Description: Relax dependencies
Forwarded: not-needed
Last-Update: 2023-06-20
---
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,9 +17,9 @@
 description = "Newtype wrapper providing a convenient representation of four-character-code values"
 license = "MIT/Apache-2.0"
 repository = "https://github.com/dholroyd/four-cc"
-[dependencies.schemars]
-version = "0.8.8"
-optional = true
+#[dependencies.schemars]
+#version = "0.8.8"
+#optional = true
 
 [dependencies.serde]
 version = "1.0"
