Index: thread-local/Cargo.toml
===================================================================
--- thread-local.orig/Cargo.toml
+++ thread-local/Cargo.toml
@@ -22,13 +22,13 @@ keywords = ["thread_local", "concurrent"
 license = "Apache-2.0/MIT"
 repository = "https://github.com/Amanieu/thread_local-rs"
 
-[[bench]]
-name = "thread_local"
-harness = false
-required-features = ["criterion"]
-[dependencies.criterion]
-version = "0.3.3"
-optional = true
+#[[bench]]
+#name = "thread_local"
+#harness = false
+#required-features = ["criterion"]
+#[dependencies.criterion]
+#version = "0.3.3"
+#optional = true
 
 [dependencies.once_cell]
 version = "1.5.2"
