--- a/Cargo.toml
+++ b/Cargo.toml
@@ -123,22 +123,6 @@
     "ring",
 ]
 
-[target.'cfg(all(target_family = "wasm", target_os = "unknown"))'.dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-default-features = false
-
-[target.'cfg(all(target_family = "wasm", target_os = "unknown"))'.dependencies.ring]
-version = "0.17"
-features = ["wasm32_unknown_unknown_js"]
-
-[target.'cfg(all(target_family = "wasm", target_os = "unknown"))'.dependencies.rustls-pki-types]
-version = "1.7"
-features = ["web"]
-
-[target.'cfg(all(target_family = "wasm", target_os = "unknown"))'.dependencies.web-time]
-version = "1"
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
