Index: quinn-proto/Cargo.toml
===================================================================
--- quinn-proto.orig/Cargo.toml
+++ quinn-proto/Cargo.toml
@@ -61,7 +61,7 @@ optional = true
 default-features = false
 
 [dependencies.rustls-platform-verifier]
-version = "0.4"
+version = "0.6"
 optional = true
 
 [dependencies.slab]
@@ -92,12 +92,11 @@ version = "0.4"
 version = "1"
 
 [dev-dependencies.rcgen]
-version = "0.13"
+version = "0.14"
 
 [dev-dependencies.tracing-subscriber]
 version = "0.3.0"
 features = [
-    "env-filter",
     "fmt",
     "ansi",
     "time",
@@ -106,7 +105,7 @@ features = [
 default-features = false
 
 [dev-dependencies.wasm-bindgen-test]
-version = "0.3.45"
+version = "0.3.37"
 
 [features]
 default = [
Index: quinn-proto/src/tests/mod.rs
===================================================================
--- quinn-proto.orig/src/tests/mod.rs
+++ quinn-proto/src/tests/mod.rs
@@ -467,7 +467,7 @@ fn reject_missing_client_cert() {
     // because it's convenient.
     store.add(CERTIFIED_KEY.cert.der().clone()).unwrap();
 
-    let key = PrivatePkcs8KeyDer::from(CERTIFIED_KEY.key_pair.serialize_der());
+    let key = PrivatePkcs8KeyDer::from(CERTIFIED_KEY.signing_key.serialize_der());
     let cert = CERTIFIED_KEY.cert.der().clone();
 
     let provider = Arc::new(default_provider());
@@ -2144,7 +2144,7 @@ fn big_cert_and_key() -> (CertificateDer
 
     (
         cert.cert.into(),
-        PrivateKeyDer::Pkcs8(cert.key_pair.serialize_der().into()),
+        PrivateKeyDer::Pkcs8(cert.signing_key.serialize_der().into()),
     )
 }
 
Index: quinn-proto/src/tests/util.rs
===================================================================
--- quinn-proto.orig/src/tests/util.rs
+++ quinn-proto/src/tests/util.rs
@@ -596,7 +596,7 @@ fn server_crypto_inner(
     let (cert, key) = identity.unwrap_or_else(|| {
         (
             CERTIFIED_KEY.cert.der().clone(),
-            PrivateKeyDer::Pkcs8(CERTIFIED_KEY.key_pair.serialize_der().into()),
+            PrivateKeyDer::Pkcs8(CERTIFIED_KEY.signing_key.serialize_der().into()),
         )
     });
 
@@ -714,6 +714,6 @@ fn set_congestion_experienced(
 lazy_static! {
     pub static ref SERVER_PORTS: Mutex<RangeFrom<u16>> = Mutex::new(4433..);
     pub static ref CLIENT_PORTS: Mutex<RangeFrom<u16>> = Mutex::new(44433..);
-    pub(crate) static ref CERTIFIED_KEY: rcgen::CertifiedKey =
+    pub(crate) static ref CERTIFIED_KEY: rcgen::CertifiedKey<rcgen::KeyPair> =
         rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
 }
Index: quinn-proto/src/config.rs
===================================================================
--- quinn-proto.orig/src/config.rs
+++ quinn-proto/src/config.rs
@@ -1061,7 +1061,7 @@ impl ClientConfig {
     #[cfg(feature = "platform-verifier")]
     pub fn with_platform_verifier() -> Self {
         Self::new(Arc::new(crypto::rustls::QuicClientConfig::new(Arc::new(
-            rustls_platform_verifier::Verifier::new(),
+            rustls_platform_verifier::Verifier::new(Arc::new(rustls::crypto::ring::default_provider())).expect("Failed to load TLS certificates"),
         ))))
     }
 
