Index: gping/Cargo.toml
===================================================================
--- gping.orig/Cargo.toml
+++ gping/Cargo.toml
@@ -21,35 +21,35 @@ license = "MIT"
 repository = "https://github.com/orf/gping"
 
 [dependencies.anyhow]
-version = "1.0.75"
+version = "1.0"
 
 [dependencies.chrono]
 version = "0.4.31"
 
 [dependencies.clap]
-version = "4.4.7"
+version = "4.4.6"
 features = ["derive"]
 
 [dependencies.const_format]
-version = "0.2.32"
+version = "0.2.30"
 
 [dependencies.crossterm]
-version = "0.27.0"
+version = "0.25.0"
 
 [dependencies.dns-lookup]
-version = "2.0.0"
+version = "1"
 
 [dependencies.itertools]
-version = "0.11.0"
+version = "0.10"
 
 [dependencies.pinger]
-version = "^1.0.0"
+version = "0.13"
 
 [dependencies.shadow-rs]
 version = "0.24.1"
 
 [dependencies.tui]
-version = "0.24.0"
+version = "0.22"
 features = ["crossterm"]
 default_features = false
 package = "ratatui"
