--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,32 +27,32 @@
 version = "0.4.34"
 
 [dependencies.clap]
-version = "4.5.0"
+version = "4.4.0"
 features = ["derive"]
 
 [dependencies.const_format]
-version = "0.2.32"
+version = "0.2.30"
 
 [dependencies.crossterm]
 version = "0.27.0"
 
 [dependencies.dns-lookup]
-version = "2.0.0"
+version = "1"
 
 [dependencies.itertools]
-version = "0.12.1"
+version = ">= 0.10.1, < 1.0"
 
 [dependencies.pinger]
 version = "^1.1.1"
 
 [dependencies.shadow-rs]
-version = "0.26.1"
+version = "0.24.1"
 
 [dependencies.tui]
-version = "0.24.0"
+version = "0.23.0"
 features = ["crossterm"]
 default_features = false
 package = "ratatui"
 
 [build-dependencies.shadow-rs]
-version = "0.26.1"
+version = "0.24.1"
