Index: gping/Cargo.toml
===================================================================
--- gping.orig/Cargo.toml
+++ gping/Cargo.toml
@@ -21,3 +21,3 @@
 [dependencies.clap]
-version = "4.4.7"
+version = "4.4.6"
 features = ["derive"]
@@ -30,3 +30,3 @@
 [dependencies.const_format]
-version = "0.2.32"
+version = "0.2.30"
 
@@ -40,3 +40,3 @@
 [dependencies.dns-lookup]
-version = "2.0.0"
+version = "1"
 
@@ -50,3 +50,3 @@
 [dependencies.itertools]
-version = "0.11.0"
+version = "0.12"
 
@@ -60,3 +60,3 @@
 [dependencies.pinger]
-version = "^1.0.0"
+version = "0.13"
 
@@ -70,3 +70,3 @@
 [dependencies.tui]
-version = "0.24.0"
+version = "0.23"
 features = ["crossterm"]
