--- a/Cargo.toml
+++ b/Cargo.toml
@@ -103,5 +103,5 @@
-[target.'cfg(any(target_os = "ios", target_os = "tvos", target_os = "visionos"))'.dependencies.objc2]
+[disabled.target.'cfg(any(target_os = "ios", target_os = "tvos", target_os = "visionos"))'.dependencies.objc2]
 version = "0.6"
 
-[target.'cfg(any(target_os = "ios", target_os = "tvos", target_os = "visionos"))'.dependencies.objc2-foundation]
+[disabled.target.'cfg(any(target_os = "ios", target_os = "tvos", target_os = "visionos"))'.dependencies.objc2-foundation]
 version = "0.3"
@@ -116,15 +116,15 @@
-[target.'cfg(target_family = "wasm")'.dependencies.web-sys]
+[disabled.target.'cfg(target_family = "wasm")'.dependencies.web-sys]
 version = "0.3"
 features = ["Window"]
 
-[target.'cfg(target_os = "android")'.dependencies.jni]
+[disabled.target.'cfg(target_os = "android")'.dependencies.jni]
 version = "0.21"
 
-[target.'cfg(target_os = "android")'.dependencies.ndk-context]
+[disabled.target.'cfg(target_os = "android")'.dependencies.ndk-context]
 version = "0.1"
 
-[target.'cfg(target_os = "android")'.dev-dependencies.ndk-glue]
+[disabled.target.'cfg(target_os = "android")'.dev-dependencies.ndk-glue]
 version = ">= 0.3, <= 0.7"
 
-[target.'cfg(target_os = "macos")'.dependencies.core-foundation]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.core-foundation]
 version = "0.10"
