--- a/Cargo.toml
+++ b/Cargo.toml
@@ -97,11 +97,6 @@
 [dependencies.pin-project-lite]
 version = "0.2"
 
-[dependencies.real-async-native-tls]
-version = "0.5.0"
-optional = true
-package = "async-native-tls"
-
 [dependencies.real-async-tls]
 version = "0.13"
 features = ["client"]
@@ -125,17 +120,17 @@
 package = "tokio-openssl"
 
 [dependencies.real-tokio-rustls]
-version = "0.26"
+version = "0.24"
 optional = true
 default-features = false
 package = "tokio-rustls"
 
 [dependencies.rustls-native-certs]
-version = "0.8"
+version = "0.6"
 optional = true
 
 [dependencies.rustls-pki-types]
-version = "1.0.1"
+version = "1.10"
 optional = true
 
 [dependencies.tokio]
@@ -148,10 +143,6 @@
 version = "0.24"
 default-features = false
 
-[dependencies.webpki-roots]
-version = "0.26"
-optional = true
-
 [dev-dependencies.async-std]
 version = "1.0"
 features = [
@@ -160,7 +151,7 @@
 ]
 
 [dev-dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 
 [dev-dependencies.futures]
 version = "0.3"
@@ -168,21 +159,14 @@
 [dev-dependencies.futures-channel]
 version = "0.3"
 
-[dev-dependencies.http-body-util]
-version = "0.1"
-
 [dev-dependencies.hyper]
-version = "1.0"
+version = "0.14"
 features = [
     "http1",
     "server",
 ]
 default-features = false
 
-[dev-dependencies.hyper-util]
-version = "0.1"
-features = ["tokio"]
-
 [dev-dependencies.tokio]
 version = "1.0"
 features = ["full"]
@@ -201,11 +185,6 @@
     "rustls-pki-types",
     "tungstenite/__rustls-tls",
 ]
-async-native-tls = [
-    "async-std-runtime",
-    "real-async-native-tls",
-    "tungstenite/native-tls",
-]
 async-std-runtime = [
     "async-std",
     "handshake",
@@ -221,12 +200,6 @@
     "handshake",
 ]
 handshake = ["tungstenite/handshake"]
-tokio-native-tls = [
-    "tokio-runtime",
-    "real-tokio-native-tls",
-    "real-native-tls",
-    "tungstenite/native-tls",
-]
 tokio-openssl = [
     "tokio-runtime",
     "real-tokio-openssl",
@@ -241,9 +214,5 @@
     "__rustls-tls",
     "rustls-native-certs",
 ]
-tokio-rustls-webpki-roots = [
-    "__rustls-tls",
-    "webpki-roots",
-]
 url = ["tungstenite/url"]
 verbose-logging = []
