--- a/Cargo.toml
+++ b/Cargo.toml
@@ -125,3 +120,3 @@
 [dependencies.rustls-native-certs]
-version = "0.8"
+version = "0.6"
 optional = true
@@ -148,10 +143,6 @@
 version = "0.24"
 default-features = false
 
-[dependencies.webpki-roots]
-version = "0.26"
-optional = true
-
 [dev-dependencies.async-std]
 version = "1.0"
 features = [
@@ -160,7 +151,7 @@
 ]
 
 [dev-dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 
 [dev-dependencies.futures]
 version = "0.3"
@@ -241,9 +214,5 @@
     "__rustls-tls",
     "rustls-native-certs",
 ]
-tokio-rustls-webpki-roots = [
-    "__rustls-tls",
-    "webpki-roots",
-]
 url = ["tungstenite/url"]
 verbose-logging = []
