--- a/Cargo.toml
+++ b/Cargo.toml
@@ -103,11 +103,11 @@
 default-features = false
 
 [dependencies.gio]
-version = "0.20"
+version = "0.21"
 optional = true
 
 [dependencies.glib]
-version = "0.20"
+version = "0.21"
 optional = true
 
 [dependencies.log]
@@ -171,10 +171,6 @@
 version = "0.24"
 default-features = false
 
-[dependencies.webpki-roots]
-version = "0.26"
-optional = true
-
 [dev-dependencies.async-std]
 version = "1.0"
 features = [
@@ -183,7 +183,7 @@
 ]
 
 [dev-dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 
 [dev-dependencies.futures]
 version = "0.3"
@@ -268,9 +268,5 @@
     "__rustls-tls",
     "rustls-native-certs",
 ]
-tokio-rustls-webpki-roots = [
-    "__rustls-tls",
-    "webpki-roots",
-]
 url = ["tungstenite/url"]
 verbose-logging = []
