Index: async-tungstenite/Cargo.toml
===================================================================
--- async-tungstenite.orig/Cargo.toml
+++ async-tungstenite/Cargo.toml
@@ -154,7 +154,7 @@ default-features = false
 package = "tokio-rustls"
 
 [dependencies.rustls-native-certs]
-version = "0.8"
+version = "0.6"
 optional = true
 
 [dependencies.rustls-pki-types]
Index: async-tungstenite/src/tokio/rustls.rs
===================================================================
--- async-tungstenite.orig/src/tokio/rustls.rs
+++ async-tungstenite/src/tokio/rustls.rs
@@ -48,14 +48,14 @@ where
 
                     #[cfg(feature = "tokio-rustls-native-certs")]
                     {
-                        let mut native_certs = rustls_native_certs::load_native_certs();
-                        if let Some(err) = native_certs.errors.drain(..).next() {
+                        let native_certs = rustls_native_certs::load_native_certs();
+                        if let Err(err) = native_certs {
                             return Err(std::io::Error::new(std::io::ErrorKind::Other, err).into());
                         }
-                        let native_certs = native_certs.certs;
+                        let native_certs = native_certs.unwrap();
                         let total_number = native_certs.len();
                         let (number_added, number_ignored) =
-                            root_store.add_parsable_certificates(native_certs);
+                            root_store.add_parsable_certificates(native_certs.into_iter().map(|cert| cert.0.into()));
                         log::debug!("Added {number_added}/{total_number} native root certificates (ignored {number_ignored})");
                     }
                     #[cfg(all(
