From: Maytham Alsudany <maytha8thedev@gmail.com>
Forwarded: no
Description: Use tungstenite 0.26

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -156,7 +156,7 @@
 default-features = false
 
 [dependencies.tungstenite]
-version = "0.24"
+version = "0.26"
 default-features = false
 
 [dev-dependencies.async-std]
@@ -191,7 +191,7 @@
 features = ["full"]
 
 [dev-dependencies.tungstenite]
-version = "0.24"
+version = "0.26"
 features = ["url"]
 
 [dev-dependencies.url]
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -312,11 +312,11 @@
     }
 
     /// Close the underlying web socket
-    pub async fn close(&mut self, msg: Option<CloseFrame<'_>>) -> Result<(), WsError>
+    pub async fn close(&mut self, msg: Option<CloseFrame>) -> Result<(), WsError>
     where
         S: AsyncRead + AsyncWrite + Unpin,
     {
-        let msg = msg.map(|msg| msg.into_owned());
+        let msg = msg.map(|msg| msg.to_owned());
         self.send(Message::Close(msg)).await
     }
 }
