Index: rust-prometheus-client-0.22.2/Cargo.toml
===================================================================
--- rust-prometheus-client-0.22.2.orig/Cargo.toml
+++ rust-prometheus-client-0.22.2/Cargo.toml
@@ -90,7 +90,7 @@ features = [
 ]
 
 [dev-dependencies.pyo3]
-version = "0.20"
+version = "0.22"
 
 [dev-dependencies.quickcheck]
 version = "1"
Index: rust-prometheus-client-0.22.2/src/encoding/text.rs
===================================================================
--- rust-prometheus-client-0.22.2.orig/src/encoding/text.rs
+++ rust-prometheus-client-0.22.2/src/encoding/text.rs
@@ -920,7 +920,7 @@ mod tests {
 
         println!("{:?}", input);
         Python::with_gil(|py| {
-            let parser = PyModule::from_code(
+            let parser = PyModule::from_code_bound(
                 py,
                 r#"
 from prometheus_client.openmetrics.parser import text_string_to_metric_families
