Index: prometheus-client/Cargo.toml
===================================================================
--- prometheus-client.orig/Cargo.toml
+++ prometheus-client/Cargo.toml
@@ -61,11 +61,11 @@ version = "0.12"
 version = "0.4.1"
 
 [dependencies.prost]
-version = "0.11.0"
+version = "0.12"
 optional = true
 
 [dependencies.prost-types]
-version = "0.11.0"
+version = "0.12"
 optional = true
 
 [dev-dependencies.actix-web]
@@ -111,7 +111,7 @@ features = [
 ]
 
 [build-dependencies.prost-build]
-version = "0.11.0"
+version = "0.12"
 optional = true
 
 [features]
