Index: rust-prometheus-client-0.22.2/Cargo.toml
===================================================================
--- rust-prometheus-client-0.22.2.orig/Cargo.toml
+++ rust-prometheus-client-0.22.2/Cargo.toml
@@ -61,16 +61,13 @@ version = "0.12"
 version = "0.4.1"
 
 [dependencies.prost]
-version = "0.11.0"
+version = "0.12"
 optional = true
 
 [dependencies.prost-types]
-version = "0.11.0"
+version = "0.12"
 optional = true
 
-[dev-dependencies.actix-web]
-version = "4"
-
 [dev-dependencies.async-std]
 version = "1"
 features = ["attributes"]
@@ -78,11 +75,0 @@
-[dev-dependencies.http-types]
-version = "2"
-
-[dev-dependencies.hyper]
-version = "0.14.16"
-features = [
-    "server",
-    "http1",
-    "tcp",
-]
-
@@ -98,9 +84,6 @@ version = "1"
 [dev-dependencies.rand]
 version = "0.8.4"
 
-[dev-dependencies.tide]
-version = "0.16"
-
 [dev-dependencies.tokio]
 version = "1"
 features = [
@@ -111,7 +94,7 @@ features = [
 ]
 
 [build-dependencies.prost-build]
-version = "0.11.0"
+version = "0.12"
 optional = true
 
 [features]
