adapted from 46462cc99c24e9eb5d4abbcb3abc6e459e2a836b upstream

Index: kuznyechik/src/sse2/backends.rs
===================================================================
--- kuznyechik.orig/src/sse2/backends.rs
+++ kuznyechik/src/sse2/backends.rs
@@ -9,6 +9,10 @@ use cipher::{
     typenum::Unsigned,
     BlockBackend, BlockSizeUser, ParBlocks, ParBlocksSizeUser,
 };
+#[cfg(target_arch = "x86")]
+use core::arch::x86::*;
+ 
+#[cfg(target_arch = "x86_64")]
 use core::arch::x86_64::*;
 
 pub(super) type RoundKeys = [__m128i; 10];
