--- a/Cargo.toml
+++ b/Cargo.toml
@@ -185,8 +185,8 @@
 optional = true
 
 [dependencies.rand]
-version = "0.9"
-features = ["thread_rng"]
+version = "0.8"
+features = ["std_rng"]
 default-features = false
 
 [dependencies.serde]
--- a/src/file/api/mod.rs
+++ b/src/file/api/mod.rs
@@ -84,7 +84,7 @@
 impl Keyring {
     #[allow(clippy::new_without_default)]
     pub(crate) fn new() -> Self {
-        let salt = rand::rng().random::<[u8; DEFAULT_SALT_SIZE]>().to_vec();
+        let salt = rand::thread_rng().r#gen::<[u8; DEFAULT_SALT_SIZE]>().to_vec();
 
         Self {
             salt_size: salt.len() as u32,
@@ -119,8 +119,8 @@
         mtime: Option<std::time::SystemTime>,
     ) -> Result<(), Error> {
         let tmp_path = if let Some(parent) = path.as_ref().parent() {
-            let rnd: String = rand::rng()
-                .sample_iter(&rand::distr::Alphanumeric)
+            let rnd: String = rand::thread_rng()
+                .sample_iter(&rand::distributions::Alphanumeric)
                 .take(16)
                 .map(char::from)
                 .collect();
@@ -284,7 +284,7 @@
 
     // Reset Keyring content
     pub(crate) fn reset(&mut self) {
-        let salt = rand::rng().random::<[u8; DEFAULT_SALT_SIZE]>().to_vec();
+        let salt = rand::thread_rng().r#gen::<[u8; DEFAULT_SALT_SIZE]>().to_vec();
         self.salt_size = salt.len() as u32;
         self.salt = salt;
         self.iteration_count = DEFAULT_ITERATION_COUNT;
--- a/src/secret.rs
+++ b/src/secret.rs
@@ -65,7 +65,7 @@
     pub fn random() -> Result<Self, getrandom::Error> {
         let mut secret = [0; 64];
         // Equivalent of `ring::rand::SecureRandom`
-        getrandom::fill(&mut secret)?;
+        getrandom::getrandom(&mut secret)?;
 
         Ok(Self::blob(secret))
     }
