--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@
 exclude = ["org.freedesktop.Secrets.xml"]
 autolib = false
 autobins = false
-autoexamples = false
+autoexamples = true
 autotests = false
 autobenches = false
 description = "James Bond went on a new mission and this time as a Secret Service provider"
@@ -101,20 +101,6 @@
 name = "oo7"
 path = "src/lib.rs"
 
-[[example]]
-name = "basic"
-path = "examples/basic.rs"
-required-features = ["tokio"]
-
-[[example]]
-name = "basic_2"
-path = "examples/basic_2.rs"
-required-features = ["tokio"]
-
-[[example]]
-name = "dbus_service"
-path = "examples/dbus_service.rs"
-required-features = ["tokio"]
 
 [dependencies.aes]
 version = "0.8"
@@ -162,7 +148,7 @@
 version = "1.1"
 
 [dependencies.futures-lite]
-version = "2.6"
+version = "2.3"
 optional = true
 
 [dependencies.futures-util]
