Forwarded: not-needed
Last-Update: 2025-03-04
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -183,3 +183,3 @@
 [dependencies.ron]
-version = "0.8"
+version = "0.7"
 optional = true
@@ -204,3 +204,3 @@
 [dependencies.winnow]
-version = "0.7.0"
+version = "0.6"
 
@@ -215,3 +215,3 @@
 [dev-dependencies.float-cmp]
-version = "0.10"
+version = "0.9"
 
@@ -228,3 +228,3 @@
 [dev-dependencies.notify]
-version = "7.0"
+version = "6.1"
 
