Index: config/Cargo.toml
===================================================================
--- config.orig/Cargo.toml
+++ config/Cargo.toml
@@ -28,7 +28,7 @@ version = "1.0"
 version = "7.0.0"
 
 [dependencies.rust-ini]
-version = "0.18"
+version = "0.21"
 optional = true
 
 [dependencies.serde]
Index: config/tests/file_ini.rs
===================================================================
--- config.orig/tests/file_ini.rs
+++ config/tests/file_ini.rs
@@ -64,7 +64,7 @@ fn test_error_parse() {
     assert_eq!(
         res.unwrap_err().to_string(),
         format!(
-            r#"2:0 expecting "[Some('='), Some(':')]" but found EOF. in {}"#,
+            r#"3:1 expecting "[Some('='), Some(':')]" but found EOF. in {}"#,
             path.display()
         )
     );
