Index: cpp-macros/Cargo.toml
===================================================================
--- cpp-macros.orig/Cargo.toml
+++ cpp-macros/Cargo.toml
@@ -37,7 +37,7 @@ features = ["docs-only"]
 proc-macro = true
 
 [dependencies.aho-corasick]
-version = "0.7.15"
+version = "1.0"
 
 [dependencies.byteorder]
 version = "1.0"
Index: cpp-macros/src/lib.rs
===================================================================
--- cpp-macros.orig/src/lib.rs
+++ cpp-macros/src/lib.rs
@@ -70,7 +70,7 @@ fn read_metadata(file: File) -> io::Resu
     let mut file = BufReader::new(file);
     let end = {
         const AUTO_KEYWORD: &[&[u8]] = &[&cpp_common::STRUCT_METADATA_MAGIC];
-        let aut = aho_corasick::AhoCorasick::new(AUTO_KEYWORD);
+        let aut = aho_corasick::AhoCorasick::new(AUTO_KEYWORD).unwrap();
         let found = aut.stream_find_iter(&mut file).next().expect(
             r#"
 -- rust-cpp fatal error --
