Index: ruma-common/Cargo.toml
===================================================================
--- ruma-common.orig/Cargo.toml
+++ ruma-common/Cargo.toml
@@ -39,13 +39,13 @@ harness = false
 required-features = ["criterion"]
 
 [dependencies.base64]
-version = "0.13.0"
+version = "0.21.0"
 
 [dependencies.bytes]
 version = "1.0.1"
 
 [dependencies.criterion]
-version = "0.3.3"
+version = "0.3.6"
 optional = true
 
 [dependencies.form_urlencoded]
@@ -56,7 +56,7 @@ version = "0.2.6"
 optional = true
 
 [dependencies.html5ever]
-version = "0.25.2"
+version = "0.26.0"
 optional = true
 
 [dependencies.http]
@@ -149,7 +149,7 @@ version = "0.2.2"
 version = "1.0.2"
 
 [dev-dependencies.trybuild]
-version = "1.0.42"
+version = "1.0.71"
 
 [features]
 api = [
@@ -167,11 +167,6 @@ default = [
     "server",
 ]
 events = ["dep:thiserror"]
-js = [
-    "dep:js-sys",
-    "getrandom?/js",
-    "uuid?/js",
-]
 markdown = ["pulldown-cmark"]
 rand = [
     "dep:rand_crate",
@@ -207,6 +202,3 @@ unstable-sanitize = [
     "dep:phf",
 ]
 
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.js-sys]
-version = "0.3"
-optional = true
Index: ruma-common/src/serde/base64.rs
===================================================================
--- ruma-common.orig/src/serde/base64.rs
+++ ruma-common/src/serde/base64.rs
@@ -2,6 +2,10 @@
 
 use std::{fmt, marker::PhantomData};
 
+use base64::{
+    engine::{general_purpose, GeneralPurpose, GeneralPurposeConfig},
+    Engine,
+};
 use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 
 /// A wrapper around `B` (usually `Vec<u8>`) that (de)serializes from / to a base64 string.
@@ -25,7 +29,7 @@ pub trait Base64Config {
 }
 
 #[doc(hidden)]
-pub struct Conf(base64::Config);
+pub struct Conf(base64::alphabet::Alphabet);
 
 /// Standard base64 character set without padding.
 ///
@@ -36,8 +40,7 @@ pub struct Conf(base64::Config);
 pub struct Standard;
 
 impl Base64Config for Standard {
-    // See https://github.com/matrix-org/matrix-spec/issues/838
-    const CONF: Conf = Conf(base64::STANDARD_NO_PAD.decode_allow_trailing_bits(true));
+    const CONF: Conf = Conf(base64::alphabet::STANDARD);
 }
 
 /// Url-safe base64 character set without padding.
@@ -49,7 +52,14 @@ impl Base64Config for Standard {
 pub struct UrlSafe;
 
 impl Base64Config for UrlSafe {
-    const CONF: Conf = Conf(base64::URL_SAFE_NO_PAD.decode_allow_trailing_bits(true));
+    const CONF: Conf = Conf(base64::alphabet::URL_SAFE);
+}
+
+impl<C: Base64Config, B> Base64<C, B> {
+    // See https://github.com/matrix-org/matrix-spec/issues/838
+    const CONFIG: GeneralPurposeConfig =
+        general_purpose::NO_PAD.with_decode_allow_trailing_bits(true);
+    const ENGINE: GeneralPurpose = GeneralPurpose::new(&C::CONF.0, Self::CONFIG);
 }
 
 impl<C: Base64Config, B: AsRef<[u8]>> Base64<C, B> {
@@ -65,7 +75,7 @@ impl<C: Base64Config, B: AsRef<[u8]>> Ba
 
     /// Encode the bytes contained in this `Base64` instance to unpadded base64.
     pub fn encode(&self) -> String {
-        base64::encode_config(&self.bytes, C::CONF.0)
+        Self::ENGINE.encode(self.as_bytes())
     }
 }
 
@@ -84,7 +94,7 @@ impl<C: Base64Config> Base64<C> {
 
     /// Parse some base64-encoded data to create a `Base64` instance.
     pub fn parse(encoded: impl AsRef<[u8]>) -> Result<Self, Base64DecodeError> {
-        base64::decode_config(encoded, C::CONF.0).map(Self::new).map_err(Base64DecodeError)
+        Self::ENGINE.decode(encoded).map(Self::new).map_err(Base64DecodeError)
     }
 }
 
