--- a/src/lib.rs
+++ b/src/lib.rs
@@ -415,6 +415,7 @@
         target_os = "solaris",
     ))]
     #[test]
+    #[ignore]
     fn test_no_ulimit() -> Result<()> {
         // This test depends on the dev environment being run uncontained.
         let info = sys_info::mem_info()?;
@@ -425,6 +426,7 @@
     }
 
     #[test]
+    #[ignore]
     fn test_ulimit() -> Result<()> {
         // Page size rounding
         let limit = read_test_process(Some(99_999_744))?;
