Index: gix/Cargo.toml
===================================================================
--- gix.orig/Cargo.toml
+++ gix/Cargo.toml
@@ -326,22 +326,6 @@ blocking-http-transport-curl = [
     "blocking-network-client",
     "gix-transport/http-client-curl",
 ]
-blocking-http-transport-reqwest = [
-    "blocking-network-client",
-    "gix-transport/http-client-reqwest",
-]
-blocking-http-transport-reqwest-native-tls = [
-    "blocking-http-transport-reqwest",
-    "gix-transport/http-client-reqwest-native-tls",
-]
-blocking-http-transport-reqwest-rust-tls = [
-    "blocking-http-transport-reqwest",
-    "gix-transport/http-client-reqwest-rust-tls",
-]
-blocking-http-transport-reqwest-rust-tls-trust-dns = [
-    "blocking-http-transport-reqwest",
-    "gix-transport/http-client-reqwest-rust-tls-trust-dns",
-]
 blocking-network-client = [
     "gix-protocol/blocking-client",
     "gix-pack/streaming-input",
@@ -425,7 +409,7 @@ pack-cache-lru-dynamic = ["gix-pack/pack
 pack-cache-lru-static = ["gix-pack/pack-cache-lru-static"]
 parallel = ["gix-features/parallel"]
 parallel-walkdir = ["gix-features/fs-walkdir-parallel"]
-progress-tree = ["prodash/progress-tree"]
+progress-tree = ["dep:prodash", "prodash?/progress-tree"]
 revision = [
     "gix-revision/describe",
     "gix-revision/merge_base",
