diff --git a/Cargo.toml b/Cargo.toml
index 5a15fb7..594db7a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -330,18 +330,6 @@ blocking-http-transport-reqwest = [
     "blocking-network-client",
     "gix-transport/http-client-reqwest",
 ]
-blocking-http-transport-reqwest-native-tls = [
-    "blocking-http-transport-reqwest",
-    "gix-transport/http-client-reqwest-native-tls",
-]
-blocking-http-transport-reqwest-rust-tls = [
-    "blocking-http-transport-reqwest",
-    "gix-transport/http-client-reqwest-rust-tls",
-]
-blocking-http-transport-reqwest-rust-tls-trust-dns = [
-    "blocking-http-transport-reqwest",
-    "gix-transport/http-client-reqwest-rust-tls-trust-dns",
-]
 blocking-network-client = [
     "gix-protocol/blocking-client",
     "gix-pack/streaming-input",
@@ -425,7 +413,7 @@ pack-cache-lru-dynamic = ["gix-pack/pack-cache-lru-dynamic"]
 pack-cache-lru-static = ["gix-pack/pack-cache-lru-static"]
 parallel = ["gix-features/parallel"]
 parallel-walkdir = ["gix-features/fs-walkdir-parallel"]
-progress-tree = ["prodash/progress-tree"]
+progress-tree = ["dep:prodash", "prodash?/progress-tree"]
 revision = [
     "gix-revision/describe",
     "gix-revision/merge_base",
