--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,11 +72,11 @@ version = "0.1"
 [target.'cfg(not(any(target_os = "windows", target_vendor = "apple")))'.dependencies.openssl-sys]
 version = "0.9.81"
 
-[target.'cfg(target_os = "macos")'.dependencies.tempfile]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.tempfile]
 version = "3.1.0"
 
-[target.'cfg(target_os = "windows")'.dependencies.schannel]
+[disabled.target.'cfg(target_os = "windows")'.dependencies.schannel]
 version = "0.1.17"
 
-[target.'cfg(target_vendor = "apple")'.dependencies.libc]
+[disabled.target.'cfg(target_vendor = "apple")'.dependencies.libc]
 version = "0.2"
