Index: native-tls/src/test.rs
===================================================================
--- native-tls.orig/src/test.rs
+++ native-tls/src/test.rs
@@ -19,7 +19,6 @@ macro_rules! p {
 mod tests {
     use super::*;
 
-    #[test]
     fn connect_google() {
         let builder = p!(TlsConnector::new());
         let s = p!(TcpStream::connect("google.com:443"));
@@ -34,14 +33,12 @@ mod tests {
         assert!(result.ends_with(b"</HTML>\r\n") || result.ends_with(b"</html>"));
     }
 
-    #[test]
     fn connect_bad_hostname() {
         let builder = p!(TlsConnector::new());
         let s = p!(TcpStream::connect("google.com:443"));
         builder.connect("goggle.com", s).unwrap_err();
     }
 
-    #[test]
     fn connect_bad_hostname_ignored() {
         let builder = p!(TlsConnector::builder()
             .danger_accept_invalid_hostnames(true)
