Index: native-tls/Cargo.toml
===================================================================
--- native-tls.orig/Cargo.toml
+++ native-tls/Cargo.toml
@@ -49,13 +49,13 @@ name = "simple-server-pkcs8"
 path = "examples/simple-server-pkcs8.rs"
 
 [dev-dependencies.pem]
-version = "1.0"
+version = "3.0"
 
 [dev-dependencies.hex]
 version = "0.4"
 
 [dev-dependencies.base64]
-version = "0.21"
+version = "0.22"
 
 [features]
 alpn = []
