Index: xkb/src/key.rs
===================================================================
--- xkb.orig/src/key.rs
+++ xkb/src/key.rs
@@ -270,10 +270,10 @@ pub const dead_belowdiaeresis: Keysym =
 pub const dead_invertedbreve: Keysym = Keysym(XKB_KEY_dead_invertedbreve);
 pub const dead_belowcomma: Keysym = Keysym(XKB_KEY_dead_belowcomma);
 pub const dead_currency: Keysym = Keysym(XKB_KEY_dead_currency);
-pub const dead_lowline: Keysym = Keysym(XKB_KEY_dead_lowline);
-pub const dead_aboveverticalline: Keysym = Keysym(XKB_KEY_dead_aboveverticalline);
-pub const dead_belowverticalline: Keysym = Keysym(XKB_KEY_dead_belowverticalline);
-pub const dead_longsolidusoverlay: Keysym = Keysym(XKB_KEY_dead_longsolidusoverlay);
+//pub const dead_lowline: Keysym = Keysym(XKB_KEY_dead_lowline);
+//pub const dead_aboveverticalline: Keysym = Keysym(XKB_KEY_dead_aboveverticalline);
+//pub const dead_belowverticalline: Keysym = Keysym(XKB_KEY_dead_belowverticalline);
+//pub const dead_longsolidusoverlay: Keysym = Keysym(XKB_KEY_dead_longsolidusoverlay);
 pub const dead_a: Keysym = Keysym(XKB_KEY_dead_a);
 pub const dead_A: Keysym = Keysym(XKB_KEY_dead_A);
 pub const dead_e: Keysym = Keysym(XKB_KEY_dead_e);
