--- a/src/lib.rs
+++ b/src/lib.rs
@@ -313,6 +313,7 @@
     fn test_testdir_current_rnd_dir() {
         let path;
         {
+            std::env::set_current_dir("/tmp").unwrap();
             let dir = TestDir::current_rnd();
 
             // Dir created
@@ -332,6 +333,7 @@
     fn test_testdir_current_dir() {
         let path;
         {
+            std::env::set_current_dir("/tmp").unwrap();
             let dir = TestDir::current("a/b/c");
 
             // Dir created
