diff --git a/Cargo.toml b/Cargo.toml
index fdf59a4..90742df 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,25 +32,25 @@ name = "github_actions_expressions"
 path = "src/lib.rs"
 
 [dependencies.anyhow]
-version = "1.0.99"
+version = "1"
 
 [dependencies.itertools]
-version = "0.14.0"
+version = "0.13"
 
 [dependencies.pest]
-version = "2.8.1"
+version = "2"
 
 [dependencies.pest_derive]
-version = "2.8.1"
+version = "2"
 
 [dependencies.serde_json]
-version = "1.0.143"
+version = "1"
 
 [dependencies.subfeature]
 version = "0.0.3"
 
 [dev-dependencies.pretty_assertions]
-version = "1.4.1"
+version = "1"
 
 [lints.clippy]
 dbg_macro = "warn"
