--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,17 +58,17 @@
 version = "2.0"
 
 [dependencies.gdk]
-version = "0.9"
+version = "0.10"
 package = "gdk4"
 
 [dependencies.glib]
-version = "0.20"
+version = "0.21"
 
 [dependencies.glib-sys]
-version = "0.20"
+version = "0.21"
 
 [dependencies.gtk]
-version = "0.9"
+version = "0.10"
 package = "gtk4"
 
 [dependencies.gtk4-layer-shell-sys]
@@ -78,15 +78,15 @@
 version = "0.2"
 
 [dev-dependencies.gio]
-version = "0.20"
+version = "0.21"
 
 [dev-dependencies.libadwaita]
-version = "0.7"
+version = "0.8"
 
 [features]
 
 [target."cfg(docsrs)".dependencies.gdk]
-version = "0.9"
+version = "0.10"
 features = [
     "gl",
     "v4_16",
