Index: meow-cli/Cargo.toml
===================================================================
--- meow-cli.orig/Cargo.toml
+++ meow-cli/Cargo.toml
@@ -39,17 +39,17 @@ name = "meow-cli"
 path = "src/main.rs"
 
 [dependencies.clap]
-version = "4.5.45"
+version = "4.5.23"
 features = ["derive"]
 
 [dependencies.clap-num]
 version = "1.2.0"
 
 [dependencies.glob]
-version = "0.3.3"
+version = "0.3.2"
 
 [dependencies.include_dir]
-version = "0.7.4"
+version = "0.7.3"
 features = ["glob"]
 
 [dependencies.rand]
