diff --git a/Cargo.toml b/Cargo.toml
index 698d40c..9c7fcdf 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,13 +40,13 @@ repository = "https://github.com/magiclen/zbar-rust"
 resolver = "2"
 
 [dependencies.enum-ordinalize]
-version = "3.1"
+version = "3.1.11"
 
 [dependencies.libc]
 version = "0.2.43"
 
 [dev-dependencies.image]
-version = "0.24"
+version = "0.23"
 
 [dev-dependencies.qrcode-generator]
 version = "4.1"
