From f5f1bbb8dcb9f7159a9096066a923c4f1e398fb4 Mon Sep 17 00:00:00 2001
From: Dandan Zhang <zhangdandan@loongson.cn>
Date: Wed, 26 Feb 2025 07:19:17 +0000
Subject: [PATCH] Add loongarch64 support

---
 raw/src/lib.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lib.rs b/src/lib.rs
index 6172033..4733f4c 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -57,6 +57,7 @@ cfg_if! {
         target_arch = "sparc",
         target_arch = "sparc64",
         target_arch = "aarch64",
+        target_arch = "loongarch64",
     ))] {
         pub use x32::*;
     } else if #[cfg(any(
