From 94b1277e9499dacc6e584dd7fdceb43360eceff6 Mon Sep 17 00:00:00 2001
From: Olivier Goffart <ogoffart@sixtyfps.io>
Date: Wed, 29 Dec 2021 12:22:01 +0100
Subject: [PATCH] Update gettext-rs dependency
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,7 +27,7 @@
 optional = true

 [dependencies.gettext-rs]
-version = "0.5"
+version = "0.7"
 features = ["gettext-system"]
 optional = true

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -346,7 +346,10 @@ pub mod internal {
 
     #[cfg(feature = "gettext-rs")]
     pub fn init<T: Into<Vec<u8>>>(module: &'static str, dir: T) {
-        gettextrs::bindtextdomain::<Vec<u8>>(domain_from_module(module).into(), dir.into());
+        // FIXME: change T from `Into<Vec<u8>> to `Into<PathBuf>`
+        let dir = String::from_utf8(dir.into()).unwrap();
+        // FIXME: don't ignore errors
+        let _ = gettextrs::bindtextdomain(domain_from_module(module), dir);
 
         static START: std::sync::Once = std::sync::Once::new();
         START.call_once(|| {
