Index: grep/Cargo.toml
===================================================================
--- grep.orig/Cargo.toml
+++ grep/Cargo.toml
@@ -29,7 +29,6 @@ version = "0.1.1"
 
 [dependencies.grep-pcre2]
 version = "0.1.2"
-optional = true
 
 [dependencies.grep-printer]
 version = "0.1.1"
@@ -58,5 +57,4 @@ version = "2.2.2"
 
 [features]
 avx-accel = ["grep-searcher/avx-accel"]
-pcre2 = ["grep-pcre2"]
 simd-accel = ["grep-searcher/simd-accel"]
