--- a/Cargo.toml
+++ b/Cargo.toml
@@ -104,10 +104,6 @@
     "dep:is_executable",
     "clap/unstable-ext",
 ]
-unstable-shell-tests = [
-    "dep:completest",
-    "dep:completest-pty",
-]
 
 [lib]
 name = "clap_complete"
@@ -140,13 +136,6 @@
 version = "0.7.0"
 optional = true
 
-[dependencies.completest]
-version = "0.4.2"
-optional = true
-
-[dependencies.completest-pty]
-version = "0.5.5"
-optional = true
 
 [dependencies.is_executable]
 version = "1.0.1"
