only in patch2:
Index: signal-hook/Cargo.toml
===================================================================
--- signal-hook.orig/Cargo.toml
+++ signal-hook/Cargo.toml
@@ -28,8 +28,8 @@ version = "^0.2"
 
 [dependencies.signal-hook-registry]
 version = "^1.4"
-[dev-dependencies.serial_test]
-version = "^0.5"
+#[dev-dependencies.serial_test]
+#version = "^0.5"
 [build-dependencies.cc]
 version = "^1"
 optional = true
