Index: signal-hook/Cargo.toml
===================================================================
--- signal-hook.orig/Cargo.toml
+++ signal-hook/Cargo.toml
@@ -33,9 +33,6 @@ optional = true
 [dependencies.signal-hook-registry]
 version = "~1.2"
 
-[dev-dependencies.version-sync]
-version = "~0.8"
-
 [features]
 mio-support = ["mio"]
 tokio-support = []
Index: signal-hook/tests/version.rs
===================================================================
--- signal-hook.orig/tests/version.rs
+++ signal-hook/tests/version.rs
@@ -1,4 +1,4 @@
-#[macro_use]
+/*#[macro_use]
 extern crate version_sync;
 
 #[test]
@@ -9,4 +9,4 @@ fn test_readme_deps() {
 #[test]
 fn test_html_root_url() {
     assert_html_root_url_updated!("src/lib.rs");
-}
+}*/
