diff --git a/Cargo.toml b/Cargo.toml
index 0282ac3..46376ba 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -62,16 +62,6 @@ version = "3"
 [target.'cfg(not(target_family = "wasm"))'.dependencies.home]
 version = "0.5.5"
 
-[target."cfg(windows)".dev-dependencies.known-folders]
-version = "1.3.1"
-
-[target."cfg(windows)".dev-dependencies.windows]
-version = "0.61.3"
-features = ["Win32_System_Threading"]
-
-[target."cfg(windows)".dev-dependencies.winreg]
-version = "0.55.0"
-
 [lints.clippy]
 bool_to_int_with_if = "allow"
 borrow_as_ptr = "allow"
