diff --git a/Cargo.toml b/Cargo.toml
index 46376ba..335cd73 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,7 +44,7 @@ default-features = false
 version = "^0.1.13"
 
 [dependencies.gix-validate]
-version = "^0.10.0"
+version = "0.9"
 
 [dependencies.once_cell]
 version = "1.21.3"
@@ -53,7 +53,7 @@ version = "1.21.3"
 version = "2.0.0"
 
 [dev-dependencies.serial_test]
-version = "3.1.0"
+version = ">= 2.0.0, < 4"
 default-features = false
 
 [dev-dependencies.tempfile]
