gate tests on the features they need

diff --git a/src/rr/dnssec/rdata/tsig.rs b/src/rr/dnssec/rdata/tsig.rs
index 1b1ea4d..55aac38 100644
--- a/src/rr/dnssec/rdata/tsig.rs
+++ b/src/rr/dnssec/rdata/tsig.rs
@@ -895,6 +895,7 @@ mod tests {
         assert_eq!(tbs, tbv);
     }
 
+    #[cfg(any(feature = "dnssec-ring", feature = "dnssec-openssl"))]
     #[test]
     fn test_sign_encode_id_changed() {
         let mut message = Message::new();
diff --git a/tests/dnssec_presentation_format_tests.rs b/tests/dnssec_presentation_format_tests.rs
index a8bf67f..10697a0 100644
--- a/tests/dnssec_presentation_format_tests.rs
+++ b/tests/dnssec_presentation_format_tests.rs
@@ -30,6 +30,7 @@ fn test_dnskey_display() {
     assert_eq!(result, exp_result);
 }
 
+#[cfg(any(feature = "dnssec-ring", feature = "dnssec-openssl"))]
 #[test]
 #[allow(deprecated)]
 fn test_ds_display() {
