bump ring to version packaged in Debian

diff --git a/Cargo.toml b/Cargo.toml
index dca02ad..66ffcd8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -139,7 +139,7 @@ default-features = false
 version = "0.8"
 
 [dependencies.ring]
-version = "0.16"
+version = "0.17"
 features = ["std"]
 optional = true
 
diff --git a/src/rr/dnssec/key_format.rs b/src/rr/dnssec/key_format.rs
index 6aead61..19359f6 100644
--- a/src/rr/dnssec/key_format.rs
+++ b/src/rr/dnssec/key_format.rs
@@ -91,7 +91,7 @@ impl KeyFormat {
                     } else {
                         &ECDSA_P384_SHA384_FIXED_SIGNING
                     };
-                    let key = EcdsaKeyPair::from_pkcs8(ring_algorithm, bytes)?;
+                    let key = EcdsaKeyPair::from_pkcs8(ring_algorithm, bytes, &ring::rand::SystemRandom::new())?;
 
                     Ok(KeyPair::from_ecdsa(key))
                 }
diff --git a/src/rr/dnssec/rdata/tsig.rs b/src/rr/dnssec/rdata/tsig.rs
index 1b1ea4d..4c0e3bc 100644
--- a/src/rr/dnssec/rdata/tsig.rs
+++ b/src/rr/dnssec/rdata/tsig.rs
@@ -635,9 +635,9 @@ impl TsigAlgorithm {
         use TsigAlgorithm::*;
 
         let len = match self {
-            HmacSha256 => hmac::HMAC_SHA256.digest_algorithm().output_len,
-            HmacSha384 => hmac::HMAC_SHA384.digest_algorithm().output_len,
-            HmacSha512 => hmac::HMAC_SHA512.digest_algorithm().output_len,
+            HmacSha256 => hmac::HMAC_SHA256.digest_algorithm().output_len(),
+            HmacSha384 => hmac::HMAC_SHA384.digest_algorithm().output_len(),
+            HmacSha512 => hmac::HMAC_SHA512.digest_algorithm().output_len(),
             _ => return Err(ProtoErrorKind::TsigUnsupportedMacAlgorithm(self.clone()).into()),
         };
 
