From: Maytham Alsudany <maytha8thedev@gmail.com>
Forwarded: not-needed
Subject: Use http 1.x

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,5 +70,5 @@
 
 [dependencies.h2]
-version = "0.3.0"
+version = "0.4"
 features = ["stream"]
 optional = true
@@ -80,3 +80,3 @@
 [dependencies.h3]
-version = "0.0.3"
+version = "0.0.6"
 optional = true
@@ -90,3 +90,3 @@
 [dependencies.h3-quinn]
-version = "0.0.4"
+version = "0.0.7"
 optional = true
@@ -100,3 +100,3 @@
 [dependencies.http]
-version = "0.2"
+version = "1"
 optional = true
