diff --git a/src/private/layout_test.rs b/src/private/layout_test.rs
index 35cbf22..7a743c7 100644
--- a/src/private/layout_test.rs
+++ b/src/private/layout_test.rs
@@ -20,6 +20,7 @@
 // THE SOFTWARE.
 
 use crate::private::layout::PointerReader;
+use alloc::vec::Vec;
 
 fn test_at_alignments(words: &[crate::Word], verify: &dyn Fn(PointerReader)) {
     verify(PointerReader::get_root_unchecked(words.as_ptr() as *const u8));
