Index: unicode-bidi/Cargo.toml
===================================================================
--- unicode-bidi.orig/Cargo.toml
+++ unicode-bidi/Cargo.toml
@@ -47,14 +47,6 @@ name = "conformance_tests"
 path = "tests/conformance_tests.rs"
 required-features = ["hardcoded-data"]
 
-[dependencies.flame]
-version = "0.2"
-optional = true
-
-[dependencies.flamer]
-version = "0.4"
-optional = true
-
 [dependencies.serde]
 version = ">=0.8, <2.0"
 features = ["derive"]
@@ -70,10 +62,6 @@ default = [
     "std",
     "hardcoded-data",
 ]
-flame_it = [
-    "flame",
-    "flamer",
-]
 hardcoded-data = []
 std = []
 unstable = []
