Index: unicode-bidi/Cargo.toml
===================================================================
--- unicode-bidi.orig/Cargo.toml
+++ unicode-bidi/Cargo.toml
@@ -42,14 +42,6 @@ repository = "https://github.com/servo/u
 [lib]
 name = "unicode_bidi"
 
-[dependencies.flame]
-version = "0.2"
-optional = true
-
-[dependencies.flamer]
-version = "0.4"
-optional = true
-
 [dependencies.serde]
 version = ">=0.8, <2.0"
 features = ["derive"]
@@ -65,10 +57,6 @@ default = [
     "std",
     "hardcoded-data",
 ]
-flame_it = [
-    "flame",
-    "flamer",
-]
 hardcoded-data = []
 std = []
 unstable = []
