--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,6 +31,7 @@
 ]
 license = "MIT/Apache-2.0"
 repository = "https://github.com/georust/geojson"
+doctest = false
 
 [package.metadata.docs.rs]
 all-features = true
@@ -67,23 +68,8 @@
 name = "roundtrip"
 path = "tests/roundtrip.rs"
 
-[[bench]]
-name = "parse"
-path = "benches/parse.rs"
-harness = false
-
-[[bench]]
-name = "serialize"
-path = "benches/serialize.rs"
-harness = false
-
-[[bench]]
-name = "to_geo_types"
-path = "benches/to_geo_types.rs"
-harness = false
-
 [dependencies.geo-types]
-version = "0.7.13"
+version = "0.7.11"
 features = ["serde"]
 optional = true
 
