--- a/Cargo.toml
+++ b/Cargo.toml
@@ -183,9 +183,6 @@
 version = "1.0.3"
 default-features = false
 
-[dev-dependencies.regex-test]
-version = "0.1.0"
-
 [profile.bench]
 debug = 2
 
--- a/tests/suite_string.rs
+++ b/tests/suite_string.rs
@@ -1,12 +1,12 @@
 use {
     anyhow::Result,
     regex::{Regex, RegexBuilder},
-    regex_test::{
+    /*regex_test::{
         CompiledRegex, Match, RegexTest, Span, TestResult, TestRunner,
-    },
+    },*/
 };
 
-/// Tests the default configuration of the hybrid NFA/DFA.
+/* /// Tests the default configuration of the hybrid NFA/DFA.
 #[test]
 fn default() -> Result<()> {
     let mut runner = TestRunner::new()?;
@@ -46,9 +46,9 @@
         }
         name => TestResult::fail(&format!("unrecognized test name: {name}")),
     }
-}
+} */
 
-/// Converts the given regex test to a closure that searches with a
+/* /// Converts the given regex test to a closure that searches with a
 /// `bytes::Regex`. If the test configuration is unsupported, then a
 /// `CompiledRegex` that skips the test is returned.
 fn compiler(
@@ -110,4 +110,4 @@
     // a match always gives a non-zero number of groups with the first group
     // being non-None.
     regex_test::Captures::new(0, spans).unwrap()
-}
+} */
--- a/tests/lib.rs
+++ b/tests/lib.rs
@@ -16,7 +16,7 @@
     // Nothing to blacklist yet!
 ];
 
-fn suite() -> anyhow::Result<regex_test::RegexTests> {
+/*fn suite() -> anyhow::Result<regex_test::RegexTests> {
     let _ = env_logger::try_init();
 
     let mut tests = regex_test::RegexTests::new();
@@ -55,4 +55,4 @@
     load!("fowler/repetition");
 
     Ok(tests)
-}
+}*/
--- a/tests/suite_bytes.rs
+++ b/tests/suite_bytes.rs
@@ -1,12 +1,12 @@
 use {
     anyhow::Result,
     regex::bytes::{Regex, RegexBuilder},
-    regex_test::{
+    /*regex_test::{
         CompiledRegex, Match, RegexTest, Span, TestResult, TestRunner,
-    },
+    },*/
 };
 
-/// Tests the default configuration of the hybrid NFA/DFA.
+/* /// Tests the default configuration of the hybrid NFA/DFA.
 #[test]
 fn default() -> Result<()> {
     let mut runner = TestRunner::new()?;
@@ -105,4 +105,4 @@
     // a match always gives a non-zero number of groups with the first group
     // being non-None.
     regex_test::Captures::new(0, spans).unwrap()
-}
+} */
--- a/tests/suite_bytes_set.rs
+++ b/tests/suite_bytes_set.rs
@@ -1,10 +1,10 @@
 use {
     anyhow::Result,
     regex::bytes::{RegexSet, RegexSetBuilder},
-    regex_test::{CompiledRegex, RegexTest, TestResult, TestRunner},
+    //regex_test::{CompiledRegex, RegexTest, TestResult, TestRunner},
 };
 
-/// Tests the default configuration of the hybrid NFA/DFA.
+/* /// Tests the default configuration of the hybrid NFA/DFA.
 #[test]
 fn default() -> Result<()> {
     let mut runner = TestRunner::new()?;
@@ -68,4 +68,4 @@
         .line_terminator(test.line_terminator())
         .build()?;
     Ok(CompiledRegex::compiled(move |test| run_test(&re, test)))
-}
+} */
--- a/tests/suite_string_set.rs
+++ b/tests/suite_string_set.rs
@@ -1,10 +1,10 @@
 use {
     anyhow::Result,
     regex::{RegexSet, RegexSetBuilder},
-    regex_test::{CompiledRegex, RegexTest, TestResult, TestRunner},
+    //regex_test::{CompiledRegex, RegexTest, TestResult, TestRunner},
 };
 
-/// Tests the default configuration of the hybrid NFA/DFA.
+/* /// Tests the default configuration of the hybrid NFA/DFA.
 #[test]
 fn default() -> Result<()> {
     let mut runner = TestRunner::new()?;
@@ -75,4 +75,4 @@
         .line_terminator(test.line_terminator())
         .build()?;
     Ok(CompiledRegex::compiled(move |test| run_test(&re, test)))
-}
+} */
