Index: compiler-builtins/Cargo.toml
===================================================================
--- compiler-builtins.orig/Cargo.toml
+++ compiler-builtins/Cargo.toml
@@ -27,6 +27,7 @@ documentation = "https://docs.rs/compile
 readme = "README.md"
 license = "MIT AND Apache-2.0 WITH LLVM-exception AND (MIT OR Apache-2.0)"
 repository = "https://github.com/rust-lang/compiler-builtins"
+rust-version = "1.88"
 
 [features]
 c = ["dep:cc"]
