--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,6 +39,6 @@
 default-features = false
 
 [dependencies.toml_edit]
-version = "0.22.8"
+version = ">= 0.21"
 features = ["parse"]
 default-features = false
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -95,7 +95,7 @@
     let mut contents = String::new();
     file.read_to_string(&mut contents).ok()?;
 
-    let doc = contents.parse::<toml_edit::DocumentMut>().ok()?;
+    let doc = contents.parse::<toml_edit::Document>().ok()?;
     let release_date = doc
         .get("package")
         .and_then(|package| package.as_table_like())

