Index: laurel/Cargo.toml
===================================================================
--- laurel.orig/Cargo.toml
+++ laurel/Cargo.toml
@@ -33,10 +33,6 @@ license = "GPL-3.0-or-later"
 repository = "https://github.com/threathunters-io/laurel"
 autobenches = false
 
-[[bench]]
-name = "parse"
-harness = false
-
 [dependencies.anyhow]
 version = "1"
 
@@ -105,9 +101,6 @@ version = "0.5"
 [dev-dependencies.divan]
 version = "0.1.13"
 
-[dev-dependencies.gperftools]
-version = "0"
-
 [build-dependencies.bindgen]
 version = ">= 0.60"
 
