From 220c8829a9837934082c2578e3ae3085a1a5443e Mon Sep 17 00:00:00 2001
From: Wolfgang Silbermayr <wolfgang@silbermayr.at>
Date: Mon, 14 Jan 2019 21:22:56 +0100
Subject: [PATCH] Update nix dependency to 0.13

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index f799d15..b24c632 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,7 +29,7 @@ harness = false
 [features]
 termination = []
 [target."cfg(unix)".dependencies.nix]
-version = "0.11"
+version = "0.13"
 [target."cfg(windows)".dependencies.winapi]
 version = "0.3"
 features = ["consoleapi", "handleapi", "synchapi", "winbase"]
-- 
2.20.1

