Index: ctrlc/Cargo.toml
===================================================================
--- ctrlc.orig/Cargo.toml
+++ ctrlc/Cargo.toml
@@ -32,8 +32,8 @@ harness = false
 [features]
 termination = []
 [target."cfg(unix)".dependencies.nix]
-version = "0.24"
-features = ["fs", "signal"]
+version = "0.23"
+#features = ["fs", "signal"]
 default-features = false
 [target."cfg(windows)".dependencies.winapi]
 version = "0.3"
