--- a/Cargo.toml
+++ b/Cargo.toml
@@ -220,54 +220,6 @@
 name = "zst"
 path = "tests/zst.rs"
 
-[[bench]]
-name = "append"
-path = "benches/append.rs"
-
-[[bench]]
-name = "bench1"
-path = "benches/bench1.rs"
-
-[[bench]]
-name = "chunks"
-path = "benches/chunks.rs"
-
-[[bench]]
-name = "construct"
-path = "benches/construct.rs"
-
-[[bench]]
-name = "gemv_gemm"
-path = "benches/gemv_gemm.rs"
-
-[[bench]]
-name = "higher-order"
-path = "benches/higher-order.rs"
-
-[[bench]]
-name = "iter"
-path = "benches/iter.rs"
-
-[[bench]]
-name = "numeric"
-path = "benches/numeric.rs"
-
-[[bench]]
-name = "par_rayon"
-path = "benches/par_rayon.rs"
-
-[[bench]]
-name = "reserve"
-path = "benches/reserve.rs"
-
-[[bench]]
-name = "to_shape"
-path = "benches/to_shape.rs"
-
-[[bench]]
-name = "zip"
-path = "benches/zip.rs"
-
 [dependencies.approx]
 version = "0.5"
 optional = true
