--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,3 +36,3 @@ default-features = false
 [dependencies.async-channel]
-version = "2.1.1"
+version = ">= 1.9.0"
 optional = true
@@ -40,3 +40,3 @@ optional = true
 [dependencies.crossterm]
-version = "0.27.0"
+version = ">= 0.27.0, < 0.30"
 features = ["events"]
@@ -60,3 +60,3 @@ default-features = false
 [dependencies.futures-lite]
-version = "2.1.0"
+version = ">= 1.12.0"
 optional = true
