diff --git a/Cargo.toml b/Cargo.toml
index e1d1dc6..a5b8150 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -38,7 +38,7 @@ version = "1.1.1"
 optional = true
 
 [dependencies.crossterm]
-version = "0.26.1"
+version = "0.25"
 optional = true
 default-features = false
 
@@ -61,12 +61,12 @@ version = "1.4.0"
 optional = true
 
 [dependencies.termion]
-version = "2.0.1"
+version = "1"
 optional = true
 default-features = false
 
 [dependencies.tui]
-version = "0.20.1"
+version = "0.22"
 optional = true
 default-features = false
 package = "ratatui"
diff --git a/src/terminal.rs b/src/terminal.rs
index 1bba472..f33c8d3 100644
--- a/src/terminal.rs
+++ b/src/terminal.rs
@@ -81,10 +81,10 @@ mod _impl {
     impl<T: io::Write> AlternateRawScreen<termion::raw::RawTerminal<T>> {
         pub fn try_from(write: T) -> Result<Self, io::Error> {
             use termion::raw::IntoRawMode;
-            use termion::screen::IntoAlternateScreen;
+            use termion::screen::AlternateScreen;
             let write = write.into_raw_mode()?;
             Ok(AlternateRawScreen {
-                inner: write.into_alternate_screen()?,
+                inner: AlternateScreen::from(write),
             })
         }
     }
