Index: pleaser/Cargo.toml
===================================================================
--- pleaser.orig/Cargo.toml
+++ pleaser/Cargo.toml
@@ -33,7 +33,8 @@ version = "0.2"
 version = "0.2"
 
 [dependencies.nix]
-version = "0.26"
+version = ">= 0.27"
+features = ["signal", "user", "fs", "term", "hostname"]
 
 [dependencies.pam]
 version = "0.7"
Index: pleaser/src/lib.rs
===================================================================
--- pleaser.orig/src/lib.rs
+++ pleaser/src/lib.rs
@@ -1506,13 +1506,13 @@ pub fn challenge_password(ro: &RunOption
 
                 let tty = std::fs::File::open("/dev/tty");
                 if tty.is_ok() {
-                    let term_res = nix::sys::termios::tcgetattr(tty.as_ref().unwrap().as_raw_fd());
+                    let term_res = nix::sys::termios::tcgetattr(tty.as_ref().unwrap());
                     if let Ok(mut term) = term_res {
                         term.local_flags
                             .set(nix::sys::termios::LocalFlags::ECHO, true);
 
                         let res = nix::sys::termios::tcsetattr(
-                            tty.as_ref().unwrap().as_raw_fd(),
+                            tty.as_ref().unwrap(),
                             nix::sys::termios::SetArg::TCSANOW,
                             &term,
                         );
