Description: syslog 6.0 pid type change
Author: Ed Neville <ed-debian@s5h.net>
Origin: upstream
Forwarded: not-needed
Last-Update: 2021-12-23

--- rust-pleaser-0.5.1.orig/src/lib.rs
+++ rust-pleaser-0.5.1/src/lib.rs
@@ -1565,7 +1565,7 @@ pub fn log_action(service: &str, result:
         facility: Facility::LOG_USER,
         hostname: None,
         process: service.into(),
-        pid: process::id() as i32,
+        pid: process::id(),
     };
 
     let cwd = match env::current_dir() {
