Index: cbindgen/Cargo.toml
===================================================================
--- cbindgen.orig/Cargo.toml
+++ cbindgen/Cargo.toml
@@ -66,7 +66,7 @@ optional = true
 version = "0.4"
 
 [dependencies.indexmap]
-version = "2.1.0"
+version = ">= 1"
 
 [dependencies.log]
 version = "0.4"
@@ -107,7 +107,7 @@ version = "0.8.8"
 version = "1.4.0"
 
 [dev-dependencies.serial_test]
-version = "2.0.0"
+version = ">= 0.5.0, < 3"
 default-features = false
 
 [features]
