Index: cbindgen/Cargo.toml
===================================================================
--- cbindgen.orig/Cargo.toml
+++ cbindgen/Cargo.toml
@@ -71,7 +71,7 @@ optional = true
 version = "0.4"
 
 [dependencies.indexmap]
-version = "2.1.0"
+version = ">= 1"
 
 [dependencies.log]
 version = "0.4"
@@ -112,5 +112,5 @@ version = "0.8.8"
 version = "1.4.0"
 
 [dev-dependencies.serial_test]
-version = "2.0.0"
+version = ">= 0.5.0, < 3"
 default-features = false
