Last-Updated: 2025-08-08
Author: Blair Noctis <ncts@debian.org>
Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -103,11 +103,11 @@
-windows-native = [
-    "windows-sys/Win32_Devices_DeviceAndDriverInstallation",
-    "windows-sys/Win32_Devices_HumanInterfaceDevice",
-    "windows-sys/Win32_Devices_Properties",
-    "windows-sys/Win32_Security",
-    "windows-sys/Win32_Storage_EnhancedStorage",
-    "windows-sys/Win32_Storage_FileSystem",
-    "windows-sys/Win32_System_IO",
-    "windows-sys/Win32_System_Threading",
-    "windows-sys/Win32_UI_Shell_PropertiesSystem",
-]
+#windows-native = [
+#    "windows-sys/Win32_Devices_DeviceAndDriverInstallation",
+#    "windows-sys/Win32_Devices_HumanInterfaceDevice",
+#    "windows-sys/Win32_Devices_Properties",
+#    "windows-sys/Win32_Security",
+#    "windows-sys/Win32_Storage_EnhancedStorage",
+#    "windows-sys/Win32_Storage_FileSystem",
+#    "windows-sys/Win32_System_IO",
+#    "windows-sys/Win32_System_Threading",
+#    "windows-sys/Win32_UI_Shell_PropertiesSystem",
+#]
@@ -128 +128 @@
-[target."cfg(windows)".dependencies.windows-sys]
+[disabled.target."cfg(windows)".dependencies.windows-sys]
