--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,3 +54,3 @@
 
-[[bench]]
+[[disabled.bench]]
 name = "jwt"
@@ -87,3 +87,3 @@
 
-[target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dev-dependencies.criterion]
+[disabled.target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dev-dependencies.criterion]
 version = "0.4"
@@ -91,3 +91,3 @@
 
-[target.'cfg(not(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi")))))'.dev-dependencies.criterion]
+[disabled.target.'cfg(not(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi")))))'.dev-dependencies.criterion]
 version = "0.4"
 
