--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,110 +50,6 @@
 name = "plotters"
 path = "src/lib.rs"
 
-[[example]]
-name = "3d-plot"
-path = "examples/3d-plot.rs"
-
-[[example]]
-name = "3d-plot2"
-path = "examples/3d-plot2.rs"
-
-[[example]]
-name = "animation"
-path = "examples/animation.rs"
-
-[[example]]
-name = "area-chart"
-path = "examples/area-chart.rs"
-
-[[example]]
-name = "blit-bitmap"
-path = "examples/blit-bitmap.rs"
-
-[[example]]
-name = "boxplot"
-path = "examples/boxplot.rs"
-
-[[example]]
-name = "chart"
-path = "examples/chart.rs"
-
-[[example]]
-name = "colormaps"
-path = "examples/colormaps.rs"
-
-[[example]]
-name = "console"
-path = "examples/console.rs"
-
-[[example]]
-name = "customized_coord"
-path = "examples/customized_coord.rs"
-
-[[example]]
-name = "errorbar"
-path = "examples/errorbar.rs"
-
-[[example]]
-name = "full_palette"
-path = "examples/full_palette.rs"
-
-[[example]]
-name = "histogram"
-path = "examples/histogram.rs"
-
-[[example]]
-name = "mandelbrot"
-path = "examples/mandelbrot.rs"
-
-[[example]]
-name = "matshow"
-path = "examples/matshow.rs"
-
-[[example]]
-name = "nested_coord"
-path = "examples/nested_coord.rs"
-
-[[example]]
-name = "normal-dist"
-path = "examples/normal-dist.rs"
-
-[[example]]
-name = "normal-dist2"
-path = "examples/normal-dist2.rs"
-
-[[example]]
-name = "pie"
-path = "examples/pie.rs"
-
-[[example]]
-name = "relative_size"
-path = "examples/relative_size.rs"
-
-[[example]]
-name = "sierpinski"
-path = "examples/sierpinski.rs"
-
-[[example]]
-name = "slc-temp"
-path = "examples/slc-temp.rs"
-
-[[example]]
-name = "snowflake"
-path = "examples/snowflake.rs"
-
-[[example]]
-name = "stock"
-path = "examples/stock.rs"
-
-[[example]]
-name = "tick_control"
-path = "examples/tick_control.rs"
-
-[[example]]
-name = "two-scales"
-path = "examples/two-scales.rs"
-
 [[bench]]
 name = "benchmark"
 path = "benches/main.rs"
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -310,7 +310,7 @@
 
 And the following code draws a quadratic function. `src/main.rs`,
 
-```rust
+```rust,ignore
 use plotters::prelude::*;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let root = BitMapBackend::new("plotters-doc-data/0.png", (640, 480)).into_drawing_area();
@@ -458,7 +458,7 @@
 ### Drawing Backends
 Plotters can use different drawing backends, including SVG, BitMap, and even real-time rendering. For example, a bitmap drawing backend.
 
-```rust
+```rust,ignore
 use plotters::prelude::*;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     // Create a 800*600 bitmap and start drawing
@@ -480,7 +480,7 @@
 
 Besides that, the drawing area also allows for a customized coordinate system, by doing so, the coordinate mapping is done by the drawing area automatically.
 
-```rust
+```rust,ignore
 use plotters::prelude::*;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let root_drawing_area =
@@ -508,7 +508,7 @@
 
 To learn more about the element system, please read the [element module documentation](./element/index.html).
 
-```rust
+```rust,ignore
 use plotters::prelude::*;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let root = BitMapBackend::new("plotters-doc-data/3.png", (300, 200)).into_drawing_area();
@@ -534,7 +534,7 @@
 
 For example, we can have an element which includes a dot and its coordinate.
 
-```rust
+```rust,ignore
 use plotters::prelude::*;
 use plotters::coord::types::RangedCoordf32;
 
@@ -576,7 +576,7 @@
 For example, you can define the label areas, meshes, and put a data series onto the drawing area with the help
 of the chart context object.
 
-```rust
+```rust,ignore
 use plotters::prelude::*;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let root = BitMapBackend::new("plotters-doc-data/5.png", (640, 480)).into_drawing_area();
