--- a/Cargo.toml
+++ b/Cargo.toml
@@ -153,9 +153,6 @@
     "CanvasRenderingContext2d",
 ]
 
-[target."cfg(all(target_arch = \"wasm32\", not(target_os = \"wasi\")))".dev-dependencies.wasm-bindgen-test]
-version = "0.3.24"
-
 [target."cfg(not(all(target_arch = \"wasm32\", not(target_os = \"wasi\"))))".dependencies.ab_glyph]
 version = "0.2.12"
 optional = true
@@ -187,7 +184,7 @@
 optional = true
 
 [target."cfg(not(all(target_arch = \"wasm32\", not(target_os = \"wasi\"))))".dependencies.ttf-parser]
-version = "0.17.0"
+version = "0.24"
 optional = true
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.rand]
