--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,37 +40,6 @@
 name = "figment"
 path = "src/lib.rs"
 
-[[test]]
-name = "tuple-struct"
-path = "tests/tuple-struct.rs"
-
-[[test]]
-name = "lossy_values"
-path = "tests/lossy_values.rs"
-
-[[test]]
-name = "cargo"
-path = "tests/cargo.rs"
-
-[[test]]
-name = "empty-env-vars"
-path = "tests/empty-env-vars.rs"
-
-[[test]]
-name = "camel-case"
-path = "tests/camel-case.rs"
-
-[[test]]
-name = "enum"
-path = "tests/enum.rs"
-
-[[test]]
-name = "yaml-enum"
-path = "tests/yaml-enum.rs"
-
-[[test]]
-name = "tagged"
-path = "tests/tagged.rs"
 
 [dependencies.parking_lot]
 version = "0.12"
