--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,4 +68,4 @@
 
-[dependencies.serde_yaml_ng]
-version = "0.10.0"
+[dependencies.serde_yaml]
+version = "0.9"
 
--- a/src/carapace_spec.rs
+++ b/src/carapace_spec.rs
@@ -61,7 +61,7 @@ impl Generator for Spec {
 
     fn generate(&self, cmd: &clap::Command, buf: &mut dyn std::io::Write) {
         let command = command_for(cmd, true);
-        let serialized = serde_yaml_ng::to_string(&command).unwrap();
+        let serialized = serde_yaml::to_string(&command).unwrap();
         buf.write_all(serialized.as_bytes())
             .expect("Failed to write to generated file");
     }
