Index: rust-crc32fast-1.5.0/benches/bench.rs
===================================================================
--- rust-crc32fast-1.5.0.orig/benches/bench.rs
+++ rust-crc32fast-1.5.0/benches/bench.rs
@@ -66,6 +66,7 @@ bencher::benchmark_group!(
     bench_kilobyte_baseline,
     bench_megabyte_baseline
 );
+#[cfg(feature = "std")]
 bencher::benchmark_group!(
     bench_specialized,
     bench_kilobyte_specialized,
@@ -77,4 +78,7 @@ bencher::benchmark_group!(
     bench_combine_32,
     bench_combine_64
 );
+#[cfg(feature = "std")]
 bencher::benchmark_main!(bench_baseline, bench_specialized, bench_combine);
+#[cfg(not(feature = "std"))]
+benchmark_main!(bench_baseline);
Index: rust-crc32fast-1.5.0/src/baseline.rs
===================================================================
--- rust-crc32fast-1.5.0.orig/src/baseline.rs
+++ rust-crc32fast-1.5.0/src/baseline.rs
@@ -70,6 +70,9 @@ pub(crate) fn update_slow(prev: u32, buf
 
 #[cfg(test)]
 mod test {
+    #[cfg(not(feature = "std"))]
+    use alloc::vec::Vec;
+    
     #[test]
     fn slow() {
         assert_eq!(super::update_slow(0, b""), 0);
Index: rust-crc32fast-1.5.0/src/lib.rs
===================================================================
--- rust-crc32fast-1.5.0.orig/src/lib.rs
+++ rust-crc32fast-1.5.0/src/lib.rs
@@ -39,6 +39,9 @@
 use core::fmt;
 use core::hash;
 
+#[cfg(all(test,not(feature = "std")))]
+extern crate alloc;
+
 mod baseline;
 mod combine;
 mod specialized;
@@ -181,6 +184,9 @@ impl hash::Hasher for Hasher {
 mod test {
     use super::Hasher;
 
+    #[cfg(not(feature = "std"))]
+    use alloc::vec::Vec;
+    
     quickcheck::quickcheck! {
         fn combine(bytes_1: Vec<u8>, bytes_2: Vec<u8>) -> bool {
             let mut hash_a = Hasher::new();
Index: rust-crc32fast-1.5.0/src/specialized/pclmulqdq.rs
===================================================================
--- rust-crc32fast-1.5.0.orig/src/specialized/pclmulqdq.rs
+++ rust-crc32fast-1.5.0/src/specialized/pclmulqdq.rs
@@ -186,6 +186,7 @@ unsafe fn get(a: &mut &[u8]) -> arch::__
 
 #[cfg(test)]
 mod test {
+    #[cfg(feature = "std")]
     quickcheck::quickcheck! {
         fn check_against_baseline(init: u32, chunks: Vec<(Vec<u8>, usize)>) -> bool {
             let mut baseline = super::super::super::baseline::State::new(init);
Index: rust-crc32fast-1.5.0/src/specialized/aarch64.rs
===================================================================
--- rust-crc32fast-1.5.0.orig/src/specialized/aarch64.rs
+++ rust-crc32fast-1.5.0/src/specialized/aarch64.rs
@@ -79,6 +79,9 @@ pub unsafe fn calculate(crc: u32, data:
 
 #[cfg(test)]
 mod test {
+    #[cfg(not(feature = "std"))]
+    use alloc::vec::Vec;
+
     quickcheck::quickcheck! {
         fn check_against_baseline(init: u32, chunks: Vec<(Vec<u8>, usize)>) -> bool {
             let mut baseline = super::super::super::baseline::State::new(init);
