Index: netlink-packet-route-0.19/Cargo.toml
===================================================================
--- netlink-packet-route-0.19.orig/Cargo.toml
+++ netlink-packet-route-0.19/Cargo.toml
@@ -24,9 +24,6 @@ keywords = [
 license = "MIT"
 repository = "https://github.com/rust-netlink/netlink-packet-route"
 
-[[example]]
-name = "dump_packet_links"
-
 [dependencies.anyhow]
 version = "1.0.31"
 
