Index: netlink-packet-route-0.19/Cargo.toml
===================================================================
--- netlink-packet-route-0.19.orig/Cargo.toml
+++ netlink-packet-route-0.19/Cargo.toml
@@ -49,8 +49,5 @@ version = "0.5.2"
 [dev-dependencies.netlink-sys]
 version = "0.8.5"
 
-[dev-dependencies.pretty_assertions]
-version = "0.7.2"
-
 [features]
 rich_nlas = []
Index: netlink-packet-route-0.19/src/lib.rs
===================================================================
--- netlink-packet-route-0.19.orig/src/lib.rs
+++ netlink-packet-route-0.19/src/lib.rs
@@ -66,7 +66,3 @@ pub use self::message::{RouteNetlinkMess
 
 #[macro_use]
 extern crate netlink_packet_utils;
-
-#[cfg(test)]
-#[macro_use]
-extern crate pretty_assertions;
