Index: petgraph/Cargo.toml
===================================================================
--- petgraph.orig/Cargo.toml
+++ petgraph/Cargo.toml
@@ -55,11 +55,6 @@ default-features = false
 [dependencies.indexmap]
 version = "1.7"
 
-[dependencies.quickcheck]
-version = "0.8"
-optional = true
-default-features = false
-
 [dependencies.serde]
 version = "1.0"
 optional = true
@@ -82,7 +77,6 @@ version = "0.5.5"
 [features]
 all = [
     "unstable",
-    "quickcheck",
     "matrix_graph",
     "stable_graph",
     "graphmap",
