--- a/benches/common/factories.rs
+++ b/benches/common/factories.rs
@@ -265,14 +265,17 @@
     graph()
 }
 
+#[cfg(feature = "stable_graph")]
 pub fn stable_graph<Ty: EdgeType>() -> GraphFactory<Ty, StableGraph<(), (), Ty>> {
     GraphFactory::new()
 }
 
+#[cfg(feature = "stable_graph")]
 pub fn stable_ungraph() -> GraphFactory<Undirected, StableGraph<(), (), Undirected>> {
     stable_graph()
 }
 
+#[cfg(feature = "stable_graph")]
 pub fn stable_digraph() -> GraphFactory<Directed, StableGraph<(), (), Directed>> {
     stable_graph()
 }
--- a/benches/stable_graph.rs
+++ b/benches/stable_graph.rs
@@ -1,3 +1,4 @@
+#![cfg(feature = "stable_graph")]
 #![feature(test)]
 
 extern crate petgraph;
--- a/benches/matrix_graph.rs
+++ b/benches/matrix_graph.rs
@@ -1,3 +1,4 @@
+#![cfg(feature = "matrix_graph")]
 #![feature(test)]
 
 extern crate petgraph;
