Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -371,7 +371,7 @@ rand = "0.7"
 regex = "1.0"
 sha1 = { version="0.6", features=["std"] }
 tempfile = "3.2.0"
-time = "0.1"
+time = "0"
 unindent = "0.1"
 uucore = { version=">=0.0.11", package="uucore", path="src/uucore", features=["entries", "process"] }
 walkdir = "2.2"
Index: coreutils/src/uu/install/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/install/Cargo.toml
+++ coreutils/src/uu/install/Cargo.toml
@@ -19,14 +19,14 @@ path = "src/install.rs"
 
 [dependencies]
 clap = { version = "2.33", features = ["wrap_help"] }
-filetime = "0.2"
+filetime = "0"
 file_diff = "1.0.0"
 libc = ">= 0.2"
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore", features=["fs", "mode", "perms", "entries"] }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
 
 [dev-dependencies]
-time = "0.1.40"
+time = "0"
 
 [[bin]]
 name = "install"
Index: coreutils/src/uu/touch/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/touch/Cargo.toml
+++ coreutils/src/uu/touch/Cargo.toml
@@ -17,7 +17,7 @@ path = "src/touch.rs"
 [dependencies]
 filetime = "0.2.1"
 clap = { version = "2.33", features = ["wrap_help"] }
-time = "0.1.40"
+time = "0"
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore", features=["libc"] }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
 
Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -22,7 +22,7 @@ dunce = "1.0.0"
 wild = "2.0"
 # * optional
 thiserror = { version="1.0", optional=true }
-time = { version="<= 0.1.43", optional=true }
+time = { version="0", optional=true }
 # * "problem" dependencies (pinned)
 data-encoding = { version="2.1", optional=true }
 data-encoding-macro = { version="0.1.12", optional=true }
