Index: coreutils/src/uu/hashsum/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/hashsum/Cargo.toml
+++ coreutils/src/uu/hashsum/Cargo.toml
@@ -26,7 +26,7 @@ regex-syntax = "0.6.7"
 sha1 = "0.6.0"
 sha2 = "0.6.0"
 sha3 = "0.6.0"
-blake2b_simd = "0.5.11"
+blake2b_simd = "1"
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore" }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
 
