Index: coreutils/src/uu/expr/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/expr/Cargo.toml
+++ coreutils/src/uu/expr/Cargo.toml
@@ -19,7 +19,7 @@ clap = { version = "2.33", features = ["
 libc = "0.2.42"
 num-bigint = "0.4.0"
 num-traits = "0.2.14"
-onig = "~4.3.2"
+onig = { version = "6", default-features = false }
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore" }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
 
