Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -248,7 +248,7 @@ clap = { version = "2.33", features = ["
 lazy_static = { version="1.3" }
 textwrap = { version="0.14", features=["terminal_size"] }
 uucore = { version=">=0.0.11", package="uucore", path="src/uucore" }
-selinux = { version="0.2.3", optional = true }
+selinux = { version="0.2", optional = true }
 # * uutils
 uu_test  = { optional=true, version="0.0.12", package="uu_test", path="src/uu/test" }
 #
Index: coreutils/src/uu/cp/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/cp/Cargo.toml
+++ coreutils/src/uu/cp/Cargo.toml
@@ -23,7 +23,7 @@ clap = { version = "2.33", features = ["
 filetime = "0.2"
 libc = "0.2.85"
 quick-error = "1.2.3"
-selinux = { version="0.2.3", optional=true }
+selinux = { version="0.2", optional=true }
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore", features=["entries", "fs", "perms", "mode"] }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
 walkdir = "2.2"
Index: coreutils/src/uu/id/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/id/Cargo.toml
+++ coreutils/src/uu/id/Cargo.toml
@@ -18,7 +18,7 @@ path = "src/id.rs"
 clap = { version = "2.33", features = ["wrap_help"] }
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore", features=["entries", "process"] }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
-selinux = { version="0.2.1", optional = true }
+selinux = { version="0.2", optional = true }
 
 [[bin]]
 name = "id"
Index: coreutils/src/uu/ls/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/ls/Cargo.toml
+++ coreutils/src/uu/ls/Cargo.toml
@@ -27,7 +27,7 @@ uucore = { version = ">=0.0.8", package
 uucore_procs = { version=">=0.0.8", package = "uucore_procs", path = "../../uucore_procs" }
 once_cell = "1.7.2"
 atty = "0.2"
-selinux = { version="0.2.1", optional = true }
+selinux = { version="0.2", optional = true }
 
 [target.'cfg(unix)'.dependencies]
 lazy_static = "1.4.0"
