Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -32,7 +32,7 @@ once_cell = "1.8.0"
 os_display = "0.1.0"
 
 [target.'cfg(unix)'.dependencies]
-walkdir = { version="2.3.2", optional=true }
+walkdir = { version="2", optional=true }
 nix = { version="0.23.1", optional=true }
 
 [dev-dependencies]
