Index: coreutils/src/uu/uniq/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/uniq/Cargo.toml
+++ coreutils/src/uu/uniq/Cargo.toml
@@ -16,8 +16,8 @@ path = "src/uniq.rs"
 
 [dependencies]
 clap = { version = "2.33", features = ["wrap_help"] }
-strum = "0.21"
-strum_macros = "0.21"
+strum = "0"
+strum_macros = "0"
 uucore = { version=">=0.0.11", package="uucore", path="../../uucore" }
 uucore_procs = { version=">=0.0.8", package="uucore_procs", path="../../uucore_procs" }
 
