Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -246,7 +246,7 @@ test = [ "uu_test" ]
 [dependencies]
 clap = { version = "2.33", features = ["wrap_help"] }
 lazy_static = { version="1.3" }
-textwrap = { version="0.14", features=["terminal_size"] }
+textwrap = { version="0.11", features=["terminal_size"] }
 uucore = { version=">=0.0.11", package="uucore", path="src/uucore" }
 selinux = { version="0.2", optional = true }
 # * uutils
