Index: coreutils/src/uu/more/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/more/Cargo.toml
+++ coreutils/src/uu/more/Cargo.toml
@@ -21,7 +21,7 @@ uucore_procs = { version=">=0.0.8", pack
 crossterm = ">=0.19"
 atty = "0.2"
 unicode-width = "0.1.7"
-unicode-segmentation = "1.7.1"
+unicode-segmentation = "1"
 
 [target.'cfg(target_os = "redox")'.dependencies]
 redox_termios = "0.1"
