--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,45 +54,6 @@
 name = "tungstenite"
 path = "src/lib.rs"
 
-[[example]]
-name = "autobahn-client"
-path = "examples/autobahn-client.rs"
-required-features = ["handshake"]
-
-[[example]]
-name = "autobahn-server"
-path = "examples/autobahn-server.rs"
-required-features = ["handshake"]
-
-[[example]]
-name = "callback-error"
-path = "examples/callback-error.rs"
-required-features = ["handshake"]
-
-[[example]]
-name = "client"
-path = "examples/client.rs"
-required-features = ["handshake"]
-
-[[example]]
-name = "server"
-path = "examples/server.rs"
-required-features = ["handshake"]
-
-[[example]]
-name = "srv_accept_unmasked_frames"
-path = "examples/srv_accept_unmasked_frames.rs"
-required-features = ["handshake"]
-
-[[bench]]
-name = "buffer"
-path = "benches/buffer.rs"
-harness = false
-
-[[bench]]
-name = "write"
-path = "benches/write.rs"
-harness = false
 
 [dependencies.byteorder]
 version = "1.3.2"
