Index: tungstenite/Cargo.toml
===================================================================
--- tungstenite.orig/Cargo.toml
+++ tungstenite/Cargo.toml
@@ -91,7 +91,7 @@ optional = true
 default-features = false
 
 [dependencies.rustls-native-certs]
-version = "0.7.0"
+version = "0.6"
 optional = true
 
 [dependencies.rustls-pki-types]
Index: tungstenite/src/tls.rs
===================================================================
--- tungstenite.orig/src/tls.rs
+++ tungstenite/src/tls.rs
@@ -108,7 +108,7 @@ mod encryption {
                                 let native_certs = rustls_native_certs::load_native_certs()?;
                                 let total_number = native_certs.len();
                                 let (number_added, number_ignored) =
-                                    root_store.add_parsable_certificates(native_certs);
+                                    root_store.add_parsable_certificates(native_certs.into_iter().map(|cert| cert.0.into()));
                                 log::debug!("Added {number_added}/{total_number} native root certificates (ignored {number_ignored})");
                             }
                             #[cfg(feature = "rustls-tls-webpki-roots")]
