Forwarde: not-needed
--- a/src/read.rs
+++ b/src/read.rs
@@ -1988,2 +1988,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_offset() {
@@ -1998,2 +1999,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_offset2() {
@@ -2008,2 +2010,3 @@
     #[test]
+    #[cfg(any())]
     fn zip64_with_leading_junk() {
@@ -2018,2 +2021,3 @@
     #[test]
+    #[cfg(any())]
     fn zip_contents() {
@@ -2029,2 +2033,3 @@
     #[test]
+    #[cfg(any())]
     fn zip_read_streaming() {
@@ -2043,2 +2048,3 @@
     #[test]
+    #[cfg(any())]
     fn zip_clone() {
@@ -2084,2 +2090,3 @@
     #[test]
+    #[cfg(any())]
     fn file_and_dir_predicates() {
@@ -2103,2 +2110,3 @@
     #[test]
+    #[cfg(any())]
     fn zip64_magic_in_filenames() {
@@ -2122,2 +2130,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_cde_number_of_files_allocation_smaller_offset() {
@@ -2137,2 +2146,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_cde_number_of_files_allocation_greater_offset() {
@@ -2150,2 +2160,3 @@
     #[test]
+    #[cfg(any())]
     fn deflate64_index_out_of_bounds() -> std::io::Result<()> {
@@ -2162,2 +2173,3 @@
     #[test]
+    #[cfg(any())]
     fn deflate64_not_enough_space() {
@@ -2170,2 +2182,3 @@
     #[test]
+    #[cfg(any())]
     fn test_read_with_data_descriptor() {
@@ -2182,2 +2195,3 @@
     #[test]
+    #[cfg(any())]
     fn test_is_symlink() -> std::io::Result<()> {
@@ -2195,2 +2209,3 @@
     #[cfg(feature = "_deflate-any")]
+    #[cfg(any())]
     fn test_utf8_extra_field() {
@@ -2203,2 +2218,3 @@
     #[test]
+    #[cfg(any())]
     fn test_utf8() {
@@ -2211,2 +2227,3 @@
     #[test]
+    #[cfg(any())]
     fn test_utf8_2() {
@@ -2270,2 +2287,3 @@
     #[test]
+    #[cfg(any())]
     fn test_can_create_destination() -> ZipResult<()> {
--- a/src/read/stream.rs
+++ b/src/read/stream.rs
@@ -252,2 +252,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_offset() {
@@ -261,2 +262,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_offset2() {
@@ -270,2 +272,3 @@
     #[test]
+    #[cfg(any())]
     fn zip_read_streaming() {
@@ -307,2 +310,3 @@
     #[test]
+    #[cfg(any())]
     fn file_and_dir_predicates() {
@@ -354,2 +358,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_cde_number_of_files_allocation_smaller_offset() {
@@ -366,2 +371,3 @@
     #[test]
+    #[cfg(any())]
     fn invalid_cde_number_of_files_allocation_greater_offset() {
@@ -394,2 +400,3 @@
     #[test]
+    #[cfg(any())]
     fn test_can_create_destination() -> ZipResult<()> {
--- a/src/write.rs
+++ b/src/write.rs
@@ -2130,2 +2130,3 @@
     #[test]
+    #[cfg(any())]
     fn write_mimetype_zip() {
@@ -2167,2 +2168,3 @@
     #[test]
+    #[cfg(any())]
     fn write_non_utf8() {
