Index: zip/Cargo.toml
===================================================================
--- zip.orig/Cargo.toml
+++ zip/Cargo.toml
@@ -71,7 +71,7 @@ default = [
     "deflate64",
     "deflate",
     "lzma",
-    "ppmd",
+#    "ppmd",
     "time",
     "zstd",
     "xz",
@@ -100,1 +100,1 @@
-ppmd = ["dep:ppmd-rust"]
+#ppmd = ["dep:ppmd-rust"]
@@ -186,9 +186,9 @@ default-features = false
 version = "0.12"
 optional = true
 
-[dependencies.ppmd-rust]
-version = "1.2"
-optional = true
+#[dependencies.ppmd-rust]
+#version = "1.2"
+#optional = true
 
 [dependencies.sha1]
 version = "0.10"
