Index: zip/Cargo.toml
===================================================================
--- zip.orig/Cargo.toml
+++ zip/Cargo.toml
@@ -78,7 +78,7 @@ default = [
 ]
 deflate = [
     "deflate-zopfli",
-    "deflate-flate2-zlib-rs",
+    "deflate-flate2-zlib",
 ]
 deflate-flate2 = [
     "_deflate-any",
@@ -88,10 +88,6 @@ deflate-flate2-zlib = [
     "deflate-flate2",
     "flate2/zlib",
 ]
-deflate-flate2-zlib-rs = [
-    "deflate-flate2",
-    "flate2/zlib-rs",
-]
 deflate-zopfli = [
     "dep:zopfli",
     "_deflate-any",
