Forwarde: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -129,3 +129,3 @@
 [dependencies.bzip2]
-version = "0.5.0"
+version = "0.4"
 optional = true
@@ -144,3 +144,3 @@
 [dependencies.deflate64]
-version = "0.1.9"
+version = "0.1"
 optional = true
@@ -153,5 +153,5 @@
 [dependencies.getrandom]
-version = "0.3.1"
+version = "0.2"
 features = [
-    "wasm_js",
+    #"wasm_js",
     "std",
@@ -232,5 +232,5 @@
 [dev-dependencies.getrandom]
-version = "0.3.1"
+version = "0.2.1"
 features = [
-    "wasm_js",
+    #"wasm_js",
     "std",
@@ -253,3 +253,3 @@
 [target.'cfg(any(all(target_arch = "arm", target_pointer_width = "32"), target_arch = "mips", target_arch = "powerpc"))'.dependencies.crossbeam-utils]
-version = "0.8.21"
+version = "0.8"
 
--- a/src/aes.rs
+++ b/src/aes.rs
@@ -238,3 +238,3 @@
         let mut salt = vec![0; salt_length];
-        getrandom::fill(&mut salt)?;
+        getrandom::getrandom(&mut salt)?;
         encrypted_file_header.write_all(&salt)?;
