diff --git a/Cargo.toml b/Cargo.toml
index 5081607..d592267 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,13 +47,13 @@ version = "~0.13"
 optional = true
 
 [dependencies.constant_time_eq]
-version = "~0.2.1"
+version = "0.1"
 
 [dependencies.hmac]
 version = "~0.12.1"
 
 [dependencies.image]
-version = "~0.24.2"
+version = "~0.23"
 features = ["png"]
 optional = true
 default-features = false
@@ -63,7 +63,7 @@ version = "~1.8"
 optional = true
 
 [dependencies.rand]
-version = "~0.8.5"
+version = "~0.8"
 features = [
     "std_rng",
     "std",
@@ -72,12 +72,12 @@ optional = true
 default-features = false
 
 [dependencies.serde]
-version = "1.0"
+version = "1"
 features = ["derive"]
 optional = true
 
-[dependencies.sha-1]
-version = "~0.10.0"
+[dependencies.sha1]
+version = "~0.10"
 
 [dependencies.sha2]
 version = "~0.10.2"
@@ -87,7 +87,7 @@ version = "^2.2.2"
 optional = true
 
 [dependencies.urlencoding]
-version = "^2.1.0"
+version = "1"
 optional = true
 
 [features]
