Index: linux-raw-sys/Cargo.toml
===================================================================
--- linux-raw-sys.orig/Cargo.toml
+++ linux-raw-sys/Cargo.toml
@@ -92,10 +92,6 @@ netlink = []
 no_std = []
 prctl = []
 ptrace = []
-rustc-dep-of-std = [
-    "core",
-    "no_std",
-]
 std = []
 system = []
 xdp = []
@@ -104,11 +100,6 @@ xdp = []
 name = "linux_raw_sys"
 path = "src/lib.rs"
 
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [dev-dependencies.libc]
 version = "0.2.100"
 
