Index: io-uring/src/sys/sys.rs
===================================================================
--- io-uring.orig/src/sys/sys.rs
+++ io-uring/src/sys/sys.rs
@@ -182,6 +182,13 @@ fn bindgen_test_layout___kernel_timespec
         16usize,
         concat!("Size of: ", stringify!(__kernel_timespec))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<__kernel_timespec>(),
+        4usize,
+        concat!("Alignment of ", stringify!(__kernel_timespec))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<__kernel_timespec>(),
         8usize,
@@ -224,6 +231,13 @@ fn bindgen_test_layout_open_how() {
         24usize,
         concat!("Size of: ", stringify!(open_how))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<open_how>(),
+        4usize,
+        concat!("Alignment of ", stringify!(open_how))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<open_how>(),
         8usize,
@@ -342,6 +356,13 @@ fn bindgen_test_layout_io_uring_sqe__bin
         8usize,
         concat!("Size of: ", stringify!(io_uring_sqe__bindgen_ty_1))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_sqe__bindgen_ty_1>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_sqe__bindgen_ty_1))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_sqe__bindgen_ty_1>(),
         8usize,
@@ -442,6 +463,13 @@ fn bindgen_test_layout_io_uring_sqe__bin
         8usize,
         concat!("Size of: ", stringify!(io_uring_sqe__bindgen_ty_2))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_sqe__bindgen_ty_2>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_sqe__bindgen_ty_2))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_sqe__bindgen_ty_2>(),
         8usize,
@@ -925,6 +953,16 @@ fn bindgen_test_layout_io_uring_sqe__bin
             stringify!(io_uring_sqe__bindgen_ty_6__bindgen_ty_1)
         )
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_sqe__bindgen_ty_6__bindgen_ty_1>(),
+        4usize,
+        concat!(
+            "Alignment of ",
+            stringify!(io_uring_sqe__bindgen_ty_6__bindgen_ty_1)
+        )
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_sqe__bindgen_ty_6__bindgen_ty_1>(),
         8usize,
@@ -964,6 +1002,13 @@ fn bindgen_test_layout_io_uring_sqe__bin
         16usize,
         concat!("Size of: ", stringify!(io_uring_sqe__bindgen_ty_6))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_sqe__bindgen_ty_6>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_sqe__bindgen_ty_6))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_sqe__bindgen_ty_6>(),
         8usize,
@@ -1008,6 +1053,13 @@ fn bindgen_test_layout_io_uring_sqe() {
         64usize,
         concat!("Size of: ", stringify!(io_uring_sqe))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_sqe>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_sqe))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_sqe>(),
         8usize,
@@ -1177,6 +1229,13 @@ fn bindgen_test_layout_io_uring_cqe() {
         16usize,
         concat!("Size of: ", stringify!(io_uring_cqe))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_cqe>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_cqe))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_cqe>(),
         8usize,
@@ -1247,6 +1306,13 @@ fn bindgen_test_layout_io_sqring_offsets
         40usize,
         concat!("Size of: ", stringify!(io_sqring_offsets))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_sqring_offsets>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_sqring_offsets))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_sqring_offsets>(),
         8usize,
@@ -1365,6 +1431,13 @@ fn bindgen_test_layout_io_cqring_offsets
         40usize,
         concat!("Size of: ", stringify!(io_cqring_offsets))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_cqring_offsets>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_cqring_offsets))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_cqring_offsets>(),
         8usize,
@@ -1484,6 +1557,13 @@ fn bindgen_test_layout_io_uring_params()
         120usize,
         concat!("Size of: ", stringify!(io_uring_params))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_params>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_params))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_params>(),
         8usize,
@@ -1636,6 +1716,13 @@ fn bindgen_test_layout_io_uring_files_up
         16usize,
         concat!("Size of: ", stringify!(io_uring_files_update))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_files_update>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_files_update))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_files_update>(),
         8usize,
@@ -1691,6 +1778,13 @@ fn bindgen_test_layout_io_uring_rsrc_reg
         32usize,
         concat!("Size of: ", stringify!(io_uring_rsrc_register))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_rsrc_register>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_rsrc_register))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_rsrc_register>(),
         8usize,
@@ -1764,6 +1858,13 @@ fn bindgen_test_layout_io_uring_rsrc_upd
         16usize,
         concat!("Size of: ", stringify!(io_uring_rsrc_update))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_rsrc_update>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_rsrc_update))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_rsrc_update>(),
         8usize,
@@ -1820,6 +1921,13 @@ fn bindgen_test_layout_io_uring_rsrc_upd
         32usize,
         concat!("Size of: ", stringify!(io_uring_rsrc_update2))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_rsrc_update2>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_rsrc_update2))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_rsrc_update2>(),
         8usize,
@@ -2168,6 +2276,13 @@ fn bindgen_test_layout_io_uring_buf() {
         16usize,
         concat!("Size of: ", stringify!(io_uring_buf))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_buf>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_buf))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_buf>(),
         8usize,
@@ -2245,6 +2360,16 @@ fn bindgen_test_layout_io_uring_buf_ring
             stringify!(io_uring_buf_ring__bindgen_ty_1__bindgen_ty_1)
         )
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_buf_ring__bindgen_ty_1__bindgen_ty_1>(),
+        4usize,
+        concat!(
+            "Alignment of ",
+            stringify!(io_uring_buf_ring__bindgen_ty_1__bindgen_ty_1)
+        )
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_buf_ring__bindgen_ty_1__bindgen_ty_1>(),
         8usize,
@@ -2337,7 +2462,7 @@ fn bindgen_test_layout_io_uring_buf_ring
     );
     assert_eq!(
         ::core::mem::align_of::<io_uring_buf_ring__bindgen_ty_1__bindgen_ty_2>(),
-        8usize,
+        ::core::mem::align_of::<u64>(),
         concat!(
             "Alignment of ",
             stringify!(io_uring_buf_ring__bindgen_ty_1__bindgen_ty_2)
@@ -2371,6 +2496,13 @@ fn bindgen_test_layout_io_uring_buf_ring
         16usize,
         concat!("Size of: ", stringify!(io_uring_buf_ring__bindgen_ty_1))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_buf_ring__bindgen_ty_1>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_buf_ring__bindgen_ty_1))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_buf_ring__bindgen_ty_1>(),
         8usize,
@@ -2393,6 +2525,13 @@ fn bindgen_test_layout_io_uring_buf_ring
         16usize,
         concat!("Size of: ", stringify!(io_uring_buf_ring))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_buf_ring>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_buf_ring))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_buf_ring>(),
         8usize,
@@ -2426,6 +2565,13 @@ fn bindgen_test_layout_io_uring_buf_reg(
         40usize,
         concat!("Size of: ", stringify!(io_uring_buf_reg))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_buf_reg>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_buf_reg))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_buf_reg>(),
         8usize,
@@ -2506,6 +2652,13 @@ fn bindgen_test_layout_io_uring_getevent
         24usize,
         concat!("Size of: ", stringify!(io_uring_getevents_arg))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_getevents_arg>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_getevents_arg))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_getevents_arg>(),
         8usize,
@@ -2573,6 +2726,13 @@ fn bindgen_test_layout_io_uring_sync_can
         64usize,
         concat!("Size of: ", stringify!(io_uring_sync_cancel_reg))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_sync_cancel_reg>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_sync_cancel_reg))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_sync_cancel_reg>(),
         8usize,
@@ -2666,6 +2826,13 @@ fn bindgen_test_layout_io_uring_file_ind
         16usize,
         concat!("Size of: ", stringify!(io_uring_file_index_range))
     );
+    #[cfg(target_arch = "x86")]
+    assert_eq!(
+        ::core::mem::align_of::<io_uring_file_index_range>(),
+        4usize,
+        concat!("Alignment of ", stringify!(io_uring_file_index_range))
+    );
+    #[cfg(not(target_arch = "x86"))]
     assert_eq!(
         ::core::mem::align_of::<io_uring_file_index_range>(),
         8usize,
@@ -2785,7 +2952,7 @@ fn bindgen_test_layout_futex_waitv() {
     );
     assert_eq!(
         ::core::mem::align_of::<futex_waitv>(),
-        8usize,
+        ::core::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(futex_waitv))
     );
     assert_eq!(
