Index: io-uring/src/opcode.rs
===================================================================
--- io-uring.orig/src/opcode.rs
+++ io-uring/src/opcode.rs
@@ -152,7 +152,7 @@ opcode! {
         sqe.__bindgen_anon_2.addr = iovec as _;
         sqe.len = len;
         sqe.__bindgen_anon_1.off = offset;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         sqe.__bindgen_anon_4.buf_group = buf_group;
         Entry(sqe)
     }
@@ -189,7 +189,7 @@ opcode! {
         sqe.__bindgen_anon_2.addr = iovec as _;
         sqe.len = len;
         sqe.__bindgen_anon_1.off = offset;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         Entry(sqe)
     }
 }
@@ -263,7 +263,7 @@ opcode! {
         sqe.__bindgen_anon_2.addr = buf as _;
         sqe.len = len;
         sqe.__bindgen_anon_1.off = offset;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         sqe.__bindgen_anon_4.buf_index = buf_index;
         Entry(sqe)
     }
@@ -306,7 +306,7 @@ opcode! {
         sqe.__bindgen_anon_2.addr = buf as _;
         sqe.len = len;
         sqe.__bindgen_anon_1.off = offset;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         sqe.__bindgen_anon_4.buf_index = buf_index;
         Entry(sqe)
     }
@@ -925,7 +925,7 @@ opcode! {
         sqe.__bindgen_anon_2.addr = buf as _;
         sqe.len = len;
         sqe.__bindgen_anon_1.off = offset;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         sqe.__bindgen_anon_4.buf_group = buf_group;
         Entry(sqe)
     }
@@ -973,7 +973,7 @@ opcode! {
         sqe.__bindgen_anon_2.addr = buf as _;
         sqe.len = len;
         sqe.__bindgen_anon_1.off = offset;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         Entry(sqe)
     }
 }
@@ -1730,7 +1730,7 @@ opcode! {
         sqe.fd = domain as _;
         sqe.__bindgen_anon_1.off = socket_type as _;
         sqe.len = protocol as _;
-        sqe.__bindgen_anon_3.rw_flags = flags;
+        sqe.__bindgen_anon_3.rw_flags = flags as _;
         if let Some(dest) = file_index {
             sqe.__bindgen_anon_5.file_index = dest.kernel_index_arg();
         }
@@ -2321,7 +2321,7 @@ opcode! {
         sqe.len = len;
         sqe.__bindgen_anon_4.buf_index = buf_index;
         sqe.ioprio = ioprio;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         Entry(sqe)
     }
 }
@@ -2352,7 +2352,7 @@ opcode! {
         sqe.len = len;
         sqe.__bindgen_anon_4.buf_index = buf_index;
         sqe.ioprio = ioprio;
-        sqe.__bindgen_anon_3.rw_flags = rw_flags;
+        sqe.__bindgen_anon_3.rw_flags = rw_flags as _;
         Entry(sqe)
     }
 }
Index: io-uring/src/types.rs
===================================================================
--- io-uring.orig/src/types.rs
+++ io-uring/src/types.rs
@@ -48,7 +48,7 @@ use std::marker::PhantomData;
 use std::num::NonZeroU32;
 use std::os::unix::io::RawFd;
 
-pub use sys::__kernel_rwf_t as RwFlags;
+pub use libc::__kernel_rwf_t as RwFlags;
 pub use sys::{
     io_uring_region_desc, io_uring_zcrx_area_reg, io_uring_zcrx_cqe, io_uring_zcrx_ifq_reg,
     io_uring_zcrx_rqe, IORING_MEM_REGION_TYPE_USER, IORING_ZCRX_AREA_SHIFT, IOU_PBUF_RING_INC,
