Index: kvm-bindings/src/x86_64/serialize.rs
===================================================================
--- kvm-bindings.orig/src/x86_64/serialize.rs
+++ kvm-bindings/src/x86_64/serialize.rs
@@ -5,10 +5,10 @@ use bindings::{
     kvm_clock_data, kvm_cpuid2, kvm_cpuid_entry2, kvm_debugregs, kvm_dtable, kvm_irqchip,
     kvm_irqchip__bindgen_ty_1, kvm_lapic_state, kvm_mp_state, kvm_msr_entry, kvm_msrs,
     kvm_pit_channel_state, kvm_pit_state2, kvm_regs, kvm_segment, kvm_sregs, kvm_vcpu_events,
-    kvm_xcr, kvm_xcrs, kvm_xsave,
+    kvm_xcr, kvm_xcrs, kvm_xsave, kvm_ioapic_state__bindgen_ty_1,
 };
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
-use zerocopy::{transmute, IntoBytes, FromBytes, FromZeroes, Immutable};
+use zerocopy::{transmute, IntoBytes, FromBytes, FromZeros, Immutable};
 
 serde_impls!(
     kvm_regs,
