Index: terminal-size/Cargo.toml
===================================================================
--- terminal-size.orig/Cargo.toml
+++ terminal-size/Cargo.toml
@@ -31,9 +31,9 @@ repository = "https://github.com/eminenc
 version = "0.35.7"
 features = ["termios"]
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.36.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Console",
-]
+# [target."cfg(windows)".dependencies.windows-sys]
+# version = "0.36.0"
+# features = [
+#     "Win32_Foundation",
+#     "Win32_System_Console",
+# ]
