bump dependencies to what's in Debian

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,13 +22,13 @@
 license = "MPL-2.0"
 repository = "https://github.com/rust-rspec/rspec"
 [dependencies.colored]
-version = "2.0"
+version = "3.0"
 
 [dependencies.derive-new]
 version = "0.5"
 
 [dependencies.derive_builder]
-version = "0.9"
+version = "0.20"
 
 [dependencies.expectest]
 version = "0.12"
@@ -38,7 +38,7 @@
 version = "1.5"
 
 [dependencies.time]
-version = "0.2"
+version = "0.3"
 [build-dependencies.clippy]
 version = "0.0.153"
 optional = true
