--- rust-proptest-1.5.0.orig/src/num/float_samplers.rs
+++ rust-proptest-1.5.0/src/num/float_samplers.rs
@@ -352,6 +352,7 @@ macro_rules! float_sampler {
                     }
 
                     #[test]
+                    #[cfg(not(target_arch = "x86"))]
                     fn no_value_between_float_and_next_down(val in finite()) {
                         prop_assume!(val > $typ::MIN);
                         let prev = next_down(val);
