Description: fix FTBFS on all platforms
Author: Eric Long <i@hack3r.moe>
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -21,18 +21,16 @@
 //! you are switching.
 //!
 
-#![feature(untagged_unions)]
-
 #![cfg_attr(not(test), no_std)]
 #[cfg(not(test))]
 extern crate core as std;
 
+use std::mem::ManuallyDrop;
 use std::ops::{Deref, DerefMut};
 
-#[allow(unions_with_drop_fields)]
 #[derive(Copy)]
 union UnionFlag<T> {
-    value: T,
+    value: ManuallyDrop<T>,
 }
 
 impl<T: Clone> Clone for UnionFlag<T> {
@@ -54,7 +52,7 @@
     /// Create a new **NoDrop**.
     #[inline]
     pub fn new(value: T) -> Self {
-        NoDrop(UnionFlag { value: value })
+        NoDrop(UnionFlag { value: ManuallyDrop::new(value) })
     }
 
     /// Extract the inner value.
@@ -63,7 +61,7 @@
     #[inline]
     pub fn into_inner(self) -> T {
         unsafe {
-            self.0.value
+            ManuallyDrop::into_inner(self.0.value)
         }
     }
 }
