--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,14 +33,10 @@ repository = "https://gitlab.com/timvise
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies.derive_builder]
-version = "0.10"
+version = "0.9"
 
 [dependencies.rand]
 version = "0.8"
 
 [dependencies.thiserror]
-version = "1.0.31"
-
-[target."cfg(target_arch = \"wasm32\")".dependencies.getrandom]
-version = "0.2"
-features = ["js"]
+version = "1.0"
