--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,10 +57,10 @@
 
-[dev-dependencies.snapbox]
-version = "0.6.7"
-features = [
-    "diff",
-    "dir",
-    "term-svg",
-    "regex",
-]
+#[dev-dependencies.snapbox]
+#version = "0.6.7"
+#features = [
+#    "diff",
+#    "dir",
+#    "term-svg",
+#    "regex",
+#]
 
--- a/src/core/partial_version.rs
+++ b/src/core/partial_version.rs
@@ -185,3 +185,3 @@
 
-#[cfg(test)]
+#[cfg(all(test, feature = "enough-snapbox"))]
 mod test {
--- a/src/manifest/rust_version.rs
+++ b/src/manifest/rust_version.rs
@@ -105,3 +105,3 @@
 
-#[cfg(test)]
+#[cfg(all(test, feature = "enough-snapbox"))]
 mod test {
