diff --git a/Cargo.toml b/Cargo.toml
index 96f0fff..004a8c5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,7 +36,7 @@ repository = "https://github.com/cortex/ripasso/"
 name = "ripasso"
 path = "src/lib.rs"
 
-[[bench]]
+[[disabled.bench]]
 name = "library_benchmark"
 path = "benches/library_benchmark.rs"
 harness = false
@@ -99,7 +99,7 @@ features = [
     "alloc",
 ]
 
-[dev-dependencies.criterion]
+[disabled.dev-dependencies.criterion]
 version = "0.5"
 
 [dev-dependencies.flate2]
