diff --git a/src/tests/pass.rs b/src/tests/pass.rs
index e8fca62..283cef3 100644
--- a/src/tests/pass.rs
+++ b/src/tests/pass.rs
@@ -1919,9 +1919,8 @@ fn test_verify_gpg_id_files_untrusted_key_in_keyring() {
 
     assert!(result.is_err());
 
-    assert_eq!(
-        Error::GenericDyn("No valid signature".to_owned()),
-        result.err().unwrap()
+    assert!(
+        result.err().unwrap().to_string().contains("No valid signature")
     );
 }
 
