Last-Update: 2025-03-04
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,3 +84,3 @@
 [dependencies.sequoia-gpg-agent]
-version = "0.4.0"
+version = "0.5"
 
--- a/src/crypto.rs
+++ b/src/crypto.rs
@@ -528,3 +528,3 @@
                     let key = cert.primary_key();
-                    let mut pair = sequoia_gpg_agent::gnupg::KeyPair::new(
+                    let mut pair = sequoia_gpg_agent::KeyPair::new_for_gnupg_context(
                         self.ctx
