Index: curl/tests/multi.rs
===================================================================
--- curl.orig/tests/multi.rs
+++ curl/tests/multi.rs
@@ -84,7 +84,7 @@ fn smoke2() {
     assert_eq!(done, 2);
 }
 
-#[test]
+/*#[test]
 fn upload_lots() {
     use curl::multi::{Events, Socket, SocketEvents};
 
@@ -227,7 +227,7 @@ fn upload_lots() {
 
     let mut e = t!(m.remove(e));
     assert_eq!(t!(e.response_code()), 200);
-}
+}*/
 
 // Tests passing raw file descriptors to Multi::wait. The test is limited to Linux only as the
 // semantics of the underlying poll(2) system call used by curl apparently differ on other
