Index: curl/tests/protocols.rs
===================================================================
--- curl.orig/tests/protocols.rs
+++ curl/tests/protocols.rs
@@ -1,13 +1,3 @@
-#[cfg(all(feature = "static-curl", not(feature = "protocol-ftp")))]
-#[test]
-fn static_with_ftp_disabled() {
-    assert!(curl::Version::get()
-        .protocols()
-        .filter(|&p| p == "ftp")
-        .next()
-        .is_none());
-}
-
 #[cfg(all(feature = "static-curl", feature = "protocol-ftp"))]
 #[test]
 fn static_with_ftp_enabled() {
