Index: curl/Cargo.toml
===================================================================
--- curl.orig/Cargo.toml
+++ curl/Cargo.toml
@@ -78,7 +78,6 @@ version = "2.0.3"
 
 [features]
 default = ["ssl"]
-force-system-lib-on-osx = ["curl-sys/force-system-lib-on-osx"]
 http2 = ["curl-sys/http2"]
 ssl = [
     "openssl-sys",
@@ -96,16 +95,6 @@ optional = true
 version = "0.9.43"
 optional = true
 
-[target."cfg(target_env = \"msvc\")".dependencies.schannel]
-version = "0.1.13"
-
-[target."cfg(target_env = \"msvc\")".dependencies.winapi]
-version = "0.3"
-features = [
-    "libloaderapi",
-    "wincrypt",
-]
-
 [badges.appveyor]
 repository = "alexcrichton/curl-rust"
 
