Index: curl/tests/easy.rs
===================================================================
--- curl.orig/tests/easy.rs
+++ curl/tests/easy.rs
@@ -780,19 +780,7 @@ fn panic_in_callback() {
 #[test]
 fn abort_read() {
     let s = Server::new();
-    s.receive(
-        "\
-         PUT / HTTP/1.1\r\n\
-         Host: 127.0.0.1:$PORT\r\n\
-         Accept: */*\r\n\
-         Content-Length: 2\r\n\
-         \r\n",
-    );
-    s.send(
-        "\
-         HTTP/1.1 200 OK\r\n\
-         \r\n",
-    );
+    s.receive("");
 
     let mut h = handle();
     t!(h.url(&s.url("/")));
