--- a/Cargo.toml	2018-09-21 18:00:31.522682296 +0000
+++ b/Cargo.toml	2018-09-21 18:01:35.962553903 +0000
@@ -46,13 +46,7 @@
 
 [target."cfg(all(unix, not(target_os = \"macos\")))".dependencies.openssl-sys]
 version = "0.9.33"
-[target."cfg(target_env=\"msvc\")".dependencies.kernel32-sys]
-version = "0.2.2"
 
-[target."cfg(target_env=\"msvc\")".dependencies.schannel]
-version = "0.1.13"
-[target."cfg(windows)".dependencies.winapi]
-version = "0.2.7"
 [badges.appveyor]
 repository = "alexcrichton/curl-rust"
 
