--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,11 +52,6 @@
 name = "test_load"
 path = "tests/test_load.rs"
 
-[[bench]]
-name = "bench_cap_find"
-path = "benches/bench_cap_find.rs"
-harness = false
-required-features = ["img"]
 
 [dependencies.g2p]
 version = "1.0"
@@ -69,9 +64,6 @@
 [dependencies.lru]
 version = "0.12"
 
-[dev-dependencies.criterion]
-version = "0.5"
-
 [dev-dependencies.image]
 version = "0.25"
 
