Index: msedit/Cargo.toml
===================================================================
--- msedit.orig/Cargo.toml
+++ msedit/Cargo.toml
@@ -63,23 +63,6 @@ default-features = false
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59"
-features = [
-    "Win32_Globalization",
-    "Win32_Security",
-    "Win32_Storage_FileSystem",
-    "Win32_System_Console",
-    "Win32_System_Diagnostics_Debug",
-    "Win32_System_IO",
-    "Win32_System_LibraryLoader",
-    "Win32_System_Memory",
-    "Win32_System_Threading",
-]
-
-[target."cfg(windows)".build-dependencies.winresource]
-version = "0.1.22"
-
 [profile.bench]
 lto = "thin"
 codegen-units = 16
