Index: msedit/src/lib.rs
===================================================================
--- msedit.orig/src/lib.rs
+++ msedit/src/lib.rs
@@ -1,16 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-#![feature(
-    allocator_api,
-    breakpoint,
-    cold_path,
-    let_chains,
-    linked_list_cursors,
-    maybe_uninit_fill,
-    maybe_uninit_slice,
-    maybe_uninit_uninit_array_transpose
-)]
 #![allow(clippy::missing_transmute_annotations, clippy::new_without_default, stable_features)]
 
 #[macro_use]
