Index: msedit/src/bin/edit/main.rs
===================================================================
--- msedit.orig/src/bin/edit/main.rs
+++ msedit/src/bin/edit/main.rs
@@ -1,8 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-#![feature(allocator_api, let_chains, linked_list_cursors, string_from_utf8_lossy_owned)]
-
 mod documents;
 mod draw_editor;
 mod draw_filepicker;
