tIndex: parking-lot-core/Cargo.toml
===================================================================
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,9 +26,9 @@
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/Amanieu/parking_lot"
 
-[dependencies.backtrace]
-version = "0.3.60"
-optional = true
+#[dependencies.backtrace]
+#version = "0.3.60"
+#optional = true
 
 [dependencies.cfg-if]
 version = "1.0.0"
@@ -48,7 +48,7 @@
 deadlock_detection = [
     "petgraph",
     "thread-id",
-    "backtrace",
+    #"backtrace",
 ]
 nightly = []
 
