Index: parking-lot-core/Cargo.toml
===================================================================
--- parking-lot-core.orig/Cargo.toml
+++ parking-lot-core/Cargo.toml
@@ -44,7 +44,7 @@ nightly = []
 [target."cfg(target_os = \"cloudabi\")".dependencies.cloudabi]
 version = "0.0.3"
 [target."cfg(target_os = \"redox\")".dependencies.redox_syscall]
-version = "0.1"
+version = ">= 0.1, < 0.3"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.55"
 [target."cfg(windows)".dependencies.winapi]
