--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,20 +23,5 @@ readme = "README.md"
 keywords = ["text", "unicode", "script", "language"]
 license = "MIT/Apache-2.0"
 repository = "https://github.com/unicode-rs/unicode-script"
-[dependencies.compiler_builtins]
-version = "0.1"
-optional = true
-
-[dependencies.core]
-version = "1.0"
-optional = true
-package = "rustc-std-workspace-core"
-
-[dependencies.std]
-version = "1.0"
-optional = true
-package = "rustc-std-workspace-std"
-
 [features]
 bench = []
-rustc-dep-of-std = ["std", "core", "compiler_builtins"]
