Index: futures-core/src/task/__internal/atomic_waker.rs
===================================================================
--- futures-core.orig/src/task/__internal/atomic_waker.rs
+++ futures-core/src/task/__internal/atomic_waker.rs
@@ -42,7 +42,7 @@ use core::sync::atomic;
 /// Here is a simple example providing a `Flag` that can be signalled manually
 /// when it is ready.
 ///
-/// ```
+/// ``` ignore
 /// use futures::future::Future;
 /// use futures::task::{Context, Poll, AtomicWaker};
 /// use std::sync::Arc;
@@ -234,7 +234,7 @@ impl AtomicWaker {
     ///
     /// Here is how `register` is used when implementing a flag.
     ///
-    /// ```
+    /// ``` ignore
     /// use futures::future::Future;
     /// use futures::task::{Context, Poll, AtomicWaker};
     /// use std::sync::atomic::AtomicBool;
