Index: futures-core/Cargo.toml
===================================================================
--- futures-core.orig/Cargo.toml
+++ futures-core/Cargo.toml
@@ -29,10 +29,10 @@ rustdoc-args = [
     "docsrs",
 ]
 
-[dependencies.portable-atomic]
-version = "1"
-optional = true
-default-features = false
+#[dependencies.portable-atomic]
+#version = "1"
+#optional = true
+#default-features = false
 
 [dev-dependencies]
 
