Index: tracing-attributes/Cargo.toml
===================================================================
--- tracing-attributes.orig/Cargo.toml
+++ tracing-attributes/Cargo.toml
@@ -75,7 +75,6 @@ version = "0.1.35"
 
 [dev-dependencies.tracing-subscriber]
 version = "0.3.0"
-features = ["env-filter"]
 
 [dev-dependencies.trybuild]
 version = "1.0.64"
Index: tracing-attributes/tests/err.rs
===================================================================
--- tracing-attributes.orig/tests/err.rs
+++ tracing-attributes/tests/err.rs
@@ -5,1 +5,1 @@
-use tracing_subscriber::filter::EnvFilter;
+//use tracing_subscriber::filter::EnvFilter;
Index: tracing-attributes/tests/ret.rs
===================================================================
--- tracing-attributes.orig/tests/ret.rs
+++ tracing-attributes/tests/ret.rs
@@ -8,1 +8,1 @@
-use tracing_subscriber::EnvFilter;
+//use tracing_subscriber::EnvFilter;
