Index: tracing-attributes/Cargo.toml
===================================================================
--- tracing-attributes.orig/Cargo.toml
+++ tracing-attributes/Cargo.toml
@@ -74,10 +74,6 @@ name = "err"
 path = "tests/err.rs"
 
 [[test]]
-name = "fields"
-path = "tests/fields.rs"
-
-[[test]]
 name = "follows_from"
 path = "tests/follows_from.rs"
 
Index: tracing-attributes/tests/ui/pass/type_shadowing.rs
===================================================================
--- tracing-attributes.orig/tests/ui/pass/type_shadowing.rs
+++ tracing-attributes/tests/ui/pass/type_shadowing.rs
@@ -8,10 +8,10 @@ enum Bar {
     Foo,
 }
 
-#[tracing::instrument]
+/*#[tracing::instrument]
 fn this_is_fine() -> Foo {
     // glob import imports Bar::Foo, shadowing Foo
     use Bar::*;
-}
+}*/
 
 fn main() {}
