Index: tracing-attributes/tests/async_fn.rs
===================================================================
--- tracing-attributes.orig/tests/async_fn.rs
+++ tracing-attributes/tests/async_fn.rs
@@ -4,7 +4,7 @@ use std::{future::Future, pin::Pin, sync
 use tracing::subscriber::with_default;
 use tracing_attributes::instrument;
 use tracing_mock::{expect, subscriber};
-use tracing_test::{block_on_future, PollN};
+//use tracing_test::{block_on_future, PollN};
 
 #[instrument]
 async fn test_async_fn(polls: usize) -> Result<(), ()> {
Index: tracing-attributes/tests/err.rs
===================================================================
--- tracing-attributes.orig/tests/err.rs
+++ tracing-attributes/tests/err.rs
@@ -4,7 +4,7 @@ use tracing_attributes::instrument;
 use tracing_mock::*;
 //use tracing_subscriber::filter::EnvFilter;
 use tracing_subscriber::layer::SubscriberExt;
-use tracing_test::{block_on_future, PollN};
+//use tracing_test::{block_on_future, PollN};
 
 use std::convert::TryFrom;
 use std::num::TryFromIntError;
Index: tracing-attributes/tests/follows_from.rs
===================================================================
--- tracing-attributes.orig/tests/follows_from.rs
+++ tracing-attributes/tests/follows_from.rs
@@ -1,7 +1,7 @@
 use tracing::{subscriber::with_default, Id, Level, Span};
 use tracing_attributes::instrument;
 use tracing_mock::{expect, subscriber};
-use tracing_test::block_on_future;
+//use tracing_test::block_on_future;
 
 #[instrument(follows_from = causes, skip(causes))]
 fn with_follows_from_sync(causes: impl IntoIterator<Item = impl Into<Option<Id>>>) {}
Index: tracing-attributes/tests/ret.rs
===================================================================
--- tracing-attributes.orig/tests/ret.rs
+++ tracing-attributes/tests/ret.rs
@@ -6,7 +6,7 @@ use tracing_attributes::instrument;
 use tracing_mock::{expect, subscriber};
 use tracing_subscriber::layer::SubscriberExt;
 //use tracing_subscriber::EnvFilter;
-use tracing_test::block_on_future;
+//use tracing_test::block_on_future;
 
 #[instrument(ret)]
 fn ret() -> i32 {
