Index: async-compression/Cargo.toml
===================================================================
--- async-compression.orig/Cargo.toml
+++ async-compression/Cargo.toml
@@ -179,7 +179,7 @@ version = ">= 6, < 9"
 optional = true
 
 [dependencies.bzip2]
-version = "0.6"
+version = ">= 0.4, < 0.7"
 optional = true
 
 [dependencies.deflate64]
@@ -246,10 +246,10 @@ version = "0.9"
 version = "1"
 
 [dev-dependencies.proptest-derive]
-version = "0.6"
+version = "0.4"
 
 [dev-dependencies.rand]
-version = "0.9"
+version = ">= 0.8, < 0.10"
 
 [dev-dependencies.tokio]
 version = "1.38.2"
