diff --git a/Cargo.toml b/Cargo.toml
index 0e6828c..8202722 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,8 +34,8 @@ keywords = [
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/drakulix/simplelog.rs"
 
-[dependencies.ansi_term]
-version = "0.12"
+[dependencies.nu-ansi-term]
+version = "0.50"
 optional = true
 
 [dependencies.log]
diff --git a/src/loggers/logging.rs b/src/loggers/logging.rs
index a09bff3..3d06392 100644
--- a/src/loggers/logging.rs
+++ b/src/loggers/logging.rs
@@ -7,16 +7,16 @@ use std::thread;
 use termcolor::Color;
 
 #[cfg(all(feature = "termcolor", feature = "ansi_term"))]
-pub fn termcolor_to_ansiterm(color: &Color) -> Option<ansi_term::Color> {
+pub fn termcolor_to_ansiterm(color: &Color) -> Option<nu_ansi_term::Color> {
     match color {
-        Color::Black => Some(ansi_term::Color::Black),
-        Color::Red => Some(ansi_term::Color::Red),
-        Color::Green => Some(ansi_term::Color::Green),
-        Color::Yellow => Some(ansi_term::Color::Yellow),
-        Color::Blue => Some(ansi_term::Color::Blue),
-        Color::Magenta => Some(ansi_term::Color::Purple),
-        Color::Cyan => Some(ansi_term::Color::Cyan),
-        Color::White => Some(ansi_term::Color::White),
+        Color::Black => Some(nu_ansi_term::Color::Black),
+        Color::Red => Some(nu_ansi_term::Color::Red),
+        Color::Green => Some(nu_ansi_term::Color::Green),
+        Color::Yellow => Some(nu_ansi_term::Color::Yellow),
+        Color::Blue => Some(nu_ansi_term::Color::Blue),
+        Color::Magenta => Some(nu_ansi_term::Color::Purple),
+        Color::Cyan => Some(nu_ansi_term::Color::Cyan),
+        Color::White => Some(nu_ansi_term::Color::White),
         _ => None,
     }
 }
