Index: memchr/Cargo.toml
===================================================================
--- memchr.orig/Cargo.toml
+++ memchr/Cargo.toml
@@ -56,7 +56,6 @@ alloc = []
 default = ["std"]
 libc = []
 logging = ["dep:log"]
-rustc-dep-of-std = ["core"]
 std = ["alloc"]
 use_std = ["std"]
 
@@ -65,11 +64,6 @@ name = "memchr"
 path = "src/lib.rs"
 bench = false
 
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [dependencies.log]
 version = "0.4.20"
 optional = true
