Description: relax package depdendency versions
Author: Ananthu C V <weepingclown@disroot.org>
Author: Peter Michael Green <plugwash@debian.org>
Forwarded: not-needed
Last-Update: 2024-12-27
Index: miette/Cargo.toml
===================================================================
--- miette.orig/Cargo.toml
+++ miette/Cargo.toml
@@ -60,7 +60,7 @@ version = "=7.4.0"
 optional = true
 
 [dependencies.owo-colors]
-version = "4.0.0"
+version = ">= 3.5.0"
 optional = true
 
 [dependencies.serde]
