Index: num-format/Cargo.toml
===================================================================
--- num-format.orig/Cargo.toml
+++ num-format/Cargo.toml
@@ -74,21 +74,21 @@ optional = true
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
 optional = true
-[target."cfg(windows)".dependencies.num-format-windows]
-version = "0.3"
-optional = true
+# # [target."cfg(windows)".dependencies.num-format-windows]
+# # version = "0.3"
+# optional = true
 
-[target."cfg(windows)".dependencies.widestring]
-version = "0.4"
-optional = true
+# [target."cfg(windows)".dependencies.widestring]
+# version = "0.4"
+# optional = true
 
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = ["winnls"]
-optional = true
-[badges.maintenance]
-status = "actively-developed"
+# [target."cfg(windows)".dependencies.winapi]
+# version = "0.3"
+# features = ["winnls"]
+# optional = true
+# [badges.maintenance]
+# status = "actively-developed"
 
-[badges.travis-ci]
-branch = "master"
-repository = "bcmyers/num-format"
+# [badges.travis-ci]
+# branch = "master"
+# repository = "bcmyers/num-format"
