Index: num-format/Cargo.toml
===================================================================
--- num-format.orig/Cargo.toml
+++ num-format/Cargo.toml
@@ -61,7 +61,7 @@ default = ["std"]
 std = ["arrayvec/default"]
 with-num-bigint = ["num-bigint", "std"]
 with-serde = ["serde/derive"]
-with-system-locale = ["cfg-if", "encoding_rs", "lazy_static", "libc"]
+with-system-locale = ["cfg-if", "encoding_rs", "lazy_static", "libc", "std"]
 [target."cfg(unix)".dependencies.cfg-if]
 version = "0.1"
 optional = true
