Index: num-format/Cargo.toml
===================================================================
--- num-format.orig/Cargo.toml
+++ num-format/Cargo.toml
@@ -29,8 +29,7 @@ version = "0.7"
 default-features = false
 
 [dependencies.itoa]
-version = "0.4"
-features = ["i128"]
+version = "1"
 default-features = false
 
 [dependencies.lazy_static]
@@ -38,7 +37,7 @@ version = "1.2"
 optional = true
 
 [dependencies.num-bigint]
-version = "0.2"
+version = "0.4"
 optional = true
 
 [dependencies.serde]
@@ -52,14 +51,14 @@ version = "0.1"
 version = "1.2"
 
 [dev-dependencies.rand]
-version = "0.6"
+version = "0.8"
 
 [dev-dependencies.serde_json]
 version = "1.0"
 
 [features]
 default = ["std"]
-std = ["arrayvec/default", "itoa/default", "itoa/i128"]
+std = ["arrayvec/default"]
 with-num-bigint = ["num-bigint", "std"]
 with-serde = ["serde/derive"]
 with-system-locale = ["cfg-if", "encoding_rs", "lazy_static", "libc"]
