diff --git a/tests/roundtrip/test_primitives.rs b/tests/roundtrip/test_primitives.rs
index 58fbd77..14f1425 100644
--- a/tests/roundtrip/test_primitives.rs
+++ b/tests/roundtrip/test_primitives.rs
@@ -17,9 +17,13 @@ macro_rules! test_primitive {
 
 test_primitive!(test_isize_neg, -100isize, isize);
 test_primitive!(test_isize_pos, 100isize, isize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_isize_min, isize::min_value(), isize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_isize_max, isize::max_value(), isize);
 
 test_primitive!(test_usize, 100usize, usize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_usize_min, usize::min_value(), usize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_usize_max, usize::max_value(), usize);
diff --git a/tests/schema/container_extension/test_max_size.rs b/tests/schema/container_extension/test_max_size.rs
index d179b0e..b096a7d 100644
--- a/tests/schema/container_extension/test_max_size.rs
+++ b/tests/schema/container_extension/test_max_size.rs
@@ -52,6 +52,7 @@ fn max_serialized_size_primitives() {
 }
 
 #[test]
+#[cfg(target_pointer_width = "64")]
 fn max_serialized_size_built_in_types() {
     test_ok::<core::ops::RangeFull>(0);
     test_ok::<core::ops::RangeInclusive<u8>>(2);
@@ -83,6 +84,7 @@ fn max_serialized_size_built_in_types() {
 }
 
 #[test]
+#[cfg(target_pointer_width = "64")]
 fn max_serialized_size_derived_types() {
     #[derive(BorshSchema)]
     pub struct Empty;
