diff --git a/tests/roundtrip/test_primitives.rs b/tests/roundtrip/test_primitives.rs
index 58fbd77..14f1425 100644
--- a/tests/roundtrip/test_primitives.rs
+++ b/tests/roundtrip/test_primitives.rs
@@ -17,9 +17,13 @@ macro_rules! test_primitive {
 
 test_primitive!(test_isize_neg, -100isize, isize);
 test_primitive!(test_isize_pos, 100isize, isize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_isize_min, isize::min_value(), isize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_isize_max, isize::max_value(), isize);
 
 test_primitive!(test_usize, 100usize, usize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_usize_min, usize::min_value(), usize);
+#[cfg(target_pointer_width = "64")]
 test_primitive!(test_usize_max, usize::max_value(), usize);
