diff --git a/Cargo.toml b/Cargo.toml
index babacc2..01e0fa8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -80,7 +80,7 @@ version = "1.0"
 optional = true
 
 [dependencies.loom]
-version = "0.7"
+version = "0.5"
 optional = true
 
 [dependencies.saa]
@@ -94,13 +94,13 @@ version = "1.0"
 optional = true
 
 [dev-dependencies.criterion]
-version = "0.7"
+version = "0.5"
 
 [dev-dependencies.futures]
 version = "0.3"
 
 [dev-dependencies.proptest]
-version = "1.7"
+version = "1"
 
 [dev-dependencies.serde_test]
 version = "1.0"
