Index: ntpd/src/daemon/keyexchange.rs
===================================================================
--- ntpd.orig/src/daemon/keyexchange.rs
+++ ntpd/src/daemon/keyexchange.rs
@@ -617,7 +617,7 @@ fn certificates_from_bufread(mut reader:
     ntp_proto::tls_utils::pemfile::certs(&mut reader).collect()
 }
 
-#[cfg(test)]
+/*#[cfg(test)]
 mod tests {
     use std::{io::Cursor, net::SocketAddr, path::PathBuf};
 
@@ -1349,4 +1349,4 @@ mod tests {
         let error = server_error_record(NtsRecord::INTERNAL_SERVER_ERROR).await;
         assert!(matches!(error, KEE::InternalServerError));
     }
-}
+}*/
