Index: ntpd/Cargo.toml
===================================================================
--- ntpd.orig/Cargo.toml
+++ ntpd/Cargo.toml
@@ -233,15 +233,15 @@ name = "ntpd"
 path = "src/lib.rs"
 
 [[bin]]
-name = "ntp-daemon"
+name = "ntp-rs-daemon"
 path = "bin/ntp-daemon.rs"
 
 [[bin]]
-name = "ntp-ctl"
+name = "ntp-rs-ctl"
 path = "bin/ntp-ctl.rs"
 
 [[bin]]
-name = "ntp-metrics-exporter"
+name = "ntp-rs-metrics-exporter"
 path = "bin/ntp-metrics-exporter.rs"
 
 [dependencies.async-trait]
