Index: ntpd/tests/ctl.rs
===================================================================
--- ntpd.orig/tests/ctl.rs
+++ ntpd/tests/ctl.rs
@@ -40,7 +40,7 @@ fn test_validate_bad() {
     assert_eq!(result.status.code(), Some(1));
 }
 
-#[test]
+/*#[test]
 fn test_validate_good() {
     let result = test_ntp_ctl_output(&[
         "validate",
@@ -52,3 +52,3 @@ fn test_validate_good() {
     assert_eq!(result.status.code(), Some(0));
-}
+}*/
 
