Description: Annotate built-using
Forwarded: not-needed
Last-Update: 2020-12-18
--- a/build.rs
+++ b/build.rs
@@ -212,2 +212,3 @@
     println!("cargo:rustc-cfg=blake3_avx2_ffi");
+    println!("dh-cargo:deb-built-using=blake3_sse2_sse41_avx2_assembly=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
     let mut build = new_build();
@@ -238,2 +239,3 @@
     println!("cargo:rustc-cfg=blake3_avx512_ffi");
+    println!("dh-cargo:deb-built-using=blake3_avx512_intrinsics=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
     let mut build = new_build();
@@ -258,2 +260,3 @@
     println!("cargo:rustc-cfg=blake3_avx512_ffi");
+    println!("dh-cargo:deb-built-using=blake3_avx512_assembly=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
     let mut build = new_build();
@@ -293,2 +296,3 @@
     build.compile("blake3_neon");
+    println!("dh-cargo:deb-built-using=blake3_neon=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
 }
