--- a/Cargo.toml
+++ b/Cargo.toml
@@ -71,12 +71,6 @@
 version = "1.0"
 optional = true
 
-[dependencies.rkyv]
-version = "0.7.42"
-features = ["alloc"]
-optional = true
-default-features = false
-
 [dependencies.serde]
 version = "1.0.25"
 optional = true
@@ -102,10 +96,6 @@
 [dev-dependencies.rayon]
 version = "1.0"
 
-[dev-dependencies.rkyv]
-version = "0.7.42"
-features = ["validation"]
-
 [dev-dependencies.serde_test]
 version = "1.0"
 
