--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,3 +64,3 @@
 [dependencies.dirs]
-version = "6"
+version = "5"
 
@@ -70,3 +70,3 @@
 [dependencies.fd-lock]
-version = "4.0.0"
+version = "3"
 
@@ -112,3 +112,3 @@
 [dependencies.ruzstd]
-version = "0.8"
+version = "0.7"
 
@@ -116,8 +116,2 @@
 version = "2"
-features = [
-    "crypto-rust",
-    "allow-experimental-crypto",
-    "allow-variable-time-crypto",
-]
-default-features = false
 
--- a/src/pkgs/archlinux.rs
+++ b/src/pkgs/archlinux.rs
@@ -2,7 +2,7 @@ use crate::container::Container;
 use crate::errors::*;
 use crate::pkgs::Pkg;
 use peekread::{BufPeekReader, PeekRead};
-use ruzstd::decoding::StreamingDecoder;
+use ruzstd::StreamingDecoder;
 use std::fmt::Write;
 use std::io::{BufRead, BufReader, Read};
 use std::time::SystemTime;
