Index: repro-env/Cargo.toml
===================================================================
--- repro-env.orig/Cargo.toml
+++ repro-env/Cargo.toml
@@ -62,13 +62,13 @@ version = "0.1.0"
 version = "2.4.0"
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
 [dependencies.env_logger]
 version = "0.11"
 
 [dependencies.fd-lock]
-version = "4.0.0"
+version = "3"
 
 [dependencies.flate2]
 version = "1.0.26"
@@ -101,7 +101,7 @@ default-features = false
 version = "0.1.1"
 
 [dependencies.reqwest]
-version = "0.12"
+version = "0.12.12"
 features = [
     "rustls-tls-native-roots",
     "socks",
@@ -110,16 +110,10 @@ features = [
 default-features = false
 
 [dependencies.ruzstd]
-version = "0.8"
+version = "0.7"
 
 [dependencies.sequoia-openpgp]
 version = "2"
-features = [
-    "crypto-rust",
-    "allow-experimental-crypto",
-    "allow-variable-time-crypto",
-]
-default-features = false
 
 [dependencies.serde]
 version = "1"
Index: repro-env/src/pkgs/archlinux.rs
===================================================================
--- repro-env.orig/src/pkgs/archlinux.rs
+++ repro-env/src/pkgs/archlinux.rs
@@ -2,7 +2,7 @@ use crate::container::Container;
 use crate::errors::*;
 use crate::pkgs::Pkg;
 use peekread::{BufPeekReader, PeekRead};
-use ruzstd::decoding::StreamingDecoder;
+use ruzstd::StreamingDecoder;
 use std::fmt::Write;
 use std::io::{BufRead, BufReader, Read};
 use std::time::SystemTime;
