--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,16 +21,16 @@
 repository = "https://github.com/kpcyrd/repro-env"
 
 [dependencies.anyhow]
-version = "1.0.71"
+version = "1"
 
 [dependencies.ar]
 version = "0.9.0"
 
 [dependencies.bytes]
-version = "1.4.0"
+version = "1.2"
 
 [dependencies.clap]
-version = "4.3.8"
+version = "4.1"
 features = ["derive"]
 
 [dependencies.clap_complete]
@@ -40,16 +40,16 @@
 version = "0.1.0"
 
 [dependencies.data-encoding]
-version = "2.4.0"
+version = "2.3"
 
 [dependencies.dirs]
-version = "5.0.1"
+version = "4"
 
 [dependencies.env_logger]
-version = "0.10"
+version = "0.9"
 
 [dependencies.flate2]
-version = "1.0.26"
+version = "1.0.25"
 
 [dependencies.hex]
 version = "0.4.3"
@@ -75,10 +75,9 @@
 version = "0.1.1"
 
 [dependencies.reqwest]
-version = "0.11.18"
+version = "0.11.14"
 features = [
     "rustls-tls-native-roots",
-    "tokio-socks",
 ]
 default-features = false
 
@@ -93,10 +92,10 @@
 version = "1"
 
 [dependencies.sha1]
-version = "0.10.5"
+version = "0.10"
 
 [dependencies.sha2]
-version = "0.10.7"
+version = "0.10"
 
 [dependencies.tar]
 version = "0.4.38"
@@ -115,7 +114,7 @@
 ]
 
 [dependencies.toml]
-version = "0.7.5"
+version = "0.5"
 
 [dependencies.urlencoding]
-version = "2.1.2"
+version = "2.1"
