--- a/Cargo.toml
+++ b/Cargo.toml
@@ -59,7 +59,7 @@
 version = "0.10.1"
 
 [dependencies.clap]
-version = "4.5.20"
+version = "4.5.16"
 features = ["derive"]
 
 [dependencies.color-eyre]
@@ -69,10 +69,10 @@
 version = "0.4.4"
 
 [dependencies.crossterm]
-version = "0.28.1"
+version = ">= 0.27.0, < 0.30"
 
 [dependencies.data-encoding]
-version = "2.6.0"
+version = "2.5.0"
 
 [dependencies.derive_builder]
 version = "0.20.1"
@@ -84,7 +84,7 @@
 version = "0.3.13"
 
 [dependencies.getrandom]
-version = "0.2.15"
+version = "0.2.12"
 
 [dependencies.hex]
 version = "0.4.3"
@@ -99,15 +99,16 @@
 version = "0.14.1"
 
 [dependencies.ratatui]
-version = "0.29.0"
+version = "0.28.1"
 features = ["all-widgets"]
 
 [dependencies.rpassword]
-version = "7.3.1"
+version = "7.2.0"
 
 [dependencies.rust-argon2]
 version = "2.1.0"
 
+
 [dependencies.scrypt]
 version = "0.11.0"
 
--- a/src/interface/app.rs
+++ b/src/interface/app.rs
@@ -257,7 +257,7 @@
                     .borders(Borders::TOP | Borders::BOTTOM)
                     .title(self.title.as_str()),
             )
-            .row_highlight_style(
+            .highlight_style(
                 Style::default()
                     .bg(Color::White)
                     .fg(Color::Black)
