Index: proc-macro-hack/Cargo.toml
===================================================================
--- proc-macro-hack.orig/Cargo.toml
+++ proc-macro-hack/Cargo.toml
@@ -25,11 +25,6 @@ targets = ["x86_64-unknown-linux-gnu"]
 
 [lib]
 proc-macro = true
-[dev-dependencies.demo-hack]
-version = "0.0.5"
-
-[dev-dependencies.demo-hack-impl]
-version = "0.0.5"
 
 [dev-dependencies.quote]
 version = "1.0"
Index: proc-macro-hack/src/lib.rs
===================================================================
--- proc-macro-hack.orig/src/lib.rs
+++ proc-macro-hack/src/lib.rs
@@ -68,7 +68,7 @@
 //! procedural macro from the implementation crate. The re-export also carries a
 //! \#\[proc_macro_hack\] attribute.
 //!
-//! ```
+//! ```ignore
 //! use proc_macro_hack::proc_macro_hack;
 //!
 //! /// Add one to an expression.
@@ -102,7 +102,7 @@
 //!
 //! [&raquo; example of a downstream crate][example]
 //!
-//! ```
+//! ```ignore
 //! use demo_hack::add_one;
 //!
 //! fn main() {
