Index: tracing/Cargo.toml
===================================================================
--- tracing.orig/Cargo.toml
+++ tracing/Cargo.toml
@@ -62,38 +62,10 @@ name = "tracing"
 path = "src/lib.rs"
 
 [[test]]
-name = "enabled"
-path = "tests/enabled.rs"
-
-[[test]]
-name = "event"
-path = "tests/event.rs"
-
-[[test]]
-name = "filter_caching_is_lexically_scoped"
-path = "tests/filter_caching_is_lexically_scoped.rs"
-
-[[test]]
-name = "filters_are_not_reevaluated_for_the_same_span"
-path = "tests/filters_are_not_reevaluated_for_the_same_span.rs"
-
-[[test]]
-name = "filters_are_reevaluated_for_different_call_sites"
-path = "tests/filters_are_reevaluated_for_different_call_sites.rs"
-
-[[test]]
-name = "filters_dont_leak"
-path = "tests/filters_dont_leak.rs"
-
-[[test]]
 name = "future_send"
 path = "tests/future_send.rs"
 
 [[test]]
-name = "instrument"
-path = "tests/instrument.rs"
-
-[[test]]
 name = "macro_imports"
 path = "tests/macro_imports.rs"
 
@@ -105,38 +77,6 @@ path = "tests/macros.rs"
 name = "macros_incompatible_concat"
 path = "tests/macros_incompatible_concat.rs"
 
-[[test]]
-name = "max_level_hint"
-path = "tests/max_level_hint.rs"
-
-[[test]]
-name = "missed_register_callsite"
-path = "tests/missed_register_callsite.rs"
-
-[[test]]
-name = "multiple_max_level_hints"
-path = "tests/multiple_max_level_hints.rs"
-
-[[test]]
-name = "no_subscriber"
-path = "tests/no_subscriber.rs"
-
-[[test]]
-name = "register_callsite_deadlock"
-path = "tests/register_callsite_deadlock.rs"
-
-[[test]]
-name = "scoped_clobbers_default"
-path = "tests/scoped_clobbers_default.rs"
-
-[[test]]
-name = "span"
-path = "tests/span.rs"
-
-[[test]]
-name = "subscriber"
-path = "tests/subscriber.rs"
-
 [[bench]]
 name = "baseline"
 path = "benches/baseline.rs"
