Index: tracing/Cargo.toml
===================================================================
--- tracing.orig/Cargo.toml
+++ tracing/Cargo.toml
@@ -105,7 +105,6 @@ release_max_level_off = []
 release_max_level_trace = []
 release_max_level_warn = []
 std = ["tracing-core/std"]
-valuable = ["tracing-core/valuable"]
 
 [target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
 version = "^0.3"
