Index: tracing/Cargo.toml
===================================================================
--- tracing.orig/Cargo.toml
+++ tracing/Cargo.toml
@@ -74,2 +74,3 @@
 name = "dispatch_get_clone"
+required-features = ["std"]
 path = "benches/dispatch_get_clone.rs"
@@ -80,2 +80,3 @@
 name = "dispatch_get_ref"
+required-features = ["std"]
 path = "benches/dispatch_get_ref.rs"
@@ -90,2 +90,3 @@
 name = "empty_span"
+required-features = ["std"]
 path = "benches/empty_span.rs"
@@ -100,2 +11,3 @@
 name = "enter_span"
+required-features = ["std"]
 path = "benches/enter_span.rs"
@@ -110,2 +110,3 @@
 name = "event"
+required-features = ["std"]
 path = "benches/event.rs"
@@ -120,2 +120,3 @@
 name = "span_fields"
+required-features = ["std"]
 path = "benches/span_fields.rs"
@@ -130,2 +130,3 @@
 name = "span_no_fields"
+required-features = ["std"]
 path = "benches/span_no_fields.rs"
@@ -140,2 +140,3 @@
 name = "span_repeated"
+required-features = ["std"]
 path = "benches/span_repeated.rs"
Index: tracing/benches/shared.rs
===================================================================
--- tracing.orig/benches/shared.rs
+++ tracing/benches/shared.rs
@@ -7,6 +7,7 @@ use std::{
     sync::{Mutex, MutexGuard},
 };
 
+#[cfg(feature = "std")]
 pub fn for_all_recording(
     group: &mut criterion::BenchmarkGroup<'_, WallTime>,
     mut iter: impl FnMut(&mut Bencher<'_, WallTime>),
@@ -32,6 +33,7 @@ pub fn for_all_recording(
     group.bench_function("global", &mut iter);
 }
 
+#[cfg(feature = "std")]
 pub fn for_all_dispatches(
     group: &mut criterion::BenchmarkGroup<'_, WallTime>,
     mut iter: impl FnMut(&mut Bencher<'_, WallTime>),
