Description: relaxing dependencies
Author: Ed Neville <ed-debian@s5h.net>
Origin: other
Forwarded: not-needed
Last-Update: 2021-12-23

--- rust-syslog-6.0.1.orig/Cargo.toml
+++ rust-syslog-6.0.1/Cargo.toml
@@ -19,19 +19,19 @@ keywords = ["syslog", "logs", "logging"]
 license = "MIT"
 repository = "https://github.com/Geal/rust-syslog"
 [dependencies.error-chain]
-version = "0.12.2"
+version = "0.12"
 default-features = false
 
 [dependencies.hostname]
-version = "0.3.1"
+version = "0.3"
 
 [dependencies.libc]
-version = "0.2.112"
+version = "0.2"
 
 [dependencies.log]
-version = "0.4.8"
+version = "0.4"
 features = ["std"]
 
 [dependencies.time]
-version = "0.3.5"
+version = "0.3"
 features = ["local-offset", "formatting"]
