Description: Disable failing test on arm64/ppc64le
Bug: https://github.com/tomaka/shared_library/issues/17
Last-Update: 2022-05-14
---
--- a/src/dynamic_library.rs
+++ b/src/dynamic_library.rs
@@ -164,7 +164,15 @@ mod test {
     use std::path::Path;
 
     #[test]
-    #[cfg_attr(any(windows, target_os = "android"), ignore)] // FIXME #8818, #10379
+    #[cfg_attr(
+        any(
+            windows,
+            target_os = "android",
+            target_arch = "aarch64",
+            target_arch = "powerpc64"
+        ),
+        ignore
+    )] // FIXME #8818, #10379
     fn test_loading_cosine() {
         // The math library does not need to be loaded since it is already
         // statically linked in
