--- a/build.rs
+++ b/build.rs
@@ -150,6 +150,7 @@
     } else {
         println!("cargo:rustc-link-search={out_dir}");
     }
+    println!("dh-cargo:deb-built-using=avscasm=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
     println!("cargo:rustc-link-lib=static=avscasm");
     rerun_dir("src/asm/x86");
 }
@@ -218,8 +219,9 @@
         std::fs::write(hash_path, &hash[..]).unwrap();
     } else {
         println!("cargo:rustc-link-search={out_dir}");
-        println!("cargo:rustc-link-lib=static=avsc-aarch64");
     }
+    println!("dh-cargo:deb-built-using=avsc-aarch64=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("cargo:rustc-link-lib=static=avsc-aarch64");
     rerun_dir("src/asm/arm");
 }
 
