--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,29 +32,10 @@
     "cc",
     "libc",
 ]
-binary = [
-    "clap",
-    "serialize",
-]
 default = [
-    "binary",
     "asm",
 ]
-devel = [
-    "console",
-    "fern",
-]
-ffmpeg = ["ffmpeg-the-third"]
 libc = ["dep:libc"]
-serialize = [
-    "serde",
-    "serde_json",
-]
-tracing = [
-    "tracing-subscriber",
-    "tracing-chrome",
-    "dep:tracing",
-]
 
 [lib]
 name = "av_scenechange"
@@ -77,28 +58,6 @@
 [dependencies.arrayvec]
 version = "0.7.6"
 
-[dependencies.clap]
-version = "4.0.22"
-features = ["derive"]
-optional = true
-
-[dependencies.console]
-version = "0.15"
-optional = true
-
-[dependencies.fern]
-version = "0.7"
-optional = true
-
-[dependencies.ffmpeg-the-third]
-version = "3.0.0"
-features = [
-    "codec",
-    "format",
-]
-optional = true
-default-features = false
-
 [dependencies.libc]
 version = "0.2.172"
 optional = true
@@ -119,43 +78,12 @@
 [dependencies.rayon]
 version = "1.10.0"
 
-[dependencies.serde]
-version = "1.0.123"
-features = ["derive"]
-optional = true
-
-[dependencies.serde_json]
-version = "1.0.62"
-optional = true
-
 [dependencies.thiserror]
 version = "2.0.12"
 
-[dependencies.tracing]
-version = "0.1.40"
-optional = true
-
-[dependencies.tracing-chrome]
-version = "0.7.1"
-optional = true
-
-[dependencies.tracing-subscriber]
-version = "0.3.18"
-optional = true
-
 [dependencies.v_frame]
 version = "0.3.8"
 
-[dependencies.vapoursynth]
-version = "0.4.0"
-features = [
-    "vsscript-functions",
-    "vapoursynth-functions",
-    "vapoursynth-api-32",
-    "vsscript-api-31",
-]
-optional = true
-
 [dependencies.y4m]
 version = "0.8.0"
 
