Index: sequoia-autocrypt/Cargo.toml
===================================================================
--- sequoia-autocrypt.orig/Cargo.toml
+++ sequoia-autocrypt/Cargo.toml
@@ -72,11 +72,3 @@ features = [
     "__implicit-crypto-backend-for-tests",
 ]
 default-features = false
-
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "2"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
