Description: relax depdendency versions
Author: Ananthu C V <weepingclown@disroot.org>
Forwarded: not-needed
Last-Update: 2024-09-22

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,19 +35,19 @@
 repository = "https://github.com/http-rs/async-h1"
 
 [dependencies.async-channel]
-version = "1.5.1"
+version = "2.3.1"
 
 [dependencies.async-dup]
 version = "1.2.2"
 
 [dependencies.async-global-executor]
-version = "2.3.1"
+version = "3.0"
 
 [dependencies.async-io]
-version = "1.13.0"
+version = "2.3.3"
 
 [dependencies.futures-lite]
-version = "1.13.0"
+version = "2.3"
 
 [dependencies.http-types]
 version = "2.9.0"
@@ -67,4 +67,4 @@
 features = ["attributes"]
 
 [dev-dependencies.pretty_assertions]
-version = "0.6.1"
+version = "1.4.0"
