--- a/build.rs
+++ b/build.rs
@@ -1,4 +1,6 @@
 fn main() {
+    use std::env;
+
     let mut build = cc::Build::new();
 
     build
@@ -62,4 +64,6 @@
     println!("cargo:rerun-if-changed=include/simsimd/probability.h");
     println!("cargo:rerun-if-changed=include/simsimd/binary.h");
     println!("cargo:rerun-if-changed=include/simsimd/types.h");
+
+    println!("dh-cargo:deb-built-using=simsimd=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
 }
