--- a/build.rs
+++ b/build.rs
@@ -1,4 +1,5 @@
 fn main() -> Result<(), cc::Error> {
+    use std::env;
     let mut build = cc::Build::new();
 
     build
@@ -66,5 +67,7 @@
     println!("cargo:rerun-if-changed=include/simsimd/probability.h");
     println!("cargo:rerun-if-changed=include/simsimd/binary.h");
     println!("cargo:rerun-if-changed=include/simsimd/types.h");
+
+    println!("dh-cargo:deb-built-using=simsimd=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
     Ok(())
 }
