Index: crypto-common/Cargo.toml
===================================================================
--- crypto-common.orig/Cargo.toml
+++ crypto-common/Cargo.toml
@@ -40,12 +40,12 @@ version = "0.14.4"
 features = ["more_lengths"]
 
 [dependencies.rand_core]
-version = "0.6"
+version = "0.9"
 optional = true
 
 [dependencies.typenum]
 version = "1.14"
 
 [features]
-getrandom = ["rand_core/getrandom"]
+getrandom = ["rand_core/os_rng"]
 std = []
