--- a/Cargo.toml
+++ b/Cargo.toml
@@ -63,22 +63,22 @@
 name = "search_strings"
 path = "tests/search_strings.rs"
 
-[[bench]]
+[[disable.bench]]
 name = "composite"
 path = "benches/composite.rs"
 harness = false
 
-[[bench]]
+[[disable.bench]]
 name = "fuzzy"
 path = "benches/fuzzy.rs"
 harness = false
 
-[[bench]]
+[[disable.bench]]
 name = "path_normalization"
 path = "benches/path_normalization.rs"
 harness = false
 
-[[bench]]
+[[disable.bench]]
 name = "toks"
 path = "benches/toks.rs"
 harness = false
@@ -244,7 +244,7 @@
 version = "0.5"
 optional = true
 
-[dev-dependencies.glassbench]
+[disable.dev-dependencies.glassbench]
 version = "0.4.4"
 
 [build-dependencies.clap]
