--- a/Cargo.toml
+++ b/Cargo.toml
@@ -174,7 +174,7 @@
 version = "3.2"
 
 [dependencies.termimad]
-version = "0.30.1"
+version = "0.29"
 
 [dependencies.terminal-clipboard]
 version = "0.4.1"
--- a/src/command/panel_input.rs
+++ b/src/command/panel_input.rs
@@ -65,8 +65,8 @@
             area.left += 1;
         }
         self.input_field.set_area(area);
-        let cursor_pos = self.input_field.display_on(w)?;
-        Ok(cursor_pos)
+        let _cursor_pos: () = self.input_field.display_on(w)?;
+        Ok(None)
     }
 
     /// consume the event to
