--- a/man/page
+++ b/man/page
@@ -152,9 +152,6 @@
 .TP
 .SH BUGS
 .PP
-.B broot
-is known to be slow on most \fIWindows\fR installations.
-.PP
 On unix and mac platforms, most problems you may encounter are related to some terminals or terminal multiplexers which either intercepts some standard TTY instructions or break buffering or size querying. The list of shortcuts you can define in the config file is thus dependent of your system.
 .SH AUTHOR
 .B broot
