--- a/Cargo.toml
+++ b/Cargo.toml
@@ -109,10 +109,10 @@
 version = "1.4"
 
 [dependencies.cli-log]
-version = "2.1"
+version = "2"
 
 [dependencies.crokey]
-version = "1.2"
+version = "1.0"
 
 [dependencies.custom_error]
 version = "1.6"
@@ -121,7 +121,7 @@
 version = "2.2.3"
 
 [dependencies.directories]
-version = "4.0"
+version = "5"
 
 [dependencies.file-size]
 version = "1.0.3"
@@ -130,7 +130,7 @@
 version = "0.1"
 
 [dependencies.git2]
-version = "0.20"
+version = ">= 0.19, < 0.21"
 default-features = false
 
 [dependencies.glob]
@@ -152,7 +152,7 @@
 version = "0.2"
 
 [dependencies.lru]
-version = "0.16"
+version = "0.12"
 
 [dependencies.memmap2]
 version = "0.9"
@@ -161,13 +161,13 @@
 version = "1.18"
 
 [dependencies.opener]
-version = "0.8"
+version = "0.6"
 
 [dependencies.pathdiff]
 version = "0.2"
 
 [dependencies.phf]
-version = "0.13"
+version = "0.11"
 features = ["macros"]
 
 [dependencies.rayon]
@@ -209,7 +209,7 @@
 version = "3.2"
 
 [dependencies.termimad]
-version = "0.34"
+version = "0.33"
 
 [dependencies.terminal-clipboard]
 version = "0.4.1"
@@ -219,7 +219,7 @@
 version = "1.8"
 
 [dependencies.toml]
-version = "0.9"
+version = "0.8"
 
 [dependencies.trash]
 version = "3.1.2"
@@ -235,7 +235,7 @@
 version = "0.15"
 
 [dependencies.which]
-version = "4.4.0"
+version = "8"
 
 [dependencies.xterm-query]
 version = "0.5"
