--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,7 +51,7 @@
 version = "0.21"
 
 [dependencies.bet]
-version = "1.0.3"
+version = "1.0"
 
 [dependencies.char_reader]
 version = "0.1"
@@ -60,20 +60,20 @@
 version = "0.4"
 
 [dependencies.clap]
-version = "4.4"
+version = "4.5"
 features = [
     "derive",
     "cargo",
 ]
 
 [dependencies.clap-help]
-version = "1.3"
+version = "1.2"
 
 [dependencies.cli-log]
 version = "2.0"
 
 [dependencies.crokey]
-version = "1.1"
+version = "1.0"
 
 [dependencies.crossbeam]
 version = "0.8"
@@ -85,7 +85,7 @@
 version = "2.2.3"
 
 [dependencies.directories]
-version = "4.0"
+version = "5.0"
 
 [dependencies.file-size]
 version = "1.0.3"
@@ -94,7 +94,7 @@
 version = "0.1"
 
 [dependencies.git2]
-version = "0.19"
+version = "0.18"
 default-features = false
 
 [dependencies.glob]
@@ -132,7 +132,7 @@
 features = ["macros"]
 
 [dependencies.rayon]
-version = "1.9"
+version = "1.8"
 
 [dependencies.resvg]
 version = "0.36"
@@ -158,7 +158,7 @@
 version = "1.0.2"
 
 [dependencies.strict]
-version = "0.1.4"
+version = "0.2"
 
 [dependencies.syntect]
 version = "4.6.1"
@@ -168,20 +168,20 @@
 version = "3.2"
 
 [dependencies.termimad]
-version = "0.30"
+version = "0.29"
 
 [dependencies.terminal-clipboard]
 version = "0.4.1"
 optional = true
 
 [dependencies.terminal-light]
-version = "1.5"
+version = "1.4"
 
 [dependencies.toml]
 version = "0.8"
 
 [dependencies.trash]
-version = "3.1.2"
+version = "3.3"
 optional = true
 
 [dependencies.umask]
@@ -191,21 +191,21 @@
 version = "0.1.10"
 
 [dependencies.which]
-version = "4.4.0"
+version = "4.2"
 
 [dependencies.xterm-query]
-version = "0.3"
+version = "0.4"
 optional = true
 
 [build-dependencies.clap]
-version = "4.4"
+version = "4.5"
 features = [
     "derive",
     "cargo",
 ]
 
 [build-dependencies.clap_complete]
-version = "4.4"
+version = "4.5"
 
 [build-dependencies.clap_mangen]
 version = "0.2.12"
@@ -220,4 +220,4 @@
 version = "0.11.0"
 
 [target."cfg(unix)".dependencies.uzers]
-version = "0.11.3"
+version = "0.12"
