Description: Relax dependencies in Cargo.toml
Author: Eddie Billoir <lechatp@outlook.fr>
Index: const-panic-proc-macros/Cargo.toml
===================================================================
--- const-panic-proc-macros.orig/Cargo.toml
+++ const-panic-proc-macros/Cargo.toml
@@ -48,7 +48,7 @@ version = "1.0.37"
 version = "2.0"
 
 [dependencies.unicode-xid]
-version = "0.2.5"
+version = "0.2"
 
 [badges.travis-ci]
 repository = "rodrimati1992/const_format_crates/"
