Index: grcov/src/defs.rs
===================================================================
--- grcov.orig/src/defs.rs
+++ grcov/src/defs.rs
@@ -1,4 +1,4 @@
-use crossbeam::queue::MsQueue;
+use crossbeam::sync::MsQueue;
 use libc;
 use std::collections::{BTreeMap, HashMap};
 use std::path::PathBuf;
Index: grcov/src/main.rs
===================================================================
--- grcov.orig/src/main.rs
+++ grcov/src/main.rs
@@ -4,7 +4,7 @@ extern crate num_cpus;
 extern crate serde_json;
 extern crate tempfile;
 
-use crossbeam::queue::MsQueue;
+use crossbeam::sync::MsQueue;
 use serde_json::Value;
 use std::alloc::System;
 use std::collections::HashMap;
