Index: grcov/build.rs
===================================================================
--- grcov.orig/build.rs
+++ grcov/build.rs
@@ -68,6 +68,9 @@ fn get_llvm_libdir() -> String {
 fn main() {
     let mut build = cc::Build::new();
 
+    println!("dh-cargo:deb-built-using=bzip2=1={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=llvmgcov=1={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+
     if !cfg!(target_env = "msvc") {
         build.flag("-Wno-unused-parameter");
     }
