Index: grcov/Cargo.toml
===================================================================
--- grcov.orig/Cargo.toml
+++ grcov/Cargo.toml
@@ -26,7 +26,7 @@ categories = ["command-line-utilities",
 license = "MPL-2.0"
 repository = "https://github.com/mozilla/grcov"
 [dependencies.crossbeam]
-version = "^0.4"
+version = "^0.3"
 
 [dependencies.globset]
 version = "^0.4"
@@ -38,7 +38,7 @@ version = "^0.2"
 version = "^1.8"
 
 [dependencies.md5]
-version = "^0.6"
+version = "^0.3"
 
 [dependencies.semver]
 version = "^0.9"
@@ -60,9 +60,7 @@ version = "^2.2"
 version = "^0.8"
 
 [dependencies.zip]
-version = "^0.4"
-features = ["deflate-zlib"]
-default-features = false
+version = "^0.5"
 [dev-dependencies.regex]
 version = "^1.0"
 [build-dependencies.cc]
