Index: grcov/Cargo.toml
===================================================================
--- grcov.orig/Cargo.toml
+++ grcov/Cargo.toml
@@ -24,13 +24,13 @@ categories = ["command-line-utilities",
 license = "MPL-2.0"
 repository = "https://github.com/mozilla/grcov"
 [dependencies.crossbeam]
-version = "^0.4"
+version = "^0.3"
 
 [dependencies.globset]
 version = "^0.4"
 
 [dependencies.md5]
-version = "^0.6"
+version = "^0.3"
 
 [dependencies.num_cpus]
 version = "^1.8"
@@ -58,9 +58,7 @@ version = "^2.2"
 version = "^0.8"
 
 [dependencies.zip]
-version = "^0.4"
-features = ["deflate-zlib"]
-default-features = false
+version = "^0.5"
 [dev-dependencies.regex]
 version = "^1.0"
 [badges.appveyor]
