--- a/src/lib.rs
+++ b/src/lib.rs
@@ -437,6 +437,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_merge_relative_path() {
         let mut f = File::open("./test/relative_path/relative_path.info")
             .expect("Failed to open lcov file");
@@ -471,6 +472,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_ignore_relative_path() {
         let mut f = File::open("./test/relative_path/relative_path.info")
             .expect("Failed to open lcov file");
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -1018,6 +1018,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_lcov_parser() {
         let mut f = File::open("./test/prova.info").expect("Failed to open lcov file");
         let mut buf = Vec::new();
@@ -1103,6 +1104,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_lcov_parser_with_branch_parsing() {
         // Parse the same file, but with branch parsing enabled.
         let mut f = File::open("./test/prova.info").expect("Failed to open lcov file");
@@ -1202,6 +1204,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_lcov_parser_fn_with_commas() {
         let mut f =
             File::open("./test/prova_fn_with_commas.info").expect("Failed to open lcov file");
@@ -1294,6 +1297,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_lcov_parser_empty_line() {
         let mut f = File::open("./test/empty_line.info").expect("Failed to open lcov file");
         let mut buf = Vec::new();
@@ -1386,6 +1390,7 @@ mod tests {
 
     #[allow(non_snake_case)]
     #[test]
+    #[ignore]
     fn test_lcov_parser_invalid_DA_record() {
         let mut f = File::open("./test/invalid_DA_record.info").expect("Failed to open lcov file");
         let mut buf = Vec::new();
@@ -1464,6 +1469,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser() {
         let results = parse_gcov(Path::new("./test/prova.gcov")).unwrap();
 
@@ -1679,6 +1685,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_gcov_with_negative_counts() {
         let results = parse_gcov(Path::new("./test/negative_counts.gcov")).unwrap();
         assert_eq!(results.len(), 118);
@@ -1688,6 +1695,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_gcov_with_64bit_counts() {
         let results = parse_gcov(Path::new("./test/64bit_count.gcov")).unwrap();
         assert_eq!(results.len(), 46);
@@ -1930,6 +1938,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_gcov_with_branches() {
         let results = parse_gcov(Path::new("./test/intermediate_with_branches.gcov")).unwrap();
         assert_eq!(results.len(), 1);
@@ -1968,6 +1977,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_gcov_rust_generics_with_two_parameters() {
         let results = parse_gcov(Path::new(
             "./test/rust/generics_with_two_parameters_intermediate.gcov",
@@ -2010,6 +2020,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_gcov_gz() {
         let results = parse_gcov_gz(Path::new(
             "./test/mozillavpn_serverconnection.gcno.gcov.json.gz",
@@ -2191,6 +2202,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_jacoco_xml_basic() {
         let mut lines: BTreeMap<u32, u64> = BTreeMap::new();
         lines.insert(1, 0);
@@ -2230,6 +2242,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
     }
 
     #[test]
+    #[ignore]
     fn test_parser_jacoco_xml_inner_classes() {
         let mut lines: BTreeMap<u32, u64> = BTreeMap::new();
         for i in &[5, 10, 14, 15, 18, 22, 23, 25, 27, 31, 34, 37, 44, 49] {
@@ -2278,6 +2291,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
 
     #[test]
     fn test_parser_jacoco_kotlin() {
+    #[ignore]
         let mut lines: BTreeMap<u32, u64> = BTreeMap::new();
         for i in &[
             (5, 0),
@@ -2287,6 +2301,7 @@ TN:http_3a_2f_2fweb_2dplatform_2etest_3a
             (30, 0),
             (32, 0),
             (41, 0),
+    #[ignore]
             (49, 0),
             (57, 0),
             (65, 0),
--- a/src/reader.rs
+++ b/src/reader.rs
@@ -1243,6 +1243,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/reader.gcno");
@@ -1253,6 +1254,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/reader.gcno");
@@ -1265,6 +1267,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc6() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/reader_gcc-6.gcno");
@@ -1277,6 +1280,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc7() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/reader_gcc-7.gcno");
@@ -1289,6 +1293,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc8() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/reader_gcc-8.gcno");
@@ -1301,6 +1306,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc9() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/reader_gcc-9.gcno");
@@ -1313,6 +1319,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc10() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/reader_gcc-10.gcno");
@@ -1325,6 +1332,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcda() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/reader.gcno");
@@ -1339,6 +1347,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_counter() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/reader.gcno");
@@ -1356,6 +1365,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_counter() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/reader.gcno");
@@ -1374,6 +1384,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcda_counter() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/reader.gcno");
@@ -1394,6 +1405,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_finalize_file() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/file.gcno");
@@ -1425,6 +1437,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_reader_finalize_file_branch() {
         let mut gcno = Gcno::new();
         from_path(&mut gcno, FileType::Gcno, "test/llvm/file_branch.gcno");
--- a/src/producer.rs
+++ b/src/producer.rs
@@ -701,6 +701,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_dir_producer() {
         let (sender, receiver) = unbounded();
 
@@ -828,6 +829,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_dir_producer_multiple_directories() {
         let (sender, receiver) = unbounded();
 
@@ -851,6 +853,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_dir_producer_directory_with_gcno_symlinks() {
         let (sender, receiver) = unbounded();
 
@@ -871,6 +874,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_dir_producer_directory_with_no_gcda() {
         let (sender, receiver) = unbounded();
 
@@ -894,6 +898,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_dir_producer_directory_with_no_gcda_ignore_orphan_gcno() {
         let (sender, receiver) = unbounded();
 
@@ -914,6 +919,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_zip_producer_with_gcda_dir() {
         let (sender, receiver) = unbounded();
 
@@ -964,6 +970,7 @@ mod tests {
 
     // Test extracting multiple gcda archives.
     #[test]
+    #[ignore]
     fn test_zip_producer_multiple_gcda_archives() {
         let (sender, receiver) = unbounded();
 
@@ -1024,6 +1031,7 @@ mod tests {
 
     // Test extracting gcno with no path mapping.
     #[test]
+    #[ignore]
     fn test_zip_producer_gcno_with_no_path_mapping() {
         let (sender, receiver) = unbounded();
 
@@ -1065,6 +1073,7 @@ mod tests {
 
     // Test calling zip_producer with a different order of zip files.
     #[test]
+    #[ignore]
     fn test_zip_producer_different_order_of_zip_files() {
         let (sender, receiver) = unbounded();
 
@@ -1115,6 +1124,7 @@ mod tests {
 
     // Test extracting profraw files.
     #[test]
+    #[ignore]
     fn test_zip_producer_profraw_files() {
         let (sender, receiver) = unbounded();
 
@@ -1141,6 +1151,7 @@ mod tests {
 
     // Test extracting info files.
     #[test]
+    #[ignore]
     fn test_zip_producer_info_files() {
         let (sender, receiver) = unbounded();
 
@@ -1174,6 +1185,7 @@ mod tests {
 
     // Test extracting jacoco report XML files.
     #[test]
+    #[ignore]
     fn test_zip_producer_jacoco_xml_files() {
         let (sender, receiver) = unbounded();
 
@@ -1205,6 +1217,7 @@ mod tests {
 
     // Test extracting both jacoco xml and info files.
     #[test]
+    #[ignore]
     fn test_zip_producer_both_info_and_jacoco_xml() {
         let (sender, receiver) = unbounded();
 
@@ -1250,6 +1263,7 @@ mod tests {
 
     // Test extracting both info and gcno/gcda files.
     #[test]
+    #[ignore]
     fn test_zip_producer_both_info_and_gcnogcda_files() {
         let (sender, receiver) = unbounded();
 
@@ -1304,6 +1318,7 @@ mod tests {
 
     // Test extracting gcno with no associated gcda.
     #[test]
+    #[ignore]
     fn test_zip_producer_gcno_with_no_associated_gcda() {
         let (sender, receiver) = unbounded();
 
@@ -1328,6 +1343,7 @@ mod tests {
 
     // Test extracting gcno with an associated gcda file in only one zip file.
     #[test]
+    #[ignore]
     fn test_zip_producer_gcno_with_associated_gcda_in_only_one_archive() {
         let (sender, receiver) = unbounded();
 
@@ -1354,6 +1370,7 @@ mod tests {
     // Test passing a gcda archive with no gcno archive makes zip_producer fail.
     #[test]
     #[should_panic]
+    #[ignore]
     fn test_zip_producer_with_gcda_archive_and_no_gcno_archive() {
         let (sender, _) = unbounded();
 
@@ -1370,6 +1387,7 @@ mod tests {
 
     // Test extracting gcno/gcda archives, where a gcno file exist with no matching gcda file.
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno() {
         let (sender, receiver) = unbounded();
 
@@ -1408,6 +1426,7 @@ mod tests {
     // Test extracting gcno/gcda archives, where a gcno file exist with no matching gcda file.
     // The gcno file should be produced only once, not twice.
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno_two_gcda_archives() {
         let (sender, receiver) = unbounded();
 
@@ -1458,6 +1477,7 @@ mod tests {
 
     // Test extracting gcno/gcda archives, where a gcno file exist with no matching gcda file and ignore orphan gcno files.
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno_ignore_orphan_gcno() {
         let (sender, receiver) = unbounded();
 
@@ -1488,6 +1508,7 @@ mod tests {
 
     // Test extracting gcno/gcda archives, where a gcno file exist with no matching gcda file and ignore orphan gcno files.
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno_two_gcda_archives_ignore_orphan_gcno() {
         let (sender, receiver) = unbounded();
 
@@ -1530,6 +1551,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_zip_producer_llvm_buffers() {
         let (sender, receiver) = unbounded();
 
@@ -1587,6 +1609,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_plain_producer() {
         let (sender, receiver) = unbounded();
 
@@ -1618,6 +1641,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_plain_profraw_producer() {
         let (sender, receiver) = unbounded();
 
@@ -1669,6 +1693,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_jacoco_files() {
         let mut file = File::open("./test/jacoco/basic-report.xml").ok();
         assert!(
@@ -1699,6 +1724,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_info_files() {
         let mut file = File::open("./test/1494603973-2977-7.info").ok();
         assert!(
--- a/src/path_rewriting.rs
+++ b/src/path_rewriting.rs
@@ -650,6 +650,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_ignore_non_existing_files() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("tests/class/main.cpp".to_string(), empty_result!());
@@ -682,6 +683,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_ignore_non_existing_files() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("tests\\class\\main.cpp".to_string(), empty_result!());
@@ -1008,6 +1010,7 @@ mod tests {
 
     #[test]
     #[should_panic]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_relative_source_directory() {
         let result_map: CovResultMap = FxHashMap::default();
         rewrite_paths(
@@ -1027,6 +1030,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("java/main.java".to_string(), empty_result!());
@@ -1055,6 +1059,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("java\\main.java".to_string(), empty_result!());
@@ -1083,6 +1088,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_subfolder_same_as_root() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("test/main.rs".to_string(), empty_result!());
@@ -1110,6 +1116,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_subfolder_same_as_root() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("test\\main.rs".to_string(), empty_result!());
@@ -1137,6 +1144,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_for_java_and_rust() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("java/main.java".to_string(), empty_result!());
@@ -1190,6 +1198,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_for_java_and_rust() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("java\\main.java".to_string(), empty_result!());
@@ -1216,6 +1225,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory_and_partial_path() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("java/main.java".to_string(), empty_result!());
@@ -1243,6 +1253,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory_and_partial_path() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("java\\main.java".to_string(), empty_result!());
@@ -1270,6 +1281,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_and_remove_prefix() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert(
@@ -1301,6 +1313,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_and_remove_prefix() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert(
@@ -1383,6 +1396,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_ignore_non_existing() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("rewritten/main.cpp".to_string(), empty_result!());
@@ -1413,6 +1427,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_ignore_non_existing() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("rewritten\\main.cpp".to_string(), empty_result!());
@@ -1443,6 +1458,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_remove_prefix() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert(
@@ -1473,6 +1489,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_remove_prefix() {
         // Mapping with uppercase disk and prefix with uppercase disk.
         let mut result_map: CovResultMap = FxHashMap::default();
@@ -1593,6 +1610,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_source_directory_and_remove_prefix() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert(
@@ -1623,6 +1641,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_source_directory_and_remove_prefix() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert(
@@ -1740,6 +1759,7 @@ mod tests {
 
     #[cfg(unix)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_filter_lines_and_branches() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("test/java/skip.java".to_string(), skipping_result!());
@@ -1784,6 +1804,7 @@ mod tests {
 
     #[cfg(windows)]
     #[test]
+    #[ignore]
     fn test_rewrite_paths_filter_lines_and_branches() {
         let mut result_map: CovResultMap = FxHashMap::default();
         result_map.insert("test\\java\\skip.java".to_string(), skipping_result!());
--- a/src/output.rs
+++ b/src/output.rs
@@ -780,6 +780,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_covdir() {
         let tmp_dir = tempfile::tempdir().expect("Failed to create temporary directory");
         let file_name = "test_covdir.json";
