--- a/src/lib.rs
+++ b/src/lib.rs
@@ -429,2 +429,3 @@
     #[test]
+    #[ignore]
     fn test_merge_relative_path() {
@@ -463,2 +464,3 @@
     #[test]
+    #[ignore]
     fn test_ignore_relative_path() {
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -878,2 +878,3 @@
     #[test]
+    #[ignore]
     fn test_lcov_parser() {
@@ -963,2 +964,3 @@
     #[test]
+    #[ignore]
     fn test_lcov_parser_with_branch_parsing() {
@@ -1062,2 +1064,3 @@
     #[test]
+    #[ignore]
     fn test_lcov_parser_fn_with_commas() {
@@ -1154,2 +1157,3 @@
     #[test]
+    #[ignore]
     fn test_lcov_parser_empty_line() {
@@ -1246,2 +1250,3 @@
     #[test]
+    #[ignore]
     fn test_lcov_parser_invalid_DA_record() {
@@ -1271,2 +1276,3 @@
     #[test]
+    #[ignore]
     fn test_parser() {
@@ -1486,2 +1492,3 @@
     #[test]
+    #[ignore]
     fn test_parser_gcov_with_negative_counts() {
@@ -1495,2 +1502,3 @@
     #[test]
+    #[ignore]
     fn test_parser_gcov_with_64bit_counts() {
@@ -1737,2 +1745,3 @@
     #[test]
+    #[ignore]
     fn test_parser_gcov_with_branches() {
@@ -1775,2 +1784,3 @@
     #[test]
+    #[ignore]
     fn test_parser_gcov_rust_generics_with_two_parameters() {
@@ -1817,2 +1827,3 @@
     #[test]
+    #[ignore]
     fn test_parser_gcov_gz() {
@@ -1998,2 +2009,3 @@
     #[test]
+    #[ignore]
     fn test_parser_jacoco_xml_basic() {
@@ -2037,2 +2049,3 @@
     #[test]
+    #[ignore]
     fn test_parser_jacoco_xml_inner_classes() {
@@ -2085,2 +2098,3 @@
     #[should_panic]
+    #[ignore]
     fn test_parser_jacoco_xml_non_top_level_classes_panics() {
@@ -2094,2 +2108,3 @@
     #[should_panic]
+    #[ignore]
     fn test_parser_jacoco_xml_full_report_with_non_top_level_classes_panics() {
--- a/src/reader.rs
+++ b/src/reader.rs
@@ -1250,2 +1250,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno() {
@@ -1260,2 +1261,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda() {
@@ -1272,2 +1274,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc6() {
@@ -1284,2 +1287,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc7() {
@@ -1296,2 +1300,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc8() {
@@ -1308,2 +1313,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc9() {
@@ -1320,2 +1326,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcc10() {
@@ -1332,2 +1339,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcda() {
@@ -1346,2 +1354,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_counter() {
@@ -1363,2 +1372,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_counter() {
@@ -1381,2 +1391,3 @@
     #[test]
+    #[ignore]
     fn test_reader_gcno_gcda_gcda_counter() {
@@ -1401,2 +1412,3 @@
     #[test]
+    #[ignore]
     fn test_reader_finalize_file() {
@@ -1432,2 +1444,3 @@
     #[test]
+    #[ignore]
     fn test_reader_finalize_file_branch() {
--- a/src/producer.rs
+++ b/src/producer.rs
@@ -651,2 +651,3 @@
     #[test]
+    #[ignore]
     fn test_dir_producer() {
@@ -771,2 +772,3 @@
     #[test]
+    #[ignore]
     fn test_dir_producer_multiple_directories() {
@@ -794,2 +796,3 @@
     #[test]
+    #[ignore]
     fn test_dir_producer_directory_with_gcno_symlinks() {
@@ -814,2 +817,3 @@
     #[test]
+    #[ignore]
     fn test_dir_producer_directory_with_no_gcda() {
@@ -837,2 +841,3 @@
     #[test]
+    #[ignore]
     fn test_dir_producer_directory_with_no_gcda_ignore_orphan_gcno() {
@@ -857,2 +862,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_with_gcda_dir() {
@@ -907,2 +913,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_multiple_gcda_archives() {
@@ -967,2 +974,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_gcno_with_no_path_mapping() {
@@ -1008,2 +1016,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_different_order_of_zip_files() {
@@ -1058,2 +1067,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_profraw_files() {
@@ -1084,2 +1094,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_info_files() {
@@ -1117,2 +1128,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_jacoco_xml_files() {
@@ -1148,2 +1160,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_both_info_and_jacoco_xml() {
@@ -1193,2 +1206,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_both_info_and_gcnogcda_files() {
@@ -1247,2 +1261,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_gcno_with_no_associated_gcda() {
@@ -1271,2 +1286,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_gcno_with_associated_gcda_in_only_one_archive() {
@@ -1297,2 +1313,3 @@
     #[should_panic]
+    #[ignore]
     fn test_zip_producer_with_gcda_archive_and_no_gcno_archive() {
@@ -1313,2 +1330,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno() {
@@ -1351,2 +1369,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno_two_gcda_archives() {
@@ -1401,2 +1420,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno_ignore_orphan_gcno() {
@@ -1431,2 +1451,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_no_matching_gcno_two_gcda_archives_ignore_orphan_gcno() {
@@ -1473,2 +1494,3 @@
     #[test]
+    #[ignore]
     fn test_zip_producer_llvm_buffers() {
@@ -1530,2 +1552,3 @@
     #[test]
+    #[ignore]
     fn test_plain_producer() {
@@ -1561,2 +1584,3 @@
     #[test]
+    #[ignore]
     fn test_plain_profraw_producer() {
@@ -1612,2 +1636,3 @@
     #[test]
+    #[ignore]
     fn test_jacoco_files() {
@@ -1642,2 +1667,3 @@
     #[test]
+    #[ignore]
     fn test_info_files() {
--- a/src/path_rewriting.rs
+++ b/src/path_rewriting.rs
@@ -560,2 +560,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_ignore_non_existing_files() {
@@ -592,2 +593,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_ignore_non_existing_files() {
@@ -918,2 +920,3 @@
     #[should_panic]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_relative_source_directory() {
@@ -937,2 +940,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory() {
@@ -965,2 +969,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory() {
@@ -993,2 +998,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_subfolder_same_as_root() {
@@ -1020,2 +1026,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_subfolder_same_as_root() {
@@ -1047,2 +1054,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_for_java_and_rust() {
@@ -1073,2 +1081,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_for_java_and_rust() {
@@ -1099,2 +1108,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory_and_partial_path() {
@@ -1126,2 +1136,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_absolute_source_directory_and_partial_path() {
@@ -1153,2 +1164,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_and_remove_prefix() {
@@ -1184,2 +1196,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_and_remove_prefix() {
@@ -1266,2 +1279,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_ignore_non_existing() {
@@ -1296,2 +1310,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_ignore_non_existing() {
@@ -1326,2 +1341,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_remove_prefix() {
@@ -1356,2 +1372,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_remove_prefix() {
@@ -1476,2 +1493,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_source_directory_and_remove_prefix() {
@@ -1506,2 +1524,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_rewrite_path_using_mapping_and_source_directory_and_remove_prefix() {
@@ -1623,2 +1642,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_filter_lines_and_branches() {
@@ -1666,2 +1686,3 @@
     #[test]
+    #[ignore]
     fn test_rewrite_paths_filter_lines_and_branches() {
--- a/src/output.rs
+++ b/src/output.rs
@@ -767,2 +767,3 @@
     #[test]
+    #[ignore]
     fn test_covdir() {
