--- a/Cargo.toml
+++ b/Cargo.toml
@@ -108,7 +108,7 @@ version = "0.4"
 version = "0.10"
 
 [dependencies.num_cpus]
-version = "1.17"
+version = "1"
 
 [dependencies.quick-xml]
 version = "0.38"
@@ -120,7 +120,7 @@ version = "1.10"
 version = "1.11"
 
 [dependencies.rustc-hash]
-version = "2.1"
+version = "2"
 
 [dependencies.rustc_version]
 version = "0.4"
@@ -142,14 +142,14 @@ version = "0.12"
 version = "1.15"
 
 [dependencies.symbolic-common]
-version = "12.16"
+version = "12"
 
 [dependencies.symbolic-demangle]
-version = "12.16"
+version = "12"
 default-features = false
 
 [dependencies.tabled]
-version = "0.20"
+version = "0.18"
 
 [dependencies.tempfile]
 version = "3.20"
@@ -165,7 +165,7 @@ features = ["v4"]
 version = "2.5"
 
 [dependencies.zip]
-version = "4.3"
+version = "5"
 default-features = false
 
 [dev-dependencies.pretty_assertions]
