--- grcov.orig/Cargo.toml
+++ grcov/Cargo.toml
@@ -53,2 +53,1 @@
-deflate-miniz = ["zip/deflate-miniz"]
-deflate-zlib = ["zip/deflate-zlib"]
+deflate-zlib = ["zip/deflate-flate2-zlib"]
@@ -116,3 +116,3 @@
 [dependencies.quick-xml]
-version = "0.37"
+version = "0.36"
 
@@ -125,3 +125,3 @@
 [dependencies.rustc-hash]
-version = "2.1"
+version = "2"
 
@@ -147,6 +147,6 @@
 [dependencies.symbolic-common]
-version = "12.14"
+version = "12"
 
 [dependencies.symbolic-demangle]
-version = "12.14"
+version = "12"
 default-features = false
@@ -170,3 +170,3 @@
 [dependencies.zip]
-version = "2.5"
+version = "5"
 default-features = false
--- grcov.orig/src/parser.rs
+++ grcov/src/parser.rs
@@ -15,3 +15,3 @@
 use quick_xml::encoding::Decoder;
-use quick_xml::encoding::EncodingError;
+//use quick_xml::encoding::EncodingError;
 use quick_xml::events::attributes::AttrError;
@@ -47,2 +47,3 @@
 
+#[cfg(any())]
 impl From<EncodingError> for ParserError {
