diff --git a/tests/choose.rs b/tests/choose.rs
index 4b6780a..de066fc 100644
--- a/tests/choose.rs
+++ b/tests/choose.rs
@@ -214,6 +214,7 @@ fn status_error() {
 }
 
 #[test]
+#[ignore = "Test fails in Ubuntu because of rust-coreutils require-utility-to-be-invoked-at-matching-path.patch"]
 fn default() {
   let tmp = temptree! {
     justfile: "foo:\n echo foo\n",
diff --git a/tests/edit.rs b/tests/edit.rs
index 2d9840a..3b7c1bd 100644
--- a/tests/edit.rs
+++ b/tests/edit.rs
@@ -96,6 +96,7 @@ fn status_error() {
 
 /// Test that editor is $VISUAL, $EDITOR, or "vim" in that order
 #[test]
+#[ignore = "Test fails in Ubuntu because of rust-coreutils require-utility-to-be-invoked-at-matching-path.patch"]
 fn editor_precedence() {
   let tmp = temptree! {
     justfile: JUSTFILE,
