--- a/Cargo.toml
+++ b/Cargo.toml
@@ -106,6 +106,6 @@
 [dependencies.env_logger]
-version = "0.11.0"
+version = "0.10.0"
 
 [dependencies.heck]
-version = "0.5.0"
+version = "0.4.0"
 
@@ -130,3 +130,3 @@
 [dependencies.regex]
-version = "1.10.4"
+version = "1.10"
 
@@ -156,3 +156,3 @@
 [dependencies.snafu]
-version = "0.8.0"
+version = "0.7.1"
 
@@ -191,2 +191,2 @@
 [dev-dependencies.which]
-version = "6.0.0"
+version = "4"
