--- a/Cargo.toml
+++ b/Cargo.toml
@@ -107,6 +107,6 @@
 [dependencies.env_logger]
-version = "0.11.0"
+version = "0.10.0"
 
 [dependencies.heck]
-version = "0.5.0"
+version = "0.4.0"
 
@@ -131,3 +131,3 @@
 [dependencies.regex]
-version = "1.10.4"
+version = "1.10"
 
@@ -157,3 +157,3 @@
 [dependencies.snafu]
-version = "0.8.0"
+version = "0.7.1"
 
@@ -192,2 +192,2 @@
 [dev-dependencies.which]
-version = "6.0.0"
+version = "4"
