Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -81,3 +81,3 @@
 [dependencies.clap]
-version = "4.0.0"
+version = "4.5"
 features = [
@@ -89,3 +89,3 @@
 [dependencies.clap_complete]
-version = "4.0.0"
+version = "4.5"
 
@@ -111,3 +111,3 @@
 [dependencies.heck]
-version = "0.5.0"
+version = "0.4.0"
 
@@ -132,3 +132,3 @@
 [dependencies.regex]
-version = "1.10.4"
+version = "1.10"
 
@@ -158,3 +158,3 @@
 [dependencies.snafu]
-version = "0.8.0"
+version = "0.7.1"
 
@@ -174,3 +174,3 @@
 [dependencies.unicode-width]
-version = "0.1.0"
+version = "0.1.13"
 
@@ -190,3 +190,3 @@
 [dev-dependencies.which]
-version = "6.0.0"
+version = "4"
 
