Description: Work around clap_complete behavior change
Forwarded: not-needed
Last-Update: 2024-12-24
--- a/src/completions.rs
+++ b/src/completions.rs
@@ -143,3 +143,3 @@
   (
-    r"'*--set=[Override <VARIABLE> with <VALUE>]:VARIABLE:_default:VARIABLE:_default' \",
+    r"'*--set=[Override <VARIABLE> with <VALUE>]:VARIABLE: :VARIABLE: ' \",
     r"'*--set=[Override <VARIABLE> with <VALUE>]: :(_just_variables)' \",
@@ -147,4 +147,4 @@
   (
-    r"'()-s+[Show recipe at <PATH>]:PATH:_default' \
-'()--show=[Show recipe at <PATH>]:PATH:_default' \",
+    r"'()-s+[Show recipe at <PATH>]:PATH: ' \
+'()--show=[Show recipe at <PATH>]:PATH: ' \",
     r"'-s+[Show recipe at <PATH>]: :(_just_commands)' \
@@ -154,3 +154,3 @@
     "'*::ARGUMENTS -- Overrides and recipe(s) to run, defaulting to the first recipe in the \
-     justfile:_default' \\
+     justfile:' \\
 && ret=0",
