--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,9 +55,9 @@
 optional = true
 default-features = false
 
-[dependencies.borsh]
-version = "1"
-optional = true
+#[dependencies.borsh]
+#version = "1"
+#optional = true
 
 [dependencies.bytes]
 version = "1"
@@ -169,7 +169,7 @@
 
 [features]
 arbitrary = ["dep:arbitrary"]
-borsh = ["dep:borsh"]
+#borsh = ["dep:borsh"]
 bytes = ["dep:bytes"]
 default = ["std"]
 diesel = ["dep:diesel"]
