Index: lsd/Cargo.toml
===================================================================
--- lsd.orig/Cargo.toml
+++ lsd/Cargo.toml
@@ -121,13 +121,13 @@ version = "0.11"
 [target."cfg(unix)".dependencies.xattr]
 version = "1"
 
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "aclapi",
-    "accctrl",
-    "winnt",
-    "winerror",
-    "securitybaseapi",
-    "winbase",
-]
+# [target."cfg(windows)".dependencies.winapi]
+# version = "0.3"
+# features = [
+#     "aclapi",
+#     "accctrl",
+#     "winnt",
+#     "winerror",
+#     "securitybaseapi",
+#     "winbase",
+# ]
