--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,7 +41,7 @@
 version = "0.2"
 
 [dependencies.clap]
-version = "4.3.*"
+version = "4"
 features = [
     "derive",
     "wrap_help",
@@ -64,7 +64,7 @@
 version = "0.2.*"
 
 [dependencies.lscolors]
-version = "0.16.0"
+version = "0.20"
 
 [dependencies.once_cell]
 version = "1.17.1"
@@ -83,7 +83,7 @@
 version = "0.1"
 
 [dependencies.terminal_size]
-version = "0.3"
+version = "0.4"
 
 [dependencies.thiserror]
 version = "1.0"
@@ -104,7 +104,8 @@
 version = "2.5"
 
 [dependencies.yaml-rust]
-version = "0.4.*"
+version = "0.10"
+package = "yaml-rust2"
 
 [dev-dependencies.assert_cmd]
 version = "2"
@@ -122,11 +123,11 @@
 version = "3"
 
 [build-dependencies.clap]
-version = "4.3.*"
+version = "4"
 features = ["derive"]
 
 [build-dependencies.clap_complete]
-version = "4.3"
+version = "4"
 
 [build-dependencies.version_check]
 version = "0.9.*"
@@ -137,22 +138,22 @@
 sudo = []
 
 [target."cfg(not(all(windows, target_arch = \"x86\", target_env = \"gnu\")))".dependencies.git2]
-version = "0.18"
+version = "0.20"
 optional = true
 default-features = false
 
 [target."cfg(unix)".dependencies.users]
-version = "0.11.3"
+version = "0.12"
 package = "uzers"
 
 [target."cfg(unix)".dependencies.xattr]
 version = "1"
 
-[target."cfg(windows)".dependencies.windows]
-version = "0.43.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Security_Authorization",
-    "Win32_Storage_FileSystem",
-    "Win32_System_Memory",
-]
+#[target."cfg(windows)".dependencies.windows]
+#version = "0.43.0"
+#features = [
+#    "Win32_Foundation",
+#    "Win32_Security_Authorization",
+#    "Win32_Storage_FileSystem",
+#    "Win32_System_Memory",
+#]
