diff --git a/Cargo.toml b/Cargo.toml
index 4555fff..cf776d8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,12 +22,6 @@ categories = ["command-line-utilities"]
 license = "Apache-2.0"
 repository = "https://github.com/lsd-rs/lsd"
 
-[profile.release]
-lto = true
-codegen-units = 1
-debug = 0
-strip = true
-
 [[bin]]
 name = "lsd"
 path = "src/main.rs"
@@ -37,7 +31,7 @@ version = "0.4.*"
 features = ["unstable-locales"]
 
 [dependencies.chrono-humanize]
-version = "0.1.*"
+version = "0.2"
 
 [dependencies.clap]
 version = "4.1"
@@ -47,11 +41,11 @@ features = [
 ]
 
 [dependencies.crossterm]
-version = "0.24.0"
+version = "0.27"
 features = ["serde"]
 
 [dependencies.dirs]
-version = "3.0.*"
+version = ">= 3, < 6"
 
 [dependencies.globset]
 version = "0.4.*"
@@ -63,7 +57,7 @@ version = "0.2.2"
 version = "0.2.*"
 
 [dependencies.lscolors]
-version = "0.9.0"
+version = "0.16"
 
 [dependencies.once_cell]
 version = "1.17.1"
@@ -73,16 +67,16 @@ version = "1.0"
 features = ["derive"]
 
 [dependencies.serde_yaml]
-version = "0.8"
+version = ">= 0.8, < 1.0"
 
 [dependencies.sys-locale]
-version = "0.2.4"
+version = "0.3"
 
 [dependencies.term_grid]
 version = "0.1.*"
 
 [dependencies.terminal_size]
-version = "0.1.*"
+version = "0.3"
 
 [dependencies.thiserror]
 version = "1.0"
@@ -91,16 +85,16 @@ version = "1.0"
 version = "0.1.*"
 
 [dependencies.url]
-version = "2.1.*"
+version = "2"
 
 [dependencies.vsort]
-version = "0.1.0"
+version = "0.2"
 
 [dependencies.wild]
-version = "2.0.*"
+version = "2"
 
 [dependencies.xdg]
-version = "2.1.*"
+version = "2"
 
 [dependencies.yaml-rust]
 version = "0.4.*"
@@ -112,7 +106,7 @@ version = "1"
 version = "1"
 
 [dev-dependencies.predicates]
-version = "1.0.1"
+version = "1"
 
 [dev-dependencies.serial_test]
 version = "0.5"
@@ -136,7 +130,7 @@ no-git = []
 sudo = []
 
 [target."cfg(not(all(windows, target_arch = \"x86\", target_env = \"gnu\")))".dependencies.git2]
-version = "0.16"
+version = "0.18"
 optional = true
 default-features = false
 
