Index: libpulse-binding/Cargo.toml
===================================================================
--- libpulse-binding.orig/Cargo.toml
+++ libpulse-binding/Cargo.toml
@@ -27,6 +27,9 @@ keywords = [
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/jnqnfe/pulse-binding-rust"
 
+[lib]
+doctest = false
+
 [package.metadata.docs.rs]
 all-features = false
 no-default-features = true
@@ -74,8 +77,3 @@ pa_v8 = [
     "pa_v6",
     "libpulse-sys/pa_v8",
 ]
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = ["winsock2"]
-default-features = false
