Forwarded: not-needed
Last-Update: 2024-11-01
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,4 +50,4 @@
-[[example]]
-name = "reqwest"
-path = "examples/reqwest.rs"
-required-features = ["http"]
+#[[example]]
+#name = "reqwest"
+#path = "examples/reqwest.rs"
+#required-features = ["http"]
@@ -98,3 +98,3 @@
-[dev-dependencies.reqwest]
-version = "0.12.7"
-features = ["blocking"]
+#[dev-dependencies.reqwest]
+#version = "0.12.7"
+#features = ["blocking"]
