Description: Adapt dependencies for only http 1 available
Forwarded: not-needed
Last-Update: 2024-11-01
Index: http-auth/Cargo.toml
===================================================================
--- http-auth.orig/Cargo.toml
+++ http-auth/Cargo.toml
@@ -50,7 +50,7 @@ path = "src/lib.rs"
 [[example]]
 name = "reqwest"
 path = "examples/reqwest.rs"
-required-features = ["http"]
+required-features = ["http10"]
 
 [dependencies.base64]
 version = "0.22.0"
@@ -64,9 +64,6 @@ optional = true
 version = "0.4"
 optional = true
 
-[dependencies.http]
-version = "0.2.5"
-optional = true
 
 [dependencies.http10]
 version = "1.0.0"
