From 6ddb4c4954ab32c1880ce901e4a7348ac856a9af Mon Sep 17 00:00:00 2001
From: Wolfgang Silbermayr <wolfgang@silbermayr.at>
Date: Wed, 26 Dec 2018 09:45:39 +0100
Subject: [PATCH] Remove futures preview

---
 Cargo.toml | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index a0c52b3..03403d0 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,18 +32,6 @@ version = "1.0"
 [dependencies.fragile]
 version = "0.3"
 
-[dependencies.futures-channel-preview]
-version = "0.2"
-optional = true
-
-[dependencies.futures-core-preview]
-version = "0.2"
-optional = true
-
-[dependencies.futures-util-preview]
-version = "0.2"
-optional = true
-
 [dependencies.gio-sys]
 version = "0.7.0"
 
@@ -70,7 +58,6 @@ optional = true
 [features]
 dox = ["gio-sys/dox", "glib/dox"]
 embed-lgpl-docs = ["gtk-rs-lgpl-docs"]
-futures = ["futures-core-preview", "futures-channel-preview", "futures-util-preview", "glib/futures"]
 purge-lgpl-docs = ["gtk-rs-lgpl-docs"]
 v2_34 = ["gio-sys/v2_34", "glib/v2_34"]
 v2_36 = ["v2_34", "gio-sys/v2_36", "glib/v2_36"]
-- 
2.20.1

