--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,16 +24,16 @@
 [lib]
 proc-macro = true
 [dependencies.proc-macro2]
-version = "0.4.8"
+version = "1"
 
 [dependencies.quote]
-version = "0.6.3"
+version = "1"
 
 [dependencies.syn]
-version = "0.15.0"
+version = "1"
 
 [dependencies.synstructure]
-version = "0.10.0"
+version = "0.11.0"
 [dev-dependencies.failure]
 version = "0.1.0"
 
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -120,7 +120,7 @@
         }
 
         let format_string = match msg.nested[0] {
-            syn::NestedMeta::Meta(syn::Meta::NameValue(ref nv)) if nv.ident == "display" => {
+            syn::NestedMeta::Meta(syn::Meta::NameValue(ref nv)) if nv.path.is_ident("display") => {
                 nv.lit.clone()
             }
             _ => {
@@ -131,44 +131,21 @@
             }
         };
         let args = msg.nested.iter().skip(1).map(|arg| match *arg {
-            syn::NestedMeta::Literal(syn::Lit::Int(ref i)) => {
-                let bi = &v.bindings()[i.value() as usize];
+            syn::NestedMeta::Lit(syn::Lit::Int(ref i)) => {
+                let bi = &v.bindings()[i.base10_parse::<usize>()?];
                 Ok(quote!(#bi))
             }
-            syn::NestedMeta::Meta(syn::Meta::Word(ref id)) => {
-                let id_s = id.to_string();
-                if id_s.starts_with("_") {
-                    if let Ok(idx) = id_s[1..].parse::<usize>() {
-                        let bi = match v.bindings().get(idx) {
-                            Some(bi) => bi,
-                            None => {
-                                return Err(Error::new(
-                                    arg.span(),
-                                    &format!(
-                                        "display attempted to access field `{}` in `{}::{}` which \
-                                     does not exist (there are {} field{})",
-                                        idx,
-                                        s.ast().ident,
-                                        v.ast().ident,
-                                        v.bindings().len(),
-                                        if v.bindings().len() != 1 { "s" } else { "" }
-                                    )
-                                ));
-                            }
-                        };
-                        return Ok(quote!(#bi));
-                    }
-                }
+            syn::NestedMeta::Meta(syn::Meta::Path(ref path)) => {
                 for bi in v.bindings() {
-                    if bi.ast().ident.as_ref() == Some(id) {
+                    if path.is_ident(&(bi.ast().ident)) {
                         return Ok(quote!(#bi));
                     }
                 }
                 return Err(Error::new(
                     arg.span(),
                     &format!(
-                        "Couldn't find field `{}` in `{}::{}`",
-                        id,
+                        "Couldn't find field `{:?}` in `{}::{}`",
+                        path,
                         s.ast().ident,
                         v.ast().ident
                     )
@@ -192,8 +169,8 @@
 fn find_error_msg(attrs: &[syn::Attribute]) -> Result<Option<syn::MetaList>, Error> {
     let mut error_msg = None;
     for attr in attrs {
-        if let Some(meta) = attr.interpret_meta() {
-            if meta.name() == "fail" {
+        if let Ok(meta) = attr.parse_meta() {
+            if meta.path().is_ident("fail") {
                 if error_msg.is_some() {
                     return Err(Error::new(
                         meta.span(),
@@ -223,7 +200,7 @@
                 segments: ref path, ..
             },
         }) => path.last().map_or(false, |s| {
-            s.value().ident == "Backtrace" && s.value().arguments.is_empty()
+            s.ident == "Backtrace" && s.arguments.is_empty()
         }),
         _ => false,
     }
@@ -232,18 +209,18 @@
 fn is_cause(bi: &&synstructure::BindingInfo) -> bool {
     let mut found_cause = false;
     for attr in &bi.ast().attrs {
-        if let Some(meta) = attr.interpret_meta() {
-            if meta.name() == "cause" {
+        if let Ok(meta) = attr.parse_meta() {
+            if meta.path().is_ident("cause") {
                 if found_cause {
                     panic!("Cannot have two `cause` attributes");
                 }
                 found_cause = true;
             }
-            if meta.name() == "fail" {
+            if meta.path().is_ident("fail") {
                 if let syn::Meta::List(ref list) = meta {
                     if let Some(ref pair) = list.nested.first() {
-                        if let &&syn::NestedMeta::Meta(syn::Meta::Word(ref word)) = pair.value() {
-                            if word == "cause" {
+                        if let &&syn::NestedMeta::Meta(syn::Meta::Path(ref path)) = pair {
+                            if path.is_ident("cause") {
                                 if found_cause {
                                     panic!("Cannot have two `cause` attributes");
                                 }
