Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -43,7 +43,7 @@ optional = true
 default-features = false
 
 [dependencies.console]
-version = "0.11.2"
+version = "0.8"
 
 [dependencies.content_inspector]
 version = "0.2.4"
@@ -60,7 +60,7 @@ version = "0.12"
 default-features = false
 
 [dependencies.git2]
-version = "0.13"
+version = "0.11"
 optional = true
 default-features = false
 
