Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -113,7 +113,7 @@ version = "2.33"
 optional = true
 
 [features]
-application = ["atty", "clap", "dirs", "git", "lazy_static", "paging", "wild", "regex-onig"]
+application = ["atty", "clap", "dirs", "git", "lazy_static", "paging"] #, "wild", "regex-onig"]
 default = ["application"]
 git = ["git2"]
 paging = ["shell-words"]
