Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -37,9 +37,9 @@ optional = true
 [dependencies.bincode]
 version = "1.0"
 
-[dependencies.bugreport]
-version = "0.4"
-optional = true
+#[dependencies.bugreport]
+#version = "0.4"
+#optional = true
 
 [dependencies.clap]
 version = "2.34"
Index: bat/src/bin/bat/main.rs
===================================================================
--- bat.orig/src/bin/bat/main.rs
+++ bat/src/bin/bat/main.rs
@@ -273,13 +273,13 @@ fn invoke_bugreport(app: &App) {
 fn run() -> Result<bool> {
     let app = App::new()?;
 
-    if app.matches.is_present("diagnostic") {
+/*    if app.matches.is_present("diagnostic") {
         #[cfg(feature = "bugreport")]
         invoke_bugreport(&app);
         #[cfg(not(feature = "bugreport"))]
         println!("bat has been built without the 'bugreport' feature. The '--diagnostic' option is not available.");
         return Ok(true);
-    }
+    }*/
 
     match app.matches.subcommand() {
         ("cache", Some(cache_matches)) => {
