Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -34,12 +34,12 @@ version = "^0.12.1"
 
 [dependencies.atty]
 version = "0.2.14"
-optional = true
+#optional = true
 
 [dependencies.clap]
 version = "2.33"
 features = ["suggestions", "color", "wrap_help"]
-optional = true
+#optional = true
 default-features = false
 
 [dependencies.console]
@@ -50,7 +50,7 @@ version = "0.2.4"
 
 [dependencies.dirs]
 version = "3.0"
-optional = true
+#optional = true
 
 [dependencies.encoding]
 version = "0.2"
@@ -61,7 +61,7 @@ default-features = false
 
 [dependencies.git2]
 version = "0.11"
-optional = true
+#optional = true
 default-features = false
 
 [dependencies.globset]
@@ -69,14 +69,14 @@ version = "0.4"
 
 [dependencies.lazy_static]
 version = "1.4"
-optional = true
+#optional = true
 
-[dependencies.path_abs]
-version = "0.5"
-default-features = false
+#[dependencies.path_abs]
+#version = "0.5"
+#default-features = false
 
 [dependencies.semver]
-version = "0.11"
+version = "0.9"
 
 [dependencies.serde]
 version = "1.0"
@@ -87,7 +87,7 @@ version = "0.8"
 
 [dependencies.shell-words]
 version = "1.0.0"
-optional = true
+#optional = true
 
 [dependencies.syntect]
 version = "4.4.0"
@@ -97,9 +97,9 @@ default-features = false
 [dependencies.unicode-width]
 version = "0.1.8"
 
-[dependencies.wild]
-version = "2.0"
-optional = true
+#[dependencies.wild]
+#version = "2.0"
+#optional = true
 [dev-dependencies.assert_cmd]
 version = "1.0.1"
 
@@ -110,12 +110,13 @@ version = "1.0.5"
 version = "0.3"
 [build-dependencies.clap]
 version = "2.33"
-optional = true
+#optional = true
 
-[features]
-application = ["atty", "clap", "dirs", "git", "lazy_static", "paging", "wild", "regex-onig"]
-default = ["application"]
-git = ["git2"]
-paging = ["shell-words"]
-regex-fancy = ["syntect/regex-fancy"]
-regex-onig = ["syntect/regex-onig"]
+#[features]
+#application = ["atty", "clap", "dirs", "git", "lazy_static", "paging"]
+#, "wild", "regex-onig"]
+#default = ["application"]
+#git = ["git2"]
+#paging = ["shell-words"]
+#regex-fancy = ["syntect/regex-fancy"]
+#regex-onig = ["syntect/regex-onig"]
