Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -51,7 +51,7 @@ version = "1.0"
 version = "1.1.0"
 
 [dependencies.clap]
-version = "3.2.20"
+version = "3"
 features = [
     "std",
     "suggestions",
@@ -153,7 +153,7 @@ version = "3.3.0"
 version = "0.2.0"
 
 [build-dependencies.clap]
-version = "3.2.20"
+version = "3"
 optional = true
 
 [features]
