Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -155,9 +155,9 @@ version = "0.8"
 version = "0.8.0"
 optional = true
 
-[dependencies.execute]
-version = "0.2.13"
-optional = true
+#[dependencies.execute]
+#version = "0.2.13"
+#optional = true
 
 [dependencies.flate2]
 version = "1.0"
@@ -305,7 +305,7 @@ build-assets = [
 ]
 default = ["application"]
 git = ["git2"]
-lessopen = ["execute"]
+#lessopen = ["execute"]
 minimal-application = [
     "clap",
     "etcetera",
