Description: Disable unavailable digest_auth and functionality it enables
Forwarded: not-needed
Last-Update: 2025-01-09
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -118,3 +118,3 @@
 
-[dependencies.digest_auth]
+[disabled.dependencies.digest_auth]
 version = "0.3.0"
--- a/src/auth.rs
+++ b/src/auth.rs
@@ -17,3 +17,3 @@
     Basic(String, Option<String>),
-    Digest(String, String),
+    //Digest(String, String),
 }
@@ -27,2 +27,3 @@
             }
+            #[cfg(any())]
             AuthType::Digest => {
@@ -42,3 +43,3 @@
             AuthType::Bearer => Some(Auth::Bearer(entry.password)),
-            AuthType::Digest => Some(Auth::Digest(entry.login?, entry.password)),
+            //AuthType::Digest => Some(Auth::Digest(entry.login?, entry.password)),
         }
@@ -62,2 +63,3 @@
 
+#[cfg(any())]
 pub struct DigestAuthMiddleware<'a> {
@@ -67,2 +69,3 @@
 
+#[cfg(any())]
 impl<'a> DigestAuthMiddleware<'a> {
@@ -73,2 +76,3 @@
 
+#[cfg(any())]
 impl Middleware for DigestAuthMiddleware<'_> {
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -954,3 +954,3 @@
     Bearer,
-    Digest,
+    //Digest,
 }
--- a/src/main.rs
+++ b/src/main.rs
@@ -39,3 +39,3 @@
 
-use crate::auth::{Auth, DigestAuthMiddleware};
+use crate::auth::{Auth, /*DigestAuthMiddleware*/};
 use crate::buffer::Buffer;
@@ -536,3 +536,3 @@
                 Auth::Bearer(token) => request_builder.bearer_auth(token),
-                Auth::Digest(..) => request_builder,
+                //Auth::Digest(..) => request_builder,
             }
@@ -627,2 +627,3 @@
             }
+            #[cfg(any())]
             if let Some(Auth::Digest(username, password)) = &auth {
--- a/src/session.rs
+++ b/src/session.rs
@@ -239,2 +239,3 @@
                 }
+                #[cfg(any())]
                 "digest" => {
@@ -263,2 +264,3 @@
             }
+            #[cfg(any())]
             auth::Auth::Digest(username, password) => {
--- a/src/to_curl.rs
+++ b/src/to_curl.rs
@@ -338,2 +338,3 @@
             }
+            #[cfg(any())]
             AuthType::Digest => {
