Description: Disable unavailable network-interface and functionality it enables
Forwarded: not-needed
Last-Update: 2025-01-09
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -304,3 +304,3 @@
     "rustls",
-    "network-interface",
+    #"network-interface",
 ]
@@ -315,3 +315,3 @@
 ]
-network-interface = ["dep:network-interface"]
+#network-interface = ["dep:network-interface"]
 online-tests = []
@@ -324,3 +324,3 @@
 
-[target."cfg(not(any(target_os = \"android\", target_os = \"fuchsia\", target_os = \"linux\")))".dependencies.network-interface]
+[disabled.target."cfg(not(any(target_os = \"android\", target_os = \"fuchsia\", target_os = \"linux\")))".dependencies.network-interface]
 version = "1.0.0"
--- a/src/main.rs
+++ b/src/main.rs
@@ -333,12 +333,12 @@
         } else {
-            #[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
+            //#[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
             {
-                client = client.interface(name_or_ip);
+                //client = client.interface(name_or_ip);
             }
 
-            #[cfg(not(any(target_os = "android", target_os = "fuchsia", target_os = "linux")))]
+            //#[cfg(not(any(target_os = "android", target_os = "fuchsia", target_os = "linux")))]
             {
-                #[cfg(not(feature = "network-interface"))]
+                //#[cfg(not(feature = "network-interface"))]
                 return Err(anyhow!(
-                    "This binary was built without support for binding to interfaces. Enable the `network-interface` feature."
+                    "This binary was built without support for binding to interfaces."// Enable the `network-interface` feature."
                 ));
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -335,5 +335,6 @@
 
-    /// Bind to a network interface or local IP address.
+    /// Bind to a local IP address. (Debian: interface support is disabled due to
+    /// unavailable dependency.)
     ///
-    /// Example: --interface=eth0 --interface=192.168.0.2
+    /// Example: --interface=192.168.0.2
     #[clap(long, value_name = "NAME")]
