Forwarded: https://github.com/ducaale/xh/pull/402
--- a/src/generation.rs
+++ b/src/generation.rs
@@ -184,3 +184,8 @@
     let current_date = {
-        let (year, month, day) = DateTime::now_utc().date().to_calendar_date();
+        // https://reproducible-builds.org/docs/source-date-epoch/
+        let now = match std::env::var("SOURCE_DATE_EPOCH") {
+            Ok(val) => DateTime::from_unix_timestamp(val.parse::<i64>().unwrap()).unwrap(),
+            Err(_) => DateTime::now_utc(),
+        };
+        let (year, month, day) = now.date().to_calendar_date();
         format!("{}-{:02}-{:02}", year, u8::from(month), day)
