Index: rust-slice-ring-buffer-0.3.4/benches/insert.rs
===================================================================
--- rust-slice-ring-buffer-0.3.4.orig/benches/insert.rs
+++ rust-slice-ring-buffer-0.3.4/benches/insert.rs
@@ -6,7 +6,7 @@ extern crate test;
 
 use std::collections::VecDeque;
 
-const MAX_NO_ITERS: usize = 1_000_000_000;
+const MAX_NO_ITERS: usize = 1_000_000;
 
 #[bench]
 fn insert_front_vdeq(b: &mut test::Bencher) {
Index: rust-slice-ring-buffer-0.3.4/benches/pop_back.rs
===================================================================
--- rust-slice-ring-buffer-0.3.4.orig/benches/pop_back.rs
+++ rust-slice-ring-buffer-0.3.4/benches/pop_back.rs
@@ -6,7 +6,7 @@ extern crate test;
 
 use std::collections::VecDeque;
 
-const MAX_NO_ITERS: usize = 1_000_000_000;
+const MAX_NO_ITERS: usize = 1_000_000;
 
 #[bench]
 fn pop_back_std_vecdeque(b: &mut test::Bencher) {
Index: rust-slice-ring-buffer-0.3.4/benches/pop_front.rs
===================================================================
--- rust-slice-ring-buffer-0.3.4.orig/benches/pop_front.rs
+++ rust-slice-ring-buffer-0.3.4/benches/pop_front.rs
@@ -6,7 +6,7 @@ extern crate test;
 
 use std::collections::VecDeque;
 
-const MAX_NO_ITERS: usize = 2_000_000_000;
+const MAX_NO_ITERS: usize = 2_000_000;
 
 #[bench]
 fn pop_front_std_vecdeque(b: &mut test::Bencher) {
Index: rust-slice-ring-buffer-0.3.4/benches/remove.rs
===================================================================
--- rust-slice-ring-buffer-0.3.4.orig/benches/remove.rs
+++ rust-slice-ring-buffer-0.3.4/benches/remove.rs
@@ -6,7 +6,7 @@ extern crate test;
 
 use std::collections::VecDeque;
 
-const MAX_NO_ITERS: usize = 1_000_000_000;
+const MAX_NO_ITERS: usize = 1_000_000;
 
 #[bench]
 fn remove_front_vdeq(b: &mut test::Bencher) {
