Forwarded: not-needed
Last-Update: 2024-10-10
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -102,13 +102,13 @@ path = "tests/test_take.rs"
 
-[[bench]]
-name = "buf"
-path = "benches/buf.rs"
+#[[bench]]
+#name = "buf"
+#path = "benches/buf.rs"
 
-[[bench]]
-name = "bytes"
-path = "benches/bytes.rs"
+#[[bench]]
+#name = "bytes"
+#path = "benches/bytes.rs"
 
-[[bench]]
-name = "bytes_mut"
-path = "benches/bytes_mut.rs"
+#[[bench]]
+#name = "bytes_mut"
+#path = "benches/bytes_mut.rs"
 
