Index: actix-router/Cargo.toml
===================================================================
--- actix-router.orig/Cargo.toml
+++ actix-router/Cargo.toml
@@ -48,7 +48,7 @@ version = ">=0.1.5, <2"
 version = "1"
 
 [dependencies.http]
-version = "0.2.7"
+version = "1"
 optional = true
 
 [dependencies.regex]
@@ -71,7 +71,7 @@ version = "0.5"
 features = ["html_reports"]
 
 [dev-dependencies.http]
-version = "0.2.7"
+version = "1"
 
 [dev-dependencies.percent-encoding]
 version = "2.1"
