diff --git a/Cargo.toml b/Cargo.toml
index 91176c1..14f7de5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,9 +24,6 @@ repository = "https://github.com/kellpossible/cargo-i18n/tree/master/i18n-embed"
 
 [lib]
 proc-macro = true
-[dependencies.find-crate]
-version = "0.6"
-optional = true
 
 [dependencies.i18n-config]
 version = "0.4"
@@ -48,7 +45,7 @@ version = "6"
 
 [features]
 default = []
-fluent-system = ["i18n-config", "find-crate", "quote"]
-gettext-system = ["i18n-config", "find-crate", "quote"]
+fluent-system = ["i18n-config", "quote"]
+gettext-system = ["i18n-config", "quote"]
 [badges.maintenance]
 status = "actively-developed"
