--- a/build.rs
+++ b/build.rs
@@ -1,13 +1,7 @@
 extern crate version_check;
 
 fn main() {
-  match version_check::is_min_version("1.26.0") {
-    Some((true, _actual_version)) => {
-      println!("cargo:rustc-cfg=stable_i128");
-    }
-    Some(_) => (),
-    None => {
-      eprintln!("couldn't query version info from rustc");
-    }
+  if version_check::is_min_version("1.26.0").unwrap_or(false) {
+    println!("cargo:rustc-cfg=stable_i128");
   }
 }
