Description: Fix constants to match the libsoup3 values
Forwarded: https://gitlab.gnome.org/World/Rust/soup3-rs/-/issues/5
--- a/tests/abi.rs
+++ b/tests/abi.rs
@@ -277,10 +277,10 @@
     ("(gint) SOUP_COOKIE_JAR_ACCEPT_GRANDFATHERED_THIRD_PARTY", "3"),
     ("(gint) SOUP_COOKIE_JAR_ACCEPT_NEVER", "1"),
     ("(gint) SOUP_COOKIE_JAR_ACCEPT_NO_THIRD_PARTY", "2"),
-    ("SOUP_COOKIE_MAX_AGE_ONE_DAY", "0"),
+    ("SOUP_COOKIE_MAX_AGE_ONE_DAY", "86400"),
     ("SOUP_COOKIE_MAX_AGE_ONE_HOUR", "3600"),
-    ("SOUP_COOKIE_MAX_AGE_ONE_WEEK", "0"),
-    ("SOUP_COOKIE_MAX_AGE_ONE_YEAR", "0"),
+    ("SOUP_COOKIE_MAX_AGE_ONE_WEEK", "604800"),
+    ("SOUP_COOKIE_MAX_AGE_ONE_YEAR", "31556926.080000"),
     ("(gint) SOUP_DATE_COOKIE", "2"),
     ("(gint) SOUP_DATE_HTTP", "1"),
     ("(gint) SOUP_ENCODING_BYTERANGES", "5"),
