--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,13 +51,13 @@
 path = "src/lib.rs"
 
 [dependencies.bitflags]
-version = "2.6"
+version = "2.4"
 
 [dependencies.cairo-sys-rs]
 version = "0.20"
 
 [dependencies.freetype-rs]
-version = "0.37"
+version = "0.26"
 optional = true
 
 [dependencies.glib]
@@ -93,6 +93,5 @@
     "v1_16",
     "cairo-sys-rs/v1_18",
 ]
-win32-surface = ["cairo-sys-rs/win32-surface"]
 xcb = ["cairo-sys-rs/xcb"]
 xlib = ["cairo-sys-rs/xlib"]
