Index: pyo3-build-config/Cargo.toml
===================================================================
--- pyo3-build-config.orig/Cargo.toml
+++ pyo3-build-config/Cargo.toml
@@ -61,16 +61,8 @@ path = "src/lib.rs"
 [dependencies.once_cell]
 version = "1"
 
-[dependencies.python3-dll-a]
-version = "0.2.12"
-optional = true
-
 [dependencies.target-lexicon]
 version = "0.13"
 
-[build-dependencies.python3-dll-a]
-version = "0.2.12"
-optional = true
-
 [build-dependencies.target-lexicon]
 version = "0.13"
