Index: rusttype/Cargo.toml
===================================================================
--- rusttype.orig/Cargo.toml
+++ rusttype/Cargo.toml
@@ -56,7 +56,7 @@ version = "0.5"
 optional = true
 
 [dependencies.owned_ttf_parser]
-version = "0.15"
+version = ">= 0.24, < 1.0"
 default-features = false
 
 [dependencies.rustc-hash]
@@ -77,6 +77,7 @@ has-atomics = []
 libm-math = [
     "libm",
     "ab_glyph_rasterizer/libm",
+    "owned_ttf_parser/no-std-float",
 ]
 std = [
     "has-atomics",
