Index: dfrs/Cargo.toml
===================================================================
--- dfrs.orig/Cargo.toml
+++ dfrs/Cargo.toml
@@ -32,7 +32,7 @@ version = "1.0"
 version = "2.0"
 
 [dependencies.env_logger]
-version = "0.8"
+version = "0.10"
 
 [dependencies.lazy_static]
 version = "1.4"
@@ -49,5 +49,5 @@ version = "1.4"
 [dependencies.strum]
-version = "0.20"
+version = "0.25"
 
 [dependencies.strum_macros]
-version = "0.20"
+version = "0.25"
