Index: libseat-sys/Cargo.toml.orig
===================================================================
--- libseat-sys.orig/Cargo.toml.orig
+++ libseat-sys/Cargo.toml.orig
@@ -16,7 +16,7 @@ use_bindgen = ["bindgen"]
 docs_rs = []
 
 [build-dependencies]
-bindgen = { version = "0.58.1", optional = true }
+bindgen = { version = ">= 0.58.1", optional = true }
 pkg-config = { version = "0.3.19" }
 
 [package.metadata.docs.rs]
