diff --git a/Cargo.toml b/Cargo.toml
index be90ae3..56f6bc0 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -88,10 +88,10 @@ version = "1.0.219"
 version = "1.0.140"
 
 [dependencies.strum]
-version = "0.27"
+version = "0.26"
 
 [dependencies.strum_macros]
-version = "0.27"
+version = "0.26"
 
 [dependencies.thiserror]
 version = "2.0.12"
diff --git a/src/lib.rs b/src/lib.rs
index 9ad294a..09233c3 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -39,7 +39,7 @@
 //!
 //! ## Example
 //!
-//! ```rust
+//! ```ignore
 //! use serde_sarif::sarif::Message;
 //!
 //! let message = Message::default()
