Description: relax dependency versions
Author: Ananthu C V <weepingclown@disroot.org>
Forwarded: not-needed
Last-Update: 2024-06-21

--- rust-wezterm-input-types-0.1.0.orig/Cargo.toml
+++ rust-wezterm-input-types-0.1.0/Cargo.toml
@@ -22,7 +22,7 @@ repository = "https://github.com/wez/wez
 version = "1.3"
 
 [dependencies.euclid]
-version = "0.22"
+version = "0.20"
 
 [dependencies.lazy_static]
 version = "1.4"
