--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,13 +36,5 @@ build-native-freetype = ["freetype/freet
 build-native-harfbuzz = ["cc", "pkg-config"]
 default = ["build-native-harfbuzz", "build-native-freetype"]
 [target."cfg(any(target_os = \"android\", all(unix, not(target_vendor = \"apple\"))))".dependencies.freetype]
-version = "0.5"
+version = "0.7"
 default-features = false
-[target."cfg(target_vendor = \"apple\")".dependencies.core-graphics]
-version = "0.19"
-
-[target."cfg(target_vendor = \"apple\")".dependencies.core-text]
-version = "15"
-
-[target."cfg(target_vendor = \"apple\")".dependencies.foreign-types]
-version = "0.3"
