Index: b/Cargo.toml
===================================================================
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -82,7 +82,7 @@
 version = "0.2.16"
 
 [dependencies.pyo3]
-version = "0.25"
+version = ">=0.25,<0.27"
 optional = true
 
 [dependencies.smallvec]
@@ -92,7 +92,7 @@
 version = "1.0.7"
 
 [dev-dependencies.pyo3]
-version = "0.25"
+version = ">=0.25,<0.27"
 features = ["auto-initialize"]
 
 [dev-dependencies.serde]
@@ -107,7 +107,7 @@
 ]
 
 [build-dependencies.pyo3-build-config]
-version = "0.25"
+version = ">=0.25,<0.27"
 optional = true
 
 [lints.clippy]
