--- a/Cargo.toml
+++ b/Cargo.toml
@@ -59,90 +59,6 @@
 name = "nalgebra"
 path = "src/lib.rs"
 
-[[example]]
-name = "dimensional_genericity"
-path = "examples/dimensional_genericity.rs"
-
-[[example]]
-name = "homogeneous_coordinates"
-path = "examples/homogeneous_coordinates.rs"
-
-[[example]]
-name = "linear_system_resolution"
-path = "examples/linear_system_resolution.rs"
-
-[[example]]
-name = "matrix_construction"
-path = "examples/matrix_construction.rs"
-
-[[example]]
-name = "matrixcompare"
-path = "examples/matrixcompare.rs"
-required-features = ["compare"]
-
-[[example]]
-name = "mvp"
-path = "examples/mvp.rs"
-
-[[example]]
-name = "point_construction"
-path = "examples/point_construction.rs"
-
-[[example]]
-name = "raw_pointer"
-path = "examples/raw_pointer.rs"
-
-[[example]]
-name = "reshaping"
-path = "examples/reshaping.rs"
-
-[[example]]
-name = "scalar_genericity"
-path = "examples/scalar_genericity.rs"
-
-[[example]]
-name = "screen_to_view_coords"
-path = "examples/screen_to_view_coords.rs"
-
-[[example]]
-name = "transform_conversion"
-path = "examples/transform_conversion.rs"
-
-[[example]]
-name = "transform_matrix4"
-path = "examples/transform_matrix4.rs"
-
-[[example]]
-name = "transform_vector_point"
-path = "examples/transform_vector_point.rs"
-
-[[example]]
-name = "transform_vector_point3"
-path = "examples/transform_vector_point3.rs"
-
-[[example]]
-name = "transformation_pointer"
-path = "examples/transformation_pointer.rs"
-
-[[example]]
-name = "unit_wrapper"
-path = "examples/unit_wrapper.rs"
-
-[[test]]
-name = "lib"
-path = "tests/lib.rs"
-
-[[bench]]
-name = "nalgebra_bench"
-path = "benches/lib.rs"
-harness = false
-required-features = ["rand"]
-
-[dependencies.alga]
-version = "0.9"
-optional = true
-default-features = false
-
 [dependencies.approx]
 version = "0.5"
 default-features = false
@@ -151,85 +67,6 @@
 version = "1.5"
 optional = true
 
-[dependencies.glam014]
-version = "0.14"
-optional = true
-package = "glam"
-
-[dependencies.glam015]
-version = "0.15"
-optional = true
-package = "glam"
-
-[dependencies.glam016]
-version = "0.16"
-optional = true
-package = "glam"
-
-[dependencies.glam017]
-version = "0.17"
-optional = true
-package = "glam"
-
-[dependencies.glam018]
-version = "0.18"
-optional = true
-package = "glam"
-
-[dependencies.glam019]
-version = "0.19"
-optional = true
-package = "glam"
-
-[dependencies.glam020]
-version = "0.20"
-optional = true
-package = "glam"
-
-[dependencies.glam021]
-version = "0.21"
-optional = true
-package = "glam"
-
-[dependencies.glam022]
-version = "0.22"
-optional = true
-package = "glam"
-
-[dependencies.glam023]
-version = "0.23"
-optional = true
-package = "glam"
-
-[dependencies.glam024]
-version = "0.24"
-optional = true
-package = "glam"
-
-[dependencies.glam025]
-version = "0.25"
-optional = true
-package = "glam"
-
-[dependencies.glam027]
-version = "0.27"
-optional = true
-package = "glam"
-
-[dependencies.glam028]
-version = "0.28"
-optional = true
-package = "glam"
-
-[dependencies.glam029]
-version = "0.29"
-optional = true
-package = "glam"
-
-[dependencies.matrixcompare-core]
-version = "0.1"
-optional = true
-
 [dependencies.matrixmultiply]
 version = "0.3"
 optional = true
@@ -305,19 +142,13 @@
 [dependencies.typenum]
 version = "1.12"
 
-[dev-dependencies.cool_asserts]
-version = "2.0.3"
-
 [dev-dependencies.criterion]
-version = "0.4"
+version = "0.5"
 features = ["html_reports"]
 
 [dev-dependencies.itertools]
 version = "0.13"
 
-[dev-dependencies.matrixcompare]
-version = "0.3.0"
-
 [dev-dependencies.rand_isaac]
 version = "0.3"
 
@@ -333,26 +164,10 @@
 [features]
 alloc = []
 arbitrary = ["quickcheck"]
-compare = ["matrixcompare-core"]
 convert-bytemuck = [
     "bytemuck",
     "num-complex/bytemuck",
 ]
-convert-glam014 = ["glam014"]
-convert-glam015 = ["glam015"]
-convert-glam016 = ["glam016"]
-convert-glam017 = ["glam017"]
-convert-glam018 = ["glam018"]
-convert-glam019 = ["glam019"]
-convert-glam020 = ["glam020"]
-convert-glam021 = ["glam021"]
-convert-glam022 = ["glam022"]
-convert-glam023 = ["glam023"]
-convert-glam024 = ["glam024"]
-convert-glam025 = ["glam025"]
-convert-glam027 = ["glam027"]
-convert-glam028 = ["glam028"]
-convert-glam029 = ["glam029"]
 convert-mint = ["mint"]
 debug = [
     "approx/num-complex",
