Index: winit/Cargo.toml
===================================================================
--- winit.orig/Cargo.toml
+++ winit/Cargo.toml
@@ -62,3 +62,3 @@ version = "0.4"
 [dependencies.mint]
-version = "0.5.6"
+version = "0.5.5"
 optional = true
@@ -99,3 +99,3 @@ default-features = false
 [dev-dependencies.simple_logger]
-version = "4.2.0"
+version = "5"
 default_features = false
@@ -103,3 +103,3 @@ default_features = false
 [build-dependencies.cfg_aliases]
-version = "0.1.1"
+version = "0.2"
 
@@ -165,3 +165,3 @@ default-features = false
 [target."cfg(all(unix, not(any(target_os = \"redox\", target_family = \"wasm\", target_os = \"android\", target_os = \"ios\", target_os = \"macos\"))))".dependencies.calloop]
-version = "0.12.3"
+version = "0.13.0"
 
@@ -211,3 +211,3 @@ optional = true
 [target."cfg(all(unix, not(any(target_os = \"redox\", target_family = \"wasm\", target_os = \"android\", target_os = \"ios\", target_os = \"macos\"))))".dependencies.wayland-protocols]
-version = "0.31.0"
+version = "0.32.0"
 features = ["staging"]
@@ -216,3 +216,3 @@ optional = true
 [target."cfg(all(unix, not(any(target_os = \"redox\", target_family = \"wasm\", target_os = \"android\", target_os = \"ios\", target_os = \"macos\"))))".dependencies.wayland-protocols-plasma]
-version = "0.2.0"
+version = "0.3.0"
 features = ["client"]
