diff --git a/Cargo.toml b/Cargo.toml
index de4cb54..526a9e5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,7 +60,7 @@ version = "1.1.0"
 version = "0.4"
 
 [dependencies.mint]
-version = "0.5.6"
+version = "0.5.5"
 optional = true
 
 [dependencies.once_cell]
@@ -97,11 +97,11 @@ features = ["png"]
 default-features = false
 
 [dev-dependencies.simple_logger]
-version = "4.2.0"
+version = "5"
 default_features = false
 
 [build-dependencies.cfg_aliases]
-version = "0.1.1"
+version = "0.2"
 
 [features]
 default = [
