Index: winit/Cargo.toml
===================================================================
--- winit.orig/Cargo.toml
+++ winit/Cargo.toml
@@ -60,7 +60,7 @@ version = "1.1.0"
 version = "0.4"
 
 [dependencies.mint]
-version = "0.5.6"
+version = "0.5.5"
 optional = true
 
 [dependencies.once_cell]
@@ -97,11 +97,11 @@ features = ["png"]
 default-features = false
 
 [dev-dependencies.simple_logger]
-version = "4.2.0"
+version = "5"
 default_features = false
 
 [build-dependencies.cfg_aliases]
-version = "0.1.1"
+version = "0.2"
 
 [features]
 default = [
@@ -163,7 +163,7 @@ optional = true
 default-features = false
 
 [target."cfg(all(unix, not(any(target_os = \"redox\", target_family = \"wasm\", target_os = \"android\", target_os = \"ios\", target_os = \"macos\"))))".dependencies.calloop]
-version = "0.12.3"
+version = "0.13.0"
 
 [target."cfg(all(unix, not(any(target_os = \"redox\", target_family = \"wasm\", target_os = \"android\", target_os = \"ios\", target_os = \"macos\"))))".dependencies.libc]
 version = "0.2.64"
