Index: winit/Cargo.toml
===================================================================
--- winit.orig/Cargo.toml
+++ winit/Cargo.toml
@@ -127,7 +127,6 @@ default = [
     "rwh_06",
     "x11",
     "wayland",
-    "wayland-dlopen",
     "wayland-csd-adwaita",
 ]
 mint = ["dpi/mint"]
@@ -158,7 +157,6 @@ wayland-csd-adwaita-crossfont = [
     "sctk-adwaita/crossfont",
 ]
 wayland-csd-adwaita-notitle = ["sctk-adwaita"]
-wayland-dlopen = ["wayland-backend/dlopen"]
 x11 = [
     "x11-dl",
     "bytemuck",
