Index: dirs-sys-next/Cargo.toml
===================================================================
--- dirs-sys-next.orig/Cargo.toml
+++ dirs-sys-next/Cargo.toml
@@ -19,9 +19,9 @@ description = "system-level helper funct
 readme = "README.md"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/xdg-rs/dirs/tree/master/dirs-sys"
-[target."cfg(target_os = \"redox\")".dependencies.redox_users]
-version = "0.3.0"
-default-features = false
+#[target."cfg(target_os = \"redox\")".dependencies.redox_users]
+#version = "0.3.0"
+#default-features = false
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
 [target."cfg(windows)".dependencies.winapi]
