Index: actix-web/Cargo.toml
===================================================================
--- actix-web.orig/Cargo.toml
+++ actix-web/Cargo.toml
@@ -203,15 +203,6 @@ name = "worker-cpu-pin"
 path = "examples/worker-cpu-pin.rs"
 
 [[test]]
-name = "compression"
-path = "tests/compression.rs"
-required-features = [
-    "compress-brotli",
-    "compress-gzip",
-    "compress-zstd",
-]
-
-[[test]]
 name = "test-macro-import-conflict"
 path = "tests/test-macro-import-conflict.rs"
 
@@ -224,16 +215,6 @@ name = "test_httpserver"
 path = "tests/test_httpserver.rs"
 
 [[test]]
-name = "test_server"
-path = "tests/test_server.rs"
-required-features = [
-    "compress-brotli",
-    "compress-gzip",
-    "compress-zstd",
-    "cookies",
-]
-
-[[test]]
 name = "test_weird_poll"
 path = "tests/test_weird_poll.rs"
 
@@ -251,11 +232,6 @@ path = "benches/responder.rs"
 harness = false
 
 [[bench]]
-name = "server"
-path = "benches/server.rs"
-harness = false
-
-[[bench]]
 name = "service"
 path = "benches/service.rs"
 harness = false
