Index: rust-actix-web-4.11.0/src/http/header/content_length.rs
===================================================================
--- rust-actix-web-4.11.0.orig/src/http/header/content_length.rs
+++ rust-actix-web-4.11.0/src/http/header/content_length.rs
@@ -216,6 +216,7 @@ mod tests {
         assert_parse_eq::<ContentLength, _, _>([" 0 "], ContentLength(0));
 
         // large value (2^64 - 1)
+        #[cfg(target_pointer_with = "64")]// ContentLength uses a usize
         assert_parse_eq::<ContentLength, _, _>(
             ["18446744073709551615"],
             ContentLength(18_446_744_073_709_551_615),
