Index: actix-web/src/types/json.rs
===================================================================
--- actix-web.orig/src/types/json.rs
+++ actix-web/src/types/json.rs
@@ -628,10 +628,17 @@ mod tests {
             .limit(100)
             .await;
 
-        assert!(json_eq(
-            json.err().unwrap(),
+        // the precise error can vary a bit depending on dependency versions.
+        let err = json.err().unwrap();
+        let errs = format!("{}",&err);
+        assert!(errs.starts_with("JSON payload"));
+        assert!(errs.contains("limit"));
+        assert!(errs.contains("100 bytes"));
+        /*assert!(json_eq(
+            err,
             JsonPayloadError::Overflow { limit: 100 }
-        ));
+        ),"json_eq(json.err().unwrap(), JsonPayloadError::Overflow {{ limit: 100 }}) left = {}, right = {}",errs,JsonPayloadError::Overflow { limit: 100 });*/
+
 
         let (req, mut pl) = TestRequest::default()
             .insert_header((
