Index: actix-web/Cargo.toml
===================================================================
--- actix-web.orig/Cargo.toml
+++ actix-web/Cargo.toml
@@ -116,7 +116,6 @@ default = [
     "unicode",
     "compat",
 ]
-experimental-io-uring = ["actix-server/io-uring"]
 http2 = ["actix-http/http2"]
 macros = [
     "dep:actix-macros",
@@ -260,7 +259,7 @@ version = "1"
 version = "1"
 
 [dependencies.cookie]
-version = "0.16"
+version = "0.18"
 features = ["percent-encode"]
 optional = true
 
@@ -279,7 +278,7 @@ features = [
 version = "0.8"
 
 [dependencies.foldhash]
-version = "0.1"
+version = "0.2"
 
 [dependencies.futures-core]
 version = "0.3.17"
@@ -349,10 +348,10 @@ default-features = false
 features = ["macros"]
 
 [dev-dependencies.brotli]
-version = "8"
+version = ">= 6, < 9"
 
 [dev-dependencies.const-str]
-version = "0.5"
+version = "0.7"
 
 [dev-dependencies.core_affinity]
 version = "0.8"
@@ -373,10 +372,10 @@ features = ["std"]
 default-features = false
 
 [dev-dependencies.rand]
-version = "0.9"
+version = ">= 0.8.5, < 0.10"
 
 [dev-dependencies.rcgen]
-version = "0.13"
+version = "0.14"
 
 [dev-dependencies.rustls-pemfile]
 version = "2"
