--- a/Cargo.toml
+++ b/Cargo.toml
@@ -204,7 +204,2 @@
 
-[dependencies.actix-tls]
-version = "3.4"
-optional = true
-default-features = false
-
 [dependencies.actix-utils]
@@ -230,3 +225,3 @@
 [dependencies.cookie]
-version = "0.16"
+version = "0.18"
 features = ["percent-encode"]
@@ -406,7 +401,2 @@
 openssl = [
-    "__tls",
-    "http2",
-    "actix-http/openssl",
-    "actix-tls/accept",
-    "actix-tls/openssl",
 ]
@@ -414,28 +404,8 @@
 rustls-0_20 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_20",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_20",
 ]
 rustls-0_21 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_21",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_21",
 ]
 rustls-0_22 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_22",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_22",
 ]
 rustls-0_23 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_23",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_23",
 ]
