Index: actix-web/Cargo.toml
===================================================================
--- actix-web.orig/Cargo.toml
+++ actix-web/Cargo.toml
@@ -189,10 +189,6 @@ name = "test_error_propagation"
 path = "tests/test_error_propagation.rs"
 
 [[test]]
-name = "test_httpserver"
-path = "tests/test_httpserver.rs"
-
-[[test]]
 name = "test_weird_poll"
 path = "tests/test_weird_poll.rs"
 
@@ -346,12 +342,10 @@ version = "0.1.30"
 [dependencies.url]
 version = "2.5.4"
 
-[dev-dependencies.actix-test]
-version = "0.1"
-features = [
-    "openssl",
-    "rustls-0_23",
-]
+[dev-dependencies.actix-rt]
+version = "2.6"
+default-features = false
+features = ["macros"]
 
 [dev-dependencies.awc]
 version = "3"
