Index: actix-web/Cargo.toml
===================================================================
--- actix-web.orig/Cargo.toml
+++ actix-web/Cargo.toml
@@ -128,28 +128,7 @@ openssl = [
     "actix-tls/accept",
     "actix-tls/openssl",
 ]
-rustls = ["rustls-0_20"]
-rustls-0_20 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_20",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_20",
-]
-rustls-0_21 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_21",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_21",
-]
-rustls-0_22 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_22",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_22",
-]
+rustls = ["rustls-0_23"]
 rustls-0_23 = [
     "__tls",
     "http2",
