--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,8 +57,6 @@
 https = ["openssl-sys"]
 ssh = ["libssh2-sys"]
 ssh_key_from_memory = []
-vendored = []
-vendored-openssl = ["openssl-sys/vendored"]
 
 [target."cfg(unix)".dependencies.openssl-sys]
 version = "0.9"
--- a/build.rs
+++ b/build.rs
@@ -22,6 +22,8 @@
         }
     }
 
+    panic!("debian build must never use vendored libgit2!");
+
     println!("cargo:rustc-cfg=libgit2_vendored");
 
     if !Path::new("libgit2/src").exists() {
