Index: libgit2-sys/Cargo.toml
===================================================================
--- libgit2-sys.orig/Cargo.toml
+++ libgit2-sys/Cargo.toml
@@ -39,8 +39,6 @@ repository = "https://github.com/rust-la
 [features]
 https = ["openssl-sys"]
 ssh = ["libssh2-sys"]
-vendored = []
-vendored-openssl = ["openssl-sys/vendored"]
 
 [lib]
 name = "libgit2_sys"
Index: libgit2-sys/build.rs
===================================================================
--- libgit2-sys.orig/build.rs
+++ libgit2-sys/build.rs
@@ -61,6 +61,8 @@ The build is now aborting. To disable, u
         return;
     }
 
+    panic!("debian build must never use vendored libgit2!");
+
     println!("cargo:rustc-cfg=libgit2_vendored");
 
     if !Path::new("libgit2/src").exists() {
