Index: libgit2-sys/Cargo.toml
===================================================================
--- libgit2-sys.orig/Cargo.toml
+++ libgit2-sys/Cargo.toml
@@ -62,8 +62,6 @@ version = "0.3.15"
 [features]
 https = ["openssl-sys"]
 ssh = ["libssh2-sys"]
-vendored = []
-vendored-openssl = ["openssl-sys/vendored"]
 
 [target."cfg(unix)".dependencies.openssl-sys]
 version = "0.9.45"
Index: libgit2-sys/build.rs
===================================================================
--- libgit2-sys.orig/build.rs
+++ libgit2-sys/build.rs
@@ -61,6 +61,8 @@ The build is now aborting. To disable, u
         return;
     }
 
+    panic!("debian build must never use vendored libgit2!");
+
     println!("cargo:rustc-cfg=libgit2_vendored");
 
     if !Path::new("libgit2/src").exists() {
