Index: globset/Cargo.toml
===================================================================
--- globset.orig/Cargo.toml
+++ globset/Cargo.toml
@@ -59,10 +59,6 @@ features = [
 ]
 default-features = false
 
-[dependencies.serde]
-version = "1.0.104"
-optional = true
-
 [dev-dependencies.glob]
 version = "0.3.0"
 
@@ -74,5 +70,4 @@ version = "1.0.45"
 
 [features]
 default = ["log"]
-serde1 = ["serde"]
 simd-accel = []
