Description: Remove unused dependencies
Author: Ermakov Alexander <ermakovav@altgtu.ru>
Forwarded: not-needed
Last-Update: 2025-10-27

--- rust-shell-quote.orig/Cargo.toml
+++ rust-shell-quote/Cargo.toml
@@ -50,19 +50,6 @@ path = "src/lib.rs"
 version = "1"
 optional = true
 
-[dev-dependencies.criterion]
-version = "^0.5.1"
-features = ["html_reports"]
-
-[dev-dependencies.lenient_semver]
-version = "0.4.2"
-
-[dev-dependencies.semver]
-version = "1.0.23"
-
-[dev-dependencies.test-case]
-version = "3.3.1"
-
 [features]
 bash = []
 default = [
