Index: keyring/Cargo.toml
===================================================================
--- keyring.orig/Cargo.toml
+++ keyring/Cargo.toml
@@ -82,10 +82,6 @@ name = "keyring"
 path = "src/lib.rs"
 
 
-[[example]]
-name = "keyring-cli"
-path = "examples/cli.rs"
-
 [[test]]
 name = "basic"
 path = "tests/basic.rs"
@@ -123,9 +119,6 @@ version = "2"
 [dev-dependencies.rpassword]
 version = "7"
 
-[dev-dependencies.rprompt]
-version = "2"
-
 [dev-dependencies.whoami]
 version = "1.5"
 
