diff --git a/Cargo.toml b/Cargo.toml
index e0064f6..c7197bc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -115,6 +115,18 @@ optional = true
 default-features = false
 
 [dev-dependencies]
+get-size2 = { path = ".", features = [
+    "derive",
+    "bytes",
+    "chrono",
+    "chrono-tz",
+    "compact-str",
+    "hashbrown",
+    "indexmap",
+    "smallvec",
+    "thin-vec",
+    "url",
+] }
 
 [lints.clippy]
 allow_attributes = "warn"
