--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,11 +46,3 @@
 [dependencies.snafu]
 version = "0.7"
 
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.core-foundation]
-version = "0.9"
-
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.core-foundation-sys]
-version = "0.8"
-
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.io-kit-sys]
-version = "0.4"
