--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,23 +45,3 @@
 
 [dependencies.snafu]
 version = "0.7"
-
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.core-foundation]
-version = "0.9"
-
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.core-foundation-sys]
-version = "0.8"
-
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.io-kit-sys]
-version = "0.4"
-
-[target."cfg(windows)".dependencies.windows]
-version = "0.62"
-features = [
-    "Win32_Storage_FileSystem",
-    "Win32_System_SystemServices",
-    "Win32_Security",
-    "Win32_System_Ioctl",
-    "Win32_System_IO",
-    "Win32_System_WindowsProgramming",
-]
