Origin: vendor
Subject: Relax dependencies to work well with other rust crates in Debian
Forwarded: not-needed

Index: sequoia-keystore-tpm/Cargo.toml
===================================================================
--- sequoia-keystore-tpm.orig/Cargo.toml
+++ sequoia-keystore-tpm/Cargo.toml
@@ -56,7 +56,6 @@ version = "0.4.17"
 
 [dependencies.openpgp-cert-d]
 version = "0.3.4"
-default-features = false
 
 [dependencies.sequoia-ipc]
 version = "0.36"
@@ -76,7 +75,7 @@ version = "0.1.0"
 version = "1"
 
 [dependencies.serde_yaml]
-version = "0.8"
+version = "0.9"
 
 [dependencies.tokio]
 version = "1.19"
@@ -89,11 +88,6 @@ features = [
 [dev-dependencies.env_logger]
 version = ">=0.10, <0.12"
 
-[dev-dependencies.sequoia-openpgp]
-version = "2"
-features = ["compression"]
-default-features = false
-
 [dev-dependencies.tracing]
 version = "0.1"
 default-features = false
@@ -110,11 +104,3 @@ features = [
     "__implicit-crypto-backend-for-tests",
 ]
 default-features = false
-
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "2"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
