Index: cargo-metadata/Cargo.toml
===================================================================
--- cargo-metadata.orig/Cargo.toml
+++ cargo-metadata/Cargo.toml
@@ -31,10 +31,6 @@ features = ["serde1"]
 [dependencies.cargo-platform]
 version = "0.2"
 
-[dependencies.derive_builder]
-version = "< 0.13"
-optional = true
-
 [dependencies.semver]
 version = "1.0.7"
 features = ["serde"]
@@ -51,5 +47,4 @@ features = ["unbounded_depth"]
 version = "1.0.31"
 
 [features]
-builder = ["derive_builder"]
 default = []
