Index: peak-mem/Cargo.toml
===================================================================
--- peak-mem.orig/Cargo.toml
+++ peak-mem/Cargo.toml
@@ -43,3 +43,3 @@
 [dependencies.clap]
-version = "=4.5.23"
+version = "4.5.23"
 features = [
@@ -49,24 +49,24 @@ features = [
 ]
 
 [dependencies.crossterm]
-version = "=0.27.0"
+version = "0.28"
 
 [dependencies.libc]
-version = "=0.2.169"
+version = "0.2"
 
 [dependencies.nix]
 version = "=0.29.0"
 features = ["signal"]
 
 [dependencies.serde]
-version = "=1.0.217"
+version = "1.0"
 features = ["derive"]
 
 [dependencies.serde_json]
-version = "=1.0.139"
+version = "1.0"
 
 [dependencies.tokio]
-version = "=1.43.1"
+version = "1.43"
 features = ["full"]
 
 [dev-dependencies.assert_cmd]
