From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Sun, 25 Aug 2024 09:12:09 -0400
Subject: relax deps

---
 Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 2b6cd06..ab6a6a0 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,12 +50,12 @@ version = "0.3.30"
 version = "0.3.30"
 
 [dependencies.gdk]
-version = "0.8.0"
+version = "0.9.0"
 features = ["v4_12"]
 package = "gdk4"
 
 [dependencies.gio]
-version = "0.19.0"
+version = "0.20.0"
 
 [dependencies.glycin-utils]
 version = "1.0.0"
