Index: glycin/Cargo.toml
===================================================================
--- glycin.orig/Cargo.toml
+++ glycin/Cargo.toml
@@ -54,7 +54,7 @@ package = "gdk4"
 version = "0.18.1"
 
 [dependencies.glycin-utils]
-version = "0.1.0-beta.2"
+version = "0.1.0"
 
 [dependencies.lcms2]
 version = "5.6.0"
@@ -66,7 +66,7 @@ version = "4.0.1"
 version = "0.6.3"
 
 [dependencies.memmap]
-version = "0.7.0"
+version = "0.9"
 package = "memmap2"
 
 [dependencies.nix]
@@ -79,4 +79,4 @@ version = "0.8.36"
 version = "0.11.2"
 
 [dependencies.zbus]
-version = "3.13.1"
+version = "3.12"
