Description: revert
 https://gitlab.gnome.org/GNOME/glycin/-/commit/88477123
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -144,7 +144,7 @@
 package = "memmap2"
 
 [dependencies.nix]
-version = "0.30.0"
+version = "0.29.0"
 features = [
     "fs",
     "resource",
@@ -189,7 +189,7 @@
 version = "6.0.0"
 
 [dependencies.zbus]
-version = "5.10.0"
+version = ">= 5.6"
 features = [
     "p2p",
     "p2p",
--- a/src/sandbox.rs
+++ b/src/sandbox.rs
@@ -234,11 +234,11 @@
 
         // Allow FD to be passed to child process
         let mut flags = nix::fcntl::FdFlag::from_bits_truncate(nix::fcntl::fcntl(
-            &self.dbus_socket,
+            dbus_fd,
             nix::fcntl::FcntlArg::F_GETFD,
         )?);
         flags.remove(nix::fcntl::FdFlag::FD_CLOEXEC);
-        nix::fcntl::fcntl(&self.dbus_socket, nix::fcntl::FcntlArg::F_SETFD(flags))?;
+        nix::fcntl::fcntl(dbus_fd, nix::fcntl::FcntlArg::F_SETFD(flags))?;
 
         command.stderr(Stdio::piped());
         command.stdout(Stdio::piped());
