From: Andreas Schwab <schwab@suse.de>
Date: Sun, 5 Oct 2025 22:50:21 +0200
Subject: sandbox: allow riscv_hwprobe

Origin: https://gitlab.gnome.org/GNOME/glycin/-/merge_requests/316
--- a/src/sandbox.rs
+++ b/src/sandbox.rs
@@ -115,6 +115,7 @@ const ALLOWED_SYSCALLS: &[&str] = &[
     "recvfrom",
     "recvmsg",
     "restart_syscall",
+    "riscv_hwprobe",
     "rseq",
     "rt_sigaction",
     "rt_sigprocmask",
