diff --git a/Cargo.toml b/Cargo.toml
index 3105864..d1e2950 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -136,7 +136,6 @@ version = "0.3"
 features = [
     "std",
     "fmt",
-    "env-filter",
 ]
 
 [features]
@@ -185,7 +184,7 @@ serde-config = [
     "serde",
     "hickory-proto/serde-config",
 ]
-webpki-roots = ["hickory-proto/webpki-roots"]
+#webpki-roots = ["hickory-proto/webpki-roots"]
 
 [badges.codecov]
 branch = "main"
