Index: octocrab/Cargo.toml
===================================================================
--- octocrab.orig/Cargo.toml
+++ octocrab/Cargo.toml
@@ -51,18 +51,10 @@ name = "create_a_gist"
 path = "examples/create_a_gist.rs"
 
 [[example]]
-name = "create_org_secret"
-path = "examples/create_org_secret.rs"
-
-[[example]]
 name = "create_repo_from_template"
 path = "examples/create_repo_from_template.rs"
 
 [[example]]
-name = "create_repo_secret"
-path = "examples/create_repo_secret.rs"
-
-[[example]]
 name = "custom_client"
 path = "examples/custom_client.rs"
 
@@ -603,10 +595,6 @@ features = ["serde"]
 [dev-dependencies.base64]
 version = "0.22.0"
 
-[dev-dependencies.crypto_box]
-version = "0.8.2"
-features = ["seal"]
-
 [dev-dependencies.graphql_client]
 version = "0.14.0"
 
