Index: octocrab/src/lib.rs
===================================================================
--- octocrab.orig/src/lib.rs
+++ octocrab/src/lib.rs
@@ -1783,7 +1783,7 @@ mod tests {
         assert_eq!(uri.path(), "/help%20world");
     }
 
-    #[tokio::test]
+    /*#[tokio::test]
     async fn extra_headers() {
         use http::header::HeaderName;
         use wiremock::{matchers, Mock, MockServer, ResponseTemplate};
@@ -1809,7 +1809,7 @@ mod tests {
             .send()
             .await
             .unwrap();
-    }
+    }*/
 
     use super::*;
     use chrono::Duration;
Index: octocrab/Cargo.toml
===================================================================
--- octocrab.orig/Cargo.toml
+++ octocrab/Cargo.toml
@@ -184,273 +184,9 @@ name = "update_pull_request_branch"
 path = "examples/update_pull_request_branch.rs"
 
 [[test]]
-name = "actions_add_selected_repo_to_org_secret_test"
-path = "tests/actions_add_selected_repo_to_org_secret_test.rs"
-
-[[test]]
-name = "actions_delete_workflow_run_logs_test"
-path = "tests/actions_delete_workflow_run_logs_test.rs"
-
-[[test]]
-name = "actions_remove_selected_repo_from_org_secret_test"
-path = "tests/actions_remove_selected_repo_from_org_secret_test.rs"
-
-[[test]]
-name = "actions_self_hosted_runners"
-path = "tests/actions_self_hosted_runners.rs"
-
-[[test]]
-name = "actions_workflows_dispatches_test"
-path = "tests/actions_workflows_dispatches_test.rs"
-
-[[test]]
-name = "check_suites"
-path = "tests/check_suites.rs"
-
-[[test]]
-name = "code_scanning_alert_update_test"
-path = "tests/code_scanning_alert_update_test.rs"
-
-[[test]]
-name = "code_scanning_alerts_list_test"
-path = "tests/code_scanning_alerts_list_test.rs"
-
-[[test]]
-name = "code_scanning_analysis_test"
-path = "tests/code_scanning_analysis_test.rs"
-
-[[test]]
-name = "commit_associated_check_runs_tests"
-path = "tests/commit_associated_check_runs_tests.rs"
-
-[[test]]
-name = "current_user_orgs_test"
-path = "tests/current_user_orgs_test.rs"
-
-[[test]]
-name = "dependabot_alerts_test"
-path = "tests/dependabot_alerts_test.rs"
-
-[[test]]
-name = "events_test"
-path = "tests/events_test.rs"
-
-[[test]]
-name = "follow_redirect"
-path = "tests/follow_redirect.rs"
-
-[[test]]
-name = "generate_release_notes_test"
-path = "tests/generate_release_notes_test.rs"
-
-[[test]]
-name = "gists_test"
-path = "tests/gists_test.rs"
-
-[[test]]
-name = "hooks_delivery_list"
-path = "tests/hooks_delivery_list.rs"
-
-[[test]]
-name = "issues_check_assignee_test"
-path = "tests/issues_check_assignee_test.rs"
-
-[[test]]
-name = "issues_comments_delete_test"
-path = "tests/issues_comments_delete_test.rs"
-
-[[test]]
-name = "issues_comments_reactions_delete_test"
-path = "tests/issues_comments_reactions_delete_test.rs"
-
-[[test]]
-name = "issues_labels_delete_test"
-path = "tests/issues_labels_delete_test.rs"
-
-[[test]]
-name = "issues_locking_test"
-path = "tests/issues_locking_test.rs"
-
-[[test]]
-name = "issues_reactions_delete_test"
-path = "tests/issues_reactions_delete_test.rs"
-
-[[test]]
 name = "issues_timeline_tests"
 path = "tests/issues_timeline_tests.rs"
 
-[[test]]
-name = "mock_error"
-path = "tests/mock_error.rs"
-
-[[test]]
-name = "notifications_threads_test"
-path = "tests/notifications_threads_test.rs"
-
-[[test]]
-name = "org_installations_test"
-path = "tests/org_installations_test.rs"
-
-[[test]]
-name = "org_members_tests"
-path = "tests/org_members_tests.rs"
-
-[[test]]
-name = "org_secrets_test"
-path = "tests/org_secrets_test.rs"
-
-[[test]]
-name = "projects_org_create_project_test"
-path = "tests/projects_org_create_project_test.rs"
-
-[[test]]
-name = "projects_org_list_projects_test"
-path = "tests/projects_org_list_projects_test.rs"
-
-[[test]]
-name = "projects_project_delete_test"
-path = "tests/projects_project_delete_test.rs"
-
-[[test]]
-name = "projects_project_get_test"
-path = "tests/projects_project_get_test.rs"
-
-[[test]]
-name = "projects_project_update_test"
-path = "tests/projects_project_update_test.rs"
-
-[[test]]
-name = "projects_repo_create_project_test"
-path = "tests/projects_repo_create_project_test.rs"
-
-[[test]]
-name = "projects_repo_list_projects_test"
-path = "tests/projects_repo_list_projects_test.rs"
-
-[[test]]
-name = "projects_user_create_project_test"
-path = "tests/projects_user_create_project_test.rs"
-
-[[test]]
-name = "projects_user_list_projects_test"
-path = "tests/projects_user_list_projects_test.rs"
-
-[[test]]
-name = "pull_request_commits_test"
-path = "tests/pull_request_commits_test.rs"
-
-[[test]]
-name = "pull_request_review_comment_test"
-path = "tests/pull_request_review_comment_test.rs"
-
-[[test]]
-name = "pull_request_review_operations_test"
-path = "tests/pull_request_review_operations_test.rs"
-
-[[test]]
-name = "repo_contributors_test"
-path = "tests/repo_contributors_test.rs"
-
-[[test]]
-name = "repo_delete_ref_test"
-path = "tests/repo_delete_ref_test.rs"
-
-[[test]]
-name = "repo_secrets_test"
-path = "tests/repo_secrets_test.rs"
-
-[[test]]
-name = "repos_commit_test"
-path = "tests/repos_commit_test.rs"
-
-[[test]]
-name = "repos_delete_test"
-path = "tests/repos_delete_test.rs"
-
-[[test]]
-name = "repos_events_test"
-path = "tests/repos_events_test.rs"
-
-[[test]]
-name = "repos_is_collaborator_test"
-path = "tests/repos_is_collaborator_test.rs"
-
-[[test]]
-name = "repos_merges_test"
-path = "tests/repos_merges_test.rs"
-
-[[test]]
-name = "repos_releases_get_by_id"
-path = "tests/repos_releases_get_by_id.rs"
-
-[[test]]
-name = "repos_releases_get_by_tag"
-path = "tests/repos_releases_get_by_tag.rs"
-
-[[test]]
-name = "repos_releases_get_latest"
-path = "tests/repos_releases_get_latest.rs"
-
-[[test]]
-name = "repos_releases_list_test"
-path = "tests/repos_releases_list_test.rs"
-
-[[test]]
-name = "repos_stargazers_tests"
-path = "tests/repos_stargazers_tests.rs"
-
-[[test]]
-name = "secrets_alerts_test"
-path = "tests/secrets_alerts_test.rs"
-
-[[test]]
-name = "team_delete_test"
-path = "tests/team_delete_test.rs"
-
-[[test]]
-name = "team_invitations_tests"
-path = "tests/team_invitations_tests.rs"
-
-[[test]]
-name = "team_members_tests"
-path = "tests/team_members_tests.rs"
-
-[[test]]
-name = "team_repos_tests"
-path = "tests/team_repos_tests.rs"
-
-[[test]]
-name = "user_blocks_tests"
-path = "tests/user_blocks_tests.rs"
-
-[[test]]
-name = "user_deserialize_test"
-path = "tests/user_deserialize_test.rs"
-
-[[test]]
-name = "user_emails_tests"
-path = "tests/user_emails_tests.rs"
-
-[[test]]
-name = "user_git_ssh_keys_tests"
-path = "tests/user_git_ssh_keys_tests.rs"
-
-[[test]]
-name = "user_gpg_keys_tests"
-path = "tests/user_gpg_keys_tests.rs"
-
-[[test]]
-name = "user_repositories_tests"
-path = "tests/user_repositories_tests.rs"
-
-[[test]]
-name = "user_social_accounts_tests"
-path = "tests/user_social_accounts_tests.rs"
-
-[[test]]
-name = "user_ssh_signing_keys_tests"
-path = "tests/user_ssh_signing_keys_tests.rs"
-
 [dependencies.arc-swap]
 version = "1.3.0"
 
@@ -607,9 +343,6 @@ default-features = false
 [dev-dependencies.tokio-test]
 version = "0.4.2"
 
-[dev-dependencies.wiremock]
-version = "0.6.0"
-
 [features]
 default = [
     "follow-redirect",
