Index: octocrab/Cargo.toml
===================================================================
--- octocrab.orig/Cargo.toml
+++ octocrab/Cargo.toml
@@ -102,9 +102,6 @@ path = "examples/github_app_authenticati
 name = "github_app_authentication_manual"
 path = "examples/github_app_authentication_manual.rs"
 
-[[example]]
-name = "graphql_issues"
-path = "examples/graphql_issues.rs"
 
 [[example]]
 name = "is_collab"
@@ -595,9 +592,6 @@ features = ["serde"]
 [dev-dependencies.base64]
 version = "0.22.0"
 
-[dev-dependencies.graphql_client]
-version = "0.14.0"
-
 [dev-dependencies.pretty_assertions]
 version = "1.4.0"
 
