Index: octocrab/Cargo.toml
===================================================================
--- octocrab.orig/Cargo.toml
+++ octocrab/Cargo.toml
@@ -261,7 +261,7 @@ version = "0.1.3"
 features = ["http1"]
 
 [dependencies.jsonwebtoken]
-version = "9"
+version = ">= 8, < 10"
 
 [dependencies.once_cell]
 version = "1.7.2"
@@ -273,7 +273,7 @@ version = "2.2.0"
 version = "1.0.12"
 
 [dependencies.secrecy]
-version = "0.10.3"
+version = ">= 0.8, < 0.11"
 
 [dependencies.serde]
 version = "1.0.126"
