Index: b/Cargo.toml
===================================================================
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -108,7 +108,7 @@
 version = "2.3.1"
 
 [dependencies.pyo3]
-version = "0.25"
+version = ">=0.25,<0.27"
 features = [
     "chrono",
     "serde",
@@ -136,7 +136,7 @@
 version = "1"
 
 [dev-dependencies.pyo3]
-version = "0.25"
+version = ">=0.25,<0.27"
 features = ["chrono"]
 
 [dev-dependencies.serial_test]
