--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,7 +57,7 @@
 version = "0.8.5"
 
 [dev-dependencies.rstest]
-version = "0.24.0"
+version = "0.17.0"
 
 [dev-dependencies.serde]
 version = "1.0.89"
@@ -71,9 +71,6 @@
     "dep:crossbeam-channel",
     "notify/crossbeam-channel",
 ]
-default = ["macos_fsevent"]
-macos_fsevent = ["notify/macos_fsevent"]
-macos_kqueue = ["notify/macos_kqueue"]
+default = [ ]
 serde = ["notify-types/serde"]
 serialization-compat-6 = ["notify/serialization-compat-6"]
-web-time = ["notify-types/web-time"]
