Index: wl-clipboard-rs/Cargo.toml
===================================================================
--- wl-clipboard-rs.orig/Cargo.toml
+++ wl-clipboard-rs/Cargo.toml
@@ -26,7 +26,7 @@ license = "MIT/Apache-2.0"
 repository = "https://github.com/YaLTeR/wl-clipboard-rs"
 
 [dependencies.derive-new]
-version = "0.5.9"
+version = "0.5"
 
 [dependencies.libc]
 version = "0.2.119"
@@ -35,10 +35,11 @@ version = "0.2.119"
 version = "0.4.14"
 
 [dependencies.nix]
-version = "0.24.2"
+version = ">= 0.24.2, < 1.0"
+features = ["fs", "process"]
 
 [dependencies.os_pipe]
-version = "1.0.1"
+version = "1"
 
 [dependencies.tempfile]
 version = "3.3.0"
@@ -68,6 +69,11 @@ features = [
 
 [dev-dependencies.wayland-server]
 version = "0.29.4"
+features = [
+    "dlopen",
+    "use_system_lib",
+    "parking_lot",
+]
 
 [features]
 dlopen = [
