Index: bytecount/Cargo.toml
===================================================================
--- bytecount.orig/Cargo.toml
+++ bytecount/Cargo.toml
@@ -33,17 +33,17 @@ repository = "https://github.com/llogiq/
 [lib]
 bench = false
 
-[[bench]]
-name = "bench"
-harness = false
+# [[bench]]
+# name = "bench"
+# harness = false
 
 # [dependencies.packed_simd]
 # version = "0.3.8"
 # optional = true
 
-[dev-dependencies.criterion]
-version = "0.4"
-default-features = false
+# [dev-dependencies.criterion]
+# version = "0.4"
+# default-features = false
 
 [dev-dependencies.quickcheck]
 version = "1.0"
