--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,4 +43,4 @@
 
-[dependencies.memmap]
-version = "^0.7.0"
+[dependencies.memmap2]
+version = "0"
 
--- a/src/main.rs
+++ b/src/main.rs
@@ -218,7 +218,7 @@
 
     // Read the file
     let f = File::open(&args.path).expect("can't open object file");
-    let buf = unsafe { memmap::Mmap::map(&f).expect("can't memmap object file") };
+    let buf = unsafe { memmap2::Mmap::map(&f).expect("can't memmap object file") };
     let obj = object::File::parse(&*buf).expect("can't parse object file");
     let obj_arch = obj.architecture();
 
