--- a/Cargo.toml
+++ b/Cargo.toml
@@ -100,1 +100,1 @@
-color = ["ansi_term", "atty"]
+color = ["ansiterm", "atty"]
@@ -110,1 +110,1 @@
-[target."cfg(not(windows))".dependencies.ansi_term]
+[target."cfg(not(windows))".dependencies.ansiterm]
--- a/src/fmt.rs
+++ b/src/fmt.rs
@@ -1,8 +1,8 @@
 #[cfg(all(feature = "color", not(target_os = "windows")))]
-use ansi_term::ANSIString;
+use ansiterm::ANSIString;
 
 #[cfg(all(feature = "color", not(target_os = "windows")))]
-use ansi_term::Colour::{Green, Red, Yellow};
+use ansiterm::Colour::{Green, Red, Yellow};
 
 use std::env;
 use std::fmt;
@@ -169,8 +169,8 @@ impl<T: fmt::Display> fmt::Display for Format<T> {
 #[cfg(all(test, feature = "color", not(target_os = "windows")))]
 mod test {
     use super::Format;
-    use ansi_term::ANSIString;
-    use ansi_term::Colour::{Green, Red, Yellow};
+    use ansiterm::ANSIString;
+    use ansiterm::Colour::{Green, Red, Yellow};
 
     #[test]
     fn colored_output() {
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -541,7 +541,7 @@
 #![allow(bare_trait_objects, deprecated)]
 
 #[cfg(all(feature = "color", not(target_os = "windows")))]
-extern crate ansi_term;
+extern crate ansiterm;
 #[cfg(feature = "color")]
 extern crate atty;
 #[macro_use]
