diff --git a/src/core/code.rs b/src/core/code.rs
index fb87217..a31ee23 100644
--- a/src/core/code.rs
+++ b/src/core/code.rs
@@ -87,6 +87,7 @@ impl CodeSection {
     /// into a new code section encoder:
     ///
     /// ```
+    /// #[cfg(feature = "wasmparser")] {
     /// # use wasmparser::{BinaryReader, CodeSectionReader};
     /// //                  id, size, # entries, entry
     /// let code_section = [10, 6,    1,         4, 0, 65, 0, 11];
@@ -101,6 +102,7 @@ impl CodeSection {
     /// // than re-parsing and re-encoding it.
     /// let mut encoder = wasm_encoder::CodeSection::new();
     /// encoder.raw(&code_section[body_range.start..body_range.end]);
+    /// }
     /// ```
     pub fn raw(&mut self, data: &[u8]) -> &mut Self {
         data.encode(&mut self.bytes);
diff --git a/src/core/dump.rs b/src/core/dump.rs
index dd6db41..12b92dd 100644
--- a/src/core/dump.rs
+++ b/src/core/dump.rs
@@ -334,7 +334,7 @@ impl Encode for CoreDumpValue {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "wasmparser"))]
 mod tests {
     use super::*;
     use crate::Module;
diff --git a/src/core/producers.rs b/src/core/producers.rs
index 38e7f0d..e5fbfaa 100644
--- a/src/core/producers.rs
+++ b/src/core/producers.rs
@@ -116,7 +116,7 @@ impl Encode for ProducersField {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "wasmparser"))]
 mod test {
     #[test]
     fn roundtrip_example() {
diff --git a/src/core/types.rs b/src/core/types.rs
index a84ffd3..40ca95d 100644
--- a/src/core/types.rs
+++ b/src/core/types.rs
@@ -700,7 +700,7 @@ impl<'a> CoreTypeEncoder<'a> {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "wasmparser"))]
 mod tests {
     use super::*;
     use crate::Module;
diff --git a/src/lib.rs b/src/lib.rs
index d53a8e2..01fe34c 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -25,6 +25,7 @@
 //! then we would do this:
 //!
 //! ```
+//! #[cfg(feature = "wasmparser")] {
 //! use wasm_encoder::{
 //!     CodeSection, ExportKind, ExportSection, Function, FunctionSection,
 //!     Module, TypeSection, ValType,
@@ -67,6 +68,7 @@
 //!
 //! // We generated a valid Wasm module!
 //! assert!(wasmparser::validate(&wasm_bytes).is_ok());
+//! }
 //! ```
 
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
@@ -201,7 +203,7 @@ fn encode_section(sink: &mut Vec<u8>, count: u32, bytes: &[u8]) {
     sink.extend(bytes);
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "wasmparser"))]
 mod test {
     use super::*;
 
