--- a/tests/test.rs
+++ b/tests/test.rs
@@ -7,7 +7,7 @@
 use std::{collections::HashMap, path::Path, str};
 
 use anyhow::{Context as _, Result};
-use build_context::TARGET;
+//use build_context::TARGET;
 use cargo_config2::*;
 use helper::*;
 
@@ -262,7 +262,7 @@
     let _config = de(&fixtures_path().join("reference")).unwrap();
 }
 
-#[rustversion::attr(not(nightly), ignore)]
+/*#[rustversion::attr(not(nightly), ignore)]
 #[cfg_attr(miri, ignore)] // Miri doesn't support pipe2 (inside duct::Expression::read)
 #[test]
 fn cargo_config_json() {
@@ -309,4 +309,4 @@
     assert!(stderr.contains("--cfg b"), "actual:\n---\n{stderr}\n---\n");
 
     Ok(())
-}
+}*/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,9 +56,6 @@
 [dev-dependencies.anyhow]
 version = "1"
 
-[dev-dependencies.build-context]
-version = "0.1"
-
 [dev-dependencies.clap]
 version = "4"
 features = [
