Index: cargo-c/Cargo.toml
===================================================================
--- cargo-c.orig/Cargo.toml
+++ cargo-c/Cargo.toml
@@ -58,7 +58,7 @@ version = "0.76.0"
 version = "0.2"
 
 [dependencies.cbindgen]
-version = "0.26.0"
+version = "0.27.0"
 default-features = false
 
 [dependencies.cc]
@@ -76,7 +76,7 @@ features = [
 version = "0.3"
 
 [dependencies.itertools]
-version = "0.12"
+version = ">= 0.10, < 1.0"
 
 [dependencies.log]
 version = "0.4"
