Index: cargo-c/Cargo.toml
===================================================================
--- cargo-c.orig/Cargo.toml
+++ cargo-c/Cargo.toml
@@ -58,7 +58,7 @@ version = "0.70.0"
 version = "0.2"
 
 [dependencies.cbindgen]
-version = "0.24.3"
+version = "0.26"
 default-features = false
 
 [dependencies.cc]
@@ -76,7 +76,7 @@ features = [
 version = "0.3"
 
 [dependencies.itertools]
-version = "0.10"
+version = ">= 0.10, < 1.0"
 
 [dependencies.log]
 version = "0.4"
@@ -97,7 +97,7 @@ version = "1.0"
 version = "1.0.62"
 
 [dependencies.toml]
-version = "0.7"
+version = ">= 0.7, < 1.0"
 
 [features]
 default = []
