Index: fancy-regex/tests/oniguruma.rs
===================================================================
--- fancy-regex.orig/tests/oniguruma.rs
+++ fancy-regex/tests/oniguruma.rs
@@ -200,7 +200,7 @@ fn run_test(test: &Test) -> Option<Strin
     }
 }
 
-#[test]
+/*#[test]
 fn oniguruma() {
     let tests: Vec<Test> = parse_tests(include_str!("oniguruma/test_utf8.c"))
         .into_iter()
@@ -244,4 +244,4 @@ fn oniguruma() {
         "{} successful Oniguruma tests, {} ignored",
         success, ignored
     );
-}
+}*/
