Index: const-fn/src/lib.rs
===================================================================
--- const-fn.orig/src/lib.rs
+++ const-fn/src/lib.rs
@@ -72,7 +72,7 @@ be maintained manually)
     no_crate_inject,
     attr(
         deny(warnings, rust_2018_idioms, single_use_lifetimes),
-        allow(dead_code, unused_variables)
+        allow(dead_code, unused_variables, unexpected_cfgs)
     )
 ))]
 #![forbid(unsafe_code)]
