Index: value-bag/src/internal/sval/v2.rs
===================================================================
--- value-bag.orig/src/internal/sval/v2.rs
+++ value-bag/src/internal/sval/v2.rs
@@ -431,13 +431,13 @@ mod tests {
         );
     }
 
-    #[test]
+    /*#[test]
     #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
     fn sval2_structured() {
         let value = ValueBag::from(42u64);
 
         value_bag_sval2::test::assert_tokens(&value, &[value_bag_sval2::test::Token::U64(42)]);
-    }
+    }*/
 
     #[test]
     #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
