Index: value-bag/src/internal/sval/v2.rs
===================================================================
--- value-bag.orig/src/internal/sval/v2.rs
+++ value-bag/src/internal/sval/v2.rs
@@ -786,7 +786,7 @@ mod tests {
 
         use crate::std::vec::Vec;
 
-        #[test]
+        /*#[test]
         #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
         fn sval2_stream_borrowed_str_seq() {
             let value = ValueBag::from_seq_slice(&["a", "b", "c"]);
@@ -814,9 +814,9 @@ mod tests {
                     SeqEnd,
                 ]
             });
-        }
+        }*/
 
-        #[test]
+        /*#[test]
         #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
         fn sval2_stream_str_seq() {
             let value = ValueBag::from_fill(&|slot: Slot| slot.fill_seq_slice(&["a", "b", "c"]));
@@ -844,7 +844,7 @@ mod tests {
                     SeqEnd,
                 ]
             });
-        }
+        }*/
 
         #[test]
         #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
