Index: value-bag/Cargo.toml
===================================================================
--- value-bag.orig/Cargo.toml
+++ value-bag/Cargo.toml
@@ -34,8 +34,6 @@ repository = "https://github.com/sval-rs
 features = [
     "std",
     "error",
-    "sval",
-    "serde",
     "test",
 ]
 
@@ -48,24 +46,12 @@ optional = true
 default-features = false
 package = "erased-serde"
 
-[dependencies.serde1_fmt]
-version = "1"
-optional = true
-package = "serde_fmt"
-
 [dependencies.serde1_lib]
 version = "1"
 optional = true
 default-features = false
 package = "serde"
 
-[dependencies.sval1_lib]
-version = "=1.0.0-alpha.5"
-features = ["fmt"]
-optional = true
-default-features = false
-package = "sval"
-
 [dev-dependencies.serde1_json]
 version = "1"
 package = "serde_json"
@@ -74,16 +60,6 @@ package = "serde_json"
 version = "1"
 package = "serde_test"
 
-[dev-dependencies.sval1_json]
-version = "=1.0.0-alpha.5"
-features = ["std"]
-package = "sval_json"
-
-[dev-dependencies.sval1_lib]
-version = "=1.0.0-alpha.5"
-features = ["test"]
-package = "sval"
-
 [dev-dependencies.wasm-bindgen]
 version = "0.2"
 
@@ -97,17 +73,6 @@ package = "version_check"
 [features]
 error = [
     "std",
-    "sval1_lib/std",
-]
-serde = ["serde1"]
-serde1 = [
-    "serde1_lib",
-    "sval1_lib/serde1",
-    "sval1_lib/alloc",
-    "erased-serde1/alloc",
-    "serde1_fmt",
 ]
 std = []
-sval = ["sval1"]
-sval1 = ["sval1_lib"]
 test = ["std"]
