Index: value-bag/Cargo.toml
===================================================================
--- value-bag.orig/Cargo.toml
+++ value-bag/Cargo.toml
@@ -75,8 +75,8 @@ sval = ["sval2"]
 sval2 = ["value-bag-sval2"]
 test = ["std"]
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen]
-version = "0.2"
+#[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen]
+#version = "0.2"
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+#[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
+#version = "0.3"
