Index: value-bag/Cargo.toml
===================================================================
--- value-bag.orig/Cargo.toml
+++ value-bag/Cargo.toml
@@ -75,11 +75,6 @@ package = "serde_json"
 version = "1"
 package = "serde_test"
 
-[dev-dependencies.sval1_json]
-version = "=1.0.0-alpha.5"
-features = ["std"]
-package = "sval_json"
-
 [dev-dependencies.sval1_lib]
 version = "=1.0.0-alpha.5"
 features = ["test"]
