--- a/Cargo.toml
+++ b/Cargo.toml
@@ -115,3 +115,3 @@
 [dependencies.rand]
-version = "0.9"
+version = "0.8"
 
@@ -131,6 +131,6 @@
 [dependencies.toml]
-version = "0.9"
+version = "0.8"
 
 [dependencies.zstd]
-version = "0.13.3"
+version = "0.13"
 
--- b/src/auth.rs
+++ a/src/auth.rs
@@ -1,7 +1,8 @@
 use crate::api;
 use crate::config::Config;
 use actix_web::HttpRequest;
-use rand::distr::{Alphanumeric, SampleString};
+use rand::distributions::Alphanumeric;
+use rand::prelude::*;
 use rebuilderd_common::api::*;
 use rebuilderd_common::errors::*;
 use std::env;
@@ -68,7 +69,11 @@ pub fn setup_auth_cookie() -> Result<String> {
         cookie
     } else {
         debug!("Generating random cookie");
-        Alphanumeric.sample_string(&mut rand::rng(), 32)
+        thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(32)
+            .map(char::from)
+            .collect()
     };
 
     let cookie_path = if let Ok(cookie_path) = env::var("REBUILDERD_COOKIE_PATH") {
