--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,7 +33,7 @@
 version = "0.2"
 
 [build-dependencies.bindgen]
-version = "0.57"
+version = ">=0.57"
 optional = true
 
 [features]
--- a/build.rs
+++ b/build.rs
@@ -81,8 +81,8 @@
             })
             .header(header.display().to_string())
             .ctypes_prefix("::libc")
-            .whitelist_type(r"^libinput_.*$")
-            .whitelist_function(r"^libinput_.*$")
+            .allowlist_type(r"^libinput_.*$")
+            .allowlist_function(r"^libinput_.*$")
             .generate()
             .unwrap();
 
