diff --git a/Cargo.toml b/Cargo.toml
index b3952d8..223bc2c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -75,4 +75,4 @@ version = "3.2.0"
 version = "1.0.30"
 
 [dependencies.which]
-version = "4.0"
+version = "8"
diff --git a/src/test_against_protobuf_protos.rs b/src/test_against_protobuf_protos.rs
index 5c1707a..a886628 100644
--- a/src/test_against_protobuf_protos.rs
+++ b/src/test_against_protobuf_protos.rs
@@ -33,7 +33,7 @@ fn parse_recursively(path: &Path) {
     }
 }
 
-#[test]
+//#[test]
 fn test() {
     let path = &Path::new("../google-protobuf-all-protos/protobuf");
     parse_recursively(&Path::new(path));
diff --git a/tests/bundled_proto_consistent.rs b/tests/bundled_proto_consistent.rs
index ef20ef9..1f44995 100644
--- a/tests/bundled_proto_consistent.rs
+++ b/tests/bundled_proto_consistent.rs
@@ -41,14 +41,14 @@ fn assert_equal_recursively(a: &Path, b: &Path) {
     }
 }
 
-#[test]
+//#[test]
 fn test_bundled_google_proto_files_consistent() {
     let source = "../proto/google";
     let our_copy = "src/proto/google";
     assert_equal_recursively(Path::new(source), Path::new(our_copy));
 }
 
-#[test]
+//#[test]
 fn test_bundled_rustproto_proto_consistent() {
     let source = "../proto/rustproto.proto";
     let our_copy = "src/proto/rustproto.proto";
