Index: num-cpus/Cargo.toml
===================================================================
--- num-cpus.orig/Cargo.toml
+++ num-cpus/Cargo.toml
@@ -25,8 +25,8 @@ categories = ["hardware-support"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/seanmonstar/num_cpus"
 
-[target."cfg(all(any(target_arch = \"x86_64\", target_arch = \"aarch64\"), target_os = \"hermit\"))".dependencies.hermit-abi]
-version = "0.2.6"
+#[target."cfg(all(any(target_arch = \"x86_64\", target_arch = \"aarch64\"), target_os = \"hermit\"))".dependencies.hermit-abi]
+#version = "0.2.6"
 
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2.26"
