commit 563350c621ade9a3d1396a59c7d7902164037745
Author: 82marbag <69267416+82marbag@users.noreply.github.com>
Date:   Tue Mar 21 17:55:22 2023 +0000

    Remove unwanted test (#147)

diff --git a/src/parse.rs b/src/parse.rs
index 20022b775b..e8264032a3 100644
--- a/src/parse.rs
+++ b/src/parse.rs
@@ -445,10 +445,3 @@ fn test_parse_iterator_invalid() {
     assert_eq!(iter.next().unwrap().unwrap(), parse("application/json").unwrap());
     assert_eq!(iter.next(), None);
 }
-
-#[test]
-fn test_parse_iterator_all_invalid() {
-    let mut iter = MimeIter::new("application/json, text/html");
-    assert_eq!(iter.next().unwrap().unwrap_err(), "application/json");
-    assert_eq!(iter.next(), None);
-}
