Index: rust-encoding-rs-0.8.15/Cargo.toml
===================================================================
--- rust-encoding-rs-0.8.15.orig/Cargo.toml
+++ rust-encoding-rs-0.8.15/Cargo.toml
@@ -31,9 +31,6 @@ version = "0.1.0"
 version = "1.0"
 optional = true
 
-[dependencies.simd]
-version = "0.2.3"
-optional = true
 [dev-dependencies.bincode]
 version = "0.8"
 
@@ -53,6 +50,5 @@ fast-legacy-encode = ["fast-hangul-encod
 less-slow-big5-hanzi-encode = []
 less-slow-gb-hanzi-encode = []
 less-slow-kanji-encode = []
-simd-accel = ["simd"]
 [badges.travis-ci]
 repository = "hsivonen/encoding_rs"
