--- a/Cargo.toml
+++ b/Cargo.toml
@@ -74,7 +74,7 @@ version = "1.5.2"
 optional = true
 
 [dependencies.parking_lot]
-version = "0.12.0"
+version = "0.11"
 optional = true
 
 [dependencies.pin-project-lite]
@@ -195,10 +195,6 @@ version = "0.3.0"
 version = "0.6.0"
 features = ["tokio"]
 
-[target."cfg(tokio_unstable)".dependencies.tracing]
-version = "0.1.25"
-features = ["std"]
-optional = true
 default-features = false
 
 [target."cfg(unix)".dependencies.libc]
@@ -213,12 +209,7 @@ optional = true
 version = "0.2.42"
 
 [target."cfg(unix)".dev-dependencies.nix]
-version = "0.24"
-features = [
-    "fs",
-    "socket",
-]
-default-features = false
+version = "0.23"
 
 [target."cfg(windows)".dependencies.winapi]
 version = "0.3.8"
