Index: tokio/Cargo.toml
===================================================================
--- tokio.orig/Cargo.toml
+++ tokio/Cargo.toml
@@ -719,18 +719,6 @@ name = "time_wasm"
 path = "tests/time_wasm.rs"
 
 [[test]]
-name = "tracing_sync"
-path = "tests/tracing_sync.rs"
-
-[[test]]
-name = "tracing_task"
-path = "tests/tracing_task.rs"
-
-[[test]]
-name = "tracing_time"
-path = "tests/tracing_time.rs"
-
-[[test]]
 name = "udp"
 path = "tests/udp.rs"
 
@@ -798,9 +786,6 @@ version = "0.4.0"
 version = "0.7"
 features = ["rt"]
 
-[target.'cfg(all(tokio_unstable, target_has_atomic = "64"))'.dev-dependencies.tracing-mock]
-version = "= 0.1.0-beta.1"
-
 [target.'cfg(all(tokio_unstable, target_os = "linux"))'.dependencies.backtrace]
 version = "0.3.58"
 optional = true
