Index: tokio/Cargo.toml
===================================================================
--- tokio.orig/Cargo.toml
+++ tokio/Cargo.toml
@@ -176,10 +176,6 @@ version = "0.4.9"
 [target."cfg(not(any(target_arch = \"wasm32\", target_arch = \"wasm64\")))".dev-dependencies.tempfile]
 version = "3.1.0"
 
-[target."cfg(target_os = \"freebsd\")".dev-dependencies.mio-aio]
-version = "0.7.0"
-features = ["tokio"]
-
 [target."cfg(tokio_taskdump)".dependencies.backtrace]
 version = "0.3.58"
 
