--- a/Cargo.toml
+++ b/Cargo.toml
@@ -96,11 +96,6 @@
     "mio/os-ext",
     "mio/net",
     "socket2",
-    "windows-sys/Win32_Foundation",
-    "windows-sys/Win32_Security",
-    "windows-sys/Win32_Storage_FileSystem",
-    "windows-sys/Win32_System_Pipes",
-    "windows-sys/Win32_System_SystemServices",
 ]
 process = [
     "bytes",
@@ -109,9 +104,6 @@
     "mio/os-ext",
     "mio/net",
     "signal-hook-registry",
-    "windows-sys/Win32_Foundation",
-    "windows-sys/Win32_System_Threading",
-    "windows-sys/Win32_System_WindowsProgramming",
 ]
 rt = []
 rt-multi-thread = ["rt"]
@@ -121,8 +113,6 @@
     "mio/net",
     "mio/os-ext",
     "signal-hook-registry",
-    "windows-sys/Win32_Foundation",
-    "windows-sys/Win32_System_Console",
 ]
 sync = []
 test-util = [
@@ -823,17 +813,6 @@
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59"
-optional = true
-
-[target."cfg(windows)".dev-dependencies.windows-sys]
-version = "0.59"
-features = [
-    "Win32_Foundation",
-    "Win32_Security_Authorization",
-]
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
