--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@
 path = "src/lib.rs"
 
 [dependencies.async-channel]
-version = "2.5"
+version = ">= 2.3"

 [dependencies.async-compat]
 version = "0.2"
@@ -44,7 +44,7 @@
 version = "0.21"
 
 [dependencies.indexmap]
-version = "2.10"
+version = ">= 2.7"
 
 [dependencies.log]
 version = "0.4"
@@ -50,10 +50,10 @@
 version = "0.4"
 
 [dependencies.num_enum]
-version = "0.7"
+version = "0.5"
 
 [dependencies.once_cell]
-version = "1.21"
+version = ">=1.20"
 
 [dependencies.transmission-client]
 version = "0.1.1"
 
