--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,13 +24,13 @@
 version = "1.6"
 
 [dependencies.async-std]
-version = "1.10"
+version = "1.12"
 
 [dependencies.gio]
-version = "0.15"
+version = "0.16"
 
 [dependencies.glib]
-version = "0.15"
+version = "0.16"
 
 [dependencies.gtk-macros]
 version = "0.3"
@@ -39,7 +39,7 @@
 version = "0.4"
 
 [dependencies.num_enum]
-version = "0.5"
+version = "0.5.7"
 
 [dependencies.once_cell]
 version = "1.7"
