Index: sysexits/src/exit_code/convert.rs
===================================================================
--- sysexits.orig/src/exit_code/convert.rs
+++ sysexits/src/exit_code/convert.rs
@@ -754,7 +754,7 @@ mod tests {
         );
         #[cfg(feature = "extended_io_error")]
         assert_eq!(
-            ExitCode::from(Error::from(ErrorKind::FilesystemQuotaExceeded)),
+            ExitCode::from(Error::from(ErrorKind::QuotaExceeded)),
             ExitCode::IoErr
         );
         #[cfg(feature = "extended_io_error")]
@@ -926,7 +926,7 @@ mod tests {
         assert_eq!(ExitCode::from(io::ErrorKind::NotSeekable), ExitCode::IoErr);
         #[cfg(feature = "extended_io_error")]
         assert_eq!(
-            ExitCode::from(io::ErrorKind::FilesystemQuotaExceeded),
+            ExitCode::from(io::ErrorKind::QuotaExceeded),
             ExitCode::IoErr
         );
         #[cfg(feature = "extended_io_error")]
