Index: git2/Cargo.toml
===================================================================
--- git2.orig/Cargo.toml
+++ git2/Cargo.toml
@@ -149,11 +149,6 @@ https = [
 ]
 ssh = ["libgit2-sys/ssh"]
 unstable = []
-vendored-libgit2 = ["libgit2-sys/vendored"]
-vendored-openssl = [
-    "openssl-sys/vendored",
-    "libgit2-sys/vendored-openssl",
-]
 zlib-ng-compat = ["libgit2-sys/zlib-ng-compat"]
 
 [target.'cfg(all(unix, not(target_os = "macos")))'.dependencies.openssl-probe]
