--- rust-dockworker-0.7.0.orig/Cargo.toml
+++ rust-dockworker-0.7.0/Cargo.toml
@@ -188,7 +188,8 @@ version = "0.1"
 
 [dependencies.hyper-rustls]
 version = "0.27"
-features = ["http2"]
+features = ["http2", "native-tokio", "http1", "tls12", "logging", "ring"]
+default-features = false
 optional = true
 
 [dependencies.hyper-tls]
@@ -212,6 +213,8 @@ optional = true
 
 [dependencies.rustls]
 version = "0.23"
+features = ["ring", "logging", "std", "tls12"]
+default-features = false
 optional = true
 
 [dependencies.rustls-pemfile]
