--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,26 +40,6 @@
     "win_sendinput_send_scancodes",
     "zippychord",
 ]
-gui = [
-    "win_manifest",
-    "kanata-parser/gui",
-    "win_sendinput_send_scancodes",
-    "win_llhook_read_scancodes",
-    "muldiv",
-    "strip-ansi-escapes",
-    "open",
-    "dep:windows-sys",
-    "winapi/processthreadsapi",
-    "native-windows-gui/tray-notification",
-    "native-windows-gui/message-window",
-    "native-windows-gui/menu",
-    "native-windows-gui/cursor",
-    "native-windows-gui/high-dpi",
-    "native-windows-gui/embed-resource",
-    "native-windows-gui/image-decoder",
-    "native-windows-gui/notice",
-    "native-windows-gui/animation-timer",
-]
 interception_driver = [
     "kanata-interception",
     "kanata-parser/interception_driver",
@@ -184,9 +164,6 @@
 [target.'cfg(target_os = "linux")'.dependencies.signal-hook]
 version = "0.3.14"
 
-[target.'cfg(target_os = "macos")'.dependencies.core-graphics]
-version = "0.24.0"
-
 [target.'cfg(target_os = "macos")'.dependencies.karabiner-driverkit]
 version = "0.1.5"
 
