From: Agathe Porte <gagath@debian.org>
Subject: [PATCH] relax dependencies

Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -102,7 +102,7 @@
 default-features = false
 
 [dependencies.miette]
-version = "5.7.0"
+version = "7.4.0"
 features = ["fancy"]
 
 [dependencies.once_cell]
@@ -139,18 +139,18 @@
 version = "0.13.0"
 
 [target.'cfg(target_os = "linux")'.dependencies.inotify]
-version = "0.10.0"
+version = "0.11.0"
 default-features = false
 
 [target.'cfg(target_os = "linux")'.dependencies.mio]
-version = "0.8.11"
+version = "1.0.2"
 features = [
     "os-poll",
     "os-ext",
 ]
 
 [target.'cfg(target_os = "linux")'.dependencies.nix]
-version = "0.26.1"
+version = "0.29.0"
 features = ["ioctl"]
 
 [target.'cfg(target_os = "linux")'.dependencies.open]
