--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,14 +30,11 @@
 version = ">= 0.53.1, < 0.56"
 default-features = false
 
-[build-dependencies.nettle-src]
-version = "3.5.1-0"
-optional = true
-
 [build-dependencies.pkg-config]
 version = "0.3"
 
 [features]
 vendored = ["nettle-src"]
+nettle-src = []
 [target."cfg(target_env = \"msvc\")".build-dependencies.vcpkg]
 version = "0.2.9"
--- a/build.rs
+++ b/build.rs
@@ -62,13 +62,6 @@
     println!("cargo:rerun-if-env-changed=NETTLE_STATIC");
     println!("cargo:rerun-if-env-changed={}", NETTLE_PREGENERATED_BINDINGS);
 
-    #[cfg(feature = "vendored")]
-    {
-        let artifacts = nettle_src::Build::new().build();
-        println!("cargo:vendored=1");
-        env::set_var("PKG_CONFIG_PATH", artifacts.lib_dir().join("pkgconfig"));
-    }
-
     let include_paths = try_vcpkg().or_else(try_pkg_config).unwrap();
 
     let out_path = Path::new(&env::var("OUT_DIR").unwrap()).join("bindings.rs");
