Description: Use libc constants for flags where available
Forwarded: https://github.com/main--/rust-timerfd/pull/3
Author: Wolfgang Silbermayr <wolfgang@silbermayr.at>
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -131,9 +131,9 @@
     TimerCancelOnSet,
 }
 
-static TFD_CLOEXEC: libc::c_int = 0o2000000;
-static TFD_NONBLOCK: libc::c_int = 0o0004000;
-static TFD_TIMER_ABSTIME: libc::c_int = 0o0000001;
+static TFD_CLOEXEC: libc::c_int = libc::O_CLOEXEC;
+static TFD_NONBLOCK: libc::c_int = libc::O_NONBLOCK;
+static TFD_TIMER_ABSTIME: libc::c_int = libc::TIMER_ABSTIME;
 static TFD_TIMER_CANCEL_ON_SET: libc::c_int = 0o0000002;
 
 mod structs;
