--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,22 +32,12 @@
 version = "1.0"
 default-features = false
 
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
 [dependencies.compiler_builtins]
 version = "0.1.2"
 optional = true
 
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
 [build-dependencies.autocfg]
 version = "1.0"
 
 [features]
 no_extern_crate_alloc = []
-rustc-dep-of-std = ["core", "alloc", "compiler_builtins", "adler/rustc-dep-of-std"]
