Index: miniz-oxide/Cargo.toml
===================================================================
--- miniz-oxide.orig/Cargo.toml
+++ miniz-oxide/Cargo.toml
@@ -42,20 +42,10 @@ name = "miniz_oxide"
 version = "1.0"
 default-features = false
 
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
 [dependencies.compiler_builtins]
 version = "0.1.2"
 optional = true
 
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [dependencies.simd-adler32]
 version = "0.3"
 optional = true
@@ -63,10 +53,4 @@ default-features = false
 
 [features]
 default = []
-rustc-dep-of-std = [
-    "core",
-    "alloc",
-    "compiler_builtins",
-    "adler/rustc-dep-of-std",
-]
 simd = ["simd-adler32"]
