Index: pathrs/Cargo.toml
===================================================================
--- pathrs.orig/Cargo.toml
+++ pathrs/Cargo.toml
@@ -42,7 +42,6 @@ capi = [
     "dep:bytemuck",
     "bitflags/bytemuck",
     "dep:rand",
-    "dep:open-enum",
 ]
 default = []
 
@@ -78,10 +77,6 @@ version = "2"
 [dependencies.once_cell]
 version = "1"
 
-[dependencies.open-enum]
-version = "0.3"
-optional = true
-
 [dependencies.rand]
 version = "0.9"
 optional = true
