Origin: https://github.com/BurntSushi/fst/commit/b8890233
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,4 +40,4 @@
 
-[dev-dependencies.memmap]
-version = "0.7"
+[dev-dependencies.memmap2]
+version = ">=0.7, <1"
 
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -92,7 +92,7 @@
 Instead, only constant memory is required during construction.
 
 For the search phase, we use the
-[`memmap`](https://crates.io/memmap)
+[`memmap2`](https://crates.io/memmap2)
 crate to make the file available as a `&[u8]` without necessarily reading it
 all into memory (the operating system will automatically handle that for you).
 
@@ -102,7 +102,7 @@
 use std::io;
 
 use fst::{IntoStreamer, Streamer, Map, MapBuilder};
-use memmap::Mmap;
+use memmap2::Mmap;
 
 // This is where we'll write our map to.
 let mut wtr = io::BufWriter::new(File::create("map.fst")?);
