--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,23 +44,23 @@
 version = "0.11.1"
 
 [dependencies.freetype-rs]
-version = "0.37"
+version = "0.26"
 
 [dependencies.input]
-version = "0.8"
+version = "0.9"
 
 [dependencies.input-linux]
-version = "0.7"
+version = "0.6"
 features = ["serde"]
 
 [dependencies.input-linux-sys]
-version = "0.9"
+version = "0.8"
 
 [dependencies.libc]
 version = "0.2"
 
 [dependencies.librsvg]
-version = "~2.59.1"
+version = "2.59"
 
 [dependencies.nix]
 version = "0.29"
