--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,10 +20,10 @@
 autotests = false
 autobenches = false
 description = "The most basic dynamic function row daemon possible"
-homepage = "https://github.com/WhatAmISupposedToPutHere/tiny-dfr"
+homepage = "https://github.com/AsahiLinux/tiny-dfr"
 readme = "README.md"
 license = "MIT AND Apache-2.0"
-repository = "https://github.com/WhatAmISupposedToPutHere/tiny-dfr"
+repository = "https://github.com/AsahiLinux/tiny-dfr"
 
 [[bin]]
 name = "tiny-dfr"
