Index: num-bigint/Cargo.toml
===================================================================
--- num-bigint.orig/Cargo.toml
+++ num-bigint/Cargo.toml
@@ -58,12 +58,12 @@ features = ["i128"]
 default-features = false
 
 [dependencies.quickcheck]
-version = "1"
+version = "0.9"
 optional = true
 default-features = false
 
 [dependencies.rand]
-version = "0.8"
+version = "0.7"
 optional = true
 default-features = false
 
