From: Maytham Alsudany <maytha8thedev@gmail.com>
Forwarded: not-needed
Description: Use rcgen 0.11

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -88,7 +88,7 @@
 default-features = false
 
 [dev-dependencies.rcgen]
-version = "0.13"
+version = "0.11"
 
 [dev-dependencies.rustls]
 version = "0.23"
--- a/src/tests/mod.rs
+++ b/src/tests/mod.rs
@@ -142,7 +142,7 @@
 pub fn build_certs() -> (CertificateDer<'static>, PrivateKeyDer<'static>) {
     let cert = rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
     (
-        cert.cert.into(),
-        PrivateKeyDer::Pkcs8(cert.key_pair.serialize_der().into()),
+        CertificateDer::from(cert.serialize_der().unwrap()),
+        PrivateKeyDer::Pkcs8(cert.get_key_pair().serialize_der().into()),
     )
 }
