--- a/src/system/audit.rs
+++ b/src/system/audit.rs
@@ -355,7 +355,8 @@
         assert!(traversed_secure_open("/bin/hello.txt", &user).is_err());
     }
 
-    #[test]
+    // Debian: this makes assumptions that are note valid in autopkgtest context
+    #[allow(dead_code)]
     fn test_traverse_secure_open_positive() {
         use crate::common::resolve::CurrentUser;
         use crate::system::{GroupId, UserId};
