Index: rust-code-analysis/build.rs
===================================================================
--- rust-code-analysis.orig/build.rs
+++ rust-code-analysis/build.rs
@@ -151,6 +151,22 @@ fn build_dir(dir: &str, language: &str)
 }
 
 fn main() {
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-c=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-ccomments=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-cpp=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-c-sharp=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-css=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-go=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-html=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-java=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-javascript=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-mozcpp=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozjs=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-preproc=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-python=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-rust=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-typescript=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+
     if !is_enums() {
         mk_predef("c_macros", "PREDEFINED_MACROS");
         mk_predef("c_specials", "SPECIALS");
