Index: rust-code-analysis/build.rs
===================================================================
--- rust-code-analysis.orig/build.rs
+++ rust-code-analysis/build.rs
@@ -151,6 +151,22 @@ fn build_dir(dir: &str, language: &str)
 }
 
 fn main() {
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-c=$s={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-ccomments=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-cpp=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-c-sharp=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-css=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-go=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-html=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-java=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-javascript=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-mozcpp=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-mozjs=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-preproc=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-python=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-rust=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+    println!("dh-cargo:deb-built-using=tree-sitter-mozcpp-typescript=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+
     if !is_enums() {
         mk_predef("c_macros", "PREDEFINED_MACROS");
         mk_predef("c_specials", "SPECIALS");
