--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -324,3 +324,3 @@
             let handle = actix_rt::spawn(async move {
-                let f = tokio_uring::fs::File::create("test.txt").await.unwrap();
+                let f = tokio_uring::fs::File::create("/tmp/test.txt").await.unwrap();
                 let buf = b"Hello World!";
@@ -333,3 +333,3 @@
 
-                std::fs::remove_file("test.txt").unwrap();
+                std::fs::remove_file("/tmp/test.txt").unwrap();
             });
