Index: netlink-packet-route/Cargo.toml
===================================================================
--- netlink-packet-route.orig/Cargo.toml
+++ netlink-packet-route/Cargo.toml
@@ -37,30 +37,6 @@ rich_nlas = []
 name = "netlink_packet_route"
 path = "src/lib.rs"
 
-[[example]]
-name = "dump_neighbour_tables"
-path = "examples/dump_neighbour_tables.rs"
-
-[[example]]
-name = "dump_neighbours"
-path = "examples/dump_neighbours.rs"
-
-[[example]]
-name = "dump_packet_link_bridge_vlan"
-path = "examples/dump_packet_link_bridge_vlan.rs"
-
-[[example]]
-name = "dump_packet_links"
-path = "examples/dump_packet_links.rs"
-
-[[example]]
-name = "dump_rules"
-path = "examples/dump_rules.rs"
-
-[[example]]
-name = "new_rule"
-path = "examples/new_rule.rs"
-
 [dependencies.anyhow]
 version = "1.0.31"
 
