Index: netlink-packet-route/Cargo.toml
===================================================================
--- netlink-packet-route.orig/Cargo.toml
+++ netlink-packet-route/Cargo.toml
@@ -82,8 +82,5 @@ version = "0.5.2"
 [dev-dependencies.netlink-sys]
 version = "0.8.5"
 
-[dev-dependencies.pretty_assertions]
-version = "0.7.2"
-
 [features]
 rich_nlas = []
Index: netlink-packet-route/src/lib.rs
===================================================================
--- netlink-packet-route.orig/src/lib.rs
+++ netlink-packet-route/src/lib.rs
@@ -67,9 +67,5 @@ pub use self::message::{RouteNetlinkMess
 #[macro_use]
 extern crate netlink_packet_utils;
 
-#[cfg(test)]
-#[macro_use]
-extern crate pretty_assertions;
-
 #[macro_use]
 extern crate bitflags;
