Index: netlink-packet-route/Cargo.toml
===================================================================
--- netlink-packet-route.orig/Cargo.toml
+++ netlink-packet-route/Cargo.toml
@@ -62,8 +62,5 @@ version = "1.4.0"
 [dev-dependencies.netlink-sys]
 version = "0.8.4"
 
-[dev-dependencies.pretty_assertions]
-version = "0.7.2"
-
 [features]
 rich_nlas = []
Index: netlink-packet-route/src/lib.rs
===================================================================
--- netlink-packet-route.orig/src/lib.rs
+++ netlink-packet-route/src/lib.rs
@@ -20,7 +20,3 @@ pub use self::rtnl::*;
 #[cfg(test)]
 #[macro_use]
 extern crate lazy_static;
-
-#[cfg(test)]
-#[macro_use]
-extern crate pretty_assertions;
