Index: netlink-packet-route/src/address/tests/ipv6.rs
===================================================================
--- netlink-packet-route.orig/src/address/tests/ipv6.rs
+++ netlink-packet-route/src/address/tests/ipv6.rs
@@ -13,6 +13,7 @@ use crate::AddressFamily;
 // TODO(Gris Ge): Need test for `AddressAttribute::Anycast` and `Multicast`.
 
 #[test]
+#[cfg_attr(target_arch = "s390x", ignore = "debci runner does not have ipv6")]
 fn test_addr_flag_stable_privacy() {
     let nla = AddressAttribute::Flags(
         AddressFlags::Permanent | AddressFlags::StablePrivacy,
@@ -36,6 +37,7 @@ fn test_addr_flag_stable_privacy() {
 }
 
 #[test]
+#[cfg_attr(target_arch = "s390x", ignore = "debci runner does not have ipv6")]
 fn test_get_loopback_ipv6_addr() {
     let raw = vec![
         0x0a, 0x80, 0x80, 0xfe, 0x01, 0x00, 0x00, 0x00, 0x14, 0x00, 0x01, 0x00,
