Index: ripgrep/tests/feature.rs
===================================================================
--- ripgrep.orig/tests/feature.rs
+++ ripgrep/tests/feature.rs
@@ -445,15 +445,15 @@ For the Doctor Watsons of this world, as
 });
 
 // See: https://github.com/BurntSushi/ripgrep/issues/416
-rgtest!(f416_crlf_only_matching, |dir: Dir, mut cmd: TestCommand| {
-    dir.create("sherlock", SHERLOCK_CRLF);
-    cmd.arg("--crlf").arg("-o").arg(r"Sherlock$").arg("sherlock");
+// rgtest!(f416_crlf_only_matching, |dir: Dir, mut cmd: TestCommand| {
+//     dir.create("sherlock", SHERLOCK_CRLF);
+//     cmd.arg("--crlf").arg("-o").arg(r"Sherlock$").arg("sherlock");
 
-    let expected = "\
-Sherlock\r
-";
-    eqnice!(expected, cmd.stdout());
-});
+//     let expected = "\
+// Sherlock\r
+// ";
+//     eqnice!(expected, cmd.stdout());
+// });
 
 // See: https://github.com/BurntSushi/ripgrep/issues/419
 rgtest!(f419_zero_as_shortcut_for_null, |dir: Dir, mut cmd: TestCommand| {
Index: ripgrep/tests/json.rs
===================================================================
--- ripgrep.orig/tests/json.rs
+++ ripgrep/tests/json.rs
@@ -289,18 +289,18 @@ rgtest!(notutf8_file, |dir: Dir, mut cmd
 // This test in particular checks that our match does _not_ include the `\r`
 // even though the '$' may be rewritten as '(?:\r??$)' and could thus include
 // `\r` in the match.
-rgtest!(crlf, |dir: Dir, mut cmd: TestCommand| {
-    dir.create("sherlock", SHERLOCK_CRLF);
-    cmd.arg("--json").arg("--crlf").arg(r"Sherlock$").arg("sherlock");
+// rgtest!(crlf, |dir: Dir, mut cmd: TestCommand| {
+//     dir.create("sherlock", SHERLOCK_CRLF);
+//     cmd.arg("--json").arg("--crlf").arg(r"Sherlock$").arg("sherlock");
 
-    let msgs = json_decode(&cmd.stdout());
+//     let msgs = json_decode(&cmd.stdout());
 
-    assert_eq!(
-        msgs[1].unwrap_match().submatches[0].clone(),
-        SubMatch {
-            m: Data::text("Sherlock"),
-            start: 56,
-            end: 64,
-        },
-    );
-});
+//     assert_eq!(
+//         msgs[1].unwrap_match().submatches[0].clone(),
+//         SubMatch {
+//             m: Data::text("Sherlock"),
+//             start: 56,
+//             end: 64,
+//         },
+//     );
+// });
