Index: ripgrep/Cargo.toml
===================================================================
--- ripgrep.orig/Cargo.toml
+++ ripgrep/Cargo.toml
@@ -180,7 +180,7 @@ version = "2"
 
 [features]
 pcre2 = ["grep/pcre2"]
-simd-accel = ["grep/simd-accel"]
+#simd-accel = ["grep/simd-accel"]
 
 [target."cfg(all(target_env = \"musl\", target_pointer_width = \"64\"))".dependencies.jemallocator]
 version = "0.5.0"
