Index: rust-gio-sys-0.14.0/Cargo.toml
===================================================================
--- rust-gio-sys-0.14.0.orig/Cargo.toml
+++ rust-gio-sys-0.14.0/Cargo.toml
@@ -71,7 +71,7 @@ version = "1.0.0"
 [dev-dependencies.tempfile]
 version = "3"
 [build-dependencies.system-deps]
-version = "3"
+version = "6"
 
 [features]
 dox = []
@@ -85,5 +85,5 @@ v2_62 = ["v2_60"]
 v2_64 = ["v2_62"]
 v2_66 = ["v2_64"]
 [target."cfg(windows)".dependencies.winapi]
-version = "0.3.9"
+version = "0.3.8"
 features = ["ws2def", "winsock2"]
