Description: relax dependency versions
Author: Ananthu C V <weepingclown@debian.org>
Forwarded: https://github.com/jondot/merge-struct/pull/1
Last-Update: 2025-08-16
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,6 +34,6 @@
 [dev-dependencies.insta]
-version = "1.17.1"
+version = "1.39"
 features = [
-    "backtrace",
     "redactions",
+    "yaml"
 ]
@@ -42 +42,5 @@
 version = "1"
+
+[dev-dependencies.serde]
+version = "1"
+features = ["derive"]
\ No newline at end of file
