--- a/Cargo.toml
+++ b/Cargo.toml
@@ -104,10 +104,6 @@
 [dependencies.trust-dns-proto]
 version = "0.21.1"
 
-[dependencies.trust-dns-resolver]
-version = "0.21.1"
-features = ["serde-config"]
-optional = true
 [dev-dependencies.tokio]
 version = "1.0"
 features = ["macros", "rt"]
@@ -115,13 +111,8 @@
 [features]
 backtrace = ["trust-dns-proto/backtrace"]
 dns-over-https = ["h2", "http", "trust-dns-proto/dns-over-https"]
-dns-over-https-rustls = ["dns-over-https", "trust-dns-proto/dns-over-https-rustls", "trust-dns-client/dns-over-https-rustls", "trust-dns-resolver/dns-over-https-rustls", "dns-over-rustls", "tokio-rustls"]
-dns-over-rustls = ["dns-over-tls", "dnssec-ring", "rustls", "trust-dns-proto/dns-over-rustls", "trust-dns-client/dns-over-rustls", "trust-dns-resolver/dns-over-rustls", "tokio-rustls"]
 dns-over-tls = []
 dnssec = []
-dnssec-openssl = ["dnssec", "openssl", "trust-dns-client/dnssec-openssl", "trust-dns-proto/dnssec-openssl", "trust-dns-resolver/dnssec-openssl"]
-dnssec-ring = ["dnssec", "trust-dns-client/dnssec-ring", "trust-dns-proto/dnssec-ring", "trust-dns-resolver/dnssec-ring"]
-resolver = ["trust-dns-resolver"]
 testing = []
 [badges.codecov]
 branch = "main"
