Index: cargo-lichking/Cargo.toml
===================================================================
--- cargo-lichking.orig/Cargo.toml
+++ cargo-lichking/Cargo.toml
@@ -26,7 +26,7 @@ features = ["std"]
 default-features = false
 
 [dependencies.cargo_metadata]
-version = "0.9.1"
+version = "0.14"
 default-features = false
 
 [dependencies.clap]
Index: cargo-lichking/src/bundle.rs
===================================================================
--- cargo-lichking.orig/src/bundle.rs
+++ cargo-lichking/src/bundle.rs
@@ -534,7 +534,7 @@ fn choose(
                 "{} has no candidate texts for license {} in {}",
                 package.name,
                 license,
-                package.manifest_path.parent().unwrap().display()
+                package.manifest_path.parent().unwrap().as_std_path().display()
             );
             context.missing_license = true;
             return Ok(None);
Index: cargo-lichking/src/licensed.rs
===================================================================
--- cargo-lichking.orig/src/licensed.rs
+++ cargo-lichking/src/licensed.rs
@@ -11,7 +11,7 @@ impl Licensed for Package {
         self.license
             .as_ref()
             .and_then(|license| license.parse::<License>().ok())
-            .or_else(|| self.license_file().map(License::File))
+            .or_else(|| self.license_file().map(|x| License::File(x.into_std_path_buf())))
             .unwrap_or_default()
     }
 }
