Relax deps to what is in Debian

diff --git a/Cargo.toml b/Cargo.toml
index ffe64de..6557dfc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,10 +29,10 @@ repository = "https://github.com/Byron/gitoxide"
 doctest = false
 
 [dependencies.gix-command]
-version = "^0.3.0"
+version = "0.2"
 
 [dependencies.gix-config-value]
-version = "^0.14.1"
+version = "0.14"
 
 [dependencies.parking_lot]
 version = "0.12.1"
