diff --git a/Cargo.toml b/Cargo.toml
index b6eaba4..2647356 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,6 +47,7 @@ features = ["example_generated"]
 [features]
 example_generated = []
 std = []
+rustc-dep-of-std = ["core", "compiler_builtins"]
 
 [lib]
 name = "bitflags"
@@ -83,6 +84,15 @@ version = "1.0.103"
 optional = true
 default-features = false
 
+[dependencies.core]
+version = "1.0.0"
+optional = true
+package = "rustc-std-workspace-core"
+
+[dependencies.compiler_builtins]
+version = "0.1.2"
+optional = true
+
 [dev-dependencies.arbitrary]
 version = "1.0"
 features = ["derive"]
