--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,9 +60,6 @@
 name = "custom_bits_type"
 path = "examples/custom_bits_type.rs"
 
-[[example]]
-name = "custom_derive"
-path = "examples/custom_derive.rs"
 
 [[example]]
 name = "fmt"
@@ -76,9 +73,6 @@
 name = "serde"
 path = "examples/serde.rs"
 
-[[bench]]
-name = "parse"
-path = "benches/parse.rs"
 
 [dependencies.arbitrary]
 version = "1.0"
@@ -125,6 +119,4 @@
 [dev-dependencies.trybuild]
 version = "1.0.18"
 
-[dev-dependencies.zerocopy]
-version = "0.8"
-features = ["derive"]
+
