Index: prometheus-client-derive-encode/tests/lib.rs
===================================================================
--- prometheus-client-derive-encode.orig/tests/lib.rs
+++ prometheus-client-derive-encode/tests/lib.rs
@@ -1,3 +1,4 @@
+/*
 use prometheus_client::encoding::text::encode;
 use prometheus_client::encoding::{EncodeLabelSet, EncodeLabelValue};
 use prometheus_client::metrics::counter::Counter;
@@ -173,3 +174,5 @@ fn flatten() {
         + "# EOF\n";
     assert_eq!(expected, buffer);
 }
+*/
+
