Description: dropping dependencies needed for only windows  systems
Author: Ananthu C V <weepingclown@disroot.org>
Forwarded: not-needed
Last-Update: 2024-10-13
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -170 +170 @@
-    "windows-sys/Win32_System_Threading",
+#    "windows-sys/Win32_System_Threading",
@@ -177,6 +177,6 @@
-[target.'cfg(target_os = "windows")'.dependencies.windows-sys]
-version = "0.36.1"
-features = [
-    "Win32_Foundation",
-    "Win32_Networking_WinSock",
-]
+#[target.'cfg(target_os = "windows")'.dependencies.windows-sys]
+#version = "0.36.1"
+#features = [
+#    "Win32_Foundation",
+#    "Win32_Networking_WinSock",
+#]
@@ -184,2 +184,2 @@
-[target.'cfg(target_os = "windows")'.dev-dependencies.eui48]
-version = "1.1"
+#[target.'cfg(target_os = "windows")'.dev-dependencies.eui48]
+#version = "1.1"
