Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -632,12 +632,0 @@ tests = [
-[target.'cfg(target_arch = "wasm32")'.dependencies.backon]
-version = "1.2"
-features = ["gloo-timers-sleep"]
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.tokio]
-version = "1.27"
-features = ["time"]
-
