Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -435,25 +435,5 @@ features = [
     "tracing-log",
 ]
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.backon]
-version = "1.2"
-features = ["gloo-timers-sleep"]
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.tokio]
-version = "1.46"
-features = ["time"]
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.uuid]
-version = "1.14"
-features = [
-    "serde",
-    "v4",
-    "js",
-]
-
 [lints.clippy]
 unused_async = "warn"
