Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -70,7 +70,6 @@ layers-throttle = ["dep:governor"]
 layers-tracing = ["dep:tracing"]
 services-aliyun-drive = []
 services-alluxio = []
-services-atomicserver = ["dep:atomic_lib"]
 services-azblob = [
     "dep:sha2",
     "dep:reqsign",
@@ -277,10 +276,6 @@ optional = true
 [dependencies.async-trait]
 version = "0.1.68"
 
-[dependencies.atomic_lib]
-version = "0.39.0"
-optional = true
-
 [dependencies.await-tree]
 version = "0.2"
 optional = true
