--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -307,3 +307,3 @@
 [dependencies.ouroboros]
-version = "0.18.4"
+version = ">= 0.15.6, < 0.19"
 optional = true
@@ -323,3 +323,3 @@
 [dependencies.prometheus-client]
-version = "0.23.1"
+version = ">= 0.22.2, < 0.24"
 optional = true
@@ -327,3 +327,3 @@
 [dependencies.prost]
-version = "0.13"
+version = ">= 0.12.6, < 0.14"
 optional = true
@@ -352,3 +352,3 @@
 [dependencies.rocksdb]
-version = "0.21.0"
+version = "0.23"
 optional = true
@@ -431,3 +431,2 @@
 features = [
-    "async",
     "async_tokio",
@@ -439,3 +438,3 @@
 [dev-dependencies.libtest-mimic]
-version = "0.8"
+version = ">= 0.7, < 0.9"
 
