Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -305,7 +305,7 @@ features = [
 optional = true
 
 [dependencies.ouroboros]
-version = "0.18.4"
+version = ">= 0.15.6, < 0.19"
 optional = true
 
 [dependencies.percent-encoding]
@@ -321,15 +321,15 @@ features = ["process"]
 optional = true
 
 [dependencies.prometheus-client]
-version = "0.23.1"
+version = ">= 0.22.2, < 0.24"
 optional = true
 
 [dependencies.prost]
-version = "0.13"
+version = ">= 0.12.6, < 0.14"
 optional = true
 
 [dependencies.quick-xml]
-version = "0.36"
+version = "0"
 features = [
     "serialize",
     "overlapped-lists",
@@ -350,7 +350,7 @@ features = ["stream"]
 default-features = false
 
 [dependencies.rocksdb]
-version = "0.21.0"
+version = "0.23"
 optional = true
 default-features = false
 
@@ -429,7 +429,6 @@ features = [
 [dev-dependencies.criterion]
 version = "0.5"
 features = [
-    "async",
     "async_tokio",
 ]
 
@@ -437,7 +436,7 @@ features = [
 version = "0.15"
 
 [dev-dependencies.libtest-mimic]
-version = "0.8"
+version = ">= 0.7, < 0.9"
 
 [dev-dependencies.pretty_assertions]
 version = "1"
