Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -134,15 +134,6 @@ services-postgresql = [
     "dep:sqlx",
     "sqlx?/postgres",
 ]
-services-redis = [
-    "dep:redis",
-    "dep:bb8",
-    "redis?/tokio-rustls-comp",
-]
-services-redis-native-tls = [
-    "services-redis",
-    "redis?/tokio-native-tls-comp",
-]
 services-rocksdb = [
     "dep:rocksdb",
     "internal-tokio-rt",
@@ -348,15 +339,6 @@ features = [
 version = "0.8"
 optional = true
 
-[dependencies.redis]
-version = "0.27"
-features = [
-    "cluster-async",
-    "tokio-comp",
-    "connection-manager",
-]
-optional = true
-
 [dependencies.reqsign]
 version = "0.16.1"
 optional = true
