Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -49,7 +49,6 @@ default = [
 executors-tokio = ["tokio/rt"]
 internal-tokio-rt = ["tokio/rt-multi-thread"]
 layers-chaos = ["dep:rand"]
-layers-fastmetrics = ["dep:fastmetrics"]
 layers-mime-guess = ["dep:mime_guess"]
 layers-prometheus = ["dep:prometheus"]
 layers-prometheus-client = ["dep:prometheus-client"]
@@ -226,10 +225,6 @@ optional = true
 version = "0.15"
 optional = true
 
-[dependencies.fastmetrics]
-version = "0.3.0"
-optional = true
-
 [dependencies.flume]
 version = "0.11"
 optional = true
