Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -146,7 +146,6 @@ services-sqlite = [
     "dep:ouroboros",
 ]
 services-supabase = []
-services-surrealdb = ["dep:surrealdb"]
 services-swift = []
 services-tikv = ["tikv-client"]
 services-upyun = [
@@ -360,11 +359,6 @@ version = "0.8.0"
 features = ["runtime-tokio-rustls"]
 optional = true
 
-[dependencies.surrealdb]
-version = "2"
-features = ["protocol-http"]
-optional = true
-
 [dependencies.tikv-client]
 version = "0.3.0"
 optional = true
