Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -104,11 +104,6 @@ services-gcs = [
     "reqsign?/reqwest_request",
 ]
 services-gdrive = ["internal-path-cache"]
-services-ghac = [
-    "dep:ghac",
-    "dep:prost",
-    "services-azblob",
-]
 services-github = []
 services-gridfs = [
     "dep:mongodb",
@@ -315,10 +310,6 @@ features = [
 ]
 default-features = false
 
-[dependencies.ghac]
-version = "0.2.0"
-optional = true
-
 [dependencies.governor]
 version = "0.6.0"
 features = ["std"]
