Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -49,7 +49,6 @@ executors-tokio = ["tokio/rt"]
 internal-tokio-rt = ["tokio/rt-multi-thread"]
 layers-blocking = ["internal-tokio-rt"]
 layers-chaos = ["dep:rand"]
-layers-dtrace = ["dep:probe"]
 layers-mime-guess = ["dep:mime_guess"]
 layers-prometheus = ["dep:prometheus"]
 layers-prometheus-client = ["dep:prometheus-client"]
@@ -280,10 +279,6 @@ optional = true
 [dependencies.percent-encoding]
 version = "2"
 
-[dependencies.probe]
-version = "0.5.1"
-optional = true
-
 [dependencies.prometheus]
 version = "0.13"
 features = ["process"]
