Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -109,10 +109,6 @@ services-koofr = []
 services-lakefs = []
 services-memcached = ["dep:bb8"]
 services-memory = []
-services-monoiofs = [
-    "dep:monoio",
-    "dep:flume",
-]
 services-mysql = [
     "dep:sqlx",
     "sqlx?/mysql",
@@ -310,16 +306,6 @@ version = "0.10"
 version = "2.0.5"
 optional = true
 
-[dependencies.monoio]
-version = "0.2.4"
-features = [
-    "sync",
-    "mkdirat",
-    "unlinkat",
-    "renameat",
-]
-optional = true
-
 [dependencies.once_cell]
 version = "1"
 
