Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -102,7 +102,6 @@ services-ipfs = ["dep:prost"]
 services-ipmfs = []
 services-koofr = []
 services-lakefs = []
-services-memcached = ["dep:bb8"]
 services-memory = []
 services-mysql = [
     "dep:sqlx",
@@ -135,11 +134,6 @@ services-s3 = [
     "dep:crc32c",
 ]
 services-seafile = []
-services-sftp = [
-    "dep:openssh",
-    "dep:openssh-sftp-client",
-    "dep:bb8",
-]
 services-sqlite = [
     "dep:sqlx",
     "sqlx?/sqlite",
@@ -208,10 +202,6 @@ features = ["tokio-sleep"]
 [dependencies.base64]
 version = "0.22"
 
-[dependencies.bb8]
-version = "0.9"
-optional = true
-
 [dependencies.bytes]
 version = "1.6"
 
