Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -109,7 +109,6 @@ services-gridfs = [
     "dep:mongodb-internal-macros",
 ]
 services-hdfs = ["dep:hdrs"]
-services-hdfs-native = ["hdfs-native"]
 services-http = []
 services-huggingface = []
 services-icloud = ["internal-path-cache"]
@@ -309,10 +308,6 @@ features = [
 ]
 default-features = false
 
-[dependencies.hdfs-native]
-version = "0.10"
-optional = true
-
 [dependencies.hdrs]
 version = "0.3.2"
 features = ["async_file"]
