Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -207,17 +207,17 @@ name = "opendal"
 path = "src/lib.rs"
 bench = false
 
-[[bench]]
-name = "ops"
-path = "benches/ops/main.rs"
-harness = false
-required-features = ["tests"]
-
-[[bench]]
-name = "types"
-path = "benches/types/main.rs"
-harness = false
-required-features = ["tests"]
+#[[bench]]
+#name = "ops"
+#path = "benches/ops/main.rs"
+#harness = false
+#required-features = ["tests"]
+
+#[[bench]]
+#name = "types"
+#path = "benches/types/main.rs"
+#harness = false
+#required-features = ["tests"]
 
 [dependencies.anyhow]
 version = "1.0.30"
@@ -481,9 +481,6 @@ version = "0.8"
 [dev-dependencies.sha2]
 version = "0.10"
 
-[dev-dependencies.size]
-version = "0.4"
-
 [dev-dependencies.tokio]
 version = "1.27"
 features = [
