Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -55,14 +55,6 @@ layers-chaos = ["dep:rand"]
 layers-dtrace = ["dep:probe"]
 layers-metrics = ["dep:metrics"]
 layers-mime-guess = ["dep:mime_guess"]
-layers-otel-metrics = [
-    "dep:opentelemetry",
-    "opentelemetry/metrics",
-]
-layers-otel-trace = [
-    "dep:opentelemetry",
-    "opentelemetry/trace",
-]
 layers-prometheus = ["dep:prometheus"]
 layers-prometheus-client = ["dep:prometheus-client"]
 layers-throttle = ["dep:governor"]
@@ -441,10 +433,6 @@ features = [
 ]
 optional = true
 
-[dependencies.opentelemetry]
-version = "0.28"
-optional = true
-
 [dependencies.ouroboros]
 version = "0.18.4"
 optional = true
@@ -599,13 +587,0 @@ version = "0.15"
-[dev-dependencies.opentelemetry]
-version = "0.28"
-features = ["trace"]
-default-features = false
-
-[dev-dependencies.opentelemetry-otlp]
-version = "0.28"
-features = ["grpc-tonic"]
-
-[dev-dependencies.opentelemetry_sdk]
-version = "0.28"
-features = ["rt-tokio"]
-
@@ -629,9 +604,6 @@ features = [
     "rt-multi-thread",
 ]
 
-[dev-dependencies.tracing-opentelemetry]
-version = "0.29.0"
-
 [dev-dependencies.tracing-subscriber]
 version = "0.3"
 features = [
