Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -108,7 +108,6 @@ services-gridfs = [
     "dep:mongodb",
     "dep:mongodb-internal-macros",
 ]
-services-hdfs = ["dep:hdrs"]
 services-http = []
 services-huggingface = []
 services-icloud = ["internal-path-cache"]
@@ -308,11 +307,6 @@ features = [
 ]
 default-features = false
 
-[dependencies.hdrs]
-version = "0.3.2"
-features = ["async_file"]
-optional = true
-
 [dependencies.hmac]
 version = "0.12.1"
 optional = true
