Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -147,7 +147,6 @@ services-sqlite = [
 ]
 services-supabase = []
 services-swift = []
-services-tikv = ["tikv-client"]
 services-upyun = [
     "dep:hmac",
     "dep:sha1",
@@ -359,11 +358,6 @@ version = "0.8.0"
 features = ["runtime-tokio-rustls"]
 optional = true
 
-[dependencies.tikv-client]
-version = "0.3.0"
-optional = true
-default-features = false
-
 [dependencies.tokio]
 version = "1.27"
 features = [
