Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -52,7 +52,6 @@ layers-async-backtrace = ["dep:async-bac
 layers-blocking = ["internal-tokio-rt"]
 layers-chaos = ["dep:rand"]
 layers-dtrace = ["dep:probe"]
-layers-metrics = ["dep:metrics"]
 layers-mime-guess = ["dep:mime_guess"]
 layers-prometheus = ["dep:prometheus"]
 layers-prometheus-client = ["dep:prometheus-client"]
@@ -320,10 +319,6 @@ version = "0.4"
 [dependencies.md-5]
 version = "0.10"
 
-[dependencies.metrics]
-version = "0.24"
-optional = true
-
 [dependencies.mime_guess]
 version = "2.0.5"
 optional = true
