Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -53,7 +53,6 @@ layers-await-tree = ["dep:await-tree"]
 layers-blocking = ["internal-tokio-rt"]
 layers-chaos = ["dep:rand"]
 layers-dtrace = ["dep:probe"]
-layers-fastrace = ["dep:fastrace"]
 layers-metrics = ["dep:metrics"]
 layers-mime-guess = ["dep:mime_guess"]
 layers-otel-metrics = [
@@ -338,10 +337,6 @@ version = "0.14"
 features = ["tls"]
 optional = true
 
-[dependencies.fastrace]
-version = "0.7.1"
-optional = true
-
 [dependencies.flume]
 version = "0.11"
 optional = true
@@ -598,7 +598,0 @@
-[dev-dependencies.fastrace]
-version = "0.7"
-features = ["enable"]
-
-[dev-dependencies.fastrace-jaeger]
-version = "0.7"
-
