Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -49,7 +49,6 @@ executors-tokio = ["tokio/rt"]
 internal-path-cache = ["dep:moka"]
 internal-tokio-rt = ["tokio/rt-multi-thread"]
 layers-async-backtrace = ["dep:async-backtrace"]
-layers-await-tree = ["dep:await-tree"]
 layers-blocking = ["internal-tokio-rt"]
 layers-chaos = ["dep:rand"]
 layers-dtrace = ["dep:probe"]
@@ -267,10 +266,6 @@ optional = true
 [dependencies.async-trait]
 version = "0.1.68"
 
-[dependencies.await-tree]
-version = "0.2"
-optional = true
-
 [dependencies.backon]
 version = "1.2"
 features = ["tokio-sleep"]
