Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -80,7 +80,6 @@ services-b2 = []
 services-cacache = ["dep:cacache"]
 services-chainsafe = []
 services-cloudflare-kv = []
-services-compfs = ["dep:compio"]
 services-cos = [
     "dep:reqsign",
     "reqsign?/services-tencent",
@@ -293,16 +292,6 @@ features = [
 ]
 default-features = false
 
-[dependencies.compio]
-version = "0.12.0"
-features = [
-    "runtime",
-    "bytes",
-    "polling",
-    "dispatcher",
-]
-optional = true
-
 [dependencies.crc32c]
 version = "0.6.6"
 optional = true
