Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -116,7 +116,6 @@ services-koofr = []
 services-lakefs = []
 services-memcached = ["dep:bb8"]
 services-memory = []
-services-mini-moka = ["dep:mini-moka"]
 services-moka = ["dep:moka"]
 services-mongodb = [
     "dep:mongodb",
@@ -323,10 +322,6 @@ version = "0.10"
 version = "2.0.5"
 optional = true
 
-[dependencies.mini-moka]
-version = "0.10"
-optional = true
-
 [dependencies.moka]
 version = "0.12"
 features = [
