Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -130,10 +130,6 @@ services-oss = [
     "reqsign?/reqwest_request",
 ]
 services-pcloud = []
-services-persy = [
-    "dep:persy",
-    "internal-tokio-rt",
-]
 services-postgresql = [
     "dep:sqlx",
     "sqlx?/postgres",
@@ -328,10 +324,6 @@ optional = true
 [dependencies.percent-encoding]
 version = "2"
 
-[dependencies.persy]
-version = "1.4.6"
-optional = true
-
 [dependencies.probe]
 version = "0.5.1"
 optional = true
