Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -134,10 +134,6 @@ services-postgresql = [
     "dep:sqlx",
     "sqlx?/postgres",
 ]
-services-redb = [
-    "dep:redb",
-    "internal-tokio-rt",
-]
 services-redis = [
     "dep:redis",
     "dep:bb8",
@@ -352,10 +348,6 @@ features = [
 version = "0.8"
 optional = true
 
-[dependencies.redb]
-version = "2"
-optional = true
-
 [dependencies.redis]
 version = "0.27"
 features = [
