Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -141,11 +141,6 @@ services-s3 = [
     "dep:crc32c",
 ]
 services-seafile = []
-services-sftp = [
-    "dep:openssh",
-    "dep:openssh-sftp-client",
-    "dep:bb8",
-]
 services-sqlite = [
     "dep:sqlx",
     "sqlx?/sqlite",
@@ -286,18 +281,6 @@ version = "0.10"
 version = "2.0.5"
 optional = true
 
-[dependencies.openssh]
-version = "0.11.0"
-optional = true
-
-[dependencies.openssh-sftp-client]
-version = "0.15.2"
-features = [
-    "openssh",
-    "tracing",
-]
-optional = true
-
 [dependencies.ouroboros]
 version = "0.18.4"
 optional = true
