Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -145,10 +145,6 @@ services-sftp = [
     "dep:openssh-sftp-client",
     "dep:bb8",
 ]
-services-sled = [
-    "dep:sled",
-    "internal-tokio-rt",
-]
 services-sqlite = [
     "dep:sqlx",
     "sqlx?/sqlite",
@@ -367,4 +363,0 @@
-[dependencies.sled]
-version = "0.34.7"
-optional = true
-
