Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -105,4 +105,0 @@
-services-gridfs = [
-    "dep:mongodb",
-    "dep:mongodb-internal-macros",
-]
@@ -115,10 +111,6 @@ services-koofr = []
 services-lakefs = []
 services-memcached = ["dep:bb8"]
 services-memory = []
-services-mongodb = [
-    "dep:mongodb",
-    "dep:mongodb-internal-macros",
-]
 services-monoiofs = [
     "dep:monoio",
     "dep:flume",
@@ -320,14 +312,6 @@ version = "0.10"
 version = "2.0.5"
 optional = true
 
-[dependencies.mongodb]
-version = ">=3,<3.2.0"
-optional = true
-
-[dependencies.mongodb-internal-macros]
-version = ">=3,<3.2.0"
-optional = true
-
 [dependencies.monoio]
 version = "0.2.4"
 features = [
