Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -113,11 +113,6 @@ services-mysql = [
     "dep:sqlx",
     "sqlx?/mysql",
 ]
-services-nebula-graph = [
-    "dep:rust-nebula",
-    "dep:bb8",
-    "dep:snowflaked",
-]
 services-obs = [
     "dep:reqsign",
     "reqsign?/services-huaweicloud",
@@ -354,5 +349,0 @@
-[dependencies.rust-nebula]
-version = "^0.0.2"
-features = ["graph"]
-optional = true
-
@@ -360,5 +360,0 @@
-[dependencies.snowflaked]
-version = "1"
-features = ["sync"]
-optional = true
-
