Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -56,7 +56,6 @@ layers-metrics = ["dep:metrics"]
 layers-mime-guess = ["dep:mime_guess"]
 layers-prometheus = ["dep:prometheus"]
 layers-prometheus-client = ["dep:prometheus-client"]
-layers-throttle = ["dep:governor"]
 layers-tracing = ["dep:tracing"]
 services-aliyun-drive = []
 services-alluxio = []
@@ -310,11 +309,6 @@ features = [
 ]
 default-features = false
 
-[dependencies.governor]
-version = "0.6.0"
-features = ["std"]
-optional = true
-
 [dependencies.hdfs-native]
 version = "0.10"
 optional = true
