Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -46,7 +46,6 @@ default = [
     "services-memory",
 ]
 executors-tokio = ["tokio/rt"]
-internal-path-cache = ["dep:moka"]
 internal-tokio-rt = ["tokio/rt-multi-thread"]
 layers-async-backtrace = ["dep:async-backtrace"]
 layers-blocking = ["internal-tokio-rt"]
@@ -101,7 +100,6 @@ services-gcs = [
     "reqsign?/services-google",
     "reqsign?/reqwest_request",
 ]
-services-gdrive = ["internal-path-cache"]
 services-github = []
 services-gridfs = [
     "dep:mongodb",
@@ -109,14 +107,12 @@ services-gridfs = [
 ]
 services-http = []
 services-huggingface = []
-services-icloud = ["internal-path-cache"]
 services-ipfs = ["dep:prost"]
 services-ipmfs = []
 services-koofr = []
 services-lakefs = []
 services-memcached = ["dep:bb8"]
 services-memory = []
-services-moka = ["dep:moka"]
 services-mongodb = [
     "dep:mongodb",
     "dep:mongodb-internal-macros",
@@ -322,14 +318,6 @@ version = "0.10"
 version = "2.0.5"
 optional = true
 
-[dependencies.moka]
-version = "0.12"
-features = [
-    "future",
-    "sync",
-]
-optional = true
-
 [dependencies.mongodb]
 version = ">=3,<3.2.0"
 optional = true
