Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -90,11 +90,6 @@ services-fs = [
     "tokio/fs",
     "internal-tokio-rt",
 ]
-services-ftp = [
-    "dep:suppaftp",
-    "dep:bb8",
-    "dep:async-tls",
-]
 services-gcs = [
     "dep:reqsign",
     "reqsign?/services-google",
@@ -365,16 +360,6 @@ version = "0.8.0"
 features = ["runtime-tokio-rustls"]
 optional = true
 
-[dependencies.suppaftp]
-version = "6.0.3"
-features = [
-    "async-secure",
-    "rustls",
-    "async-rustls",
-]
-optional = true
-default-features = false
-
 [dependencies.surrealdb]
 version = "2"
 features = ["protocol-http"]
