--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -210,5 +210,5 @@
-[[bench]]
-name = "ops"
-path = "benches/ops/main.rs"
-harness = false
-required-features = ["tests"]
+#[[bench]]
+#name = "ops"
+#path = "benches/ops/main.rs"
+#harness = false
+#required-features = ["tests"]
@@ -216,4 +216,4 @@
-[[bench]]
-name = "types"
-path = "benches/types/main.rs"
-harness = false
+#[[bench]]
+#name = "types"
+#path = "benches/types/main.rs"
+#harness = false
@@ -483,3 +482,0 @@
-
-[dev-dependencies.size]
-version = "0.4"
