Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -89,7 +89,6 @@ services-d1 = []
 services-dashmap = ["dep:dashmap"]
 services-dbfs = []
 services-dropbox = []
-services-foundationdb = ["dep:foundationdb"]
 services-fs = [
     "tokio/fs",
     "internal-tokio-rt",
@@ -308,14 +307,6 @@ optional = true
 version = "0.11"
 optional = true
 
-[dependencies.foundationdb]
-version = "0.9.0"
-features = [
-    "embedded-fdb-include",
-    "fdb-7_3",
-]
-optional = true
-
 [dependencies.futures]
 version = "0.3"
 features = [
