Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -58,8 +58,0 @@
-layers-otel-metrics = [
-    "dep:opentelemetry",
-    "opentelemetry/metrics",
-]
-layers-otel-trace = [
-    "dep:opentelemetry",
-    "opentelemetry/trace",
-]
@@ -444,4 +436,0 @@
-[dependencies.opentelemetry]
-version = "0.30.0"
-optional = true
-
@@ -599,13 +587,0 @@
-[dev-dependencies.opentelemetry]
-version = "0.30.0"
-features = ["trace"]
-default-features = false
-
-[dev-dependencies.opentelemetry-otlp]
-version = "0.30.0"
-features = ["grpc-tonic"]
-
-[dev-dependencies.opentelemetry_sdk]
-version = "0.30.0"
-features = ["rt-tokio"]
-
@@ -632,3 +607,0 @@
-[dev-dependencies.tracing-opentelemetry]
-version = "0.31.0"
-
