Index: opendal/Cargo.toml
===================================================================
--- opendal.orig/Cargo.toml
+++ opendal/Cargo.toml
@@ -56,1 +56,0 @@
-layers-fastrace = ["dep:fastrace"]
@@ -341,4 +340,0 @@
-[dependencies.fastrace]
-version = "0.7.1"
-optional = true
-
@@ -598,7 +598,0 @@
-[dev-dependencies.fastrace]
-version = "0.7"
-features = ["enable"]
-
-[dev-dependencies.fastrace-jaeger]
-version = "0.7"
-
