Index: opendal.bak/Cargo.toml
===================================================================
--- opendal.bak.orig/Cargo.toml
+++ opendal.bak/Cargo.toml
@@ -90,10 +90,6 @@ services-d1 = []
 services-dashmap = ["dep:dashmap"]
 services-dbfs = []
 services-dropbox = []
-services-etcd = [
-    "dep:etcd-client",
-    "dep:bb8",
-]
 services-foundationdb = ["dep:foundationdb"]
 services-fs = [
     "tokio/fs",
@@ -319,11 +315,6 @@ optional = true
 version = "0.15"
 optional = true
 
-[dependencies.etcd-client]
-version = "0.14"
-features = ["tls"]
-optional = true
-
 [dependencies.flume]
 version = "0.11"
 optional = true
