Index: normpath/Cargo.toml
===================================================================
--- normpath.orig/Cargo.toml
+++ normpath/Cargo.toml
@@ -68,17 +68,7 @@ version = "3.2"
 
 [features]
 localization = [
-    "windows-sys/Win32_UI_Shell",
-    "windows-sys/Win32_UI_WindowsAndMessaging",
 ]
 
 [target."cfg(not(windows))".dev-dependencies.libc]
 version = "0.2"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.48"
-features = ["Win32_Storage_FileSystem"]
-
-[target."cfg(windows)".dev-dependencies.windows-sys]
-version = "0.48"
-features = ["Win32_Foundation"]
