--- a/Cargo.toml
+++ b/Cargo.toml
@@ -62,4 +62,2 @@
 localization = [
-    "windows-sys/Win32_UI_Shell",
-    "windows-sys/Win32_UI_WindowsAndMessaging",
 ]
@@ -93,1 +93,1 @@
-[target."cfg(windows)".dependencies.windows-sys]
+[disabled.target."cfg(windows)".dependencies.windows-sys]
@@ -97,1 +97,1 @@
-[target."cfg(windows)".dev-dependencies.normpath_macros]
+[disabled.target."cfg(windows)".dev-dependencies.normpath_macros]
@@ -100,1 +100,1 @@
-[target."cfg(windows)".dev-dependencies.windows-sys]
+[disabled.target."cfg(windows)".dev-dependencies.windows-sys]
