Index: glutin/Cargo.toml
===================================================================
--- glutin.orig/Cargo.toml
+++ glutin/Cargo.toml
@@ -52,7 +52,7 @@ version = "0.4"
 version = "0.1"
 
 [target."cfg(any(target_os = \"linux\", target_os = \"freebsd\", target_os = \"dragonfly\", target_os = \"netbsd\", target_os = \"openbsd\"))".dependencies.parking_lot]
-version = "0.11"
+version = "0.12"
 
 [target."cfg(any(target_os = \"linux\", target_os = \"freebsd\", target_os = \"dragonfly\", target_os = \"netbsd\", target_os = \"openbsd\"))".dependencies.wayland-client]
 version = "0.29"
