diff --git a/Cargo.toml b/Cargo.toml
index 6ede7a2..bce9298 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,10 +33,6 @@ targets = ["x86_64-unknown-linux-gnu"]
 [lib]
 proc-macro = true
 
-[dependencies.clang-ast]
-version = "0.1"
-optional = true
-
 [dependencies.flate2]
 version = "1.0"
 optional = true
@@ -70,7 +66,6 @@ version = "1.0"
 [features]
 experimental-async-fn = []
 experimental-enum-variants-from-header = [
-    "clang-ast",
     "flate2",
     "memmap",
     "serde",
