Origin: upstream, 1.0.169
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,32 +39,12 @@ targets = ["x86_64-unknown-linux-gnu"]
 
 [features]
 experimental-async-fn = []
-experimental-enum-variants-from-header = [
-    "clang-ast",
-    "flate2",
-    "memmap",
-    "serde",
-    "serde_derive",
-    "serde_json",
-]
 
 [lib]
 name = "cxxbridge_macro"
 path = "src/lib.rs"
 proc-macro = true
 
-[dependencies.clang-ast]
-version = "0.1.18"
-optional = true
-
-[dependencies.flate2]
-version = "1.0.26"
-optional = true
-
-[dependencies.memmap]
-version = "0.7"
-optional = true
-
 [dependencies.proc-macro2]
 version = "1.0.74"
 
@@ -74,18 +54,6 @@ version = "1.0.35"
 [dependencies.rustversion]
 version = "1"
 
-[dependencies.serde]
-version = "1.0.166"
-optional = true
-
-[dependencies.serde_derive]
-version = "1.0.166"
-optional = true
-
-[dependencies.serde_json]
-version = "1.0.100"
-optional = true
-
 [dependencies.syn]
 version = "2.0.46"
 features = ["full"]
