From: Maytham Alsudany <maytha8thedev@gmail.com>
Forwarded: not-needed
Description: Disable badssl test
 This test accesses the internet.

Index: tokio-rustls/Cargo.toml
===================================================================
--- tokio-rustls.orig/Cargo.toml
+++ tokio-rustls/Cargo.toml
@@ -42,7 +42,7 @@ name = "tokio_rustls"
 path = "src/lib.rs"
 doctest = false
 
-[[test]]
+[[disabled.test]]
 name = "badssl"
 path = "tests/badssl.rs"
 
@@ -87,9 +87,6 @@ features = ["pem"]
 version = "1.0"
 features = ["full"]
 
-[dev-dependencies.rustls-native-certs]
-version = "0.8"
-
 [features]
 default = [
     "logging",
