From: Maytham Alsudany <maytha8thedev@gmail.com>
Forwarded: not-needed
Description: Use rcgen 0.11

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -80,7 +80,7 @@
 version = "1.1"
 
 [dev-dependencies.rcgen]
-version = "0.13"
+version = "0.11"
 features = ["pem"]
 
 [dev-dependencies.tokio]
--- a/tests/certs/main.rs
+++ b/tests/certs/main.rs
@@ -12,35 +12,33 @@
 #[test]
 #[ignore]
 fn regenerate_certs() {
-    let root_key = KeyPair::generate().unwrap();
-    let root_ca = issuer_params("Rustls Robust Root")
-        .self_signed(&root_key)
-        .unwrap();
+    let root_key = KeyPair::generate(&rcgen::PKCS_ECDSA_P256_SHA256).unwrap();
+    let mut root_ca_params = issuer_params("Rustls Robust Root");
+    root_ca_params.key_pair = Some(root_key);
+    let root_ca = rcgen::Certificate::from_params(root_ca_params).unwrap();
 
     let mut root_file = File::create("tests/certs/root.pem").unwrap();
-    root_file.write_all(root_ca.pem().as_bytes()).unwrap();
+    root_file.write_all(root_ca.serialize_pem().unwrap().as_bytes()).unwrap();
 
-    let intermediate_key = KeyPair::generate().unwrap();
-    let intermediate_ca = issuer_params("Rustls Robust Root - Rung 2")
-        .signed_by(&intermediate_key, &root_ca, &root_key)
-        .unwrap();
+    let intermediate_key = KeyPair::generate(&rcgen::PKCS_ECDSA_P256_SHA256).unwrap();
+    let mut intermediate_ca_params = issuer_params("Rustls Robust Root - Rung 2");
+    intermediate_ca_params.key_pair = Some(intermediate_key);
+    let intermediate_ca = rcgen::Certificate::from_params(intermediate_ca_params).unwrap();
 
-    let end_entity_key = KeyPair::generate().unwrap();
+    let end_entity_key = KeyPair::generate(&rcgen::PKCS_ECDSA_P256_SHA256).unwrap();
     let mut end_entity_params =
-        CertificateParams::new(vec![utils::TEST_SERVER_DOMAIN.to_string()]).unwrap();
+        CertificateParams::new(vec![utils::TEST_SERVER_DOMAIN.to_string()]);
     end_entity_params.is_ca = IsCa::ExplicitNoCa;
     end_entity_params.extended_key_usages = vec![
         ExtendedKeyUsagePurpose::ServerAuth,
         ExtendedKeyUsagePurpose::ClientAuth,
     ];
-    let end_entity = end_entity_params
-        .signed_by(&end_entity_key, &intermediate_ca, &intermediate_key)
-        .unwrap();
+    let end_entity = rcgen::Certificate::from_params(end_entity_params).unwrap();
 
     let mut chain_file = File::create("tests/certs/chain.pem").unwrap();
-    chain_file.write_all(end_entity.pem().as_bytes()).unwrap();
+    chain_file.write_all(end_entity.serialize_pem_with_signer(&intermediate_ca).unwrap().as_bytes()).unwrap();
     chain_file
-        .write_all(intermediate_ca.pem().as_bytes())
+        .write_all(intermediate_ca.serialize_pem_with_signer(&root_ca).unwrap().as_bytes())
         .unwrap();
 
     let mut key_file = File::create("tests/certs/end.key").unwrap();
