--- a/build.rs
+++ b/build.rs
@@ -11,7 +11,7 @@ fn main() {
 
     // Tell cargo to invalidate the built crate whenever the wrapper changes
     println!("cargo:rerun-if-changed=wrapper.h");
-
+    println!("dh-cargo:deb-built-using=libspa-rs-reexports=0={}", env::var("CARGO_MANIFEST_DIR").unwrap());
     let builder = bindgen::Builder::default()
         .header("wrapper.h")
         // Tell cargo to invalidate the built crate whenever any of the
