Forwarded: not-needed
Last-Update: 2025-03-31
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51 +51 @@
-[[test]]
+[[disabled.test]]
@@ -59 +59 @@
-[[test]]
+[[disabled.test]]
@@ -151 +151 @@
-[dev-dependencies.webpki-roots]
+[disabled.dev-dependencies.webpki-roots]
@@ -216,4 +216,4 @@
-webpki-roots = [
-    "hickory-client/webpki-roots",
-    "hickory-server/webpki-roots",
-]
+#webpki-roots = [
+#    "hickory-client/webpki-roots",
+#    "hickory-server/webpki-roots",
+#]
