Forwarded: not-needed
Last-Update: 2025-03-31
--- a/tests/named_https_tests.rs
+++ b/tests/named_https_tests.rs
@@ -28,2 +28,3 @@
 #[test]
+#[cfg(any())]
 fn test_example_https_toml_startup() {
--- a/tests/named_openssl_tests.rs
+++ b/tests/named_openssl_tests.rs
@@ -31,2 +31,3 @@
 #[test]
+#[cfg(any())]
 fn test_example_tls_toml_startup() {
@@ -36,2 +37,3 @@
 #[test]
+#[cfg(any())]
 fn test_example_tls_rustls_and_openssl_toml_startup() {
--- a/tests/named_quic_tests.rs
+++ b/tests/named_quic_tests.rs
@@ -22,2 +22,3 @@
 #[test]
+#[cfg(any())]
 fn test_example_quic_toml_startup() {
--- a/tests/named_rustls_tests.rs
+++ b/tests/named_rustls_tests.rs
@@ -31,2 +31,3 @@
 #[test]
+#[cfg(any())]
 fn test_example_tls_toml_startup() {
--- a/tests/named_test_rsa_dnssec.rs
+++ b/tests/named_test_rsa_dnssec.rs
@@ -105,2 +105,3 @@
 #[cfg(feature = "dnssec-openssl")]
+#[cfg(any())]
 fn test_rsa_sha256() {
@@ -116,2 +117,3 @@
 #[cfg(feature = "dnssec-openssl")]
+#[cfg(any())]
 fn test_rsa_sha512() {
@@ -127,2 +129,3 @@
 #[cfg(feature = "dnssec-openssl")]
+#[cfg(any())]
 fn test_ecdsa_p256() {
@@ -138,2 +141,3 @@
 #[cfg(feature = "dnssec-ring")]
+#[cfg(any())]
 fn test_ecdsa_p256_pkcs8() {
@@ -149,2 +153,3 @@
 #[cfg(feature = "dnssec-openssl")]
+#[cfg(any())]
 fn test_ecdsa_p384() {
@@ -160,2 +165,3 @@
 #[cfg(feature = "dnssec-ring")]
+#[cfg(any())]
 fn test_ecdsa_p384_pkcs8() {
@@ -171,2 +177,3 @@
 #[cfg(feature = "dnssec-ring")]
+#[cfg(any())]
 fn test_ed25519() {
@@ -183,2 +190,3 @@
 #[allow(deprecated)]
+#[cfg(any())]
 fn test_rsa_sha1_fails() {
@@ -195,2 +203,3 @@
 #[test]
+#[cfg(any())]
 fn test_dnssec_restart_with_update_journal() {
@@ -231,2 +240,3 @@
 #[test]
+#[cfg(any())]
 fn test_dnssec_restart_with_update_journal_dep() {
@@ -266,2 +276,3 @@
 #[test]
+#[cfg(any())]
 fn crypto_self_test() {
--- a/tests/named_tests.rs
+++ b/tests/named_tests.rs
@@ -180,2 +180,3 @@
 #[test]
+#[cfg(any())]
 fn test_nxdomain_where_no_name_exists() {
