diff --git a/Cargo.toml b/Cargo.toml
index 368aa5e..15376a8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,23 +49,28 @@ required-features = ["cli"]
 [[example]]
 name = "cache"
 path = "examples/cache.rs"
+test = false
 
 [[example]]
 name = "cpu"
 path = "examples/cpu.rs"
+test = false
 
 [[example]]
 name = "serialize_deserialize"
 path = "examples/serialize_deserialize.rs"
 required-features = ["serde_json", "serialize"]
+test = false
 
 [[example]]
 name = "topology"
 path = "examples/topology.rs"
+test = false
 
 [[example]]
 name = "tsc_frequency"
 path = "examples/tsc_frequency.rs"
+test = false
 
 [dependencies.bitflags]
 version = "2.0"
