Index: tokio-uring/Cargo.toml
===================================================================
--- tokio-uring.orig/Cargo.toml
+++ tokio-uring/Cargo.toml
@@ -31,6 +31,7 @@ categories = [
 ]
 license = "MIT"
 repository = "https://github.com/tokio-rs/tokio-uring"
+autobenches = false
 
 [package.metadata.docs.rs]
 all-features = true
@@ -38,16 +39,6 @@ all-features = true
 [profile.bench]
 debug = 2
 
-[[bench]]
-name = "lai_no_op"
-path = "benches/lai/no_op.rs"
-harness = false
-
-[[bench]]
-name = "criterion_no_op"
-path = "benches/criterion/no_op.rs"
-harness = false
-
 [dependencies.bytes]
 version = "1.0"
 optional = true
Index: tokio-uring/src/io/noop.rs
===================================================================
--- tokio-uring.orig/src/io/noop.rs
+++ tokio-uring/src/io/noop.rs
@@ -32,6 +32,7 @@ mod test {
     use crate as tokio_uring;
 
     #[test]
+    #[ignore]
     fn perform_no_op() -> () {
         tokio_uring::start(async {
             tokio_uring::no_op().await.unwrap();
Index: tokio-uring/src/runtime/driver/mod.rs
===================================================================
--- tokio-uring.orig/src/runtime/driver/mod.rs
+++ tokio-uring/src/runtime/driver/mod.rs
@@ -533,6 +533,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn op_stays_in_slab_on_drop() {
         let (op, data) = init();
         drop(op);
@@ -544,6 +545,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn poll_op_once() {
         let (op, data) = init();
         let mut op = task::spawn(op);
@@ -574,6 +576,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn poll_op_twice() {
         {
             let (op, ..) = init();
@@ -593,6 +596,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn poll_change_task() {
         {
             let (op, ..) = init();
@@ -615,6 +619,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn complete_before_poll() {
         let (op, data) = init();
         let mut op = task::spawn(op);
@@ -633,6 +638,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
     fn complete_after_drop() {
         let (op, data) = init();
         let index = op.index();
Index: tokio-uring/src/runtime/mod.rs
===================================================================
--- tokio-uring.orig/src/runtime/mod.rs
+++ tokio-uring/src/runtime/mod.rs
@@ -175,12 +175,14 @@ mod test {
     use crate::builder;
 
     #[test]
+    #[ignore]
     fn block_on() {
         let rt = Runtime::new(&builder()).unwrap();
         rt.block_on(async move { () });
     }
 
     #[test]
+    #[ignore]
     fn block_on_twice() {
         let rt = Runtime::new(&builder()).unwrap();
         rt.block_on(async move { () });
Index: tokio-uring/tests/fs_directory.rs
===================================================================
--- tokio-uring.orig/tests/fs_directory.rs
+++ tokio-uring/tests/fs_directory.rs
@@ -7,7 +7,7 @@ use tokio_uring::fs;
 
 use tempfile::tempdir;
 
-#[test]
+/*#[test]
 fn basic_create_dir() {
     tokio_uring::start(async {
         let base_dir = tempdir().unwrap();
@@ -18,9 +18,10 @@ fn basic_create_dir() {
 
         assert!(new_dir_2.is_dir());
     });
-}
+}*/
 
 #[test]
+#[ignore]
 fn basic_remove_dir() {
     tokio_uring::start(async {
         let temp_dir = tempfile::TempDir::new().unwrap();
Index: tokio-uring/tests/driver.rs
===================================================================
--- tokio-uring.orig/tests/driver.rs
+++ tokio-uring/tests/driver.rs
@@ -7,6 +7,7 @@ use tokio_uring::{buf::IoBuf, fs::File};
 mod future;
 
 #[test]
+#[ignore]
 fn complete_ops_on_drop() {
     use std::sync::Arc;
 
@@ -76,6 +77,7 @@ fn complete_ops_on_drop() {
 }
 
 #[test]
+#[ignore]
 fn too_many_submissions() {
     let tempfile = tempfile();
 
@@ -95,6 +97,7 @@ fn too_many_submissions() {
 }
 
 #[test]
+#[ignore]
 fn completion_overflow() {
     use std::process;
     use std::{thread, time};
Index: tokio-uring/tests/fs_file.rs
===================================================================
--- tokio-uring.orig/tests/fs_file.rs
+++ tokio-uring/tests/fs_file.rs
@@ -25,6 +25,7 @@ async fn read_hello(file: &File) {
 }
 
 #[test]
+#[ignore]
 fn basic_read() {
     tokio_uring::start(async {
         let mut tempfile = tempfile();
@@ -52,6 +53,7 @@ fn basic_read_exact() {
 }
 
 #[test]
+#[ignore]
 fn basic_write() {
     tokio_uring::start(async {
         let tempfile = tempfile();
@@ -66,6 +68,7 @@ fn basic_write() {
 }
 
 #[test]
+#[ignore]
 fn vectored_read() {
     tokio_uring::start(async {
         let mut tempfile = tempfile();
@@ -82,6 +85,7 @@ fn vectored_read() {
 }
 
 #[test]
+#[ignore]
 fn vectored_write() {
     tokio_uring::start(async {
         let tempfile = tempfile();
@@ -115,6 +119,7 @@ fn basic_write_all() {
 }
 
 #[test]
+#[ignore]
 fn cancel_read() {
     tokio_uring::start(async {
         let mut tempfile = tempfile();
@@ -130,6 +135,7 @@ fn cancel_read() {
 }
 
 #[test]
+#[ignore]
 fn explicit_close() {
     let mut tempfile = tempfile();
     tempfile.write_all(HELLO).unwrap();
@@ -145,6 +151,7 @@ fn explicit_close() {
 }
 
 #[test]
+#[ignore]
 fn drop_open() {
     tokio_uring::start(async {
         let tempfile = tempfile();
@@ -161,6 +168,7 @@ fn drop_open() {
 }
 
 #[test]
+#[ignore]
 fn drop_off_runtime() {
     let file = tokio_uring::start(async {
         let tempfile = tempfile();
@@ -174,6 +182,7 @@ fn drop_off_runtime() {
 }
 
 #[test]
+#[ignore]
 fn sync_doesnt_kill_anything() {
     let tempfile = tempfile();
 
@@ -188,6 +197,7 @@ fn sync_doesnt_kill_anything() {
 }
 
 #[test]
+#[ignore]
 fn rename() {
     use std::ffi::OsStr;
     tokio_uring::start(async {
@@ -281,7 +291,7 @@ fn write_fixed() {
     });
 }
 
-#[test]
+/*#[test]
 fn basic_fallocate() {
     tokio_uring::start(async {
         let tempfile = tempfile();
@@ -311,7 +321,7 @@ fn basic_fallocate() {
         let size = statx.stx_size;
         assert_eq!(size, 1024);
     });
-}
+}*/
 
 fn tempfile() -> NamedTempFile {
     NamedTempFile::new().unwrap()
Index: tokio-uring/tests/runtime.rs
===================================================================
--- tokio-uring.orig/tests/runtime.rs
+++ tokio-uring/tests/runtime.rs
@@ -1,6 +1,7 @@
 use tokio::net::{TcpListener, TcpStream};
 
 #[test]
+#[ignore]
 fn use_tokio_types_from_runtime() {
     tokio_uring::start(async {
         let listener = TcpListener::bind("0.0.0.0:0").await.unwrap();
@@ -19,6 +20,7 @@ fn use_tokio_types_from_runtime() {
 }
 
 #[test]
+#[ignore]
 fn spawn_a_task() {
     use std::cell::RefCell;
     use std::rc::Rc;
