Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -131,6 +131,6 @@
 [dependencies.prost]
-version = "0.14.0"
+version = "0.12"
 
 [dependencies.prost-reflect]
-version = "0.16.0"
+version = "0.15.0"
 features = [
@@ -141,3 +141,3 @@
 [dependencies.prost-types]
-version = "0.14.0"
+version = "0.12"
 
@@ -158,6 +158,6 @@
 [dev-dependencies.prost-build]
-version = "0.14.0"
+version = "0.12"
 
 [dev-dependencies.prost-reflect]
-version = "0.16.0"
+version = "0.15.0"
 features = ["serde"]
