Index: hyperfine/Cargo.toml
===================================================================
--- hyperfine.orig/Cargo.toml
+++ hyperfine/Cargo.toml
@@ -113,11 +113,11 @@ version = "4.4.12"
 [build-dependencies.clap_complete]
 version = "4.2.1"
 
-[features]
-windows_process_extensions_main_thread_handle = []
+#[features]
+#windows_process_extensions_main_thread_handle = []
 
-[target."cfg(all(windows, not(windows_process_extensions_main_thread_handle)))".dependencies.once_cell]
-version = "1.19"
+#[target."cfg(all(windows, not(windows_process_extensions_main_thread_handle)))".dependencies.once_cell]
+#version = "1.19"
 
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2"
@@ -126,12 +126,12 @@ version = "0.2"
 version = "0.29"
 features = ["zerocopy"]
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_System_JobObjects",
-    "Win32_System_LibraryLoader",
-    "Win32_System_Threading",
-]
+# [target."cfg(windows)".dependencies.windows-sys]
+# version = "0.59"
+# features = [
+#     "Win32_Foundation",
+#     "Win32_Security",
+#     "Win32_System_JobObjects",
+#     "Win32_System_LibraryLoader",
+#     "Win32_System_Threading",
+# ]
