Index: zerocopy-derive/tests/deprecated.rs
===================================================================
--- zerocopy-derive.orig/tests/deprecated.rs
+++ zerocopy-derive/tests/deprecated.rs
@@ -45,4 +45,4 @@ test!(Enum => #[repr(u8)] enum Enum { A,
 
 test!(Struct => #[repr(C)] struct Struct; => TryFromBytes, FromZeros, FromBytes, KnownLayout, Immutable, IntoBytes, Unaligned);
 
-test!(Union => #[repr(C)] union Union{ a: (), } => TryFromBytes, FromZeros, FromBytes, KnownLayout, Immutable, IntoBytes, Unaligned);
+//test!(Union => #[repr(C)] union Union{ a: (), } => TryFromBytes, FromZeros, FromBytes, KnownLayout, Immutable, IntoBytes, Unaligned);
Index: zerocopy-derive/tests/union_to_bytes.rs
===================================================================
--- zerocopy-derive.orig/tests/union_to_bytes.rs
+++ zerocopy-derive/tests/union_to_bytes.rs
@@ -18,7 +18,7 @@ include!("include.rs");
 //   - no padding (size of union equals size of each field type)
 // - `repr(packed)`
 
-#[derive(imp::IntoBytes, Clone, Copy)]
+/*#[derive(imp::IntoBytes, Clone, Copy)]
 #[repr(C)]
 union CZst {
     a: (),
@@ -72,4 +72,4 @@ union CMultibytePacked {
     c: f32,
 }
 
-util_assert_impl_all!(CMultibytePacked: imp::IntoBytes);
+util_assert_impl_all!(CMultibytePacked: imp::IntoBytes);*/
