--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,21 +57,13 @@ optional = true
 version = "0.2"
 
 [dependencies.x11]
-version = "2.20"
+version = "2.19"
 optional = true
 
 [dev-dependencies.gir-format-check]
-version = "^0.1"
+version = "0.1"
 
 [features]
-dox = [
-    "ffi/dox",
-    "gdk/dox",
-    "gio/dox",
-    "glib/dox",
-    "egl",
-    "xlib",
-]
 egl = ["khronos-egl"]
 v4_10 = [
     "ffi/v4_10",
