From 796f9acbcbe9f23150afb35004719d365c58364c Mon Sep 17 00:00:00 2001
From: QiangHeisenberg <liziqiang5@huawei.com>
Date: Fri, 28 Apr 2023 17:08:04 +0800
Subject: [PATCH] fix  test failed

---
 lib.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib.rs b/lib.rs
index e0197568a6c..7336f982fc8 100644
--- a/lib.rs
+++ b/lib.rs
@@ -199,7 +199,7 @@ impl Registry {
     /// let mut handle = Easy::new();
     /// // If connecting to crates.io, a user-agent is required.
     /// handle.useragent("my_crawler (example.com/info)");
-    /// let mut reg = Registry::new_handle(String::from("https://crates.io"), None, handle);
+    /// let mut reg = Registry::new_handle(String::from("https://crates.io"), None, handle, true);
     /// ```
     pub fn new_handle(
         host: String,
