--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,10 +40,6 @@ name = "glib"
 name = "subclass_compiletest"
 required-features = ["compiletests"]
 
-[[bench]]
-name = "gstring"
-harness = false
-
 [dependencies.bitflags]
 version = "1.0"
 
@@ -74,7 +70,7 @@ optional = true
 package = "gio-sys"
 
 [dependencies.glib-macros]
-version = "0.17.3"
+version = "0.17"
 
 [dependencies.gobject_ffi]
 version = "0.17"
@@ -106,25 +102,20 @@ features = [
 version = "1"
 
 [dev-dependencies.criterion]
-version = "0.4.0"
+version = "0.5"
 
 [dev-dependencies.gir-format-check]
-version = "^0.1"
+version = "0.1"
 
 [dev-dependencies.tempfile]
 version = "3"
 
-[dev-dependencies.trybuild2]
+[dev-dependencies.trybuild]
 version = "1"
 
 [features]
 compiletests = []
 default = ["gio"]
-dox = [
-    "ffi/dox",
-    "gobject_ffi/dox",
-    "log_macros",
-]
 gio = ["gio_ffi"]
 log = ["rs-log"]
 log_macros = ["log"]
