commit 2e684b4793883a37c3b2864e47dcfbd7517afb7c
Author: Nick Fitzgerald <fitzgen@gmail.com>
Date:   Mon Jan 24 11:06:31 2022 -0800

    Fix tests for non 64-bit architectures
    
    Fixes #95

diff --git a/tests/derive.rs b/tests/derive.rs
old mode 100644
new mode 100755
index 9dfbbd5b8..adf118855
--- a/tests/derive.rs
+++ b/tests/derive.rs
@@ -164,7 +164,7 @@ fn one_lifetime() {
     assert_eq!("abc", lifetime.alpha);
 
     let (lower, upper) = <OneLifetime as Arbitrary>::size_hint(0);
-    assert_eq!(lower, 8);
+    assert_eq!(lower, std::mem::size_of::<usize>());
     assert_eq!(upper, None);
 }
 
@@ -183,6 +183,6 @@ fn two_lifetimes() {
     assert_eq!("def", lifetime.beta);
 
     let (lower, upper) = <TwoLifetimes as Arbitrary>::size_hint(0);
-    assert_eq!(lower, 16);
+    assert_eq!(lower, std::mem::size_of::<usize>() * 2);
     assert_eq!(upper, None);
 }
