Index: sccache/Cargo.toml
===================================================================
--- sccache.orig/Cargo.toml
+++ sccache/Cargo.toml
@@ -87,7 +87,7 @@ version = "0.9"
 version = "0.1"
 
 [dependencies.base64]
-version = "0.21"
+version = "0.22"
 
 [dependencies.bincode]
 version = "1"
@@ -119,7 +119,7 @@ version = "5.0.1"
 version = "0.8"
 
 [dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 
 [dependencies.filetime]
 version = "0.2"
@@ -137,7 +137,7 @@ version = "2.11"
 version = "0.3"
 
 [dependencies.gzp]
-version = "0.11.3"
+version = "1.0"
 features = ["deflate_rust"]
 default-features = false
 
@@ -165,7 +165,7 @@ features = [
 optional = true
 
 [dependencies.itertools]
-version = "0.12"
+version = "0.13"
 
 [dependencies.jobserver]
 version = "0.1"
@@ -211,7 +211,7 @@ version = "1.16"
 version = "0.4"
 
 [dependencies.object]
-version = "0.32"
+version = "0.36"
 
 [dependencies.once_cell]
 version = "1.19"
@@ -333,7 +333,7 @@ version = "6"
 default-features = false
 
 [dependencies.zip]
-version = "0.6"
+version = "2.5"
 default-features = false
 
 [dependencies.zstd]
