--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,17 +44,17 @@
 name = "yaml_rust2"
 path = "src/lib.rs"
 
-[[bin]]
-name = "dump_events"
-path = "tools/dump_events.rs"
+#[[bin]]
+#name = "dump_events"
+#path = "tools/dump_events.rs"
 
-[[bin]]
-name = "run_bench"
-path = "tools/run_bench.rs"
+#[[bin]]
+#name = "run_bench"
+#path = "tools/run_bench.rs"
 
-[[bin]]
-name = "time_parse"
-path = "tools/time_parse.rs"
+#[[bin]]
+#name = "time_parse"
+#path = "tools/time_parse.rs"
 
 [[example]]
 name = "dump_yaml"
