--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@
 optional = true
 
 [dev-dependencies.rustix]
-version = "0.38"
+version = "1"
 features = ["event"]
 
 [features]
--- a/src/context.rs
+++ b/src/context.rs
@@ -385,7 +385,7 @@
     /// let mut input = Libinput::new_with_udev(Interface);
     /// input.udev_assign_seat("seat0").unwrap();
     ///
-    /// while poll(&mut [PollFd::new(&input, PollFlags::IN)], -1).is_ok() {
+    /// while poll(&mut [PollFd::new(&input, PollFlags::IN)], None).is_ok() {
     ///     input.dispatch().unwrap();
     ///     for event in &mut input {
     ///         // do some processing...
