--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,5 +39,5 @@
 
-[[bench]]
-name = "benchmarks"
-harness = false
+#[[bench]]
+#name = "benchmarks"
+#harness = false
 
@@ -91,4 +91,4 @@
 
-[dev-dependencies.criterion]
-version = "0.5.1"
+#[dev-dependencies.criterion]
+#version = "0.5.1"
 
--- a/benches/benchmarks.rs
+++ b/benches/benchmarks.rs
@@ -1,2 +1,2 @@
-use serde::{Deserialize, Serialize};
+/*use serde::{Deserialize, Serialize};
 #[cfg(feature = "serde_bytes")]
@@ -147,2 +147,2 @@
 criterion_group!(benches, big_array_ser_and_de, fixed_size_array);
-criterion_main!(benches);
+criterion_main!(benches);*/
