--- a/Cargo.toml
+++ b/Cargo.toml
@@ -159,11 +159,6 @@
 name = "value_value"
 path = "tests/value_value.rs"
 
-[[bench]]
-name = "benchmarks"
-path = "benches/benchmarks.rs"
-harness = false
-
 [dependencies.arrayvec]
 version = "0.7.4"
 features = ["serde"]
@@ -233,9 +228,6 @@
 ]
 default-features = false
 
-[dev-dependencies.criterion]
-version = "0.7.0"
-
 [dev-dependencies.rand]
 version = ">= 0.8"
 
--- a/benches/benchmarks.rs
+++ b/benches/benchmarks.rs
@@ -1,4 +1,4 @@
-use serde::{Deserialize, Serialize};
+/*use serde::{Deserialize, Serialize};
 #[cfg(feature = "serde_bytes")]
 use serde_bytes::ByteBuf;
 use std::{collections::HashMap, hint::black_box, vec};
@@ -205,4 +205,4 @@
     signature_parse,
     object_path_parse
 );
-criterion_main!(benches);
+criterion_main!(benches); */
