Index: zvariant/Cargo.toml
===================================================================
--- zvariant.orig/Cargo.toml
+++ zvariant/Cargo.toml
@@ -11,7 +11,7 @@
 
 [package]
 edition = "2021"
-rust-version = "1.75"
+rust-version = "1.70"
 name = "zvariant"
 version = "4.0.0"
 authors = ["Zeeshan Ali Khan <zeeshanak@gnome.org>"]
@@ -37,9 +37,6 @@ all-features = true
 [lib]
 bench = false
 
-[[bench]]
-name = "benchmarks"
-harness = false
 
 [dependencies.arrayvec]
 version = "0.7.2"
@@ -93,7 +90,7 @@ version = "=4.0.0"
 version = "0.5"
 
 [dev-dependencies.glib]
-version = "0.18"
+version = "0.19"
 
 [dev-dependencies.rand]
 version = "0.8.5"
