--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,14 +65,14 @@
 features = ["derive"]
 
 [dependencies.serde_bytes]
-version = "0.11.14"
+version = "0.11"
 optional = true
 
 [dependencies.static_assertions]
 version = "1.1.0"
 
 [dependencies.time]
-version = "0.3.36"
+version = "0.3"
 features = ["serde"]
 optional = true
 
@@ -82,7 +82,7 @@
 optional = true
 
 [dependencies.uuid]
-version = "1.8.0"
+version = "1.6"
 features = ["serde"]
 optional = true
 
