--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,3 +67,3 @@
 [dependencies.serde_bytes]
-version = "0.11.14"
+version = "0.11"
 optional = true
@@ -74,3 +74,3 @@
 [dependencies.time]
-version = "0.3.36"
+version = "0.3"
 features = ["serde"]
@@ -84,3 +84,3 @@
 [dependencies.uuid]
-version = "1.8.0"
+version = "1.6"
 features = ["serde"]
@@ -104,3 +104,3 @@
 [dev-dependencies.serde_repr]
-version = "0.1.19"
+version = "0.1"
 
