Index: wayland-client/Cargo.toml
===================================================================
--- wayland-client.orig/Cargo.toml
+++ wayland-client/Cargo.toml
@@ -48,14 +48,2 @@ path = "src/lib.rs"
 
-[[example]]
-name = "list_globals"
-path = "examples/list_globals.rs"
-
-[[example]]
-name = "list_globals_no_dispatch"
-path = "examples/list_globals_no_dispatch.rs"
-
-[[example]]
-name = "simple_window"
-path = "examples/simple_window.rs"
-
 [dependencies.bitflags]
