--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,6 +29,8 @@ categories = [
 license = "MIT"
 repository = "https://github.com/smithay/wayland-rs"
 
+autoexamples = false
+
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = [
@@ -40,7 +42,7 @@ rustdoc-args = [
 version = "1.2"
 
 [dependencies.calloop]
-version = "0.10.5"
+version = "0.10"
 optional = true
 
 [dependencies.log]
@@ -58,10 +60,10 @@ version = "0.1.0"
 version = "0.30.0"
 
 [dev-dependencies.futures-channel]
-version = "0.3.16"
+version = "0.3"
 
 [dev-dependencies.futures-util]
 version = "0.3"
 
 [dev-dependencies.tempfile]
-version = "3.2"
+version = "3.3"
