--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,5 +36,2 @@
 
-[dependencies.advisory-lock]
-version = "0.3.0"
-
 [dependencies.anyhow]
@@ -43,3 +40,3 @@
 [dependencies.async-stream]
-version = "0.3.6"
+version = "0.3"
 
@@ -66,6 +63,6 @@
 [dependencies.colored]
-version = "3"
+version = "2"
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
@@ -81,11 +78,3 @@
 [dependencies.getrandom]
-version = "0.3"
-
-[dependencies.gix]
-version = "0.70"
-optional = true
-
-[dependencies.gix-object]
-version = "0.47"
-optional = true
+version = "0.2"
 
@@ -98,9 +87,3 @@
 [dependencies.ipnetwork]
-version = "0.21"
-
-[dependencies.irc]
-version = "1"
-features = ["tls-rust"]
-optional = true
-default-features = false
+version = "0.17"
 
@@ -110,3 +93,3 @@
 [dependencies.lru]
-version = "0.13"
+version = "0.12"
 
@@ -119,3 +102,3 @@
 [dependencies.num-format]
-version = "0.4.4"
+version = "0.4"
 
@@ -131,9 +114,2 @@
 version = "1.21"
-features = [
-    "compression",
-    "crypto-rust",
-    "allow-experimental-crypto",
-    "allow-variable-time-crypto",
-]
-default-features = false
 
@@ -175,8 +151,2 @@
 default = [
-    "git",
-    "irc",
-]
-git = [
-    "gix",
-    "gix-object",
 ]
