Index: apt-swarm/Cargo.toml
===================================================================
--- apt-swarm.orig/Cargo.toml
+++ apt-swarm/Cargo.toml
@@ -28,13 +28,8 @@ repository = "https://github.com/kpcyrd/
 
 [features]
 default = [
-    "git",
     "irc",
 ]
-git = [
-    "gix",
-    "gix-object",
-]
 
 [lib]
 name = "apt_swarm"
@@ -71,10 +66,6 @@ features = ["serde"]
 [dependencies.bytes]
 version = "1.4.0"
 
-[dependencies.bzip2]
-version = "0.5.1"
-features = ["libbz2-rs-sys"]
-
 [dependencies.chrono]
 version = "0.4"
 features = [
@@ -94,10 +85,10 @@ features = [
 version = "4.0.7"
 
 [dependencies.colored]
-version = "3"
+version = "2"
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
 [dependencies.env_logger]
 version = "0.11"
@@ -109,15 +100,7 @@ version = "2"
 version = "0.3.26"
 
 [dependencies.getrandom]
-version = "0.3"
-
-[dependencies.gix]
-version = "0.70"
-optional = true
-
-[dependencies.gix-object]
-version = "0.47"
-optional = true
+version = "0.2"
 
 [dependencies.hex]
 version = "0.4.3"
@@ -126,10 +109,10 @@ version = "0.4.3"
 version = "2"
 
 [dependencies.ipnetwork]
-version = "0.21"
+version = "0.17"
 
 [dependencies.irc]
-version = "1"
+version = "1.1"
 features = ["tls-rust"]
 optional = true
 default-features = false
@@ -138,7 +121,7 @@ default-features = false
 version = "0.4.17"
 
 [dependencies.lru]
-version = "0.13"
+version = "0.12"
 
 [dependencies.lz4_flex]
 version = "0.11.3"
@@ -150,7 +133,7 @@ version = "2.5.0"
 version = "0.4"
 
 [dependencies.reqwest]
-version = "0.12"
+version = "0.12.15"
 features = [
     "rustls-tls",
     "socks",
@@ -164,13 +147,6 @@ default-features = false
 
 [dependencies.sequoia-openpgp]
 version = "1.21"
-features = [
-    "compression",
-    "crypto-rust",
-    "allow-experimental-crypto",
-    "allow-variable-time-crypto",
-]
-default-features = false
 
 [dependencies.serde]
 version = "1.0.152"
@@ -209,9 +185,6 @@ version = "2.5"
 [dev-dependencies.tempfile]
 version = "3.4.0"
 
-[target.'cfg(target_os = "openbsd")'.dependencies.pledge]
-version = "0.4.2"
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
