--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,3 +43,3 @@
 [dependencies.async-stream]
-version = "0.3.6"
+version = "0.3"
 
@@ -55,6 +55,2 @@
 
-[dependencies.bzip2]
-version = "0.5.1"
-features = ["libbz2-rs-sys"]
-
 [dependencies.chrono]
@@ -78,6 +74,6 @@
 [dependencies.colored]
-version = "3"
+version = "2"
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
@@ -93,11 +89,3 @@
 [dependencies.getrandom]
-version = "0.3"
-
-[dependencies.gix]
-version = "0.70"
-optional = true
-
-[dependencies.gix-object]
-version = "0.47"
-optional = true
+version = "0.2"
 
@@ -110,3 +98,3 @@
 [dependencies.ipnetwork]
-version = "0.21"
+version = "0.17"
 
@@ -122,3 +110,3 @@
 [dependencies.lru]
-version = "0.13"
+version = "0.12"
 
@@ -131,3 +119,3 @@
 [dependencies.num-format]
-version = "0.4.4"
+version = "0.4"
 
@@ -143,9 +131,2 @@
 version = "1.21"
-features = [
-    "compression",
-    "crypto-rust",
-    "allow-experimental-crypto",
-    "allow-variable-time-crypto",
-]
-default-features = false
 
@@ -190,9 +171,4 @@
 default = [
-    "git",
     "irc",
 ]
-git = [
-    "gix",
-    "gix-object",
-]
 
