--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,9 +30,4 @@
 default = [
-    "git",
     "irc",
 ]
-git = [
-    "gix",
-    "gix-object",
-]
 
@@ -73,6 +68,2 @@
 
-[dependencies.bzip2]
-version = "0.5.1"
-features = ["libbz2-rs-sys"]
-
 [dependencies.chrono]
@@ -96,6 +87,6 @@
 [dependencies.colored]
-version = "3"
+version = "2"
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
@@ -111,11 +102,3 @@
 [dependencies.getrandom]
-version = "0.3"
-
-[dependencies.gix]
-version = "0.70"
-optional = true
-
-[dependencies.gix-object]
-version = "0.47"
-optional = true
+version = "0.2"
 
@@ -128,3 +111,3 @@
 [dependencies.ipnetwork]
-version = "0.21"
+version = "0.17"
 
@@ -140,3 +123,3 @@
 [dependencies.lru]
-version = "0.13"
+version = "0.12"
 
@@ -162,9 +145,2 @@
 version = "1.21"
-features = [
-    "compression",
-    "crypto-rust",
-    "allow-experimental-crypto",
-    "allow-variable-time-crypto",
-]
-default-features = false
 
@@ -207,5 +183,2 @@
 
-[target.'cfg(target_os = "openbsd")'.dependencies.pledge]
-version = "0.4.2"
-
 [lints.rust.unexpected_cfgs]
