--- a/Cargo.toml
+++ b/Cargo.toml
@@ -81,7 +81,7 @@
 version = "0.24"
 
 [dependencies.image_hasher]
-version = "2.0"
+version = "1"
 
 [dependencies.log]
 version = "0.4.22"
--- a/src/help_combo_box.rs
+++ b/src/help_combo_box.rs
@@ -137,10 +137,10 @@
         eng_name: "DoubleGradient",
         hash_alg: HashAlg::DoubleGradient,
     },
-    ImageHashTypeStruct {
-        eng_name: "Median",
-        hash_alg: HashAlg::Median,
-    },
+    // ImageHashTypeStruct {
+    //     eng_name: "Median",
+    //     hash_alg: HashAlg::Median,
+    // },
 ];
 
 pub const IMAGES_HASH_SIZE_COMBO_BOX: [i32; 4] = [8, 16, 32, 64];
--- a/src/connect_things/connect_settings.rs
+++ b/src/connect_things/connect_settings.rs
@@ -182,7 +182,7 @@
                                     HashAlg::DoubleGradient,
                                     HashAlg::VertGradient,
                                     HashAlg::Mean,
-                                    HashAlg::Median,
+                                    // HashAlg::Median,
                                 ] {
                                     let (mut messages, loaded_items) = load_cache_from_file_generalized_by_path::<czkawka_core::similar_images::ImagesEntry>(
                                         &get_similar_images_cache_file(hash_size, hash_alg, image_filter),
