Index: object/tests/read/coff.rs
===================================================================
--- object.orig/tests/read/coff.rs
+++ object/tests/read/coff.rs
@@ -2,7 +2,7 @@ use object::{pe, read, Object, ObjectSec
 use std::fs;
 use std::path::PathBuf;
 
-#[cfg(feature = "coff")]
+/*#[cfg(feature = "coff")]
 #[test]
 fn coff_extended_relocations() {
     let path_to_obj: PathBuf = ["testfiles", "coff", "relocs_overflow.o"].iter().collect();
@@ -20,4 +20,4 @@ fn coff_extended_relocations() {
     };
     let relocations = code_section.relocations().collect::<Vec<_>>();
     assert_eq!(relocations.len(), 65536);
-}
+}*/
Index: object/tests/read/elf.rs
===================================================================
--- object.orig/tests/read/elf.rs
+++ object/tests/read/elf.rs
@@ -12,7 +12,7 @@ fn get_buildid(path: &Path) -> Result<Op
         .map(|option| option.map(ToOwned::to_owned))
 }
 
-#[cfg(feature = "std")]
+/*#[cfg(feature = "std")]
 #[test]
 /// Regression test: used to attempt to allocate 5644418395173552131 bytes
 fn get_buildid_bad_elf() {
@@ -44,4 +44,4 @@ fn get_buildid_less_bad_elf() {
         buildid,
         b"\xf9\xc0\xc6\x05\xd3\x76\xbb\xa5\x7e\x02\xf5\x74\x50\x9d\x16\xcc\xe9\x9c\x1b\xf1"
     );
-}
+}*/
Index: object/tests/read/macho.rs
===================================================================
--- object.orig/tests/read/macho.rs
+++ object/tests/read/macho.rs
@@ -3,7 +3,7 @@ use object::{Object, ObjectSection as _}
 
 // Test that we can read compressed sections in Mach-O files as produced
 // by the Go compiler.
-#[cfg(feature = "std")]
+/* #[cfg(feature = "std")]
 #[test]
 fn test_go_macho() {
     let macho_testfiles = std::path::Path::new("testfiles/macho");
@@ -46,4 +46,4 @@ fn test_go_macho() {
             }
         }
     }
-}
+}*/
