Index: object/Cargo.toml
===================================================================
--- object.orig/Cargo.toml
+++ object/Cargo.toml
@@ -56,4 +56,8 @@ path = "tests/integration.rs"
 name = "parse_self"
 path = "tests/parse_self.rs"
 
+[dependencies.ahash]
+version = "0.8"
+default-features = false
+optional = true
 
@@ -71,10 +75,8 @@ version = "1.0.13"
 optional = true
 
 [dependencies.hashbrown]
-version = "0.15.0"
-features = ["default-hasher"]
+version = ">= 0.14.0, < 0.16"
 optional = true
-default-features = false
 
 [dependencies.indexmap]
 version = "2.0"
@@ -164,6 +166,7 @@ write_core = [
     "dep:crc32fast",
     "dep:indexmap",
     "dep:hashbrown",
+    "dep:ahash",
 ]
 write_std = [
     "write_core",
Index: object/src/write/string.rs
===================================================================
--- object.orig/src/write/string.rs
+++ object/src/write/string.rs
@@ -3,7 +3,7 @@ use alloc::vec::Vec;
 #[cfg(feature = "write_std")]
 type IndexSet<K> = indexmap::IndexSet<K>;
 #[cfg(not(feature = "write_std"))]
-type IndexSet<K> = indexmap::IndexSet<K, hashbrown::DefaultHashBuilder>;
+type IndexSet<K> = indexmap::IndexSet<K,core::hash::BuildHasherDefault<ahash::AHasher>>;
 
 /// An identifier for an entry in a string table.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
