Index: object/Cargo.toml
===================================================================
--- object.orig/Cargo.toml
+++ object/Cargo.toml
@@ -56,19 +56,6 @@ path = "tests/integration.rs"
 name = "parse_self"
 path = "tests/parse_self.rs"
 
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
-[dependencies.compiler_builtins]
-version = "0.1.2"
-optional = true
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
 
 [dependencies.crc32fast]
 version = "1.2"
@@ -154,12 +141,6 @@ read = [
     "unaligned",
 ]
 read_core = []
-rustc-dep-of-std = [
-    "core",
-    "compiler_builtins",
-    "alloc",
-    "memchr/rustc-dep-of-std",
-]
 std = ["memchr/std"]
 unaligned = []
 unstable = []
