Index: object/Cargo.toml
===================================================================
--- object.orig/Cargo.toml
+++ object/Cargo.toml
@@ -98,11 +98,6 @@ default-features = false
 version = "0.7.0"
 optional = true
 
-[dependencies.wasmparser]
-version = "0.218.0"
-optional = true
-default-features = false
-
 [features]
 all = [
     "read",
@@ -110,7 +105,6 @@ all = [
     "build",
     "std",
     "compression",
-    "wasm",
 ]
 archive = []
 build = [
@@ -144,7 +138,6 @@ doc = [
     "elf",
     "macho",
     "pe",
-    "wasm",
     "xcoff",
 ]
 elf = []
@@ -174,7 +167,6 @@ unstable-all = [
     "all",
     "unstable",
 ]
-wasm = ["dep:wasmparser"]
 write = [
     "write_std",
     "coff",
