--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,7 +65,3 @@
 ]
 static = []
 stock-zlib = []
-zlib-ng = [
-    "libc",
-    "cmake",
-]
--- a/build.rs
+++ b/build.rs
@@ -10,7 +10,7 @@
 
     let host_and_target_contain = |s| host.contains(s) && target.contains(s);
 
-    let want_ng = cfg!(feature = "zlib-ng") && !cfg!(feature = "stock-zlib");
+    let want_ng = false; cfg!(feature = "zlib-ng") && !cfg!(feature = "stock-zlib");
 
     if want_ng && target != "wasm32-unknown-unknown" {
         return build_zlib_ng(&target, true);
