Index: prodash/Cargo.toml
===================================================================
--- prodash.orig/Cargo.toml
+++ prodash/Cargo.toml
@@ -83,7 +83,7 @@ version = "2.0.0"
 optional = true
 
 [dependencies.is-terminal]
-version = "0.4.9"
+version = "0.4"
 optional = true
 
 [dependencies.log]
@@ -111,7 +111,7 @@ optional = true
 default-features = false
 
 [dependencies.tui]
-version = "0.20.1"
+version = "0.23"
 optional = true
 default-features = false
 package = "ratatui"
@@ -157,7 +157,7 @@ version = "0.3.4"
 default-features = false
 
 [dev-dependencies.is-terminal]
-version = "0.4.9"
+version = "0.4"
 
 [dev-dependencies.once_cell]
 version = "1.4.0"
