--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,6 +23,9 @@ keywords = ["ffi", "pipewire", "multimed
 categories = ["api-bindings", "multimedia"]
 license = "MIT"
 repository = "https://gitlab.freedesktop.org/pipewire/pipewire-rs"
+[lib]
+doctest = false
+
 [package.metadata.system-deps.libpipewire]
 name = "libpipewire-0.3"
 version = "0.3"
@@ -30,7 +33,7 @@ version = "0.3"
 version = "0"
 package = "libspa-sys"
 [build-dependencies.bindgen]
-version = "0.59"
+version = "0.60"
 
 [build-dependencies.system-deps]
 version = "6"
