Index: resvg-0.43.0/examples/draw_bboxes.rs
===================================================================
--- resvg-0.43.0.orig/examples/draw_bboxes.rs	2006-07-24 03:21:28.000000000 +0200
+++ resvg-0.43.0/examples/draw_bboxes.rs	2024-08-11 13:32:57.277752805 +0200
@@ -1,3 +1,7 @@
+#[cfg(not(all(feature = "text", any(feature = "memmap-fonts", feature = "system-fonts"))))]
+fn main() {}
+
+#[cfg(all(feature = "text", any(feature = "memmap-fonts", feature = "system-fonts")))]
 fn main() {
     let args: Vec<String> = std::env::args().collect();
     if !(args.len() == 3 || args.len() == 5) {
Index: resvg-0.43.0/examples/minimal.rs
===================================================================
--- resvg-0.43.0.orig/examples/minimal.rs	2006-07-24 03:21:28.000000000 +0200
+++ resvg-0.43.0/examples/minimal.rs	2024-08-11 13:32:46.366717256 +0200
@@ -1,3 +1,7 @@
+#[cfg(not(all(feature = "text", any(feature = "memmap-fonts", feature = "system-fonts"))))]
+fn main() {}
+
+#[cfg(all(feature = "text", any(feature = "memmap-fonts", feature = "system-fonts")))]
 fn main() {
     let args: Vec<String> = std::env::args().collect();
     if args.len() != 3 {
