--- a/src/read.rs
+++ b/src/read.rs
@@ -57,2 +57,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_dylib() {
@@ -64,2 +65,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_exe() {
@@ -75,2 +77,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_ar() {
@@ -82,2 +85,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_not_fat() {
@@ -95,2 +99,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_extract_dylib() {
@@ -107,2 +112,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_extract_exe() {
@@ -119,2 +125,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_reader_extract_ar() {
--- a/src/write.rs
+++ b/src/write.rs
@@ -403,2 +403,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_writer_add_exe() {
@@ -419,2 +420,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_writer_add_duplicated_arch() {
@@ -427,2 +429,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_writer_add_fat() {
@@ -436,2 +439,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_writer_add_archive() {
@@ -453,2 +457,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_writer_add_llvm_bitcode() {
@@ -469,2 +474,3 @@
     #[test]
+    #[cfg(any())]
     fn test_fat_writer_remove() {
