Index: starship/Cargo.toml
===================================================================
--- starship.orig/Cargo.toml
+++ starship/Cargo.toml
@@ -40,10 +40,6 @@ features = ["tls", "form"]
 optional = true
 default-features = false
 
-[dependencies.battery]
-version = "0.7.5"
-optional = true
-
 [dependencies.byte-unit]
 version = "3.0.3"
 
@@ -131,7 +127,7 @@ version = "0.5.2"
 version = "3.1.0"
 
 [features]
-default = ["battery", "http"]
+#default = ["battery", "http"]
 http = ["attohttpc"]
 tls-vendored = ["native-tls/vendored"]
 [badges.is-it-maintained-issue-resolution]
