Index: starship/Cargo.toml
===================================================================
--- starship.orig/Cargo.toml
+++ starship/Cargo.toml
@@ -34,10 +34,6 @@ path = "src/main.rs"
 [dependencies.ansi_term]
 version = "0.12.1"
 
-[dependencies.battery]
-version = "0.7.5"
-optional = true
-
 [dependencies.byte-unit]
 version = "3.0.3"
 
@@ -120,7 +116,6 @@ version = "0.4"
 version = "3.1.0"
 
 [features]
-default = ["battery"]
 [badges.is-it-maintained-issue-resolution]
 repository = "starship/starship"
 
