--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,5 +23,4 @@
 [dependencies.shell-words]
 version = "1.1.0"
 
-[dev-dependencies.env-lock]
-version = "0.1.0"
+
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -259,6 +259,7 @@
     use std::{ffi::OsStr, path::PathBuf};
 
     /// Test loading from a static source that overrides the environment
+    /*
     #[test]
     fn source_priority() {
         let builder = {
@@ -273,9 +274,10 @@
                 .source(Some("default"))
         };
         assert_cmd(builder, "priority", &[]);
-    }
+    }*/
 
     /// Test loading from the `VISUAL` env var
+    /*
     #[test]
     fn source_visual() {
         let builder = {
@@ -286,9 +288,10 @@
             EditorBuilder::new().environment().source(Some("default"))
         };
         assert_cmd(builder, "visual", &[]);
-    }
+    }*/
 
     /// Test loading from the `EDITOR` env var
+    /*
     #[test]
     fn source_editor() {
         let builder = {
@@ -299,9 +302,10 @@
             EditorBuilder::new().environment().source(Some("default"))
         };
         assert_cmd(builder, "editor", &[]);
-    }
+    }*/
 
     /// Test loading from a fallback value, with lower precedence than the env
+    /*
     #[test]
     fn source_default() {
         let builder = {
@@ -312,7 +316,7 @@
             EditorBuilder::new().environment().source(Some("default"))
         };
         assert_cmd(builder, "default", &[]);
-    }
+    }*/
 
     /// Test included paths as extra arguments
     #[test]
@@ -338,6 +342,7 @@
     }
 
     /// Test when all options are undefined
+    /*
     #[test]
     fn error_no_command() {
         let _guard = env_lock::lock_env([
@@ -348,7 +353,7 @@
             EditorBuilder::new().environment().source(None::<&str>),
             "Edit command not defined in any of the listed sources",
         );
-    }
+    } */
 
     /// Test when the command exists but is the empty string
     #[test]
