Description: Revert
 https://gitlab.gnome.org/GNOME/glycin/-/commit/884771235437d2
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,7 +57,7 @@
 package = "memmap2"
 
 [dependencies.nix]
-version = "0.30.0"
+version = "0.29.0"
 features = [
     "fs",
     "resource",
--- a/src/shared_memory.rs
+++ b/src/shared_memory.rs
@@ -1,3 +1,4 @@
+use std::ffi::CString;
 use std::ops::{Deref, DerefMut};
 use std::os::fd::{AsRawFd, OwnedFd};
 
@@ -12,8 +13,9 @@
 impl SharedMemory {
     pub fn new(size: u64) -> Result<Self, Error> {
         let memfd = nix::sys::memfd::memfd_create(
-            c"glycin-frame",
-            nix::sys::memfd::MFdFlags::MFD_CLOEXEC | nix::sys::memfd::MFdFlags::MFD_ALLOW_SEALING,
+            &CString::new("glycin-frame").unwrap(),
+            nix::sys::memfd::MemFdCreateFlag::MFD_CLOEXEC
+                | nix::sys::memfd::MemFdCreateFlag::MFD_ALLOW_SEALING,
         )
         .expect("Failed to create memfd");
 
