Index: cookie-store/Cargo.toml
===================================================================
--- cookie-store.orig/Cargo.toml
+++ cookie-store/Cargo.toml
@@ -65,7 +65,7 @@ version = "1.0.147"
 version = "1.0.87"
 
 [dependencies.time]
-version = "0.3.16"
+version = "=0.3.9"
 
 [dependencies.url]
 version = "2.3.1"
@@ -75,4 +75,3 @@ default = ["public_suffix"]
 log_secure_cookie_values = []
 preserve_order = ["indexmap"]
 public_suffix = ["publicsuffix"]
-wasm-bindgen = ["time/wasm-bindgen"]
