Index: cookie-store/Cargo.toml
===================================================================
--- cookie-store.orig/Cargo.toml
+++ cookie-store/Cargo.toml
@@ -38,14 +38,14 @@ rustdoc-args = [
 ]
 
 [dependencies.cookie]
-version = "0.16.1"
+version = ">= 0.16.1, < 1.0"
 features = ["percent-encode"]
 
 [dependencies.idna]
-version = "0.3.0"
+version = ">= 0.3.0, < 0.5"
 
 [dependencies.indexmap]
-version = "1.9.1"
+version = ">= 1.9.1"
 optional = true
 
 [dependencies.log]
