Index: b/Cargo.toml
===================================================================
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,7 +68,7 @@
 bench = false
 
 [dependencies.serde]
-version = "1.0.228"
+version = "1.0.219"
 optional = true
 default-features = false
 
@@ -78,14 +78,14 @@
 default-features = false
 
 [dev-dependencies.criterion]
-version = "0.7.0"
+version = "0.5.1"
 features = ["html_reports"]
 
 [dev-dependencies.pretty_assertions]
 version = "1.4.1"
 
 [dev-dependencies.serde_json]
-version = "1.0.145"
+version = "1.0.143"
 
 [dev-dependencies.static_assertions]
 version = "1.1.0"
