Index: findutils/src/find/matchers/mod.rs
===================================================================
--- findutils.orig/src/find/matchers/mod.rs
+++ findutils/src/find/matchers/mod.rs
@@ -1675,11 +1675,11 @@ mod tests {
         assert!(file.is_ok());
 
         // test error when file no permission
-        #[cfg(unix)]
+        /*#[cfg(unix)]
         {
             let result = get_or_create_file("/etc/shadow");
             assert!(result.is_err());
-        }
+        }*/
 
         let _ = fs::remove_file("test_data/get_or_create_file_test");
     }
Index: findutils/src/find/mod.rs
===================================================================
--- findutils.orig/src/find/mod.rs
+++ findutils/src/find/mod.rs
@@ -1069,7 +1069,7 @@ mod tests {
         }
     }
 
-    #[test]
+    /*#[test]
     #[cfg(target_os = "linux")]
     fn test_no_permission_file_error() {
         use std::{path::Path, process::Command};
@@ -1099,7 +1099,7 @@ mod tests {
                 .output()
                 .expect("cannot set file permission");
         }
-    }
+    }*/
 
     #[test]
     #[cfg(target_os = "linux")]
