Index: sniffglue/src/lib.rs
===================================================================
--- sniffglue.orig/src/lib.rs
+++ sniffglue/src/lib.rs
@@ -67,7 +67,7 @@ mod tests {
             },
             IPv4(IPv4Header {
                 version: 4,
-                ihl: 20,
+                ihl: 5,
                 tos: 0,
                 length: 442,
                 id: 20073,
Index: sniffglue/src/centrifuge/sll.rs
===================================================================
--- sniffglue.orig/src/centrifuge/sll.rs
+++ sniffglue/src/centrifuge/sll.rs
@@ -55,7 +55,7 @@ mod tests {
             Ether::IPv4(
                 IPv4Header {
                     version: 4,
-                    ihl: 20,
+                    ihl: 5,
                     tos: 0,
                     length: 94,
                     id: 63161,
Index: sniffglue/Cargo.toml
===================================================================
--- sniffglue.orig/Cargo.toml
+++ sniffglue/Cargo.toml
@@ -75,7 +75,7 @@ version = "1.6"
 version = "0.1.3"
 
 [dependencies.pktparse]
-version = "0.7"
+version = "0.7.1"
 features = ["serde"]
 
 [dependencies.reduce]
