--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,7 +24,7 @@
 [lib]
 doc = false
 [dependencies.ansi_term]
-version = "0.11"
+version = "0.12"
 
 [dependencies.atty]
 version = "0.2"
@@ -48,7 +48,7 @@
 version = "0.4"
 
 [dependencies.nix]
-version = "0.14"
+version = "0.16"
 
 [dependencies.nom]
 version = "4.0"
@@ -79,13 +79,13 @@
 version = "0.8"
 
 [dependencies.structopt]
-version = "0.2"
+version = "0.3"
 
 [dependencies.threadpool]
 version = "1.7"
 
 [dependencies.tls-parser]
-version = "0.7"
+version = "0.8"
 
 [dependencies.toml]
 version = "0.5"
