Index: sniffglue/Cargo.toml
===================================================================
--- sniffglue.orig/Cargo.toml
+++ sniffglue/Cargo.toml
@@ -39,10 +39,10 @@ version = "1"
 version = "0.2"
 
 [dependencies.base64]
-version = "0.13"
+version = "0.21"
 
 [dependencies.bstr]
-version = "0.2.16"
+version = "1.0"
 
 [dependencies.dhcp4r]
 version = "0.2.0"
@@ -54,7 +54,7 @@ version = "2.0"
 version = "0.8"
 
 [dependencies.env_logger]
-version = "0.9"
+version = "0.10"
 
 [dependencies.libc]
 version = "0.2"
@@ -63,7 +63,7 @@ version = "0.2"
 version = "0.4"
 
 [dependencies.nix]
-version = "0.23"
+version = "0.26"
 
 [dependencies.nom]
 version = "7"
@@ -104,7 +104,8 @@ version = "0.5"
 
 [dependencies.users]
 version = "0.11"
+package = "uzers"
 [target."cfg(target_os=\"linux\")".dependencies.syscallz]
-version = "0.16"
+version = "0.17"
 [badges.travis-ci]
 repository = "kpcyrd/sniffglue"
