Description: No dependency on Windows
Author: John Goerzen <jgoerzen@complete.org>
Last-Update: 2023-06-13 mjt@tls.msk.ru

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,3 +51,2 @@
     "hermit-abi",
-    "windows-sys",
 ]
@@ -94,11 +93 @@
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_Networking_WinSock",
-    "Win32_Security",
-    "Win32_System_IO",
-]
-optional = true
