From: =?UTF-8?q?Jeremy=20B=C3=ADcha?= <jbicha@ubuntu.com>
Date: Sat, 20 Sep 2025 10:48:10 -0400
Subject: Revert symbol additition from gstreamer1.0 1.27

which isn't in Debian Unstable yet
to allow the cross_validate_constants_with_c test
to keep passing
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -732,18 +732,6 @@
 pub const GST_LOCK_FLAG_EXCLUSIVE: GstLockFlags = 4;
 pub const GST_LOCK_FLAG_LAST: GstLockFlags = 256;
 
-pub type GstLogContextFlags = c_uint;
-pub const GST_LOG_CONTEXT_FLAG_NONE: GstLogContextFlags = 0;
-pub const GST_LOG_CONTEXT_FLAG_THROTTLE: GstLogContextFlags = 1;
-
-pub type GstLogContextHashFlags = c_uint;
-pub const GST_LOG_CONTEXT_DEFAULT: GstLogContextHashFlags = 0;
-pub const GST_LOG_CONTEXT_IGNORE_OBJECT: GstLogContextHashFlags = 1;
-pub const GST_LOG_CONTEXT_IGNORE_FORMAT: GstLogContextHashFlags = 2;
-pub const GST_LOG_CONTEXT_IGNORE_FILE: GstLogContextHashFlags = 4;
-pub const GST_LOG_CONTEXT_USE_LINE_NUMBER: GstLogContextHashFlags = 8;
-pub const GST_LOG_CONTEXT_USE_STRING_ARGS: GstLogContextHashFlags = 16;
-
 pub type GstMapFlags = c_uint;
 pub const GST_MAP_READ: GstMapFlags = 1;
 pub const GST_MAP_WRITE: GstMapFlags = 2;
@@ -2047,24 +2035,6 @@
     }
 }
 
-#[repr(C)]
-#[allow(dead_code)]
-pub struct _GstLogContext {
-    _data: [u8; 0],
-    _marker: core::marker::PhantomData<(*mut u8, core::marker::PhantomPinned)>,
-}
-
-pub type GstLogContext = _GstLogContext;
-
-#[repr(C)]
-#[allow(dead_code)]
-pub struct _GstLogContextBuilder {
-    _data: [u8; 0],
-    _marker: core::marker::PhantomData<(*mut u8, core::marker::PhantomPinned)>,
-}
-
-pub type GstLogContextBuilder = _GstLogContextBuilder;
-
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub struct GstMapInfo {
@@ -2633,7 +2663,6 @@
     pub reference: *mut GstCaps,
     pub timestamp: GstClockTime,
     pub duration: GstClockTime,
-    pub info: *mut GstStructure,
 }
 
 impl ::std::fmt::Debug for GstReferenceTimestampMeta {
@@ -2673,7 +2642,6 @@
             .field("reference", &self.reference)
             .field("timestamp", &self.timestamp)
             .field("duration", &self.duration)
-            .field("info", &self.info)
             .finish()
     }
 }
@@ -4418,20 +4386,6 @@
     //=========================================================================
     pub fn gst_lock_flags_get_type() -> GType;
 
-    //=========================================================================
-    // GstLogContextFlags
-    //=========================================================================
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_flags_get_type() -> GType;
-
-    //=========================================================================
-    // GstLogContextHashFlags
-    //=========================================================================
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_hash_flags_get_type() -> GType;
-
     //=========================================================================
     // GstMapFlags
     //=========================================================================
@@ -5596,50 +5550,6 @@
     pub fn gst_iterator_push(it: *mut GstIterator, other: *mut GstIterator);
     pub fn gst_iterator_resync(it: *mut GstIterator);
 
-    //=========================================================================
-    // GstLogContext
-    //=========================================================================
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_free(ctx: *mut GstLogContext);
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_get_category(context: *mut GstLogContext) -> *mut GstDebugCategory;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_reset(ctx: *mut GstLogContext);
-
-    //=========================================================================
-    // GstLogContextBuilder
-    //=========================================================================
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_builder_build(builder: *mut GstLogContextBuilder) -> *mut GstLogContext;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_builder_set_category(
-        builder: *mut GstLogContextBuilder,
-        category: *mut GstDebugCategory,
-    ) -> *mut GstLogContextBuilder;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_builder_set_hash_flags(
-        builder: *mut GstLogContextBuilder,
-        flags: GstLogContextHashFlags,
-    ) -> *mut GstLogContextBuilder;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_builder_set_interval(
-        builder: *mut GstLogContextBuilder,
-        interval: GstClockTime,
-    ) -> *mut GstLogContextBuilder;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_log_context_builder_new(
-        category: *mut GstDebugCategory,
-        flags: GstLogContextFlags,
-    ) -> *mut GstLogContextBuilder;
-
     //=========================================================================
     // GstMemory
     //=========================================================================
@@ -6174,12 +6084,6 @@
         api: GType,
         aggregator: GstAllocationMetaParamsAggregator,
     );
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_meta_api_type_tags_contain_only(
-        api: GType,
-        valid_tags: *mut *const c_char,
-    ) -> gboolean;
     #[cfg(feature = "v1_24")]
     #[cfg_attr(docsrs, doc(cfg(feature = "v1_24")))]
     pub fn gst_meta_deserialize(
@@ -6928,13 +6832,6 @@
         fieldname: *const c_char,
         value: *mut gboolean,
     ) -> gboolean;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_structure_get_caps(
-        structure: *const GstStructure,
-        fieldname: *const c_char,
-        caps: *mut *const GstCaps,
-    ) -> gboolean;
     pub fn gst_structure_get_clock_time(
         structure: *const GstStructure,
         fieldname: *const c_char,
@@ -7154,9 +7051,6 @@
         subset: *const GstStructure,
         superset: *const GstStructure,
     ) -> gboolean;
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_structure_is_writable(structure: *const GstStructure) -> gboolean;
     pub fn gst_structure_map_in_place(
         structure: *mut GstStructure,
         func: GstStructureMapFunc,
@@ -9709,35 +9603,9 @@
         id: *const c_char,
         message_string: *const c_char,
     );
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_debug_log_id_literal_with_context(
-        ctx: *mut GstLogContext,
-        level: GstDebugLevel,
-        file: *const c_char,
-        function: *const c_char,
-        line: c_int,
-        id: *const c_char,
-        message: *const c_char,
-    );
     //#[cfg(feature = "v1_22")]
     //#[cfg_attr(docsrs, doc(cfg(feature = "v1_22")))]
     //pub fn gst_debug_log_id_valist(category: *mut GstDebugCategory, level: GstDebugLevel, file: *const c_char, function: *const c_char, line: c_int, id: *const c_char, format: *const c_char, args: /*Unimplemented*/va_list);
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_debug_log_id_with_context(
-        ctx: *mut GstLogContext,
-        level: GstDebugLevel,
-        file: *const c_char,
-        function: *const c_char,
-        line: c_int,
-        id: *const c_char,
-        format: *const c_char,
-        ...
-    );
-    //#[cfg(feature = "v1_28")]
-    //#[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    //pub fn gst_debug_log_id_with_context_valist(ctx: *mut GstLogContext, level: GstDebugLevel, file: *const c_char, function: *const c_char, line: c_int, id: *const c_char, format: *const c_char, args: /*Unimplemented*/va_list);
     #[cfg(feature = "v1_20")]
     #[cfg_attr(docsrs, doc(cfg(feature = "v1_20")))]
     pub fn gst_debug_log_literal(
@@ -9749,33 +9617,7 @@
         object: *mut gobject::GObject,
         message_string: *const c_char,
     );
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_debug_log_literal_with_context(
-        ctx: *mut GstLogContext,
-        level: GstDebugLevel,
-        file: *const c_char,
-        function: *const c_char,
-        line: c_int,
-        object: *mut gobject::GObject,
-        message: *const c_char,
-    );
     //pub fn gst_debug_log_valist(category: *mut GstDebugCategory, level: GstDebugLevel, file: *const c_char, function: *const c_char, line: c_int, object: *mut gobject::GObject, format: *const c_char, args: /*Unimplemented*/va_list);
-    #[cfg(feature = "v1_28")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    pub fn gst_debug_log_with_context(
-        ctx: *mut GstLogContext,
-        level: GstDebugLevel,
-        file: *const c_char,
-        function: *const c_char,
-        line: c_int,
-        object: *mut gobject::GObject,
-        format: *const c_char,
-        ...
-    );
-    //#[cfg(feature = "v1_28")]
-    //#[cfg_attr(docsrs, doc(cfg(feature = "v1_28")))]
-    //pub fn gst_debug_log_with_context_valist(ctx: *mut GstLogContext, level: GstDebugLevel, file: *const c_char, function: *const c_char, line: c_int, object: *mut gobject::GObject, format: *const c_char, args: /*Unimplemented*/va_list);
     #[cfg(feature = "v1_26")]
     #[cfg_attr(docsrs, doc(cfg(feature = "v1_26")))]
     pub fn gst_debug_print_object(ptr: gconstpointer) -> *mut c_char;
--- a/tests/abi.rs
+++ b/tests/abi.rs
@@ -699,20 +699,6 @@
             alignment: align_of::<GstLockFlags>(),
         },
     ),
-    (
-        "GstLogContextFlags",
-        Layout {
-            size: size_of::<GstLogContextFlags>(),
-            alignment: align_of::<GstLogContextFlags>(),
-        },
-    ),
-    (
-        "GstLogContextHashFlags",
-        Layout {
-            size: size_of::<GstLogContextHashFlags>(),
-            alignment: align_of::<GstLogContextHashFlags>(),
-        },
-    ),
     (
         "GstMapFlags",
         Layout {
@@ -1781,14 +1767,6 @@
     ("(guint) GST_LOCK_FLAG_READ", "1"),
     ("GST_LOCK_FLAG_READWRITE", "3"),
     ("(guint) GST_LOCK_FLAG_WRITE", "2"),
-    ("(guint) GST_LOG_CONTEXT_DEFAULT", "0"),
-    ("(guint) GST_LOG_CONTEXT_FLAG_NONE", "0"),
-    ("(guint) GST_LOG_CONTEXT_FLAG_THROTTLE", "1"),
-    ("(guint) GST_LOG_CONTEXT_IGNORE_FILE", "4"),
-    ("(guint) GST_LOG_CONTEXT_IGNORE_FORMAT", "2"),
-    ("(guint) GST_LOG_CONTEXT_IGNORE_OBJECT", "1"),
-    ("(guint) GST_LOG_CONTEXT_USE_LINE_NUMBER", "8"),
-    ("(guint) GST_LOG_CONTEXT_USE_STRING_ARGS", "16"),
     ("(guint) GST_MAP_FLAG_LAST", "65536"),
     ("(guint) GST_MAP_READ", "1"),
     ("GST_MAP_READWRITE", "3"),
--- a/tests/constant.c
+++ b/tests/constant.c
@@ -298,14 +298,6 @@
     PRINT_CONSTANT((guint) GST_LOCK_FLAG_READ);
     PRINT_CONSTANT(GST_LOCK_FLAG_READWRITE);
     PRINT_CONSTANT((guint) GST_LOCK_FLAG_WRITE);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_DEFAULT);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_FLAG_NONE);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_FLAG_THROTTLE);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_IGNORE_FILE);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_IGNORE_FORMAT);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_IGNORE_OBJECT);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_USE_LINE_NUMBER);
-    PRINT_CONSTANT((guint) GST_LOG_CONTEXT_USE_STRING_ARGS);
     PRINT_CONSTANT((guint) GST_MAP_FLAG_LAST);
     PRINT_CONSTANT((guint) GST_MAP_READ);
     PRINT_CONSTANT(GST_MAP_READWRITE);
--- a/tests/layout.c
+++ b/tests/layout.c
@@ -79,8 +79,6 @@
     printf("%s;%zu;%zu\n", "GstIteratorResult", sizeof(GstIteratorResult), alignof(GstIteratorResult));
     printf("%s;%zu;%zu\n", "GstLibraryError", sizeof(GstLibraryError), alignof(GstLibraryError));
     printf("%s;%zu;%zu\n", "GstLockFlags", sizeof(GstLockFlags), alignof(GstLockFlags));
-    printf("%s;%zu;%zu\n", "GstLogContextFlags", sizeof(GstLogContextFlags), alignof(GstLogContextFlags));
-    printf("%s;%zu;%zu\n", "GstLogContextHashFlags", sizeof(GstLogContextHashFlags), alignof(GstLogContextHashFlags));
     printf("%s;%zu;%zu\n", "GstMapFlags", sizeof(GstMapFlags), alignof(GstMapFlags));
     printf("%s;%zu;%zu\n", "GstMapInfo", sizeof(GstMapInfo), alignof(GstMapInfo));
     printf("%s;%zu;%zu\n", "GstMemory", sizeof(GstMemory), alignof(GstMemory));
