Index: exa/Cargo.toml
===================================================================
--- exa.orig/Cargo.toml
+++ exa/Cargo.toml
@@ -37,27 +37,27 @@
 path = "src/bin/main.rs"
 doc = false
 [dependencies.ansi_term]
-version = "0.8.0"
+version = "0.11.0"

 [dependencies.datetime]
 version = "0.4.5"

 [dependencies.env_logger]
-version = "0.3"
+version = "0.6"

 [dependencies.getopts]
 version = "0.2.14"

 [dependencies.git2]
-version = "0.6.4"
+version = "0.8"
 optional = true
 default-features = false

 [dependencies.glob]
-version = "0.2"
+version = "0.3"

 [dependencies.lazy_static]
-version = "0.2"
+version = "1.2"

 [dependencies.libc]
 version = "0.2.9"
@@ -66,7 +66,7 @@
 version = "0.2.1"

 [dependencies.log]
-version = "0.3"
+version = "0.4"

 [dependencies.natord]
 version = "1.0.7"
@@ -90,7 +90,7 @@
 version = "0.1.4"

 [dependencies.users]
-version = "0.5.2"
+version = "0.9"

 [dependencies.zoneinfo_compiled]
 version = "0.4.5"
