--- matchit.orig/Cargo.toml
+++ matchit/Cargo.toml
@@ -38,5 +37,0 @@
-[[bench]]
-name = "bench"
-path = "benches/bench.rs"
-harness = false
-
@@ -44,9 +39,0 @@
-[dev-dependencies.actix-router]
-version = "0.5"
-
-[dev-dependencies.criterion]
-version = "0.5"
-
-[dev-dependencies.gonzales]
-version = "0.0.3-beta"
-
@@ -57,3 +43,0 @@
-[dev-dependencies.path-tree]
-version = "0.8"
-
@@ -62,6 +45,0 @@
-
-[dev-dependencies.route-recognizer]
-version = "0.3"
-
-[dev-dependencies.routefinder]
-version = "0.5"
