Index: grep-searcher/Cargo.toml
===================================================================
--- grep-searcher.orig/Cargo.toml
+++ grep-searcher/Cargo.toml
@@ -22,7 +22,7 @@ keywords = ["regex", "grep", "egrep", "s
 license = "Unlicense/MIT"
 repository = "https://github.com/BurntSushi/ripgrep"
 [dependencies.bytecount]
-version = "0.3.2"
+version = "0.4.0"
 
 [dependencies.encoding_rs]
 version = "0.8.6"
