--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,7 @@
 license = "Unlicense/MIT"
 repository = "https://github.com/BurntSushi/ripgrep"
 [dependencies.bytecount]
-version = "0.3.2"
+version = "0.4"
 
 [dependencies.encoding_rs]
 version = "0.8.6"
@@ -48,5 +48,5 @@
 version = "1.0.5"
 
 [features]
-avx-accel = ["bytecount/avx-accel"]
-simd-accel = ["bytecount/simd-accel", "encoding_rs/simd-accel"]
+avx-accel = ["bytecount/runtime-dispatch-simd"]
+simd-accel = ["bytecount/generic-simd", "encoding_rs/simd-accel"]
