Index: wayland-sys-0.29/Cargo.toml
===================================================================
--- wayland-sys-0.29.orig/Cargo.toml
+++ wayland-sys-0.29/Cargo.toml
@@ -28,10 +28,6 @@ all-features = true
 version = "0.5"
 optional = true
 
-[dependencies.lazy_static]
-version = "1.0.2"
-optional = true
-
 [dependencies.libc]
 version = "0.2"
 optional = true
@@ -46,10 +42,6 @@ version = "0.3.7"
 [features]
 client = ["dlib"]
 cursor = ["client"]
-dlopen = [
-    "dlib",
-    "lazy_static",
-]
 egl = ["client"]
 server = [
     "libc",
