--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,7 +27,7 @@
 repository = "https://github.com/rust-netlink/rtnetlink"
 
 [dependencies.async-global-executor]
-version = "2.0.2"
+version = ">= 2.0.2, < 4"
 optional = true
 
 [dependencies.futures]
@@ -75,10 +75,10 @@
 features = ["attributes"]
 
 [dev-dependencies.env_logger]
-version = "0.10.0"
+version = "0.11"
 
 [dev-dependencies.ipnetwork]
-version = "0.18.0"
+version = "0.21.0"
 
 [dev-dependencies.macaddr]
 version = "1.0"
