Index: rtnetlink/Cargo.toml
===================================================================
--- rtnetlink.orig/Cargo.toml
+++ rtnetlink/Cargo.toml
@@ -75,10 +75,10 @@ version = "1.9.0"
 features = ["attributes"]
 
 [dev-dependencies.env_logger]
-version = "0.10.0"
+version = "0.11"
 
 [dev-dependencies.ipnetwork]
-version = "0.18.0"
+version = "0.17.0"
 
 [dev-dependencies.macaddr]
 version = "1.0"
