Index: rustyline/Cargo.toml
===================================================================
--- rustyline.orig/Cargo.toml
+++ rustyline/Cargo.toml
@@ -160,7 +160,6 @@ default = [
 derive = ["rustyline-derive"]
 with-dirs = ["home"]
 with-file-history = ["fd-lock"]
-with-fuzzy = ["skim"]
 with-sqlite-history = ["rusqlite"]
 
 [target."cfg(unix)".dependencies.nix]
@@ -179,11 +178,6 @@ version = "0.3"
 optional = true
 default-features = false
 
-[target."cfg(unix)".dependencies.skim]
-version = "0.10"
-optional = true
-default-features = false
-
 [target."cfg(unix)".dependencies.utf8parse]
 version = "0.2"
 
