--- a/Cargo.toml
+++ b/Cargo.toml
@@ -74,3 +74,3 @@
 [dev-dependencies.rand]
-version = "0.9"
+version = "0.8"
 
diff --git a/src/transform.rs b/src/transform.rs
index cb6987e..b3505ff 100644
--- a/src/transform.rs
+++ b/src/transform.rs
@@ -881,7 +881,7 @@ mod tests {
     fn test_transform_rgb8() {
         let mut srgb_profile = ColorProfile::new_srgb();
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..255);
+        let random_point_x = rand::thread_rng().gen_range(0..255);
         let transform = bt2020_profile
             .create_transform_8bit(
                 Layout::Rgb,
@@ -935,7 +935,7 @@ mod tests {
     fn test_transform_rgba8() {
         let srgb_profile = ColorProfile::new_srgb();
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..255);
+        let random_point_x = rand::thread_rng().gen_range(0..255);
         let transform = bt2020_profile
             .create_transform_8bit(
                 Layout::Rgba,
@@ -953,7 +953,7 @@ mod tests {
     fn test_transform_gray_to_rgb8() {
         let gray_profile = ColorProfile::new_gray_with_gamma(2.2f32);
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..255);
+        let random_point_x = rand::thread_rng().gen_range(0..255);
         let transform = gray_profile
             .create_transform_8bit(
                 Layout::Gray,
@@ -971,7 +971,7 @@ mod tests {
     fn test_transform_gray_to_rgba8() {
         let srgb_profile = ColorProfile::new_gray_with_gamma(2.2f32);
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..255);
+        let random_point_x = rand::thread_rng().gen_range(0..255);
         let transform = srgb_profile
             .create_transform_8bit(
                 Layout::Gray,
@@ -989,7 +989,7 @@ mod tests {
     fn test_transform_gray_to_gray_alpha8() {
         let srgb_profile = ColorProfile::new_gray_with_gamma(2.2f32);
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..255);
+        let random_point_x = rand::thread_rng().gen_range(0..255);
         let transform = srgb_profile
             .create_transform_8bit(
                 Layout::Gray,
@@ -1007,7 +1007,7 @@ mod tests {
     fn test_transform_rgb10() {
         let srgb_profile = ColorProfile::new_srgb();
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..((1 << 10) - 1));
+        let random_point_x = rand::thread_rng().gen_range(0..((1 << 10) - 1));
         let transform = bt2020_profile
             .create_transform_10bit(
                 Layout::Rgb,
@@ -1025,7 +1025,7 @@ mod tests {
     fn test_transform_rgb12() {
         let srgb_profile = ColorProfile::new_srgb();
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..((1 << 12) - 1));
+        let random_point_x = rand::thread_rng().gen_range(0..((1 << 12) - 1));
         let transform = bt2020_profile
             .create_transform_12bit(
                 Layout::Rgb,
@@ -1043,7 +1043,7 @@ mod tests {
     fn test_transform_rgb16() {
         let srgb_profile = ColorProfile::new_srgb();
         let bt2020_profile = ColorProfile::new_bt2020();
-        let random_point_x = rand::rng().random_range(0..((1u32 << 16u32) - 1u32)) as u16;
+        let random_point_x = rand::thread_rng().gen_range(0..((1u32 << 16u32) - 1u32)) as u16;
         let transform = bt2020_profile
             .create_transform_16bit(
                 Layout::Rgb,
