Index: ring/Cargo.toml
===================================================================
--- ring.orig/Cargo.toml
+++ ring/Cargo.toml
@@ -217,7 +217,6 @@ std = ["alloc"]
 test_logging = []
 unstable-testing-arm-no-hw = []
 unstable-testing-arm-no-neon = []
-wasm32_unknown_unknown_js = ["getrandom/js"]
 
 [target."cfg(all(any(target_os = \"android\", target_os = \"linux\"), any(target_arch = \"aarch64\", target_arch = \"arm\")))".dependencies.libc]
 version = "0.2.148"
