Index: ring/Cargo.toml
===================================================================
--- ring.orig/Cargo.toml
+++ ring/Cargo.toml
@@ -216,7 +216,0 @@ unstable-testing-arm-no-hw = []
-[target.'cfg(all(all(target_arch = "aarch64", target_endian = "little"), target_os = "windows"))'.dependencies.windows-sys]
-version = "0.52"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Threading",
-]
-
