--- a/src/lib.rs
+++ b/src/lib.rs
@@ -24,18 +24,18 @@ pub type backtrace_error_callback =
 pub enum backtrace_state {}
 
 extern {
-    #[link_name = "__rbt_backtrace_create_state"]
+    #[link_name = "backtrace_create_state"]
     pub fn backtrace_create_state(filename: *const c_char,
                                   threaded: c_int,
                                   error: backtrace_error_callback,
                                   data: *mut c_void) -> *mut backtrace_state;
-    #[link_name = "__rbt_backtrace_syminfo"]
+    #[link_name = "backtrace_syminfo"]
     pub fn backtrace_syminfo(state: *mut backtrace_state,
                              addr: uintptr_t,
                              cb: backtrace_syminfo_callback,
                              error: backtrace_error_callback,
                              data: *mut c_void) -> c_int;
-    #[link_name = "__rbt_backtrace_pcinfo"]
+    #[link_name = "backtrace_pcinfo"]
     pub fn backtrace_pcinfo(state: *mut backtrace_state,
                             addr: uintptr_t,
                             cb: backtrace_full_callback,
