--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,4 +70,4 @@
 
-[dependencies.copypasta-ext]
-version = "0.4.4"
+[dependencies.arboard]
+version = "3.4.0"
 
--- a/src/app.rs
+++ b/src/app.rs
@@ -8,4 +8,2 @@
 use crate::widgets::StatefulList;
-use copypasta_ext::display::DisplayServer as ClipboardDisplayServer;
-use copypasta_ext::ClipboardProviderExt;
 use enum_iterator::Sequence;
@@ -141,3 +139,3 @@
 	style: Style,
-	clipboard: Option<Box<dyn ClipboardProviderExt>>,
+	clipboard: Option<arboard::Clipboard>,
 }
@@ -170,9 +168,7 @@
 			style,
-			clipboard: match ClipboardDisplayServer::select().try_context() {
-				None => {
-					eprintln!("failed to initialize clipboard, no suitable clipboard provider found");
-					None
-				}
-				clipboard => clipboard,
-			},
+			clipboard: arboard::Clipboard::new()
+				.map_err(|e| {
+					eprintln!("failed to initialize clipboard: {e:?}");
+					e
+				}).ok(),
 		}
@@ -227,3 +223,3 @@
 		if let Some(clipboard) = self.clipboard.as_mut() {
-			if let Ok(contents) = clipboard.get_contents() {
+			if let Ok(contents) = clipboard.get_text() {
 				return contents;
@@ -241,4 +237,4 @@
 		if let Some(clipboard) = self.clipboard.as_mut() {
-			if let Err(e) = clipboard.set_contents(contents.to_string()) {
-				eprintln!("{e}");
+			if let Err(e) = clipboard.set_text(contents) {
+				eprintln!("failed to set clipboard: {e}");
 			}
