--- a/Cargo.toml
+++ b/Cargo.toml
@@ -90,6 +90,8 @@
 
 [dependencies.ipnetwork]
 version = ">=0.12.2, <0.22.0"
+default_features = false
+features = ["serde"]
 optional = true
 
 [dependencies.itoa]
@@ -100,5 +100,5 @@
 [dependencies.libsqlite3-sys]
 version = ">=0.17.2, <0.32.0"
-features = ["bundled_bindings"]
+features = ["buildtime_bindgen"]
 optional = true

@@ -110,4 +110,0 @@
-[dependencies.mysqlclient-src]
-version = "0.1.0"
-optional = true
-
@@ -131,4 +127,0 @@
-[dependencies.pq-src]
-version = "0.3"
-optional = true
-
@@ -171,6 +163,8 @@
 
 [dev-dependencies.ipnetwork]
 version = ">=0.12.2, <0.22.0"
+default_features = false
+features = ["serde"]
 
 [dev-dependencies.quickcheck]
 version = "1.0.3"
@@ -189,14 +181,6 @@
     "32-column-tables",
     "diesel_derives/64-column-tables",
 ]
-__with_asan_tests = [
-    "libsqlite3-sys?/bundled",
-    "libsqlite3-sys?/with-asan",
-    "pq-sys?/bundled",
-    "pq-src?/with-asan",
-    "mysqlclient-sys?/bundled",
-    "mysqlclient-src?/with-asan",
-]
 chrono = [
     "diesel_derives/chrono",
     "dep:chrono",
