--- a/Cargo.toml
+++ b/Cargo.toml
@@ -91,7 +91,7 @@
 optional = true
 
 [dependencies.ipnetwork]
-version = ">=0.12.2, <0.22.0"
+version = "0.17"
 optional = true
 
 [dependencies.itoa]
@@ -103,20 +103,16 @@
 optional = true
 
 [dependencies.libsqlite3-sys]
-version = ">=0.17.2, <0.32.0"
-features = ["bundled_bindings"]
-optional = true
-
-[dependencies.mysqlclient-src]
-version = "0.1.0"
+version = "0.26"
+features = ["buildtime-bindgen"]
 optional = true
 
 [dependencies.mysqlclient-sys]
-version = ">=0.2.5, <0.5.0"
+version = "0.2.5"
 optional = true
 
 [dependencies.num-bigint]
-version = ">=0.2.0, <0.5.0"
+version = "0.4.6"
 optional = true
 
 [dependencies.num-integer]
@@ -131,12 +127,8 @@
 version = "2.1.0"
 optional = true
 
-[dependencies.pq-src]
-version = "0.3"
-optional = true
-
 [dependencies.pq-sys]
-version = ">=0.4.0, <0.8.0"
+version = "0.4.8"
 optional = true
 
 [dependencies.quickcheck]
@@ -144,7 +136,7 @@
 optional = true
 
 [dependencies.r2d2]
-version = ">= 0.8.2, < 0.9.0"
+version = "0.8.10"
 optional = true
 
 [dependencies.serde_json]
@@ -171,7 +163,7 @@
 version = "0.15"
 
 [dev-dependencies.ipnetwork]
-version = ">=0.12.2, <0.22.0"
+version = "0.17"
 
 [dev-dependencies.quickcheck]
 version = "1.0.3"
@@ -189,14 +181,6 @@
     "32-column-tables",
     "diesel_derives/64-column-tables",
 ]
-__with_asan_tests = [
-    "libsqlite3-sys?/bundled",
-    "libsqlite3-sys?/with-asan",
-    "pq-sys?/bundled",
-    "pq-src?/with-asan",
-    "mysqlclient-sys?/bundled",
-    "mysqlclient-src?/with-asan",
-]
 chrono = [
     "diesel_derives/chrono",
     "dep:chrono",
