diff --git a/Cargo.toml b/Cargo.toml
index d03a197..cd5c7bf 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -73,7 +73,7 @@ version = "2.5.0"
 optional = true
 
 [dependencies.ipnetwork]
-version = ">=0.12.2, <0.21.0"
+version = "0.17"
 optional = true
 
 [dependencies.itoa]
@@ -145,11 +145,8 @@ optional = true
 [dev-dependencies.cfg-if]
 version = "1"
 
-[dev-dependencies.dotenvy]
-version = "0.15"
-
 [dev-dependencies.ipnetwork]
-version = ">=0.12.2, <0.21.0"
+version = "0.17"
 
 [dev-dependencies.quickcheck]
 version = "1.0.3"
