Index: diesel/src/doctest_setup.rs
===================================================================
--- diesel.orig/src/doctest_setup.rs
+++ diesel/src/doctest_setup.rs
@@ -1,5 +1,5 @@
 use diesel::prelude::*;
-use dotenvy::dotenv;
+use dotenv::dotenv;
 
 cfg_if::cfg_if! {
     if #[cfg(feature = "postgres")] {
Index: diesel/src/mysql/connection/mod.rs
===================================================================
--- diesel.orig/src/mysql/connection/mod.rs
+++ diesel/src/mysql/connection/mod.rs
@@ -238,13 +238,13 @@ impl MysqlConnection {
 
 #[cfg(test)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use super::*;
     use std::env;
 
     fn connection() -> MysqlConnection {
-        dotenvy::dotenv().ok();
+        dotenv::dotenv().ok();
         let database_url = env::var("MYSQL_UNIT_TEST_DATABASE_URL")
             .or_else(|_| env::var("MYSQL_DATABASE_URL"))
             .or_else(|_| env::var("DATABASE_URL"))
Index: diesel/src/mysql/types/date_and_time/chrono.rs
===================================================================
--- diesel.orig/src/mysql/types/date_and_time/chrono.rs
+++ diesel/src/mysql/types/date_and_time/chrono.rs
@@ -133,7 +133,7 @@ impl FromSql<Date, Mysql> for NaiveDate
 #[cfg(test)]
 mod tests {
     extern crate chrono;
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use self::chrono::{Duration, NaiveDate, NaiveTime, Utc};
 
Index: diesel/src/mysql/types/date_and_time/time.rs
===================================================================
--- diesel.orig/src/mysql/types/date_and_time/time.rs
+++ diesel/src/mysql/types/date_and_time/time.rs
@@ -208,7 +208,7 @@ impl FromSql<Date, Mysql> for NaiveDate
 
 #[cfg(test)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
     extern crate time;
 
     use time::{
Index: diesel/src/pg/connection/mod.rs
===================================================================
--- diesel.orig/src/pg/connection/mod.rs
+++ diesel/src/pg/connection/mod.rs
@@ -420,7 +420,7 @@ mod private {
 
 #[cfg(test)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use super::*;
     use crate::dsl::sql;
Index: diesel/src/pg/expression/extensions/interval_dsl.rs
===================================================================
--- diesel.orig/src/pg/expression/extensions/interval_dsl.rs
+++ diesel/src/pg/expression/extensions/interval_dsl.rs
@@ -246,7 +246,7 @@ impl IntervalDsl for f64 {
 // that's fine for this test code
 #[allow(clippy::items_after_statements)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
     extern crate quickcheck;
 
     use self::quickcheck::quickcheck;
Index: diesel/src/pg/transaction.rs
===================================================================
--- diesel.orig/src/pg/transaction.rs
+++ diesel/src/pg/transaction.rs
@@ -377,7 +377,7 @@ impl QueryFragment<Pg> for Deferrable {
 
 #[test]
 fn test_transaction_builder_generates_correct_sql() {
-    extern crate dotenvy;
+    extern crate dotenv;
 
     macro_rules! assert_sql {
         ($query:expr, $sql:expr) => {
@@ -388,8 +388,8 @@ fn test_transaction_builder_generates_co
         };
     }
 
-    let database_url = dotenvy::var("PG_DATABASE_URL")
-        .or_else(|_| dotenvy::var("DATABASE_URL"))
+    let database_url = dotenv::var("PG_DATABASE_URL")
+        .or_else(|_| dotenv::var("DATABASE_URL"))
         .expect("DATABASE_URL must be set in order to run tests");
     let mut conn = PgConnection::establish(&database_url).unwrap();
 
Index: diesel/src/pg/types/date_and_time/chrono.rs
===================================================================
--- diesel.orig/src/pg/types/date_and_time/chrono.rs
+++ diesel/src/pg/types/date_and_time/chrono.rs
@@ -137,7 +137,7 @@ impl FromSql<Date, Pg> for NaiveDate {
 #[cfg(test)]
 mod tests {
     extern crate chrono;
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use self::chrono::{Duration, FixedOffset, NaiveDate, NaiveTime, TimeZone, Utc};
 
Index: diesel/src/pg/types/date_and_time/std_time.rs
===================================================================
--- diesel.orig/src/pg/types/date_and_time/std_time.rs
+++ diesel/src/pg/types/date_and_time/std_time.rs
@@ -60,7 +60,7 @@ fn duration_to_usecs(duration: Duration)
 
 #[cfg(test)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use std::time::{Duration, SystemTime, UNIX_EPOCH};
 
Index: diesel/src/pg/types/date_and_time/time.rs
===================================================================
--- diesel.orig/src/pg/types/date_and_time/time.rs
+++ diesel/src/pg/types/date_and_time/time.rs
@@ -121,7 +121,7 @@ impl FromSql<Date, Pg> for NaiveDate {
 
 #[cfg(test)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use crate::dsl::{now, sql};
     use crate::prelude::*;
Index: diesel/src/sqlite/types/date_and_time/chrono.rs
===================================================================
--- diesel.orig/src/sqlite/types/date_and_time/chrono.rs
+++ diesel/src/sqlite/types/date_and_time/chrono.rs
@@ -184,7 +184,7 @@ impl<TZ: TimeZone> ToSql<TimestamptzSqli
 #[cfg(test)]
 mod tests {
     extern crate chrono;
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use self::chrono::{
         DateTime, Duration, FixedOffset, NaiveDate, NaiveDateTime, NaiveTime, TimeZone, Timelike,
Index: diesel/src/sqlite/types/date_and_time/time.rs
===================================================================
--- diesel.orig/src/sqlite/types/date_and_time/time.rs
+++ diesel/src/sqlite/types/date_and_time/time.rs
@@ -224,7 +224,7 @@ impl ToSql<TimestamptzSqlite, Sqlite> fo
 
 #[cfg(test)]
 mod tests {
-    extern crate dotenvy;
+    extern crate dotenv;
 
     use super::time::{
         macros::{date, datetime},
Index: diesel/src/test_helpers.rs
===================================================================
--- diesel.orig/src/test_helpers.rs
+++ diesel/src/test_helpers.rs
@@ -1,4 +1,4 @@
-extern crate dotenvy;
+extern crate dotenv;
 
 use crate::prelude::*;
 
@@ -37,8 +37,8 @@ cfg_if! {
         }
 
         pub fn database_url() -> String {
-            dotenvy::var("MYSQL_UNIT_TEST_DATABASE_URL")
-                .or_else(|_| dotenvy::var("DATABASE_URL"))
+            dotenv::var("MYSQL_UNIT_TEST_DATABASE_URL")
+                .or_else(|_| dotenv::var("DATABASE_URL"))
                 .expect("DATABASE_URL must be set in order to run tests")
         }
     } else {
@@ -65,7 +65,7 @@ pub fn pg_connection_no_transaction() ->
 
 #[cfg(feature = "postgres")]
 pub fn pg_database_url() -> String {
-    dotenvy::var("PG_DATABASE_URL")
-        .or_else(|_| dotenvy::var("DATABASE_URL"))
+    dotenv::var("PG_DATABASE_URL")
+        .or_else(|_| dotenv::var("DATABASE_URL"))
         .expect("DATABASE_URL must be set in order to run tests")
 }
