Index: clap-3/Cargo.toml
===================================================================
--- clap-3.orig/Cargo.toml
+++ clap-3/Cargo.toml
@@ -342,10 +342,6 @@ optional = true
 [dependencies.bitflags]
 version = "1.2"
 
-[dependencies.clap_derive]
-version = "=3.2.18"
-optional = true
-
 [dependencies.clap_lex]
 version = "0.3"
 
@@ -409,7 +405,6 @@ color = [
     "termcolor",
 ]
 debug = [
-    "clap_derive/debug",
     "backtrace",
 ]
 default = [
@@ -418,9 +413,6 @@ default = [
     "suggestions",
 ]
 deprecated = []
-derive = [
-    "clap_derive"
-]
 env = []
 std = ["indexmap/std"]
 suggestions = ["strsim"]
