--- a/Cargo.toml
+++ b/Cargo.toml
@@ -354,7 +354,7 @@
 version = "0.2.2"
 
 [dependencies.indexmap]
-version = "1.0"
+version = "1"
 
 [dependencies.once_cell]
 version = "1.12.0"
@@ -373,11 +373,11 @@
 optional = true
 
 [dependencies.terminal_size]
-version = "0.2.1"
+version = "< 0.3"
 optional = true
 
 [dependencies.textwrap]
-version = "0.16"
+version = "< 0.17"
 features = []
 default-features = false
 
