--- a/Cargo.toml
+++ b/Cargo.toml
@@ -347,10 +347,10 @@
 optional = true
 
 [dependencies.clap_lex]
-version = "0.6"
+version = "0.7"
 
 [dependencies.indexmap]
-version = "1.0"
+version = ">= 1.0"
 
 [dependencies.once_cell]
 version = "1.12.0"
@@ -360,7 +360,7 @@
 optional = true
 
 [dependencies.strsim]
-version = "0.10"
+version = "0.11"
 optional = true
 
 [dependencies.termcolor]
@@ -368,11 +368,11 @@
 optional = true
 
 [dependencies.terminal_size]
-version = "0.2.1"
+version = ">= 0.2.1"
 optional = true
 
 [dependencies.textwrap]
-version = "0.16"
+version = "< 0.17"
 features = []
 default-features = false
 
@@ -381,8 +381,9 @@
 optional = true
 
 [dependencies.yaml-rust]
-version = "0.4.1"
+version = "0.10"
 optional = true
+package = "yaml-rust2"
 
 [dev-dependencies.humantime]
 version = "2"
