Index: tealdeer/Cargo.toml
===================================================================
--- tealdeer.orig/Cargo.toml
+++ tealdeer/Cargo.toml
@@ -38,7 +38,7 @@ package = "app_dirs2"
 version = "0.2"
 
 [dependencies.clap]
-version = "=3.0.0-rc.11"
+version = "3"
 features = ["std", "derive", "suggestions", "color"]
 default-features = false
 
@@ -67,7 +67,7 @@ version = "0.5.1"
 version = "2.0.1"
 
 [dependencies.zip]
-version = "0.5"
+version = "0.6"
 features = ["deflate"]
 default-features = false
 [dev-dependencies.assert_cmd]
