Description: fix version

Last-Update: 2024-02-19
---

--- rust-circular-buffer-1.0.0.orig/Cargo.toml
+++ rust-circular-buffer-1.0.0/Cargo.toml
@@ -80,14 +80,14 @@ optional = true
 default-features = false
 
 [dev-dependencies.criterion]
-version = "0.3"
+version = "0.5"
 features = ["html_reports"]
 
 [dev-dependencies.drop-tracker]
 version = "0.1.3"
 
 [dev-dependencies.futures-lite]
-version = "2.6"
+version = ">= 2.3"
 
 [dev-dependencies.rand]
 version = "0.8"
