Index: linux-audit-parser/Cargo.toml
===================================================================
--- linux-audit-parser.orig/Cargo.toml
+++ linux-audit-parser/Cargo.toml
@@ -56,7 +56,7 @@ version = "1.0.209"
 optional = true
 
 [dependencies.serde_with]
-version = "3.12.0"
+version = "3.11"
 features = ["macros"]
 optional = true
 default-features = false
@@ -69,4 +69,4 @@ version = "1.6"
 features = ["alloc"]
 
 [dev-dependencies.serde_test]
-version = "1.0.177"
+version = "1.0.170"
