Index: geo-types/src/arbitrary.rs
===================================================================
--- geo-types.orig/src/arbitrary.rs
+++ geo-types/src/arbitrary.rs
@@ -2,7 +2,8 @@ use crate::{
     Coord, CoordFloat, Geometry, GeometryCollection, LineString, MultiLineString, MultiPoint,
     MultiPolygon, Point, Polygon, Rect, Triangle,
 };
-use std::mem;
+use core::mem;
+use alloc::vec::Vec;
 
 impl<'a, T> arbitrary::Arbitrary<'a> for Coord<T>
 where
