Origin: vendor
Subject: patch out Windows dependencies
Forwarded: not-needed

this should be handled automatically by debcargo...

Index: sequoia-wot/Cargo.toml
===================================================================
--- sequoia-wot.orig/Cargo.toml
+++ sequoia-wot/Cargo.toml
@@ -83,13 +83,5 @@ features = [
 ]
 default-features = false
 
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "1"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
-
 [badges.maintenance]
 status = "actively-developed"
