Origin: vendor
Subject: enable crypto-nettle feature
Forwarded: not-needed


Index: sequoia-wot/Cargo.toml
===================================================================
--- sequoia-wot.orig/Cargo.toml
+++ sequoia-wot/Cargo.toml
@@ -65,6 +65,7 @@ version = "1"
 [dependencies.sequoia-cert-store]
 version = "0.7"
 default-features = false
+features = ["crypto-nettle"]
 
 [dependencies.sequoia-openpgp]
 version = "2"
