Origin: vendor
Subject: enable crypto-nettle feature
Forwarded: not-needed


Index: sequoia-wot/Cargo.toml
===================================================================
--- sequoia-wot.orig/Cargo.toml
+++ sequoia-wot/Cargo.toml
@@ -60,6 +60,7 @@ version = "0.4"
 [dependencies.sequoia-openpgp]
 version = "1.20"
 default-features = false
+features = ["crypto-nettle"]
 
 [dependencies.thiserror]
 version = "1.0.2"
