Origin: vendor
Subject: enable crypto-nettle feature
Forwarded: not-needed


Index: sequoia-wot/Cargo.toml
===================================================================
--- sequoia-wot.orig/Cargo.toml
+++ sequoia-wot/Cargo.toml
@@ -66,6 +66,7 @@ default-features = false
 [dependencies.sequoia-openpgp]
 version = "1.20"
 default-features = false
+features = ["crypto-nettle"]
 
 [dependencies.thiserror]
 version = ">=1, <3"
