Index: condure/Cargo.toml
===================================================================
--- condure.orig/Cargo.toml
+++ condure/Cargo.toml
@@ -46,7 +46,7 @@ name = "condure"
 name = "server"
 harness = false
 [dependencies.arrayvec]
-version = "0.5"
+version = "0.7"
 
 [dependencies.base64]
 version = ">= 0.12, < 0.14"
Index: condure/src/server.rs
===================================================================
--- condure.orig/src/server.rs
+++ condure/src/server.rs
@@ -193,7 +193,7 @@ fn async_local_channel<T>(
     (s, r)
 }
 
-fn gen_id(id: usize, ckey: usize, next_cid: &mut u32) -> ArrayString<[u8; 32]> {
+fn gen_id(id: usize, ckey: usize, next_cid: &mut u32) -> ArrayString<32> {
     let mut buf = [0; 32];
     let mut c = io::Cursor::new(&mut buf[..]);
 
@@ -205,7 +205,7 @@ fn gen_id(id: usize, ckey: usize, next_c
 
     *next_cid += 1;
 
-    ArrayString::from_str(s).unwrap()
+    ArrayString::from_str(s).unwrap()
 }
 
 impl Shutdown for TcpStream {
@@ -248,14 +248,14 @@ impl ZhttpSender for channel::LocalSende
 
 struct StreamLocalSenders {
     out: channel::LocalSender<zmq::Message>,
-    out_stream: channel::LocalSender<(ArrayVec<[u8; 64]>, zmq::Message)>,
+    out_stream: channel::LocalSender<(ArrayVec<u8, 64>, zmq::Message)>,
     out_stream_can_write: Cell<bool>,
 }
 
 impl StreamLocalSenders {
     fn new(
         out: channel::LocalSender<zmq::Message>,
-        out_stream: channel::LocalSender<(ArrayVec<[u8; 64]>, zmq::Message)>,
+        out_stream: channel::LocalSender<(ArrayVec<u8, 64>, zmq::Message)>,
     ) -> Self {
         Self {
             out,
@@ -334,7 +334,7 @@ impl Stream {
 }
 
 struct Connection {
-    id: ArrayString<[u8; 32]>,
+    id: ArrayString<32>,
     stream: Stream,
     conn: ServerConnection,
     want: Want,
@@ -536,7 +536,7 @@ impl Connection {
     }
 
     fn process_with_stream<S: Read + Write + Shutdown>(
-        id: &ArrayString<[u8; 32]>,
+        id: &ArrayString<32>,
         conn: &mut ServerConnection,
         want: &mut Want,
         stream: &mut S,
@@ -600,7 +600,7 @@ impl<'a> BatchGroup<'a, '_> {
 
 struct Batch {
     nodes: Slab<list::Node<usize>>,
-    addrs: Vec<(ArrayVec<[u8; 64]>, list::List)>,
+    addrs: Vec<(ArrayVec<u8, 64>, list::List)>,
     addr_index: usize,
     group_ids: arena::ReusableVec,
     last_group_ckeys: Vec<usize>,
@@ -722,7 +722,7 @@ enum BatchType {
 }
 
 struct ConnectionItem {
-    id: ArrayString<[u8; 32]>,
+    id: ArrayString<32>,
     stop: Option<AsyncLocalSender<()>>,
     zreceiver_sender:
         Option<AsyncLocalSender<(arena::Rc<zhttppacket::OwnedResponse>, Option<u32>)>>,
@@ -783,7 +783,7 @@ impl Connections {
         stop: AsyncLocalSender<()>,
         zreceiver_sender: AsyncLocalSender<(arena::Rc<zhttppacket::OwnedResponse>, Option<u32>)>,
         shared: Option<arena::Rc<ServerStreamSharedData>>,
-    ) -> Result<(usize, ArrayString<[u8; 32]>), ()> {
+    ) -> Result<(usize, ArrayString<32>), ()> {
         let items = &mut *self.items.borrow_mut();
         let c = &mut *self.inner.borrow_mut();
 
@@ -825,7 +825,7 @@ impl Connections {
         items.nodes.remove(nkey);
     }
 
-    fn regen_id(&self, worker_id: usize, ckey: usize) -> ArrayString<[u8; 32]> {
+    fn regen_id(&self, worker_id: usize, ckey: usize) -> ArrayString<32> {
         let nkey = ckey;
 
         let items = &mut *self.items.borrow_mut();
@@ -963,7 +963,7 @@ impl Connections {
         &self,
         from: &str,
         btype: BatchType,
-    ) -> Option<(usize, ArrayVec<[u8; 64]>, zmq::Message)> {
+    ) -> Option<(usize, ArrayVec<u8, 64>, zmq::Message)> {
         let items = &mut *self.items.borrow_mut();
         let nodes = &mut items.nodes;
         let batch = &mut items.batch;
@@ -1008,7 +1008,7 @@ impl Connections {
 
             let data = &data[..size];
 
-            let mut addr = ArrayVec::<[u8; 64]>::new();
+            let mut addr = ArrayVec::<u8, 64>::new();
             if addr.try_extend_from_slice(group.addr()).is_err() {
                 error!("failed to prepare addr");
                 continue;
@@ -1051,7 +1051,7 @@ struct ConnectionReqOpts {
 struct ConnectionStreamOpts {
     messages_max: usize,
     sender: channel::LocalSender<zmq::Message>,
-    sender_stream: channel::LocalSender<(ArrayVec<[u8; 64]>, zmq::Message)>,
+    sender_stream: channel::LocalSender<(ArrayVec<u8, 64>, zmq::Message)>,
     stream_shared_mem: Rc<arena::RcMemory<ServerStreamSharedData>>,
 }
 
@@ -1705,7 +1705,7 @@ impl Worker {
         done: AsyncLocalSender<zhttpsocket::AsyncClientStreamHandle>,
         instance_id: Rc<String>,
         zstream_out_receiver: AsyncLocalReceiver<zmq::Message>,
-        zstream_out_stream_receiver: AsyncLocalReceiver<(ArrayVec<[u8; 64]>, zmq::Message)>,
+        zstream_out_stream_receiver: AsyncLocalReceiver<(ArrayVec<u8, 64>, zmq::Message)>,
         stream_handle: zhttpsocket::AsyncClientStreamHandle,
         stream_maxconn: usize,
         conns: Rc<Connections>,
@@ -1862,7 +1862,7 @@ impl Worker {
         done: channel::LocalSender<usize>,
         worker_id: usize,
         ckey: usize,
-        mut cid: ArrayString<[u8; 32]>,
+        mut cid: ArrayString<32>,
         mut stream: Stream,
         peer_addr: SocketAddr,
         zreceiver: channel::LocalReceiver<(arena::Rc<zhttppacket::OwnedResponse>, Option<u32>)>,
@@ -2111,7 +2111,7 @@ impl Worker {
         stop: AsyncLocalReceiver<()>,
         _done: AsyncLocalSender<()>,
         instance_id: Rc<String>,
-        sender: channel::LocalSender<(ArrayVec<[u8; 64]>, zmq::Message)>,
+        sender: channel::LocalSender<(ArrayVec<u8, 64>, zmq::Message)>,
         conns: Rc<Connections>,
     ) {
         debug!("worker {}: task started: keep_alives", id);
Index: condure/src/tls.rs
===================================================================
--- condure.orig/src/tls.rs
+++ condure/src/tls.rs
@@ -337,7 +337,7 @@ impl TlsAcceptor {
 pub struct TlsStream {
     stream: Stream<'static>,
     tcp_stream: Box<TcpStream>,
-    id: ArrayString<[u8; 32]>,
+    id: ArrayString<32>,
 }
 
 impl TlsStream {
Index: condure/src/zhttpsocket.rs
===================================================================
--- condure.orig/src/zhttpsocket.rs
+++ condure/src/zhttpsocket.rs
@@ -294,7 +294,7 @@ struct ReqPipeEnd {
 struct StreamPipeEnd {
     sender: channel::Sender<arena::Arc<zmq::Message>>,
     receiver_any: channel::Receiver<zmq::Message>,
-    receiver_addr: channel::Receiver<(ArrayVec<[u8; 64]>, zmq::Message)>,
+    receiver_addr: channel::Receiver<(ArrayVec<u8, 64>, zmq::Message)>,
 }
 
 struct AsyncReqPipeEnd {
@@ -305,28 +305,28 @@ struct AsyncReqPipeEnd {
 struct AsyncStreamPipeEnd {
     sender: AsyncSender<arena::Arc<zmq::Message>>,
     receiver_any: AsyncReceiver<zmq::Message>,
-    receiver_addr: AsyncReceiver<(ArrayVec<[u8; 64]>, zmq::Message)>,
+    receiver_addr: AsyncReceiver<(ArrayVec<u8, 64>, zmq::Message)>,
 }
 
 enum ControlRequest {
     Stop,
     SetClientReq(Vec<SpecInfo>),
     SetClientStream(Vec<SpecInfo>, Vec<SpecInfo>, Vec<SpecInfo>),
-    AddClientReqHandle(ReqPipeEnd, ArrayString<[u8; 8]>),
-    AddClientStreamHandle(StreamPipeEnd, ArrayString<[u8; 8]>),
+    AddClientReqHandle(ReqPipeEnd, ArrayString<8>),
+    AddClientStreamHandle(StreamPipeEnd, ArrayString<8>),
 }
 
 type ControlResponse = Result<(), String>;
 
 struct ReqPipe {
     pe: AsyncReqPipeEnd,
-    filter: ArrayString<[u8; 8]>,
+    filter: ArrayString<8>,
     valid: Cell<bool>,
 }
 
 struct StreamPipe {
     pe: AsyncStreamPipeEnd,
-    filter: ArrayString<[u8; 8]>,
+    filter: ArrayString<8>,
     valid: Cell<bool>,
 }
 
@@ -398,7 +398,7 @@ impl ReqHandles {
         self.nodes.len()
     }
 
-    fn add(&mut self, pe: AsyncReqPipeEnd, filter: ArrayString<[u8; 8]>) {
+    fn add(&mut self, pe: AsyncReqPipeEnd, filter: ArrayString<8>) {
         assert!(self.nodes.len() < self.nodes.capacity());
 
         let key = self.nodes.insert(list::Node::new(ReqPipe {
@@ -513,7 +513,7 @@ struct StreamHandles {
     nodes: Slab<list::Node<StreamPipe>>,
     list: list::List,
     recv_any_scratch: RefCell<RecvScratch<zmq::Message>>,
-    recv_addr_scratch: RefCell<RecvScratch<(ArrayVec<[u8; 64]>, zmq::Message)>>,
+    recv_addr_scratch: RefCell<RecvScratch<(ArrayVec<u8, 64>, zmq::Message)>>,
     need_cleanup: Cell<bool>,
 }
 
@@ -532,7 +532,7 @@ impl StreamHandles {
         self.nodes.len()
     }
 
-    fn add(&mut self, pe: AsyncStreamPipeEnd, filter: ArrayString<[u8; 8]>) {
+    fn add(&mut self, pe: AsyncStreamPipeEnd, filter: ArrayString<8>) {
         assert!(self.nodes.len() < self.nodes.capacity());
 
         let key = self.nodes.insert(list::Node::new(StreamPipe {
@@ -582,7 +582,7 @@ impl StreamHandles {
         }
     }
 
-    async fn recv_addr(&self) -> (ArrayVec<[u8; 64]>, zmq::Message) {
+    async fn recv_addr(&self) -> (ArrayVec<u8, 64>, zmq::Message) {
         let mut scratch = self.recv_addr_scratch.borrow_mut();
 
         let (mut tasks, slice_scratch) = scratch.get();
@@ -1281,7 +1281,7 @@ impl AsyncClientReqHandle {
 
 pub struct ClientStreamHandle {
     sender_any: channel::Sender<zmq::Message>,
-    sender_addr: channel::Sender<(ArrayVec<[u8; 64]>, zmq::Message)>,
+    sender_addr: channel::Sender<(ArrayVec<u8, 64>, zmq::Message)>,
     receiver: channel::Receiver<arena::Arc<zmq::Message>>,
 }
 
@@ -1336,7 +1336,7 @@ impl ClientStreamHandle {
 
 pub struct AsyncClientStreamHandle {
     sender_any: AsyncSender<zmq::Message>,
-    sender_addr: AsyncSender<(ArrayVec<[u8; 64]>, zmq::Message)>,
+    sender_addr: AsyncSender<(ArrayVec<u8, 64>, zmq::Message)>,
     receiver: AsyncReceiver<arena::Arc<zmq::Message>>,
 }
 
@@ -1365,7 +1365,7 @@ impl AsyncClientStreamHandle {
 
     pub async fn send_to_addr(
         &self,
-        addr: ArrayVec<[u8; 64]>,
+        addr: ArrayVec<u8, 64>,
         msg: zmq::Message,
     ) -> Result<(), io::Error> {
         match self.sender_addr.send((addr, msg)).await {
Index: condure/src/connection.rs
===================================================================
--- condure.orig/src/connection.rs
+++ condure/src/connection.rs
@@ -395,7 +395,7 @@ enum ServerReqState {
 }
 
 pub struct ServerReqConnection {
-    id: ArrayString<[u8; 32]>,
+    id: ArrayString<32>,
     peer_addr: Option<SocketAddr>,
     secure: bool,
     timeout: Duration,
@@ -985,7 +985,7 @@ enum ServerStreamState {
 }
 
 struct ServerStreamSharedDataInner {
-    to_addr: Option<ArrayVec<[u8; 64]>>,
+    to_addr: Option<ArrayVec<u8, 64>>,
     out_seq: u32,
 }
 
@@ -1023,7 +1023,7 @@ impl ServerStreamSharedData {
         s.out_seq = 0;
     }
 
-    fn set_to_addr(&self, addr: Option<ArrayVec<[u8; 64]>>) {
+    fn set_to_addr(&self, addr: Option<ArrayVec<u8, 64>>) {
         let s = &mut *self.inner.borrow_mut();
 
         s.to_addr = addr;
@@ -1047,7 +1047,7 @@ impl ServerStreamSharedData {
 }
 
 struct ServerStreamData {
-    id: ArrayString<[u8; 32]>,
+    id: ArrayString<32>,
     peer_addr: Option<SocketAddr>,
     secure: bool,
     client_timeout: Duration,
@@ -1059,7 +1059,7 @@ struct ServerStreamData {
     cont_len: usize,
     cont_left: usize,
     websocket: bool,
-    ws_accept: Option<ArrayString<[u8; 28]>>, // base64_encode(sha1_hash) = 28 bytes
+    ws_accept: Option<ArrayString<28>>, // base64_encode(sha1_hash) = 28 bytes
     in_seq: u32,
     in_credits: u32,
     out_credits: u32,
