Index: ntp-proto/Cargo.toml
===================================================================
--- ntp-proto.orig/Cargo.toml
+++ ntp-proto/Cargo.toml
@@ -76,29 +76,6 @@ version = "0.21.0"
 optional = true
 package = "rustls"
 
-[dependencies.rustls22]
-version = "0.22.0"
-features = [
-    "ring",
-    "logging",
-    "tls12",
-]
-optional = true
-default-features = false
-package = "rustls"
-
-[dependencies.rustls23]
-version = "0.23.0"
-features = [
-    "ring",
-    "logging",
-    "std",
-    "tls12",
-]
-optional = true
-default-features = false
-package = "rustls"
-
 [dependencies.serde]
 version = "1.0.145"
 features = ["derive"]
@@ -127,14 +104,3 @@ rustls21 = [
     "dep:rustls-pemfile1",
     "dep:rustls-native-certs6",
 ]
-rustls22 = [
-    "dep:rustls22",
-    "dep:rustls-pemfile2",
-    "dep:rustls-native-certs7",
-    "dep:rustls-pki-types",
-]
-rustls23 = [
-    "dep:rustls23",
-    "dep:rustls-pemfile2",
-    "dep:rustls-native-certs7",
-]
