--- ntp-proto.orig/Cargo.toml
+++ ntp-proto/Cargo.toml
@@ -54,7 +54,2 @@
 
-[dependencies.rustls-native-certs7]
-version = "0.7"
-optional = true
-package = "rustls-native-certs"
-
 [dependencies.rustls-pemfile1]
@@ -64,11 +59,2 @@
 
-[dependencies.rustls-pemfile2]
-version = "2.0"
-optional = true
-package = "rustls-pemfile"
-
-[dependencies.rustls-pki-types]
-version = "1.2"
-optional = true
-
 [dependencies.rustls21]
@@ -78,25 +64,2 @@
 
-[dependencies.rustls22]
-version = "0.22.0"
-features = [
-    "ring",
-    "logging",
-    "tls12",
-]
-optional = true
-default-features = false
-package = "rustls"
-
-[dependencies.rustls23]
-version = "0.23.0"
-features = [
-    "ring",
-    "logging",
-    "std",
-    "tls12",
-]
-optional = true
-default-features = false
-package = "rustls"
-
 [dependencies.serde]
@@ -129,12 +92 @@
 ]
-rustls22 = [
-    "dep:rustls22",
-    "dep:rustls-pemfile2",
-    "dep:rustls-native-certs7",
-    "dep:rustls-pki-types",
-]
-rustls23 = [
-    "dep:rustls23",
-    "dep:rustls-pemfile2",
-    "dep:rustls-native-certs7",
-]
