Index: ntp-proto/Cargo.toml
===================================================================
--- ntp-proto.orig/Cargo.toml
+++ ntp-proto/Cargo.toml
@@ -119,9 +119,9 @@ __internal-fuzz = [
     "__internal-api",
 ]
 __internal-test = ["__internal-api"]
-default = ["rustls23"]
+default = ["rustls21"]
 ntpv5 = []
-nts-pool = ["rustls23"]
+nts-pool = ["rustls21"]
 rustls21 = [
     "dep:rustls21",
     "dep:rustls-pemfile1",
