Index: ntp-proto/Cargo.toml
===================================================================
--- ntp-proto.orig/Cargo.toml
+++ ntp-proto/Cargo.toml
@@ -52,11 +52,6 @@ version = "0.6"
 optional = true
 package = "rustls-native-certs"
 
-[dependencies.rustls-native-certs7]
-version = "0.7"
-optional = true
-package = "rustls-native-certs"
-
 [dependencies.rustls-pemfile1]
 version = "1.0"
 optional = true
