--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,10 +70,7 @@
     "ciborium",
     "serde",
     "chrono",
-    "wasm-bindgen",
-    "serde-wasm-bindgen",
     "clap",
-    "crossterm",
     "uriparse",
     "base64-url",
     "regex-syntax",
@@ -135,7 +132,7 @@
 version = "0.2.1"
 
 [dependencies.itertools]
-version = "0.14.0"
+version = "0"
 
 [dependencies.lexical-core]
 version = "1.0.5"
@@ -158,7 +155,7 @@
 default-features = false
 
 [dependencies.regex-syntax]
-version = "0.8.5"
+version = "0"
 optional = true
 
 [dependencies.serde]
@@ -191,31 +188,6 @@
 [dev-dependencies.wasm-bindgen-test]
 version = "0.3.34"
 
-[target.'cfg(not(target_arch = "wasm32"))'.dependencies.crossterm]
-version = "0.28.1"
-optional = true
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.console_error_panic_hook]
-version = "0.1.6"
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.js-sys]
-version = "0.3.64"
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.serde-wasm-bindgen]
-version = "0.6.5"
-optional = true
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen]
-version = "0.2"
-optional = true
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen-test]
-version = "0.3.34"
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.web-sys]
-version = "0.3.64"
-features = ["console"]
-
 [profile.release]
 opt-level = "s"
 lto = true
