--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,7 +64,7 @@
 default-features = false
 
 [dependencies.base64-url]
-version = "2.0.0"
+version = "3"
 optional = true
 
 [dependencies.chrono]
@@ -95,10 +95,10 @@
 version = "0.2.1"
 
 [dependencies.itertools]
-version = "0.11.0"
+version = "0"
 
 [dependencies.lexical-core]
-version = "0.8.3"
+version = "0"
 
 [dependencies.log]
 version = "0.4.14"
@@ -118,7 +118,7 @@
 default-features = false
 
 [dependencies.regex-syntax]
-version = "0.7.1"
+version = "0"
 optional = true
 
 [dependencies.serde]
@@ -169,32 +169,11 @@
     "ciborium",
     "serde",
     "chrono",
-    "wasm-bindgen",
-    "serde-wasm-bindgen",
     "clap",
-    "crossterm",
     "uriparse",
     "base64-url",
     "regex-syntax",
 ]
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.crossterm]
-version = "0.27.0"
-optional = true
-
-[target."cfg(target_arch = \"wasm32\")".dependencies.console_error_panic_hook]
-version = "0.1.6"
-
-[target."cfg(target_arch = \"wasm32\")".dependencies.serde-wasm-bindgen]
-version = "0.5.0"
-optional = true
-
-[target."cfg(target_arch = \"wasm32\")".dependencies.wasm-bindgen]
-version = "0.2"
-optional = true
-
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3.25"
-
 [badges.maintenance]
 status = "actively-developed"
