--- rust-smawk-0.3.1.orig/Cargo.toml
+++ rust-smawk-0.3.1/Cargo.toml
@@ -34,8 +31,6 @@ version = "0.8"
 [dev-dependencies.rand_chacha]
 version = "0.3"
 
-[dev-dependencies.version-sync]
-version = "0.9"
 [badges.appveyor]
 repository = "mgeisler/smawk"
 
--- rust-smawk-0.3.1.orig/tests/version-numbers.rs
+++ rust-smawk-0.3.1/tests/version-numbers.rs
@@ -1,4 +1,4 @@
-#[test]
+/*#[test]
 fn test_readme_deps() {
     version_sync::assert_markdown_deps_updated!("README.md");
 }
@@ -7,3 +7,4 @@ fn test_readme_deps() {
 fn test_html_root_url() {
     version_sync::assert_html_root_url_updated!("src/lib.rs");
 }
+*/
\ No newline at end of file
