--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,7 +35,7 @@
 optional = true
 
 [dev-dependencies.simple_logger]
-version = "4.2.0"
+version = "1"
 
 [features]
 default = []
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -124,14 +124,14 @@
 
             assert_eq!(&"Hello, there!", borrowed_thing);
 
-            SimpleLogger::new().init().unwrap_or(());
+            SimpleLogger::new().with_utc_timestamps().init().unwrap_or(());
 
             have_fun("Alice".to_string(), "Bob".to_string());
         }
 
         #[test]
         fn it_works_with_parameters() {
-            SimpleLogger::new().init().unwrap_or(());
+            SimpleLogger::new().with_utc_timestamps().init().unwrap_or(());
 
             have_fun_with_parameters(
                 Parameter {
