Index: rustix/Cargo.toml
===================================================================
--- rustix.orig/Cargo.toml
+++ rustix/Cargo.toml
@@ -45,6 +45,7 @@ categories = [
 ]
 license = "Apache-2.0 WITH LLVM-exception OR Apache-2.0 OR MIT"
 repository = "https://github.com/bytecodealliance/rustix"
+autobenches = false
 
 [package.metadata.docs.rs]
 features = ["all-apis"]
@@ -68,10 +69,6 @@ targets = [
     "wasm32-wasi",
 ]
 
-[[bench]]
-name = "mod"
-harness = false
-
 [dependencies.bitflags]
 version = "2.3.3"
 default-features = false
