--- a/Cargo.toml
+++ b/Cargo.toml
@@ -150,10 +150,10 @@
 name = "rustix"
 path = "src/lib.rs"
 
-[[bench]]
-name = "mod"
-path = "benches/mod.rs"
-harness = false
+#[[bench]]
+#name = "mod"
+#path = "benches/mod.rs"
+#harness = false
 
 [dependencies.bitflags]
 version = "2.4.0"
@@ -202,4 +202,4 @@
 
-[target.'cfg(all(criterion, not(any(target_os = "emscripten", target_os = "wasi"))))'.dev-dependencies.criterion]
-version = "0.4"
+#[target.'cfg(all(criterion, not(any(target_os = "emscripten", target_os = "wasi"))))'.dev-dependencies.criterion]
+#version = "0.4"
 
