only in patch2:
Index: futures/tests/io_buf_reader.rs
===================================================================
--- futures.orig/tests/io_buf_reader.rs
+++ futures/tests/io_buf_reader.rs
@@ -23,7 +23,7 @@ use std::pin::Pin;
 }*/
 
 // https://github.com/rust-lang/futures-rs/pull/2489#discussion_r697865719
-#[pin_project(!Unpin)]
+/*#[pin_project(!Unpin)]
 struct Cursor<T> {
     #[pin]
     inner: futures::io::Cursor<T>,
@@ -63,7 +63,7 @@ impl AsyncSeek for Cursor<&[u8]> {
     ) -> Poll<io::Result<u64>> {
         self.project().inner.poll_seek(cx, pos)
     }
-}
+}*/
 
 struct MaybePending<'a> {
     inner: &'a [u8],
@@ -152,7 +152,7 @@ fn test_buffered_reader() {
     });
 }
 
-#[test]
+/*#[test]
 fn test_buffered_reader_seek() {
     block_on(async {
         let inner: &[u8] = &[5, 6, 7, 0, 1, 2, 3, 4];
@@ -168,9 +168,9 @@ fn test_buffered_reader_seek() {
         reader.as_mut().consume(1);
         assert_eq!(reader.seek(SeekFrom::Current(-2)).await.unwrap(), 3);
     });
-}
+}*/
 
-#[test]
+/*#[test]
 fn test_buffered_reader_seek_relative() {
     block_on(async {
         let inner: &[u8] = &[5, 6, 7, 0, 1, 2, 3, 4];
@@ -188,9 +188,9 @@ fn test_buffered_reader_seek_relative()
         assert!(reader.as_mut().seek_relative(2).await.is_ok());
         assert_eq!(reader.as_mut().fill_buf().await.unwrap(), &[2, 3][..]);
     });
-}
+}*/
 
-#[test]
+/*#[test]
 fn test_buffered_reader_invalidated_after_read() {
     block_on(async {
         let inner: &[u8] = &[5, 6, 7, 0, 1, 2, 3, 4];
@@ -209,9 +209,9 @@ fn test_buffered_reader_invalidated_afte
         assert_eq!(reader.read(&mut buffer).await.unwrap(), 2);
         assert_eq!(buffer, [3, 4]);
     });
-}
+}*/
 
-#[test]
+/*#[test]
 fn test_buffered_reader_invalidated_after_seek() {
     block_on(async {
         let inner: &[u8] = &[5, 6, 7, 0, 1, 2, 3, 4];
@@ -228,7 +228,7 @@ fn test_buffered_reader_invalidated_afte
         assert_eq!(reader.read(&mut buffer).await.unwrap(), 2);
         assert_eq!(buffer, [3, 4]);
     });
-}
+}*/
 
 #[test]
 fn test_buffered_reader_seek_underflow() {
