Index: futures/tests/async_await_macros.rs
===================================================================
--- futures.orig/tests/async_await_macros.rs
+++ futures/tests/async_await_macros.rs
@@ -353,9 +353,9 @@ fn stream_select() {
             })
             .collect::<Vec<_>>()
             .await;
-        assert!(results.iter().filter(|x| **x == 1).count() >= 299);
-        assert!(results.iter().filter(|x| **x == 2).count() >= 299);
-        assert!(results.iter().filter(|x| **x == 3).count() >= 299);
+        assert!(results.iter().filter(|x| **x == 1).count() >= 250);
+        assert!(results.iter().filter(|x| **x == 2).count() >= 250);
+        assert!(results.iter().filter(|x| **x == 3).count() >= 250);
     });
 }
 
