Index: futures/Cargo.toml
===================================================================
--- futures.orig/Cargo.toml
+++ futures/Cargo.toml
@@ -323,9 +323,6 @@ version = "1.0.11"
 [dev-dependencies.static_assertions]
 version = "1"
 
-[dev-dependencies.tokio]
-version = "0.1.11"
-
 [features]
 alloc = [
     "futures-core/alloc",
@@ -339,11 +336,6 @@ async-await = [
     "futures-util/async-await-macro",
 ]
 bilock = ["futures-util/bilock"]
-cfg-target-has-atomic = []
-compat = [
-    "std",
-    "futures-util/compat",
-]
 default = [
     "std",
     "async-await",
@@ -353,10 +345,6 @@ executor = [
     "std",
     "futures-executor/std",
 ]
-io-compat = [
-    "compat",
-    "futures-util/io-compat",
-]
 std = [
     "alloc",
     "futures-core/std",
