Index: futures/tests/_require_features.rs
===================================================================
--- futures.orig/tests/_require_features.rs
+++ futures/tests/_require_features.rs
@@ -2,8 +2,6 @@
     feature = "std",
     feature = "alloc",
     feature = "async-await",
-    feature = "compat",
-    feature = "io-compat",
     feature = "executor",
     feature = "thread-pool",
 )))]
