Index: xterm-query/Cargo.toml
===================================================================
--- xterm-query.orig/Cargo.toml
+++ xterm-query/Cargo.toml
@@ -44,8 +44,8 @@ path = "examples/kitty/main.rs"
 version = "1.0"
 
 [dev-dependencies.crossterm]
-version = "0.21"
+version = ">= 0.27, < 0.30"
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.29"
+version = ">= 0.29, < 1.0"
 features = ["poll"]
