From 59dbd6a1cdb2bd3325f644d94da8e08ff7958ef2 Mon Sep 17 00:00:00 2001
From: Konstantin Senkevich <konstsen@gmail.com>
Date: Mon, 24 Aug 2020 16:36:01 +0300
Subject: [PATCH 08/19] Changelog note.

---
 CHANGELOG | 1 +
 1 file changed, 1 insertion(+)

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -12,6 +12,7 @@
 - Set default axis bounds for case where all values are equal (Issue #36)
 - Optimise tick generation to jump straight to range (Issue #42)
 - Fix text rendering of scatter plots
+- Replace failure dependency with thiserror (Issue #63).
 
 ## 0.5.0 - 2020-03-14
 ### Added
