--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@
 # will likely look very different (and much more reasonable)
 
 [package]
-edition = "2018"
+edition = "2021"
 name = "plotlib"
 version = "0.5.1"
 authors = ["Matt Williams <matt@milliams.com>"]
@@ -21,10 +21,9 @@
 categories = ["visualization", "science"]
 license = "MIT"
 repository = "https://github.com/milliams/plotlib"
-[dependencies.failure]
-version = "0.1.7"
+
+[dependencies.thiserror]
+version = ">= 1.0.30"
 
 [dependencies.svg]
-version = "0.7.1"
-[badges.travis-ci]
-repository = "milliams/plotlib"
+version = ">= 0.13.0"
