diff --git a/Cargo.toml b/Cargo.toml
index 4a466f3..cedcfce 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -111,10 +111,10 @@ name = "test_derive_input"
 path = "tests/test_derive_input.rs"
 required-features = ["full", "extra-traits", "derive"]
 
-[[test]]
-name = "test_expr"
-path = "tests/test_expr.rs"
-required-features = ["full", "extra-traits", "derive"]
+#[[test]]
+#name = "test_expr"
+#path = "tests/test_expr.rs"
+#required-features = ["full", "extra-traits", "derive"]
 
 [[test]]
 name = "test_generics"
@@ -206,10 +206,10 @@ name = "test_ty"
 path = "tests/test_ty.rs"
 required-features = ["full", "extra-traits", "derive"]
 
-[[test]]
-name = "test_unparenthesize"
-path = "tests/test_unparenthesize.rs"
-required-features = ["full", "visit-mut", "parsing", "derive"]
+#[[test]]
+#name = "test_unparenthesize"
+#path = "tests/test_unparenthesize.rs"
+#required-features = ["full", "visit-mut", "parsing", "derive"]
 
 [[test]]
 name = "test_visibility"
