Index: syn/tests/test_unparenthesize.rs
===================================================================
--- syn.orig/tests/test_unparenthesize.rs
+++ syn/tests/test_unparenthesize.rs
@@ -15,7 +15,7 @@ mod macros;
 
 mod repo;
 
-#[test]
+/*#[test]
 fn test_unparenthesize() {
     repo::rayon_init();
     repo::clone_rust();
@@ -28,7 +28,7 @@ fn test_unparenthesize() {
     if failed > 0 {
         panic!("{} failures", failed);
     }
-}
+}*/
 
 struct FlattenParens;
 
