diff --git a/tests/test_unparenthesize.rs b/tests/test_unparenthesize.rs
index 5fa2e59..7974200 100644
--- a/tests/test_unparenthesize.rs
+++ b/tests/test_unparenthesize.rs
@@ -23,7 +23,7 @@ mod macros;
 mod common;
 mod repo;
 
-#[test]
+/*#[test]
 fn test_unparenthesize() {
     repo::rayon_init();
     repo::clone_rust();
@@ -36,7 +36,7 @@ fn test_unparenthesize() {
     if failed > 0 {
         panic!("{} failures", failed);
     }
-}
+}*/
 
 fn test(path: &Path, failed: &AtomicUsize) {
     let content = fs::read_to_string(path).unwrap();
