--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,7 +16,6 @@ name = "syn"
 version = "2.0.18"
 authors = ["David Tolnay <dtolnay@gmail.com>"]
 include = [
-    "/benches/**",
     "/Cargo.toml",
     "/LICENSE-APACHE",
     "/LICENSE-MIT",
@@ -55,30 +54,13 @@ features = [
     "extra-traits",
 ]
 
-[lib]
-doc-scrape-examples = false
-
-[[bench]]
-name = "rust"
-harness = false
-required-features = [
-    "full",
-    "parsing",
-]
-
-[[bench]]
-name = "file"
-required-features = [
-    "full",
-    "parsing",
-]
 
 [dependencies.proc-macro2]
-version = "1.0.59"
+version = "1.0.58"
 default-features = false
 
 [dependencies.quote]
-version = "1.0.28"
+version = "1.0.27"
 optional = true
 default-features = false
 
@@ -113,9 +95,6 @@ features = ["blocking"]
 [dev-dependencies.rustversion]
 version = "1"
 
-[dev-dependencies.syn-test-suite]
-version = "0"
-
 [dev-dependencies.tar]
 version = "0.4.16"
 
@@ -144,6 +123,5 @@ proc-macro = [
     "proc-macro2/proc-macro",
     "quote/proc-macro",
 ]
-test = ["syn-test-suite/all-features"]
 visit = []
 visit-mut = []
