Index: syn/Cargo.toml
===================================================================
--- syn.orig/Cargo.toml
+++ syn/Cargo.toml
@@ -131,18 +131,10 @@ name = "test_path"
 path = "tests/test_path.rs"
 
 [[test]]
-name = "test_precedence"
-path = "tests/test_precedence.rs"
-
-[[test]]
 name = "test_receiver"
 path = "tests/test_receiver.rs"
 
 [[test]]
-name = "test_round_trip"
-path = "tests/test_round_trip.rs"
-
-[[test]]
 name = "test_shebang"
 path = "tests/test_shebang.rs"
 
@@ -180,15 +172,6 @@ path = "benches/file.rs"
 required-features = [
     "full",
     "parsing",
-]
-
-[[bench]]
-name = "rust"
-path = "benches/rust.rs"
-harness = false
-required-features = [
-    "full",
-    "parsing",
 ]
 
 [dependencies.proc-macro2]
