diff --git a/Cargo.toml b/Cargo.toml
index 8293b96..17612e4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -80,7 +80,6 @@
     "proc-macro2/proc-macro",
     "quote?/proc-macro",
 ]
-test = ["syn-test-suite/all-features"]
 visit = []
 visit-mut = []
 
@@ -245,9 +244,6 @@
 [dev-dependencies.rustversion]
 version = "1"
 
-[dev-dependencies.syn-test-suite]
-version = "0"
-
 [dev-dependencies.termcolor]
 version = "1"
 
