diff --git a/Cargo.toml b/Cargo.toml
index c61c9bf..27e2f4b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -239,9 +239,9 @@ version = "1"
 [target."cfg(not(miri))".dev-dependencies.rayon]
 version = "1"
 
-[target."cfg(not(miri))".dev-dependencies.reqwest]
-version = "0.12"
-features = ["blocking"]
+#[target."cfg(not(miri))".dev-dependencies.reqwest]
+#version = "0.12"
+#features = ["blocking"]
 
 [target."cfg(not(miri))".dev-dependencies.tar]
 version = "0.4.16"
diff --git a/benches/file.rs b/benches/file.rs
index 6167488..8ecfc06 100644
--- a/benches/file.rs
+++ b/benches/file.rs
@@ -1,3 +1,4 @@
+/*
 // $ cargo bench --features full,test --bench file
 
 #![feature(rustc_private, test)]
@@ -56,4 +57,4 @@ fn create_token_buffer(b: &mut Bencher) {
 fn parse_file(b: &mut Bencher) {
     let tokens = get_tokens();
     b.iter(|| syn::parse2::<syn::File>(tokens.clone()));
-}
+}*/
diff --git a/benches/rust.rs b/benches/rust.rs
index b5edab1..2e0ecbf 100644
--- a/benches/rust.rs
+++ b/benches/rust.rs
@@ -1,3 +1,7 @@
+fn main() {
+    eprintln!("Debian: this bench is disabled due to dependency on reqwest.\n\
+    To run it, please unapply debian/patches/remove-reqwest.patch.");
+/*
 // $ cargo bench --features full,test --bench rust
 //
 // Syn only, useful for profiling:
@@ -184,5 +188,5 @@ fn main() {
             elapsed.subsec_millis(),
         );
     }
-    eprintln!();
+    eprintln!();*/
 }
diff --git a/tests/repo/mod.rs b/tests/repo/mod.rs
index 8157345..ee2fa7e 100644
--- a/tests/repo/mod.rs
+++ b/tests/repo/mod.rs
@@ -488,6 +488,7 @@ pub fn rayon_init() {
 }
 
 pub fn clone_rust() {
+    return;
     let needs_clone = match fs::read_to_string("tests/rust/COMMIT") {
         Err(_) => true,
         Ok(contents) => contents.trim() != REVISION,
@@ -533,7 +534,7 @@ pub fn clone_rust() {
 }
 
 fn download_and_unpack() -> Result<()> {
-    let url = format!(
+    /*let url = format!(
         "https://github.com/rust-lang/rust/archive/{}.tar.gz",
         REVISION
     );
@@ -559,6 +560,6 @@ fn download_and_unpack() -> Result<()> {
         entry.unpack(&out)?;
     }
 
-    fs::write("tests/rust/COMMIT", REVISION)?;
+    fs::write("tests/rust/COMMIT", REVISION)?;*/
     Ok(())
 }
diff --git a/tests/test_precedence.rs b/tests/test_precedence.rs
index 33af274..2338767 100644
--- a/tests/test_precedence.rs
+++ b/tests/test_precedence.rs
@@ -62,7 +62,7 @@ mod repo;
 #[path = "../src/scan_expr.rs"]
 mod scan_expr;
 
-#[test]
+//#[test]
 fn test_rustc_precedence() {
     repo::rayon_init();
     repo::clone_rust();
diff --git a/tests/test_round_trip.rs b/tests/test_round_trip.rs
index 3657da2..18474d2 100644
--- a/tests/test_round_trip.rs
+++ b/tests/test_round_trip.rs
@@ -50,7 +50,7 @@ mod macros;
 mod common;
 mod repo;
 
-#[test]
+//#[test]
 fn test_round_trip() {
     repo::rayon_init();
     repo::clone_rust();
