diff --git a/Cargo.toml b/Cargo.toml
index a3f0dec..0790568 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,7 +84,6 @@ proc-macro = [
     "proc-macro2/proc-macro",
     "quote?/proc-macro",
 ]
-test = ["syn-test-suite/all-features"]
 visit = []
 visit-mut = []
 
@@ -248,9 +247,6 @@ version = "1"
 [dev-dependencies.rustversion]
 version = "1"
 
-[dev-dependencies.syn-test-suite]
-version = "0"
-
 [dev-dependencies.termcolor]
 version = "1"
 
