--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,8 +44,8 @@
 name = "image_webp"
 path = "src/lib.rs"
 
-[dependencies.byteorder-lite]
-version = "0.1.0"
+[dependencies.byteorder]
+version = "1.5"
 
 [dependencies.quick-error]
 version = "2.0.1"
--- a/src/decoder.rs
+++ b/src/decoder.rs
@@ -1,4 +1,4 @@
-use byteorder_lite::{LittleEndian, ReadBytesExt};
+use byteorder::{LittleEndian, ReadBytesExt};
 use quick_error::quick_error;
 
 use std::collections::HashMap;
--- a/src/extended.rs
+++ b/src/extended.rs
@@ -1,6 +1,6 @@
 use super::lossless::LosslessDecoder;
 use crate::decoder::DecodingError;
-use byteorder_lite::ReadBytesExt;
+use byteorder::ReadBytesExt;
 use std::io::{BufRead, Read};
 
 use crate::alpha_blending::do_alpha_blending;
--- a/src/vp8.rs
+++ b/src/vp8.rs
@@ -10,7 +10,7 @@
 //! * [rfc-6386](http://tools.ietf.org/html/rfc6386) - The VP8 Data Format and Decoding Guide
 //! * [VP8.pdf](http://static.googleusercontent.com/media/research.google.com/en//pubs/archive/37073.pdf) - An overview of of the VP8 format
 
-use byteorder_lite::{LittleEndian, ReadBytesExt};
+use byteorder::{LittleEndian, ReadBytesExt};
 use std::default::Default;
 use std::io::Read;
 
