Index: sha3/Cargo.toml
===================================================================
--- sha3.orig/Cargo.toml
+++ sha3/Cargo.toml
@@ -21,21 +21,21 @@ categories = ["cryptography", "no-std"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/RustCrypto/hashes"
 [dependencies.block-buffer]
-version = "0.7"
+version = "0.9"
 
 [dependencies.byte-tools]
 version = "0.3"
 
 [dependencies.digest]
-version = "0.8"
+version = "0.9"
 
 [dependencies.keccak]
 version = "0.1"
 
 [dependencies.opaque-debug]
-version = "0.2"
+version = "0.3"
 [dev-dependencies.digest]
-version = "0.8"
+version = "0.9"
 features = ["dev"]
 
 [dev-dependencies.hex-literal]
