--- a/Cargo.toml
+++ b/Cargo.toml
@@ -183,8 +183,6 @@
 version = "2"
 
 [features]
-aws-lc-rs = ["proto/aws-lc-rs"]
-aws-lc-rs-fips = ["proto/aws-lc-rs-fips"]
 default = [
     "log",
     "platform-verifier",
@@ -213,18 +211,6 @@
     "tokio/net",
 ]
 rustls = ["rustls-ring"]
-rustls-aws-lc-rs = [
-    "dep:rustls",
-    "aws-lc-rs",
-    "proto/rustls-aws-lc-rs",
-    "proto/aws-lc-rs",
-]
-rustls-aws-lc-rs-fips = [
-    "dep:rustls",
-    "aws-lc-rs-fips",
-    "proto/rustls-aws-lc-rs-fips",
-    "proto/aws-lc-rs-fips",
-]
 rustls-log = ["rustls?/logging"]
 rustls-ring = [
     "dep:rustls",
