Index: quinn/Cargo.toml
===================================================================
--- quinn.orig/Cargo.toml
+++ quinn/Cargo.toml
@@ -54,7 +54,7 @@ harness = false
 required-features = ["tls-rustls"]
 
 [dependencies.async-io]
-version = "1.6"
+version = "2"
 optional = true
 
 [dependencies.async-std]
@@ -114,13 +114,14 @@ features = ["derive"]
 version = "3"
 
 [dev-dependencies.directories-next]
-version = "2"
+version = "5"
+package = "directories"
 
 [dev-dependencies.rand]
 version = "0.8"
 
 [dev-dependencies.rcgen]
-version = "0.10.0"
+version = "0.11.0"
 
 [dev-dependencies.rustls-pemfile]
 version = "1.0.0"
