Origin: vendor
Subject: patch out Windows dependencies
Forwarded: not-needed

this should be handled automatically by debcargo...

Index: sequoia-keystore-gpg-agent/Cargo.toml
===================================================================
--- sequoia-keystore-gpg-agent.orig/Cargo.toml
+++ sequoia-keystore-gpg-agent/Cargo.toml
@@ -106,11 +106,3 @@ features = [
     "__implicit-crypto-backend-for-tests",
 ]
 default-features = false
-
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "2"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
