--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,11 +37,9 @@
     "dep:objc2-core-graphics",
     "dep:objc2-core-foundation",
     "image",
-    "windows-sys",
     "core-graphics",
 ]
 wayland-data-control = ["wl-clipboard-rs"]
-windows-sys = ["windows-sys/Win32_Graphics_Gdi"]
 wl-clipboard-rs = ["dep:wl-clipboard-rs"]
 
 [lib]
@@ -149,29 +147,4 @@
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.clipboard-win]
-version = "5.3.1"
-features = ["std"]
 
-[target."cfg(windows)".dependencies.image]
-version = "0.25"
-features = [
-    "png",
-    "bmp",
-]
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.log]
-version = "0.4"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = ">=0.52.0, <0.61.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_System_DataExchange",
-    "Win32_System_Memory",
-    "Win32_System_Ole",
-    "Win32_UI_Shell",
-]
