--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,14 +30,10 @@
 repository = "https://github.com/1Password/arboard"
 
 [features]
-core-graphics = ["dep:objc2-core-graphics"]
 default = ["image-data"]
 image = ["dep:image"]
 image-data = [
-    "dep:objc2-core-graphics",
-    "dep:objc2-core-foundation",
     "image",
-    "core-graphics",
 ]
 wayland-data-control = ["wl-clipboard-rs"]
 wl-clipboard-rs = ["dep:wl-clipboard-rs"]
@@ -95,16 +91,16 @@
 [target.'cfg(all(unix, not(any(target_os="macos", target_os="android", target_os="emscripten"))))'.dependencies.x11rb]
 version = "0.13"
 
-[target.'cfg(target_os = "macos")'.dependencies.image]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.image]
 version = "0.25"
 features = ["tiff"]
 optional = true
 default-features = false
 
-[target.'cfg(target_os = "macos")'.dependencies.objc2]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2]
 version = "0.6.0"
 
-[target.'cfg(target_os = "macos")'.dependencies.objc2-app-kit]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-app-kit]
 version = "0.3.0"
 features = [
     "std",
@@ -115,7 +111,7 @@
 ]
 default-features = false
 
-[target.'cfg(target_os = "macos")'.dependencies.objc2-core-foundation]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-core-foundation]
 version = "0.3.0"
 features = [
     "std",
@@ -124,7 +120,7 @@
 optional = true
 default-features = false
 
-[target.'cfg(target_os = "macos")'.dependencies.objc2-core-graphics]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-core-graphics]
 version = "0.3.0"
 features = [
     "std",
@@ -135,7 +131,7 @@
 optional = true
 default-features = false
 
-[target.'cfg(target_os = "macos")'.dependencies.objc2-foundation]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-foundation]
 version = "0.3.0"
 features = [
     "std",
