--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68 +68 @@
-core-graphics = ["dep:objc2-core-graphics"]
+#core-graphics = ["dep:objc2-core-graphics"]
@@ -72,2 +72,2 @@
-    "dep:objc2-core-graphics",
-    "dep:objc2-core-foundation",
+    #"dep:objc2-core-graphics",
+    #"dep:objc2-core-foundation",
@@ -76 +76 @@
-    "core-graphics",
+    #"core-graphics",
@@ -104 +104 @@
-[target.'cfg(target_os = "macos")'.dependencies.image]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.image]
@@ -110 +110 @@
-[target.'cfg(target_os = "macos")'.dependencies.objc2]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2]
@@ -113 +113 @@
-[target.'cfg(target_os = "macos")'.dependencies.objc2-app-kit]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-app-kit]
@@ -124 +124 @@
-[target.'cfg(target_os = "macos")'.dependencies.objc2-core-foundation]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-core-foundation]
@@ -133 +133 @@
-[target.'cfg(target_os = "macos")'.dependencies.objc2-core-graphics]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-core-graphics]
@@ -144 +144 @@
-[target.'cfg(target_os = "macos")'.dependencies.objc2-foundation]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.objc2-foundation]
