--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47 +47 @@
-    "core-graphics",
+#    "core-graphics",
@@ -72,30 +72,30 @@
-[target."cfg(target_os = \"macos\")".dependencies.core-graphics]
-version = "0.23"
-optional = true
-
-[target."cfg(target_os = \"macos\")".dependencies.image]
-version = "0.25"
-features = ["tiff"]
-optional = true
-default-features = false
-
-[target."cfg(target_os = \"macos\")".dependencies.objc2]
-version = "0.5.1"
-features = ["relax-void-encoding"]
-
-[target."cfg(target_os = \"macos\")".dependencies.objc2-app-kit]
-version = "0.2.0"
-features = [
-    "NSPasteboard",
-    "NSPasteboardItem",
-    "NSImage",
-]
-
-[target."cfg(target_os = \"macos\")".dependencies.objc2-foundation]
-version = "0.2.0"
-features = [
-    "NSArray",
-    "NSString",
-    "NSEnumerator",
-    "NSGeometry",
-]
+#[target."cfg(target_os = \"macos\")".dependencies.core-graphics]
+#version = "0.23"
+#optional = true
+
+#[target."cfg(target_os = \"macos\")".dependencies.image]
+#version = "0.25"
+#features = ["tiff"]
+#optional = true
+#default-features = false
+
+#[target."cfg(target_os = \"macos\")".dependencies.objc2]
+#version = "0.5.1"
+#features = ["relax-void-encoding"]
+
+#[target."cfg(target_os = \"macos\")".dependencies.objc2-app-kit]
+#version = "0.2.0"
+#features = [
+#    "NSPasteboard",
+#    "NSPasteboardItem",
+#    "NSImage",
+#]
+
+#[target."cfg(target_os = \"macos\")".dependencies.objc2-foundation]
+#version = "0.2.0"
+#features = [
+#    "NSArray",
+#    "NSString",
+#    "NSEnumerator",
+#    "NSGeometry",
+#]
