--- a/Cargo.toml
+++ b/Cargo.toml
@@ -75 +75 @@
-    "windows-sys",
+    #"windows-sys",
@@ -79 +79 @@
-windows-sys = ["dep:windows-sys"]
+#windows-sys = ["dep:windows-sys"]
@@ -156 +156 @@
-[target."cfg(windows)".dependencies.clipboard-win]
+[disabled.target."cfg(windows)".dependencies.clipboard-win]
@@ -160 +160 @@
-[target."cfg(windows)".dependencies.image]
+[disabled.target."cfg(windows)".dependencies.image]
@@ -166 +166 @@
-[target."cfg(windows)".dependencies.log]
+[disabled.target."cfg(windows)".dependencies.log]
@@ -169 +169 @@
-[target."cfg(windows)".dependencies.windows-sys]
+[disabled.target."cfg(windows)".dependencies.windows-sys]
