--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,3 +53,3 @@
 [target."cfg(all(unix, not(any(target_os=\"macos\", target_os=\"android\", target_os=\"emscripten\"))))".dependencies.image]
-version = "0.25"
+version = "0.24"
 features = ["png"]
--- a/src/platform/linux/mod.rs
+++ b/src/platform/linux/mod.rs
@@ -32,5 +32,5 @@
 			image.width as u32,
 			image.height as u32,
-			image::ExtendedColorType::Rgba8,
+			image::ColorType::Rgba8,
 		)
 		.map_err(|_| Error::ConversionFailure)?;
