Index: flate2/src/lib.rs
===================================================================
--- flate2.orig/src/lib.rs
+++ flate2/src/lib.rs
@@ -91,7 +91,7 @@
 #![deny(missing_docs)]
 #![deny(missing_debug_implementations)]
 #![allow(trivial_numeric_casts)]
-#![cfg_attr(test, deny(warnings))]
+//#![cfg_attr(test, deny(warnings))]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
 
 #[cfg(not(feature = "any_impl",))]
