--- a/Cargo.toml
+++ b/Cargo.toml
@@ -110,10 +110,10 @@
 optional = true
 
 [dependencies.rand]
-version = "0.9"
+version = "0.8"
 features = [
     "std",
-    "thread_rng",
+    "std_rng",
 ]
 default-features = false
 
--- a/src/desktop/handle_token.rs
+++ b/src/desktop/handle_token.rs
@@ -3,7 +3,7 @@
     fmt::{self, Debug, Display},
 };
 
-use rand::prelude::*;
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
 use serde::{Deserialize, Serialize};
 #[cfg(feature = "backend")]
 use zbus::zvariant::OwnedObjectPath;
@@ -38,9 +38,9 @@
 
 impl Default for HandleToken {
     fn default() -> Self {
-        let mut rng = rand::rng();
+        let mut rng = thread_rng();
         let token: String = (&mut rng)
-            .sample_iter(rand::distr::Alphanumeric)
+            .sample_iter(Alphanumeric)
             .take(10)
             .map(char::from)
             .collect();
