Index: ashpd/Cargo.toml
===================================================================
--- ashpd.orig/Cargo.toml
+++ ashpd/Cargo.toml
@@ -48,11 +48,6 @@ version = "0.3"
 [dependencies.futures-util]
 version = "0.3"
 
-[dependencies.gdk3wayland]
-version = "0.17"
-optional = true
-package = "gdkwayland"
-
 [dependencies.gdk3x11]
 version = "0.17"
 optional = true
@@ -155,14 +150,6 @@ async-std = [
     "dep:async-std",
 ]
 default = ["async-std"]
-gtk3 = [
-    "gtk3_x11",
-    "gtk3_wayland",
-]
-gtk3_wayland = [
-    "gdk3wayland",
-    "dep:gtk3",
-]
 gtk3_x11 = [
     "gdk3x11",
     "dep:gtk3",
