From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Wed, 21 Aug 2024 17:19:54 -0400
Subject: relax deps

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index d8b8739..2157e56 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -132,7 +132,7 @@ version = "0.31"
 optional = true
 
 [dependencies.wayland-protocols]
-version = "0.32"
+version = "0.31"
 features = [
     "unstable",
     "client",
