--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,7 @@ repository = "https://github.com/bilelmo
 version = "0.6"
 
 [dependencies.gdk]
-version = "0.13"
+version = "0.14"
 optional = true
 
 [dependencies.serde]
@@ -42,14 +42,14 @@ version = "0.19"
 version = "0.19"
 
 [dependencies.zbus]
-version = "1.1"
+version = "1.9"
 
 [dependencies.zvariant]
-version = "2.2"
+version = "2"
 features = ["enumflags2"]
 
 [dependencies.zvariant_derive]
-version = "2.2"
+version = "2"
 
 [features]
 gdk_color = ["gdk"]
