--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,32 +49,32 @@ version = "0.3"
 version = "0.3"
 
 [dependencies.gdk3wayland]
-version = "0.17"
+version = "0.16"
 optional = true
 package = "gdkwayland"
 
 [dependencies.gdk3x11]
-version = "0.17"
+version = "0.16"
 optional = true
 package = "gdkx11"
 
 [dependencies.gdk4wayland]
-version = "0.6"
+version = "0.5"
 optional = true
 package = "gdk4-wayland"
 
 [dependencies.gdk4x11]
-version = "0.6"
+version = "0.5"
 optional = true
 package = "gdk4-x11"
 
 [dependencies.gtk3]
-version = "0.17"
+version = "0.16"
 optional = true
 package = "gtk"
 
 [dependencies.gtk4]
-version = "0.6"
+version = "0.5"
 optional = true
 
 [dependencies.libc]
