Index: ashpd/src/backend/builder.rs
===================================================================
--- ashpd.orig/src/backend/builder.rs
+++ ashpd/src/backend/builder.rs
@@ -168,7 +168,6 @@ impl Builder {
 
     pub async fn build(self) -> Result<()> {
         let cnx = zbus::Connection::session().await?;
-        cnx.request_name_with_flags(self.name, self.flags).await?;
 
         #[cfg(feature = "tokio")]
         let spawn = self.spawn.unwrap_or(Arc::new(super::spawn::TokioSpawner));
@@ -319,6 +318,8 @@ impl Builder {
                 .await?;
         }
 
+        cnx.request_name_with_flags(self.name, self.flags).await?;
+
         Ok(())
     }
 }
