Index: bindgen/Cargo.toml
===================================================================
--- bindgen.orig/Cargo.toml
+++ bindgen/Cargo.toml
@@ -37,13 +37,13 @@ required-features = ["clap"]
 version = "1.0.3"
 
 [dependencies.cexpr]
-version = "0.3.3"
+version = "0.4"
 
 [dependencies.cfg-if]
 version = "0.1.0"
 
 [dependencies.clang-sys]
-version = "0.28.0"
+version = "1"
 features = ["runtime", "clang_6_0"]
 
 [dependencies.clap]
@@ -51,7 +51,7 @@ version = "2"
 optional = true
 
 [dependencies.env_logger]
-version = "0.6"
+version = "0.7"
 optional = true
 
 [dependencies.lazy_static]
