Index: arrayvec/Cargo.toml
===================================================================
--- arrayvec.orig/Cargo.toml
+++ arrayvec/Cargo.toml
@@ -59,10 +59,6 @@ name = "arrayvec"
 path = "src/lib.rs"
 
 [[test]]
-name = "borsh"
-path = "tests/borsh.rs"
-
-[[test]]
 name = "serde"
 path = "tests/serde.rs"
 
@@ -80,11 +76,6 @@ name = "extend"
 path = "benches/extend.rs"
 harness = false
 
-[dependencies.borsh]
-version = "1.2.0"
-optional = true
-default-features = false
-
 [dependencies.serde]
 version = "1.0"
 optional = true
