--- a/Cargo.toml
+++ b/Cargo.toml
@@ -93,7 +93,7 @@
 version = "2.4.1"
 
 [dependencies.image]
-version = "0.25.7"
+version = "0.25"
 optional = true
 default-features = false
 
@@ -111,7 +111,7 @@
 package = "memmap2"
 
 [dependencies.nix]
-version = "0.30.0"
+version = "0.29.0"
 features = [
     "fs",
     "resource",
@@ -131,7 +131,7 @@
 features = ["derive"]
 
 [dependencies.thiserror]
-version = "2.0.3"
+version = "1"
 
 [dependencies.tokio]
 version = "1.35.1"
@@ -143,7 +143,7 @@
 optional = true
 
 [dependencies.zbus]
-version = "5.10.0"
+version = "5.6.0"
 features = ["p2p"]
 default-features = false
 
