--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@
 
 [package]
 edition = "2021"
-rust-version = "1.70"
+rust-version = "1.66"
 name = "glycin-utils"
 version = "0.1.0-beta.4"
 authors = ["Sophie Herold"]
@@ -39,11 +39,11 @@ version = "0.7.0"
 features = ["gettext-system"]
 
 [dependencies.image]
-version = "0.24.7"
+version = "0.24"
 optional = true
 
 [dependencies.memmap]
-version = "0.7.0"
+version = "0.5.0"
 package = "memmap2"
 
 [dependencies.nix]
@@ -54,7 +54,7 @@ version = "1.0.162"
 features = ["derive"]
 
 [dependencies.zbus]
-version = "3.13.1"
+version = "3.12"
 
 [features]
 image-rs = ["image"]

