diff --git a/Cargo.toml b/Cargo.toml
index 63fbab6..6354a00 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -101,7 +101,7 @@ default-features = false
 version = "0.2.152"
 
 [dependencies.libseccomp]
-version = "0.4.0"
+version = ">= 0.3.0"
 
 [dependencies.log]
 version = "0.4.0"
@@ -111,7 +111,7 @@ version = "0.9.0"
 package = "memmap2"
 
 [dependencies.nix]
-version = "0.30.0"
+version = ">= 0.29"
 features = [
     "fs",
     "resource",
@@ -143,7 +143,7 @@ features = [
 optional = true
 
 [dependencies.zbus]
-version = "5.10.0"
+version = ">= 5.3"
 features = ["p2p"]
 default-features = false
 
