Index: rust-libm-0.2.1/src/math/rem_pio2.rs
===================================================================
--- rust-libm-0.2.1.orig/src/math/rem_pio2.rs
+++ rust-libm-0.2.1/src/math/rem_pio2.rs
@@ -50,7 +50,9 @@ pub(crate) fn rem_pio2(x: f64) -> (i32,
 
     fn medium(x: f64, ix: u32) -> (i32, f64, f64) {
         /* rint(x/(pi/2)), Assume round-to-nearest. */
-        let f_n = x as f64 * INV_PIO2 + TO_INT - TO_INT;
+        //use to_bits and from_bits to force rounding to the storage format,
+        //otherwise this doesn't work on x87
+        let f_n = f64::from_bits((x as f64 * INV_PIO2 + TO_INT).to_bits()) - TO_INT;
         let n = f_n as i32;
         let mut r = x - f_n * PIO2_1;
         let mut w = f_n * PIO2_1T; /* 1st round, good to 85 bits */
