--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,10 +43,9 @@ version = "0.4.31"
 [dependencies.chrono-tz]
-version = "0.8.5"
+version = "0.10"
 
 [dependencies.codespan-reporting]
-version = "0.11"
+version = "0.12"
 
 [dependencies.heck]
-version = "0.4.1"
-features = ["unicode"]
+version = "0.5"
 
@@ -60,3 +59,3 @@ version = "0.1"
 [dependencies.itertools]
-version = "0.12"
+version = ">= 0.10"
 
@@ -66,6 +65,6 @@ version = "0.2.152"
 [dependencies.num-format]
-version = "0.4.4"
+version = ">= 0.4"
 
 [dependencies.num-integer]
-version = "0.1.45"
+version = ">= 0.1.44"
 
