Description: relax dependency versions
Author: Ananthu C V <weepingclown@disroot.org>
Author: Peter Michael Green <plugwash@debian.org>
Forwarded: not-needed
Last-Update: 2025-09-09

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,7 +34,9 @@
 features = ["serde"]
 
 [dependencies.ipnetwork]
-version = "0.20.0"
+version = ">= 0.17, < 0.22"
+default_features = false
+features = ["serde"]
 
 [dependencies.libc]
 version = "0.2.147"
@@ -45,7 +45,7 @@
 optional = true
 
 [dependencies.pcap]
-version = "1.1.0"
+version = "2.0.0"
 optional = true
 
 [dependencies.pnet_base]
