Index: reqsign/Cargo.toml
===================================================================
--- reqsign.orig/Cargo.toml
+++ reqsign/Cargo.toml
@@ -101,11 +101,6 @@ path = "src/lib.rs"
 name = "main"
 path = "tests/main.rs"
 
-[[bench]]
-name = "aws"
-path = "benches/aws.rs"
-harness = false
-
 [dependencies.anyhow]
 version = "1"
 
@@ -189,12 +184,6 @@ features = ["oid"]
 version = "0.8.9"
 optional = true
 
-[dev-dependencies.aws-credential-types]
-version = "1.1.8"
-
-[dev-dependencies.aws-sigv4]
-version = "1.2.0"
-
 [dev-dependencies.criterion]
 version = "0.5"
 features = [
Index: reqsign/src/aws/v4.rs
===================================================================
--- reqsign.orig/src/aws/v4.rs
+++ reqsign/src/aws/v4.rs
@@ -376,7 +376,7 @@ fn generate_signing_key(secret: &str, ti
     sign_request
 }
 
-#[cfg(test)]
+/*#[cfg(test)]
 mod tests {
     use std::time::SystemTime;
 
@@ -891,4 +891,4 @@ mod tests {
 
         Ok(())
     }
-}
+}*/
