Index: openssl-sys/Cargo.toml
===================================================================
--- openssl-sys.orig/Cargo.toml
+++ openssl-sys/Cargo.toml
@@ -37,18 +37,10 @@ repository = "https://github.com/sfackle
 [package.metadata.pkg-config]
 openssl = "1.0.1"
 
-[features]
-aws-lc = ["dep:aws-lc-sys"]
-
 [lib]
 name = "openssl_sys"
 path = "src/lib.rs"
 
-[dependencies.aws-lc-sys]
-version = "0.27"
-features = ["ssl"]
-optional = true
-
 [dependencies.libc]
 version = "0.2"
 
