Relax bindgen dependency to allow any 0.x version greater than 0.72.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@
 version = "0.2"
 
 [build-dependencies.bindgen]
-version = "0.69.0"
+version = ">= 0.72, < 1.0"
 features = ["experimental"]
 optional = true
 
--- a/build/run_bindgen.rs
+++ b/build/run_bindgen.rs
@@ -1,7 +1,7 @@
 #[cfg(feature = "bindgen")]
 use bindgen::callbacks::{MacroParsingBehavior, ParseCallbacks};
 #[cfg(feature = "bindgen")]
-use bindgen::{MacroTypeVariation, RustTarget};
+use bindgen::{callbacks::ItemInfo, MacroTypeVariation, RustTarget};
 use std::io::Write;
 use std::path::PathBuf;
 #[cfg(not(feature = "bindgen"))]
@@ -354,7 +354,8 @@
         MacroParsingBehavior::Ignore
     }
 
-    fn item_name(&self, original_item_name: &str) -> Option<String> {
+    fn item_name(&self, item_info: ItemInfo) -> Option<String> {
+        let original_item_name = item_info.name;
         match original_item_name {
             // Our original definitions of these are wrong, so rename to avoid breakage
             "CRYPTO_EX_new"
