From: =?UTF-8?q?Jeremy=20B=C3=ADcha?= <jbicha@ubuntu.com>
Date: Sat, 20 Sep 2025 10:48:10 -0400
Subject: Revert symbol additition from gstreamer 1.27

which isn't in Debian Unstable yet
to allow the cross_validate tests to keep passing
--- a/tests/abi.rs
+++ b/tests/abi.rs
@@ -1338,7 +1338,6 @@
     ("(gint) GST_VIDEO_FORMAT_AYUV64", "40"),
     ("(gint) GST_VIDEO_FORMAT_BGR", "16"),
     ("(gint) GST_VIDEO_FORMAT_BGR10A2_LE", "85"),
-    ("(gint) GST_VIDEO_FORMAT_BGR10x2_LE", "140"),
     ("(gint) GST_VIDEO_FORMAT_BGR15", "32"),
     ("(gint) GST_VIDEO_FORMAT_BGR16", "30"),
     ("(gint) GST_VIDEO_FORMAT_BGRA", "12"),
@@ -1386,7 +1386,7 @@
     ("(gint) GST_VIDEO_FORMAT_I422_12LE", "75"),
     ("(gint) GST_VIDEO_FORMAT_IYU1", "38"),
     ("(gint) GST_VIDEO_FORMAT_IYU2", "63"),
-    ("GST_VIDEO_FORMAT_LAST", "142"),
+    ("GST_VIDEO_FORMAT_LAST", "139"),
     ("(gint) GST_VIDEO_FORMAT_MT2110R", "116"),
     ("(gint) GST_VIDEO_FORMAT_MT2110T", "115"),
     ("(gint) GST_VIDEO_FORMAT_NV12", "23"),
@@ -1400,7 +1400,6 @@
     ("(gint) GST_VIDEO_FORMAT_NV12_8L128", "111"),
     ("(gint) GST_VIDEO_FORMAT_NV16", "51"),
     ("(gint) GST_VIDEO_FORMAT_NV16_10LE32", "80"),
-    ("(gint) GST_VIDEO_FORMAT_NV16_10LE40", "139"),
     ("(gint) GST_VIDEO_FORMAT_NV21", "24"),
     ("(gint) GST_VIDEO_FORMAT_NV24", "52"),
     ("(gint) GST_VIDEO_FORMAT_NV61", "60"),
@@ -1414,7 +1414,6 @@
     ("(gint) GST_VIDEO_FORMAT_RBGA", "133"),
     ("(gint) GST_VIDEO_FORMAT_RGB", "15"),
     ("(gint) GST_VIDEO_FORMAT_RGB10A2_LE", "86"),
-    ("(gint) GST_VIDEO_FORMAT_RGB10x2_LE", "141"),
     ("(gint) GST_VIDEO_FORMAT_RGB15", "31"),
     ("(gint) GST_VIDEO_FORMAT_RGB16", "29"),
     ("(gint) GST_VIDEO_FORMAT_RGB8P", "35"),
--- a/tests/constant.c
+++ b/tests/constant.c
@@ -288,7 +288,6 @@
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_AYUV64);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_BGR);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_BGR10A2_LE);
-    PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_BGR10x2_LE);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_BGR15);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_BGR16);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_BGRA);
@@ -350,7 +350,6 @@
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV12_8L128);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV16);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV16_10LE32);
-    PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV16_10LE40);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV21);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV24);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_NV61);
@@ -364,7 +363,6 @@
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RBGA);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RGB);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RGB10A2_LE);
-    PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RGB10x2_LE);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RGB15);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RGB16);
     PRINT_CONSTANT((gint) GST_VIDEO_FORMAT_RGB8P);
