Index: merge/examples/args.rs
===================================================================
--- merge.orig/examples/args.rs
+++ merge/examples/args.rs
@@ -1,5 +1,6 @@
 // SPDX-FileCopyrightText: 2020 Robin Krahl <robin.krahl@ireas.org>
 // SPDX-License-Identifier: CC0-1.0
+#![cfg(feature = "derive")]
 
 use merge::Merge;
 use serde::Deserialize;
Index: merge/examples/user.rs
===================================================================
--- merge.orig/examples/user.rs
+++ merge/examples/user.rs
@@ -1,6 +1,6 @@
 // SPDX-FileCopyrightText: 2020 Robin Krahl <robin.krahl@ireas.org>
 // SPDX-License-Identifier: CC0-1.0
-
+#![cfg(all(feature = "derive", feature = "std"))]
 use merge::Merge;
 
 #[derive(Merge)]
Index: merge/tests/derive.rs
===================================================================
--- merge.orig/tests/derive.rs
+++ merge/tests/derive.rs
@@ -550,6 +550,7 @@ fn test_unnamed_fields_skip() {
     test(S::new(None, None), S::new(None, None), S::new(None, None));
 }
 
+#[cfg(feature = "num")]
 #[test]
 fn test_default_strategy() {
     #[derive(Debug, Merge, PartialEq)]
Index: merge/tests/strategies.rs
===================================================================
--- merge.orig/tests/strategies.rs
+++ merge/tests/strategies.rs
@@ -20,6 +20,7 @@ fn test_option_overwrite_none() {
     test(S(None), S(None), S(None));
 }
 
+#[cfg(feature = "num")]
 #[test]
 fn test_option_recursive() {
     #[derive(Debug, Merge, PartialEq)]
