--- a/Cargo.toml
+++ b/Cargo.toml
@@ -96,9 +96,4 @@
 
-[dependencies.rustls-native-certs-06]
-version = "0.6"
-optional = true
-package = "rustls-native-certs"
-
 [dependencies.rustls-native-certs-07]
-version = "0.7"
+version = "0.8"
 optional = true
@@ -127,17 +122,2 @@
 
-[dependencies.tokio-rustls-023]
-version = "0.23"
-optional = true
-package = "tokio-rustls"
-
-[dependencies.tokio-rustls-024]
-version = "0.24"
-optional = true
-package = "tokio-rustls"
-
-[dependencies.tokio-rustls-025]
-version = "0.25"
-optional = true
-package = "tokio-rustls"
-
 [dependencies.tokio-rustls-026]
@@ -156,17 +136,2 @@
 
-[dependencies.webpki-roots-022]
-version = "0.22"
-optional = true
-package = "webpki-roots"
-
-[dependencies.webpki-roots-025]
-version = "0.25"
-optional = true
-package = "webpki-roots"
-
-[dependencies.webpki-roots-026]
-version = "0.26"
-optional = true
-package = "webpki-roots"
-
 [dev-dependencies.actix-codec]
@@ -189,3 +154,3 @@
 [dev-dependencies.itertools]
-version = "0.12"
+version = "0.13"
 
@@ -195,3 +160,3 @@
 [dev-dependencies.rcgen]
-version = "0.12"
+version = "0.14"
 
@@ -204,5 +169,2 @@
 
-[dev-dependencies.trust-dns-resolver]
-version = "0.23"
-
 [features]
@@ -219,33 +181,3 @@
 ]
-rustls = ["rustls-0_20"]
-rustls-0_20 = ["rustls-0_20-webpki-roots"]
-rustls-0_20-native-roots = [
-    "tokio-rustls-023",
-    "dep:rustls-native-certs-06",
-]
-rustls-0_20-webpki-roots = [
-    "tokio-rustls-023",
-    "webpki-roots-022",
-]
-rustls-0_21 = ["rustls-0_21-webpki-roots"]
-rustls-0_21-native-roots = [
-    "tokio-rustls-024",
-    "dep:rustls-native-certs-06",
-]
-rustls-0_21-webpki-roots = [
-    "tokio-rustls-024",
-    "webpki-roots-025",
-]
-rustls-0_22 = [
-    "dep:tokio-rustls-025",
-    "dep:rustls-pki-types-1",
-]
-rustls-0_22-native-roots = [
-    "rustls-0_22",
-    "dep:rustls-native-certs-07",
-]
-rustls-0_22-webpki-roots = [
-    "rustls-0_22",
-    "dep:webpki-roots-026",
-]
+rustls = ["rustls-0_23"]
 rustls-0_23 = [
@@ -258,6 +190,2 @@
 ]
-rustls-0_23-webpki-roots = [
-    "rustls-0_23",
-    "dep:webpki-roots-026",
-]
 uri = [
@@ -266,11 +194 @@
 ]
-
-[target."cfg(any())".dependencies.rustls-021]
-version = "0.21.6"
-optional = true
-package = "rustls"
-
-[target."cfg(any())".dependencies.rustls-webpki-0101]
-version = "0.101.4"
-optional = true
-package = "rustls-webpki"
