--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,7 +67,7 @@
 version = "0.2"
 
 [dependencies.dashmap]
-version = "5.4.0"
+version = ">= 5.4.0"
 
 [dependencies.futures]
 version = "0.3.21"
@@ -79,16 +79,16 @@
 version = "0.2"
 
 [dependencies.rustc-hash]
-version = "1.1.0"
+version = "2"
 
 [dependencies.static_assertions]
 version = "1.1.0"
 
 [dev-dependencies.core_affinity]
-version = "0.5.10"
+version = "0.8"
 
 [dev-dependencies.criterion]
-version = "0.3.4"
+version = "0.5"
 features = ["html_reports"]
 
 [dev-dependencies.futures]
