--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,1 +70,1 @@
-[dependencies.derive-visitor]
+[disabled.dependencies.derive-visitor]
@@ -82,1 +82,1 @@
-[dependencies.proptest]
+[disabled.dependencies.proptest]
@@ -91,1 +91,1 @@
-[dependencies.rkyv]
+[disabled.dependencies.rkyv]
@@ -106,1 +106,1 @@
-[dependencies.speedy]
+[disabled.dependencies.speedy]
@@ -118,20 +118,6 @@
     "rand/std",
     "rand/std_rng",
 ]
-rkyv = ["rkyv_32"]
-rkyv_16 = [
-    "dep:rkyv",
-    "rkyv?/size_16",
-]
-rkyv_32 = [
-    "dep:rkyv",
-    "rkyv?/size_32",
-]
-rkyv_64 = [
-    "dep:rkyv",
-    "rkyv?/size_64",
-]
-rkyv_ck = ["rkyv?/validation"]
 serde = [
     "dep:serde",
     "rand?/serde1",
