Index: backtrace/Cargo.toml
===================================================================
--- backtrace.orig/Cargo.toml
+++ backtrace/Cargo.toml
@@ -104,7 +104,7 @@ features = ["derive"]
 optional = true
 
 [dev-dependencies.libloading]
-version = "0.7"
+version = "0.8"
 
 [features]
 coresymbolication = []
@@ -127,7 +127,7 @@ version = "0.2.156"
 default-features = false
 
 [target.'cfg(not(all(windows, target_env = "msvc", not(target_vendor = "uwp"))))'.dependencies.miniz_oxide]
-version = "0.8"
+version = ">=0.7.1, <1.0"
 default-features = false
 
 [target.'cfg(not(all(windows, target_env = "msvc", not(target_vendor = "uwp"))))'.dependencies.object]
