Index: backtrace/Cargo.toml
===================================================================
--- backtrace.orig/Cargo.toml
+++ backtrace/Cargo.toml
@@ -58,7 +58,7 @@ harness = false
 required-features = ["std"]
 
 [dependencies.addr2line]
-version = "0.17.0"
+version = "0.18.0"
 default-features = false
 
 [dependencies.cfg-if]
@@ -74,7 +74,7 @@ version = "0.2.94"
 default-features = false
 
 [dependencies.miniz_oxide]
-version = "0.5.0"
+version = "0.4.0"
 default-features = false
 
 [dependencies.object]
