--- a/Cargo.toml
+++ b/Cargo.toml
@@ -138,7 +138,7 @@
 version = "0.6.0"
 
 [dependencies.tokio]
-version = "1.40"
+version = "1.39"
 features = ["rt"]
 optional = true
 
@@ -146,7 +146,7 @@
 version = "0.8"
 
 [dependencies.unicode-width]
-version = "0.2"
+version = "0.1"
 
 [features]
 default = ["docker"]
@@ -162,7 +162,7 @@
 version = "0.14"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies.regex]
-version = "1.11"
+version = "1.10"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies.uzers]
 version = "0.12"
