--- a/Cargo.toml
+++ b/Cargo.toml
@@ -118,10 +118,10 @@
 version = "0.15.11"
 
 [dependencies.directories]
-version = "6.0.0"
+version = "5"
 
 [dependencies.dockworker]
-version = "0.6.0"
+version = "0.7"
 optional = true
 
 [dependencies.getch]
@@ -158,7 +158,7 @@
 version = "0.8"
 
 [dependencies.unicode-width]
-version = "0.2"
+version = "0.1"
 
 [build-dependencies.anyhow]
 version = "1.0"
@@ -183,7 +183,7 @@
 version = "0.12"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies.which]
-version = "7"
+version = "8"
 
 [profile.release]
 lto = true
