--- a/Cargo.toml
+++ b/Cargo.toml
@@ -107,7 +107,7 @@
 version = "0.15.10"
 
 [dependencies.directories]
-version = "6.0.0"
+version = "5"
 
 [dependencies.dockworker]
 version = "0.5.1"
@@ -139,7 +139,7 @@
 version = "0.6.2"
 
 [dependencies.tokio]
-version = "1.43"
+version = "1.42"
 features = ["rt"]
 optional = true
 
@@ -147,7 +147,7 @@
 version = "0.8"
 
 [dependencies.unicode-width]
-version = "0.2"
+version = "0.1"
 
 [features]
 default = ["docker"]
@@ -169,7 +169,7 @@
 version = "0.12"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies.which]
-version = "7"
+version = "6"
 
 [badges.codecov]
 branch = "master"
