Index: pprof/Cargo.toml
===================================================================
--- pprof.orig/Cargo.toml
+++ pprof/Cargo.toml
@@ -20,6 +20,7 @@ documentation = "https://docs.rs/pprof/"
 readme = "README.md"
 license = "Apache-2.0"
 repository = "https://github.com/tikv/pprof-rs"
+autoexamples = false
 
 [package.metadata.docs.rs]
 all-features = true
@@ -29,13 +30,6 @@ name = "flamegraph"
 required-features = ["flamegraph"]
 
 [[example]]
-name = "profile_proto_with_prost"
-required-features = [
-    "protobuf",
-    "prost-codec",
-]
-
-[[example]]
 name = "profile_proto_with_protobuf_codec"
 required-features = [
     "protobuf",
