--- a/tests/all.rs
+++ b/tests/all.rs
@@ -377,6 +377,7 @@
 
 #[tokio::test]
 #[cfg(all(unix, feature = "xattr"))]
+#[ignore = "known to be flaky"]
 async fn xattrs() {
     // If /tmp is a tmpfs, xattr will fail
     // The xattr crate's unit tests also use /var/tmp for this reason
@@ -394,6 +395,7 @@
 
 #[tokio::test]
 #[cfg(all(unix, feature = "xattr"))]
+#[ignore = "known to be flaky"]
 async fn no_xattrs() {
     // If /tmp is a tmpfs, xattr will fail
     // The xattr crate's unit tests also use /var/tmp for this reason
