--- a/Cargo.toml
+++ b/Cargo.toml
@@ -191,7 +191,7 @@
 harness = false
 
 [dependencies.cairo-rs]
-version = "0.20"
+version = "0.21"
 features = [
     "v1_16",
     "png",
@@ -216,10 +216,10 @@
 version = "0.9.0"
 
 [dependencies.gio]
-version = "0.20"
+version = "0.21"
 
 [dependencies.glib]
-version = "0.20"
+version = "0.21"
 
 [dependencies.image]
 version = "0.25.0"
@@ -253,11 +253,11 @@
 version = "0.2"
 
 [dependencies.pango]
-version = "0.20"
+version = "0.21"
 features = ["v1_46"]
 
 [dependencies.pangocairo]
-version = "0.20"
+version = "0.21"

 [dependencies.rayon]
 version = "1"
@@ -306,7 +306,7 @@
 version = "0.5"
 
 [dev-dependencies.lopdf]
-version = "0.33"
+version = "0.34"
 
 [dev-dependencies.matches]
 version = "0.1"
@@ -337,7 +337,6 @@
 version = "7.0.0"
 
 [features]
-capi = []
 test-utils = ["yeslogic-fontconfig-sys"]
 
 [target.'cfg(all(not(target_os = "macos"), not(target_os = "windows")))'.dependencies.yeslogic-fontconfig-sys]
--- a/src/io.rs
+++ b/src/io.rs
@@ -107,7 +107,7 @@ pub fn acquire_data(
         let file = GFile::for_uri(uri);
         let (contents, _etag) = file.load_contents(cancellable)?;
 
-        let (content_type, _uncertain) = gio::content_type_guess(Some(uri), &contents);
+        let (content_type, _uncertain) = gio::content_type_guess(Some(uri), &*contents);
 
         let mime_type = if let Some(mime_type_str) = gio::content_type_get_mime_type(&content_type)
         {
