--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,6 +32,7 @@
 readme = "README.md"
 license = "LGPL-2.1-or-later"
 repository = "https://gitlab.gnome.org/GNOME/librsvg/"
+autobenches = false
 
 [package.metadata.docs.rs]
 rustc-args = [
@@ -93,33 +94,6 @@
 [lib]
 name = "rsvg"
 
-[[bench]]
-name = "box_blur"
-harness = false
-
-[[bench]]
-name = "composite"
-harness = false
-
-[[bench]]
-name = "lighting"
-harness = false
-
-[[bench]]
-name = "path_parser"
-harness = false
-
-[[bench]]
-name = "pixel_iterators"
-harness = false
-
-[[bench]]
-name = "pixel_ops"
-harness = false
-
-[[bench]]
-name = "srgb"
-harness = false
 
 [dependencies.cairo-rs]
 version = "0.19"
@@ -138,7 +112,7 @@
 version = "~0.31"
 
 [dependencies.data-url]
-version = "0.3.0"
+version = "0.1.0"
 
 [dependencies.encoding_rs]
 version = "0.8.32"
@@ -156,7 +130,7 @@
 version = "0.24.7"
 
 [dependencies.itertools]
-version = "0.12.0"
+version = "0.10.0"
 
 [dependencies.language-tags]
 version = "0.3.1"
@@ -171,7 +145,7 @@
 version = "0.11.0"
 
 [dependencies.nalgebra]
-version = "0.32.1"
+version = "0.33"
 
 [dependencies.num-traits]
 version = "0.2"
@@ -187,7 +161,7 @@
 version = "1"
 
 [dependencies.rctree]
-version = "0.6.0"
+version = "0.3.0"
 
 [dependencies.regex]
 version = "1.7.1"
@@ -260,10 +234,4 @@
 [build-dependencies.system-deps]
 version = "6.0.0"
 
-[features]
-c-api = []
-test-utils = ["yeslogic-fontconfig-sys"]
 
-[target."cfg(all(not(target_os = \"macos\"), not(target_os = \"windows\")))".dependencies.yeslogic-fontconfig-sys]
-version = "4.0.1"
-optional = true
