Description: Disable Windows dependencies, only used in doctests
Forwarded: not-needed
Last-Update: 2025-11-25
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -73 +73 @@
-[dev-dependencies.windows-sys]
+[disabled.dev-dependencies.windows-sys]
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -124,3 +124,3 @@
 //!
-//! ```rust
+//! ```rust,ignore
 //! # #[cfg(any(not(windows), not(feature = "alloc")))]
@@ -166,3 +166,3 @@
 //!
-//! ```rust
+//! ```rust,ignore
 //! # #[cfg(any(not(windows), not(feature = "alloc")))]
