Index: dashmap/Cargo.toml
===================================================================
--- dashmap.orig/Cargo.toml
+++ dashmap/Cargo.toml
@@ -72,12 +72,6 @@ version = "1.0.188"
 features = ["derive"]
 optional = true
 
-[dependencies.typesize]
-version = "0.1.8"
-optional = true
-default-features = false
-
 [features]
 inline = ["hashbrown/inline-more"]
 raw-api = []
-typesize = ["dep:typesize"]
