diff --git a/benches/mmap/mod.rs b/benches/mmap/mod.rs
index bbf3ab3..5871c9e 100644
--- a/benches/mmap/mod.rs
+++ b/benches/mmap/mod.rs
@@ -16,3 +16,5 @@ use criterion::{black_box, Criterion};
 use vm_memory::{ByteValued, Bytes, GuestAddress, GuestMemory};
-
+#[cfg(target_pointer_width = "32")]
+const REGION_SIZE: usize = 0x1000_0000;
+#[cfg(not(target_pointer_width = "32"))]
 const REGION_SIZE: usize = 0x8000_0000;
diff --git a/src/bitmap/mod.rs b/src/bitmap/mod.rs
index 51318ed..10d5ef6 100644
--- a/src/bitmap/mod.rs
+++ b/src/bitmap/mod.rs
@@ -237,3 +237,3 @@ pub(crate) mod tests {
 
-        let val = 1u64;
+        let val = 1usize;
 
diff --git a/src/bytes.rs b/src/bytes.rs
index 6274c3a..1fd7841 100644
--- a/src/bytes.rs
+++ b/src/bytes.rs
@@ -403,4 +403,4 @@ pub(crate) mod tests {
             {
-                if align_of::<T>() != 1 {
-                    assert_eq!(T::from_slice(unaligned_data), None);
+                if (i % align_of::<T>()) != 0 {
+                    assert_eq!(T::from_slice(unaligned_data), None, "{}::from_slice(unaligned_data) with alignment offset {} was not none", std::any::type_name::<T>(),i);
                     assert_eq!(T::from_mut_slice(unaligned_data), None);
-- 
2.39.5

