diff --git a/Cargo.toml b/Cargo.toml
index b29afe9..725ab19 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,11 +49,6 @@ xen = [
 name = "vm_memory"
 path = "src/lib.rs"
 
-[[bench]]
-name = "main"
-path = "benches/main.rs"
-harness = false
-
 [dependencies.arc-swap]
 version = "1.0.0"
 optional = true
@@ -74,4 +69,2 @@
 
-[dev-dependencies.criterion]
-version = "0.5.0"
 
@@ -87,7 +79,3 @@ features = [
     "errhandlingapi",
     "sysinfoapi",
 ]
-
-[profile.bench]
-lto = true
-codegen-units = 1
