diff --git a/src/mmap.rs b/src/mmap.rs
index 48d9a56..9e6dc8f 100644
--- a/src/mmap.rs
+++ b/src/mmap.rs
@@ -1269,2 +1269,3 @@ mod tests {
     #[test]
+    #[cfg(not(target_arch = "powerpc64"))]
     fn test_retrieve_fd_backing_memory_region() {
@@ -1296,2 +1297,3 @@ mod tests {
     #[cfg(unix)]
+    #[cfg(not(target_arch = "powerpc64"))]
     fn test_retrieve_offset_from_fd_backing_memory_region() {
@@ -1515,2 +1517,3 @@ mod tests {
     #[test]
+    #[cfg(not(target_arch = "powerpc64"))]
     fn test_dirty_tracking() {
diff --git a/src/mmap_unix.rs b/src/mmap_unix.rs
index 14ceb80..116b8ee 100644
--- a/src/mmap_unix.rs
+++ b/src/mmap_unix.rs
@@ -545,2 +545,3 @@ mod tests {
     #[cfg(not(miri))] // Miri cannot mmap files
+    #[cfg(not(target_arch = "powerpc64"))]
     fn test_mmap_region_build() {
@@ -635,2 +636,3 @@ mod tests {
     #[cfg(not(miri))] // Miri cannot mmap files
+    #[cfg(not(target_arch = "powerpc64"))]
     fn test_mmap_region_fds_overlap() {
@@ -665,2 +667,3 @@ mod tests {
     #[test]
+    #[cfg(not(target_arch = "powerpc64"))]
     fn test_dirty_tracking() {
-- 
2.39.5

