--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,5 +42,2 @@
 
-[dev-dependencies.xshell]
-version = "0.2.7"
-
 [features]
--- a/lib.rs
+++ b/lib.rs
@@ -38,6 +38,7 @@ pub const fn is_release() -> bool {
     matches!(build_channel(), BuildModel::Release)
 }
 
+/*
 #[cfg(test)]
 mod tests {
     use xshell::{cmd, Shell};
@@ -80,3 +81,4 @@ fn main() {
         cmd!(sh, "rm -rf ../{repo}").run().unwrap();
     }
 }
+*/
