Index: zerocopy/tests/trybuild.rs
===================================================================
--- zerocopy.orig/tests/trybuild.rs
+++ zerocopy/tests/trybuild.rs
@@ -20,36 +20,36 @@
 //   is enabled.
 #![cfg(feature = "derive")]
 
-use testutil::{set_rustflags_w_warnings, ToolchainVersion};
+//use testutil::{set_rustflags_w_warnings, ToolchainVersion};
 
 #[test]
 #[cfg_attr(miri, ignore)]
 fn ui() {
-    let version = ToolchainVersion::extract_from_pwd().unwrap();
+    //let version = ToolchainVersion::extract_from_pwd().unwrap();
     // See the doc comment on this method for an explanation of what this does
     // and why we store source files in different directories.
-    let source_files_dirname = version.get_ui_source_files_dirname_and_maybe_print_warning();
+    //let source_files_dirname = version.get_ui_source_files_dirname_and_maybe_print_warning();
 
     // Set `-Wwarnings` in the `RUSTFLAGS` environment variable to ensure that
     // `.stderr` files reflect what the typical user would encounter.
-    set_rustflags_w_warnings();
+    //set_rustflags_w_warnings();
 
-    let t = trybuild::TestCases::new();
-    t.compile_fail(format!("tests/{}/*.rs", source_files_dirname));
+    //let t = trybuild::TestCases::new();
+    //t.compile_fail(format!("tests/{}/*.rs", source_files_dirname));
 }
 
 #[test]
 #[cfg_attr(miri, ignore)]
 fn ui_invalid_impls() {
-    let version = ToolchainVersion::extract_from_pwd().unwrap();
+    //let version = ToolchainVersion::extract_from_pwd().unwrap();
     // See the doc comment on this method for an explanation of what this does
     // and why we store source files in different directories.
-    let source_files_dirname = version.get_ui_source_files_dirname_and_maybe_print_warning();
+    //let source_files_dirname = version.get_ui_source_files_dirname_and_maybe_print_warning();
 
     // Set `-Wwarnings` in the `RUSTFLAGS` environment variable to ensure that
     // `.stderr` files reflect what the typical user would encounter.
-    set_rustflags_w_warnings();
+    //set_rustflags_w_warnings();
 
-    let t = trybuild::TestCases::new();
-    t.compile_fail(format!("tests/{}/invalid-impls/*.rs", source_files_dirname));
+    //let t = trybuild::TestCases::new();
+    //t.compile_fail(format!("tests/{}/invalid-impls/*.rs", source_files_dirname));
 }
