--- a/Cargo.toml
+++ b/Cargo.toml
@@ -38,17 +38,6 @@
 [dependencies.libudev-sys]
 version = "0.1.4"
 
-[dependencies.mio06]
-version = "^0.6.21"
-optional = true
-package = "mio"
-
-[dependencies.mio07]
-version = "0.7"
-features = ["os-ext"]
-optional = true
-package = "mio"
-
 [dependencies.mio08]
 version = "0.8"
 features = ["os-ext"]
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -7,10 +7,6 @@
 extern crate io_lifetimes;
 extern crate libc;
 pub extern crate libudev_sys as ffi;
-#[cfg(feature = "mio06")]
-extern crate mio06;
-#[cfg(feature = "mio07")]
-extern crate mio07;
 #[cfg(feature = "mio08")]
 extern crate mio08;
 
--- a/src/monitor.rs
+++ b/src/monitor.rs
@@ -7,10 +7,6 @@
 use std::os::unix::io::{AsRawFd, RawFd};
 
 use io_lifetimes::{AsFd, BorrowedFd};
-#[cfg(feature = "mio06")]
-use mio06::{event::Evented, unix::EventedFd, Poll, PollOpt, Ready, Token};
-#[cfg(feature = "mio07")]
-use mio07::{event::Source, unix::SourceFd, Interest, Registry, Token};
 #[cfg(feature = "mio08")]
 use mio08::{event::Source, unix::SourceFd, Interest, Registry, Token};
 
@@ -307,34 +303,7 @@
     }
 }
 
-#[cfg(feature = "mio06")]
-impl Evented for Socket {
-    fn register(
-        &self,
-        poll: &Poll,
-        token: Token,
-        interest: Ready,
-        opts: PollOpt,
-    ) -> std::io::Result<()> {
-        EventedFd(&self.as_raw_fd()).register(poll, token, interest, opts)
-    }
-
-    fn reregister(
-        &self,
-        poll: &Poll,
-        token: Token,
-        interest: Ready,
-        opts: PollOpt,
-    ) -> std::io::Result<()> {
-        EventedFd(&self.as_raw_fd()).reregister(poll, token, interest, opts)
-    }
-
-    fn deregister(&self, poll: &Poll) -> std::io::Result<()> {
-        EventedFd(&self.as_raw_fd()).deregister(poll)
-    }
-}
-
-#[cfg(any(feature = "mio07", feature = "mio08"))]
+#[cfg(any(feature = "mio08"))]
 impl Source for Socket {
     fn register(
         &mut self,
