diff --git a/Cargo.toml b/Cargo.toml
index 32d74d9..bd6378f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,10 +35,6 @@ repository = "https://github.com/rust-lang/packed_simd"
 [dependencies.cfg-if]
 version = "1.0.0"
 
-[dependencies.core_arch]
-version = "0.1.5"
-optional = true
-
 [dependencies.libm]
 version = "0.1.4"
 
@@ -54,10 +50,6 @@ default = []
 into_bits = []
 libcore_neon = []
 
-[target."cfg(target_arch = \"x86_64\")".dependencies.sleef-sys]
-version = "0.1.2"
-optional = true
-
 [target.wasm32-unknown-unknown.dev-dependencies.wasm-bindgen]
 version = "=0.2.73"
 
