commit 76916f1a8518540b90063832ace546d5efdb13ca
Author: Jubilee Young <workingjubilee@gmail.com>
Date:   Sat May 15 20:39:43 2021 -0700

    Let packed_simd to be sloppy

diff --git a/.travis.yml b/.travis.yml
index 011eddde0..0ffc06afd 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -173,12 +173,6 @@ jobs:
         rustup toolchain install nightly -c rustfmt --allow-downgrade
         ci/all.sh check_fmt || true
       stage: tools
-    - name: "clippy"
-      install: true
-      script: |
-        rustup component add clippy
-        ci/all.sh clippy
-      stage: tools
 
   allow_failures:
     # FIXME: ISPC cannot be found?
diff --git a/examples/Cargo.toml b/examples/Cargo.toml
index 681326351..b05d4b22b 100644
--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -1,3 +1,6 @@
+# FIXME: Many members of this workspace, including aobench, mandelbrot, and stencil,
+# currently trigger a "null pointer deref" warning.
+# This is likely due to unsoundness inside packed_simd.
 [workspace]
 members = [
     "aobench",
diff --git a/examples/aobench/src/lib.rs b/examples/aobench/src/lib.rs
index 7fc7f9bda..d6bbb4cf5 100644
--- a/examples/aobench/src/lib.rs
+++ b/examples/aobench/src/lib.rs
@@ -2,7 +2,9 @@
 //!
 //! Based on [aobench](https://code.google.com/archive/p/aobench/) by Syoyo
 //! Fujita.
-#![deny(warnings, rust_2018_idioms)]
+// FIXME: Null pointer deref warning triggered in this example,
+// likely inside a macro expansion deriving from packed_simd.
+#![deny(rust_2018_idioms)]
 #![allow(non_snake_case, non_camel_case_types)]
 #![allow(
     clippy::many_single_char_names,
diff --git a/examples/aobench/src/main.rs b/examples/aobench/src/main.rs
index 3dd5ec88b..9dbb1f467 100644
--- a/examples/aobench/src/main.rs
+++ b/examples/aobench/src/main.rs
@@ -2,7 +2,7 @@
 //!
 //! Based on [aobench](https://code.google.com/archive/p/aobench/) by Syoyo
 //! Fujita.
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 
 use aobench_lib::*;
 use std::path::PathBuf;
diff --git a/examples/dot_product/src/lib.rs b/examples/dot_product/src/lib.rs
index 7f053a8f3..77d170c0d 100644
--- a/examples/dot_product/src/lib.rs
+++ b/examples/dot_product/src/lib.rs
@@ -1,5 +1,5 @@
 //! Vector dot product
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![feature(custom_inner_attributes)]
 #![allow(clippy::must_use_candidate, clippy::float_cmp)]
 
diff --git a/examples/fannkuch_redux/src/main.rs b/examples/fannkuch_redux/src/main.rs
index 912b4d66b..24cdec49f 100644
--- a/examples/fannkuch_redux/src/main.rs
+++ b/examples/fannkuch_redux/src/main.rs
@@ -1,4 +1,4 @@
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 
 use fannkuch_redux_lib::*;
 
diff --git a/examples/mandelbrot/src/lib.rs b/examples/mandelbrot/src/lib.rs
index 581fd02ca..196edc8b0 100644
--- a/examples/mandelbrot/src/lib.rs
+++ b/examples/mandelbrot/src/lib.rs
@@ -2,7 +2,9 @@
 //!
 //! [bg]: https://benchmarksgame-team.pages.debian.net/benchmarksgame/description/mandelbrot.html#mandelbrot
 
-#![deny(warnings, rust_2018_idioms)]
+// FIXME: Null pointer deref warning triggered in this example,
+// likely inside a macro expansion deriving from packed_simd.
+#![deny(rust_2018_idioms)]
 #![allow(
     clippy::cast_precision_loss,
     clippy::cast_sign_loss,
diff --git a/examples/mandelbrot/src/main.rs b/examples/mandelbrot/src/main.rs
index 6f82f05ac..7bee4a2f7 100644
--- a/examples/mandelbrot/src/main.rs
+++ b/examples/mandelbrot/src/main.rs
@@ -2,7 +2,7 @@
 //!
 //! [bg]: https://benchmarksgame-team.pages.debian.net/benchmarksgame/description/mandelbrot.html#mandelbrot
 
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 
 use mandelbrot_lib::*;
 use std::io;
diff --git a/examples/matrix_inverse/src/lib.rs b/examples/matrix_inverse/src/lib.rs
index 1f9bab3f8..fc3dbe210 100644
--- a/examples/matrix_inverse/src/lib.rs
+++ b/examples/matrix_inverse/src/lib.rs
@@ -1,6 +1,6 @@
 //! 4x4 matrix inverse
 #![feature(custom_inner_attributes)]
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![allow(clippy::must_use_candidate)]
 
 pub mod scalar;
diff --git a/examples/nbody/src/lib.rs b/examples/nbody/src/lib.rs
index d15854d5d..87c08b79c 100644
--- a/examples/nbody/src/lib.rs
+++ b/examples/nbody/src/lib.rs
@@ -1,7 +1,7 @@
 //! The N-body benchmark from the [benchmarks game][bg].
 //!
 //! [bg]: https://benchmarksgame-team.pages.debian.net/benchmarksgame/description/nbody.html#nbody
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![allow(
     clippy::similar_names,
     clippy::excessive_precision,
diff --git a/examples/nbody/src/main.rs b/examples/nbody/src/main.rs
index 3f53b2fe5..c5bc128eb 100644
--- a/examples/nbody/src/main.rs
+++ b/examples/nbody/src/main.rs
@@ -1,7 +1,7 @@
 //! The N-body benchmark from the [benchmarks game][bg].
 //!
 //! [bg]: https://benchmarksgame-team.pages.debian.net/benchmarksgame/description/nbody.html#nbody
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 
 fn run<O: std::io::Write>(o: &mut O, n: usize, alg: usize) {
     let (energy_before, energy_after) = nbody_lib::run(n, alg);
diff --git a/examples/options_pricing/src/lib.rs b/examples/options_pricing/src/lib.rs
index 138249909..86f5f0248 100644
--- a/examples/options_pricing/src/lib.rs
+++ b/examples/options_pricing/src/lib.rs
@@ -1,4 +1,4 @@
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![allow(
     clippy::inline_always,
     clippy::many_single_char_names,
diff --git a/examples/slice_sum/src/main.rs b/examples/slice_sum/src/main.rs
index 18b3692aa..2b3e440ad 100644
--- a/examples/slice_sum/src/main.rs
+++ b/examples/slice_sum/src/main.rs
@@ -1,4 +1,4 @@
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 
 use packed_simd::f32x8 as f32s;
 use std::{mem, slice};
diff --git a/examples/spectral_norm/src/lib.rs b/examples/spectral_norm/src/lib.rs
index 1134fb0b6..dac35341d 100644
--- a/examples/spectral_norm/src/lib.rs
+++ b/examples/spectral_norm/src/lib.rs
@@ -1,5 +1,5 @@
 //! Spectral Norm
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![allow(non_snake_case, non_camel_case_types)]
 #![allow(
     clippy::cast_precision_loss,
diff --git a/examples/spectral_norm/src/main.rs b/examples/spectral_norm/src/main.rs
index d16e6de99..7bbdf7cbf 100644
--- a/examples/spectral_norm/src/main.rs
+++ b/examples/spectral_norm/src/main.rs
@@ -1,5 +1,3 @@
-#![deny(warnings)]
-
 extern crate spectral_norm_lib;
 use spectral_norm_lib::*;
 
diff --git a/examples/stencil/src/lib.rs b/examples/stencil/src/lib.rs
index b922bc14f..1d7aeb40d 100644
--- a/examples/stencil/src/lib.rs
+++ b/examples/stencil/src/lib.rs
@@ -1,5 +1,7 @@
 #![feature(custom_inner_attributes, stmt_expr_attributes)]
-#![deny(warnings, rust_2018_idioms)]
+// FIXME: Null pointer deref warning triggered in this example,
+// likely inside a macro expansion deriving from packed_simd.
+#![deny(rust_2018_idioms)]
 #![allow(
     clippy::similar_names,
     clippy::cast_precision_loss,
diff --git a/micro_benchmarks/benches/mask_reductions.rs b/micro_benchmarks/benches/mask_reductions.rs
index e0b43a993..7d7ade52b 100644
--- a/micro_benchmarks/benches/mask_reductions.rs
+++ b/micro_benchmarks/benches/mask_reductions.rs
@@ -1,5 +1,5 @@
 //! Benchmarks for the mask reductions `all`, `any`, and `none`.
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![feature(test)]
 
 use packed_simd::*;
diff --git a/verify/verify/src/lib.rs b/verify/verify/src/lib.rs
index ad87e56cd..b222c88f1 100644
--- a/verify/verify/src/lib.rs
+++ b/verify/verify/src/lib.rs
@@ -1,7 +1,7 @@
 // FIXME: these types are unsound in C FFI already
 // See https://github.com/rust-lang/rust/issues/53346
 #![allow(improper_ctypes_definitions)]
-#![deny(warnings, rust_2018_idioms)]
+#![deny(rust_2018_idioms)]
 #![cfg_attr(test, feature(avx512_target_feature, abi_vectorcall, llvm_asm))]
 
 #[cfg(test)]
