Index: anstream/examples/dump-stream.rs
===================================================================
--- anstream.orig/examples/dump-stream.rs
+++ anstream/examples/dump-stream.rs
@@ -55,7 +55,7 @@ fn style(
 }
 
 fn print_number(stdout: &mut impl Write, fixed: u8, style: anstyle::Style) -> std::io::Result<()> {
-    write!(stdout, "{style}{fixed:>3X}{style:#}",)
+    write!(stdout, "{style:?}{fixed:>3X}{style:#?}",)
 }
 
 #[derive(Default)]
