--- a/Cargo.toml
+++ b/Cargo.toml
@@ -89,7 +89,7 @@
 version = "^2.2.0"
 
 [dependencies.rustix]
-version = "0.38.22"
+version = "1"
 features = [
     "mm",
     "fs",
@@ -101,7 +101,7 @@
 default-features = false
 
 [dev-dependencies.rustix]
-version = "0.38.22"
+version = "1"
 features = [
     "event",
     "mm",
--- a/examples/syncobj.rs
+++ b/examples/syncobj.rs
@@ -49,5 +49,5 @@
     // let future = async move { afd.readable().await.unwrap().retain_ready() };
     // future.await;
     let mut poll_fds = [rustix::event::PollFd::new(&fd, PollFlags::IN)];
-    rustix::event::poll(&mut poll_fds, -1).unwrap();
+    rustix::event::poll(&mut poll_fds, None).unwrap();
 }
