--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,21 +61,10 @@
 
 [features]
 default = ["latest"]
-embedded-libheif = ["libheif-sys/embedded-libheif"]
-latest = ["v1_20"]
-v1_17 = ["libheif-sys/v1_17"]
-v1_18 = [
-    "v1_17",
-    "libheif-sys/v1_18",
-]
+latest = ["v1_19"]
 v1_19 = [
-    "v1_18",
     "libheif-sys/v1_19",
 ]
-v1_20 = [
-    "v1_19",
-    "libheif-sys/v1_20",
-]
 
 [lib]
 name = "libheif_rs"
