--- a/tests/heif_struct_test.rs
+++ b/tests/heif_struct_test.rs
@@ -9,6 +9,7 @@ fn get_version() {
 }
 
 #[test]
+#[ignore = "broken under debian"]
 fn get_encoder_descriptors() {
     let lib_heif = LibHeif::new();
     let descriptors = lib_heif.encoder_descriptors(100, None, None);
@@ -16,6 +17,7 @@ fn get_encoder_descriptors() {
 }
 
 #[test]
+#[ignore = "broken under debian"]
 fn get_encoder() {
     let lib_heif = LibHeif::new();
     let descriptors = lib_heif.encoder_descriptors(1, None, None);
