--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,11 +57,6 @@
 name = "mod"
 path = "tests/mod.rs"
 
-[[bench]]
-name = "perf"
-path = "benches/perf.rs"
-harness = false
-
 [dependencies.async-trait]
 version = "0.1"
 
@@ -150,10 +145,6 @@
 [dependencies.weezl]
 version = "0.1.7"
 
-[dev-dependencies.criterion]
-version = "0.6"
-features = ["html_reports"]
-
 [dev-dependencies.env_logger]
 version = "0.11"
 
