--- a/Cargo.toml
+++ b/Cargo.toml
@@ -89,7 +89,7 @@
 version = "0.25"
 
 [dependencies.itertools]
-version = "0.14"
+version = "0.13"
 
 [dependencies.lazy_static]
 version = "1"
@@ -113,7 +113,7 @@
 version = "0.8"
 
 [dependencies.num_enum]
-version = "0.7"
+version = "0.5"
 
 [dependencies.rayon]
 version = "1"
@@ -149,7 +149,7 @@
 version = "0.11"
 
 [dev-dependencies.serde_yaml]
-version = "0.8"
+version = "0.9"
 
 [build-dependencies.glob]
 version = "0.3"
