--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,9 +14,9 @@
 name = "apple-nvram"
 version = "0.3.0"
 description = "A library to parse and write apple-formatted nvram entries"
-homepage = "https://github.com/WhatAmISupposedToPutHere/asahi-nvram"
+homepage = "https://github.com/AsahiLinux/asahi-nvram"
 license = "MIT"
-repository = "https://github.com/WhatAmISupposedToPutHere/asahi-nvram"
+repository = "https://github.com/AsahiLinux/asahi-nvram"
 
 [dependencies.adler32]
 version = "1"
