Index: similar/Cargo.toml
===================================================================
--- similar.orig/Cargo.toml
+++ similar/Cargo.toml
@@ -84,7 +84,7 @@ required-features = [
 ]
 
 [dependencies.bstr]
-version = "0.2.14"
+version = "1"
 optional = true
 default-features = false
 
@@ -101,10 +101,10 @@ optional = true
 version = "0.15.0"
 
 [dev-dependencies.insta]
-version = "1.10.0"
+version = "1.5.2"
 
 [dev-dependencies.serde_json]
-version = "1.0.68"
+version = "1.0.82"
 
 [features]
 bytes = [
@@ -118,4 +118,5 @@ unicode = [
     "text",
     "unicode-segmentation",
     "bstr/unicode",
+    "bstr/std",
 ]
