--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,11 +52,6 @@ default-features = false
 version = "0.3.17"
 default-features = false
 
-[dependencies.futures_01]
-version = "0.1.25"
-optional = true
-package = "futures"
-
 [dependencies.memchr]
 version = "2.2"
 optional = true
@@ -68,7 +63,7 @@ version = "0.2.4"
 version = "0.1.0"
 
 [dependencies.proc-macro-hack]
-version = "0.5.19"
+version = "0.5"
 optional = true
 
 [dependencies.proc-macro-nested]
@@ -94,12 +89,9 @@ async-await-macro = ["async-await", "fut
 bilock = []
 cfg-target-has-atomic = []
 channel = ["std", "futures-channel"]
-compat = ["std", "futures_01"]
 default = ["std", "async-await", "async-await-macro"]
 io = ["std", "futures-io", "memchr"]
-io-compat = ["io", "compat", "tokio-io"]
 read-initializer = ["io", "futures-io/read-initializer", "futures-io/unstable"]
 sink = ["futures-sink"]
 std = ["alloc", "futures-core/std", "futures-task/std", "slab"]
-unstable = ["futures-core/unstable", "futures-task/unstable"]
 write-all-vectored = ["io"]
