Index: litemap/Cargo.toml
===================================================================
--- litemap.orig/Cargo.toml
+++ litemap/Cargo.toml
@@ -67,12 +67,6 @@ name = "store"
 path = "tests/store.rs"
 required-features = ["testing"]
 
-[[bench]]
-name = "litemap"
-path = "benches/litemap.rs"
-harness = false
-required-features = ["serde"]
-
 [dependencies.databake]
 version = "0.2.0"
 optional = true
@@ -101,7 +95,6 @@ version = "1.0.45"
 
 [features]
 alloc = []
-bench = ["serde"]
 databake = ["dep:databake"]
 default = ["alloc"]
 serde = ["dep:serde"]
