Index: litemap/Cargo.toml
===================================================================
--- litemap.orig/Cargo.toml
+++ litemap/Cargo.toml
@@ -57,24 +57,6 @@ name = "litemap"
 path = "src/lib.rs"
 bench = false
 
-[[example]]
-name = "language_names_hash_map"
-path = "examples/language_names_hash_map.rs"
-
-[[example]]
-name = "language_names_lite_map"
-path = "examples/language_names_lite_map.rs"
-
-[[example]]
-name = "litemap_bincode"
-path = "examples/litemap_bincode.rs"
-required-features = ["serde"]
-
-[[example]]
-name = "litemap_postcard"
-path = "examples/litemap_postcard.rs"
-required-features = ["serde"]
-
 [[test]]
 name = "serde"
 path = "tests/serde.rs"
