Index: litemap/Cargo.toml
===================================================================
--- litemap.orig/Cargo.toml
+++ litemap/Cargo.toml
@@ -76,10 +76,6 @@ path = "examples/litemap_postcard.rs"
 required-features = ["serde"]
 
 [[test]]
-name = "rkyv"
-path = "tests/rkyv.rs"
-
-[[test]]
 name = "serde"
 path = "tests/serde.rs"
 required-features = ["serde"]
@@ -114,10 +110,6 @@ version = "1.0.1"
 features = ["use-std"]
 default-features = false
 
-[dev-dependencies.rkyv]
-version = "0.7"
-features = ["validation"]
-
 [dev-dependencies.serde]
 version = "1.0.110"
 default-features = false
