Index: instant/Cargo.toml
===================================================================
--- instant.orig/Cargo.toml
+++ instant/Cargo.toml
@@ -27,7 +27,8 @@ version = "0.3"
 [features]
 inaccurate = []
 now = []
-wasm-bindgen = ["js-sys", "wasm-bindgen_rs", "web-sys"]
+wasm-bindgen = ["js-sys", "wasm-bindgen_rs"] 
+#, "web-sys"]
 [target.asmjs-unknown-emscripten.dependencies.js-sys]
 version = "0.3"
 optional = true
@@ -41,10 +42,10 @@ version = "0.2"
 optional = true
 package = "wasm-bindgen"
 
-[target.asmjs-unknown-emscripten.dependencies.web-sys]
-version = "0.3"
-features = ["Window", "Performance", "PerformanceTiming"]
-optional = true
+#[target.asmjs-unknown-emscripten.dependencies.web-sys]
+#version = "0.3"
+#features = ["Window", "Performance", "PerformanceTiming"]
+#optional = true
 [target.wasm32-unknown-emscripten.dependencies.js-sys]
 version = "0.3"
 optional = true
@@ -58,10 +59,10 @@ version = "0.2"
 optional = true
 package = "wasm-bindgen"
 
-[target.wasm32-unknown-emscripten.dependencies.web-sys]
-version = "0.3"
-features = ["Window", "Performance", "PerformanceTiming"]
-optional = true
+#[target.wasm32-unknown-emscripten.dependencies.web-sys]
+#version = "0.3"
+#features = ["Window", "Performance", "PerformanceTiming"]
+#optional = true
 [target.wasm32-unknown-unknown.dependencies.js-sys]
 version = "0.3"
 optional = true
@@ -75,7 +76,7 @@ version = "0.2"
 optional = true
 package = "wasm-bindgen"
 
-[target.wasm32-unknown-unknown.dependencies.web-sys]
-version = "0.3"
-features = ["Window", "Performance", "PerformanceTiming"]
-optional = true
+#[target.wasm32-unknown-unknown.dependencies.web-sys]
+#version = "0.3"
+#features = ["Window", "Performance", "PerformanceTiming"]
+#optional = true
