Index: instant/Cargo.toml
===================================================================
--- instant.orig/Cargo.toml
+++ instant/Cargo.toml
@@ -33,9 +33,9 @@ wasm-bindgen = ["js-sys", "wasm-bindgen_
 version = "0.3"
 optional = true
 
-[target.asmjs-unknown-emscripten.dependencies.stdweb]
-version = "0.4"
-optional = true
+# [target.asmjs-unknown-emscripten.dependencies.stdweb]
+# version = "0.4"
+# optional = true
 
 [target.asmjs-unknown-emscripten.dependencies.wasm-bindgen_rs]
 version = "0.2"
@@ -50,9 +50,9 @@ package = "wasm-bindgen"
 version = "0.3"
 optional = true
 
-[target.wasm32-unknown-emscripten.dependencies.stdweb]
-version = "0.4"
-optional = true
+# [target.wasm32-unknown-emscripten.dependencies.stdweb]
+# version = "0.4"
+# optional = true
 
 [target.wasm32-unknown-emscripten.dependencies.wasm-bindgen_rs]
 version = "0.2"
@@ -67,9 +67,9 @@ package = "wasm-bindgen"
 version = "0.3"
 optional = true
 
-[target.wasm32-unknown-unknown.dependencies.stdweb]
-version = "0.4"
-optional = true
+# [target.wasm32-unknown-unknown.dependencies.stdweb]
+# version = "0.4"
+# optional = true
 
 [target.wasm32-unknown-unknown.dependencies.wasm-bindgen_rs]
 version = "0.2"
