--- a/tests/test_std.rs
+++ b/tests/test_std.rs
@@ -57,6 +57,7 @@
     assert_eq!(it.size_hint(), (6, Some(6)));
 }
 
+#[cfg(feature = "use_std")]
 #[test]
 fn duplicates_by() {
     let xs = ["aaa", "bbbbb", "aa", "ccc", "bbbb", "aaaaa", "cccc"];
@@ -67,6 +68,7 @@
     it::assert_equal(ys_rev.iter(), xs.iter().duplicates_by(|x| x[..2].to_string()).rev());
 }
 
+#[cfg(feature = "use_std")]
 #[test]
 fn duplicates() {
     let xs = [0, 1, 2, 3, 2, 1, 3];
@@ -91,6 +93,7 @@
     assert_eq!(ys_rev, xs.iter().duplicates().rev().cloned().collect_vec());
 }
 
+#[cfg(feature = "use_std")]
 #[test]
 fn unique_by() {
     let xs = ["aaa", "bbbbb", "aa", "ccc", "bbbb", "aaaaa", "cccc"];
@@ -101,6 +104,7 @@
     it::assert_equal(ys_rev.iter(), xs.iter().unique_by(|x| x[..2].to_string()).rev());
 }
 
+#[cfg(feature = "use_std")]
 #[test]
 fn unique() {
     let xs = [0, 1, 2, 3, 2, 1, 3];
@@ -236,6 +240,7 @@
     }
 }
 
+#[cfg(feature = "use_std")]
 #[test]
 fn all_unique() {
     assert!("ABCDEFGH".chars().all_unique());
@@ -1007,6 +1012,7 @@
     });
 }
 
+#[cfg(feature = "use_std")]
 #[test]
 fn extrema_set() {
     use std::cmp::Ordering;
--- a/tests/quick.rs
+++ b/tests/quick.rs
@@ -944,12 +944,14 @@
     }
 }
 
+#[cfg(feature = "use_std")]
 quickcheck! {
     fn size_duplicates(it: Iter<i8>) -> bool {
         correct_size_hint(it.duplicates())
     }
 }
 
+#[cfg(feature = "use_std")]
 quickcheck! {
     fn size_unique(it: Iter<i8>) -> bool {
         correct_size_hint(it.unique())
@@ -1317,6 +1319,7 @@
     }
 }
 
+#[cfg(feature = "use_std")]
 quickcheck! {
     fn consistent_grouping_map_with_by(a: Vec<u8>, modulo: u8) -> () {
         let modulo = if modulo == 0 { 1 } else { modulo }; // Avoid `% 0`
@@ -1714,6 +1717,7 @@
     true
 }
 
+#[cfg(feature = "use_std")]
 quickcheck! {
     #[cfg(doc)]
     fn fused_combination(a: Iter<i16>) -> bool
@@ -1800,6 +1804,7 @@
     }
 }
 
+#[cfg(feature = "use_std")]
 quickcheck! {
     fn min_set_contains_min(a: Vec<(usize, char)>) -> bool {
         let result_set = a.iter().min_set();
