--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,6 +40,11 @@
 overflow-checks = true
 strip = "debuginfo"
 
+[[bin]]
+name = "wormhole-rs"
+path = "cli/src/main.rs"
+
+
 [dependencies.async-io]
 version = "2.2.0"
 
@@ -75,6 +75,25 @@
 [dependencies.crypto_secretbox]
 version = "0.1.1"
 
+[dependencies.clap]
+version =  "3.1.5"
+features = ["cargo", "derive", "wrap_help"]
+
+[dependencies.clap_complete]
+version = "3.1.1"
+
+[dependencies.cli-clipboard]
+version = "0.4.0"
+
+[dependencies.console]
+version = "0.15.0"
+
+[dependencies.color-eyre]
+version = "0.6.0"
+
+[dependencies.ctrlc]
+version = "3.2.1"
+
 [dependencies.derive_more]
 version = "0.99.0"
 features = [
@@ -84,6 +103,9 @@
 ]
 default-features = false
 
+[dependencies.dialoguer]
+version = "0.10.0"
+
 [dependencies.futures]
 version = "0.3.12"
 
@@ -101,6 +123,9 @@
 version = "0.10.1"
 optional = true
 
+[dependencies.indicatif]
+version = "0.17.0"
+
 [dependencies.libc]
 version = "0.2.101"
 
@@ -115,9 +140,15 @@
 version = "0.6.2"
 optional = true
 
+[dependencies.number_prefix]
+version = "0.4.0"
+
 [dependencies.percent-encoding]
 version = "2.1.0"
 
+[dependencies.qr2term]
+version = "0.3.0"
+
 [dependencies.rand]
 version = "0.8.3"
 

