--- a/Cargo.toml
+++ b/Cargo.toml
@@ -154,7 +154,7 @@
 features = ["serde"]
 
 [dependencies.zxcvbn]
-version = "3.1.0"
+version = "2.2"
 optional = true
 
 [dev-dependencies.eyre]
--- a/src/core.rs
+++ b/src/core.rs
@@ -970,7 +970,7 @@
                 .collect::<Vec<_>>()
         });
 
-        zxcvbn::zxcvbn(password, &words[..])
+        zxcvbn::zxcvbn(password, &words[..]).expect("REASON")
     }
 }
 
