--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,7 +49,6 @@
 all = [
     "default",
     "forwarding",
-    "fuzzy-complete",
 ]
 default = [
     "transit",
@@ -62,7 +61,6 @@
     "transit",
     "dep:rmp-serde",
 ]
-fuzzy-complete = ["fuzzt"]
 tls = ["async-tungstenite/async-tls"]
 transfer = [
     "transit",
@@ -116,10 +114,6 @@
 [dependencies.futures]
 version = "0.3.12"
 
-[dependencies.fuzzt]
-version = "0.3.1"
-optional = true
-
 [dependencies.hex]
 version = "0.4.2"
 features = ["serde"]
--- a/src/core/wordlist.rs
+++ b/src/core/wordlist.rs
@@ -189,6 +189,7 @@

     #[test]
     #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
+    #[ignore = "needs fuzzt"]
     fn test_completion() {
         let words: Vec<Vec<String>> = vec![
             vec_strings("purple green yellow"),
@@ -216,6 +217,7 @@
     }

     #[test]
+    #[ignore = "needs fuzzt"]
     #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
     fn test_choose_words_matches_completion() {
         let few_words: Vec<Vec<String>> = vec![vec_strings("purple"), vec_strings("sausages")];
@@ -255,6 +257,7 @@
     }

     #[test]
+    #[ignore = "needs fuzzt"]
     #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
     #[cfg(feature = "fuzzy-complete")]
     fn test_completion_fuzzy() {
@@ -290,6 +293,7 @@
     }

     #[test]
+    #[ignore = "needs fuzzt"]
     #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
     fn test_wormhole_code_normal_completions() {
         let list = Wordlist::default_wordlist(2);
