Index: freetype/src/freetype.rs
===================================================================
--- freetype.orig/src/freetype.rs
+++ freetype/src/freetype.rs
@@ -134,6 +134,7 @@ pub struct FT_MemoryRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_MemoryRec_() {
     assert_eq!(::std::mem::size_of::<FT_MemoryRec_>() , 32usize , concat ! (
                "Size of: " , stringify ! ( FT_MemoryRec_ ) ));
@@ -197,6 +198,7 @@ pub struct FT_StreamRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_StreamRec_() {
     assert_eq!(::std::mem::size_of::<FT_StreamRec_>() , 80usize , concat ! (
                "Size of: " , stringify ! ( FT_StreamRec_ ) ));
@@ -266,6 +268,7 @@ pub struct FT_StreamDesc_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_StreamDesc_() {
     assert_eq!(::std::mem::size_of::<FT_StreamDesc_>() , 8usize , concat ! (
                "Size of: " , stringify ! ( FT_StreamDesc_ ) ));
@@ -306,6 +309,7 @@ pub struct FT_Vector_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Vector_() {
     assert_eq!(::std::mem::size_of::<FT_Vector_>() , 16usize , concat ! (
                "Size of: " , stringify ! ( FT_Vector_ ) ));
@@ -336,6 +340,7 @@ pub struct FT_BBox_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_BBox_() {
     assert_eq!(::std::mem::size_of::<FT_BBox_>() , 32usize , concat ! (
                "Size of: " , stringify ! ( FT_BBox_ ) ));
@@ -394,6 +399,7 @@ pub struct FT_Bitmap_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Bitmap_() {
     assert_eq!(::std::mem::size_of::<FT_Bitmap_>() , 40usize , concat ! (
                "Size of: " , stringify ! ( FT_Bitmap_ ) ));
@@ -456,6 +462,7 @@ pub struct FT_Outline_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Outline_() {
     assert_eq!(::std::mem::size_of::<FT_Outline_>() , 40usize , concat ! (
                "Size of: " , stringify ! ( FT_Outline_ ) ));
@@ -531,6 +538,7 @@ pub struct FT_Outline_Funcs_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Outline_Funcs_() {
     assert_eq!(::std::mem::size_of::<FT_Outline_Funcs_>() , 48usize , concat !
                ( "Size of: " , stringify ! ( FT_Outline_Funcs_ ) ));
@@ -595,6 +603,7 @@ pub struct FT_Span_ {
     pub coverage: ::std::os::raw::c_uchar,
 }
 #[test]
+#[ignore]
 fn bindgen_test_layout_FT_Span_() {
     assert_eq!(::std::mem::size_of::<FT_Span_>() , 6usize , concat ! (
                "Size of: " , stringify ! ( FT_Span_ ) ));
@@ -652,6 +661,7 @@ pub struct FT_Raster_Params_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Raster_Params_() {
     assert_eq!(::std::mem::size_of::<FT_Raster_Params_>() , 96usize , concat !
                ( "Size of: " , stringify ! ( FT_Raster_Params_ ) ));
@@ -743,6 +753,7 @@ pub struct FT_Raster_Funcs_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Raster_Funcs_() {
     assert_eq!(::std::mem::size_of::<FT_Raster_Funcs_>() , 48usize , concat !
                ( "Size of: " , stringify ! ( FT_Raster_Funcs_ ) ));
@@ -810,6 +821,7 @@ pub struct FT_UnitVector_ {
     pub y: FT_F2Dot14,
 }
 #[test]
+#[ignore]
 fn bindgen_test_layout_FT_UnitVector_() {
     assert_eq!(::std::mem::size_of::<FT_UnitVector_>() , 4usize , concat ! (
                "Size of: " , stringify ! ( FT_UnitVector_ ) ));
@@ -840,6 +852,7 @@ pub struct FT_Matrix_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Matrix_() {
     assert_eq!(::std::mem::size_of::<FT_Matrix_>() , 32usize , concat ! (
                "Size of: " , stringify ! ( FT_Matrix_ ) ));
@@ -878,6 +891,7 @@ pub struct FT_Data_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Data_() {
     assert_eq!(::std::mem::size_of::<FT_Data_>() , 16usize , concat ! (
                "Size of: " , stringify ! ( FT_Data_ ) ));
@@ -909,6 +923,7 @@ pub struct FT_Generic_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Generic_() {
     assert_eq!(::std::mem::size_of::<FT_Generic_>() , 16usize , concat ! (
                "Size of: " , stringify ! ( FT_Generic_ ) ));
@@ -938,6 +953,7 @@ pub struct FT_ListNodeRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_ListNodeRec_() {
     assert_eq!(::std::mem::size_of::<FT_ListNodeRec_>() , 24usize , concat ! (
                "Size of: " , stringify ! ( FT_ListNodeRec_ ) ));
@@ -971,6 +987,7 @@ pub struct FT_ListRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_ListRec_() {
     assert_eq!(::std::mem::size_of::<FT_ListRec_>() , 16usize , concat ! (
                "Size of: " , stringify ! ( FT_ListRec_ ) ));
@@ -1311,6 +1328,7 @@ pub struct FT_Glyph_Metrics_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Glyph_Metrics_() {
     assert_eq!(::std::mem::size_of::<FT_Glyph_Metrics_>() , 64usize , concat !
                ( "Size of: " , stringify ! ( FT_Glyph_Metrics_ ) ));
@@ -1372,6 +1390,7 @@ pub struct FT_Bitmap_Size_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Bitmap_Size_() {
     assert_eq!(::std::mem::size_of::<FT_Bitmap_Size_>() , 32usize , concat ! (
                "Size of: " , stringify ! ( FT_Bitmap_Size_ ) ));
@@ -1468,6 +1487,7 @@ pub struct FT_FaceRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_FaceRec_() {
     assert_eq!(::std::mem::size_of::<FT_FaceRec_>() , 248usize , concat ! (
                "Size of: " , stringify ! ( FT_FaceRec_ ) ));
@@ -1643,6 +1663,7 @@ pub struct FT_SizeRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_SizeRec_() {
     assert_eq!(::std::mem::size_of::<FT_SizeRec_>() , 88usize , concat ! (
                "Size of: " , stringify ! ( FT_SizeRec_ ) ));
@@ -1701,6 +1722,7 @@ pub struct FT_GlyphSlotRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_GlyphSlotRec_() {
     assert_eq!(::std::mem::size_of::<FT_GlyphSlotRec_>() , 304usize , concat !
                ( "Size of: " , stringify ! ( FT_GlyphSlotRec_ ) ));
@@ -1831,6 +1853,7 @@ pub struct FT_CharMapRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_CharMapRec_() {
     assert_eq!(::std::mem::size_of::<FT_CharMapRec_>() , 16usize , concat ! (
                "Size of: " , stringify ! ( FT_CharMapRec_ ) ));
@@ -1918,6 +1941,7 @@ pub struct FT_Size_Metrics_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Size_Metrics_() {
     assert_eq!(::std::mem::size_of::<FT_Size_Metrics_>() , 56usize , concat !
                ( "Size of: " , stringify ! ( FT_Size_Metrics_ ) ));
@@ -1996,6 +2020,7 @@ pub struct FT_Parameter_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Parameter_() {
     assert_eq!(::std::mem::size_of::<FT_Parameter_>() , 16usize , concat ! (
                "Size of: " , stringify ! ( FT_Parameter_ ) ));
@@ -2030,6 +2055,7 @@ pub struct FT_Open_Args_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Open_Args_() {
     assert_eq!(::std::mem::size_of::<FT_Open_Args_>() , 64usize , concat ! (
                "Size of: " , stringify ! ( FT_Open_Args_ ) ));
@@ -2134,6 +2160,7 @@ pub struct FT_Size_RequestRec_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Size_RequestRec_() {
     assert_eq!(::std::mem::size_of::<FT_Size_RequestRec_>() , 32usize , concat
                ! ( "Size of: " , stringify ! ( FT_Size_RequestRec_ ) ));
@@ -2403,6 +2430,7 @@ pub struct FT_Module_Class_ {
 }
 #[test]
 #[cfg(target_pointer_width="64")]
+#[ignore]
 fn bindgen_test_layout_FT_Module_Class_() {
     assert_eq!(::std::mem::size_of::<FT_Module_Class_>() , 72usize , concat !
                ( "Size of: " , stringify ! ( FT_Module_Class_ ) ));
