Description: Add loongarch support
Last-Update: 2024-04-09
Forwarded: yes

--- rust-jemalloc-sys-0.3.2.orig/jemalloc/include/jemalloc/internal/jemalloc_internal_types.h
+++ rust-jemalloc-sys-0.3.2/jemalloc/include/jemalloc/internal/jemalloc_internal_types.h
@@ -110,6 +110,9 @@ typedef int malloc_cpuid_t;
 #  ifdef __le32__
 #    define LG_QUANTUM		4
 #  endif
+#  ifdef __loongarch__
+#    define LG_QUANTUM		4
+#  endif
 #  ifndef LG_QUANTUM
 #    error "Unknown minimum alignment for architecture; specify via "
 	 "--with-lg-quantum"
--- rust-jemalloc-sys-0.3.2.orig/rep/include/jemalloc/internal/quantum.h
+++ rust-jemalloc-sys-0.3.2/rep/include/jemalloc/internal/quantum.h
@@ -61,6 +61,9 @@
 #  ifdef __le32__
 #    define LG_QUANTUM		4
 #  endif
+#  ifdef __loongarch__
+#    define LG_QUANTUM		4
+#  endif
 #  ifndef LG_QUANTUM
 #    error "Unknown minimum alignment for architecture; specify via "
 	 "--with-lg-quantum"

