Index: gimli/Cargo.toml
===================================================================
--- gimli.orig/Cargo.toml
+++ gimli/Cargo.toml
@@ -42,19 +42,6 @@ required-features = ["read", "std"]
 [[example]]
 name = "dwarf-validate"
 required-features = ["read", "std"]
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
-[dependencies.compiler_builtins]
-version = "0.1.2"
-optional = true
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
 
 [dependencies.fallible-iterator]
 version = "0.2.0"
@@ -100,7 +87,6 @@ version = "2"
 default = ["read", "write", "std", "fallible-iterator", "endian-reader"]
 endian-reader = ["stable_deref_trait"]
 read = []
-rustc-dep-of-std = ["core", "alloc", "compiler_builtins"]
 std = ["fallible-iterator/std", "stable_deref_trait/std"]
 write = ["indexmap"]
 [badges.coveralls]
