Index: nftables/Cargo.toml
===================================================================
--- nftables.orig/Cargo.toml
+++ nftables/Cargo.toml
@@ -59,7 +59,8 @@ path = "src/lib.rs"
 name = "nftables"
 path = "src/main.rs"
 
-[[test]]
+# requires dataset-stable
+[[disabled.test]]
 name = "deserialize"
 path = "tests/deserialize.rs"
 harness = false
@@ -118,7 +119,7 @@ features = [
 ]
 optional = true
 
-[dev-dependencies.datatest-stable]
+[disabled.dev-dependencies.datatest-stable]
 version = "0.3.2"
 
 [dev-dependencies.serial_test]
