Index: cxx/Cargo.toml
===================================================================
--- cxx.orig/Cargo.toml
+++ cxx/Cargo.toml
@@ -110,7 +110,7 @@ path = "tests/unique_ptr.rs"
 version = "=1.0.141"
 
 [dependencies.foldhash]
-version = "0.1"
+version = ">= 0.1, < 0.3"
 default-features = false
 
 [dependencies.link-cplusplus]
@@ -122,9 +122,6 @@ version = "=1.0.141"
 [dev-dependencies.cxx-gen]
 version = "0.7"
 
-[dev-dependencies.cxx-test-suite]
-version = "0"
-
 [dev-dependencies.rustversion]
 version = "1.0.13"
 
