Index: accesskit-consumer/Cargo.toml
===================================================================
--- accesskit-consumer.orig/Cargo.toml
+++ accesskit-consumer/Cargo.toml
@@ -37,9 +37,9 @@ name = "accesskit_consumer"
 path = "src/lib.rs"
 
 [dependencies.accesskit]
-version = "0.21.1"
+version = ">= 0.21.0"
 
 [dependencies.hashbrown]
-version = "0.15"
+version = ">= 0.14"
 features = ["default-hasher"]
 default-features = false
