diff --git a/src/node.rs b/src/node.rs
index 3b9577e..f2450a2 100644
--- a/src/node.rs
+++ b/src/node.rs
@@ -19,7 +19,7 @@ impl<'a> NodeList<'a> {
     /// This is intended for queries that are expected to optionally yield a single node.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # use serde_json_path::AtMostOneError;
@@ -53,7 +53,7 @@ impl<'a> NodeList<'a> {
     /// This is intended for queries that are expected to yield exactly one node.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # use serde_json_path::ExactlyOneError;
@@ -87,7 +87,7 @@ impl<'a> NodeList<'a> {
     /// This is intended for queries that are expected to yield zero or more nodes.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -122,7 +122,7 @@ impl<'a> NodeList<'a> {
     /// Returns the first node in the [`NodeList`], or `None` if it is empty
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -140,7 +140,7 @@ impl<'a> NodeList<'a> {
     /// Returns the last node in the [`NodeList`], or `None` if it is empty
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -159,7 +159,7 @@ impl<'a> NodeList<'a> {
     /// out of bounds.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -180,7 +180,7 @@ impl<'a> NodeList<'a> {
     /// This is intended for queries that are expected to optionally yield a single node.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -267,7 +267,7 @@ impl<'a> LocatedNodeList<'a> {
     /// This is intended for queries that are expected to optionally yield a single node.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -298,7 +298,7 @@ impl<'a> LocatedNodeList<'a> {
     /// This is intended for queries that are expected to yield a single node.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -326,7 +326,7 @@ impl<'a> LocatedNodeList<'a> {
     /// This is intended for queries that are expected to yield zero or more nodes.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -362,7 +362,7 @@ impl<'a> LocatedNodeList<'a> {
     /// over just nodes, see [`nodes`][LocatedNodeList::nodes].
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::{json, Value};
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -385,7 +385,7 @@ impl<'a> LocatedNodeList<'a> {
     /// Get an iterator over the locations of nodes within a [`LocatedNodeList`]
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -414,7 +414,7 @@ impl<'a> LocatedNodeList<'a> {
     /// See also, [`dedup_in_place`][LocatedNodeList::dedup_in_place].
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -437,7 +437,7 @@ impl<'a> LocatedNodeList<'a> {
     /// See also, [`dedup`][LocatedNodeList::dedup].
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), serde_json_path::ParseError> {
@@ -461,7 +461,7 @@ impl<'a> LocatedNodeList<'a> {
     /// Return the first entry in the [`LocatedNodeList`], or `None` if it is empty
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # use serde_json_path::LocatedNode;
@@ -481,7 +481,7 @@ impl<'a> LocatedNodeList<'a> {
     /// Return the last entry in the [`LocatedNodeList`], or `None` if it is empty
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # use serde_json_path::LocatedNode;
@@ -502,7 +502,7 @@ impl<'a> LocatedNodeList<'a> {
     /// given index is out of bounds.
     ///
     /// # Usage
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # use serde_json_path::LocatedNode;
@@ -636,7 +636,6 @@ impl ExactlyOneError {
 mod tests {
     use crate::node::{LocatedNodeList, NodeList};
     use serde_json::{json, to_value};
-    use serde_json_path::JsonPath;
 
     #[test]
     fn test_send() {
@@ -651,11 +650,4 @@ mod tests {
         assert_sync::<NodeList>();
         assert_sync::<LocatedNodeList>();
     }
-
-    #[test]
-    fn test_serialize() {
-        let v = json!([1, 2, 3, 4]);
-        let q = JsonPath::parse("$.*").expect("valid query").query(&v);
-        assert_eq!(to_value(q).expect("serialize"), v);
-    }
 }
diff --git a/src/path.rs b/src/path.rs
index d2c3d19..bda003a 100644
--- a/src/path.rs
+++ b/src/path.rs
@@ -31,7 +31,7 @@ impl<'a> NormalizedPath<'a> {
     /// [`serde_json::Value::pointer_mut`] methods.
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -78,7 +78,7 @@ impl<'a> NormalizedPath<'a> {
     /// Note that [`NormalizedPath`] also implements [`IntoIterator`]
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -100,7 +100,7 @@ impl<'a> NormalizedPath<'a> {
     /// Get the [`PathElement`] at `index`, or `None` if the index is out of bounds
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -124,7 +124,7 @@ impl<'a> NormalizedPath<'a> {
     /// Get the first [`PathElement`], or `None` if the path is empty
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -143,7 +143,7 @@ impl<'a> NormalizedPath<'a> {
     /// Get the last [`PathElement`], or `None` if the path is empty
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
@@ -175,7 +175,7 @@ impl Display for NormalizedPath<'_> {
     /// as per the [JSONPath Specification][norm-paths]
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use serde_json::json;
     /// # use serde_json_path::JsonPath;
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
