From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:13:48 +0100
Subject: Remove benches

---
 Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -332,13 +332,6 @@
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.assert_cmd]
 version = "2.0"
 
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.6"
-
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.criterion]
-version = "0.6"
-default-features = false
-
 [target."cfg(unix)".dependencies.signal-hook]
 version = "0.3"
 optional = true
