From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:15:26 +0100
Subject: Remove wasm feature

---
 Cargo.toml | 5 -----
 1 file changed, 5 deletions(-)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -92,7 +92,6 @@
     "dep:tracing",
 ]
 unstable = []
-wasm = ["wasm-bindgen"]
 
 [lib]
 name = "rav1e"
@@ -271,10 +270,6 @@
 [dependencies.v_frame]
 version = "0.3.7"
 
-[dependencies.wasm-bindgen]
-version = "0.2.90"
-optional = true
-
 [dependencies.y4m]
 version = "0.8"
 optional = true
