From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:15:48 +0100
Subject: Remove fuzzing feature

---
 Cargo.toml | 13 -------------
 1 file changed, 13 deletions(-)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -309,21 +309,6 @@
 features = ["parallel"]
 optional = true
 
-[target."cfg(fuzzing)".dependencies.arbitrary]
-version = "1.3"
-
-[target."cfg(fuzzing)".dependencies.interpolate_name]
-version = "0.2.4"
-
-[target."cfg(fuzzing)".dependencies.libfuzzer-sys]
-version = "0.4.7"
-
-[target."cfg(fuzzing)".dependencies.rand]
-version = "0.9"
-
-[target."cfg(fuzzing)".dependencies.rand_chacha]
-version = "0.9"
-
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.assert_cmd]
 version = "2.0"
 
