From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:15:26 +0100
Subject: Remove wasm feature

---
 Cargo.toml | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index a1ff12d..97f4391 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -158,10 +158,6 @@ optional = true
 [dependencies.v_frame]
 version = "0.2.5"
 
-[dependencies.wasm-bindgen]
-version = "0.2.63"
-optional = true
-
 [dependencies.y4m]
 version = "0.7"
 optional = true
@@ -217,7 +213,6 @@ scenechange = []
 serialize = ["serde", "toml", "v_frame/serialize", "arrayvec/serde"]
 signal_support = ["signal-hook"]
 unstable = []
-wasm = ["wasm-bindgen"]
 [target."cfg(any(decode_test, decode_test_dav1d))".dependencies.system-deps]
 version = "~3.1.2"
 [target."cfg(fuzzing)".dependencies.arbitrary]
