From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:15:48 +0100
Subject: Remove fuzzing feature

---
 Cargo.toml | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 97f4391..afed08c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -215,20 +215,7 @@ signal_support = ["signal-hook"]
 unstable = []
 [target."cfg(any(decode_test, decode_test_dav1d))".dependencies.system-deps]
 version = "~3.1.2"
-[target."cfg(fuzzing)".dependencies.arbitrary]
-version = "0.4"
-
-[target."cfg(fuzzing)".dependencies.interpolate_name]
-version = "0.2.2"
-
-[target."cfg(fuzzing)".dependencies.libfuzzer-sys]
-version = "0.3"
-
-[target."cfg(fuzzing)".dependencies.rand]
-version = "0.8"
 
-[target."cfg(fuzzing)".dependencies.rand_chacha]
-version = "0.3"
 [target."cfg(unix)".dependencies.signal-hook]
 version = "0.3"
 optional = true
