From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:16:10 +0100
Subject: Bump console

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index e35cc77..56fb959 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -86,7 +86,7 @@ optional = true
 default-features = false
 
 [dependencies.console]
-version = "0.14"
+version = "0.15"
 optional = true
 
 [dependencies.crossbeam]
