--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,9 +64,7 @@
     "asm",
     "threading",
     "signal_support",
-    "git_version",
 ]
-git_version = ["built/git2"]
 serialize = [
     "serde",
     "toml",
@@ -255,10 +253,6 @@
 [dev-dependencies.semver]
 version = "1.0"
 
-[build-dependencies.built]
-version = "0.8.0"
-features = []
-
 [build-dependencies.cc]
 version = "1.0"
 features = ["parallel"]
--- a/build.rs
+++ b/build.rs
@@ -246,8 +246,6 @@
 
 #[allow(unused_variables)]
 fn main() {
-  built::write_built_file().expect("Failed to acquire build-time information");
-
   let arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
   let os = env::var("CARGO_CFG_TARGET_OS").unwrap();
   // let env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
--- a/src/bin/common.rs
+++ b/src/bin/common.rs
@@ -24,11 +24,6 @@
 use std::path::PathBuf;
 use std::sync::OnceLock;
 
-pub mod built_info {
-  // The file has been placed there by the build script.
-  include!(concat!(env!("OUT_DIR"), "/built.rs"));
-}
-
 #[derive(Clap)]
 #[clap(
   name = "rav1e",
@@ -295,8 +290,8 @@
     format!(
       "{}\n{} {}\nCompiled CPU Features: {}\nRuntime Assembly Support: {}{}\nThreading: {}\nUnstable Features: {}\nCompiler Flags: {}",
       get_version(),
-      built_info::RUSTC_VERSION,
-      built_info::TARGET,
+      "unknown rustc version",
+      "unknown target",
       option_env!("CARGO_CFG_TARGET_FEATURE").unwrap_or("(None)"),
       if cfg!(feature = "asm") { "Enabled" } else { "Disabled" },
       if cfg!(feature = "asm") {
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -49,11 +49,6 @@
 pub use crate::frame::Frame;
 pub use crate::util::{CastFromPrimitive, Pixel, PixelType};
 
-pub(crate) mod built_info {
-  // The file has been placed there by the build script.
-  include!(concat!(env!("OUT_DIR"), "/built.rs"));
-}
-
 mod serialize {
   cfg_if::cfg_if! {
     if #[cfg(feature="serialize")] {
