From: Sebastian Ramacher <sramacher@debian.org>
Date: Thu, 29 Dec 2022 11:16:18 +0100
Subject: Bump dav1d-sys

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 56fb959..883cd38 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -94,7 +94,7 @@ version = "0.8"
 optional = true
 
 [dependencies.dav1d-sys]
-version = "0.3.4"
+version = "0.7"
 optional = true
 
 [dependencies.fern]
