--- a/Cargo.toml
+++ b/Cargo.toml
@@ -150,9 +150,6 @@
 [dependencies.paste]
 version = "1.0"
 
-[dependencies.profiling]
-version = "1"
-
 [dependencies.rayon]
 version = "1"
 
@@ -180,18 +177,6 @@
 version = "0.8"
 optional = true
 
-[dependencies.tracing]
-version = "0.1.40"
-optional = true
-
-[dependencies.tracing-chrome]
-version = "0.7.1"
-optional = true
-
-[dependencies.tracing-subscriber]
-version = "0.3.18"
-optional = true
-
 [dependencies.v_frame]
 version = "0.3.7"
 
@@ -269,12 +254,6 @@
 ]
 signal_support = ["signal-hook"]
 threading = []
-tracing = [
-    "profiling/profile-with-tracing",
-    "tracing-subscriber",
-    "tracing-chrome",
-    "dep:tracing",
-]
 
 [target."cfg(any(decode_test, decode_test_dav1d))".dependencies.system-deps]
 version = "6"
