--- a/src/errors.rs
+++ b/src/errors.rs
@@ -22,7 +22,7 @@
 //!
 //! ## Read a MO file (IOError)
 //!
-//! ```rust
+//! ```ignore
 //! use rspolib::{mofile, MOFile, errors::IOError, MAGIC};
 //! use rspolib_testing::create_binary_content;
 //!
@@ -77,7 +77,7 @@
 ///
 /// ## Unsupported MO revision number
 ///
-/// ```rust
+/// ```ignore
 /// use rspolib::{mofile, errors::IOError, MAGIC};
 /// use rspolib_testing::create_binary_content;
 ///
@@ -93,7 +93,7 @@
 ///
 /// ## Incorrect magic number
 ///
-/// ```
+/// ```ignore
 /// use rspolib::{mofile, errors::IOError};
 /// use rspolib_testing::create_binary_content;
 ///
--- a/src/moparser.rs
+++ b/src/moparser.rs
@@ -324,6 +324,7 @@
     }
 }
 
+#[cfg(any())]
 #[cfg(test)]
 mod tests {
     use super::*;
