--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,3 +28,3 @@
 [dependencies.rand]
-version = "0.6"
+version = ">= 0.6, < 0.9"
 optional = true
@@ -38,3 +38,3 @@
 [build-dependencies.rustc_version]
-version = "0.2"
+version = "0.4"
 
--- a/build.rs
+++ b/build.rs
@@ -2,8 +2,6 @@
 extern crate semver;
-use rustc_version::{version_meta, Channel};
-use semver::Version;
+use rustc_version::{version, version_meta, Channel, Version};
 
 pub fn main() {
-    let version = version_meta().unwrap();
-    let channel = match version.channel {
+    let channel = match version_meta().unwrap().channel {
         Channel::Dev | Channel::Nightly => "unstable",
@@ -11,4 +9,5 @@
     };
+
     println!("cargo:rustc-cfg=extprim_channel=\"{}\"", channel);
-    if version.semver >= Version::new(1, 26, 0) {
+    if version().unwrap() >= Version::parse("1.26.0").unwrap() {
         println!("cargo:rustc-cfg=extprim_has_stable_i128");
