--- a/src/report/junit/mod.rs
+++ b/src/report/junit/mod.rs
@@ -230,21 +230,21 @@
         let doc = XmlDocument::new(suite);
         assert_eq!(
             doc.to_string().unwrap(),
-            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>\
             <testsuite tests=\"3\" errors=\"1\" failures=\"1\">\
                 <testcase id=\"test.hurl\" name=\"test.hurl\" time=\"0.230\" />\
                 <testcase id=\"test.hurl\" name=\"test.hurl\" time=\"0.230\">\
                     <failure>Assert status code\n  \
-                    --&gt; test.hurl:2:10\n   \
+                    --> test.hurl:2:10\n   \
                       |\n   \
                       | GET http://localhost:8000/not_found\n \
                     2 | HTTP/1.0 200\n   \
-                      |          ^^^ actual value is &lt;404&gt;\n   \
+                      |          ^^^ actual value is &lt;404>\n   \
                       |\
                     </failure>\
                 </testcase>\
                 <testcase id=\"test.hurl\" name=\"test.hurl\" time=\"0.230\">\
-                    <error>HTTP connection\n  --&gt; test.hurl:1:5\n   |\n 1 | GET http://localhost:8000/not_found\n   |     ^^^^^^^^^^^^^^ (6) Could not resolve host: unknown\n   |\
+                    <error>HTTP connection\n  --> test.hurl:1:5\n   |\n 1 | GET http://localhost:8000/not_found\n   |     ^^^^^^^^^^^^^^ (6) Could not resolve host: unknown\n   |\
                     </error>\
                 </testcase>\
             </testsuite>"
--- a/src/report/junit/testcase.rs
+++ b/src/report/junit/testcase.rs
@@ -118,7 +118,7 @@
         let doc = XmlDocument::new(element);
         assert_eq!(
             doc.to_string().unwrap(),
-            r#"<?xml version="1.0" encoding="UTF-8"?><testcase id="test.hurl" name="test.hurl" time="0.230" />"#
+            r#"<?xml version="1.0" encoding="utf-8"?><testcase id="test.hurl" name="test.hurl" time="0.230" />"#
         );
     }
 
@@ -156,12 +156,12 @@
         let doc = XmlDocument::new(element);
         assert_eq!(
             doc.to_string().unwrap(),
-            r#"<?xml version="1.0" encoding="UTF-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><failure>Assert status code
-  --&gt; test.hurl:2:10
+            r#"<?xml version="1.0" encoding="utf-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><failure>Assert status code
+  --> test.hurl:2:10
    |
    | GET http://localhost:8000/not_found
  2 | HTTP/1.0 200
-   |          ^^^ actual value is &lt;404&gt;
+   |          ^^^ actual value is &lt;404>
    |</failure></testcase>"#
         );
     }
@@ -198,8 +198,8 @@
         let doc = XmlDocument::new(element);
         assert_eq!(
             doc.to_string().unwrap(),
-            r#"<?xml version="1.0" encoding="UTF-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><error>HTTP connection
-  --&gt; test.hurl:1:5
+            r#"<?xml version="1.0" encoding="utf-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><error>HTTP connection
+  --> test.hurl:1:5
    |
  1 | GET http://unknown
    |     ^^^^^^^^^^^^^^ (6) Could not resolve host: unknown
--- a/src/report/junit/xml/writer.rs
+++ b/src/report/junit/xml/writer.rs
@@ -209,7 +209,7 @@
         let doc = XmlDocument { root: Some(root) };
         assert_eq!(
             doc.to_string().unwrap(),
-            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>\
             <catalog>\
                 <book id=\"bk101\">\
                     <author>Gambardella, Matthew</author>\
