--- a/src/report/junit/mod.rs
+++ b/src/report/junit/mod.rs
@@ -232,3 +232,3 @@
             doc.to_string().unwrap(),
-            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>\
             <testsuite tests=\"3\" errors=\"1\" failures=\"1\">\
@@ -237,3 +237,3 @@
                     <failure>Assert status code\n  \
-                    --&gt; test.hurl:2:10\n   \
+                    --> test.hurl:2:10\n   \
                       |\n   \
@@ -241,3 +241,3 @@
                     2 | HTTP/1.0 200\n   \
-                      |          ^^^ actual value is &lt;404&gt;\n   \
+                      |          ^^^ actual value is &lt;404>\n   \
                       |\
@@ -246,3 +246,3 @@
                 <testcase id=\"test.hurl\" name=\"test.hurl\" time=\"0.230\">\
-                    <error>HTTP connection\n  --&gt; test.hurl:1:5\n   |\n 1 | GET http://localhost:8000/not_found\n   |     ^^^^^^^^^^^^^^ (6) Could not resolve host: unknown\n   |\
+                    <error>HTTP connection\n  --> test.hurl:1:5\n   |\n 1 | GET http://localhost:8000/not_found\n   |     ^^^^^^^^^^^^^^ (6) Could not resolve host: unknown\n   |\
                     </error>\
--- a/src/report/junit/testcase.rs
+++ b/src/report/junit/testcase.rs
@@ -120,3 +120,3 @@
             doc.to_string().unwrap(),
-            r#"<?xml version="1.0" encoding="UTF-8"?><testcase id="test.hurl" name="test.hurl" time="0.230" />"#
+            r#"<?xml version="1.0" encoding="utf-8"?><testcase id="test.hurl" name="test.hurl" time="0.230" />"#
         );
@@ -158,4 +158,4 @@
             doc.to_string().unwrap(),
-            r#"<?xml version="1.0" encoding="UTF-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><failure>Assert status code
-  --&gt; test.hurl:2:10
+            r#"<?xml version="1.0" encoding="utf-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><failure>Assert status code
+  --> test.hurl:2:10
    |
@@ -163,3 +163,3 @@
  2 | HTTP/1.0 200
-   |          ^^^ actual value is &lt;404&gt;
+   |          ^^^ actual value is &lt;404>
    |</failure></testcase>"#
@@ -200,4 +200,4 @@
             doc.to_string().unwrap(),
-            r#"<?xml version="1.0" encoding="UTF-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><error>HTTP connection
-  --&gt; test.hurl:1:5
+            r#"<?xml version="1.0" encoding="utf-8"?><testcase id="test.hurl" name="test.hurl" time="0.230"><error>HTTP connection
+  --> test.hurl:1:5
    |
--- a/src/report/junit/xml/writer.rs
+++ b/src/report/junit/xml/writer.rs
@@ -211,3 +211,3 @@
             doc.to_string().unwrap(),
-            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>\
             <catalog>\
