--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,3 +49,3 @@
 [dependencies.brotli]
-version = "7.0.0"
+version = "6"
 
@@ -65,6 +65,6 @@
 [dependencies.curl]
-version = "0.4.47"
+version = "0.4"
 
 [dependencies.curl-sys]
-version = "0.4.78"
+version = "0.4"
 
@@ -89,3 +89,3 @@
 [dependencies.libflate]
-version = "2.1.0"
+version = "*"
 
@@ -121,6 +121,6 @@
 [dependencies.url]
-version = "2.5.4"
+version = "2.5.2"
 
 [dependencies.uuid]
-version = "1.11.0"
+version = "1.10.0"
 features = [
@@ -131,6 +131,6 @@
 [dependencies.xml-rs]
-version = "0.8.24"
+version = "0.8"
 
 [build-dependencies.cc]
-version = "1.2.2"
+version = "1.1"
 
@@ -143,6 +143,3 @@
 [target."cfg(unix)".dependencies.termion]
-version = "4.0.3"
-
-[target."cfg(windows)".build-dependencies.winres]
-version = "0.1.12"
+version = "*"
 
