--- hurl.orig/Cargo.toml
+++ hurl/Cargo.toml
@@ -56,6 +56,6 @@
 [dependencies.brotli]
-version = "7.0.0"
+version = "6"
 
 [dependencies.chrono]
-version = "0.4.40"
+version = "0.4.39"
 features = ["clock"]
@@ -64,3 +64,3 @@
 [dependencies.clap]
-version = "4.5.32"
+version = "4.5.23"
 features = [
@@ -71,6 +71,6 @@
 [dependencies.curl]
-version = "0.4.47"
+version = "0.4"
 
 [dependencies.curl-sys]
-version = "0.4.80"
+version = "0.4"
 
@@ -92,6 +92,6 @@
 [dependencies.libflate]
-version = "2.1.0"
+version = "*"
 
 [dependencies.libxml]
-version = "0.3.3"
+version = "0.3.5"
 
@@ -107,3 +107,3 @@
 [dependencies.serde]
-version = "1.0.219"
+version = "1.0.217"
 features = ["derive"]
@@ -111,3 +111,3 @@
 [dependencies.serde_json]
-version = "1.0.140"
+version = "1.0.139"
 features = ["arbitrary_precision"]
@@ -121,9 +121,9 @@
 [dependencies.terminal_size]
-version = "0.4.2"
+version = "0.4.1"
 
 [dependencies.url]
-version = "2.5.4"
+version = "2.5.2"
 
 [dependencies.uuid]
-version = "1.16.0"
+version = "1.10.0"
 features = [
@@ -134,3 +134,3 @@
 [dependencies.xml-rs]
-version = "0.8.25"
+version = "0.8.19"
 
@@ -140,6 +140,3 @@
 [target."cfg(unix)".dependencies.termion]
-version = "4.0.4"
-
-[target."cfg(windows)".build-dependencies.winres]
-version = "0.1.12"
+version = ">= 1.5.6, <5.0"
 
