--- a/Cargo.toml
+++ b/Cargo.toml
@@ -62,9 +63,6 @@
 version = "0.24"
 package = "gstreamer-video"
 
-[dev-dependencies.gst-check]
-version = "0.24"
-package = "gstreamer-check"
 
 [build-dependencies.gst-plugin-version-helper]
 version = "0.8"
--- a/tests/gifenc.rs
+++ b/tests/gifenc.rs
@@ -20,6 +20,7 @@
     });
 }
 
+/* gst_check isn't in Debian
 #[test]
 fn test_encode_rgba() {
     init();
@@ -70,3 +71,4 @@
         assert!(!buffer.flags().contains(gst::BufferFlags::DELTA_UNIT))
     }
 }
+*/
