--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,11 +34,11 @@
 repository = "https://github.com/sunfishcode/rustix-openpty"
 
 [dev-dependencies.rustix]
-version = "0.38.0"
+version = "1"
 features = ["termios"]
 
 [target."cfg(any(target_os = \"android\", target_os = \"linux\"))".dependencies.rustix]
-version = "0.38.0"
+version = "1"
 features = [
     "alloc",
     "fs",
@@ -58,7 +58,7 @@
 default-features = false
 
 [target."cfg(not(any(target_os = \"android\", target_os = \"linux\")))".dependencies.rustix]
-version = "0.38.0"
+version = "1"
 features = [
     "fs",
     "termios",
