Index: time/src/lib.rs
===================================================================
--- time.orig/src/lib.rs
+++ time/src/lib.rs
@@ -312,8 +312,6 @@ mod rand;
 #[allow(missing_copy_implementations, missing_debug_implementations)]
 pub mod serde;
 mod sys;
-#[cfg(test)]
-mod tests;
 mod time;
 mod utc_offset;
 pub mod util;
Index: time/Cargo.toml
===================================================================
--- time.orig/Cargo.toml
+++ time/Cargo.toml
@@ -33,10 +33,6 @@ debug = 0
 [lib]
 bench = false
 
-[[bench]]
-name = "benchmarks"
-path = "benchmarks/main.rs"
-harness = false
 [dependencies.itoa]
 version = "0.4.7"
 optional = true
@@ -59,9 +55,6 @@ default-features = false
 version = "1.0.126"
 default-features = false
 
-[dev-dependencies.serde_test]
-version = "1.0.126"
-
 [features]
 alloc = []
 default = ["std"]
@@ -70,12 +63,5 @@ local-offset = ["std"]
 parsing = []
 serde-human-readable = ["serde", "formatting", "parsing"]
 std = ["alloc"]
-[target."cfg(__ui_tests)".dev-dependencies.trybuild]
-version = "=1.0.34"
 [target."cfg(any(target_os = \"linux\", unsound_local_offset))".dependencies.libc]
 version = "0.2.98"
-[target."cfg(bench)".dev-dependencies.criterion]
-version = "0.3.5"
-
-[target."cfg(bench)".dev-dependencies.criterion-cycles-per-byte]
-version = "0.1.2"
