Index: time/Cargo.toml
===================================================================
--- time.orig/Cargo.toml
+++ time/Cargo.toml
@@ -41,12 +41,6 @@ harness = false
 version = "0.4.7"
 optional = true
 
-[dependencies.quickcheck-dep]
-version = "1.0.3"
-optional = true
-default-features = false
-package = "quickcheck"
-
 [dependencies.rand]
 version = "0.8.4"
 optional = true
@@ -79,7 +73,6 @@ large-dates = ["time-macros/large-dates"
 local-offset = ["std"]
 macros = ["time-macros"]
 parsing = []
-quickcheck = ["quickcheck-dep", "alloc"]
 serde-human-readable = ["serde", "formatting", "parsing"]
 std = ["alloc"]
 [target."cfg(__ui_tests)".dev-dependencies.trybuild]
