--- a/Cargo.toml
+++ b/Cargo.toml
@@ -88,7 +88,7 @@
 version = "=0.1.2"
 
 [dependencies.time-macros]
-version = "=0.2.18"
+version = "=0.2.16"
 optional = true
 
 [dev-dependencies.num-conv]
@@ -101,12 +101,6 @@
 version = "0.8.4"
 default-features = false
 
-[dev-dependencies.rstest]
-version = "0.18.2"
-default-features = false
-
-[dev-dependencies.rstest_reuse]
-version = "0.6.0"
 
 [dev-dependencies.serde]
 version = "1.0.184"
@@ -120,7 +114,7 @@
 version = "1.0.126"
 
 [dev-dependencies.time-macros]
-version = "=0.2.18"
+version = "=0.2.16"
 
 [features]
 alloc = ["serde?/alloc"]
