Index: mozim/Cargo.toml
===================================================================
--- mozim.orig/Cargo.toml
+++ mozim/Cargo.toml
@@ -55,7 +55,7 @@ version = "0.4.17"
 version = "1.2.13"
 
 [dependencies.nix]
-version = "0.27.0"
+version = "0.29.0"
 features = [
     "poll",
     "time",
Index: mozim/src/event.rs
===================================================================
--- mozim.orig/src/event.rs
+++ mozim/src/event.rs
@@ -234,11 +234,13 @@ impl DhcpEpoll {
     }
 
     fn poll(&self, wait_time: isize) -> Result<Vec<DhcpV4Event>, DhcpError> {
+        use nix::sys::epoll::EpollTimeout;
         let mut events: [EpollEvent; EVENT_BUFFER_COUNT] =
             [EpollEvent::empty(); EVENT_BUFFER_COUNT];
 
         loop {
-            match self.fd.wait(&mut events, 1000 * wait_time) {
+            let timeout: EpollTimeout = ((1000 * wait_time) as u64).try_into().unwrap();
+            match self.fd.wait(&mut events, timeout) {
                 Ok(c) => {
                     let mut ret = Vec::new();
                     for i in &events[..c] {
Index: mozim/src/client_async.rs
===================================================================
--- mozim.orig/src/client_async.rs
+++ mozim/src/client_async.rs
@@ -13,7 +13,7 @@ use nix::poll::{PollFd, PollFlags};
 
 use crate::{DhcpError, DhcpV4Client, DhcpV4Config, DhcpV4Lease, ErrorKind};
 
-const POLL_TIMEOUT: libc::c_int = 1000; // milliseconds
+const POLL_TIMEOUT: u16 = 1000; // milliseconds
 
 #[derive(Debug)]
 struct ShareState {
@@ -107,7 +107,7 @@ impl std::ops::Drop for DhcpV4ClientAsyn
 fn poll_thread(fd: RawFd, share_state: Arc<Mutex<ShareState>>) {
     let fd = unsafe { BorrowedFd::borrow_raw(fd) };
     let mut poll_fds = [PollFd::new(
-        &fd,
+        fd,
         PollFlags::POLLIN
             | PollFlags::POLLOUT
             | PollFlags::POLLHUP
