Description: only support the tonic version in Debian
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -71,56 +71,11 @@
     "sync",
 ]
 
-[dependencies.tonic010]
-version = "0.10"
-optional = true
-package = "tonic"
-
-[dependencies.tonic011]
-version = "0.11"
-optional = true
-package = "tonic"
-
 [dependencies.tonic012]
 version = "0.12"
 optional = true
 package = "tonic"
 
-[dependencies.tonic013]
-version = "0.13"
-optional = true
-package = "tonic"
-
-[dependencies.tonic014]
-version = "0.14"
-optional = true
-package = "tonic"
-
-[dependencies.tonic05]
-version = "0.5"
-optional = true
-package = "tonic"
-
-[dependencies.tonic06]
-version = "0.6"
-optional = true
-package = "tonic"
-
-[dependencies.tonic07]
-version = "0.7"
-optional = true
-package = "tonic"
-
-[dependencies.tonic08]
-version = "0.8"
-optional = true
-package = "tonic"
-
-[dependencies.tonic09]
-version = "0.9"
-optional = true
-package = "tonic"
-
 [dependencies.vsock]
 version = "0.5.1"
 
diff --git a/src/tonic_support.rs b/src/tonic_support.rs
index ef8bf31..8b45949 100644
--- a/src/tonic_support.rs
+++ b/src/tonic_support.rs
@@ -34,13 +34,4 @@ macro_rules! tonic_connected {
     };
 }
 
-tonic_connected!(tonic05 "tonic05");
-tonic_connected!(tonic06 "tonic06");
-tonic_connected!(tonic07 "tonic07");
-tonic_connected!(tonic08 "tonic08");
-tonic_connected!(tonic09 "tonic09");
-tonic_connected!(tonic010 "tonic010");
-tonic_connected!(tonic011 "tonic011");
 tonic_connected!(tonic012 "tonic012");
-tonic_connected!(tonic013 "tonic013");
-tonic_connected!(tonic014 "tonic014");
