Index: tokio-tungstenite/Cargo.toml
===================================================================
--- tokio-tungstenite.orig/Cargo.toml
+++ tokio-tungstenite/Cargo.toml
@@ -176,11 +176,6 @@ native-tls = [
     "tungstenite/native-tls",
     "handshake",
 ]
-native-tls-vendored = [
-    "native-tls",
-    "native-tls-crate/vendored",
-    "tungstenite/native-tls-vendored",
-]
 rustls-tls-native-roots = [
     "__rustls-tls",
     "rustls-native-certs",
