Index: tokio-tungstenite/Cargo.toml
===================================================================
--- tokio-tungstenite.orig/Cargo.toml
+++ tokio-tungstenite/Cargo.toml
@@ -78,7 +78,7 @@ optional = true
 default-features = false
 
 [dependencies.rustls-native-certs]
-version = "0.8.0"
+version = "0.6"
 optional = true
 
 [dependencies.rustls-pki-types]
Index: tokio-tungstenite/src/tls.rs
===================================================================
--- tokio-tungstenite.orig/src/tls.rs
+++ tokio-tungstenite/src/tls.rs
@@ -95,15 +95,19 @@ mod encryption {
                             let mut root_store = RootCertStore::empty();
                             #[cfg(feature = "rustls-tls-native-roots")]
                             {
-                                let rustls_native_certs::CertificateResult {
-                                    certs, errors, ..
-                                } = rustls_native_certs::load_native_certs();
+                                let certs = rustls_native_certs::load_native_certs();
+                                let mut errors = Vec::new();
 
-                                if !errors.is_empty() {
+                                let certs = match certs {
+                                  Err(error) => {
+                                    errors.push(error);
                                     log::warn!(
                                         "native root CA certificate loading errors: {errors:?}"
                                     );
-                                }
+                                    Vec::new()
+                                  },
+                                  Ok(certs) => certs
+                                };
 
                                 // Not finding any native root CA certificates is not fatal if the
                                 // "rustls-tls-webpki-roots" feature is enabled.
@@ -114,7 +118,7 @@ mod encryption {
 
                                 let total_number = certs.len();
                                 let (number_added, number_ignored) =
-                                    root_store.add_parsable_certificates(certs);
+                                    root_store.add_parsable_certificates(certs.into_iter().map(|cert| cert.0.into()));
                                 log::debug!("Added {number_added}/{total_number} native root certificates (ignored {number_ignored})");
                             }
                             #[cfg(feature = "rustls-tls-webpki-roots")]
