Index: hickory-resolver/Cargo.toml
===================================================================
--- hickory-resolver.orig/Cargo.toml
+++ hickory-resolver/Cargo.toml
@@ -124,7 +124,7 @@ features = ["logging", "std", "tls12"]
 optional = true
 
 [dependencies.rustls-native-certs]
-version = "0.7"
+version = "0.6"
 optional = true
 
 [dependencies.serde]
Index: hickory-resolver/src/tls/dns_over_rustls.rs
===================================================================
--- hickory-resolver.orig/src/tls/dns_over_rustls.rs
+++ hickory-resolver/src/tls/dns_over_rustls.rs
@@ -37,7 +37,7 @@ pub(crate) static CLIENT_CONFIG: Lazy<Re
         use proto::error::ProtoErrorKind;
 
         let (added, ignored) =
-            root_store.add_parsable_certificates(rustls_native_certs::load_native_certs()?);
+            root_store.add_parsable_certificates(rustls_native_certs::load_native_certs()?.into_iter().map(|cert| cert.0.into()));
 
         if ignored > 0 {
             tracing::warn!(
