Index: nix/src/sys/socket/mod.rs
===================================================================
--- nix.orig/src/sys/socket/mod.rs
+++ nix/src/sys/socket/mod.rs
@@ -1850,6 +1850,7 @@ mod test {
 
     #[cfg_attr(qemu, ignore)]
     #[test]
+    #[cfg(not(target_arch = "s390x"))]
     fn test_recvmm2() -> crate::Result<()> {
         use crate::sys::socket::{
             sendmsg, setsockopt, socket, sockopt::Timestamping, MsgFlags, SockFlag, SockType,
