Index: tree-sitter-loader/Cargo.toml
===================================================================
--- tree-sitter-loader.orig/Cargo.toml
+++ tree-sitter-loader/Cargo.toml
@@ -27,10 +27,10 @@ version = "1.0"
 version = "^1.0.58"
 
 [dependencies.dirs]
-version = "3.0"
+version = ">= 3.0, < 6.0"
 
 [dependencies.libloading]
-version = "0.7"
+version = ">= 0.7, < 1.0"
 
 [dependencies.once_cell]
 version = "1.7"
