Index: libheif-sys/Cargo.toml
===================================================================
--- libheif-sys.orig/Cargo.toml
+++ libheif-sys/Cargo.toml
@@ -60,24 +60,12 @@ file = "CHANGELOG.md"
 replace = "{{date}}"
 search = "ReleaseDate"
 
-[package.metadata.system-deps.libheif.v1_17]
-version = "1.17"
-
-[package.metadata.system-deps.libheif.v1_18]
-version = "1.18"
-
-[package.metadata.system-deps.libheif.v1_19]
-version = "1.19"
-
 [package.metadata.system-deps.libheif.v1_20]
 version = "1.20"
 
 [features]
 default = ["latest"]
 latest = ["v1_20"]
-v1_17 = []
-v1_18 = []
-v1_19 = []
 v1_20 = []
 
 [lib]
