--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,15 +72,6 @@
 [package.metadata.system-deps.libheif.v1_20]
 version = "1.20"
 
-[package.metadata.vcpkg]
-dependencies = ["libheif[aom]"]
-git = "https://github.com/microsoft/vcpkg"
-tag = "2025.07.25"
-
-[package.metadata.vcpkg.target.x86_64-pc-windows-msvc]
-dependencies = ["libheif[aom]"]
-triplet = "x64-windows-static-md"
-
 [features]
 embedded-libheif = ["dep:cmake"]
 latest = ["v1_20"]
@@ -122,9 +113,3 @@
 
 [build-dependencies.system-deps]
 version = "7.0"
-
-[target."cfg(windows)".build-dependencies.vcpkg]
-version = "0.2.15"
-
-[target."cfg(windows)".build-dependencies.walkdir]
-version = "2.5"
