Index: sequoia-keystore/Cargo.toml
===================================================================
--- sequoia-keystore.orig/Cargo.toml
+++ sequoia-keystore/Cargo.toml
@@ -67,10 +67,6 @@ version = "0.6"
 version = "0.4"
 optional = true
 
-[dependencies.sequoia-keystore-openpgp-card]
-version = "0.1"
-optional = true
-
 [dependencies.sequoia-keystore-softkeys]
 version = "0.6"
 optional = true
@@ -126,7 +122,6 @@ default = [
     "gpg-agent",
 ]
 gpg-agent = ["dep:sequoia-keystore-gpg-agent"]
-openpgp-card = ["dep:sequoia-keystore-openpgp-card"]
 softkeys = ["dep:sequoia-keystore-softkeys"]
 tpm = ["dep:sequoia-keystore-tpm"]
 
