From 111cb4b9cb9ab868f0570c6c36aabda4b0a1fe42 Mon Sep 17 00:00:00 2001
From: Wolfgang Silbermayr <wolfgang@silbermayr.at>
Date: Mon, 28 Jan 2019 20:35:50 +0100
Subject: [PATCH] Update nix dependency to 0.13

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 2de358a..6933600 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,7 +28,7 @@ path = "src/lib.rs"
 version = "0.2.28"
 
 [dependencies.nix]
-version = "0.11"
+version = "0.13"
 
 [dependencies.quick-error]
 version = "1.2.0"
-- 
2.20.1

