Index: redox-syscall/src/lib.rs
===================================================================
--- redox-syscall.orig/src/lib.rs
+++ redox-syscall/src/lib.rs
@@ -9,6 +9,7 @@ pub use self::daemon::*;
 pub use self::data::*;
 pub use self::error::*;
 pub use self::flag::*;
+#[cfg(target_os = "redox")]
 pub use self::io::*;
 pub use self::number::*;
 pub use self::scheme::*;
@@ -53,6 +54,7 @@ pub mod error;
 pub mod flag;
 
 /// Functions for low level hardware control
+#[cfg(target_os = "redox")]
 pub mod io;
 
 /// Call numbers used by each system call
