Index: toml-edit/Cargo.toml
===================================================================
--- toml-edit.orig/Cargo.toml
+++ toml-edit/Cargo.toml
@@ -110,5 +109,0 @@ required-features = [
-[[test]]
-name = "decoder_compliance"
-path = "tests/decoder_compliance.rs"
-harness = false
-required-features = ["parse"]
@@ -116,8 +110,0 @@ required-features = ["parse"]
-[[test]]
-name = "encoder_compliance"
-path = "tests/encoder_compliance.rs"
-harness = false
-required-features = [
-    "parse",
-    "display",
-]
@@ -125,8 +111,0 @@ required-features = [
-[[test]]
-name = "invalid"
-path = "tests/invalid.rs"
-harness = false
-required-features = [
-    "parse",
-    "display",
-]
@@ -178,6 +156,0 @@ version = "0.6.0"
-
-[dev-dependencies.toml-test-data]
-version = "1.11.0"
-
-[dev-dependencies.toml-test-harness]
-version = "0.4.8"
