Index: toml-edit/tests/convert.rs
===================================================================
--- toml-edit.orig/tests/convert.rs
+++ toml-edit/tests/convert.rs
@@ -1,4 +1,4 @@
-use pretty_assertions::assert_eq;
+use std::assert_eq;
 
 use toml_edit::{Document, Item, Value};
 
Index: toml-edit/tests/test_edit.rs
===================================================================
--- toml-edit.orig/tests/test_edit.rs
+++ toml-edit/tests/test_edit.rs
@@ -1,4 +1,4 @@
-use pretty_assertions::assert_eq;
+use std::assert_eq;
 use std::fmt;
 use std::iter::FromIterator;
 use toml_edit::{array, table, value, Document, Item, Key, Table, Value};
Index: toml-edit/Cargo.toml
===================================================================
--- toml-edit.orig/Cargo.toml
+++ toml-edit/Cargo.toml
@@ -126,9 +126,6 @@ version = "1"
 features = ["derive"]
 optional = true
 
-[dev-dependencies.pretty_assertions]
-version = "1.0.0"
-
 [dev-dependencies.serde_json]
 version = "1.0.44"
 
Index: toml-edit/src/parser/mod.rs
===================================================================
--- toml-edit.orig/src/parser/mod.rs
+++ toml-edit/src/parser/mod.rs
@@ -157,7 +157,7 @@ mod tests {
     use crate::parser::*;
     use combine::stream::position::Stream;
     use combine::*;
-    use pretty_assertions::assert_eq;
+    use std::assert_eq;
     use std::fmt;
     // Copied from https://github.com/colin-kiegel/rust-pretty-assertions/issues/24
     /// Wrapper around string slice that makes debug output `{:?}` to print string same way as `{}`.
Index: toml-edit/tests/pretty.rs
===================================================================
--- toml-edit.orig/tests/pretty.rs
+++ toml-edit/tests/pretty.rs
@@ -1,6 +1,6 @@
 #![cfg(feature = "easy")]
 
-use pretty_assertions::assert_eq;
+use std::assert_eq;
 
 const PRETTY_STD: &str = r#"[example]
 array = [
Index: toml-edit/tests/test_valid.rs
===================================================================
--- toml-edit.orig/tests/test_valid.rs
+++ toml-edit/tests/test_valid.rs
@@ -1,4 +1,4 @@
-use pretty_assertions::assert_eq;
+use std::assert_eq;
 use serde_json::Map as JsonMap;
 use serde_json::Value as Json;
 
