--- a/Cargo.toml
+++ b/Cargo.toml
@@ -106,7 +106,7 @@
 ]
 
 [dependencies.indexmap]
-version = "2.0.0"
+version = ">= 1.9.1"
 features = ["std"]
 
 [dependencies.kstring]
@@ -119,7 +119,7 @@
 optional = true
 
 [dependencies.serde_spanned]
-version = "0.6.5"
+version = ">= 0.6.4"
 features = ["serde"]
 optional = true
 
@@ -131,13 +131,13 @@
 optional = true
 
 [dev-dependencies.libtest-mimic]
-version = "0.7.0"
+version = "0.6.0"
 
 [dev-dependencies.serde_json]
 version = "1.0.115"
 
 [dev-dependencies.snapbox]
-version = "0.4.16"
+version = "0.4.8"
 features = ["harness"]
 
 [features]
