Index: toml-edit/Cargo.toml
===================================================================
--- toml-edit.orig/Cargo.toml
+++ toml-edit/Cargo.toml
@@ -152,9 +152,6 @@ version = "1.5.0"
 [dev-dependencies.serde_json]
 version = "1.0.116"
 
-[dev-dependencies.snapbox]
-version = "0.6.0"
-
 [features]
 default = [
     "parse",
Index: toml-edit/src/parser/mod.rs
===================================================================
--- toml-edit.orig/src/parser/mod.rs
+++ toml-edit/src/parser/mod.rs
@@ -157,8 +157,8 @@ pub(crate) mod prelude {
 #[cfg(feature = "display")]
 mod test {
     use super::*;
-    use snapbox::assert_data_eq;
-    use snapbox::prelude::*;
+    //use snapbox::assert_data_eq;
+    //use snapbox::prelude::*;
 
     #[test]
     fn documents() {
@@ -233,7 +233,7 @@ key = "value"
                 }
             };
 
-            assert_data_eq!(doc.to_string(), input.raw());
+            //assert_data_eq!(doc.to_string(), input.raw());
         }
     }
 
Index: toml-edit/tests/testsuite/main.rs
===================================================================
--- toml-edit.orig/tests/testsuite/main.rs
+++ toml-edit/tests/testsuite/main.rs
@@ -1,9 +1,9 @@
 #![recursion_limit = "256"]
 
-mod convert;
-mod datetime;
-mod edit;
+//mod convert;
+//mod datetime;
+//mod edit;
 mod float;
-mod invalid;
-mod parse;
+//mod invalid;
+//mod parse;
 mod stackoverflow;
