Index: toml-edit/tests/testsuite/stackoverflow.rs
===================================================================
--- toml-edit.orig/tests/testsuite/stackoverflow.rs
+++ toml-edit/tests/testsuite/stackoverflow.rs
@@ -2,2 +2,3 @@
 #[cfg(not(feature = "unbounded"))]
+#[ignore]
 fn array_recursion_limit() {
@@ -7,2 +7,3 @@
 #[cfg(not(feature = "unbounded"))]
+#[ignore]
 fn inline_table_recursion_limit() {
