Index: knuffel/Cargo.toml
===================================================================
--- knuffel.orig/Cargo.toml
+++ knuffel/Cargo.toml
@@ -33,11 +33,11 @@ categories = [
 license = "MIT/Apache-2.0"
 
 [dependencies.base64]
-version = "0.21.0"
+version = ">= 0.21.0"
 optional = true
 
 [dependencies.chumsky]
-version = "0.9.2"
+version = ">= 0.9.2"
 default-features = false
 
 [dependencies.knuffel-derive]
@@ -45,7 +45,7 @@ version = "^3.2.0"
 optional = true
 
 [dependencies.miette]
-version = "5.1.1"
+version = ">= 5.1.1"
 
 [dependencies.minicbor]
 version = "0.19.1"
@@ -66,7 +66,7 @@ optional = true
 version = "2.0.1"
 
 [dev-dependencies.miette]
-version = "5.1.1"
+version = ">= 5.1.1"
 features = ["fancy"]
 
 [dev-dependencies.serde_json]
