Index: knuffel/src/errors.rs
===================================================================
--- knuffel.orig/src/errors.rs
+++ knuffel/src/errors.rs
@@ -26,7 +26,7 @@ use crate::traits::{ErrorSpan, Span};
 #[error("error parsing KDL")]
 pub struct Error {
     #[source_code]
-    pub(crate) source_code: NamedSource,
+    pub(crate) source_code: NamedSource<String>,
     #[related]
     pub(crate) errors: Vec<miette::Report>,
 }
