Index: comrak/Cargo.toml
===================================================================
--- comrak.orig/Cargo.toml
+++ comrak/Cargo.toml
@@ -134,9 +134,6 @@ features = [
 optional = true
 default-features = false
 
-[target."cfg(not(target_arch=\"wasm32\"))".dev-dependencies.propfuzz]
-version = "0.0.1"
-
 [target."cfg(target_arch=\"wasm32\")".dependencies.clap]
 version = "4.0"
 features = [
Index: comrak/src/tests.rs
===================================================================
--- comrak.orig/src/tests.rs
+++ comrak/src/tests.rs
@@ -15,7 +15,6 @@ mod header_ids;
 mod options;
 mod pathological;
 mod plugins;
-mod propfuzz;
 mod regressions;
 mod shortcodes;
 mod strikethrough;
