Description: remove nightly feature
Author: Ananthu C V <weepingclown@disroot.org>
Forwarded: not-needed
Last-Update: 2024-06-26
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,9 +34,6 @@
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/asomers/mockall"
 
-[package.metadata.docs.rs]
-features = ["nightly"]
-
 [package.metadata.release]
 tag-prefix = ""
 
@@ -96,10 +93,4 @@
 version = "1.0"
 
 [dev-dependencies.tracing]
-version = "0.1.32"
-
-[features]
-nightly = [
-    "mockall_derive/nightly_derive",
-    "downcast/nightly",
-]
+version = "0.1.32"
\ No newline at end of file
