--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,10 +19,10 @@
 autotests = false
 autobenches = false
 description = "A tool to sync Wifi passwords with macos on ARM Macs"
-homepage = "https://github.com/WhatAmISupposedToPutHere/asahi-nvram"
+homepage = "https://github.com/AsahiLinux/asahi-nvram"
 readme = false
 license = "MIT"
-repository = "https://github.com/WhatAmISupposedToPutHere/asahi-nvram"
+repository = "https://github.com/AsahiLinux/asahi-nvram"
 
 [[bin]]
 name = "asahi-wifisync"
