--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,22 +55,7 @@
     "pkg-config",
     "vcpkg",
 ]
-bundled = [
-    "cc",
-    "bundled_bindings",
-]
-bundled-sqlcipher = ["bundled"]
-bundled-sqlcipher-vendored-openssl = [
-    "bundled-sqlcipher",
-    "openssl-sys/vendored",
-]
-bundled-windows = [
-    "cc",
-    "bundled_bindings",
-]
-bundled_bindings = []
 default = ["min_sqlite_version_3_14_0"]
-in_gecko = []
 min_sqlite_version_3_14_0 = [
     "pkg-config",
     "vcpkg",
@@ -83,5 +68,4 @@
 sqlcipher = []
 unlock_notify = []
 wasm32-wasi-vfs = []
-winsqlite3 = []
 with-asan = []
