Index: jiff/tests/lib.rs
===================================================================
--- jiff.orig/tests/lib.rs
+++ jiff/tests/lib.rs
@@ -1,9 +1,9 @@
 // See: https://github.com/rust-lang/rust/pull/121364
 #![allow(unknown_lints, ambiguous_negative_literals)]
 
-mod init;
-mod procmacro;
-mod tc39_262;
+//mod init;
+//mod procmacro;
+//mod tc39_262;
 
 /// The simplest possible logger that logs to stderr.
 ///
