Index: jiff/Cargo.toml
===================================================================
--- jiff.orig/Cargo.toml
+++ jiff/Cargo.toml
@@ -73,17 +73,12 @@ default = [
 logging = ["dep:log"]
 perf-inline = []
 serde = ["dep:serde"]
-static = [
-    "static-tz",
-    "jiff-static?/tzdb",
-]
-static-tz = ["dep:jiff-static"]
 std = [
     "alloc",
     "log?/std",
     "serde?/std",
 ]
-tz-fat = ["jiff-static?/tz-fat"]
+tz-fat = []
 tz-system = [
     "std",
 ]
@@ -105,10 +100,6 @@ path = "src/lib.rs"
 name = "integration"
 path = "tests/lib.rs"
 
-[dependencies.jiff-static]
-version = "0.2"
-optional = true
-
 [dependencies.jiff-tzdb]
 version = "0.1.4"
 optional = true
