--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,6 @@
 [features]
 bzip2 = ["async-compression/bzip2"]
 deflate = ["async-compression/deflate"]
-deflate64 = ["async-compression/deflate64"]
 full = [
     "chrono",
     "tokio-fs",
@@ -56,7 +55,6 @@
     "lzma",
     "zstd",
     "xz",
-    "deflate64",
 ]
 full-wasm = [
     "chrono",
@@ -133,12 +131,6 @@
 features = ["compat"]
 optional = true
 
-[dev-dependencies.actix-multipart]
-version = "0.7"
-
-[dev-dependencies.actix-web]
-version = "4"
-
 [dev-dependencies.anyhow]
 version = "1"
 
@@ -174,4 +166,4 @@
 ]
 
 [dev-dependencies.zip]
-version = "2.1.5"
+version = "5"
