diff --git a/Cargo.toml b/Cargo.toml
index a199caa..2854996 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -82,9 +82,9 @@ nightly = []
 version = "0.38.31"
 features = ["fs"]
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59"
-features = [
-    "Win32_Storage_FileSystem",
-    "Win32_Foundation",
-]
+#[target."cfg(windows)".dependencies.windows-sys]
+#version = "0.59"
+#features = [
+#    "Win32_Storage_FileSystem",
+#    "Win32_Foundation",
+#]
