Index: tempfile/Cargo.toml
===================================================================
--- tempfile.orig/Cargo.toml
+++ tempfile/Cargo.toml
@@ -31,7 +31,7 @@ version = "0.7"
 [dependencies.remove_dir_all]
 version = "0.5"
 [target."cfg(target_os = \"redox\")".dependencies.redox_syscall]
-version = "0.2"
+version = ">= 0.2, < 0.3"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.27"
 [target."cfg(windows)".dependencies.winapi]
