Subject: We don't need benchmarks in Debian packaging

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,15 +32,6 @@ repository = "https://github.com/ruma/ruma"
 [package.metadata.docs.rs]
 all-features = true
 
-[[bench]]
-name = "state_res_bench"
-harness = false
-required-features = ["criterion"]
-
-[dependencies.criterion]
-version = "0.3.3"
-optional = true
-
 [dependencies.itertools]
 version = ">= 0.10.0, < 1.0"
 
