--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,5 +48,5 @@
 [build-dependencies.cc]
 version = "1.0"
 
-[build-dependencies.temp-dir]
-version = "0.1.11"
+[build-dependencies.tempfile]
+version = "3.2.0"
--- a/build.rs
+++ b/build.rs
@@ -1,5 +1,5 @@
 extern crate cc;
-extern crate temp_dir;
+extern crate tempfile;
 
 use std::env;
 use std::ffi::OsString;
@@ -7,7 +7,7 @@
 use std::io::ErrorKind;
 use std::path::{Path, PathBuf};
 use std::process::{Command, Stdio};
-use temp_dir::TempDir;
+use tempfile::tempdir;
 
 fn env(name: &str) -> Option<String> {
     let prefix = env::var("TARGET").unwrap().to_uppercase().replace("-", "_");
@@ -191,7 +191,7 @@
 
     let target = env::var("TARGET").unwrap();
     let src = env::current_dir().unwrap();
-    let build_dir = TempDir::new().unwrap();
+    let build_dir = tempdir().unwrap();
     let build_dir = build_dir.path();
     let compiler = cc::Build::new().get_compiler();
 

