--- a/build.rs
+++ b/build.rs
@@ -100,7 +100,7 @@
 fn try_gettext_system() -> bool {
     let target = env::var("TARGET").unwrap();
 
-    if cfg!(feature = "gettext-system") || env("GETTEXT_SYSTEM").is_some() {
+    if cfg!(feature = "gettext-system") || env("GETTEXT_SYSTEM").is_some() || true{
         if (target.contains("linux") || target.contains("hurd"))
             && (target.contains("-gnu") || target.contains("-musl"))
         {

