--- a/Cargo.toml
+++ b/Cargo.toml
@@ -106,12 +106,12 @@
 version = "0.2.1"
 optional = true
 
-[dev-dependencies.codegenrs]
-version = "3.0.1"
-default-features = false
+#[dev-dependencies.codegenrs]
+#version = "3.0.1"
+#default-features = false
 
-[dev-dependencies.divan]
-version = "0.1.14"
+#[dev-dependencies.divan]
+#version = "0.1.14"
 
 [dev-dependencies.proptest]
 version = "1.4.0"
--- a/src/state/codegen.rs
+++ b/src/state/codegen.rs
@@ -3,7 +3,7 @@
 use snapbox::file;
 use vte_generate_state_changes::generate_state_changes;
 
-#[test]
+/*#[test]
 fn table() {
     let mut content = vec![];
     generate_table(&mut content).unwrap();
@@ -11,7 +11,7 @@
     let content = String::from_utf8(content).unwrap();
     let content = codegenrs::rustfmt(&content, None).unwrap();
     snapbox::assert_data_eq!(content, file!["table.rs"].raw());
-}
+}*/
 
 #[allow(clippy::write_literal)]
 fn generate_table(file: &mut impl std::io::Write) -> std::io::Result<()> {
