Index: bitflags-1/Cargo.toml
===================================================================
--- bitflags-1.orig/Cargo.toml
+++ bitflags-1/Cargo.toml
@@ -26,14 +26,7 @@ license = "MIT/Apache-2.0"
 repository = "https://github.com/bitflags/bitflags"
 [package.metadata.docs.rs]
 features = ["example_generated"]
-[dependencies.compiler_builtins]
-version = "0.1.2"
-optional = true
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
+
 [dev-dependencies.rustversion]
 version = "1.0"
 
@@ -56,4 +49,3 @@ version = "2.3"
 [features]
 default = []
 example_generated = []
-rustc-dep-of-std = ["core", "compiler_builtins"]
