--- a/Cargo.toml
+++ b/Cargo.toml
@@ -91,6 +91,6 @@ version = "6.1"
 [dependencies.vhost]
-version = "0.13.0"
+version = "0.14.0"
 
 [dependencies.vhost-user-backend]
-version = "0.17.0"
+version = "0.20.0"
 
@@ -100,3 +100,3 @@ version = "0.2.1"
 [dependencies.virtio-queue]
-version = "0.14.0"
+version = "0.16"
 
@@ -110,3 +110,3 @@ features = [
 [dependencies.vmm-sys-util]
-version = "0.12.1"
+version = "0.15"
 
