Index: virtiofsd/Cargo.toml
===================================================================
--- virtiofsd.orig/Cargo.toml
+++ virtiofsd/Cargo.toml
@@ -59,7 +59,7 @@ version = "0.6.0"
 version = "0.8.0"
 
 [dependencies.virtio-bindings]
-version = "0.1"
+version = "0.2"
 features = ["virtio-v5_0_0"]
 
 [dependencies.virtio-queue]
Index: virtiofsd/src/main.rs
===================================================================
--- virtiofsd.orig/src/main.rs
+++ virtiofsd/src/main.rs
@@ -22,7 +22,7 @@ use vhost::vhost_user::Error::PartialMes
 use vhost::vhost_user::{Listener, SlaveFsCacheReq};
 use vhost_user_backend::Error::HandleRequest;
 use vhost_user_backend::{VhostUserBackend, VhostUserDaemon, VringMutex, VringState, VringT};
-use virtio_bindings::bindings::virtio_net::*;
+use virtio_bindings::bindings::virtio_config::*;
 use virtio_bindings::bindings::virtio_ring::{
     VIRTIO_RING_F_EVENT_IDX, VIRTIO_RING_F_INDIRECT_DESC,
 };
