--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,36 +56,3 @@
 
 [target.'cfg(not(any(target_os = "macos", windows)))'.build-dependencies.pkg-config]
 version = "0.3"
-
-[target.'cfg(target_os = "macos")'.dependencies.core-foundation]
-version = "0.9.3"
-
-[target.'cfg(target_os = "macos")'.dependencies.core-foundation-sys]
-version = "0.8.4"
-
-[target.'cfg(target_os = "macos")'.dependencies.core-graphics]
-version = "0.23.1"
-
-[target.'cfg(target_os = "macos")'.dependencies.core-text]
-version = "20.1.0"
-
-[target.'cfg(target_os = "macos")'.dependencies.objc2]
-version = "0.5.1"
-
-[target.'cfg(target_os = "macos")'.dependencies.objc2-foundation]
-version = "0.2.2"
-features = [
-    "NSString",
-    "NSUserDefaults",
-    "NSValue",
-]
-
-[target.'cfg(target_os = "macos")'.dependencies.once_cell]
-version = "1.12"
-
-[target."cfg(windows)".dependencies.dwrote]
-version = "0.11"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = ["impl-default"]
