Index: sapling-streampager/Cargo.toml
===================================================================
--- sapling-streampager.orig/Cargo.toml
+++ sapling-streampager/Cargo.toml
@@ -41,10 +41,10 @@ name = "streampager"
 path = "src/lib.rs"
 
 [dependencies.bit-set]
-version = "0.5"
+version = ">= 0.8"
 
 [dependencies.dirs]
-version = "6.0"
+version = ">= 5.0"
 
 [dependencies.enum_dispatch]
 version = "0.3.13"
@@ -56,11 +56,11 @@ version = "2.2.6"
 version = "1.5"
 
 [dependencies.lru]
-version = "=0.12.4"
+version = ">= 0.12.3"
 default-features = false
 
 [dependencies.memmap2]
-version = "0.5.10"
+version = ">= 0.9.5"
 
 [dependencies.notify]
 version = "5"
@@ -87,7 +87,7 @@ version = "3.15"
 version = "0.9"
 
 [dependencies.termwiz]
-version = "0.23"
+version = ">= 0.22"
 
 [dependencies.thiserror]
 version = "2"
@@ -99,7 +99,7 @@ version = "0.8.19"
 version = "1.12.0"
 
 [dependencies.unicode-width]
-version = "=0.1.12"
+version = ">= 0.2.0"
 
 [dependencies.vec_map]
 version = "0.8"
