Index: libc/build.rs
===================================================================
--- libc.orig/build.rs
+++ libc/build.rs
@@ -303,6 +303,14 @@ fn set_cfg(cfg: &str) {
 }
 
 fn is_gnu_time64_abi() -> bool {
+    match env::var("CARGO_CFG_TARGET_ARCH") {
+        Ok(target_arch) => {
+            if target_arch == "x86" {
+                return false;
+            }
+        }
+        Err(_) => panic!("CARGO_CFG_TARGET_ARCH not set"),
+    }
     match env::var("CARGO_CFG_TARGET_ENV") {
         Ok(target_env) => {
             if target_env != "gnu" {
