Index: sval/src/value/owned.rs
===================================================================
--- sval.orig/src/value/owned.rs
+++ sval/src/value/owned.rs
@@ -898,12 +898,12 @@ mod tests {
     fn owned_value_size() {
         let size = mem::size_of::<OwnedValue>();
         let limit = {
-            #[cfg(all(target_arch = "aarch64", target_os = "macos"))]
+            #[cfg(all(target_arch = "aarch64"))]
             {
                 mem::size_of::<u64>() * 6
             }
 
-            #[cfg(not(all(target_arch = "aarch64", target_os = "macos")))]
+            #[cfg(not(all(target_arch = "aarch64")))]
             {
                 mem::size_of::<u64>() * 5
             }
