Index: vergen/Cargo.toml
===================================================================
--- vergen.orig/Cargo.toml
+++ vergen/Cargo.toml
@@ -71,9 +71,0 @@
-[dependencies.gix]
-version = "0.63.0"
-features = [
-    "revision",
-    "interrupt",
-]
-optional = true
-default-features = false
-
@@ -93,3 +88,0 @@
-[dev-dependencies.gix]
-version = "0.63.0"
-
@@ -117,10 +107,6 @@ git2 = [
     "git2-rs",
 ]
 gitcl = ["time"]
-gitoxide = [
-    "time",
-    "gix",
-]
 rustc = ["rustc_version"]
 si = ["sysinfo"]
 unstable = []
Index: vergen/src/lib.rs
===================================================================
--- vergen.orig/src/lib.rs
+++ vergen/src/lib.rs
@@ -393,3 +393,3 @@
 #[cfg(test)]
-use {gix as _, lazy_static as _, regex as _, serial_test as _, temp_env as _};
+use {lazy_static as _, regex as _, serial_test as _, temp_env as _};
 
