--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,7 +65,7 @@
 default-features = false
 
 [dev-dependencies.fastrand]
-version = "1.3.5"
+version = "2"
 default-features = false
 
 [features]
--- a/src/test_utils.rs
+++ b/src/test_utils.rs
@@ -11,9 +11,9 @@
 macro_rules! __declare_rng_ext {
     () => {
         pub trait RngExt {
-            fn as_rng(&self) -> &fastrand::Rng;
+            fn as_rng(&mut self) -> &fastrand::Rng;
 
-            fn pick<'a, T>(&self, slice: &'a [T]) -> &'a T {
+            fn pick<'a, T>(&mut self, slice: &'a [T]) -> &'a T {
                 &slice[self.as_rng().usize(0..slice.len())]
             }
 
@@ -48,7 +48,7 @@
         }
 
         impl RngExt for fastrand::Rng {
-            fn as_rng(&self) -> &fastrand::Rng {
+            fn as_rng(&mut self) -> &mut fastrand::Rng {
                 self
             }
         }
