--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,7 +33,7 @@
 version = "0.8.0"
 
 [dependencies.rustix]
-version = "0.38.22"
+version = "1"
 
 [features]
 use_bindgen = ["drm-sys/use_bindgen"]
--- a/src/ioctl.rs
+++ b/src/ioctl.rs
@@ -2,14 +2,13 @@
 
 use drm_sys::*;
 use rustix::ioctl::{
-    ioctl, Getter, NoArg, NoneOpcode, ReadOpcode, ReadWriteOpcode, Setter, Updater, WriteOpcode,
+    ioctl, opcode, Getter, NoArg, Setter, Updater,
 };
 
 macro_rules! ioctl_readwrite {
     ($name:ident, $ioty:expr, $nr:expr, $ty:ty) => {
         pub unsafe fn $name(fd: BorrowedFd, data: &mut $ty) -> io::Result<()> {
-            type Opcode = ReadWriteOpcode<$ioty, $nr, $ty>;
-            Ok(ioctl(fd, Updater::<Opcode, $ty>::new(data))?)
+            Ok(ioctl(fd, Updater::<{ opcode::read_write::<$ty>($ioty, $nr) }, $ty>::new(data))?)
         }
     };
 }
@@ -17,8 +16,7 @@
 macro_rules! ioctl_read {
     ($name:ident, $ioty:expr, $nr:expr, $ty:ty) => {
         pub unsafe fn $name(fd: BorrowedFd) -> io::Result<$ty> {
-            type Opcode = ReadOpcode<$ioty, $nr, $ty>;
-            Ok(ioctl(fd, Getter::<Opcode, $ty>::new())?)
+            Ok(ioctl(fd, Getter::<{ opcode::read::<$ty>($ioty, $nr) }, $ty>::new())?)
         }
     };
 }
@@ -26,8 +24,7 @@
 macro_rules! ioctl_write_ptr {
     ($name:ident, $ioty:expr, $nr:expr, $ty:ty) => {
         pub unsafe fn $name(fd: BorrowedFd, data: &$ty) -> io::Result<()> {
-            type Opcode = WriteOpcode<$ioty, $nr, $ty>;
-            Ok(ioctl(fd, Setter::<Opcode, $ty>::new(*data))?)
+            Ok(ioctl(fd, Setter::<{ opcode::write::<$ty>($ioty, $nr) }, $ty>::new(*data))?)
         }
     };
 }
@@ -35,8 +32,7 @@
 macro_rules! ioctl_none {
     ($name:ident, $ioty:expr, $nr:expr) => {
         pub unsafe fn $name(fd: BorrowedFd) -> io::Result<()> {
-            type Opcode = NoneOpcode<$ioty, $nr, ()>;
-            Ok(ioctl(fd, NoArg::<Opcode>::new())?)
+            Ok(ioctl(fd, NoArg::<{ opcode::none($ioty, $nr) }>::new())?)
         }
     };
 }
