Description: relax dependency version constraints
Author: Ananthu C V <weepingclown@debian.org>
Forwarded: not-needed
Last-Update: 2025-08-17
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,3 +68,3 @@
 [dependencies.comrak]
-version = "0.39"
+version = "0.41"
 default-features = false
@@ -72,3 +72,3 @@
 [dependencies.crossterm]
-version = "0.29"
+version = "0.27"
 features = [
@@ -80,6 +80,6 @@
 [dependencies.directories]
-version = "6.0"
+version = "5.0"
 
 [dependencies.fastrand]
-version = "2.3"
+version = "2.1"
 
@@ -101,3 +101,3 @@
 [dependencies.itertools]
-version = "0.14"
+version = "0.13"
 
@@ -137,3 +137,3 @@
 [dependencies.strum]
-version = "0.27"
+version = "0.26"
 features = ["derive"]
