--- a/Cargo.toml
+++ b/Cargo.toml
@@ -81,11 +81,6 @@
 name = "simd"
 path = "tests/simd.rs"
 
-[[bench]]
-name = "benchmark"
-path = "benches/benchmark.rs"
-harness = false
-
 [dependencies.multiversion-macros]
 version = "0.8.0"
 default-features = false
@@ -93,9 +88,6 @@
 [dependencies.target-features]
 version = "0.1"
 
-[dev-dependencies.criterion]
-version = "0.4"
-
 [dev-dependencies.rustversion]
 version = "1"
 
