Index: region/Cargo.toml
===================================================================
--- region.orig/Cargo.toml
+++ region/Cargo.toml
@@ -34,18 +34,6 @@ version = "1.0"
 [dependencies.libc]
 version = "0.2.153"
 
-[target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dependencies.mach2]
-version = "0.4"
-
 [target."cfg(unix)".dev-dependencies.mmap]
 version = "0.1.6"
 package = "mmap-fixed"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Memory",
-    "Win32_System_SystemInformation",
-    "Win32_System_Diagnostics_Debug",
-]
