Index: reis/Cargo.toml
===================================================================
--- reis.orig/Cargo.toml
+++ reis/Cargo.toml
@@ -22,22 +22,7 @@ keywords = [
     "wayland",
 ]
 license = "MIT"
-
-[[example]]
-name = "receive"
-required-features = ["tokio"]
-
-[[example]]
-name = "list-devices"
-required-features = ["tokio"]
-
-[[example]]
-name = "type-text"
-required-features = ["calloop"]
-
-[[example]]
-name = "reis-demo-server"
-required-features = ["calloop"]
+autoexamples = false
 
 [dependencies.calloop]
 version = "0.13.0"
