--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,29 +33,6 @@
 name = "reis"
 path = "src/lib.rs"
 
-[[example]]
-name = "list-devices"
-path = "examples/list-devices.rs"
-required-features = ["tokio"]
-
-[[example]]
-name = "receive"
-path = "examples/receive.rs"
-required-features = ["tokio"]
-
-[[example]]
-name = "receive-synchronous"
-path = "examples/receive-synchronous.rs"
-
-[[example]]
-name = "reis-demo-server"
-path = "examples/reis-demo-server.rs"
-required-features = ["calloop"]
-
-[[example]]
-name = "type-text"
-path = "examples/type-text.rs"
-required-features = ["calloop"]
 
 [dependencies.calloop]
 version = "0.13.0"
