Index: starship-battery/Cargo.toml
===================================================================
--- starship-battery.orig/Cargo.toml
+++ starship-battery/Cargo.toml
@@ -64,11 +64,11 @@ version = "0.2.19"
 default-features = false
 
 [dependencies.schemars]
-version = "1.0.4"
+version = ">= 0.8.21, < 2"
 optional = true
 
 [dependencies.serde]
-version = "1.0.225"
+version = "1.0.219"
 features = ["derive"]
 optional = true
 
