--- a/Cargo.toml
+++ b/Cargo.toml
@@ -119,7 +119,7 @@
 source-fontconfig-dlopen = ["yeslogic-fontconfig-sys/dlopen"]
 
 [target.'cfg(not(any(target_arch = "wasm32", target_family = "windows", target_os = "android", target_env = "ohos")))'.dependencies.dirs]
-version = "6.0"
+version = "5.0"
 
 [target.'cfg(not(any(target_family = "windows", target_os = "macos", target_os = "ios")))'.dependencies.freetype-sys]
 version = "0.20"
