--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,3 +48,3 @@
 [dependencies.mbox]
-version = "0.7"
+version = "0.6"
 
@@ -60,10 +60,10 @@
 version = "0.2.1"
 
 [dependencies.picky-asn1]
-version = "0.8.0"
+version = "0.7.0"
 
 [dependencies.picky-asn1-x509]
-version = "0.12.0"
+version = "0.10.0"
 
 [dependencies.regex]
 version = "1.3.9"
@@ -80,7 +80,7 @@
 features = ["zeroize_derive"]
 
 [dev-dependencies.env_logger]
-version = "0.10.0"
+version = "0.11"
 
 [dev-dependencies.sha2]
 version = "0.10.1"
