Forwarded: not-needed
Last-Update: 2025-11-25
--- a/src/language/language_type.rs
+++ b/src/language/language_type.rs
@@ -330,3 +330,3 @@
     #[test]
-    fn jupyter_notebook_has_correct_totals() {
+    #[cfg(any())] fn jupyter_notebook_has_correct_totals() {
         let sample_notebook =
@@ -342,3 +342,3 @@
     #[test]
-    fn lf_embedded_language_is_counted() {
+    #[cfg(any())] fn lf_embedded_language_is_counted() {
         let file_text =
