Index: tokei/Cargo.toml
===================================================================
--- tokei.orig/Cargo.toml
+++ tokei/Cargo.toml
@@ -37,17 +37,17 @@ version = "2.33.3"
 version = "0.5.0"
 
 [dependencies.dashmap]
-version = "4.0.1"
+version = ">= 4.0.1"
 features = ["serde"]
 
 [dependencies.dirs]
-version = "3.0.1"
+version = "5"
 
 [dependencies.encoding_rs_io]
 version = "0.1.7"
 
 [dependencies.env_logger]
-version = "0.8.2"
+version = ">= 0.8.2, < 1.0"
 features = []
 
 [dependencies.grep-searcher]
@@ -70,7 +70,7 @@ version = "0.4.0"
 version = "1.5.2"
 
 [dependencies.parking_lot]
-version = "0.11.1"
+version = "0.12"
 
 [dependencies.rayon]
 version = "1.5.0"
@@ -90,7 +90,7 @@ optional = true
 version = "1.0.61"
 
 [dependencies.serde_yaml]
-version = "0.8.15"
+version = "0.9"
 optional = true
 
 [dependencies.term_size]
@@ -99,7 +99,7 @@ version = "0.3.2"
 [dependencies.toml]
 version = "0.5.8"
 [dev-dependencies.git2]
-version = "0.13.15"
+version = ">= 0.19, < 0.21"
 features = []
 default-features = false
 
