--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,3 +39,3 @@
 [dependencies.dashmap]
-version = "4.0.1"
+version = "5"
 features = ["serde"]
@@ -43,3 +43,3 @@
 [dependencies.dirs]
-version = "3.0.1"
+version = "5"
 
@@ -49,3 +49,3 @@
 [dependencies.env_logger]
-version = "0.8.2"
+version = "0.10"
 features = []
@@ -72,3 +72,3 @@
 [dependencies.parking_lot]
-version = "0.11.1"
+version = "0.12"
 
@@ -92,3 +92,3 @@
 [dependencies.serde_yaml]
-version = "0.8.15"
+version = "0.9"
 optional = true
@@ -101,3 +101,3 @@
 [dev-dependencies.git2]
-version = "0.13.15"
+version = "0.18"
 features = []
