--- a/Cargo.toml
+++ b/Cargo.toml
@@ -168,4 +168,4 @@
 
-[dependencies.term_size]
-version = "0.3.2"
+[dependencies.terminal_size]
+version = ">= 0.3, < 1.0"
 
--- a/src/main.rs
+++ b/src/main.rs
@@ -53,3 +53,3 @@
             if cli.files {
-                term_size::dimensions().map(|(w, _)| w)
+                terminal_size::terminal_size().map(|(terminal_size::Width(w), _)| w as usize)
             } else {
