Index: addr2line/Cargo.toml
===================================================================
--- addr2line.orig/Cargo.toml
+++ addr2line/Cargo.toml
@@ -60,7 +60,7 @@ features = ["read"]
 default-features = false
 
 [dependencies.memmap2]
-version = "0.5.5"
+version = "0.9"
 optional = true
 
 [dependencies.object]
@@ -87,7 +87,7 @@ version = "3.1.6"
 version = "0.10"
 
 [dev-dependencies.libtest-mimic]
-version = "0.5.2"
+version = "0.6"
 
 [dev-dependencies.typed-arena]
 version = "2"
