--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,2 +34,3 @@
 repository = "https://github.com/gimli-rs/addr2line"
+autotests = false
 
@@ -47,11 +48,2 @@
 [[test]]
-name = "output_equivalence"
-harness = false
-required-features = ["default"]
-
-[[test]]
-name = "correctness"
-required-features = ["default"]
-
-[[test]]
 name = "parse"
@@ -95,4 +87,4 @@
 
-[dev-dependencies.backtrace]
-version = "0.3.13"
+#[dev-dependencies.backtrace]
+#version = "0.3.13"
 
@@ -105,4 +97,4 @@
 
-[dev-dependencies.libtest-mimic]
-version = "0.6.1"
+#[dev-dependencies.libtest-mimic]
+#version = "0.6.1"
 
