Index: addr2line/Cargo.toml
===================================================================
--- addr2line.orig/Cargo.toml
+++ addr2line/Cargo.toml
@@ -56,19 +56,6 @@ required-features = ["default"]
 name = "parse"
 required-features = ["std-object"]
 
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
-[dependencies.compiler_builtins]
-version = "0.1.2"
-optional = true
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
 
 [dependencies.cpp_demangle]
 version = "0.4"
@@ -129,12 +116,6 @@ default = [
     "smallvec",
     "memmap2",
 ]
-rustc-dep-of-std = [
-    "core",
-    "alloc",
-    "compiler_builtins",
-    "gimli/rustc-dep-of-std",
-]
 std = ["gimli/std"]
 std-object = [
     "std",
