Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -360,7 +360,7 @@ package = "brotli"
 version = "0.3"
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.env_logger]
-version = "0.9"
+version = "0.10"
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.hyper]
 version = "0.14"
