Index: reqwest/tests/timeouts.rs
===================================================================
--- reqwest.orig/tests/timeouts.rs
+++ reqwest/tests/timeouts.rs
@@ -89,7 +89,7 @@ async fn connect_timeout() {
     assert!(err.is_connect() && err.is_timeout());
 }
 
-#[tokio::test]
+/*#[tokio::test]
 #[cfg(not(any(target_arch = "wasm32",target_arch = "s390x")))]
 async fn connect_many_timeout_succeeds() {
     let _ = env_logger::try_init();
@@ -115,9 +115,9 @@ async fn connect_many_timeout_succeeds()
         .send()
         .await
         .unwrap();
-}
+}*/
 
-#[cfg(not(target_arch = "wasm32"))]
+/*#[cfg(not(target_arch = "wasm32"))]
 #[tokio::test]
 async fn connect_many_timeout() {
     let _ = env_logger::try_init();
@@ -146,7 +146,7 @@ async fn connect_many_timeout() {
     let err = res.unwrap_err();
 
     assert!(err.is_connect() && err.is_timeout());
-}
+}*/
 
 #[cfg(feature = "stream")]
 #[tokio::test]
