--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -163,5 +162,0 @@
-brotli = [
-    "async-compression",
-    "async-compression/brotli",
-    "tokio-util",
-]
@@ -179,10 +173,0 @@
-deflate = [
-    "async-compression",
-    "async-compression/zlib",
-    "tokio-util",
-]
-gzip = [
-    "async-compression",
-    "async-compression/gzip",
-    "tokio-util",
-]
@@ -209,6 +193,0 @@
-
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.async-compression]
-version = "0.4.0"
-features = ["tokio"]
-optional = true
-default-features = false
