--- a/Cargo.toml
+++ b/Cargo.toml
@@ -186,4 +186,0 @@
-[[example]]
-name = "blocking"
-path = "examples/blocking.rs"
-required-features = ["blocking"]
@@ -200,3 +200,0 @@
-[[example]]
-name = "form"
-path = "examples/form.rs"
@@ -210,3 +210,0 @@
-[[example]]
-name = "h3_simple"
-path = "examples/h3_simple.rs"
@@ -220,4 +220,0 @@
-[[example]]
-name = "json_dynamic"
-path = "examples/json_dynamic.rs"
-required-features = ["json"]
@@ -230,4 +230,0 @@
-[[example]]
-name = "json_typed"
-path = "examples/json_typed.rs"
-required-features = ["json"]
@@ -240,3 +240,0 @@
-[[example]]
-name = "simple"
-path = "examples/simple.rs"
@@ -250,4 +250,0 @@
-[[example]]
-name = "tor_socks"
-path = "examples/tor_socks.rs"
-required-features = ["socks"]
@@ -260,3 +260,0 @@
-[[test]]
-name = "badssl"
-path = "tests/badssl.rs"
