Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -164,7 +164,6 @@ native-tls-alpn = [
     "native-tls",
     "native-tls-crate/alpn",
 ]
-socks = ["tokio-socks"]
 stream = [
     "tokio/fs",
     "tokio-util",
@@ -263,10 +262,6 @@ default-features = false
 version = "0.3.0"
 optional = true
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.tokio-socks]
-version = "0.5.1"
-optional = true
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.tokio-util]
 version = "0.7.1"
 features = [
