Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -139,13 +139,6 @@ version = "2.2"
 
 [features]
 __internal_proxy_sys_no_cache = []
-__rustls = [
-    "hyper-rustls",
-    "tokio-rustls",
-    "rustls",
-    "__tls",
-    "rustls-pemfile",
-]
 __tls = []
 blocking = [
     "futures-util/io",
@@ -171,12 +164,6 @@ native-tls-alpn = [
     "native-tls",
     "native-tls-crate/alpn",
 ]
-rustls-tls = ["rustls-tls-native-roots"]
-rustls-tls-manual-roots = ["__rustls"]
-rustls-tls-native-roots = [
-    "rustls-native-certs",
-    "__rustls",
-]
 socks = ["tokio-socks"]
 stream = [
     "tokio/fs",
@@ -221,11 +208,6 @@ features = [
 ]
 default-features = false
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.hyper-rustls]
-version = "0.23"
-optional = true
-default-features = false
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.hyper-tls]
 version = "0.5"
 optional = true
