diff --git a/Cargo.toml b/Cargo.toml
index c6fb486..db93d6b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -163,11 +163,6 @@ blocking = [
     "tokio/rt-multi-thread",
     "tokio/sync",
 ]
-brotli = [
-    "async-compression",
-    "async-compression/brotli",
-    "tokio-util",
-]
 cookies = [
     "cookie_crate",
     "cookie_store",
@@ -179,16 +174,6 @@ default-tls = [
     "__tls",
     "tokio-native-tls",
 ]
-deflate = [
-    "async-compression",
-    "async-compression/zlib",
-    "tokio-util",
-]
-gzip = [
-    "async-compression",
-    "async-compression/gzip",
-    "tokio-util",
-]
 http3 = [
     "rustls-tls",
     "h3",
@@ -217,12 +202,6 @@ stream = [
 ]
 trust-dns = ["trust-dns-resolver"]
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.async-compression]
-version = "0.4.0"
-features = ["tokio"]
-optional = true
-default-features = false
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.cookie_crate]
 version = "0.16"
 optional = true
