Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -174,13 +174,6 @@ default-tls = [
     "__tls",
     "tokio-native-tls",
 ]
-http3 = [
-    "rustls-tls",
-    "h3",
-    "h3-quinn",
-    "quinn",
-    "futures-channel",
-]
 json = ["serde_json"]
 multipart = ["mime_guess"]
 native-tls = ["default-tls"]
@@ -218,14 +211,6 @@ optional = true
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.h2]
 version = "0.3.10"
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.h3]
-version = "0.0.2"
-optional = true
-
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.h3-quinn]
-version = "0.0.3"
-optional = true
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.http-body]
 version = "0.4.0"
 
