--- a/Cargo.toml
+++ b/Cargo.toml
@@ -300 +300 @@
-rustls-tls = ["rustls-tls-webpki-roots"]
+rustls-tls = ["rustls-tls-native-roots"] # Debian: was rustls-tls-webpki-roots
@@ -316,9 +316,9 @@
-rustls-tls-webpki-roots = [
-    "rustls-tls-webpki-roots-no-provider",
-    "__rustls-ring",
-]
-rustls-tls-webpki-roots-no-provider = [
-    "dep:webpki-roots",
-    "hyper-rustls?/webpki-tokio",
-    "__rustls",
-]
+#rustls-tls-webpki-roots = [
+#    "rustls-tls-webpki-roots-no-provider",
+#    "__rustls-ring",
+#]
+#rustls-tls-webpki-roots-no-provider = [
+#    "dep:webpki-roots",
+#    "hyper-rustls?/webpki-tokio",
+#    "__rustls",
+#]
@@ -502,3 +502,3 @@
-[target.'cfg(not(target_arch = "wasm32"))'.dependencies.webpki-roots]
-version = "0.26.0"
-optional = true
+#[target.'cfg(not(target_arch = "wasm32"))'.dependencies.webpki-roots]
+#version = "0.26.0"
+#optional = true
