diff --git a/Cargo.toml b/Cargo.toml
index c6fb486..7e8344a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -209,7 +209,6 @@ rustls-tls-native-roots = [
     "rustls-native-certs",
     "__rustls",
 ]
-socks = ["tokio-socks"]
 stream = [
     "tokio/fs",
     "tokio-util",
@@ -334,10 +333,6 @@ optional = true
 version = "0.24"
 optional = true
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.tokio-socks]
-version = "0.5.1"
-optional = true
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.tokio-util]
 version = "0.7.1"
 features = [
-- 
2.42.0

