Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -261,7 +261,7 @@ version = "0.6"
 optional = true
 
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.rustls-pemfile]
-version = "1.0"
+version = "1.0.2"
 optional = true
 
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.tokio]
@@ -293,7 +293,7 @@ default-features = false
 version = "0.3"
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.env_logger]
-version = "0.8"
+version = "0.9"
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.hyper]
 version = "0.14"
@@ -309,7 +309,7 @@ features = [
 default-features = false
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.libflate]
-version = "1.0"
+version = "0.1"
 
 [target."cfg(not(target_arch = \"wasm32\"))".dev-dependencies.serde]
 version = "1.0"
