Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -373,7 +373,7 @@
 package = "cookie"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies.cookie_store]
-version = "0.21.0"
+version = "0.22"
 optional = true
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies.encoding_rs]
@@ -511,7 +511,7 @@ package = "brotli"
 version = "0.3"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.futures-util]
 version = "0.3.28"
