Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -213,7 +213,6 @@ socks = ["tokio-socks"]
 stream = [
     "tokio/fs",
     "tokio-util",
-    "wasm-streams",
 ]
 trust-dns = ["trust-dns-resolver"]
 
@@ -390,2 +390,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dependencies.js-sys]
-version = "0.3.45"
@@ -395,2 +395,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dependencies.serde_json]
-version = "1.0"
@@ -400,2 +400,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dependencies.wasm-bindgen]
-version = "0.2.68"
@@ -405,2 +405,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dependencies.wasm-bindgen-futures]
-version = "0.4.18"
@@ -410,3 +410,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dependencies.wasm-streams]
-version = "0.4"
-optional = true
@@ -415,19 +415,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dependencies.web-sys]
-version = "0.3.25"
-features = [
-    "AbortController",
-    "AbortSignal",
-    "Headers",
-    "Request",
-    "RequestInit",
-    "RequestMode",
-    "Response",
-    "Window",
-    "FormData",
-    "Blob",
-    "BlobPropertyBag",
-    "ServiceWorkerGlobalScope",
-    "RequestCredentials",
-    "File",
-    "ReadableStream",
-]
@@ -430,3 +430,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen]
-version = "0.2.68"
-features = ["serde-serialize"]
@@ -435,2 +435,0 @@ features = [
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3"
