--- a/Cargo.toml
+++ b/Cargo.toml
@@ -173,7 +173,7 @@
 stream = [
     "tokio/fs",
     "dep:tokio-util",
-    "dep:wasm-streams",
+    # "dep:wasm-streams",
 ]
 trust-dns = []
 zstd = [
@@ -289,9 +289,9 @@
 name = "upgrade"
 path = "tests/upgrade.rs"
 
-[[test]]
-name = "wasm_simple"
-path = "tests/wasm_simple.rs"
+#[[test]]
+#name = "wasm_simple"
+#path = "tests/wasm_simple.rs"
 
 [[test]]
 name = "zstd"
@@ -593,2 +593,2 @@
-[target.'cfg(target_arch = "wasm32")'.dependencies.js-sys]
-version = "0.3.77"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.js-sys]
+#version = "0.3.77"
@@ -600,2 +600,2 @@
-[target.'cfg(target_arch = "wasm32")'.dependencies.serde_json]
-version = "1.0"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.serde_json]
+#version = "1.0"
@@ -610,2 +610,2 @@
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen]
-version = "0.2.89"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen]
+#version = "0.2.89"
@@ -620,2 +620,2 @@
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen-futures]
-version = "0.4.18"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen-futures]
+#version = "0.4.18"
@@ -630,3 +630,3 @@
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-streams]
-version = "0.4"
-optional = true
+#[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-streams]
+#version = "0.4"
+#optional = true
@@ -640,19 +640,0 @@
-[target.'cfg(target_arch = "wasm32")'.dependencies.web-sys]
-version = "0.3.28"
-features = [
-    "AbortController",
-    "AbortSignal",
-    "Headers",
-    "Request",
-    "RequestInit",
-    "RequestMode",
-    "Response",
-    "Window",
-    "FormData",
-    "Blob",
-    "BlobPropertyBag",
-    "ServiceWorkerGlobalScope",
-    "RequestCredentials",
-    "File",
-    "ReadableStream",
-]
@@ -700,3 +700,0 @@
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen]
-version = "0.2.89"
-features = ["serde-serialize"]
@@ -710,2 +710,0 @@
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+#[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen-test]
+#version = "0.3"
