--- a/Cargo.toml
+++ b/Cargo.toml
@@ -173,7 +173,7 @@
 stream = [
     "tokio/fs",
     "dep:tokio-util",
-    "dep:wasm-streams",
+    # "dep:wasm-streams",
 ]
 trust-dns = []
 zstd = [
@@ -289,9 +289,9 @@
 name = "upgrade"
 path = "tests/upgrade.rs"
 
-[[test]]
-name = "wasm_simple"
-path = "tests/wasm_simple.rs"
+#[[test]]
+#name = "wasm_simple"
+#path = "tests/wasm_simple.rs"
 
 [[test]]
 name = "zstd"
@@ -593,48 +593,48 @@
 version = "0.13"
 package = "zstd"
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.js-sys]
-version = "0.3.45"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.js-sys]
+#version = "0.3.45"
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.serde_json]
-version = "1.0"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.serde_json]
+#version = "1.0"
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen]
-version = "0.2.89"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen]
+#version = "0.2.89"
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen-futures]
-version = "0.4.18"
+#[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen-futures]
+#version = "0.4.18"
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-streams]
-version = "0.4"
-optional = true
+#[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-streams]
+#version = "0.4"
+#optional = true
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.web-sys]
-version = "0.3.28"
-features = [
-    "AbortController",
-    "AbortSignal",
-    "Headers",
-    "Request",
-    "RequestInit",
-    "RequestMode",
-    "Response",
-    "Window",
-    "FormData",
-    "Blob",
-    "BlobPropertyBag",
-    "ServiceWorkerGlobalScope",
-    "RequestCredentials",
-    "File",
-    "ReadableStream",
-]
-
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen]
-version = "0.2.89"
-features = ["serde-serialize"]
+#[target.'cfg(target_arch = "wasm32")'.dependencies.web-sys]
+#version = "0.3.28"
+#features = [
+#    "AbortController",
+#    "AbortSignal",
+#    "Headers",
+#    "Request",
+#    "RequestInit",
+#    "RequestMode",
+#    "Response",
+#    "Window",
+#    "FormData",
+#    "Blob",
+#    "BlobPropertyBag",
+#    "ServiceWorkerGlobalScope",
+#    "RequestCredentials",
+#    "File",
+#    "ReadableStream",
+#]
+
+#[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen]
+#version = "0.2.89"
+#features = ["serde-serialize"]
 
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+#[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen-test]
+#version = "0.3"
 
 [target.'cfg(target_os = "macos")'.dependencies.system-configuration]
 version = "0.6.0"
