Index: reqwest/Cargo.toml
===================================================================
--- reqwest.orig/Cargo.toml
+++ reqwest/Cargo.toml
@@ -471,7 +471,7 @@ optional = true
 default-features = false
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies.rustls-native-certs]
-version = "0.7.0"
+version = "0.6.0"
 optional = true
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies.rustls-pemfile]
Index: reqwest/src/async_impl/client.rs
===================================================================
--- reqwest.orig/src/async_impl/client.rs
+++ reqwest/src/async_impl/client.rs
@@ -538,7 +538,7 @@ impl ClientBuilder {
                             // Continue on parsing errors, as native stores often include ancient or syntactically
                             // invalid certificates, like root certificates without any X509 extensions.
                             // Inspiration: https://github.com/rustls/rustls/blob/633bf4ba9d9521a95f68766d04c22e2b01e68318/rustls/src/anchors.rs#L105-L112
-                            match root_cert_store.add(cert.into()) {
+                            match root_cert_store.add(rustls_pki_types::CertificateDer::from(cert.0)) {
                                 Ok(_) => valid_count += 1,
                                 Err(err) => {
                                     invalid_count += 1;
