Index: reqwest/tests/timeouts.rs
===================================================================
--- reqwest.orig/tests/timeouts.rs
+++ reqwest/tests/timeouts.rs
@@ -360,7 +360,12 @@ fn connect_timeout_blocking_request() {
     // never returns
     let url = "http://192.0.2.1:81/slow";
 
-    let err = client.get(url).send().unwrap_err();
+    let result = client.get(url).send();
+    if let Ok(ref result) = result {
+        // some ubuntu test runners return a 403 response here.
+        if result.status() == 403 { return }
+    }
+    let err = result.unwrap_err();
 
     assert!(err.is_timeout());
 }
