--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,3 +43,3 @@
 [dependencies.rand]
-version = "0.9.0"
+version = "0.8.5"
 optional = true
--- a/src/fake.rs
+++ b/src/fake.rs
@@ -1,6 +1,5 @@
 use crate::{PingCreationError, PingOptions, PingResult, Pinger};
 use rand::prelude::*;
-use rand::rng;
 use std::sync::mpsc;
 use std::sync::mpsc::Receiver;
 use std::thread;
@@ -31,9 +30,9 @@ impl Pinger for FakePinger {
         let sleep_time = self.options.interval;
 
         thread::spawn(move || {
-            let mut random = rng();
+            let mut random = thread_rng();
             loop {
-                let fake_seconds = random.random_range(50..150);
+                let fake_seconds = random.gen_range(50..150);
                 let ping_result = PingResult::Pong(
                     Duration::from_millis(fake_seconds),
                     format!("Fake ping line: {fake_seconds} ms"),
