--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,16 +25,10 @@ version = "1.0.72"
 version = "1.4.0"
 
 [dependencies.regex]
-version = "1.9.0"
+version = "1.7"
 
 [dependencies.thiserror]
-version = "1.0.44"
+version = "1.0"
 
 [dev-dependencies.os_info]
-version = "3.6.0"
-
-[target."cfg(windows)".dependencies.dns-lookup]
-version = "2.0.0"
-
-[target."cfg(windows)".dependencies.winping]
-version = "0.10.1"
+version = "3.5"
