Index: samply/Cargo.toml
===================================================================
--- samply.orig/Cargo.toml
+++ samply/Cargo.toml
@@ -15,7 +15,7 @@ rust-version = "1.74"
 name = "samply"
 version = "0.12.0"
 authors = ["Markus Stange <mstange@themasta.com>"]
-description = "A command line profiler for macOS and Linux."
+description = "A command line profiler for Linux."
 readme = "README.md"
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/mstange/samply/"
@@ -126,7 +126,7 @@ version = "1.37.0"
 features = [
     "rt",
     "rt-multi-thread",
-    "macros",
+#    "macros",
 ]
 
 [dependencies.tokio-util]
@@ -164,17 +164,17 @@ version = "0.2.71"
 [target."cfg(any(target_os = \"android\", target_os = \"macos\", target_os = \"linux\"))".dependencies.signal-hook]
 version = "0.3.9"
 
-[target."cfg(target_os = \"macos\")".dependencies.flate2]
-version = "1.0.23"
+# [target."cfg(target_os = \"macos\")".dependencies.flate2]
+# version = "1.0.23"
 
-[target."cfg(target_os = \"macos\")".dependencies.lazy_static]
-version = "1.4.0"
+# [target."cfg(target_os = \"macos\")".dependencies.lazy_static]
+# version = "1.4.0"
 
-[target."cfg(target_os = \"macos\")".dependencies.mach]
-version = "0.3.2"
+# [target."cfg(target_os = \"macos\")".dependencies.mach]
+# version = "0.3.2"
 
-[target."cfg(target_os = \"macos\")".dependencies.sysctl]
-version = "0.5.4"
+# [target."cfg(target_os = \"macos\")".dependencies.sysctl]
+# version = "0.5.4"
 
-[target."cfg(target_os = \"macos\")".dependencies.tempfile]
-version = "3.10.1"
+# [target."cfg(target_os = \"macos\")".dependencies.tempfile]
+# version = "3.10.1"
