Index: fs-extra/tests/file.rs
===================================================================
--- fs-extra.orig/tests/file.rs
+++ fs-extra/tests/file.rs
@@ -120,7 +120,7 @@ fn it_copy_work() {
 #[test]
 fn it_copy_not_file() {
     let mut test_file = PathBuf::from(TEST_FOLDER);
-    test_file.push("it_copy_work");
+    test_file.push("it_copy_not_file");
     let mut test_file_out = test_file.clone();
     test_file.push("test.txt");
     test_file_out.push("out");
@@ -595,7 +595,7 @@ fn it_move_work() {
 #[test]
 fn it_move_not_file() {
     let mut test_file = PathBuf::from(TEST_FOLDER);
-    test_file.push("it_move_work");
+    test_file.push("it_move_not_file");
     let mut test_file_out = test_file.clone();
     test_file.push("test.txt");
     test_file_out.push("out");
