--- rust-fs-extra-1.2.0.orig/tests/dir.rs
+++ rust-fs-extra-1.2.0/tests/dir.rs
@@ -60,7 +60,7 @@ where
     true
 }
 
-const TEST_FOLDER: &'static str = "./tests/temp/dir";
+const TEST_FOLDER: &'static str = env!("CARGO_TARGET_TMPDIR");
 
 #[test]
 fn it_create_all_work() {
--- rust-fs-extra-1.2.0.orig/tests/file.rs
+++ rust-fs-extra-1.2.0/tests/file.rs
@@ -7,7 +7,7 @@ extern crate fs_extra;
 use fs_extra::file::*;
 use fs_extra::error::*;
 
-const TEST_FOLDER: &'static str = "./tests/temp/file";
+const TEST_FOLDER: &'static str = env!("CARGO_TARGET_TMPDIR");
 
 fn files_eq<P, Q>(file1: P, file2: Q) -> Result<bool>
 where
--- rust-fs-extra-1.2.0.orig/tests/lib.rs
+++ rust-fs-extra-1.2.0/tests/lib.rs
@@ -59,7 +59,7 @@ where
     true
 }
 
-const TEST_FOLDER: &'static str = "./tests/temp/lib";
+const TEST_FOLDER: &str = env!("CARGO_TARGET_TMPDIR");
 
 #[test]
 fn it_copy_work() {
