Index: memmap2/src/lib.rs
===================================================================
--- memmap2.orig/src/lib.rs
+++ memmap2/src/lib.rs
@@ -1960,18 +1960,20 @@ mod test {
     #[test]
     #[cfg(target_os = "linux")]
     fn advise_writes_unsafely_to_part_of_map() {
-        let mut mmap = MmapMut::map_anon(8192).unwrap();
+        use libc::{sysconf, _SC_PAGESIZE};
+        let pagesize = unsafe { sysconf(_SC_PAGESIZE) as usize };
+        let mut mmap = MmapMut::map_anon(pagesize * 2).unwrap();
         mmap.as_mut().fill(255);
         let mmap = mmap.make_read_only().unwrap();
 
         let a = mmap.as_ref()[0];
-        let b = mmap.as_ref()[4096];
+        let b = mmap.as_ref()[pagesize];
         unsafe {
-            mmap.unchecked_advise_range(UncheckedAdvice::DontNeed, 4096, 4096)
+            mmap.unchecked_advise_range(UncheckedAdvice::DontNeed, pagesize, pagesize)
                 .unwrap();
         }
         let c = mmap.as_ref()[0];
-        let d = mmap.as_ref()[4096];
+        let d = mmap.as_ref()[pagesize];
 
         assert_eq!(a, 255);
         assert_eq!(b, 255);
