Index: bzip2-sys/build.rs
===================================================================
--- bzip2-sys.orig/build.rs
+++ bzip2-sys/build.rs
@@ -6,6 +6,8 @@ fn main() {
     let mut cfg = cc::Build::new();
     cfg.warnings(false);
 
+    println!("dh-cargo:deb-built-using=bz2=1={}", env::var("CARGO_MANIFEST_DIR").unwrap());
+
     if env::var("TARGET").unwrap().contains("windows") {
         cfg.define("_WIN32", None);
         cfg.define("BZ_EXPORT", None);
