--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,3 +31,3 @@
 [dependencies.bstr]
-version = "1.9.1"
+version = "1.7"
 
@@ -56,3 +56,3 @@
 [dependencies.env_logger]
-version = "0.11"
+version = "0.10"
 
@@ -72,3 +72,3 @@
 [dependencies.ratatui]
-version = "0.27"
+version = ">= 0.25, < 0.29"
 
@@ -78,3 +78,3 @@
 [dependencies.reqwest]
-version = "0.12"
+version = "0.11"
 features = [
--- a/src/http.rs
+++ b/src/http.rs
@@ -8,3 +8,2 @@
 const CONNECT_TIMEOUT: Duration = Duration::from_secs(15);
-const READ_TIMEOUT: Duration = Duration::from_secs(30);
 
@@ -20,3 +19,2 @@
             .connect_timeout(CONNECT_TIMEOUT)
-            .read_timeout(READ_TIMEOUT)
             .build()
