Index: spytrap-adb/Cargo.toml
===================================================================
--- spytrap-adb.orig/Cargo.toml
+++ spytrap-adb/Cargo.toml
@@ -61,11 +61,11 @@
 version = "4.2.1"
 
 [dependencies.crossterm]
-version = "0.29"
+version = "0.27"
 features = ["event-stream"]
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
 [dependencies.env_logger]
 version = "0.11"
@@ -84,7 +84,7 @@
 version = "0.4.14"
 
 [dependencies.ratatui]
-version = "0.29"
+version = "0.28"
 
 [dependencies.regex]
 version = "1.5.4"
