--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,7 +29,7 @@
 version = "1.0.44"
 
 [dependencies.bstr]
-version = "1.9.1"
+version = "1.7"
 
 [dependencies.chrono]
 version = "0.4.19"
@@ -54,7 +54,7 @@
 version = "5.0.0"
 
 [dependencies.env_logger]
-version = "0.11"
+version = "0.10"
 
 [dependencies.forensic-adb]
 version = "0.7"
@@ -70,13 +70,13 @@
 version = "0.4.14"
 
 [dependencies.ratatui]
-version = "0.27"
+version = "0.25"
 
 [dependencies.regex]
 version = "1.5.4"
 
 [dependencies.reqwest]
-version = "0.12"
+version = "0.11"
 features = [
     "rustls-tls-native-roots",
     "brotli",
--- a/src/http.rs
+++ b/src/http.rs
@@ -6,7 +6,6 @@
 const APP_USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"),);
 
 const CONNECT_TIMEOUT: Duration = Duration::from_secs(15);
-const READ_TIMEOUT: Duration = Duration::from_secs(30);
 
 #[derive(Debug, Clone)]
 pub struct Client {
@@ -18,7 +17,6 @@
         let http = reqwest::Client::builder()
             .user_agent(APP_USER_AGENT)
             .connect_timeout(CONNECT_TIMEOUT)
-            .read_timeout(READ_TIMEOUT)
             .build()
             .context("Failed to setup http client")?;
         Ok(Self { http })
