Index: rust-coreutils-8.0~0.0.2/GNUmakefile
===================================================================
--- rust-coreutils-8.0~0.0.2.orig/GNUmakefile
+++ rust-coreutils-8.0~0.0.2/GNUmakefile
@@ -55,6 +55,7 @@ PROGS       := \
 	cp \
 	csplit \
 	cut \
+	date \
 	df \
 	dircolors \
 	dirname \
@@ -163,6 +164,7 @@ TEST_PROGS  := \
 	cp \
 	csplit \
 	cut \
+	date \
 	dircolors \
 	dirname \
 	echo \
