Index: rust-coreutils-8.0~0.0.2/GNUmakefile
===================================================================
--- rust-coreutils-8.0~0.0.2.orig/GNUmakefile
+++ rust-coreutils-8.0~0.0.2/GNUmakefile
@@ -1,5 +1,5 @@
 # Config options
-PROFILE         ?= release
+65;6201;1cPROFILE         ?= release
 MULTICALL       ?= n
 INSTALL         ?= install
 ifneq (,$(filter install, $(MAKECMDGOALS)))
@@ -55,6 +55,7 @@ PROGS       := \
 	cp \
 	csplit \
 	cut \
+	date \
 	df \
 	dircolors \
 	dirname \
@@ -163,6 +164,7 @@ TEST_PROGS  := \
 	cp \
 	csplit \
 	cut \
+	date \
 	dircolors \
 	dirname \
 	echo \
