Index: coreutils/src/uu/date/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/date/Cargo.toml
+++ coreutils/src/uu/date/Cargo.toml
@@ -26,8 +26,8 @@ humantime_to_duration = { workspace=true
 [target.'cfg(unix)'.dependencies]
 libc = { workspace=true }
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_Foundation", "Win32_System_SystemInformation"] }
+#[target.'cfg(windows)'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_Foundation", "Win32_System_SystemInformation"] }
 
 [[bin]]
 name = "date"
Index: coreutils/src/uu/du/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/du/Cargo.toml
+++ coreutils/src/uu/du/Cargo.toml
@@ -21,8 +21,8 @@ glob = { workspace=true }
 clap = { workspace=true }
 uucore = { workspace=true }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem", "Win32_Foundation"] }
+#[target.'cfg(target_os = "windows")'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem", "Win32_Foundation"] }
 
 [[bin]]
 name = "du"
Index: coreutils/src/uu/hostname/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/hostname/Cargo.toml
+++ coreutils/src/uu/hostname/Cargo.toml
@@ -19,8 +19,8 @@ clap = { workspace=true }
 hostname = { version = "0.3", features = ["set"] }
 uucore = { workspace=true, features=["wide"] }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_Networking_WinSock", "Win32_Foundation"] }
+#[target.'cfg(target_os = "windows")'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_Networking_WinSock", "Win32_Foundation"] }
 
 [[bin]]
 name = "hostname"
Index: coreutils/src/uu/rm/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/rm/Cargo.toml
+++ coreutils/src/uu/rm/Cargo.toml
@@ -22,8 +22,8 @@ uucore = { workspace=true, features=["fs
 [target.'cfg(unix)'.dependencies]
 libc = { workspace=true }
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem"] }
+#[target.'cfg(windows)'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem"] }
 
 [[bin]]
 name = "rm"
Index: coreutils/src/uu/sync/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/sync/Cargo.toml
+++ coreutils/src/uu/sync/Cargo.toml
@@ -22,8 +22,8 @@ uucore = { workspace=true, features=["wi
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies]
 nix = { workspace=true }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem", "Win32_System_WindowsProgramming", "Win32_Foundation"] }
+#[target.'cfg(target_os = "windows")'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem", "Win32_System_WindowsProgramming", "Win32_Foundation"] }
 
 [[bin]]
 name = "sync"
Index: coreutils/src/uu/tail/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/tail/Cargo.toml
+++ coreutils/src/uu/tail/Cargo.toml
@@ -25,9 +25,9 @@ same-file = { workspace=true }
 is-terminal = { workspace=true }
 fundu = { workspace=true }
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_System_Threading", "Win32_Foundation"] }
-winapi-util = { workspace=true }
+#[target.'cfg(windows)'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_System_Threading", "Win32_Foundation"] }
+#winapi-util = { workspace=true }
 
 [dev-dependencies]
 rstest = { workspace=true }
Index: coreutils/src/uu/touch/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/touch/Cargo.toml
+++ coreutils/src/uu/touch/Cargo.toml
@@ -22,8 +22,8 @@ humantime_to_duration = { workspace=true
 time = { workspace=true, features = ["parsing", "formatting", "local-offset", "macros"] }
 uucore = { workspace=true, features=["libc"] }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem", "Win32_Foundation"] }
+#[target.'cfg(target_os = "windows")'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_Storage_FileSystem", "Win32_Foundation"] }
 
 [[bin]]
 name = "touch"
Index: coreutils/src/uu/whoami/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/whoami/Cargo.toml
+++ coreutils/src/uu/whoami/Cargo.toml
@@ -18,8 +18,8 @@ path = "src/whoami.rs"
 clap = { workspace=true }
 uucore = { workspace=true, features=["entries"] }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-windows-sys = { workspace=true, features = ["Win32_NetworkManagement_NetManagement", "Win32_System_WindowsProgramming", "Win32_Foundation"] }
+#[target.'cfg(target_os = "windows")'.dependencies]
+#windows-sys = { workspace=true, features = ["Win32_NetworkManagement_NetManagement", "Win32_System_WindowsProgramming", "Win32_Foundation"] }
 
 [target.'cfg(unix)'.dependencies]
 libc = { workspace=true }
Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -56,9 +56,9 @@ clap = { workspace=true }
 once_cell = { workspace=true }
 tempfile = { workspace=true }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-winapi-util = { version= "0.1.5", optional=true }
-windows-sys = { version = "0.48.0", optional = true, default-features = false, features = ["Win32_Storage_FileSystem", "Win32_Foundation", "Win32_System_WindowsProgramming"] }
+#[target.'cfg(target_os = "windows")'.dependencies]
+#winapi-util = { version= "0.1.5", optional=true }
+#windows-sys = { version = "0.48.0", optional = true, default-features = false, features = ["Win32_Storage_FileSystem", "Win32_Foundation", "Win32_System_WindowsProgramming"] }
 
 [features]
 default = []
Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -328,7 +328,7 @@ unicode-width = "0.1.10"
 utf-8 = "0.7.6"
 walkdir = "2.3"
 winapi-util = "0.1.5"
-windows-sys = { version="0.48.0", default-features=false }
+#windows-sys = { version="0.48.0", default-features=false }
 xattr = "1.0.0"
 zip = { version = "0.6.6", default_features=false, features=["deflate"] }
 
