Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -82,7 +82,7 @@ feat_common_core = [
   "env",
   "expand",
   "expr",
-  "factor",
+##  "factor",
   "false",
   "fmt",
   "fold",
@@ -312,7 +312,7 @@ nix = { version = "0.29", default-featur
 nom = "> 7"
 notify = ">= 5, < 7"
 num-bigint = "0.4"
-num-prime = "0.4.4"
+#num-prime = "0.4.4"
 num-traits = "0.2"
 number_prefix = "0.4"
 onig = { version = "~6", default-features = false }
@@ -409,7 +409,7 @@ echo = { optional = true, version = "0.0
 env = { optional = true, version = "0.0.30", package = "uu_env", path = "src/uu/env" }
 expand = { optional = true, version = "0.0.30", package = "uu_expand", path = "src/uu/expand" }
 expr = { optional = true, version = "0.0.30", package = "uu_expr", path = "src/uu/expr" }
-factor = { optional = true, version = "0.0.30", package = "uu_factor", path = "src/uu/factor" }
+#factor = { optional = true, version = "0.0.30", package = "uu_factor", path = "src/uu/factor" }
 false = { optional = true, version = "0.0.30", package = "uu_false", path = "src/uu/false" }
 fmt = { optional = true, version = "0.0.30", package = "uu_fmt", path = "src/uu/fmt" }
 fold = { optional = true, version = "0.0.30", package = "uu_fold", path = "src/uu/fold" }
@@ -497,7 +497,7 @@ chrono = { workspace = true }
 filetime = { workspace = true }
 glob = { workspace = true }
 libc = { workspace = true }
-num-prime = { workspace = true }
+#num-prime = { workspace = true }
 pretty_assertions = "1.4.0"
 rand = { workspace = true }
 regex = { workspace = true }
Index: coreutils/GNUmakefile
===================================================================
--- coreutils.orig/GNUmakefile
+++ coreutils/GNUmakefile
@@ -88,7 +88,6 @@ PROGS       := \
 	env \
 	expand \
 	expr \
-	factor \
 	false \
 	fmt \
 	fold \
@@ -205,7 +204,6 @@ TEST_PROGS  := \
 	echo \
 	env \
 	expr \
-	factor \
 	false \
 	fold \
 	hashsum \
Index: coreutils/src/uu/ls/src/ls.rs
===================================================================
--- coreutils.orig/src/uu/ls/src/ls.rs
+++ coreutils/src/uu/ls/src/ls.rs
@@ -25,7 +25,6 @@ use std::{
     time::Duration,
 };
 use std::{collections::HashSet, io::IsTerminal};
-use ansi_width::ansi_width;
 
 use ansi_width::ansi_width;
 use chrono::{DateTime, Local, TimeDelta};
