Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -343,7 +343,7 @@ time = { version = "0.3.36" }
 unicode-segmentation = "1"
 unicode-width = "0.2.0"
 utf-8 = "0.7.6"
-utmp-classic = "0.1.6"
+#utmp-classic = "0.1.6"
 uutils_term_grid = "0"
 walkdir = "2"
 #winapi-util = "0.1.8"
Index: coreutils/src/uu/uptime/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/uptime/Cargo.toml
+++ coreutils/src/uu/uptime/Cargo.toml
@@ -22,8 +22,8 @@ clap = { workspace = true }
 thiserror = { workspace = true }
 uucore = { workspace = true, features = ["libc", "utmpx", "uptime"] }
 
-[target.'cfg(target_os = "openbsd")'.dependencies]
-utmp-classic = { workspace = true }
+#[target.'cfg(target_os = "openbsd")'.dependencies]
+#utmp-classic = { workspace = true }
 
 [target.'cfg(target_os="windows")'.dependencies]
 windows-sys = { workspace = true, features = [
Index: coreutils/src/uu/users/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/users/Cargo.toml
+++ coreutils/src/uu/users/Cargo.toml
@@ -20,8 +20,8 @@ path = "src/users.rs"
 clap = { workspace = true }
 uucore = { workspace = true, features = ["utmpx"] }
 
-[target.'cfg(target_os = "openbsd")'.dependencies]
-utmp-classic = { workspace = true }
+#[target.'cfg(target_os = "openbsd")'.dependencies]
+#utmp-classic = { workspace = true }
 
 [[bin]]
 name = "users"
