Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -378,7 +378,7 @@ textwrap = { version = "0.16", features
 thiserror = "2.0.3"
 time = { version = "0.3.36" }
 unicode-width = "0.2.0"
-utmp-classic = "0.1.6"
+#utmp-classic = "0.1.6"
 uutils_term_grid = "0"
 walkdir = "2"
 #winapi-util = "0.1.8"
Index: coreutils/src/uu/uptime/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/uptime/Cargo.toml
+++ coreutils/src/uu/uptime/Cargo.toml
@@ -29,8 +29,8 @@ thiserror = { workspace = true }
 uucore = { workspace = true, features = ["libc", "utmpx", "uptime"] }
 fluent = { workspace = true }
 
-[target.'cfg(target_os = "openbsd")'.dependencies]
-utmp-classic = { workspace = true }
+#[target.'cfg(target_os = "openbsd")'.dependencies]
+#utmp-classic = { workspace = true }
 
 [[bin]]
 name = "uptime"
Index: coreutils/src/uu/users/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/users/Cargo.toml
+++ coreutils/src/uu/users/Cargo.toml
@@ -27,8 +27,8 @@ clap = { workspace = true }
 uucore = { workspace = true, features = ["utmpx"] }
 fluent = { workspace = true }
 
-[target.'cfg(target_os = "openbsd")'.dependencies]
-utmp-classic = { workspace = true }
+#[target.'cfg(target_os = "openbsd")'.dependencies]
+#utmp-classic = { workspace = true }
 
 [[bin]]
 name = "users"
