Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -327,7 +327,7 @@ unicode-segmentation = "1"
 unicode-width = "0.1.10"
 utf-8 = "0.7.6"
 walkdir = "2.3"
-winapi-util = "0.1.5"
+#winapi-util = "0.1.5"
 #windows-sys = { version="0.48.0", default-features=false }
 xattr = "1.0.0"
 zip = { version = "0.6.6", default_features=false, features=["deflate"] }
Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -65,8 +65,8 @@ default = []
 # * non-default features
 encoding = ["data-encoding", "data-encoding-macro", "z85", "thiserror"]
 entries = ["libc"]
-fs = ["libc", "winapi-util", "windows-sys"]
-fsext = ["libc", "time", "windows-sys"]
+fs = ["libc"] #, "winapi-util", "windows-sys"]
+fsext = ["libc", "time"] #, "windows-sys"]
 lines = []
 memo = ["itertools"]
 mode = ["libc"]
Index: coreutils/src/uu/tail/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/tail/Cargo.toml
+++ coreutils/src/uu/tail/Cargo.toml
@@ -29,8 +29,8 @@ fundu = { workspace=true }
 #windows-sys = { workspace=true, features = ["Win32_System_Threading", "Win32_Foundation"] }
 #winapi-util = { workspace=true }
 
-[dev-dependencies]
-rstest = { workspace=true }
+#[dev-dependencies]
+#rstest = { workspace=true }
 
 [[bin]]
 name = "tail"
