Index: coreutils/src/uu/dd/src/dd.rs
===================================================================
--- coreutils.orig/src/uu/dd/src/dd.rs
+++ coreutils/src/uu/dd/src/dd.rs
@@ -785,7 +785,7 @@ impl<'a> Output<'a> {
         if let Some(libc_flags) = make_linux_oflags(&settings.oflags) {
             nix::fcntl::fcntl(
                 fx.as_raw().as_raw_fd(),
-                F_SETFL(OFlag::from_bits_retain(libc_flags)),
+                F_SETFL(OFlag::from_bits(libc_flags).unwrap()),
             )?;
         }
 
