Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -268,7 +268,7 @@ test = [ "uu_test" ]
 clap = { version = "4.0", features = ["wrap_help", "cargo"] }
 clap_complete = "4.0"
 once_cell = "1"
-phf = "0.11.1"
+phf = "0"
 selinux = { version="0.3", optional = true }
 textwrap = { version="0.16.0", features=["terminal_size"] }
 uucore = { version=">=0.0.17", package="uucore", path="src/uucore" }
@@ -416,7 +416,7 @@ rust-users = { version="0", package="use
 rand_pcg = "0.3"
 
 [build-dependencies]
-phf_codegen = "0.11.1"
+phf_codegen = "0"
 
 [[bin]]
 name = "coreutils"
