Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -334,7 +334,7 @@ conv = "0.3"
 filetime = "0.2"
 glob = "0.3.0"
 libc = "0.2"
-nix = "0.20.0"
+nix = "0.19.0"
 rand = "0.7"
 regex = "1.0"
 sha1 = { version="0.6", features=["std"] }
Index: coreutils/src/uu/cat/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/cat/Cargo.toml
+++ coreutils/src/uu/cat/Cargo.toml
@@ -24,7 +24,7 @@ uucore_procs = { version=">=0.0.5", pack
 unix_socket = "0.5.0"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies]
-nix = "0.20"
+nix = "0.19"
 
 [[bin]]
 name = "cat"
Index: coreutils/src/uu/wc/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/wc/Cargo.toml
+++ coreutils/src/uu/wc/Cargo.toml
@@ -21,7 +21,7 @@ uucore_procs = { version=">=0.0.5", pack
 thiserror = "1.0"
 
 [target.'cfg(unix)'.dependencies]
-nix = "0.20"
+nix = "0.19"
 libc = "0.2"
 
 [[bin]]
