Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -315,7 +315,7 @@ num-bigint = "0.4.4"
 num-prime = "0.4.4"
 num-traits = "0.2.19"
 number_prefix = "0.4"
-onig = { version = "~6.4", default-features = false }
+onig = { version = "~6", default-features = false }
 parse_datetime = "0.8.0"
 phf = "0.11.2"
 phf_codegen = "0.11.2"
