Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -402,7 +402,7 @@ package = "uu_pathchk"
 path = "src/uu/pathchk"
 
 [dependencies.phf]
-version = "0.11.1"
+version = "0"
 
 [dependencies.pinky]
 version = "0.0.17"
@@ -757,7 +757,7 @@ package = "uucore"
 version = "2.2"
 
 [build-dependencies.phf_codegen]
-version = "0.11.1"
+version = "0"
 
 [features]
 default = ["feat_common_core"]
