commit d9c2acc2ed4c1273dbe77b76d79b3e0eb5c8c393
Author: alextibbles <45136886+alextibbles@users.noreply.github.com>
Date:   Sat Feb 12 12:12:02 2022 -0500

    update to sha 0.10.0 (#3110)
    
    * update to sha 0.10.0
    
    * correct formatting

diff --git b/Cargo.lock a/Cargo.lock
index 90b71d2a7..ef1cd2054 100644
--- b/Cargo.lock
+++ a/Cargo.lock
@@ -312,7 +312,6 @@ dependencies = [
  "conv",
  "filetime",
  "glob",
- "hex-literal",
  "lazy_static",
  "libc",
  "nix 0.23.1",
@@ -904,12 +903,6 @@ version = "0.2.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "d6a22814455d41612f41161581c2883c0c6a1c41852729b17d5ed88f01e153aa"
 
-[[package]]
-name = "hex-literal"
-version = "0.3.4"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "7ebdb29d2ea9ed0083cd8cece49bbd968021bd99b0849edb4a9a7ee0fdf6a4e0"
-
 [[package]]
 name = "hostname"
 version = "0.3.1"
@@ -1744,15 +1737,19 @@ dependencies = [
 
 [[package]]
 name = "sha1"
-version = "0.10.0"
+version = "0.6.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "04cc229fb94bcb689ffc39bd4ded842f6ff76885efede7c6d1ffb62582878bea"
+checksum = "c1da05c97445caa12d05e848c4a4fcbbea29e748ac28f7e80e9b010392063770"
 dependencies = [
- "cfg-if 1.0.0",
- "cpufeatures",
- "digest",
+ "sha1_smol",
 ]
 
+[[package]]
+name = "sha1_smol"
+version = "1.0.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "ae1a47186c03a32177042e55dbc5fd5aee900b8e0069a8d70fba96a9375cd012"
+
 [[package]]
 name = "sha2"
 version = "0.10.1"
diff --git b/Cargo.toml a/Cargo.toml
index 336729813..e9fbe42fb 100644
--- b/Cargo.toml
+++ a/Cargo.toml
@@ -372,14 +372,13 @@ libc = "0.2"
 pretty_assertions = "1"
 rand = "0.8"
 regex = "1.0"
-sha1 = { version="0.10", features=["std"] }
+sha1 = { version="0.6", features=["std"] }
 tempfile = "3.2.0"
 time = "0.1"
 unindent = "0.1"
 uucore = { version=">=0.0.11", package="uucore", path="src/uucore", features=["entries", "process"] }
 walkdir = "2.2"
 atty = "0.2"
-hex-literal = "0.3.1"
 
 [target.'cfg(target_os = "linux")'.dev-dependencies]
 rlimit = "0.4.0"
diff --git b/src/uu/hashsum/Cargo.toml a/src/uu/hashsum/Cargo.toml
index d3170689a..495e15972 100644
--- b/src/uu/hashsum/Cargo.toml
+++ a/src/uu/hashsum/Cargo.toml
@@ -23,7 +23,7 @@ memchr = "2"
 md5 = "0.3.5"
 regex = "1.0.1"
 regex-syntax = "0.6.7"
-sha1 = "0.10.0"
+sha1 = "0.6.0"
 sha2 = "0.10.1"
 sha3 = "0.10.0"
 blake2b_simd = "0.5.11"
diff --git b/src/uu/hashsum/src/digest.rs a/src/uu/hashsum/src/digest.rs
index 678c44886..c06834c74 100644
--- b/src/uu/hashsum/src/digest.rs
+++ a/src/uu/hashsum/src/digest.rs
@@ -106,19 +106,19 @@ impl Digest for blake3::Hasher {
 
 impl Digest for sha1::Sha1 {
     fn new() -> Self {
-        Self::default()
+        Self::new()
     }
 
     fn input(&mut self, input: &[u8]) {
-        digest::Digest::update(self, input);
+        self.update(input);
     }
 
     fn result(&mut self, out: &mut [u8]) {
-        digest::Digest::finalize_into_reset(self, out.into());
+        out.copy_from_slice(&self.digest().bytes());
     }
 
     fn reset(&mut self) {
-        *self = Self::new();
+        self.reset();
     }
 
     fn output_bits(&self) -> usize {
diff --git b/tests/by-util/test_factor.rs a/tests/by-util/test_factor.rs
index 7c1e540b6..bd265f4ce 100644
--- b/tests/by-util/test_factor.rs
+++ a/tests/by-util/test_factor.rs
@@ -29,8 +29,6 @@ const NUM_TESTS: usize = 100;
 
 #[test]
 fn test_parallel() {
-    use hex_literal::hex;
-    use sha1::{Digest, Sha1};
     // factor should only flush the buffer at line breaks
     let n_integers = 100_000;
     let mut input_string = String::new();
@@ -62,20 +60,13 @@ fn test_parallel() {
         .ccmd("sort")
         .arg(tmp_dir.plus("output"))
         .succeeds();
-    let mut hasher = Sha1::new();
-    hasher.update(result.stdout());
-    let hash_check = hasher.finalize();
-    assert_eq!(
-        hash_check[..],
-        hex!("cc743607c0ff300ff575d92f4ff0c87d5660c393")
-    );
+    let hash_check = sha1::Sha1::from(result.stdout()).hexdigest();
+    assert_eq!(hash_check, "cc743607c0ff300ff575d92f4ff0c87d5660c393");
 }
 
 #[test]
 fn test_first_100000_integers() {
     extern crate sha1;
-    use hex_literal::hex;
-    use sha1::{Digest, Sha1};
 
     let n_integers = 100_000;
     let mut input_string = String::new();
@@ -87,13 +78,8 @@ fn test_first_100000_integers() {
     let result = new_ucmd!().pipe_in(input_string.as_bytes()).succeeds();
 
     // `seq 0 100000 | factor | sha1sum` => "4ed2d8403934fa1c76fe4b84c5d4b8850299c359"
-    let mut hasher = Sha1::new();
-    hasher.update(result.stdout());
-    let hash_check = hasher.finalize();
-    assert_eq!(
-        hash_check[..],
-        hex!("4ed2d8403934fa1c76fe4b84c5d4b8850299c359")
-    );
+    let hash_check = sha1::Sha1::from(result.stdout()).hexdigest();
+    assert_eq!(hash_check, "4ed2d8403934fa1c76fe4b84c5d4b8850299c359");
 }
 
 #[test]
