Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -302,8 +302,8 @@ number_prefix = "0.4"
 once_cell = "1"
 onig = { version = "6", default-features = false }
 ouroboros = "0.15"
-phf = "0.11.1"
-phf_codegen = "0.11.1"
+phf = "0"
+phf_codegen = "0"
 platform-info = "2.0.1"
 quick-error = "2.0.1"
 rand = { version = "0.8", features = ["small_rng"] }
