Index: coreutils/src/uu/mv/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/mv/Cargo.toml
+++ coreutils/src/uu/mv/Cargo.toml
@@ -28,12 +28,12 @@ uucore = { workspace = true, features =
 ] }
 thiserror = { workspace = true }
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace = true, features = [
-  "Win32_Foundation",
-  "Win32_Security",
-  "Win32_Storage_FileSystem",
-] }
+#[target.'cfg(windows)'.dependencies]
+#windows-sys = { workspace = true, features = [
+#  "Win32_Foundation",
+#  "Win32_Security",
+#  "Win32_Storage_FileSystem",
+#] }
 
 [target.'cfg(unix)'.dependencies]
 libc = { workspace = true }
