Index: rust-coreutils-0.0.27/Cargo.toml
===================================================================
--- rust-coreutils-0.0.27.orig/Cargo.toml
+++ rust-coreutils-0.0.27/Cargo.toml
@@ -296,7 +296,7 @@ lscolors = { version = "0", default-feat
 ] }
 memchr = "2.7"
 memmap2 = "0"
-nix = { version = "0.28", default-features = false }
+nix = { version = "0.29", default-features = false }
 nom = "7"
 notify = ">= 5, < 7"
 num-bigint = "0.4"
Index: rust-coreutils-0.0.27/src/uucore/src/lib/features/pipes.rs
===================================================================
--- rust-coreutils-0.0.27.orig/src/uucore/src/lib/features/pipes.rs
+++ rust-coreutils-0.0.27/src/uucore/src/lib/features/pipes.rs
@@ -11,6 +11,7 @@ use std::os::unix::io::AsRawFd;
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use nix::fcntl::SpliceFFlags;
+use std::os::fd::BorrowedFd;
 
 pub use nix::{Error, Result};
 
@@ -35,9 +36,9 @@ pub fn pipe() -> Result<(File, File)> {
 #[cfg(any(target_os = "linux", target_os = "android"))]
 pub fn splice(source: &impl AsRawFd, target: &impl AsRawFd, len: usize) -> Result<usize> {
     nix::fcntl::splice(
-        source.as_raw_fd(),
+        unsafe { BorrowedFd::borrow_raw(source.as_raw_fd()) },
         None,
-        target.as_raw_fd(),
+        unsafe { BorrowedFd::borrow_raw(target.as_raw_fd()) },
         None,
         len,
         SpliceFFlags::empty(),
@@ -66,7 +67,7 @@ pub fn splice_exact(source: &impl AsRawF
 #[cfg(any(target_os = "linux", target_os = "android"))]
 pub fn vmsplice(target: &impl AsRawFd, bytes: &[u8]) -> Result<usize> {
     nix::fcntl::vmsplice(
-        target.as_raw_fd(),
+        unsafe { BorrowedFd::borrow_raw(target.as_raw_fd()) },
         &[IoSlice::new(bytes)],
         SpliceFFlags::empty(),
     )
