Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -499,7 +499,7 @@ rlimit = "0.9.1"
 
 [target.'cfg(unix)'.dev-dependencies]
 nix = { workspace=true, features=["process", "signal", "user"] }
-rust-users = { version="0.11", package="users" }
+rust-users = { version="0", package="users" }
 rand_pcg = "0.3"
 
 [build-dependencies]
