Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -347,7 +347,7 @@ linux-raw-sys = "0.11"
 lscolors = { version = "0", default-features = false, features = [
   "gnu_legacy",
 ] }
-memchr = "2.7.2"
+memchr = "2.7"
 memmap2 = "0"
 nix = { version = "0.30", default-features = false }
 nom = ">= 7"
