Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -312,8 +312,8 @@ chrono = { version = "0.4.41", default-f
   "alloc",
   "clock",
 ] }
-clap = { version = "4.5", features = ["wrap_help", "cargo", "color"] }
-clap_complete = "4.4"
+clap = { version = "4", features = ["wrap_help", "cargo", "color"] }
+clap_complete = "4"
 clap_mangen = "0.2"
 compare = "0.1.0"
 crossterm = ">= 0.25.0"
