Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -286,7 +286,7 @@ glob = "0.3"
 half = "1"
 indicatif = "0.17"
 is-terminal = "0.4"
-itertools = "0.11.0"
+itertools = "0"
 libc = "0.2.147"
 lscolors = { version = "0.14.0", default-features = false, features = [
   "nu-ansi-term",
Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -25,7 +25,7 @@ dunce = "1"
 wild = "2.1"
 glob = "0.3"
 # * optional
-itertools = { version = "0.11.0", optional = true }
+itertools = { version = "0", optional = true }
 thiserror = { workspace = true, optional = true }
 time = { workspace = true, optional = true, features = [
   "formatting",
