Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -279,8 +279,8 @@ chrono = { version = "0.4.38", default-f
   "clock",
 ] }
 chrono-tz = "0.10.0"
-clap = { version = "4.5", features = ["wrap_help", "cargo"] }
-clap_complete = "4.4"
+clap = { version = "4", features = ["wrap_help", "cargo"] }
+clap_complete = "4"
 clap_mangen = "0.2"
 compare = "0.1.0"
 coz = { version = "0.1.3" }
