Index: rust-coreutils-0.0.30/Cargo.toml
===================================================================
--- rust-coreutils-0.0.30.orig/Cargo.toml
+++ rust-coreutils-0.0.30/Cargo.toml
@@ -279,7 +279,7 @@ chrono = { version = "0.4", default-feat
   "alloc",
   "clock",
 ] }
-chrono-tz = "0.10.0"
+chrono-tz = ">= 0.8"
 clap = { version = "4", features = ["wrap_help", "cargo"] }
 clap_complete = "4"
 clap_mangen = "0.2"
