Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -319,7 +319,7 @@ compare = "0.1.0"
 crossterm = ">= 0.25.0"
 ctor = "0.5.0"
 ctrlc = { version = "3.4.7", features = ["termination"] }
-dns-lookup = { version = "3.0.0" }
+dns-lookup = { version = "1" }
 exacl = "0"
 file_diff = "1.0.0"
 filetime = "0.2.23"
