Index: rust-coreutils-0.0.30/Cargo.toml
===================================================================
--- rust-coreutils-0.0.30.orig/Cargo.toml
+++ rust-coreutils-0.0.30/Cargo.toml
@@ -321,8 +321,8 @@ phf = "0"
 phf_codegen = "0"
 platform-info = "2"
 quick-error = "2.0.1"
-rand = { version = "0.9.0", features = ["small_rng"] }
-rand_core = "0.9.0"
+rand = { version = ">= 0.8", features = ["small_rng"] }
+rand_core = ">= 0.6"
 rayon = "1"
 regex = "1"
 rstest = ">= 0.17.0"
