Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -335,7 +335,7 @@ selinux-sys = "= 0.6.13"
 signal-hook = "0.3.17"
 smallvec = { version = "1", features = ["union"] }
 tempfile = "3.15.0"
-terminal_size = "0.4.0"
+terminal_size = "0"
 textwrap = { version = "0.16.1", features = ["terminal_size"] }
 thiserror = "2.0.3"
 time = { version = "0.3.36" }
