Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -27,7 +27,7 @@ clap = { workspace = true }
 uucore_procs = { workspace = true }
 number_prefix = { workspace = true }
 dns-lookup = { workspace = true, optional = true }
-dunce = { version = "1.0.4", optional = true }
+dunce = { version = "1", optional = true }
 wild = "2.2.1"
 glob = { workspace = true }
 iana-time-zone = { workspace = true }
