Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -379,7 +379,7 @@ thiserror = "2.0.3"
 time = { version = "0.3.36" }
 unicode-width = "0.2.0"
 utmp-classic = "0.1.6"
-uutils_term_grid = "0.7"
+uutils_term_grid = "0"
 walkdir = "2"
 #winapi-util = "0.1.8"
 #windows-sys = { version = "0.61.0", default-features = false }
