Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -282,7 +282,7 @@ fs_extra = "1.3.0"
 fts-sys = "0.2"
 fundu = "1.2.0"
 gcd = "2.3"
-glob = "0.3.1"
+glob = "0.3"
 half = "1"
 indicatif = "0.17"
 is-terminal = "0.4.7"
Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -23,7 +23,7 @@ uucore_procs = { workspace = true }
 dns-lookup = { version = "2.0.2", optional = true }
 dunce = "1.0.4"
 wild = "2.1"
-glob = "0.3.1"
+glob = "0.3"
 # * optional
 itertools = { version = "0.11.0", optional = true }
 thiserror = { workspace = true, optional = true }
