Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -28,7 +28,7 @@ uucore_procs = { workspace = true }
 number_prefix = { workspace = true }
 dns-lookup = { workspace = true, optional = true }
 dunce = { version = "1", optional = true }
-wild = "2.2.1"
+wild = "2"
 glob = { workspace = true }
 iana-time-zone = { workspace = true }
 # * optional
