Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -370,7 +370,7 @@ rust-ini = ">= 0.18.0"
 same-file = "1.0.6"
 self_cell = "1"
 # FIXME we use the exact version because the new 0.5.3 requires an MSRV of 1.88
-selinux = "=0.5.2"
+selinux = "> 0.4.0"
 signal-hook = "0"
 tempfile = "3"
 terminal_size = "0"
