Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -322,7 +322,7 @@ selinux = "0.4.4"
 signal-hook = "0"
 smallvec = { version = "1", features = ["union"] }
 tempfile = "3"
-uutils_term_grid = "0.6"
+uutils_term_grid = "0"
 terminal_size = "0"
 textwrap = { version = "0.16", features = ["terminal_size"] }
 thiserror = "1.0.59"
