Index: coreutils/src/uu/stdbuf/src/libstdbuf/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/stdbuf/src/libstdbuf/Cargo.toml
+++ coreutils/src/uu/stdbuf/src/libstdbuf/Cargo.toml
@@ -20,8 +20,8 @@ crate-type = [
 ] # XXX: note: the rlib is just to prevent Cargo from spitting out a warning
 
 [dependencies]
-cpp = "0.5.10"
+cpp = "0.5"
 libc = { workspace = true }
 
 [build-dependencies]
-cpp_build = "0.5.10"
+cpp_build = "0.5"
