Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -520,7 +520,7 @@ procfs = { version = "0", default-featur
 
 [target.'cfg(unix)'.dev-dependencies]
 nix = { workspace = true, features = ["process", "signal", "user", "term"] }
-rlimit = "0.10.1"
+rlimit = "0"
 xattr = { workspace = true }
 
 # Specifically used in test_uptime::test_uptime_with_file_containing_valid_boot_time_utmpx_record
