Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -359,7 +359,7 @@ blake2b_simd = "1"
 blake3 = "1"
 sm3 = "0.4.2"
 crc32fast = "1.4.2"
-digest = "0.10.7"
+digest = "0.10"
 
 uucore = { version = "0.0.30", package = "uucore", path = "src/uucore" }
 uucore_procs = { version = "0.0.30", package = "uucore_procs", path = "src/uucore_procs" }
