Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -323,7 +323,7 @@ smallvec = { version = "1", features = [
 tempfile = "3"
 uutils_term_grid = "0.3"
 terminal_size = "0"
-textwrap = { version = "0.16.1", features = ["terminal_size"] }
+textwrap = { version = "0.16", features = ["terminal_size"] }
 thiserror = "1.0"
 time = { version = "0.3" }
 unicode-segmentation = "1"
