Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -299,7 +299,7 @@ notify = { version = "=6.0.0", features=
 num-bigint = "0.4.3"
 num-traits = "0.2.15"
 number_prefix = "0.4"
-once_cell = "1.17.2"
+once_cell = "1"
 onig = { version = "~6.4", default-features = false }
 ouroboros = "0.15.6"
 phf = "0.11.1"
