Index: coreutils/src/uu/csplit/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/csplit/Cargo.toml
+++ coreutils/src/uu/csplit/Cargo.toml
@@ -17,7 +17,7 @@ path = "src/csplit.rs"
 [dependencies]
 clap = { version = "4.0", features = ["wrap_help", "cargo"] }
 thiserror = "1.0"
-regex = "1.7.0"
+regex = "1"
 uucore = { version=">=0.0.17", package="uucore", path="../../uucore", features=["entries", "fs"] }
 
 [[bin]]
Index: coreutils/src/uu/hashsum/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/hashsum/Cargo.toml
+++ coreutils/src/uu/hashsum/Cargo.toml
@@ -20,7 +20,7 @@ clap = { version = "4.0", features = ["w
 hex = "0.4.3"
 memchr = "2"
 md-5 = "0.10.5"
-regex = "1.7.0"
+regex = "1"
 sha1 = "0.10.1"
 sha2 = "0.10.2"
 sha3 = "0.10.6"
Index: coreutils/src/uu/nl/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/nl/Cargo.toml
+++ coreutils/src/uu/nl/Cargo.toml
@@ -16,7 +16,7 @@ path = "src/nl.rs"
 
 [dependencies]
 clap = { version = "4.0", features = ["wrap_help", "cargo"] }
-regex = "1.7.0"
+regex = "1"
 uucore = { version=">=0.0.17", package="uucore", path="../../uucore" }
 
 [[bin]]
Index: coreutils/src/uu/pr/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/pr/Cargo.toml
+++ coreutils/src/uu/pr/Cargo.toml
@@ -20,7 +20,7 @@ time = { version = "0.3", features = ["l
 uucore = { version=">=0.0.17", package="uucore", path="../../uucore", features=["entries"] }
 quick-error = "2.0.1"
 itertools = "0.10.0"
-regex = "1.7"
+regex = "1"
 
 [[bin]]
 name = "pr"
Index: coreutils/src/uu/ptx/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/ptx/Cargo.toml
+++ coreutils/src/uu/ptx/Cargo.toml
@@ -16,7 +16,7 @@ path = "src/ptx.rs"
 
 [dependencies]
 clap = { version = "4.0", features = ["wrap_help", "cargo"] }
-regex = "1.7.0"
+regex = "1"
 uucore = { version=">=0.0.17", package="uucore", path="../../uucore" }
 
 [[bin]]
