Index: rust-coreutils-0.0.30/Cargo.toml
===================================================================
--- rust-coreutils-0.0.30.orig/Cargo.toml
+++ rust-coreutils-0.0.30/Cargo.toml
@@ -329,8 +329,7 @@ rstest = ">= 0.17.0"
 rust-ini = ">= 0.18.0"
 same-file = "1.0.6"
 self_cell = "1"
-# Remove the "=" once we moved to Rust edition 2024
-selinux = "= 0.5.0"
+selinux = "> 0.4.0"
 selinux-sys = "= 0.6.13"
 signal-hook = "0"
 smallvec = { version = "1", features = ["union"] }
