Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -368,7 +368,7 @@ regex = "1"
 rstest = "0.26.0"
 rust-ini = "0.21.0"
 same-file = "1.0.6"
-self_cell = "1.0.4"
+self_cell = "1"
 # FIXME we use the exact version because the new 0.5.3 requires an MSRV of 1.88
 selinux = "=0.5.2"
 signal-hook = "0"
