Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -332,7 +332,7 @@ self_cell = "1.0.4"
 # Remove the "=" once we moved to Rust edition 2024
 selinux = "= 0.5.0"
 selinux-sys = "= 0.6.13"
-signal-hook = "0.3.17"
+signal-hook = "0"
 smallvec = { version = "1", features = ["union"] }
 tempfile = "3"
 terminal_size = "0"
