Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -329,7 +329,7 @@ utf-8 = "0.7.6"
 walkdir = "2.4"
 #winapi-util = "0.1.6"
 #windows-sys = { version = "0.48.0", default-features = false }
-xattr = "1.0.1"
+xattr = "1"
 zip = { version = "0", default_features = false, features = ["deflate"] }
 
 hex = "0.4.3"
