Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -1,3 +1,4 @@
+
 # coreutils (uutils)
 # * see the repository LICENSE, README, and CONTRIBUTING files for more information
 
@@ -343,7 +344,7 @@ jiff = { version = "0.2.10", default-fea
 ] }
 libc = "0.2"
 linux-raw-sys = "0.11"
-lscolors = { version = "0.20.0", default-features = false, features = [
+lscolors = { version = "0", default-features = false, features = [
   "gnu_legacy",
 ] }
 memchr = "2.7.2"
