Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -29,8 +29,8 @@ itertools = { version="0.10.5", optional
 thiserror = { workspace=true, optional=true }
 time = { workspace=true, optional=true, features = ["formatting", "local-offset", "macros"] }
 # * "problem" dependencies (pinned)
-data-encoding = { version="2.4", optional=true }
-data-encoding-macro = { version="0.1.13", optional=true }
+data-encoding = { version="2", optional=true }
+data-encoding-macro = { version="0.1", optional=true }
 z85 = { version="3", optional=true }
 libc = { version="0.2.144", optional=true }
 once_cell = { workspace=true }
