Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -334,7 +334,7 @@ selinux = "= 0.5.0"
 selinux-sys = "= 0.6.13"
 signal-hook = "0.3.17"
 smallvec = { version = "1", features = ["union"] }
-tempfile = "3.15.0"
+tempfile = "3"
 terminal_size = "0"
 textwrap = { version = "0.16.1", features = ["terminal_size"] }
 thiserror = "2.0.3"
