Index: coreutils/src/uu/tail/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/tail/Cargo.toml
+++ coreutils/src/uu/tail/Cargo.toml
@@ -19,7 +19,8 @@ path = "src/tail.rs"
 clap = { version = "3.2", features = ["wrap_help", "cargo"] }
 libc = "0.2.132"
 memchr = "2.5.0"
-notify = { version = "=5.0.0", features=["macos_kqueue"]}
+notify = "5.0.0"
+#notify = { version = "=5.0.0", features=["macos_kqueue"]}
 uucore = { version=">=0.0.16", package="uucore", path="../../uucore", features=["ringbuffer", "lines"] }
 same-file = "1.0.6"
 
