Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -311,7 +311,7 @@ rand_core = "0.6"
 rayon = "1"
 redox_syscall = "0.3"
 regex = "1"
-rstest = "0.17.0"
+#rstest = "0.17.0"
 rust-ini = "0.18.0"
 same-file = "1.0.6"
 selinux = "0.4"
@@ -491,7 +491,7 @@ uucore = { workspace=true, features=["en
 walkdir = { workspace=true }
 is-terminal = { workspace=true }
 hex-literal = "0.4.1"
-rstest = "0.17.0"
+#rstest = "0.17.0"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dev-dependencies]
 procfs = { version = "0.15", default-features = false }
