Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -404,7 +404,7 @@ uucore = { version=">=0.0.17", package="
 walkdir = "2.2"
 atty = "0.2"
 hex-literal = "0.3.1"
-rstest = "0.16.0"
+#rstest = "0.16.0"
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dev-dependencies]
 procfs = { version = "0.14", default-features = false }
