Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -310,7 +310,7 @@ rand_core = "0.6"
 rayon = "1"
 redox_syscall = "0.4"
 regex = "1"
-rstest = "0.18.2"
+#rstest = "0.18.2"
 rust-ini = "0.19.0"
 same-file = "1.0.6"
 self_cell = "1.0.1"
@@ -489,7 +489,7 @@ unindent = "0.2"
 uucore = { workspace = true, features = ["entries", "process", "signals"] }
 walkdir = { workspace = true }
 hex-literal = "0.4.1"
-rstest = { workspace = true }
+#rstest = { workspace = true }
 
 [target.'cfg(any(target_os = "linux", target_os = "android"))'.dev-dependencies]
 procfs = { version = "0.15", default-features = false }
Index: coreutils/src/uu/tail/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/tail/Cargo.toml
+++ coreutils/src/uu/tail/Cargo.toml
@@ -31,8 +31,8 @@ windows-sys = { workspace = true, featur
 ] }
 winapi-util = { workspace = true }
 
-[dev-dependencies]
-rstest = { workspace = true }
+#[dev-dependencies]
+#rstest = { workspace = true }
 
 [[bin]]
 name = "tail"
