commit 67114ac63c580a7a7ff9e75ef7605b3d20385c91
Author: Daniel Hofstetter <daniel.hofstetter@42dh.com>
Date:   Sun Jun 11 14:05:26 2023 +0200

    date: switch to parse_datetime

Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -1,7 +1,7 @@
 # coreutils (uutils)
 # * see the repository LICENSE, README, and CONTRIBUTING files for more information
 
-# spell-checker:ignore (libs) libselinux gethostid procfs bigdecimal kqueue fundu mangen humantime uuhelp
+# spell-checker:ignore (libs) libselinux gethostid procfs bigdecimal kqueue fundu mangen datetime uuhelp
 
 [package]
 name = "coreutils"
@@ -286,7 +286,6 @@ fundu = "1"
 gcd = "2"
 glob = "0.3"
 half = "1.6"
-humantime_to_duration = "0.2.1"
 indicatif = "0.17"
 is-terminal = "0.4"
 itertools = "0"
@@ -302,6 +301,7 @@ number_prefix = "0.4"
 once_cell = "1"
 onig = { version = "6", default-features = false }
 ouroboros = "0.15"
+parse_datetime = "0.4.0"
 phf = "0"
 phf_codegen = "0"
 platform-info = "2"
Index: coreutils/src/uu/date/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/date/Cargo.toml
+++ coreutils/src/uu/date/Cargo.toml
@@ -1,4 +1,4 @@
-# spell-checker:ignore humantime
+# spell-checker:ignore datetime
 [package]
 name = "uu_date"
 version = "0.0.19"
@@ -21,7 +21,7 @@ chrono = { workspace=true }
 time = { workspace=true }
 clap = { workspace=true }
 uucore = { workspace=true }
-humantime_to_duration = { workspace=true }
+parse_datetime = { workspace=true }
 
 [target.'cfg(unix)'.dependencies]
 libc = { workspace=true }
Index: coreutils/src/uu/date/src/date.rs
===================================================================
--- coreutils.orig/src/uu/date/src/date.rs
+++ coreutils/src/uu/date/src/date.rs
@@ -6,10 +6,10 @@
 // For the full copyright and license information, please view the LICENSE
 // file that was distributed with this source code.
 
-// spell-checker:ignore (chrono) Datelike Timelike ; (format) DATEFILE MMDDhhmm ; (vars) datetime datetimes humantime
+// spell-checker:ignore (chrono) Datelike Timelike ; (format) DATEFILE MMDDhhmm ; (vars) datetime datetimes
 
 use chrono::format::{Item, StrftimeItems};
-use chrono::{DateTime, Duration as ChronoDuration, FixedOffset, Local, Offset, Utc};
+use chrono::{DateTime, Duration, FixedOffset, Local, Offset, Utc};
 #[cfg(windows)]
 use chrono::{Datelike, Timelike};
 use clap::{crate_version, Arg, ArgAction, Command};
@@ -18,7 +18,6 @@ use libc::{clock_settime, timespec, CLOC
 use std::fs::File;
 use std::io::{BufRead, BufReader};
 use std::path::PathBuf;
-use time::Duration;
 use uucore::display::Quotable;
 #[cfg(not(any(target_os = "redox")))]
 use uucore::error::FromIo;
@@ -171,7 +170,7 @@ pub fn uumain(args: impl uucore::Args) -
     };
 
     let date_source = if let Some(date) = matches.get_one::<String>(OPT_DATE) {
-        if let Ok(duration) = humantime_to_duration::from_str(date.as_str()) {
+        if let Ok(duration) = parse_datetime::from_str(date.as_str()) {
             DateSource::Human(duration)
         } else {
             DateSource::Custom(date.into())
@@ -226,13 +225,10 @@ pub fn uumain(args: impl uucore::Args) -
                 let iter = std::iter::once(date);
                 Box::new(iter)
             }
-            DateSource::Human(ref input) => {
-                // Get the current DateTime<FixedOffset> and convert the input time::Duration to chrono::Duration
-                // for things like "1 year ago"
+            DateSource::Human(relative_time) => {
+                // Get the current DateTime<FixedOffset> for things like "1 year ago"
                 let current_time = DateTime::<FixedOffset>::from(Local::now());
-                let input_chrono = ChronoDuration::seconds(input.as_seconds_f32() as i64)
-                    + ChronoDuration::nanoseconds(input.subsec_nanoseconds() as i64);
-                let iter = std::iter::once(Ok(current_time + input_chrono));
+                let iter = std::iter::once(Ok(current_time + relative_time));
                 Box::new(iter)
             }
             DateSource::File(ref path) => {
