commit 120a0e09fc968bbd3603c0a40624cbaef717269e
Author: Daniel Hofstetter <daniel.hofstetter@42dh.com>
Date:   Sun Feb 11 13:24:20 2024 +0100

    ls,date: rename chrono::Duration to TimeDelta

Index: coreutils/src/uu/date/src/date.rs
===================================================================
--- coreutils.orig/src/uu/date/src/date.rs
+++ coreutils/src/uu/date/src/date.rs
@@ -6,7 +6,7 @@
 // spell-checker:ignore (chrono) Datelike Timelike ; (format) DATEFILE MMDDhhmm ; (vars) datetime datetimes
 
 use chrono::format::{Item, StrftimeItems};
-use chrono::{DateTime, FixedOffset, Local, Offset, TimeDelta, Utc};
+use chrono::{DateTime, Duration, FixedOffset, Local, Offset, Utc};
 #[cfg(windows)]
 use chrono::{Datelike, Timelike};
 use clap::{crate_version, Arg, ArgAction, Command};
@@ -92,7 +92,7 @@ enum DateSource {
     Custom(String),
     File(PathBuf),
     Stdin,
-    Human(TimeDelta),
+    Human(Duration),
 }
 
 enum Iso8601Format {
Index: coreutils/src/uu/ls/src/ls.rs
===================================================================
--- coreutils.orig/src/uu/ls/src/ls.rs
+++ coreutils/src/uu/ls/src/ls.rs
@@ -3005,7 +3005,7 @@ fn display_date(metadata: &Metadata, con
         Some(time) => {
             //Date is recent if from past 6 months
             //According to GNU a Gregorian year has 365.2425 * 24 * 60 * 60 == 31556952 seconds on the average.
-            let recent = time + chrono::TimeDelta::seconds(31_556_952 / 2) > chrono::Local::now();
+            let recent = time + chrono::Duration::seconds(31_556_952 / 2) > chrono::Local::now();
 
             match &config.time_style {
                 TimeStyle::FullIso => time.format("%Y-%m-%d %H:%M:%S.%f %z"),
