commit ddb027231f66591dd677926ef3117a912bec1d24
Author: Daniel Hofstetter <daniel.hofstetter@42dh.com>
Date:   Wed Jan 29 14:59:47 2025 +0100

    shred: adapt to API changes of rand

Index: coreutils/src/uu/shred/src/shred.rs
===================================================================
--- coreutils.orig/src/uu/shred/src/shred.rs
+++ coreutils/src/uu/shred/src/shred.rs
@@ -189,7 +189,7 @@ impl<'a> BytesWriter<'a> {
         match pass {
             PassType::Random => match random_source {
                 RandomSource::System => Self::Random {
-                    rng: StdRng::from_os_rng(),
+                    rng: StdRng::from_entropy(),
                     buffer: [0; BLOCK_SIZE],
                 },
                 RandomSource::Read(file) => Self::RandomFile {
@@ -507,7 +507,7 @@ fn wipe_file(
                 pass_sequence.push(PassType::Random);
             }
 
-            let mut rng = rand::rng();
+            let mut rng = rand::thread_rng();
             pass_sequence[1..].shuffle(&mut rng); // randomize the order of application
             pass_sequence.push(PassType::Random); // add the last random pass
         }
