Index: coreutils/src/uu/ls/src/ls.rs
===================================================================
--- coreutils.orig/src/uu/ls/src/ls.rs
+++ coreutils/src/uu/ls/src/ls.rs
@@ -25,6 +25,7 @@ use std::{
     time::Duration,
 };
 use std::{collections::HashSet, io::IsTerminal};
+use ansi_width::ansi_width;
 
 use ansi_width::ansi_width;
 use chrono::{DateTime, Local, TimeDelta};
Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -368,6 +368,7 @@ uu_ls = { version = "0.0.30", path = "sr
 uu_base32 = { version = "0.0.30", path = "src/uu/base32" }
 
 [dependencies]
+unicode-width = { workspace = true }
 clap = { workspace = true }
 uucore = { workspace = true }
 clap_complete = { workspace = true }
