Index: coreutils/Cargo.toml
===================================================================
--- coreutils.orig/Cargo.toml
+++ coreutils/Cargo.toml
@@ -410,6 +410,7 @@ uu_base32 = { version = "0.2.2", path =
 uutests = { version = "0.2.2", package = "uutests", path = "tests/uutests" }
 
 [dependencies]
+unicode-width = { workspace = true }
 clap.workspace = true
 uucore.workspace = true
 clap_complete.workspace = true
