Index: coreutils/src/uucore/Cargo.toml
===================================================================
--- coreutils.orig/src/uucore/Cargo.toml
+++ coreutils/src/uucore/Cargo.toml
@@ -124,4 +124,4 @@ version-cmp = []
 wide = []
 custom-tz-fmt = []
 tty = []
-uptime = ["libc", "windows-sys", "utmpx", "utmp-classic", "thiserror"]
+uptime = ["libc", "utmpx", "thiserror"]
Index: coreutils/src/uu/uptime/Cargo.toml
===================================================================
--- coreutils.orig/src/uu/uptime/Cargo.toml
+++ coreutils/src/uu/uptime/Cargo.toml
@@ -25,11 +25,11 @@ uucore = { workspace = true, features =
 #[target.'cfg(target_os = "openbsd")'.dependencies]
 #utmp-classic = { workspace = true }
 
-[target.'cfg(target_os="windows")'.dependencies]
-windows-sys = { workspace = true, features = [
-  "Win32_System_RemoteDesktop",
-  "Wdk_System_SystemInformation",
-] }
+#[target.'cfg(target_os="windows")'.dependencies]
+#windows-sys = { workspace = true, features = [
+#  "Win32_System_RemoteDesktop",
+#  "Wdk_System_SystemInformation",
+#] }
 
 [[bin]]
 name = "uptime"
