Index: coreutils/src/uu/date/src/date.rs
===================================================================
--- coreutils.orig/src/uu/date/src/date.rs
+++ coreutils/src/uu/date/src/date.rs
@@ -442,10 +442,10 @@ fn set_system_datetime(_date: Zoned) ->
 /// `<https://www.gnu.org/software/libc/manual/html_node/Time-Types.html>`
 fn set_system_datetime(date: Zoned) -> UResult<()> {
     let ts = date.timestamp();
-    let timespec = timespec {
-        tv_sec: ts.as_second() as _,
-        tv_nsec: ts.subsec_nanosecond() as _,
-    };
+
+    let mut timespec: timespec = unsafe { std::mem::zeroed() };
+    timespec.tv_sec = date.timestamp() as _;
+    timespec.tv_nsec = date.timestamp_subsec_nanos() as _;
 
     let result = unsafe { clock_settime(CLOCK_REALTIME, &raw const timespec) };
 
