--- a/Cargo.toml
+++ b/Cargo.toml
@@ -78,15 +78,15 @@
 name = "timer"
 path = "tests/timer.rs"
 
-[[bench]]
-name = "io"
-path = "benches/io.rs"
-harness = false
-
-[[bench]]
-name = "timer"
-path = "benches/timer.rs"
-harness = false
+#[[bench]]
+#name = "io"
+#path = "benches/io.rs"
+#harness = false
+
+#[[bench]]
+#name = "timer"
+#path = "benches/timer.rs"
+#harness = false
 
 [dependencies.async-lock]
 version = "3"
@@ -137,10 +137,10 @@
 [dev-dependencies.blocking]
 version = "1"
 
-[dev-dependencies.criterion]
-version = "0.5"
-features = ["cargo_bench_support"]
-default-features = false
+#[dev-dependencies.criterion]
+#version = "0.5"
+#features = ["cargo_bench_support"]
+#default-features = false
 
 [dev-dependencies.getrandom]
 version = "0.2"
