--- a/Cargo.toml
+++ b/Cargo.toml
@@ -89,13 +89,13 @@
 harness = false
 
 [dependencies.async-lock]
-version = "3.0.0"
+version = "3"
 
 [dependencies.cfg-if]
 version = "1"
 
 [dependencies.concurrent-queue]
-version = "2.2.0"
+version = "2.5"
 
 [dependencies.futures-io]
 version = "0.3.28"
@@ -103,8 +103,7 @@
 default-features = false
 
 [dependencies.futures-lite]
-version = "2.0.0"
-default-features = false
+version = ">= 1.11.0"
 
 [dependencies.parking]
 version = "2.0.0"
@@ -130,21 +129,21 @@
 default-features = false
 
 [dev-dependencies.async-channel]
-version = "2.0.0"
+version = ">= 1.9.0, < 3.0"
 
 [dev-dependencies.async-net]
-version = "2.0.0"
+version = "2"
 
 [dev-dependencies.blocking]
 version = "1"
 
 [dev-dependencies.criterion]
-version = "0.4"
+version = "0.5"
 features = ["cargo_bench_support"]
 default-features = false
 
 [dev-dependencies.getrandom]
-version = "0.3"
+version = "0.2"
 
 [dev-dependencies.signal-hook]
 version = "0.3"
