Index: async-io/Cargo.toml
===================================================================
--- async-io.orig/Cargo.toml
+++ async-io/Cargo.toml
@@ -109,10 +109,3 @@ default-features = false
 
 [target."cfg(target_os = \"linux\")".dev-dependencies.timerfd]
 version = "1"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = ["Win32_Foundation"]
-
-[target."cfg(windows)".dev-dependencies.uds_windows]
-version = "1"
