Index: async-io/Cargo.toml
===================================================================
--- async-io.orig/Cargo.toml
+++ async-io/Cargo.toml
@@ -31,6 +31,22 @@ categories = [
 ]
 license = "Apache-2.0 OR MIT"
 repository = "https://github.com/smol-rs/async-io"
+autoexamples = false
+
+[[example]]
+name = "kqueue-process"
+
+[[example]]
+name = "linux-timerfd"
+
+[[example]]
+name = "windows-command"
+
+[[example]]
+name = "unix-signal"
+
+[[example]]
+name = "windows-uds"
 
 [[bench]]
 name = "io"
@@ -105,4 +121,0 @@
-[target."cfg(target_os = \"linux\")".dev-dependencies.inotify]
-version = "0.10.1"
-default-features = false
-
