--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,10 +47,6 @@
 path = "examples/kqueue-process.rs"
 
 [[example]]
-name = "linux-inotify"
-path = "examples/linux-inotify.rs"
-
-[[example]]
 name = "linux-timerfd"
 path = "examples/linux-timerfd.rs"
 
@@ -156,10 +152,6 @@
 [dev-dependencies.tempfile]
 version = "3"
 
-[target.'cfg(target_os = "linux")'.dev-dependencies.inotify]
-version = "0.11.0"
-default-features = false
-
 [target.'cfg(target_os = "linux")'.dev-dependencies.timerfd]
 version = "1"
 
