Index: wl-nl80211/Cargo.toml
===================================================================
--- wl-nl80211.orig/Cargo.toml
+++ wl-nl80211/Cargo.toml
@@ -38,14 +38,17 @@ path = "src/lib.rs"
 [[example]]
 name = "dump_nl80211_iface"
 path = "examples/dump_nl80211_iface.rs"
+required-features = ["tokio_socket"]
 
 [[example]]
 name = "dump_nl80211_station"
 path = "examples/dump_nl80211_station.rs"
+required-features = ["tokio_socket"]
 
 [[example]]
 name = "dump_nl80211_wiphy"
 path = "examples/dump_nl80211_wiphy.rs"
+required-features = ["tokio_socket"]
 
 [dependencies.anyhow]
 version = "1.0.44"
