--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,7 @@
 optional = true
 
 [dependencies.git2]
-version = "0.19.0"
+version = "0.18"
 optional = true
 default-features = false
 
@@ -62,11 +62,6 @@
     "parsing",
 ]
 
-[dependencies.tzdb]
-version = "0.6.0"
-features = ["local"]
-optional = true
-default-features = false
 
 [dev-dependencies.winnow]
 version = "0.6"
@@ -74,5 +69,4 @@
 [features]
 default = [
     "git2",
-    "tzdb",
 ]
