Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,3 +51,3 @@
     "tzdb",
-    "is_debug/std",
+    #"is_debug/std",
     "std",
@@ -88,3 +88,3 @@
 [dependencies.git2]
-version = "0.20.0"
+version = ">= 0.19, < 0.21"
 optional = true
@@ -93,3 +93,3 @@
 [dependencies.is_debug]
-version = "1.1.0"
+version = "1"
 default-features = false
@@ -120,2 +120,2 @@
 [dev-dependencies.winnow]
-version = "0.7"
+version = "0.6"
