--- a/src/date_time.rs
+++ b/src/date_time.rs
@@ -185,2 +185,3 @@
     #[test]
+    #[ignore = "tries to access /etc/localtime"]
     fn test_local_now_human_format() {
--- a/src/git.rs
+++ b/src/git.rs
@@ -571,2 +571,3 @@
     #[test]
+    #[ignore = "broken in debian"]
     fn test_git() {
@@ -594,2 +595,3 @@
     #[test]
+    #[ignore = "broken in debian"]
     fn test_current_branch() {
@@ -614,2 +616,3 @@
     #[test]
+    #[ignore = "broken in debian"]
     fn test_command_last_tag() {
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -687,2 +687,3 @@
     #[test]
+    #[ignore = "broken in debian"]
     fn test_build() -> SdResult<()> {
@@ -699,2 +700,3 @@
     #[test]
+    #[ignore = "broken in debian"]
     fn test_build_deny() -> SdResult<()> {
