Forwarded: not-needed
--- a/src/date_time.rs
+++ b/src/date_time.rs
@@ -194,2 +194,3 @@
     #[test]
+    #[ignore = "tries to access /etc/localtime"]
     fn test_local_now_human_format() {
--- a/src/git.rs
+++ b/src/git.rs
@@ -686,2 +686,3 @@
     #[test]
+    #[ignore = "unavailable in debian"]
     fn test_git() {
@@ -713,2 +714,3 @@
     #[test]
+    #[ignore = "unavailable in debian"]
     fn test_current_branch() {
@@ -733,2 +735,3 @@
     #[test]
+    #[ignore = "unavailable in debian"]
     fn test_parse_git_describe() {
