--- a/Cargo.toml
+++ b/Cargo.toml
@@ -75,3 +75,3 @@
 [dependencies.rkyv]
-version = "0.7"
+version = "0.8"
 optional = true
--- a/src/bfloat.rs
+++ b/src/bfloat.rs
@@ -42,3 +42,3 @@
 )]
-#[cfg_attr(feature = "rkyv", archive(resolver = "Bf16Resolver"))]
+#[cfg_attr(feature = "rkyv", rkyv(resolver = Bf16Resolver))]
 #[cfg_attr(feature = "bytemuck", derive(Zeroable, Pod))]
--- a/src/binary16.rs
+++ b/src/binary16.rs
@@ -41,3 +41,3 @@
 )]
-#[cfg_attr(feature = "rkyv", archive(resolver = "F16Resolver"))]
+#[cfg_attr(feature = "rkyv", rkyv(resolver = F16Resolver))]
 #[cfg_attr(feature = "bytemuck", derive(Zeroable, Pod))]
