Index: fern/Cargo.toml
===================================================================
--- fern.orig/Cargo.toml
+++ fern/Cargo.toml
@@ -70,16 +70,11 @@ version = "0.3"
 [features]
 date-based = ["chrono"]
 meta-logging-in-format = []
-reopen-03 = ["reopen", "libc"]
 syslog-6 = ["syslog6"]
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2.58"
 optional = true
 
-[target."cfg(not(windows))".dependencies.reopen]
-version = "^0.3"
-optional = true
-
 [target."cfg(not(windows))".dependencies.syslog6]
 version = "6"
 optional = true
