--- a/Cargo.toml
+++ b/Cargo.toml
@@ -92,30 +92,13 @@
 [features]
 date-based = ["chrono"]
 meta-logging-in-format = []
-reopen-03 = [
-    "reopen03",
-    "libc",
-]
-reopen-1 = [
-    "reopen1",
-    "libc",
-]
 syslog-6 = ["syslog6"]
 
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2.58"
 optional = true
 
-[target."cfg(not(windows))".dependencies.reopen03]
-version = "^0.3"
-optional = true
-package = "reopen"
 
-[target."cfg(not(windows))".dependencies.reopen1]
-version = "~1"
-features = ["signals"]
-optional = true
-package = "reopen"
 
 [target."cfg(not(windows))".dependencies.syslog6]
 version = "6"
