--- a/Cargo.toml
+++ b/Cargo.toml
@@ -92,8 +92,0 @@
-reopen-03 = [
-    "reopen03",
-    "libc",
-]
-reopen-1 = [
-    "reopen1",
-    "libc",
-]
@@ -110,4 +110,0 @@
-[target."cfg(not(windows))".dependencies.reopen03]
-version = "^0.3"
-optional = true
-package = "reopen"
@@ -120,5 +120,0 @@
-[target."cfg(not(windows))".dependencies.reopen1]
-version = "~1"
-features = ["signals"]
-optional = true
-package = "reopen"
