Index: fern/Cargo.toml
===================================================================
--- fern.orig/Cargo.toml
+++ fern/Cargo.toml
@@ -70,8 +70,0 @@ version = "3"
-reopen-03 = [
-    "reopen03",
-    "libc",
-]
-reopen-1 = [
-    "reopen1",
-    "libc",
-]
@@ -80,4 +80,0 @@ version = "3"
-[target."cfg(not(windows))".dependencies.reopen03]
-version = "^0.3"
-optional = true
-package = "reopen"
@@ -90,5 +90,0 @@ version = "3"
-[target."cfg(not(windows))".dependencies.reopen1]
-version = "~1"
-features = ["signals"]
-optional = true
-package = "reopen"
