Description: only syslog 6 is available
Author: Ed Neville <ed-debian@s5h.net>
Origin: other
Forwarded: not-needed
Last-Update: 2021-12-24

--- rust-fern-0.6.0.orig/Cargo.toml
+++ rust-fern-0.6.0/Cargo.toml
@@ -43,11 +43,7 @@ required-features = ["colored"]
 
 [[example]]
 name = "syslog"
-required-features = ["syslog-4"]
-
-[[example]]
-name = "syslog3"
-required-features = ["syslog-3"]
+required-features = ["syslog-6"]
 
 [[example]]
 name = "meta-logging"
@@ -75,8 +71,7 @@ version = "0.3"
 date-based = ["chrono"]
 meta-logging-in-format = []
 reopen-03 = ["reopen", "libc"]
-syslog-3 = ["syslog3"]
-syslog-4 = ["syslog4"]
+syslog-6 = ["syslog6"]
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2.58"
 optional = true
@@ -85,15 +80,11 @@ optional = true
 version = "^0.3"
 optional = true
 
-[target."cfg(not(windows))".dependencies.syslog3]
-version = "3"
+[target."cfg(not(windows))".dependencies.syslog6]
+version = "6"
 optional = true
 package = "syslog"
 
-[target."cfg(not(windows))".dependencies.syslog4]
-version = "4"
-optional = true
-package = "syslog"
 [badges.appveyor]
 repository = "daboross/fern"
 
