Description: only syslog 6 is available
Author: Ed Neville <ed-debian@s5h.net>
Origin: other
Forwarded: not-needed
Last-Update: 2021-12-24

Index: fern/Cargo.toml
===================================================================
--- fern.orig/Cargo.toml
+++ fern/Cargo.toml
@@ -43,11 +43,7 @@ required-features = ["colored"]
 
 [[example]]
 name = "syslog"
-required-features = ["syslog-4"]
-
-[[example]]
-name = "syslog3"
-required-features = ["syslog-3"]
+required-features = ["syslog-6"]
 
 [[example]]
 name = "meta-logging"
@@ -75,8 +71,7 @@ version = "0.3"
 date-based = ["chrono"]
 meta-logging-in-format = []
 reopen-03 = ["reopen", "libc"]
-syslog-3 = ["syslog3"]
-syslog-4 = ["syslog4"]
+syslog-6 = ["syslog6"]
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2.58"
 optional = true
@@ -85,15 +80,11 @@ optional = true
 version = "^0.3"
 optional = true
 
-[target."cfg(not(windows))".dependencies.syslog3]
-version = "3"
+[target."cfg(not(windows))".dependencies.syslog6]
+version = "6"
 optional = true
 package = "syslog"
 
-[target."cfg(not(windows))".dependencies.syslog4]
-version = "4"
-optional = true
-package = "syslog"
 [badges.appveyor]
 repository = "daboross/fern"
 
Index: fern/examples/syslog3.rs
===================================================================
--- fern.orig/examples/syslog3.rs
+++ fern/examples/syslog3.rs
@@ -1,4 +1,7 @@
-#[cfg(not(windows))]
+//This example has been disabled to allow the rest of the testsuite to
+//run
+
+/*#[cfg(not(windows))]
 // This is necessary because `fern` depends on both version 3 and 4.
 use syslog3 as syslog;
 
@@ -15,11 +18,11 @@ fn setup_logging() -> Result<(), fern::I
         .apply()?;
 
     Ok(())
-}
+}*/
 
 #[cfg(not(windows))]
 fn main() {
-    setup_logging().expect("failed to initialize logging.");
+    /*setup_logging().expect("failed to initialize logging.");
 
     // None of this will be shown in the syslog:
     for i in 0..5 {
@@ -38,7 +41,7 @@ fn main() {
 
     info!(target: "explicit-syslog", "hello to the syslog! this is rust.");
 
-    warn!("AHHH something's on fire.");
+    warn!("AHHH something's on fire.");*/
 }
 
 #[cfg(windows)]
Index: fern/examples/syslog.rs
===================================================================
--- fern.orig/examples/syslog.rs
+++ fern/examples/syslog.rs
@@ -1,6 +1,6 @@
 #[cfg(not(windows))]
 // This is necessary because `fern` depends on both version 3 and 4.
-use syslog4 as syslog;
+use syslog6 as syslog;
 
 use log::{debug, info, warn};
 
Index: fern/src/builders.rs
===================================================================
--- fern.orig/src/builders.rs
+++ fern/src/builders.rs
@@ -8,7 +8,7 @@ use std::{
 #[cfg(feature = "date-based")]
 use std::path::{Path, PathBuf};
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
+#[cfg(all(not(windows), feature = "syslog-6"))]
 use std::collections::HashMap;
 
 use log::Log;
@@ -18,8 +18,8 @@ use crate::{log_impl, Filter, FormatCall
 #[cfg(feature = "date-based")]
 use crate::log_impl::DateBasedState;
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-use crate::{Syslog4Rfc3164Logger, Syslog4Rfc5424Logger};
+#[cfg(all(not(windows), feature = "syslog-6"))]
+use crate::{Syslog6Rfc3164Logger, Syslog6Rfc5424Logger};
 
 /// The base dispatch logger.
 ///
@@ -463,17 +463,17 @@ impl Dispatch {
                     max_child_level = log::LevelFilter::Trace;
                     Some(log_impl::Output::Syslog3(log_impl::Syslog3 { inner: log }))
                 }
-                #[cfg(all(not(windows), feature = "syslog-4"))]
-                OutputInner::Syslog4Rfc3164(logger) => {
+                #[cfg(all(not(windows), feature = "syslog-6"))]
+                OutputInner::Syslog6Rfc3164(logger) => {
                     max_child_level = log::LevelFilter::Trace;
-                    Some(log_impl::Output::Syslog4Rfc3164(log_impl::Syslog4Rfc3164 {
+                    Some(log_impl::Output::Syslog6Rfc3164(log_impl::Syslog6Rfc3164 {
                         inner: Mutex::new(logger),
                     }))
                 }
-                #[cfg(all(not(windows), feature = "syslog-4"))]
-                OutputInner::Syslog4Rfc5424 { logger, transform } => {
+                #[cfg(all(not(windows), feature = "syslog-6"))]
+                OutputInner::Syslog6Rfc5424 { logger, transform } => {
                     max_child_level = log::LevelFilter::Trace;
-                    Some(log_impl::Output::Syslog4Rfc5424(log_impl::Syslog4Rfc5424 {
+                    Some(log_impl::Output::Syslog6Rfc5424(log_impl::Syslog6Rfc5424 {
                         inner: Mutex::new(logger),
                         transform,
                     }))
@@ -657,12 +657,12 @@ enum OutputInner {
     #[cfg(all(not(windows), feature = "syslog-3"))]
     Syslog3(syslog3::Logger),
     /// Passes all messages to the syslog.
-    #[cfg(all(not(windows), feature = "syslog-4"))]
-    Syslog4Rfc3164(Syslog4Rfc3164Logger),
+    #[cfg(all(not(windows), feature = "syslog-6"))]
+    Syslog6Rfc3164(Syslog6Rfc3164Logger),
     /// Sends all messages through the transform then passes to the syslog.
-    #[cfg(all(not(windows), feature = "syslog-4"))]
-    Syslog4Rfc5424 {
-        logger: Syslog4Rfc5424Logger,
+    #[cfg(all(not(windows), feature = "syslog-6"))]
+    Syslog6Rfc5424 {
+        logger: Syslog6Rfc5424Logger,
         transform: Box<
             dyn Fn(&log::Record) -> (i32, HashMap<String, HashMap<String, String>>, String)
                 + Sync
@@ -858,8 +858,8 @@ impl From<Box<syslog3::Logger>> for Outp
     }
 }
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-impl From<Syslog4Rfc3164Logger> for Output {
+#[cfg(all(not(windows), feature = "syslog-6"))]
+impl From<Syslog6Rfc3164Logger> for Output {
     /// Creates an output logger which writes all messages to the given syslog.
     ///
     /// Log levels are translated trace => debug, debug => debug, info =>
@@ -871,9 +871,9 @@ impl From<Syslog4Rfc3164Logger> for Outp
     /// This is for RFC 3164 loggers. To use an RFC 5424 logger, use the
     /// [`Output::syslog_5424`] helper method.
     ///
-    /// This requires the `"syslog-4"` feature.
-    fn from(log: Syslog4Rfc3164Logger) -> Self {
-        Output(OutputInner::Syslog4Rfc3164(log))
+    /// This requires the `"syslog-6"` feature.
+    fn from(log: Syslog6Rfc3164Logger) -> Self {
+        Output(OutputInner::Syslog6Rfc3164(log))
     }
 }
 
@@ -1094,18 +1094,18 @@ impl Output {
     /// If you're an expert on syslog logging and would like to contribute
     /// an example to put here, it would be gladly accepted!
     ///
-    /// This requires the `"syslog-4"` feature.
+    /// This requires the `"syslog-6"` feature.
     ///
     /// [the rfc]: https://tools.ietf.org/html/rfc5424
-    #[cfg(all(not(windows), feature = "syslog-4"))]
-    pub fn syslog_5424<F>(logger: Syslog4Rfc5424Logger, transform: F) -> Self
+    #[cfg(all(not(windows), feature = "syslog-6"))]
+    pub fn syslog_5424<F>(logger: Syslog6Rfc5424Logger, transform: F) -> Self
     where
         F: Fn(&log::Record) -> (i32, HashMap<String, HashMap<String, String>>, String)
             + Sync
             + Send
             + 'static,
     {
-        Output(OutputInner::Syslog4Rfc5424 {
+        Output(OutputInner::Syslog6Rfc5424 {
             logger,
             transform: Box::new(transform),
         })
@@ -1240,14 +1240,14 @@ impl fmt::Debug for OutputInner {
                 .debug_tuple("Output::Syslog3")
                 .field(&"<unprintable syslog::Logger>")
                 .finish(),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            OutputInner::Syslog4Rfc3164 { .. } => f
-                .debug_tuple("Output::Syslog4Rfc3164")
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            OutputInner::Syslog6Rfc3164 { .. } => f
+                .debug_tuple("Output::Syslog6Rfc3164")
                 .field(&"<unprintable syslog::Logger>")
                 .finish(),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            OutputInner::Syslog4Rfc5424 { .. } => f
-                .debug_tuple("Output::Syslog4Rfc5424")
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            OutputInner::Syslog6Rfc5424 { .. } => f
+                .debug_tuple("Output::Syslog6Rfc5424")
                 .field(&"<unprintable syslog::Logger>")
                 .finish(),
             OutputInner::Dispatch(ref dispatch) => {
Index: fern/src/lib.rs
===================================================================
--- fern.orig/src/lib.rs
+++ fern/src/lib.rs
@@ -207,7 +207,7 @@ use std::{
     path::Path,
 };
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
+#[cfg(all(not(windows), feature = "syslog-6"))]
 use std::collections::HashMap;
 
 pub use crate::{
@@ -222,7 +222,7 @@ mod log_impl;
 
 #[cfg(feature = "colored")]
 pub mod colors;
-#[cfg(all(not(windows), feature = "syslog-3", feature = "syslog-4"))]
+#[cfg(all(not(windows), feature = "syslog-3", feature = "syslog-6"))]
 pub mod syslog;
 
 pub mod meta;
@@ -240,14 +240,13 @@ pub type Filter = dyn Fn(&log::Metadata)
 #[cfg(feature = "date-based")]
 pub use crate::builders::DateBased;
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-type Syslog4Rfc3164Logger = syslog4::Logger<syslog4::LoggerBackend, String, syslog4::Formatter3164>;
+#[cfg(all(not(windows), feature = "syslog-6"))]
+type Syslog6Rfc3164Logger = syslog6::Logger<syslog6::LoggerBackend, syslog6::Formatter3164>;
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-type Syslog4Rfc5424Logger = syslog4::Logger<
-    syslog4::LoggerBackend,
-    (i32, HashMap<String, HashMap<String, String>>, String),
-    syslog4::Formatter5424,
+#[cfg(all(not(windows), feature = "syslog-6"))]
+type Syslog6Rfc5424Logger = syslog6::Logger<
+    syslog6::LoggerBackend,
+    syslog6::Formatter5424,
 >;
 
 /// Convenience method for opening a log file with common options.
Index: fern/src/log_impl.rs
===================================================================
--- fern.orig/src/log_impl.rs
+++ fern/src/log_impl.rs
@@ -17,8 +17,8 @@ use log::{self, Log};
 
 use crate::{Filter, Formatter};
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-use crate::{Syslog4Rfc3164Logger, Syslog4Rfc5424Logger};
+#[cfg(all(not(windows), feature = "syslog-6"))]
+use crate::{Syslog6Rfc3164Logger, Syslog6Rfc5424Logger};
 #[cfg(all(not(windows), feature = "reopen-03"))]
 use reopen;
 
@@ -64,10 +64,10 @@ pub enum Output {
     Sender(Sender),
     #[cfg(all(not(windows), feature = "syslog-3"))]
     Syslog3(Syslog3),
-    #[cfg(all(not(windows), feature = "syslog-4"))]
-    Syslog4Rfc3164(Syslog4Rfc3164),
-    #[cfg(all(not(windows), feature = "syslog-4"))]
-    Syslog4Rfc5424(Syslog4Rfc5424),
+    #[cfg(all(not(windows), feature = "syslog-6"))]
+    Syslog6Rfc3164(Syslog6Rfc3164),
+    #[cfg(all(not(windows), feature = "syslog-6"))]
+    Syslog6Rfc5424(Syslog6Rfc5424),
     Dispatch(Dispatch),
     SharedDispatch(Arc<Dispatch>),
     OtherBoxed(Box<dyn Log>),
@@ -116,14 +116,14 @@ pub struct Syslog3 {
     pub inner: syslog3::Logger,
 }
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-pub struct Syslog4Rfc3164 {
-    pub inner: Mutex<Syslog4Rfc3164Logger>,
+#[cfg(all(not(windows), feature = "syslog-6"))]
+pub struct Syslog6Rfc3164 {
+    pub inner: Mutex<Syslog6Rfc3164Logger>,
 }
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-pub struct Syslog4Rfc5424 {
-    pub inner: Mutex<Syslog4Rfc5424Logger>,
+#[cfg(all(not(windows), feature = "syslog-6"))]
+pub struct Syslog6Rfc5424 {
+    pub inner: Mutex<Syslog6Rfc5424Logger>,
     pub transform: Box<
         dyn Fn(&log::Record) -> (i32, HashMap<String, HashMap<String, String>>, String)
             + Sync
@@ -306,10 +306,10 @@ impl Log for Output {
             Output::OtherStatic(ref s) => s.enabled(metadata),
             #[cfg(all(not(windows), feature = "syslog-3"))]
             Output::Syslog3(ref s) => s.enabled(metadata),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            Output::Syslog4Rfc3164(ref s) => s.enabled(metadata),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            Output::Syslog4Rfc5424(ref s) => s.enabled(metadata),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            Output::Syslog6Rfc3164(ref s) => s.enabled(metadata),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            Output::Syslog6Rfc5424(ref s) => s.enabled(metadata),
             Output::Panic(ref s) => s.enabled(metadata),
             Output::Writer(ref s) => s.enabled(metadata),
             #[cfg(feature = "date-based")]
@@ -331,10 +331,10 @@ impl Log for Output {
             Output::OtherStatic(ref s) => s.log(record),
             #[cfg(all(not(windows), feature = "syslog-3"))]
             Output::Syslog3(ref s) => s.log(record),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            Output::Syslog4Rfc3164(ref s) => s.log(record),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            Output::Syslog4Rfc5424(ref s) => s.log(record),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            Output::Syslog6Rfc3164(ref s) => s.log(record),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            Output::Syslog6Rfc5424(ref s) => s.log(record),
             Output::Panic(ref s) => s.log(record),
             Output::Writer(ref s) => s.log(record),
             #[cfg(feature = "date-based")]
@@ -356,10 +356,10 @@ impl Log for Output {
             Output::OtherStatic(ref s) => s.flush(),
             #[cfg(all(not(windows), feature = "syslog-3"))]
             Output::Syslog3(ref s) => s.flush(),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            Output::Syslog4Rfc3164(ref s) => s.flush(),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            Output::Syslog4Rfc5424(ref s) => s.flush(),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            Output::Syslog6Rfc3164(ref s) => s.flush(),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            Output::Syslog6Rfc5424(ref s) => s.flush(),
             Output::Panic(ref s) => s.flush(),
             Output::Writer(ref s) => s.flush(),
             #[cfg(feature = "date-based")]
@@ -592,7 +592,7 @@ impl Log for Sender {
     fn flush(&self) {}
 }
 
-#[cfg(any(feature = "syslog-3", feature = "syslog-4"))]
+#[cfg(any(feature = "syslog-3", feature = "syslog-6"))]
 macro_rules! send_syslog {
     ($logger:expr, $level:expr, $message:expr) => {
         use log::Level;
@@ -622,8 +622,8 @@ impl Log for Syslog3 {
     fn flush(&self) {}
 }
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-impl Log for Syslog4Rfc3164 {
+#[cfg(all(not(windows), feature = "syslog-6"))]
+impl Log for Syslog6Rfc3164 {
     fn enabled(&self, _: &log::Metadata) -> bool {
         true
     }
@@ -640,8 +640,8 @@ impl Log for Syslog4Rfc3164 {
     fn flush(&self) {}
 }
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-impl Log for Syslog4Rfc5424 {
+#[cfg(all(not(windows), feature = "syslog-6"))]
+impl Log for Syslog6Rfc5424 {
     fn enabled(&self, _: &log::Metadata) -> bool {
         true
     }
@@ -649,6 +649,7 @@ impl Log for Syslog4Rfc5424 {
     fn log(&self, record: &log::Record) {
         fallback_on_error(record, |record| {
             let transformed = (self.transform)(record);
+            let transformed = (transformed.0 as u32,transformed.1,transformed.2);
             let mut log = self.inner.lock().unwrap_or_else(|e| e.into_inner());
             send_syslog!(log, record.level(), transformed);
 
@@ -766,8 +767,8 @@ fn backup_logging(record: &log::Record,
 enum LogError {
     Io(io::Error),
     Send(mpsc::SendError<String>),
-    #[cfg(all(not(windows), feature = "syslog-4"))]
-    Syslog4(syslog4::Error),
+    #[cfg(all(not(windows), feature = "syslog-6"))]
+    Syslog6(syslog6::Error),
 }
 
 impl fmt::Display for LogError {
@@ -775,8 +776,8 @@ impl fmt::Display for LogError {
         match *self {
             LogError::Io(ref e) => write!(f, "{}", e),
             LogError::Send(ref e) => write!(f, "{}", e),
-            #[cfg(all(not(windows), feature = "syslog-4"))]
-            LogError::Syslog4(ref e) => write!(f, "{}", e),
+            #[cfg(all(not(windows), feature = "syslog-6"))]
+            LogError::Syslog6(ref e) => write!(f, "{}", e),
         }
     }
 }
@@ -793,10 +794,10 @@ impl From<mpsc::SendError<String>> for L
     }
 }
 
-#[cfg(all(not(windows), feature = "syslog-4"))]
-impl From<syslog4::Error> for LogError {
-    fn from(error: syslog4::Error) -> Self {
-        LogError::Syslog4(error)
+#[cfg(all(not(windows), feature = "syslog-6"))]
+impl From<syslog6::Error> for LogError {
+    fn from(error: syslog6::Error) -> Self {
+        LogError::Syslog6(error)
     }
 }
 
Index: fern/src/syslog.rs
===================================================================
--- fern.orig/src/syslog.rs
+++ fern/src/syslog.rs
@@ -5,14 +5,14 @@ Be sure to depend on `syslog` and the `s
 
 ```toml
 [dependencies]
-fern = { version = "0.5", features = ["syslog-4"] }]
-syslog = "4"
+fern = { version = "0.5", features = ["syslog-6"] }]
+syslog = "6"
 ```
 
 To use `syslog`, simply create the log you want, and pass it into `Dispatch::chain`:
 
 ```no_run
-# use syslog4 as syslog;
+# use syslog6 as syslog;
 # fn setup_logging() -> Result<(), Box<dyn std::error::Error>> {
 let formatter = syslog::Formatter3164 {
     facility: syslog::Facility::LOG_USER,
@@ -71,7 +71,7 @@ However, you probably will want to forma
 configuration is easy with fern:
 
 ```no_run
-# use syslog4 as syslog;
+# use syslog6 as syslog;
 # fn setup_logging() -> Result<(), Box<dyn std::error::Error>> {
 let syslog_formatter = syslog::Formatter3164 {
     facility: syslog::Facility::LOG_USER,
@@ -116,7 +116,7 @@ One last pattern you might want to know:
 in order to work.
 
 ```no_run
-# use syslog4 as syslog;
+# use syslog6 as syslog;
 # fn setup_logging() -> Result<(), Box<dyn std::error::Error>> {
 # let formatter = syslog::Formatter3164 {
 #     facility: syslog::Facility::LOG_USER,
