Index: fern/Cargo.toml
===================================================================
--- fern.orig/Cargo.toml
+++ fern/Cargo.toml
@@ -80,10 +80,6 @@ required-features = ["syslog-6"]
 
 
 
-[[example]]
-name = "syslog7"
-path = "examples/syslog7.rs"
-required-features = ["syslog-7"]
 
 [[test]]
 name = "channel_logging"
@@ -151,7 +147,6 @@ version = "3"
 date-based = ["chrono"]
 meta-logging-in-format = []
 syslog-6 = ["syslog6"]
-syslog-7 = ["syslog7"]
 
 [target."cfg(not(windows))".dependencies.libc]
 version = "0.2.58"
@@ -163,8 +158,3 @@ optional = true
 version = "6"
 optional = true
 package = "syslog"
-
-[target."cfg(not(windows))".dependencies.syslog7]
-version = "7"
-optional = true
-package = "syslog"
