Description: only syslog 6 is available
Author: Ed Neville <ed-debian@s5h.net>
Author: Peter Michael Green <plugwash@debian.org>
Origin: other
Forwarded: not-needed
Last-Update: 2023-11-28

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,6 +36,7 @@
 categories = ["development-tools::debugging"]
 license = "MIT"
 repository = "https://github.com/daboross/fern"
+autoexamples = false
 
 [package.metadata.docs.rs]
 all-features = true
@@ -60,14 +61,6 @@
 required-features = ["syslog-6"]
 
 [[example]]
-name = "syslog4"
-required-features = ["syslog-4"]
-
-[[example]]
-name = "syslog3"
-required-features = ["syslog-3"]
-
-[[example]]
 name = "meta-logging"
 
 [dependencies.chrono]
@@ -107,8 +100,6 @@
     "reopen1",
     "libc",
 ]
-syslog-3 = ["syslog3"]
-syslog-4 = ["syslog4"]
 syslog-6 = ["syslog6"]
 
 [target."cfg(not(windows))".dependencies.libc]
@@ -126,16 +117,6 @@
 optional = true
 package = "reopen"
 
-[target."cfg(not(windows))".dependencies.syslog3]
-version = "3"
-optional = true
-package = "syslog"
-
-[target."cfg(not(windows))".dependencies.syslog4]
-version = "4"
-optional = true
-package = "syslog"
-
 [target."cfg(not(windows))".dependencies.syslog6]
 version = "6"
 optional = true
