Description: only syslog 6 is available
Author: Ed Neville <ed-debian@s5h.net>
Author: Peter Michael Green <plugwash@debian.org>
Origin: other
Forwarded: not-needed
Last-Update: 2023-11-28

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,4 +61,0 @@
-[[example]]
-name = "syslog3"
-path = "examples/syslog3.rs"
-required-features = ["syslog-3"]
@@ -70,4 +70,0 @@
-[[example]]
-name = "syslog4"
-path = "examples/syslog4.rs"
-required-features = ["syslog-4"]
@@ -107,2 +100,0 @@
-syslog-3 = ["syslog3"]
-syslog-4 = ["syslog4"]
@@ -126,16 +117,6 @@
 optional = true
 package = "reopen"
 
-[target."cfg(not(windows))".dependencies.syslog3]
-version = "3"
-optional = true
-package = "syslog"
-
-[target."cfg(not(windows))".dependencies.syslog4]
-version = "4"
-optional = true
-package = "syslog"
-
 [target."cfg(not(windows))".dependencies.syslog6]
 version = "6"
 optional = true
