Description: Relax dependency on PyO3
Author: Jelmer Vernooij <jelmer@debian.org>
Forwarded: not-needed
Last-Update: 2023-04-19
Index: pyo3-file/Cargo.toml
===================================================================
--- pyo3-file.orig/Cargo.toml
+++ pyo3-file/Cargo.toml
@@ -21,7 +21,7 @@ license = "MIT/Apache-2.0"
 repository = "https://github.com/omerbenamram/pyo3-file"
 
 [dependencies.pyo3]
-version = "0.16.4"
+version = "0.18"
 
 [dev-dependencies.skeptic]
 version = "0.13.7"
Index: pyo3-file/src/lib.rs
===================================================================
--- pyo3-file.orig/src/lib.rs
+++ pyo3-file/src/lib.rs
@@ -23,7 +23,7 @@ impl PyFileLikeObject {
         let text_io = io.getattr("TextIOBase")?;
 
         let text_io_type = text_io.extract::<&PyType>()?;
-        let is_text_io = text_io_type.is_instance(&object)?;
+        let is_text_io = object.as_ref(py).is_instance(text_io_type)?;
 
         Ok(PyFileLikeObject {
             inner: object,
