Index: rust-fslock-0.2.1.new/src/test.rs
===================================================================
--- rust-fslock-0.2.1.new.orig/src/test.rs
+++ rust-fslock-0.2.1.new/src/test.rs
@@ -56,6 +58,7 @@ fn check_try_lock_example(
     use std::process::{Command, Stdio};
 
     let child = Command::new("cargo")
+        .arg("-Zavoid-dev-deps")
         .arg("run")
         .arg("-q")
         .arg("--example")
@@ -87,6 +90,7 @@ fn check_try_lock_with_pid_example(
     use std::process::{Command, Stdio};
 
     let child = Command::new("cargo")
+        .arg("-Zavoid-dev-deps")
         .arg("run")
         .arg("-q")
         .arg("--example")
