--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,9 +26,6 @@
 [target."cfg(unix)".dependencies.libc]
 version = "^0.2.66"
 default-features = false
-[target."cfg(windows)".dependencies.winapi]
-version = "^0.3.8"
-features = ["minwindef", "minwinbase", "winbase", "errhandlingapi", "winerror", "winnt", "synchapi", "handleapi", "fileapi", "processthreadsapi"]
 [badges.maintenance]
 status = "passively-maintained"
 
