--- z-base-32.orig/Cargo.toml
+++ z-base-32/Cargo.toml
@@ -28,2 +28,3 @@
 repository = "https://github.com/matusf/z-base-32"
+autobins = false
 
@@ -40,6 +41,2 @@
 
-[[bin]]
-name = "zbase32"
-required-features = ["cli"]
-
 [dependencies.anyhow]
@@ -48,7 +45,2 @@
 
-[dependencies.clap]
-version = "4.5.1"
-features = ["derive"]
-optional = true
-
 [dependencies.pyo3]
@@ -62,6 +54,2 @@
 [features]
-cli = [
-    "dep:clap",
-    "dep:anyhow",
-]
 python = ["dep:pyo3"]
