--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,12 +27,9 @@ license = "MIT"
 repository = "https://gitlab.freedesktop.org/dbus/zbus/"
 resolver = "2"
 
-[package.metadata.docs.rs]
-all-features = true
-targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies.async-broadcast]
-version = "0.5.0"
+version = "0.5"
 
 [dependencies.async-executor]
 version = "1.5.0"
@@ -43,7 +40,7 @@ version = "1.6.0"
 optional = true
 
 [dependencies.async-io]
-version = "1.12.0"
+version = "1.11.0"
 optional = true
 
 [dependencies.async-lock]
@@ -58,29 +55,29 @@ version = "4.3.0"
 optional = true
 
 [dependencies.async-trait]
-version = "0.1.58"
+version = "0.1.56"
 
 [dependencies.byteorder]
 version = "1.4.3"
 
 [dependencies.derivative]
-version = "2.2"
+version = "2.1"
 
 [dependencies.enumflags2]
-version = "0.7.5"
+version = "0.7.7"
 features = ["serde"]
 
 [dependencies.event-listener]
 version = "2.5.3"
 
 [dependencies.futures-core]
-version = "0.3.25"
+version = "0.3.21"
 
 [dependencies.futures-sink]
-version = "0.3.25"
+version = "0.3.21"
 
 [dependencies.futures-util]
-version = "0.3.25"
+version = "0.3.21"
 features = [
     "sink",
     "std",
@@ -97,7 +94,7 @@ version = "1.4.0"
 version = "0.2"
 
 [dependencies.quick-xml]
-version = "0.27.1"
+version = "0.26"
 features = [
     "serialize",
     "overlapped-lists",
@@ -112,14 +109,14 @@ version = "1.0"
 features = ["derive"]
 
 [dependencies.serde-xml-rs]
-version = "0.4.1"
+version = "0.5"
 optional = true
 
 [dependencies.serde_repr]
-version = "0.1.9"
+version = "0.1.7"
 
 [dependencies.sha1]
-version = "0.10.5"
+version = "0.10.1"
 features = ["std"]
 
 [dependencies.static_assertions]
@@ -138,17 +135,9 @@ features = [
 ]
 optional = true
 
-[dependencies.tokio-vsock]
-version = "0.3.3"
-optional = true
-
 [dependencies.tracing]
 version = "0.1.37"
 
-[dependencies.vsock]
-version = "0.3.0"
-optional = true
-
 [dependencies.xdg-home]
 version = "1.0.0"
 
@@ -167,23 +156,15 @@ default-features = false
 version = "1.12.0"
 features = ["attributes"]
 
-[dev-dependencies.doc-comment]
-version = "0.3.3"
-
 [dev-dependencies.futures-util]
-version = "0.3.25"
+version = "0.3.21"
 
 [dev-dependencies.ntest]
-version = "0.9.0"
+version = "0.8.1"
 
 [dev-dependencies.tempfile]
 version = "3.3.0"
 
-[dev-dependencies.test-log]
-version = "0.2.11"
-features = ["trace"]
-default-features = false
-
 [dev-dependencies.tokio]
 version = "1"
 features = [
@@ -215,45 +196,17 @@ async-io = [
 ]
 chrono = ["zvariant/chrono"]
 default = ["async-io"]
-gvariant = ["zvariant/gvariant"]
 time = ["zvariant/time"]
 tokio = ["dep:tokio"]
-tokio-vsock = [
-    "dep:tokio-vsock",
-    "tokio",
-]
 url = ["zvariant/url"]
 uuid = ["zvariant/uuid"]
-vsock = [
-    "dep:vsock",
-    "dep:async-io",
-]
-windows-gdbus = []
 xml = ["serde-xml-rs"]
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.26.0"
+version = "0.26"
 features = [
     "socket",
     "uio",
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.uds_windows]
-version = "1.0.2"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "handleapi",
-    "iphlpapi",
-    "memoryapi",
-    "processthreadsapi",
-    "sddl",
-    "securitybaseapi",
-    "synchapi",
-    "tcpmib",
-    "winbase",
-    "winerror",
-    "winsock2",
-]
