--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,12 +27,9 @@ license = "MIT"
 repository = "https://github.com/dbus2/zbus/"
 resolver = "2"
 
-[package.metadata.docs.rs]
-all-features = true
-targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies.async-broadcast]
-version = "0.5.0"
+version = "0.5"
 
 [dependencies.async-executor]
 version = "1.5.0"
@@ -116,7 +113,7 @@ version = "1.0"
 features = ["derive"]
 
 [dependencies.serde-xml-rs]
-version = "0.4.1"
+version = "0.5"
 optional = true
 
 [dependencies.serde_repr]
@@ -143,17 +140,9 @@ features = [
 ]
 optional = true
 
-[dependencies.tokio-vsock]
-version = "0.3.3"
-optional = true
-
 [dependencies.tracing]
 version = "0.1.37"
 
-[dependencies.vsock]
-version = "0.3.0"
-optional = true
-
 [dependencies.xdg-home]
 version = "1.0.0"
 
@@ -179,7 +168,7 @@ version = "0.3.3"
 version = "0.3.25"
 
 [dev-dependencies.ntest]
-version = "0.9.0"
+version = "0.8.1"
 
 [dev-dependencies.tempfile]
 version = "3.3.0"
@@ -224,45 +213,18 @@ default = ["async-io"]
 gvariant = ["zvariant/gvariant"]
 time = ["zvariant/time"]
 tokio = ["dep:tokio"]
-tokio-vsock = [
-    "dep:tokio-vsock",
-    "tokio",
-]
 url = ["zvariant/url"]
 uuid = ["zvariant/uuid"]
-vsock = [
-    "dep:vsock",
-    "dep:async-io",
-]
-windows-gdbus = []
 xml = ["serde-xml-rs"]
 
 [target."cfg(target_os = \"macos\")".dependencies.async-process]
 version = "1.7.0"
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.26.0"
+version = "0.26"
 features = [
     "socket",
     "uio",
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.uds_windows]
-version = "1.0.2"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "handleapi",
-    "iphlpapi",
-    "memoryapi",
-    "processthreadsapi",
-    "sddl",
-    "securitybaseapi",
-    "synchapi",
-    "tcpmib",
-    "winbase",
-    "winerror",
-    "winsock2",
-]
