--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,3 +56,3 @@
 [dependencies.async-trait]
-version = "0.1.80"
+version = "0.1.77"
 
@@ -97,6 +97,6 @@
 [dependencies.serde_repr]
-version = "0.1.19"
+version = "0.1.12"
 
 [dependencies.sha1]
-version = "0.10.6"
+version = "0.10.5"
 features = ["std"]
@@ -107,3 +107,3 @@
 [dependencies.tokio]
-version = "1.37.0"
+version = "1.35"
 features = [
@@ -128,3 +128,3 @@
 [dependencies.vsock]
-version = "0.5.0"
+version = "0.4.0"
 optional = true
@@ -158,3 +158,3 @@
 [dev-dependencies.test-log]
-version = "0.2.16"
+version = "0.2.15"
 features = ["trace"]
@@ -163,3 +163,3 @@
 [dev-dependencies.tokio]
-version = "1.37.0"
+version = "1.35.0"
 features = [
@@ -213,7 +211,2 @@
 
-[target."cfg(any(target_os = \"macos\", windows))".dependencies.async-recursion]
-version = "1.1.1"
-
-[target."cfg(target_os = \"macos\")".dependencies.async-process]
-version = "2.2.2"
 
