Index: zbus/Cargo.toml
===================================================================
--- zbus.orig/Cargo.toml
+++ zbus/Cargo.toml
@@ -47,7 +47,7 @@ version = "2"
 optional = true
 
 [dependencies.async-lock]
-version = "2.6.0"
+version = "2.8"
 optional = true
 
 [dependencies.async-recursion]
@@ -61,7 +61,7 @@ optional = true
 version = "0.1.58"
 
 [dependencies.blocking]
-version = "1.0.2"
+version = "1.3"
 optional = true
 
 [dependencies.byteorder]
@@ -95,13 +95,13 @@ default-features = false
 version = "0.4.3"
 
 [dependencies.once_cell]
-version = "1.4.0"
+version = "1.18"
 
 [dependencies.ordered-stream]
 version = "0.2"
 
 [dependencies.quick-xml]
-version = "0.27.1"
+version = ">= 0.27.1"
 features = [
     "serialize",
     "overlapped-lists",
@@ -116,7 +116,7 @@ version = "1.0"
 features = ["derive"]
 
 [dependencies.serde-xml-rs]
-version = "0.4.1"
+version = "0.5"
 optional = true
 
 [dependencies.serde_repr]
@@ -130,7 +130,7 @@ features = ["std"]
 version = "1.1.0"
 
 [dependencies.tokio]
-version = "1.21.2"
+version = "1.29"
 features = [
     "rt",
     "net",
@@ -151,7 +151,7 @@ optional = true
 version = "0.1.37"
 
 [dependencies.vsock]
-version = "0.3.0"
+version = "0.4"
 optional = true
 
 [dependencies.xdg-home]
@@ -176,10 +176,10 @@ features = ["attributes"]
 version = "0.3.3"
 
 [dev-dependencies.futures-util]
-version = "0.3.25"
+version = "0.3"
 
 [dev-dependencies.ntest]
-version = "0.9.0"
+version = "0.8"
 
 [dev-dependencies.tempfile]
 version = "3.3.0"
@@ -196,9 +196,8 @@ features = [
 ]
 
 [dev-dependencies.tracing-subscriber]
-version = "0.3.16"
+version = "0.3"
 features = [
-    "env-filter",
     "fmt",
     "ansi",
 ]
@@ -229,11 +228,9 @@ vsock = [
     "dep:vsock",
     "dep:async-io",
 ]
-windows-gdbus = []
+
 xml = ["serde-xml-rs"]
 
-[target."cfg(target_os = \"macos\")".dependencies.async-process]
-version = "1.7.0"
 
 [target."cfg(unix)".dependencies.nix]
 version = "0.27.0"
@@ -243,22 +240,3 @@ features = [
     "user",
 ]
 default-features = false
-
-[target."cfg(windows)".dependencies.uds_windows]
-version = "1.0.2"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "handleapi",
-    "iphlpapi",
-    "memoryapi",
-    "processthreadsapi",
-    "sddl",
-    "securitybaseapi",
-    "synchapi",
-    "tcpmib",
-    "winbase",
-    "winerror",
-    "winsock2",
-]
