--- a/Cargo.toml
+++ b/Cargo.toml
@@ -228,8 +228,6 @@
 ]
 default-features = false
 
-[dev-dependencies.zbus_xml]
-version = "5.0.0"
 
 [features]
 async-io = [

--- a/tests/e2e.rs
+++ b/tests/e2e.rs
@@ -505,7 +505,7 @@
     assert_eq!(**address.get("address").unwrap(), Value::from("127.0.0.1"));
     assert_eq!(**address.get("prefix").unwrap(), Value::from(1234u32));
 }
-
+/*
 #[instrument]
 async fn my_iface_test(conn: Connection, event: Event) -> zbus::Result<u32> {
     debug!("client side starting..");
@@ -871,7 +871,7 @@
 
     proxy.quit().await?;
     Ok(val)
-}
+}*/
 /*
 #[test]
 #[timeout(15000)]
@@ -887,6 +887,7 @@
     block_on(iface_and_proxy_(true));
 }*/
 
+/*
 #[instrument]
 async fn iface_and_proxy_(#[allow(unused)] p2p: bool) {
     let event = event_listener::Event::new();
@@ -1092,4 +1093,4 @@
         Ok(false)
     );
     debug!("Bus confirmed that all names were definitely released.");
-}
+}*/

