Origin: vendor
Subject: Relax dependencies to work well with other rust crates in Debian
Forwarded: not-needed

Index: sequoia-keystore-openpgp-card/Cargo.toml
===================================================================
--- sequoia-keystore-openpgp-card.orig/Cargo.toml
+++ sequoia-keystore-openpgp-card/Cargo.toml
@@ -84,7 +84,7 @@ version = ">=0.10, <0.12"
 
 [dev-dependencies.sequoia-openpgp]
 version = "2"
-features = ["compression"]
+features = ["compression", "crypto-nettle"]
 default-features = false
 
 [dev-dependencies.serde]
@@ -110,11 +110,3 @@ features = [
     "__implicit-crypto-backend-for-tests",
 ]
 default-features = false
-
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "2"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
