Index: quick-xml/Cargo.toml
===================================================================
--- quick-xml.orig/Cargo.toml
+++ quick-xml/Cargo.toml
@@ -112,9 +112,6 @@ version = "1"
 version = "1.0"
 features = ["derive"]
 
-[dev-dependencies.serde-value]
-version = "0.7"
-
 [dev-dependencies.tokio]
 version = "1.20"
 features = [
Index: quick-xml/tests/serde-de.rs
===================================================================
--- quick-xml.orig/tests/serde-de.rs
+++ quick-xml/tests/serde-de.rs
@@ -44,7 +44,7 @@ fn string_borrow() {
 }
 
 /// Test for https://github.com/tafia/quick-xml/issues/231
-#[test]
+/*#[test]
 fn implicit_value() {
     use serde_value::Value;
 
@@ -61,7 +61,7 @@ fn implicit_value() {
             .collect()
         )
     );
-}
+}*/
 
 #[test]
 fn explicit_value() {
