Index: trust-dns-resolver/Cargo.toml
===================================================================
--- trust-dns-resolver.orig/Cargo.toml
+++ trust-dns-resolver/Cargo.toml
@@ -145,11 +145,6 @@ dns-over-openssl = [
     "trust-dns-proto/dns-over-openssl",
     "tokio-openssl",
 ]
-dns-over-quic = [
-    "rustls/quic",
-    "dns-over-rustls",
-    "trust-dns-proto/dns-over-quic",
-]
 dns-over-rustls = [
     "dns-over-tls",
     "rustls",
