--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,11 +51,6 @@
 [dependencies.parking_lot]
 version = "0.12"
 
-[dependencies.resolv-conf]
-version = "0.7.0"
-features = ["system"]
-optional = true
-
 [dependencies.rustls]
 version = "0.20.0"
 optional = true
@@ -104,7 +99,7 @@
 
 [features]
 backtrace = ["trust-dns-proto/backtrace"]
-default = ["system-config", "tokio-runtime"]
+default = ["tokio-runtime"]
 dns-over-https = ["trust-dns-proto/dns-over-https"]
 dns-over-https-rustls = ["trust-dns-proto/dns-over-https-rustls", "dns-over-rustls", "dns-over-https"]
 dns-over-native-tls = ["dns-over-tls", "tokio-native-tls", "trust-dns-proto/dns-over-native-tls"]
@@ -115,7 +110,6 @@
 dnssec-openssl = ["dnssec", "trust-dns-proto/dnssec-openssl"]
 dnssec-ring = ["dnssec", "trust-dns-proto/dnssec-ring"]
 serde-config = ["serde", "trust-dns-proto/serde-config"]
-system-config = ["ipconfig", "resolv-conf"]
 testing = []
 tokio-runtime = ["tokio/rt", "trust-dns-proto/tokio-runtime"]
 [target."cfg(windows)".dependencies.ipconfig]
