--- a/tests/main.rs
+++ b/tests/main.rs
@@ -1,5 +1,5 @@
-mod core;
-mod derive;
-mod macros;
-mod settings;
+//mod core;
+//mod derive;
+//mod macros;
+//mod settings;
 
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -422,3 +422,3 @@
 
-[dev-dependencies.testing_table]
+[disabled.dev-dependencies.testing_table]
 version = "0.2"
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -17,4 +17,4 @@
 //!
-#![cfg_attr(all(feature = "derive", feature = "std"), doc = "```")]
-#![cfg_attr(not(all(feature = "derive", feature = "std")), doc = "```ignore")]
+//#![cfg_attr(all(feature = "derive", feature = "std"), doc = "```")]
+#![/*cfg_attr(not(all(feature = "derive", feature = "std")), */doc = "```ignore"/*)*/]
 //! use tabled::{Tabled, Table};
@@ -60,4 +60,4 @@
 //!
-#![cfg_attr(feature = "std", doc = "```")]
-#![cfg_attr(not(feature = "std"), doc = "```ignore")]
+//#![cfg_attr(feature = "std", doc = "```")]
+#![/*cfg_attr(not(feature = "std"), */doc = "```ignore"/*)*/]
 //! use std::iter::once;
@@ -103,4 +103,4 @@
 //!
-#![cfg_attr(feature = "std", doc = "```")]
-#![cfg_attr(not(feature = "std"), doc = "```ignore")]
+//#![cfg_attr(feature = "std", doc = "```")]
+#![/*cfg_attr(not(feature = "std"), */doc = "```ignore"/*)*/]
 //! use tabled::Table;
@@ -127,4 +127,4 @@
 //!
-#![cfg_attr(feature = "std", doc = "```")]
-#![cfg_attr(not(feature = "std"), doc = "```ignore")]
+//#![cfg_attr(feature = "std", doc = "```")]
+#![/*cfg_attr(not(feature = "std"), */doc = "```ignore"/*)*/]
 //! use tabled::Table;
@@ -165,4 +165,4 @@
 //!
-#![cfg_attr(all(feature = "derive", feature = "std"), doc = "```")]
-#![cfg_attr(not(all(feature = "derive", feature = "std")), doc = "```ignore")]
+//#![cfg_attr(all(feature = "derive", feature = "std"), doc = "```")]
+#![/*cfg_attr(not(all(feature = "derive", feature = "std")), */doc = "```ignore"/*)*/]
 //! use tabled::{Tabled, Table};
@@ -190,4 +190,4 @@
 //!
-#![cfg_attr(all(feature = "macros", feature = "std"), doc = "```")]
-#![cfg_attr(not(all(feature = "macros", feature = "std")), doc = "```ignore")]
+//#![cfg_attr(all(feature = "macros", feature = "std"), doc = "```")]
+#![/*cfg_attr(not(all(feature = "macros", feature = "std")), */doc = "```ignore"/*)*/]
 //! use tabled::{row, col};
--- a/src/settings/object/mod.rs
+++ b/src/settings/object/mod.rs
@@ -5,3 +5,3 @@
 //!
-//! ```
+//! ```ignore
 //! use tabled::{
--- a/src/settings/peaker/mod.rs
+++ b/src/settings/peaker/mod.rs
@@ -32,3 +32,3 @@
 ///
-/// ```
+/// ```ignore
 /// # use tabled::{Table, settings::{Style, peaker::Priority, Width}};
@@ -73,3 +73,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// # use tabled::{Table, settings::{Style, peaker::Priority, Width}};
@@ -113,3 +113,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// # use tabled::{Table, settings::{Style, peaker::Priority, Width}};
@@ -146,3 +146,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// # use tabled::{Table, settings::{Style, peaker::Priority, Width}};
@@ -185,3 +185,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// # use tabled::{Table, settings::{Style, peaker::Priority, Width}};
@@ -228,3 +228,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// # use tabled::{Table, settings::{Style, peaker::Priority, Width}};
--- a/src/tables/table.rs
+++ b/src/tables/table.rs
@@ -100,3 +100,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// use tabled::{Table, Tabled};
@@ -130,3 +130,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// use tabled::{Table, Tabled};
@@ -160,3 +160,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// use tabled::Table;
@@ -185,3 +185,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// use std::iter::FromIterator;
@@ -244,3 +244,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// use tabled::{Table, Tabled};
@@ -301,3 +301,3 @@
     ///
-    /// ```
+    /// ```ignore
     /// use tabled::{Table, Tabled, settings::Style};
--- a/src/derive/display/mod.rs
+++ b/src/derive/display/mod.rs
@@ -12,3 +12,3 @@
 ///
-/// ```
+/// ```ignore
 /// use tabled::Tabled;
@@ -61,3 +61,3 @@
 ///
-/// ```
+/// ```ignore
 /// use tabled::Tabled;
@@ -106,3 +106,3 @@
 ///
-/// ```
+/// ```ignore
 /// use tabled::Tabled;
