--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,11 +57,6 @@
 path = "src/lib.rs"
 doctest = false
 
-[[bench]]
-name = "bench_main"
-path = "benches/bench_main.rs"
-harness = false
-
 [dependencies.anyhow]
 version = "1.0.98"
 
@@ -95,11 +90,11 @@
 optional = true
 
 [dependencies.tinyvec]
-version = "1.9.0"
+version = "1.6.0"
 features = ["rustc_1_55"]
 
 [dev-dependencies.criterion]
-version = "0.6.0"
+version = "0.5.1"
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
