--- rust-libbpf-rs-0.26.0~beta.1.orig/Cargo.toml
+++ rust-libbpf-rs-0.26.0~beta.1/Cargo.toml
@@ -40,14 +40,11 @@ repository = "https://github.com/libbpf/
 status = "actively-developed"
 
 [features]
-default = ["libbpf-sys/vendored-libbpf"]
+default = []
 dont-generate-test-files = []
 generate-test-files = [
-    "libbpf-sys/vendored-libbpf",
     "dep:tempfile",
 ]
-static = ["libbpf-sys/static"]
-vendored = ["libbpf-sys/vendored"]
 
 [lib]
 name = "libbpf_rs"
