Index: clap-complete-nushell/Cargo.toml
===================================================================
--- clap-complete-nushell.orig/Cargo.toml
+++ clap-complete-nushell/Cargo.toml
@@ -81,10 +81,6 @@ exactly = 1
 
 [features]
 default = []
-unstable-shell-tests = [
-    "dep:completest",
-    "dep:completest-nu",
-]
 
 [lib]
 name = "clap_complete_nushell"
@@ -111,14 +107,6 @@ default-features = false
 [dependencies.clap_complete]
 version = "4.5.51"
 
-[dependencies.completest]
-version = "0.4.0"
-optional = true
-
-[dependencies.completest-nu]
-version = "0.4.0"
-optional = true
-
 [dev-dependencies.clap]
 version = "4.0.0"
 features = [
