Index: log/Cargo.toml
===================================================================
--- log.orig/Cargo.toml
+++ log/Cargo.toml
@@ -34,15 +34,15 @@ features = [
     "kv_unstable_serde",
 ]
 
-[[test]]
-name = "filters"
-path = "tests/filters.rs"
-harness = false
+# [[test]]
+# name = "filters"
+# path = "tests/filters.rs"
+# harness = false
 
-[[test]]
-name = "macros"
-path = "tests/macros.rs"
-harness = true
+# [[test]]
+# name = "macros"
+# path = "tests/macros.rs"
+# harness = true
 
 [dependencies.serde]
 version = "1.0"
