Forwarded: not-needed
Last-Update: 2025-04-01
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -115 +115 @@
-[dependencies.atomic_float]
+[disabled.dependencies.atomic_float]
@@ -119 +119 @@
-[dependencies.claxon]
+[disabled.dependencies.claxon]
@@ -130 +130 @@
-[dependencies.hound]
+[disabled.dependencies.hound]
@@ -138 +138 @@
-[dependencies.minimp3_fixed]
+[disabled.dependencies.minimp3_fixed]
@@ -171 +171 @@
-[dev-dependencies.rstest_reuse]
+[disabled.dev-dependencies.rstest_reuse]
@@ -175 +175 @@
-cpal-shared-stdcxx = ["cpal/oboe-shared-stdcxx"]
+#cpal-shared-stdcxx = ["cpal/oboe-shared-stdcxx"]
@@ -177 +177 @@
-    "flac",
+    #"flac",
@@ -179 +179 @@
-    "wav",
+    #"wav",
@@ -182,3 +182,3 @@
-experimental = ["dep:atomic_float"]
-flac = ["claxon"]
-minimp3 = ["dep:minimp3_fixed"]
+#experimental = ["dep:atomic_float"]
+#flac = ["claxon"]
+#minimp3 = ["dep:minimp3_fixed"]
@@ -188,4 +188,4 @@
-symphonia-aiff = [
-    "symphonia/aiff",
-    "symphonia/pcm",
-]
+#symphonia-aiff = [
+#    "symphonia/aiff",
+#    "symphonia/pcm",
+#]
@@ -215,2 +215,2 @@
-wasm-bindgen = ["cpal/wasm-bindgen"]
-wav = ["hound"]
+#wasm-bindgen = ["cpal/wasm-bindgen"]
+#wav = ["hound"]
