Index: thread-id/Cargo.toml
===================================================================
--- thread-id.orig/Cargo.toml
+++ thread-id/Cargo.toml
@@ -20,8 +20,8 @@ readme = "readme.md"
 keywords = ["thread", "pthread", "getcurrentthreadid"]
 license = "MIT/Apache-2.0"
 repository = "https://github.com/ruuda/thread-id"
-[target."cfg(target_os = \"redox\")".dependencies.redox_syscall]
-version = "0.2"
+#[target."cfg(target_os = \"redox\")".dependencies.redox_syscall]
+#version = "0.1"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.6"
 [target."cfg(windows)".dependencies.winapi]
