Index: thread-id/Cargo.toml
===================================================================
--- thread-id.orig/Cargo.toml
+++ thread-id/Cargo.toml
@@ -20,7 +20,7 @@ keywords = ["thread", "pthread", "getcur
 license = "MIT/Apache-2.0"
 repository = "https://github.com/ruuda/thread-id"
 [target."cfg(target_os = \"redox\")".dependencies.redox_syscall]
-version = "0.1"
+version = ">= 0.1, < 0.3"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.6"
 [target."cfg(windows)".dependencies.winapi]
