--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,7 +44,7 @@
 harness = false
 
 [dependencies.arrayvec]
-version = "0.5"
+version = "0.7"
 
 [dependencies.termcolor]
 version = "1.1.0"
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -196,7 +196,7 @@
     }
 }
 
-pub type SmallText = arrayvec::ArrayString<[u8; 22]>;
+pub type SmallText = arrayvec::ArrayString<22>;
 
 fn append_docs<'a, 'd, T, A>(
     mut doc: &'d Doc<'a, T, A>,
