--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,20 +28,12 @@
 ]
 license = "MIT"
 repository = "https://github.com/Marwes/pretty.rs"
+autobenches = false
+autoexamples = false
 
 [package.metadata.docs.rs]
 features = ["termcolor"]
 
-[[example]]
-name = "trees"
-
-[[example]]
-name = "colored"
-required-features = ["termcolor"]
-
-[[bench]]
-name = "trees"
-harness = false
 
 [dependencies.arrayvec]
 version = "0.7"
@@ -56,9 +48,6 @@
 [dependencies.unicode-width]
 version = "0.1"
 
-[dev-dependencies.criterion]
-version = "0.4"
-
 [dev-dependencies.difference]
 version = "2"
 
