Index: lscolors/Cargo.toml
===================================================================
--- lscolors.orig/Cargo.toml
+++ lscolors/Cargo.toml
@@ -43,9 +43,9 @@ required-features = ["nu-ansi-term"]
 version = ">= 0.11, < 0.13"
 optional = true
 
-[dependencies.crossterm]
-version = "0.26"
-optional = true
+#[dependencies.crossterm]
+#version = "0.26"
+#optional = true
 
 [dependencies.nu-ansi-term]
 version = "0.47"
