Index: lscolors/src/style.rs
===================================================================
--- lscolors.orig/src/style.rs
+++ lscolors/src/style.rs
@@ -397,6 +397,7 @@ impl Style {
             is_reverse: self.font_style.reverse,
             is_hidden: self.font_style.hidden,
             is_strikethrough: self.font_style.strikethrough,
+            with_reset: false,
         }
     }
 
Index: lscolors/Cargo.toml
===================================================================
--- lscolors.orig/Cargo.toml
+++ lscolors/Cargo.toml
@@ -47,7 +47,7 @@ optional = true
 #optional = true
 
 [dependencies.nu-ansi-term]
-version = "0.49"
+version = ">= 0.48, < 1.0"
 optional = true
 
 [dev-dependencies.tempfile]
