--- ratatui.orig/Cargo.toml
+++ ratatui/Cargo.toml
@@ -169,2 +169,6 @@
 
+#! The crate provides a set of optional features that can be enabled in your `cargo.toml` file.
+#!
+#! Generally an application will only use one backend, so you should only enable one of the following features:
+## enables the [`CrosstermBackend`] backend and adds a dependency on the [Crossterm crate].
 [dependencies.crossterm]
@@ -195,2 +200,3 @@
 
+## enables the [`TermionBackend`] backend and adds a dependency on the [Termion crate].
 [dependencies.termion]
@@ -234,2 +241,3 @@
 [features]
+## enables all widgets.
 all-widgets = ["widget-calendar"]
@@ -250,3 +250,7 @@
 ]
+## enables the [`border!`] macro.
 macros = []
+#! The following optional features are available for all backends:
+## enables serialization and deserialization of style and color types using the [Serde crate].
+## This is useful if you want to save themes to a file.
 serde = [
@@ -260,2 +260,5 @@
 unstable-segment-size = []
+#! Widgets that add dependencies are gated behind feature flags to prevent unused transitive
+#! dependencies. The available features are:
+## enables the [`calendar`] widget module and adds a dependency on the [Time crate].
 widget-calendar = ["dep:time"]
