Index: ratatui/Cargo.toml
===================================================================
--- ratatui.orig/Cargo.toml
+++ ratatui/Cargo.toml
@@ -222,9 +222,6 @@ version = "1"
 features = ["derive"]
 optional = true
 
-[dependencies.stability]
-version = "0.1.1"
-
 [dependencies.strum]
 version = "0.26"
 features = ["derive"]
Index: ratatui/src/layout.rs
===================================================================
--- ratatui.orig/src/layout.rs
+++ ratatui/src/layout.rs
@@ -220,11 +220,6 @@ pub enum Direction {
 /// #536](https://github.com/ratatui-org/ratatui/issues/536) for more information.
 ///
 /// [`Table`]: crate::widgets::Table
-#[stability::unstable(
-    feature = "segment-size",
-    reason = "The name for this feature is not final and may change in the future",
-    issue = "https://github.com/ratatui-org/ratatui/issues/536"
-)]
 #[derive(Copy, Debug, Default, Display, EnumString, Clone, Eq, PartialEq, Hash)]
 pub enum SegmentSize {
     /// prefer equal chunks if other constraints are all satisfied
@@ -448,11 +443,6 @@ impl Layout {
     /// #536](https://github.com/ratatui-org/ratatui/issues/536) for more information.
     ///
     /// [`Table`]: crate::widgets::Table
-    #[stability::unstable(
-        feature = "segment-size",
-        reason = "The name for this feature is not final and may change in the future",
-        issue = "https://github.com/ratatui-org/ratatui/issues/536"
-    )]
     #[must_use = "method moves the value of self and returns the modified value"]
     pub const fn segment_size(mut self, segment_size: SegmentSize) -> Layout {
         self.segment_size = segment_size;
Index: ratatui/src/widgets/paragraph.rs
===================================================================
--- ratatui.orig/src/widgets/paragraph.rs
+++ ratatui/src/widgets/paragraph.rs
@@ -222,11 +222,6 @@ impl<'a> Paragraph<'a> {
     /// assert_eq!(paragraph.line_count(20), 1);
     /// assert_eq!(paragraph.line_count(10), 2);
     /// ```
-    #[stability::unstable(
-        feature = "rendered-line-info",
-        reason = "The design for text wrapping is not stable and might affect this API.",
-        issue = "https://github.com/ratatui-org/ratatui/issues/293"
-    )]
     pub fn line_count(&self, width: u16) -> usize {
         if width < 1 {
             return 0;
@@ -264,11 +259,6 @@ impl<'a> Paragraph<'a> {
     /// let paragraph = Paragraph::new("Hello World\nhi\nHello World!!!");
     /// assert_eq!(paragraph.line_width(), 14);
     /// ```
-    #[stability::unstable(
-        feature = "rendered-line-info",
-        reason = "The design for text wrapping is not stable and might affect this API.",
-        issue = "https://github.com/ratatui-org/ratatui/issues/293"
-    )]
     pub fn line_width(&self) -> usize {
         self.text
             .lines
Index: ratatui/src/widgets/table.rs
===================================================================
--- ratatui.orig/src/widgets/table.rs
+++ ratatui/src/widgets/table.rs
@@ -653,11 +653,6 @@ impl<'a> Table<'a> {
     /// let table = Table::new([], widths)
     ///     .segment_size(SegmentSize::LastTakesRemainder);
     /// ```
-    #[stability::unstable(
-        feature = "segment-size",
-        reason = "The name for this feature is not final and may change in the future",
-        issue = "https://github.com/ratatui-org/ratatui/issues/536"
-    )]
     pub const fn segment_size(mut self, segment_size: SegmentSize) -> Self {
         self.segment_size = segment_size;
         self
