Index: nom/Cargo.toml
===================================================================
--- nom.orig/Cargo.toml
+++ nom/Cargo.toml
@@ -54,35 +54,6 @@ lto = true
 codegen-units = 1
 debug = true
 
-[[example]]
-name = "custom_error"
-path = "examples/custom_error.rs"
-required-features = ["alloc"]
-
-[[example]]
-name = "json"
-path = "examples/json.rs"
-required-features = ["alloc"]
-
-[[example]]
-name = "json_iterator"
-path = "examples/json_iterator.rs"
-required-features = ["alloc"]
-
-[[example]]
-name = "iterator"
-path = "examples/iterator.rs"
-
-[[example]]
-name = "s_expression"
-path = "examples/s_expression.rs"
-required-features = ["alloc"]
-
-[[example]]
-name = "string"
-path = "examples/string.rs"
-required-features = ["alloc"]
-
 [[test]]
 name = "arithmetic"
 
