--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,6 +14,7 @@
 name = "dateparser"
 version = "0.2.1"
 authors = ["Rollie Ma <rollie@rollie.dev>"]
+autobenches = false
 description = "Parse dates in string formats that are commonly used"
 homepage = "https://github.com/waltzofpearls/dateparser"
 readme = "README.md"
@@ -27,10 +28,6 @@
 license = "MIT"
 repository = "https://github.com/waltzofpearls/dateparser"
 
-[[bench]]
-name = "parse"
-harness = false
-
 [dependencies.anyhow]
 version = "1.0.75"
 
@@ -45,7 +42,3 @@
 
 [dev-dependencies.chrono-tz]
 version = "0.8.4"
-
-[dev-dependencies.criterion]
-version = "0.5.1"
-features = ["html_reports"]
