diff --git a/Cargo.toml b/Cargo.toml
index ee5629d..b6b57fd 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,20 +33,20 @@ name = "parse"
 harness = false
 
 [dependencies.anyhow]
-version = "1.0.40"
+version = "1"
 
 [dependencies.chrono]
-version = "0.4.24"
+version = "0.4"
 
 [dependencies.lazy_static]
-version = "1.4.0"
+version = "1"
 
 [dependencies.regex]
-version = "1.6.0"
+version = "1"
 
 [dev-dependencies.chrono-tz]
-version = "0.6.3"
+version = "0.6"
 
 [dev-dependencies.criterion]
-version = "0.3.6"
+version = "0.3"
 features = ["html_reports"]
