Index: getrandom/tests/rdrand.rs
===================================================================
--- getrandom.orig/tests/rdrand.rs
+++ getrandom/tests/rdrand.rs
@@ -1,3 +1,4 @@
+/*
 // We only test the RDRAND-based RNG source on supported architectures.
 #![cfg(any(target_arch = "x86_64", target_arch = "x86"))]
 
@@ -12,4 +13,4 @@ mod rdrand;
 mod util;
 
 use rdrand::getrandom_inner as getrandom_impl;
-mod common;
+mod common;*/
