--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,10 +49,6 @@
 
 [features]
 custom = []
-js = [
-    "wasm-bindgen",
-    "js-sys",
-]
 rdrand = []
 rustc-dep-of-std = [
     "compiler_builtins",
@@ -62,20 +58,6 @@
 std = []
 test-in-browser = []
 
-[target."cfg(all(any(target_arch = \"wasm32\", target_arch = \"wasm64\"), target_os = \"unknown\"))".dependencies.js-sys]
-version = "0.3"
-optional = true
-
-[target."cfg(all(any(target_arch = \"wasm32\", target_arch = \"wasm64\"), target_os = \"unknown\"))".dependencies.wasm-bindgen]
-version = "0.2.62"
-optional = true
-default-features = false
-
-[target."cfg(all(any(target_arch = \"wasm32\", target_arch = \"wasm64\"), target_os = \"unknown\"))".dev-dependencies.wasm-bindgen-test]
-version = "0.3.18"
-
-default-features = false
-
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.143"
 default-features = false
