--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,19 +41,13 @@
 dummy = []
 rustc-dep-of-std = ["compiler_builtins", "core"]
 std = []
-test-in-browser = ["wasm-bindgen"]
-[target."cfg(target_os = \"wasi\")".dependencies.wasi]
-version = "0.7"
+test-in-browser = []
+stdweb = []
+wasm-bindgen = []
 [target."cfg(unix)".dependencies.libc]
-version = "0.2.64"
+version = "0.2.62"
 default-features = false
-[target.wasm32-unknown-unknown.dependencies.stdweb]
-version = "0.4.18"
-optional = true
 
-[target.wasm32-unknown-unknown.dependencies.wasm-bindgen]
-version = "0.2.29"
-optional = true
 [target.wasm32-unknown-unknown.dev-dependencies.wasm-bindgen-test]
 version = "0.2"
 [badges.appveyor]
