Index: rumqttc/Cargo.toml
===================================================================
--- rumqttc.orig/Cargo.toml
+++ rumqttc/Cargo.toml
@@ -162,7 +162,7 @@ default-features = false
 version = "1.5"
 
 [dependencies.fixedbitset]
-version = "0.5.7"
+version = ">= 0.4.2"
 
 [dependencies.flume]
 version = "0.11"
@@ -197,7 +197,7 @@ version = "2.2.0"
 optional = true
 
 [dependencies.rustls-webpki]
-version = "0.102.8"
+version = ">= 0.102.8"
 optional = true
 
 [dependencies.thiserror]
