Description:  Don't fail build on unsupported architectures.
Author: Peter Michael Green <plugwash@debian.org>

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -71,8 +71,8 @@
 #[cfg(miri)]
 mod miri;
 
-#[cfg(not(any(target_arch = "aarch64", target_arch = "x86", target_arch = "x86_64")))]
-compile_error!("This crate works only on `aarch64`, `x86`, and `x86-64` targets.");
+//#[cfg(not(any(target_arch = "aarch64", target_arch = "x86", target_arch = "x86_64")))]
+//compile_error!("This crate works only on `aarch64`, `x86`, and `x86-64` targets.");
 
 /// Create module with CPU feature detection code.
 #[macro_export]
