Index: rust-cryptoauthlib/Cargo.toml
===================================================================
--- rust-cryptoauthlib.orig/Cargo.toml
+++ rust-cryptoauthlib/Cargo.toml
@@ -35,13 +35,13 @@ features = ["serde"]
 version = "0.8.0"
 
 [dependencies.strum_macros]
-version = "0.21.1"
+version = "0.19.0"
 [dev-dependencies.serde]
 version = "1.0"
 features = ["derive"]
 
 [dev-dependencies.serial_test]
-version = "0.5.1"
+version = "0.9.0"
 
 [dev-dependencies.toml]
 version = "0.5.8"
