Index: hex/Cargo.toml
===================================================================
--- hex.orig/Cargo.toml
+++ hex/Cargo.toml
@@ -29,6 +29,8 @@ rustdoc-args = ["--cfg", "docsrs"]
 [[bench]]
 name = "hex"
 harness = false
+required-features = ["alloc"]
+
 [dependencies.serde]
 version = "1.0"
 optional = true
Index: hex/src/lib.rs
===================================================================
--- hex.orig/src/lib.rs
+++ hex/src/lib.rs
@@ -388,6 +388,9 @@ mod test {
     #[cfg(feature = "alloc")]
     use alloc::string::ToString;
     use pretty_assertions::assert_eq;
+    
+    #[cfg(feature = "alloc")]
+    use alloc::vec;
 
     #[test]
     #[cfg(feature = "alloc")]
