diff --git a/Cargo.toml b/Cargo.toml
index 7d6ef81..c0351d1 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,13 +34,13 @@ version = "1.0"
 optional = true
 default-features = false
 [dev-dependencies.criterion]
-version = "0.3"
+version = "0.5"
 
 [dev-dependencies.faster-hex]
 version = "0.10"
 
 [dev-dependencies.pretty_assertions]
-version = "0.6"
+version = "1.0"
 
 [dev-dependencies.rustc-hex]
 version = "2.1"
