diff --git a/Cargo.toml b/Cargo.toml
index 195cf66..38a7dc8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,15 +55,15 @@ path = "benches/semaphore.rs"
 harness = false
 
 [dependencies.loom]
-version = "0.7"
+version = "0.5"
 optional = true
 
 [dev-dependencies.criterion]
-version = "0.7"
+version = "0.5"
 
 [dev-dependencies.futures]
 version = "0.3"
 
 [dev-dependencies.tokio]
-version = "1.47"
+version = "1"
 features = ["full"]
