--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,9 +41,9 @@ name = "drawing"
 [[example]]
 name = "get_all_windows"
 
-[[example]]
-name = "opengl_window"
-required-features = ["glx", "xlib_xcb", "dri2"]
+#[[example]]
+#name = "opengl_window"
+#required-features = ["glx", "xlib_xcb", "dri2"]
 
 [[example]]
 name = "randr_crtc_info"
@@ -84,8 +84,8 @@ version = "0.2.102"
 version = "2.19.0"
 features = ["xlib"]
 optional = true
-[dev-dependencies.gl]
-version = "0.14.0"
+#[dev-dependencies.gl]
+#version = "0.14.0"
 
 [dev-dependencies.x11]
 version = "2.19.1"
