Index: stfu8/Cargo.toml
===================================================================
--- stfu8.orig/Cargo.toml
+++ stfu8/Cargo.toml
@@ -26,10 +26,10 @@ version = ">=1.0.0"
 [dependencies.regex]
 version = ">=0.2.0"
 [dev-dependencies.pretty_assertions]
-version = "0.4.1"
+version = ">= 0.4.1, < 0.7"
 
 [dev-dependencies.proptest]
-version = "0.3.4"
+version = ">= 0.3.4, < 0.10"
 
 [features]
 default = ["testing"]
