--- a/src/time.rs
+++ b/src/time.rs
@@ -17,2 +17,7 @@
 
+#[cfg(not(any(target_arch = "arm", target_arch = "powerpc")))]
+pub type TvUsecT = suseconds_t;
+#[cfg(any(target_arch = "arm", target_arch = "powerpc"))]
+pub type TvUsecT = i64;
+
 #[cfg(feature = "serde")]
@@ -22,5 +27,5 @@
 struct TimevalDef {
     tv_sec: time_t,
-    tv_usec: suseconds_t,
+    tv_usec: TvUsecT,
 }
 
@@ -31,3 +36,3 @@
             tv_sec: secs as time_t,
-            tv_usec: usecs as suseconds_t,
+            tv_usec: usecs as TvUsecT,
         })
@@ -60,3 +65,3 @@
     pub fn set_microseconds(&mut self, value: i64) {
-        (self.0).tv_usec = value as suseconds_t
+        (self.0).tv_usec = value as TvUsecT
     }
