--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,20 +47,13 @@
 version = "0.8"
 
 [dependencies.nix]
-version = "0.26"
+version = "0.27"
 
 [dependencies.serde]
 version = "1"
 features = ["derive"]
 optional = true
 
-[dependencies.tokio-util-0_6]
-version = "0.6"
-features = ["codec"]
-optional = true
-default-features = false
-package = "tokio-util"
-
 [dependencies.tokio-util-0_7]
 version = "0.7"
 features = ["codec"]
@@ -72,11 +65,6 @@
 codec = ["bytes"]
 dox = []
 serde = ["dep:serde"]
-tokio-util-0_6 = [
-    "dep:tokio-util-0_6",
-    "codec",
-    "bytes",
-]
 tokio-util-0_7 = [
     "dep:tokio-util-0_7",
     "codec",
