From 748d49ec4d93b5e1666c4450c6ee450577d9fab5 Mon Sep 17 00:00:00 2001
From: Junichi Uekawa <dancer@debian.org>
Date: Tue, 25 Apr 2023 08:57:14 +0900
Subject: [PATCH] Check for std when using std features.

---
 src/tests.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tests.rs b/src/tests.rs
index f91a23b..21c8e73 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -221,6 +221,7 @@ fn signed_values() {
     }
 }
 
+#[cfg(feature = "std")]
 #[test]
 fn boolean_values() {
     let bytes: Vec<u8> = (0..16).collect();
-- 
2.30.2

