--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,2 +33,3 @@
 repository = "https://github.com/hanmertens/dary_heap"
+autobenches = false
 
@@ -41,6 +42,6 @@
 
-[[bench]]
-name = "dary_heap"
-path = "benches/dary_heap.rs"
-harness = false
+#[[bench]]
+#name = "dary_heap"
+#path = "benches/dary_heap.rs"
+#harness = false
 
@@ -52,5 +53,5 @@
 
-[dev-dependencies.criterion]
-version = "0.4"
-default-features = false
+#[dev-dependencies.criterion]
+#version = "0.4"
+#default-features = false
 
