Index: owned-ttf-parser/Cargo.toml
===================================================================
--- owned-ttf-parser.orig/Cargo.toml
+++ owned-ttf-parser/Cargo.toml
@@ -38,14 +38,6 @@ path = "src/lib.rs"
 name = "face_mut"
 path = "tests/face_mut.rs"
 
-[[test]]
-name = "owned_ttf"
-path = "tests/owned_ttf.rs"
-
-[[test]]
-name = "preparse"
-path = "tests/preparse.rs"
-
 [dependencies.ttf-parser]
 version = "0.25"
 default-features = false
