--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,10 +54,10 @@
 default = ["sync"]
-fs-err2 = [
-    "dep:fs-err2",
-    "fs-err2/io_safety",
-]
-fs-err2-tokio = [
-    "fs-err2",
-    "fs-err2/tokio",
-]
+#fs-err2 = [
+#    "dep:fs-err2",
+#    "fs-err2/io_safety",
+#]
+#fs-err2-tokio = [
+#    "fs-err2",
+#    "fs-err2/tokio",
+#]
 fs-err3 = ["dep:fs-err3"]
@@ -79,3 +79,3 @@
 
-[dependencies.fs-err2]
+[disabled.dependencies.fs-err2]
 version = "2"
