--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,7 +58,7 @@
 optional = true
 
 [dependencies.fs-err]
-version = "2"
+version = "3"
 optional = true
 
 [dependencies.smol]
@@ -76,9 +76,9 @@
 features = ["attributes"]
 
 [dev-dependencies.fs-err]
-version = "2"
+version = "3"
 features = [
-    "io_safety",
+    #"io_safety",
     "tokio",
 ]
 
@@ -100,7 +100,7 @@
 default = ["sync"]
 fs-err = [
     "dep:fs-err",
-    "fs-err/io_safety",
+    #"fs-err/io_safety",
 ]
 fs-err-tokio = [
     "fs-err",
@@ -111,7 +111,7 @@
 tokio = ["dep:tokio"]
 
 [target."cfg(not(windows))".dependencies.rustix]
-version = "0.38"
+version = ">= 0.38, < 2"
 features = ["fs"]
 
 [disabled.target."cfg(windows)".dependencies.windows-sys]
