--- a/Cargo.toml
+++ b/Cargo.toml
@@ -119,3 +119,3 @@
 
-[dev-dependencies.smol-potat]
+[disabled.dev-dependencies.smol-potat]
 version = "1.1"
--- a/src/file_ext/async_impl/smol_impl.rs
+++ b/src/file_ext/async_impl/smol_impl.rs
@@ -9,3 +9,3 @@
 test_mod! {
-  smol_potat::test,
+  tokio::test,
   use crate::smol::AsyncFileExt;
--- a/src/unix/async_impl/smol_impl.rs
+++ b/src/unix/async_impl/smol_impl.rs
@@ -9,3 +9,3 @@
 test_mod! {
-  smol_potat::test,
+  tokio::test,
   use crate::smol::AsyncFileExt;
