Index: x11-clipboard/tests/window-leak.rs
===================================================================
--- x11-clipboard.orig/tests/window-leak.rs
+++ x11-clipboard/tests/window-leak.rs
@@ -26,6 +26,7 @@ mod tests {
     use super::*;
 
     #[test]
+    #[ignore = "https://github.com/quininer/x11-clipboard/issues/50"]
     fn should_work_but_does_not() -> Result<(), Error> {
         for i in 0..1000 {
             paste_to_clipboard(&format!("I have told you {} times!", i))?;
