diff --git a/Cargo.toml b/Cargo.toml
index f47cf62..07601ad 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -75,7 +75,7 @@ optional = true
 default-features = false
 
 [dependencies.fs-err]
-version = "2"
+version = "3"
 
 [dependencies.once_cell]
 version = "1.17"
@@ -85,7 +85,7 @@ version = "1"
 optional = true
 
 [dependencies.secrecy]
-version = "0.10"
+version = "0.8"
 features = ["serde"]
 optional = true
 
diff --git a/src/lib.rs b/src/lib.rs
index ef40cd9..e5fcfaa 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -77,6 +77,6 @@ pub use fs_err as fs;
 #[cfg(feature = "secrets")]
 pub use secrecy as secret;
 #[cfg(feature = "secrets")]
-pub use secrecy::{SecretBox, SecretSlice, SecretString};
+pub use secrecy::{SecretBox, SecretString};
 #[cfg(all(feature = "default", feature = "application"))]
 pub use semver::Version;
