Author: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Description: bump to keyring 3

Forwarded: https://codeberg.org/openpgp-card/state/pulls/6

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,7 +30,7 @@
 version = "0.6"
 
 [dependencies.keyring]
-version = "2"
+version = "3"
 
 [dependencies.serde]
 version = "1"
--- a/src/keyring.rs
+++ b/src/keyring.rs
@@ -20,5 +20,5 @@
 
 pub(crate) fn delete(ident: &str) -> anyhow::Result<()> {
     let entry = Entry::new(KEYRING_NAME, ident)?;
-    Ok(entry.delete_password()?)
+    Ok(entry.delete_credential()?)
 }
