Index: clap-complete-command/Cargo.toml
===================================================================
--- clap-complete-command.orig/Cargo.toml
+++ clap-complete-command/Cargo.toml
@@ -46,16 +46,11 @@ version = "4"
 version = "4"
 optional = true
 
-[dependencies.clap_complete_nushell]
-version = "4"
-optional = true
-
 [dev-dependencies.clap]
 version = "4"
 features = ["derive"]
 
 [features]
 carapace = ["dep:carapace_spec_clap"]
-default = ["nushell"]
+default = ["fig"]
 fig = ["dep:clap_complete_fig"]
-nushell = ["dep:clap_complete_nushell"]
