Description: relax depdendency versions
Author: Ananthu C V <weepingclown@disroot.org>
Forwarded: not-needed
Last-Update: 2024-06-22

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -98,7 +98,7 @@
 version = "0.2"
 
 [dependencies.ordered-float]
-version = "4.1"
+version = "2.10"
 
 [dependencies.pest]
 version = "2.1"
@@ -110,7 +110,7 @@
 version = "0.11"
 
 [dependencies.semver]
-version = "0.11"
+version = "1"
 
 [dependencies.serde]
 version = "1.0"
@@ -161,13 +161,13 @@
 default-features = false
 
 [dev-dependencies.criterion]
-version = "0.4"
+version = "0.5"
 
 [dev-dependencies.env_logger]
-version = "0.11"
+version = "0.10"
 
 [dev-dependencies.k9]
-version = "0.11"
+version = "0.12"
 
 [dev-dependencies.varbincode]
 version = "0.1"
@@ -191,7 +191,7 @@
 ]
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.26"
+version = "0.27"
 
 [target."cfg(unix)".dependencies.signal-hook]
 version = "0.3"
