Index: termwiz/Cargo.toml
===================================================================
--- termwiz.orig/Cargo.toml
+++ termwiz/Cargo.toml
@@ -208,7 +208,7 @@ version = "0.6.2"
 version = "0.2.3"
 
 [dependencies.wezterm-blob-leases]
-version = "0.1.1"
+version = "0.1"
 
 [dependencies.wezterm-color-types]
 version = "0.3"
@@ -232,7 +232,7 @@ version = "0.12.0"
 version = "0.1"
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.29"
+version = ">= 0.29, < 1.0"
 features = [
     "mman",
     "fs",
