Description: relax depdendency versions
Author: Ananthu C V <weepingclown@disroot.org>
Forwarded: not-needed
Last-Update: 2024-06-22

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,3 +48,3 @@
 [dependencies.base64]
-version = "0.21"
+version = "0.22"
 
@@ -77,3 +77,3 @@
 [dependencies.image]
-version = "0.24.6"
+version = "0.25"
 optional = true
@@ -112,3 +112,3 @@
 [dependencies.semver]
-version = "0.11"
+version = "1"
 
@@ -126,3 +126,3 @@
 [dependencies.siphasher]
-version = "0.3"
+version = "1"
 
@@ -132,3 +132,3 @@
 [dependencies.terminfo]
-version = "0.8"
+version = "0.9"
 
@@ -163,3 +163,3 @@
 [dev-dependencies.criterion]
-version = "0.4"
+version = "0.5"
 
@@ -169,3 +169,3 @@
 [dev-dependencies.k9]
-version = "0.11"
+version = "0.12"
 
@@ -193,3 +193,3 @@
 [target."cfg(unix)".dependencies.nix]
-version = "0.26"
+version = "0.29"
 
