Index: gix-config/Cargo.toml
===================================================================
--- gix-config.orig/Cargo.toml
+++ gix-config/Cargo.toml
@@ -35,16 +35,12 @@ categories = [
 ]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/Byron/gitoxide"
+autobenches = false
 
 [package.metadata.docs.rs]
 all-features = true
 features = ["document-features"]
 
-[[bench]]
-name = "large_config_file"
-path = "./benches/large_config_file.rs"
-harness = false
-
 [dependencies.bstr]
 version = "1.3.0"
 features = ["std"]
