--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,16 +27,10 @@
 repository = "https://github.com/rust-av/v_frame"
 
 [features]
-profiling = ["dep:profiling"]
 serialize = [
     "serde",
     "aligned-vec/serde",
 ]
-tracing = [
-    "profiling",
-    "dep:tracing",
-    "profiling/profile-with-tracing",
-]
 
 [lib]
 name = "v_frame"
@@ -53,18 +47,10 @@
 [dependencies.num-traits]
 version = "0.2"
 
-[dependencies.profiling]
-version = "1"
-optional = true
-
 [dependencies.serde]
 version = "1.0"
 features = ["derive"]
 optional = true
 
-[dependencies.tracing]
-version = "0.1.40"
-optional = true
-
 [dev-dependencies.criterion]
 version = "0.5"
--- a/src/plane.rs
+++ b/src/plane.rs
@@ -511,7 +511,6 @@
     /// # Panics
     ///
     /// - If the current plane's width and height are not at least `SCALE` times the `in_plane`'s
-    #[cfg_attr(feature = "profiling", profiling::function(downscale_in_place))]
     pub fn downscale_in_place<const SCALE: usize>(&self, in_plane: &mut Plane<T>) {
         let stride = in_plane.cfg.stride;
         let width = in_plane.cfg.width;
