Description: Remove rust_hawktracer and wasm-bindgen dependency
Author: Sebastian Ramacher
Last-Update: 2022-06-23
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -66,15 +66,5 @@
 version = "0.1.40"
 optional = true
 
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.6"
-
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.criterion]
-version = "0.6"
-default-features = false
-
-[target.wasm32-unknown-unknown.dependencies.wasm-bindgen]
-version = "0.2"
-
-[target.wasm32-unknown-unknown.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+[dev-dependencies.criterion]
+version = "0.5"
