diff --git a/Cargo.toml b/Cargo.toml
index 75646c1..c3aa483 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,31 +31,6 @@ categories = [
 license = "MIT"
 repository = "https://github.com/pyros2097/rust-embed"
 
-[[example]]
-name = "warp"
-path = "examples/warp.rs"
-required-features = ["warp-ex"]
-
-[[example]]
-name = "actix"
-path = "examples/actix.rs"
-required-features = ["actix"]
-
-[[example]]
-name = "rocket"
-path = "examples/rocket.rs"
-required-features = ["rocket"]
-
-[[example]]
-name = "axum"
-path = "examples/axum.rs"
-required-features = ["axum-ex"]
-
-[[example]]
-name = "axum-spa"
-path = "examples/axum-spa/main.rs"
-required-features = ["axum-ex"]
-
 [[example]]
 name = "poem"
 path = "examples/poem.rs"
@@ -71,51 +46,20 @@ name = "include_exclude"
 path = "tests/include_exclude.rs"
 required-features = ["include-exclude"]
 
-[dependencies.actix-web]
-version = "4"
-optional = true
-
-[dependencies.axum]
-version = "0.5"
-features = ["http1"]
-optional = true
-default-features = false
-
 [dependencies.hex]
 version = "0.4.3"
 optional = true
 
-[dependencies.include-flate]
-version = "0.1"
-features = ["stable"]
-optional = true
-
 [dependencies.mime_guess]
 version = "2"
 optional = true
 
-[dependencies.poem]
-version = "1.3.30"
-features = ["server"]
-optional = true
-default-features = false
-
-[dependencies.rocket]
-version = "0.5.0-rc.2"
-optional = true
-default-features = false
-
 [dependencies.rust-embed-impl]
 version = "6.3.0"
 
 [dependencies.rust-embed-utils]
 version = "7.3.0"
 
-[dependencies.salvo]
-version = "0.16"
-optional = true
-default-features = false
-
 [dependencies.tokio]
 version = "1.0"
 features = [
@@ -127,28 +71,10 @@ optional = true
 [dependencies.walkdir]
 version = "2.3.1"
 
-[dependencies.warp]
-version = "0.3"
-optional = true
-default-features = false
-
 [dev-dependencies.sha2]
-version = "0.10"
+version = "0.10.5"
 
 [features]
-actix = [
-    "actix-web",
-    "mime_guess",
-]
-axum-ex = [
-    "axum",
-    "tokio",
-    "mime_guess",
-]
-compression = [
-    "rust-embed-impl/compression",
-    "include-flate",
-]
 debug-embed = [
     "rust-embed-impl/debug-embed",
     "rust-embed-utils/debug-embed",
@@ -158,23 +84,7 @@ include-exclude = [
     "rust-embed-utils/include-exclude",
 ]
 interpolate-folder-path = ["rust-embed-impl/interpolate-folder-path"]
-poem-ex = [
-    "poem",
-    "tokio",
-    "mime_guess",
-    "hex",
-]
-salvo-ex = [
-    "salvo",
-    "tokio",
-    "mime_guess",
-    "hex",
-]
-warp-ex = [
-    "warp",
-    "tokio",
-    "mime_guess",
-]
+
 
 [badges.appveyor]
 repository = "pyros2097/rust-embed"
