Index: getopts/Cargo.toml
===================================================================
--- getopts.orig/Cargo.toml
+++ getopts/Cargo.toml
@@ -29,10 +29,6 @@ repository = "https://github.com/rust-la
 
 [features]
 default = ["unicode"]
-rustc-dep-of-std = [
-    "std",
-    "core",
-]
 unicode = ["dep:unicode-width"]
 
 [lib]
@@ -43,16 +39,6 @@ path = "src/lib.rs"
 name = "smoke"
 path = "tests/smoke.rs"
 
-[dependencies.core]
-version = "1.0"
-optional = true
-package = "rustc-std-workspace-core"
-
-[dependencies.std]
-version = "1.0"
-optional = true
-package = "rustc-std-workspace-std"
-
 [dependencies.unicode-width]
 version = "0.2.0"
 optional = true
