Description: for now, disable libz to avoid NEW trip

Index: curl-sys/Cargo.toml
===================================================================
--- curl-sys.orig/Cargo.toml
+++ curl-sys/Cargo.toml
@@ -36,5 +35,0 @@
-[dependencies.libz-sys]
-version = "1.0.18"
-features = ["libc"]
-default-features = false
-
@@ -66,4 +60,0 @@
-zlib-ng-compat = [
-    "libz-sys/zlib-ng",
-    "static-curl",
-]
