Description: for now, disable libz to avoid NEW trip

Index: curl-sys/Cargo.toml
===================================================================
--- curl-sys.orig/Cargo.toml
+++ curl-sys/Cargo.toml
@@ -33,11 +33,6 @@ version = "0.2.2"
 version = "0.1.3"
 optional = true
 
-[dependencies.libz-sys]
-version = "1.0.18"
-features = ["libc"]
-default-features = false
-
 [dependencies.rustls-ffi]
 version = "0.8"
 features = ["no_log_capture"]
@@ -61,10 +56,6 @@ static-curl = []
 static-ssl = ["openssl-sys"]
 upkeep_7_62_0 = []
 windows-static-ssl = []
-zlib-ng-compat = [
-    "libz-sys/zlib-ng",
-    "static-curl",
-]
 
 [target."cfg(all(unix, not(target_os = \"macos\")))".dependencies.openssl-sys]
 version = "0.9.64"
