--- curl-sys.orig/Cargo.toml
+++ curl-sys/Cargo.toml
@@ -63,3 +63,3 @@
 static-curl = []
-static-ssl = ["openssl-sys/vendored"]
+static-ssl = ["openssl-sys"]
 upkeep_7_62_0 = []
only in patch2:
--- curl-sys.orig/build.rs
+++ curl-sys/build.rs
@@ -28,4 +28,5 @@
 
-    // If the static-curl feature is disabled, probe for a system-wide libcurl.
-    if !cfg!(feature = "static-curl") {
+    // On Debian, always probe for a system-wide libcurl even when linking statically
+    if true {
+        if cfg!(feature = "static-curl") { println!("cargo:static=1"); }
         // OSX ships libcurl by default, so we just use that version
