Index: curl-sys/Cargo.toml
===================================================================
--- curl-sys.orig/Cargo.toml
+++ curl-sys/Cargo.toml
@@ -41,7 +41,6 @@ version = "0.3.3"
 
 [features]
 default = ["ssl"]
-force-system-lib-on-osx = []
 http2 = ["libnghttp2-sys"]
 mesalink = []
 ntlm = []
@@ -52,19 +51,11 @@ ssl = ["openssl-sys"]
 static-curl = []
 static-ssl = ["openssl-sys"]
 upkeep_7_62_0 = []
-windows-static-ssl = []
 
 [target."cfg(all(unix, not(target_os = \"macos\")))".dependencies.openssl-sys]
 version = "0.9.64"
 optional = true
 
-[target."cfg(target_env = \"msvc\")".build-dependencies.vcpkg]
-version = "0.2"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.48"
-features = ["Win32_Networking_WinSock"]
-
 [badges.appveyor]
 repository = "alexcrichton/curl-rust"
 
