Index: curl-sys/lib.rs
===================================================================
--- curl-sys.orig/lib.rs
+++ curl-sys/lib.rs
@@ -2,11 +2,11 @@
 #![doc(html_root_url = "https://docs.rs/curl-sys/0.4")]
 
 // These `extern crate` are required for conditional linkages of curl.
-#[cfg(link_libnghttp2)]
+#[cfg(feature = "http2")]
 extern crate libnghttp2_sys;
 #[cfg(link_libz)]
 extern crate libz_sys;
-#[cfg(link_openssl)]
+#[cfg(feature = "ssl")]
 extern crate openssl_sys;
 #[cfg(feature = "rustls")]
 extern crate rustls_ffi;
