Forwarded: not-needed
Last-Update: 2025-04-10
--- a/src/lex/mod.rs
+++ b/src/lex/mod.rs
@@ -1,2 +1,2 @@
-#[cfg(test)]^M
-mod tests;^M
+//#[cfg(test)]
+//mod tests;
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -20,2 +20,2 @@
-#[cfg(test)]^M
-mod tests;^M
+//#[cfg(test)]
+//mod tests;
--- a/src/parse/mod.rs
+++ b/src/parse/mod.rs
@@ -19,2 +19,2 @@
-#[cfg(test)]^M
-mod tests;^M
+//#[cfg(test)]
+//mod tests;

