--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@
 license = "Apache-2.0 OR MIT"
 repository = "https://github.com/hsivonen/chardetng"
 [dependencies.arrayvec]
-version = "0.5.1"
+version = "0.7"
 optional = true
 
 [dependencies.cfg-if]
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -2860,7 +2860,7 @@
                 // performing thread synchronization only to bail
                 // out immediately when trying a disqualified
                 // candidate.
-                let mut qualified = ArrayVec::<[_; 27]>::new();
+                let mut qualified = ArrayVec::<_,27>::new();
                 for candidate in self.candidates.iter_mut() {
                     if candidate.qualified() {
                         qualified.push(candidate);
