--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,31 +42,22 @@
 name = "tokio_rusqlite"
 path = "src/lib.rs"
 
-[[example]]
-name = "in_memory"
-path = "examples/in_memory.rs"
-
-[[example]]
-name = "multiple_tasks"
-path = "examples/multiple_tasks.rs"
 
 [dependencies.crossbeam-channel]
 version = "0.5"
 
 [dependencies.rusqlite]
-version = "0.32"
+version = "0.29"
 
 [dependencies.tokio]
 version = "1"
 features = ["sync"]
 
 [dev-dependencies.rusqlite]
-version = "0.32"
-features = ["bundled"]
+version = "0.29"
 
 [dev-dependencies.tokio]
 version = "1"
 features = ["full"]
 
 [features]
-bundled = ["rusqlite/bundled"]
