Index: zoxide/Cargo.toml
===================================================================
--- zoxide.orig/Cargo.toml
+++ zoxide/Cargo.toml
@@ -106,7 +106,7 @@ version = "1.0.32"
 version = "1"
 
 [dependencies.clap]
-version = "4.3.0"
+version = "4"
 features = ["derive"]
 
 [dependencies.color-print]
@@ -149,14 +149,14 @@ version = "0.7.0"
 version = "3.1.0"
 
 [build-dependencies.clap]
-version = "4.3.0"
+version = "4"
 features = ["derive"]
 
 [build-dependencies.clap_complete]
-version = "4.3.0"
+version = "4"
 
 [build-dependencies.clap_complete_fig]
-version = "4.3.0"
+version = "4"
 
 [build-dependencies.color-print]
 version = "0.3.4"
