Index: zoxide/Cargo.toml
===================================================================
--- zoxide.orig/Cargo.toml
+++ zoxide/Cargo.toml
@@ -11,7 +11,7 @@
 
 [package]
 edition = "2024"
-rust-version = "1.85.0"
+#rust-version = "1.85.0"
 name = "zoxide"
 version = "0.9.8"
 authors = ["Ajeet D'Souza <98ajeet@gmail.com>"]
@@ -177,8 +177,8 @@ features = [
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.which]
-version = "7.0.3"
+#[target."cfg(windows)".dependencies.which]
+#version = "7.0.3"
 
 [profile.release]
 lto = true
