Forwarded: not-needed
Author: Blair Noctis <ncts@debian.org>
Last-Update: 2025-03-04
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -90,3 +90,3 @@
 [dependencies.pulldown-cmark]
-version = "0.11"
+version = "0.10"
 optional = true
--- a/src/utils/markup/markdown.rs
+++ b/src/utils/markup/markdown.rs
@@ -86,3 +86,3 @@
                     }
-                    Tag::BlockQuote(_) => return Some(self.literal("> ")),
+                    Tag::BlockQuote => return Some(self.literal("> ")),
                     Tag::Link {
@@ -114,4 +114,4 @@
                 | Event::Code(text)
-                | Event::InlineMath(text)
-                | Event::DisplayMath(text) => {
+                /*| Event::InlineMath(text)
+                | Event::DisplayMath(text)*/ => {
                     let text = cowvert(text);
