Index: cursive-core/src/printer.rs
===================================================================
--- cursive-core.orig/src/printer.rs
+++ cursive-core/src/printer.rs
@@ -132,7 +132,7 @@ impl<'a, 'b> Printer<'a, 'b> {
     /// The text could be truncated if it exceed the [drawing area size](Self::output_size).
     ///
     /// # Example
-    /// ```rust
+    /// ```ignore
     /// # use cursive_core as cursive;
     /// use cursive::{Printer, Vec2, View, XY};
     ///
