Index: cursive-core/src/lib.rs
===================================================================
--- cursive-core.orig/src/lib.rs
+++ cursive-core/src/lib.rs
@@ -8,7 +8,6 @@
 //!
 //! [`cursive`]: https://docs.rs/cursive
 #![deny(missing_docs)]
-#![cfg_attr(feature = "doc-cfg", feature(doc_cfg))]
 
 macro_rules! new_default(
     ($c:ident<$t:ident>) => {
Index: cursive-core/src/theme.rs
===================================================================
--- cursive-core.orig/src/theme.rs
+++ cursive-core/src/theme.rs
@@ -130,7 +130,6 @@ impl Theme {
     }
 
     #[cfg(feature = "toml")]
-    #[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "toml")))]
     /// Load values from an already parsed toml [`Table`], overwriting previous values.
     ///
     /// [`Table`]: https://docs.rs/toml/latest/toml/type.Table.html
@@ -160,13 +159,11 @@ pub enum Error {
     Io(io::Error),
 
     #[cfg(feature = "toml")]
-    #[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "toml")))]
     /// An error occurred when parsing the toml content.
     Parse(toml::de::Error),
 }
 
 #[cfg(feature = "toml")]
-#[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "toml")))]
 impl From<io::Error> for Error {
     fn from(err: io::Error) -> Self {
         Error::Io(err)
@@ -174,7 +171,6 @@ impl From<io::Error> for Error {
 }
 
 #[cfg(feature = "toml")]
-#[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "toml")))]
 impl From<toml::de::Error> for Error {
     fn from(err: toml::de::Error) -> Self {
         Error::Parse(err)
@@ -185,7 +181,6 @@ impl From<toml::de::Error> for Error {
 ///
 /// Must have the `toml` feature enabled.
 #[cfg(feature = "toml")]
-#[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "toml")))]
 pub fn load_theme_file<P: AsRef<Path>>(filename: P) -> Result<Theme, Error> {
     let content = {
         let mut content = String::new();
@@ -201,7 +196,6 @@ pub fn load_theme_file<P: AsRef<Path>>(f
 ///
 /// Must have the `toml` feature enabled.
 #[cfg(feature = "toml")]
-#[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "toml")))]
 pub fn load_toml(content: &str) -> Result<Theme, Error> {
     let table = toml::de::from_str(content)?;
 
Index: cursive-core/src/utils/markup/markdown.rs
===================================================================
--- cursive-core.orig/src/utils/markup/markdown.rs
+++ cursive-core/src/utils/markup/markdown.rs
@@ -2,7 +2,6 @@
 //!
 //! Needs the `markdown` feature to be enabled.
 #![cfg(feature = "markdown")]
-#![cfg_attr(feature = "doc-cfg", doc(cfg(feature = "markdown")))]
 
 use std::borrow::Cow;
 
Index: cursive-core/src/utils/markup/ansi.rs
===================================================================
--- cursive-core.orig/src/utils/markup/ansi.rs
+++ cursive-core/src/utils/markup/ansi.rs
@@ -2,7 +2,6 @@
 //!
 //! Needs the `ansi` feature to be enabled.
 #![cfg(feature = "ansi")]
-#![cfg_attr(feature = "doc-cfg", doc(cfg(feature = "ansi")))]
 
 use crate::style::{BaseColor, Color, Effect, Style};
 use crate::utils::markup::{StyledIndexedSpan, StyledString};
Index: cursive-core/src/utils/markup/cursup.rs
===================================================================
--- cursive-core.orig/src/utils/markup/cursup.rs
+++ cursive-core/src/utils/markup/cursup.rs
@@ -15,7 +15,6 @@
 //! // Highlight a letter from the word to show a shortcut available.
 //! Button::new(cursup::parse("/red{Q}uit"), |s| s.quit());
 //! ```
-#![cfg_attr(feature = "doc-cfg", doc(cfg(feature = "cursup")))]
 
 use crate::style::Style;
 use crate::utils::markup::{StyledIndexedSpan, StyledString};
