--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,8 +48,8 @@
 [dependencies.enum-map]
 version = "2.0"
 
-[dependencies.enumset]
-version = "1.0.4"
+[dependencies.wasmer_enumset]
+version = "1"
 
 [dependencies.lazy_static]
 version = "1"
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -38,7 +38,7 @@
 pub mod reexports {
     pub use ahash;
     pub use crossbeam_channel;
-    pub use enumset;
+    pub use wasmer_enumset;
     pub use log;
     pub use time;
 
--- a/src/printer.rs
+++ b/src/printer.rs
@@ -10,7 +10,7 @@
 use crate::with::With;
 use crate::Vec2;
 
-use enumset::EnumSet;
+use wasmer_enumset::EnumSet;
 use std::cell::Cell;
 use std::cmp::min;
 use unicode_segmentation::UnicodeSegmentation;
--- a/src/theme/effect.rs
+++ b/src/theme/effect.rs
@@ -1,4 +1,4 @@
-use enumset::EnumSetType;
+use wasmer_enumset::EnumSetType;
 
 /// Text effect
 #[allow(clippy::derive_hash_xor_eq)] // We do derive it through EnumSetType
--- a/src/theme/style.rs
+++ b/src/theme/style.rs
@@ -1,7 +1,7 @@
 use std::iter::FromIterator;
 
 use super::{Color, ColorStyle, ColorType, Effect, PaletteColor};
-use enumset::EnumSet;
+use wasmer_enumset::EnumSet;
 
 /// Combine a color and an effect.
 ///
--- a/src/utils/markup/markdown.rs
+++ b/src/utils/markup/markdown.rs
@@ -29,7 +29,7 @@
     first: bool,
     stack: Vec<Style>,
     input: &'a str,
-    parser: pulldown_cmark::Parser<'a>,
+    parser: pulldown_cmark::Parser<'a, 'a>,
 }
 
 impl<'a> Parser<'a> {
