--- a/Cargo.toml
+++ b/Cargo.toml
@@ -120,27 +120,27 @@ webp = ["dep:image-webp"]
 name = "image"
 path = "src/lib.rs"
 
-[[bench]]
+[[disabled.bench]]
 name = "blur"
 path = "benches/blur.rs"
 harness = false
 
-[[bench]]
+[[disabled.bench]]
 name = "copy_from"
 path = "benches/copy_from.rs"
 harness = false
 
-[[bench]]
+[[disabled.bench]]
 name = "decode"
 path = "benches/decode.rs"
 harness = false
 
-[[bench]]
+[[disabled.bench]]
 name = "encode"
 path = "benches/encode.rs"
 harness = false
 
-[[bench]]
+[[disabled.bench]]
 name = "fast_blur"
 path = "benches/fast_blur.rs"
 harness = false
@@ -228,1 +228,1 @@
-[dev-dependencies.criterion]
+[disabled.dev-dependencies.criterion]
