Index: image/src/codecs/bmp/decoder.rs
===================================================================
--- image.orig/src/codecs/bmp/decoder.rs
+++ image/src/codecs/bmp/decoder.rs
@@ -1419,6 +1419,7 @@ mod test {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     fn read_rect() {
         let f =
             BufReader::new(std::fs::File::open("tests/images/bmp/images/Core_8_Bit.bmp").unwrap());
@@ -1457,6 +1458,7 @@ mod test {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     fn test_no_header() {
         let tests = [
             "Info_R8_G8_B8.bmp",
Index: image/src/codecs/png.rs
===================================================================
--- image.orig/src/codecs/png.rs
+++ image/src/codecs/png.rs
@@ -742,6 +742,7 @@ mod tests {
     use std::io::{BufReader, Cursor, Read};
 
     #[test]
+    #[ignore = "skipping ..."]
     fn ensure_no_decoder_off_by_one() {
         let dec = PngDecoder::new(BufReader::new(
             std::fs::File::open("tests/images/png/bugfixes/debug_triangle_corners_widescreen.png")
@@ -767,6 +768,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     fn underlying_error() {
         use std::error::Error;
 
Index: image/src/images/dynimage.rs
===================================================================
--- image.orig/src/images/dynimage.rs
+++ image/src/images/dynimage.rs
@@ -1723,6 +1723,7 @@ mod test {
 
     #[cfg(feature = "jpeg")]
     #[test]
+    #[ignore = "skipping ..."]
     fn image_dimensions() {
         let im_path = "./tests/images/jpg/progressive/cat.jpg";
         let dims = super::image_dimensions(im_path).unwrap();
@@ -1731,6 +1732,7 @@ mod test {
 
     #[cfg(feature = "png")]
     #[test]
+    #[ignore = "skipping ..."]
     fn open_16bpc_png() {
         let im_path = "./tests/images/png/16bpc/basn6a16.png";
         let image = super::open(im_path).unwrap();
@@ -2200,7 +2202,7 @@ mod test {
         }
     }
 
-    /// Check that operations that are not cicp-aware behave as such. We introduce new methods (not
+    /* /// Check that operations that are not cicp-aware behave as such. We introduce new methods (not
     /// based directly on the public imageops interface) at a later point.
     #[cfg(feature = "png")]
     #[test]
@@ -2242,5 +2244,5 @@ mod test {
             assert_ne!(result_a, result_b, "{idx}");
             assert_eq!(result_a.as_bytes(), result_b.as_bytes(), "{idx}");
         }
-    }
+    }*/
 }
Index: image/src/imageops/sample.rs
===================================================================
--- image.orig/src/imageops/sample.rs
+++ image/src/imageops/sample.rs
@@ -1616,6 +1616,7 @@ mod tests {
     use test;
 
     #[bench]
+    #[ignore = "skipping ..."]
     #[cfg(all(feature = "benchmarks", feature = "png"))]
     fn bench_resize(b: &mut test::Bencher) {
         use std::path::Path;
@@ -1628,6 +1629,7 @@ mod tests {
 
     #[test]
     #[cfg(feature = "png")]
+    #[ignore = "skipping ..."]
     fn test_resize_same_size() {
         use std::path::Path;
         let img = crate::open(Path::new("./examples/fractal.png")).unwrap();
@@ -1637,6 +1639,7 @@ mod tests {
 
     #[test]
     #[cfg(feature = "png")]
+    #[ignore = "skipping ..."]
     fn test_sample_bilinear() {
         use std::path::Path;
         let img = crate::open(Path::new("./examples/fractal.png")).unwrap();
@@ -1656,6 +1659,7 @@ mod tests {
     }
     #[test]
     #[cfg(feature = "png")]
+    #[ignore = "skipping ..."]
     fn test_sample_nearest() {
         use std::path::Path;
         let img = crate::open(Path::new("./examples/fractal.png")).unwrap();
@@ -1745,6 +1749,7 @@ mod tests {
     }
 
     #[bench]
+    #[ignore = "skipping ..."]
     #[cfg(all(feature = "benchmarks", feature = "tiff"))]
     fn bench_resize_same_size(b: &mut test::Bencher) {
         let path = concat!(
@@ -1765,6 +1770,7 @@ mod tests {
     }
 
     #[bench]
+    #[ignore = "skipping ..."]
     #[cfg(all(feature = "benchmarks", feature = "tiff"))]
     fn bench_thumbnail(b: &mut test::Bencher) {
         let path = concat!(
@@ -1779,6 +1785,7 @@ mod tests {
     }
 
     #[bench]
+    #[ignore = "skipping ..."]
     #[cfg(all(feature = "benchmarks", feature = "tiff"))]
     fn bench_thumbnail_upsize(b: &mut test::Bencher) {
         let path = concat!(
@@ -1793,6 +1800,7 @@ mod tests {
     }
 
     #[bench]
+    #[ignore = "skipping ..."]
     #[cfg(all(feature = "benchmarks", feature = "tiff"))]
     fn bench_thumbnail_upsize_irregular(b: &mut test::Bencher) {
         let path = concat!(
@@ -1807,6 +1815,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     #[cfg(feature = "png")]
     fn resize_transparent_image() {
         use super::FilterType::{CatmullRom, Gaussian, Lanczos3, Nearest, Triangle};
Index: image/benches/decode.rs
===================================================================
--- image.orig/benches/decode.rs
+++ image/benches/decode.rs
@@ -92,7 +92,7 @@ fn bench_load(c: &mut Criterion, def: &B
     let mut group = c.benchmark_group(&group_name);
     let paths = IMAGE_DIR.iter().chain(def.dir);
 
-    for file_name in def.files {
+    /*for file_name in def.files {
         let path: path::PathBuf = paths.clone().chain(iter::once(file_name)).collect();
         let buf = fs::read(path).unwrap();
         group.bench_function(file_name.to_owned(), |b| {
@@ -100,7 +100,7 @@ fn bench_load(c: &mut Criterion, def: &B
                 image::load_from_memory_with_format(&buf, def.format).unwrap();
             });
         });
-    }
+    }*/
 }
 
 const IMAGE_DIR: [&str; 3] = [".", "tests", "images"];
Index: image/src/codecs/openexr.rs
===================================================================
--- image.orig/src/codecs/openexr.rs
+++ image/src/codecs/openexr.rs
@@ -412,6 +412,7 @@ mod test {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     fn compare_exr_hdr() {
         if cfg!(not(feature = "hdr")) {
             eprintln!("warning: to run all the openexr tests, activate the hdr feature flag");
@@ -486,6 +487,7 @@ mod test {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     fn compare_rgba_rgb() {
         let exr_path = BASE_PATH
             .iter()
@@ -503,6 +505,7 @@ mod test {
     }
 
     #[test]
+    #[ignore = "skipping ..."]
     fn compare_cropped() {
         // like in photoshop, exr images may have layers placed anywhere in a canvas.
         // we don't want to load the pixels from the layer, but we want to load the pixels from the canvas.
Index: image/src/codecs/jpeg/decoder.rs
===================================================================
--- image.orig/src/codecs/jpeg/decoder.rs
+++ image/src/codecs/jpeg/decoder.rs
@@ -202,6 +202,7 @@ mod tests {
     use std::{fs, io::Cursor};
 
     #[test]
+    #[ignore = "skipping ..."]
     fn test_exif_orientation() {
         let data = fs::read("tests/images/jpg/portrait_2.jpg").unwrap();
         let mut decoder = JpegDecoder::new(Cursor::new(data)).unwrap();
Index: image/src/codecs/qoi.rs
===================================================================
--- image.orig/src/codecs/qoi.rs
+++ image/src/codecs/qoi.rs
@@ -110,6 +110,7 @@ mod tests {
     use std::fs::File;
 
     #[test]
+    #[ignore = "skipping ..."]
     fn decode_test_image() {
         let decoder = QoiDecoder::new(File::open("tests/images/qoi/basic-test.qoi").unwrap())
             .expect("Unable to read QOI file");
Index: image/src/codecs/tga/encoder.rs
===================================================================
--- image.orig/src/codecs/tga/encoder.rs
+++ image/src/codecs/tga/encoder.rs
@@ -441,6 +441,7 @@ mod tests {
         }
 
         #[test]
+        #[ignore = "skipping ..."]
         fn round_trip_bw() {
             // This example demonstrates the run-length counter being saturated
             // It should never overflow and can be 128 max
Index: image/src/imageops/mod.rs
===================================================================
--- image.orig/src/imageops/mod.rs
+++ image/src/imageops/mod.rs
@@ -546,6 +546,7 @@ mod tests {
 
     #[test]
     #[cfg(feature = "tiff")]
+    #[ignore = "skipping ..."]
     fn fast_blur_approximates_gaussian_blur_well() {
         let path = concat!(
             env!("CARGO_MANIFEST_DIR"),
Index: image/src/metadata.rs
===================================================================
--- image.orig/src/metadata.rs
+++ image/src/metadata.rs
@@ -158,7 +158,7 @@ enum ExifEndian {
     Little,
 }
 
-#[cfg(all(test, feature = "jpeg"))]
+/*#[cfg(all(test, feature = "jpeg"))]
 mod tests {
     use crate::{codecs::jpeg::JpegDecoder, ImageDecoder as _};
 
@@ -189,4 +189,4 @@ mod tests {
             .expect("Failed to extract orientation from Exif chunk after clearing it");
         assert_eq!(orientation, Orientation::NoTransforms);
     }
-}
+}*/
