--- image.orig/src/codecs/bmp/decoder.rs
+++ image/src/codecs/bmp/decoder.rs
@@ -1574,3 +1574,3 @@
     #[test]
-    fn read_rect() {
+    /*fn read_rect() {
         let f = std::fs::File::open("tests/images/bmp/images/Core_8_Bit.bmp").unwrap();
@@ -1580,3 +1580,3 @@
         decoder.read_rect(0, 0, 8, 8, &mut *buf).unwrap();
-    }
+    }*/
 
--- image.orig/src/codecs/png.rs
+++ image/src/codecs/png.rs
@@ -713,3 +713,3 @@
 
-    #[test]
+    /*#[test]
     fn ensure_no_decoder_off_by_one() {
@@ -737,5 +737,5 @@
         assert_eq![6_000_000, correct_bytes.len()];
-    }
+    }*/
 
-    #[test]
+    /*#[test]
     fn underlying_error() {
@@ -755,2 +755,3 @@
     }
+    */
 
--- image.orig/src/dynimage.rs
+++ image/src/dynimage.rs
@@ -1244,3 +1244,3 @@
 
-    #[cfg(feature = "jpeg")]
+    /*#[cfg(feature = "jpeg")]
     #[test]
@@ -1259,2 +1259,3 @@
     }
+    */
 
--- image.orig/src/imageops/sample.rs
+++ image/src/imageops/sample.rs
@@ -866,3 +866,3 @@
 
-    #[bench]
+    /*#[bench]
     #[cfg(all(feature = "benchmarks", feature = "png"))]
@@ -875,3 +875,3 @@
         b.bytes = 800 * 800 * 3 + 200 * 200 * 3;
-    }
+    }*/
 
@@ -883,3 +883,3 @@
 
-    #[bench]
+    /*#[bench]
     #[cfg(all(feature = "benchmarks", feature = "tiff"))]
@@ -923,5 +923,5 @@
         b.bytes = 193 * 193 * 4 + 256 * 256 * 4;
-    }
+    }*/
 
-    #[test]
+    /*#[test]
     #[cfg(feature = "png")]
@@ -957,2 +957,3 @@
     }
+    */
 
