--- image.orig/Cargo.toml
+++ image/Cargo.toml
@@ -50,8 +49,0 @@
-[dependencies.dav1d]
-version = "0.6.0"
-optional = true
-
-[dependencies.dcv-color-primitives]
-version = "0.4.0"
-optional = true
-
@@ -58,3 +49,0 @@
-[dependencies.exr]
-version = "1.5.0"
-optional = true
@@ -68,4 +59,0 @@
-[dependencies.mp4parse]
-version = "0.12.0"
-optional = true
-
@@ -83,8 +70,0 @@
-[dependencies.ravif]
-version = "0.8.0"
-optional = true
-
-[dependencies.rgb]
-version = "0.8.25"
-optional = true
-
@@ -118,10 +97,0 @@
-avif = ["avif-encoder"]
-avif-decoder = [
-    "mp4parse",
-    "dcv-color-primitives",
-    "dav1d",
-]
-avif-encoder = [
-    "ravif",
-    "rgb",
-]
@@ -146 +115,0 @@
-    "openexr",
@@ -156 +124,0 @@
-openexr = ["exr"]
