--- a/Cargo.toml
+++ b/Cargo.toml
@@ -90,6 +90,2 @@
 
-[dependencies.dav1d]
-version = "0.10.3"
-optional = true
-
 [dependencies.exr]
@@ -106,6 +102,2 @@
 
-[dependencies.mp4parse]
-version = "0.17.0"
-optional = true
-
 [dependencies.num-traits]
@@ -121,6 +113,2 @@
 
-[dependencies.ravif]
-version = "0.11.11"
-optional = true
-default-features = false
 
@@ -169,11 +157,2 @@
 [features]
-avif = [
-    "dep:ravif",
-    "dep:rgb",
-]
-avif-native = [
-    "dep:mp4parse",
-    "dep:dav1d",
-]
-benchmarks = []
 bmp = []
@@ -186,3 +165,2 @@
 default-formats = [
-    "avif",
     "bmp",
@@ -217,3 +195,2 @@
 ]
-nasm = ["ravif?/asm"]
 png = ["dep:png"]
@@ -223,3 +200,2 @@
     "dep:rayon",
-    "ravif?/threading",
 ]
