--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,7 +60,7 @@
 optional = true
 
 [dependencies.tiff]
-version = "0.2.0"
+version = "0.3.0"
 optional = true
 [dev-dependencies.glob]
 version = "0.2.10"
--- a/src/tiff.rs
+++ b/src/tiff.rs
@@ -46,6 +46,7 @@
             tiff::TiffError::IoError(err) => ImageError::IoError(err),
             tiff::TiffError::FormatError(desc) => ImageError::FormatError(desc.to_string()),
             tiff::TiffError::UnsupportedError(desc) => ImageError::UnsupportedError(desc.to_string()),
+            tiff::TiffError::LimitsExceeded => ImageError::InsufficientMemory,
         }
     }
 }
