Index: aliasable/Cargo.toml
===================================================================
--- aliasable.orig/Cargo.toml
+++ aliasable/Cargo.toml
@@ -26,9 +26,6 @@ repository = "https://github.com/avitex/
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
-[dependencies.aliasable_deref_trait]
-version = "0.2"
-optional = true
 
 [dependencies.stable_deref_trait]
 version = "1.2"
@@ -37,4 +34,3 @@ optional = true
 [features]
 alloc = []
 default = ["alloc"]
-traits = ["stable_deref_trait", "aliasable_deref_trait"]
