diff --git a/Cargo.toml b/Cargo.toml
index 7b1dc35..d406672 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -81,11 +81,11 @@ replace = """
 exactly = 1
 
 [dependencies.anstream]
-version = "0.3.0"
+version = "0.3"
 optional = true
 
 [dependencies.anstyle]
-version = "1.0.0"
+version = "1"
 
 [dependencies.doc-comment]
 version = "0.3"
@@ -94,18 +94,18 @@ version = "0.3"
 version = "0.8"
 
 [dependencies.predicates]
-version = "3.0.1"
+version = "2"
 features = ["diff"]
 default-features = false
 
 [dependencies.predicates-core]
-version = "1.0.6"
+version = "1"
 
 [dependencies.predicates-tree]
-version = "1.0.1"
+version = "1"
 
 [dependencies.tempfile]
-version = "3.0"
+version = "3"
 
 [features]
 color = [
