--- a/Cargo.toml
+++ b/Cargo.toml
@@ -73,7 +73,7 @@ features = ["rust_backend"]
 default-features = false
 
 [dependencies.fontdb]
-version = "0.23.0"
+version = ">= 0.22"
 optional = true
 default-features = false
 
@@ -94,7 +94,7 @@ features = ["eq-separator"]
 version = "0.20"
 
 [dependencies.rustybuzz]
-version = "0.20.1"
+version = ">= 0.18"
 optional = true
 
 [dependencies.simplecss]
@@ -107,10 +107,10 @@ version = "1.0"
 version = "0.1.1"
 
 [dependencies.svgtypes]
-version = "0.15.3"
+version = ">= 0.15"
 
 [dependencies.tiny-skia-path]
-version = "0.11.4"
+version = ">= 0.11"
 
 [dependencies.unicode-bidi]
 version = "0.3"
