--- a/src/text/colr.rs
+++ b/src/text/colr.rs
@@ -294,7 +294,7 @@
         self.svg.end_element(); // g
     }
 
-    fn push_translate(&mut self, tx: f32, ty: f32) {
+    /* fn push_translate(&mut self, tx: f32, ty: f32) {
         self.push_transform(ttf_parser::Transform::new(1.0, 0.0, 0.0, 1.0, tx, ty));
     }
 
@@ -312,7 +312,7 @@
         let x = (-skew_x * std::f32::consts::PI).tan();
         let y = (skew_y * std::f32::consts::PI).tan();
         self.push_transform(ttf_parser::Transform::new(1.0, y, x, 1.0, 0.0, 0.0));
-    }
+    } */
 
     fn push_transform(&mut self, transform: ttf_parser::Transform) {
         self.transforms_stack.push(self.transform);
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,7 +64,7 @@
 version = "0.20"
 
 [dependencies.rustybuzz]
-version = "0.14.0"
+version = "0.17"
 optional = true
 
 [dependencies.simplecss]
