--- rust-virtio-bindings-0.2.0.orig/src/virtio_blk.rs
+++ rust-virtio-bindings-0.2.0/src/virtio_blk.rs
@@ -378,7 +378,7 @@ fn bindgen_test_layout_virtio_blk_outhdr
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_blk_outhdr>(),
-        8usize,
+        ::std::mem::align_of::<virtio_blk_outhdr>(),
         concat!("Alignment of ", stringify!(virtio_blk_outhdr))
     );
     assert_eq!(
@@ -431,7 +431,7 @@ fn bindgen_test_layout_virtio_blk_discar
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_blk_discard_write_zeroes>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_blk_discard_write_zeroes))
     );
     assert_eq!(
--- rust-virtio-bindings-0.2.0.orig/src/virtio_gpu.rs
+++ rust-virtio-bindings-0.2.0/src/virtio_gpu.rs
@@ -124,7 +124,7 @@ fn bindgen_test_layout_virtio_gpu_ctrl_h
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_ctrl_hdr>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctrl_hdr))
     );
     assert_eq!(
@@ -274,7 +274,7 @@ fn bindgen_test_layout_virtio_gpu_update
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_update_cursor>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_update_cursor))
     );
     assert_eq!(
@@ -420,7 +420,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_unref>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_unref))
     );
     assert_eq!(
@@ -475,7 +475,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_create_2d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_create_2d))
     );
     assert_eq!(
@@ -549,7 +549,7 @@ fn bindgen_test_layout_virtio_gpu_set_sc
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_set_scanout>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_set_scanout))
     );
     assert_eq!(
@@ -613,7 +613,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_flush>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_flush))
     );
     assert_eq!(
@@ -678,7 +678,7 @@ fn bindgen_test_layout_virtio_gpu_transf
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_transfer_to_host_2d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_transfer_to_host_2d))
     );
     assert_eq!(
@@ -750,7 +750,7 @@ fn bindgen_test_layout_virtio_gpu_mem_en
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_mem_entry>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_mem_entry))
     );
     assert_eq!(
@@ -803,7 +803,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_attach_backing>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!(
             "Alignment of ",
             stringify!(virtio_gpu_resource_attach_backing)
@@ -859,7 +859,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_detach_backing>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!(
             "Alignment of ",
             stringify!(virtio_gpu_resource_detach_backing)
@@ -973,7 +973,7 @@ fn bindgen_test_layout_virtio_gpu_resp_d
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resp_display_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_display_info))
     );
     assert_eq!(
@@ -1105,7 +1105,7 @@ fn bindgen_test_layout_virtio_gpu_transf
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_transfer_host_3d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_transfer_host_3d))
     );
     assert_eq!(
@@ -1208,7 +1208,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_create_3d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_create_3d))
     );
     assert_eq!(
@@ -1362,7 +1362,7 @@ fn bindgen_test_layout_virtio_gpu_ctx_cr
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_ctx_create>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctx_create))
     );
     assert_eq!(
@@ -1432,7 +1432,7 @@ fn bindgen_test_layout_virtio_gpu_ctx_de
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_ctx_destroy>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctx_destroy))
     );
     assert_eq!(
@@ -1465,7 +1465,7 @@ fn bindgen_test_layout_virtio_gpu_ctx_re
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_ctx_resource>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctx_resource))
     );
     assert_eq!(
@@ -1518,7 +1518,7 @@ fn bindgen_test_layout_virtio_gpu_cmd_su
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_cmd_submit>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_cmd_submit))
     );
     assert_eq!(
@@ -1571,7 +1571,7 @@ fn bindgen_test_layout_virtio_gpu_get_ca
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_get_capset_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_get_capset_info))
     );
     assert_eq!(
@@ -1626,7 +1626,7 @@ fn bindgen_test_layout_virtio_gpu_resp_c
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resp_capset_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_capset_info))
     );
     assert_eq!(
@@ -1699,7 +1699,7 @@ fn bindgen_test_layout_virtio_gpu_get_ca
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_get_capset>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_get_capset))
     );
     assert_eq!(
@@ -1751,7 +1751,7 @@ fn bindgen_test_layout_virtio_gpu_resp_c
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resp_capset>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_capset))
     );
     assert_eq!(
@@ -1794,7 +1794,7 @@ fn bindgen_test_layout_virtio_gpu_cmd_ge
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_cmd_get_edid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_cmd_get_edid))
     );
     assert_eq!(
@@ -1847,7 +1847,7 @@ fn bindgen_test_layout_virtio_gpu_resp_e
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resp_edid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_edid))
     );
     assert_eq!(
@@ -1991,7 +1991,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_assign_uuid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_assign_uuid))
     );
     assert_eq!(
@@ -2043,7 +2043,7 @@ fn bindgen_test_layout_virtio_gpu_resp_r
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resp_resource_uuid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_resource_uuid))
     );
     assert_eq!(
@@ -2090,7 +2090,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_create_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_create_blob))
     );
     assert_eq!(
@@ -2190,7 +2190,7 @@ fn bindgen_test_layout_virtio_gpu_set_sc
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_set_scanout_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_set_scanout_blob))
     );
     assert_eq!(
@@ -2314,7 +2314,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_map_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_map_blob))
     );
     assert_eq!(
@@ -2377,7 +2377,7 @@ fn bindgen_test_layout_virtio_gpu_resp_m
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resp_map_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_map_info))
     );
     assert_eq!(
@@ -2430,7 +2430,7 @@ fn bindgen_test_layout_virtio_gpu_resour
     );
     assert_eq!(
         ::std::mem::align_of::<virtio_gpu_resource_unmap_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_unmap_blob))
     );
     assert_eq!(
--- rust-virtio-bindings-0.2.0.orig/src/virtio_ring.rs
+++ rust-virtio-bindings-0.2.0/src/virtio_ring.rs
@@ -75,7 +75,7 @@ fn bindgen_test_layout_vring_desc() {
     );
     assert_eq!(
         ::std::mem::align_of::<vring_desc>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(vring_desc))
     );
     assert_eq!(
@@ -283,12 +283,12 @@ fn bindgen_test_layout_vring() {
     let ptr = UNINIT.as_ptr();
     assert_eq!(
         ::std::mem::size_of::<vring>(),
-        32usize,
+        ::std::mem::size_of::<usize>() * 4,
         concat!("Size of: ", stringify!(vring))
     );
     assert_eq!(
         ::std::mem::align_of::<vring>(),
-        8usize,
+        ::std::mem::align_of::<usize>(),
         concat!("Alignment of ", stringify!(vring))
     );
     assert_eq!(
@@ -303,7 +303,7 @@ fn bindgen_test_layout_vring() {
     );
     assert_eq!(
         unsafe { ::std::ptr::addr_of!((*ptr).desc) as usize - ptr as usize },
-        8usize,
+        ::std::mem::size_of::<usize>(),
         concat!(
             "Offset of field: ",
             stringify!(vring),
@@ -313,7 +313,7 @@ fn bindgen_test_layout_vring() {
     );
     assert_eq!(
         unsafe { ::std::ptr::addr_of!((*ptr).avail) as usize - ptr as usize },
-        16usize,
+        ::std::mem::size_of::<usize>() * 2,
         concat!(
             "Offset of field: ",
             stringify!(vring),
@@ -323,7 +323,7 @@ fn bindgen_test_layout_vring() {
     );
     assert_eq!(
         unsafe { ::std::ptr::addr_of!((*ptr).used) as usize - ptr as usize },
-        24usize,
+        ::std::mem::size_of::<usize>() * 3,
         concat!(
             "Offset of field: ",
             stringify!(vring),
@@ -402,7 +402,7 @@ fn bindgen_test_layout_vring_packed_desc
     );
     assert_eq!(
         ::std::mem::align_of::<vring_packed_desc>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(vring_packed_desc))
     );
     assert_eq!(
