diff --git a/src/virtio_blk.rs b/src/virtio_blk.rs
index a2e5fb2..0ca068c 100644
--- a/src/virtio_blk.rs
+++ b/src/virtio_blk.rs
@@ -550,3 +550,3 @@ fn bindgen_test_layout_virtio_blk_outhdr() {
         ::std::mem::align_of::<virtio_blk_outhdr>(),
-        8usize,
+        ::std::mem::align_of::<virtio_blk_outhdr>(),
         concat!("Alignment of ", stringify!(virtio_blk_outhdr))
@@ -606,3 +606,3 @@ fn bindgen_test_layout_virtio_blk_zone_descriptor() {
         ::std::mem::align_of::<virtio_blk_zone_descriptor>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_blk_zone_descriptor))
@@ -698,3 +698,3 @@ fn bindgen_test_layout_virtio_blk_zone_report() {
         ::std::mem::align_of::<virtio_blk_zone_report>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_blk_zone_report))
@@ -760,3 +760,3 @@ fn bindgen_test_layout_virtio_blk_discard_write_zeroes() {
         ::std::mem::align_of::<virtio_blk_discard_write_zeroes>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_blk_discard_write_zeroes))
diff --git a/src/virtio_gpu.rs b/src/virtio_gpu.rs
index ba83c90..f73ccaa 100644
--- a/src/virtio_gpu.rs
+++ b/src/virtio_gpu.rs
@@ -126,3 +126,3 @@ fn bindgen_test_layout_virtio_gpu_ctrl_hdr() {
         ::std::mem::align_of::<virtio_gpu_ctrl_hdr>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctrl_hdr))
@@ -276,3 +276,3 @@ fn bindgen_test_layout_virtio_gpu_update_cursor() {
         ::std::mem::align_of::<virtio_gpu_update_cursor>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_update_cursor))
@@ -422,3 +422,3 @@ fn bindgen_test_layout_virtio_gpu_resource_unref() {
         ::std::mem::align_of::<virtio_gpu_resource_unref>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_unref))
@@ -477,3 +477,3 @@ fn bindgen_test_layout_virtio_gpu_resource_create_2d() {
         ::std::mem::align_of::<virtio_gpu_resource_create_2d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_create_2d))
@@ -551,3 +551,3 @@ fn bindgen_test_layout_virtio_gpu_set_scanout() {
         ::std::mem::align_of::<virtio_gpu_set_scanout>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_set_scanout))
@@ -615,3 +615,3 @@ fn bindgen_test_layout_virtio_gpu_resource_flush() {
         ::std::mem::align_of::<virtio_gpu_resource_flush>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_flush))
@@ -680,3 +680,3 @@ fn bindgen_test_layout_virtio_gpu_transfer_to_host_2d() {
         ::std::mem::align_of::<virtio_gpu_transfer_to_host_2d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_transfer_to_host_2d))
@@ -752,3 +752,3 @@ fn bindgen_test_layout_virtio_gpu_mem_entry() {
         ::std::mem::align_of::<virtio_gpu_mem_entry>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_mem_entry))
@@ -805,3 +805,3 @@ fn bindgen_test_layout_virtio_gpu_resource_attach_backing() {
         ::std::mem::align_of::<virtio_gpu_resource_attach_backing>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!(
@@ -861,3 +861,3 @@ fn bindgen_test_layout_virtio_gpu_resource_detach_backing() {
         ::std::mem::align_of::<virtio_gpu_resource_detach_backing>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!(
@@ -975,3 +975,3 @@ fn bindgen_test_layout_virtio_gpu_resp_display_info() {
         ::std::mem::align_of::<virtio_gpu_resp_display_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_display_info))
@@ -1107,3 +1107,3 @@ fn bindgen_test_layout_virtio_gpu_transfer_host_3d() {
         ::std::mem::align_of::<virtio_gpu_transfer_host_3d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_transfer_host_3d))
@@ -1210,3 +1210,3 @@ fn bindgen_test_layout_virtio_gpu_resource_create_3d() {
         ::std::mem::align_of::<virtio_gpu_resource_create_3d>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_create_3d))
@@ -1364,3 +1364,3 @@ fn bindgen_test_layout_virtio_gpu_ctx_create() {
         ::std::mem::align_of::<virtio_gpu_ctx_create>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctx_create))
@@ -1434,3 +1434,3 @@ fn bindgen_test_layout_virtio_gpu_ctx_destroy() {
         ::std::mem::align_of::<virtio_gpu_ctx_destroy>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctx_destroy))
@@ -1467,3 +1467,3 @@ fn bindgen_test_layout_virtio_gpu_ctx_resource() {
         ::std::mem::align_of::<virtio_gpu_ctx_resource>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_ctx_resource))
@@ -1520,3 +1520,3 @@ fn bindgen_test_layout_virtio_gpu_cmd_submit() {
         ::std::mem::align_of::<virtio_gpu_cmd_submit>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_cmd_submit))
@@ -1573,3 +1573,3 @@ fn bindgen_test_layout_virtio_gpu_get_capset_info() {
         ::std::mem::align_of::<virtio_gpu_get_capset_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_get_capset_info))
@@ -1628,3 +1628,3 @@ fn bindgen_test_layout_virtio_gpu_resp_capset_info() {
         ::std::mem::align_of::<virtio_gpu_resp_capset_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_capset_info))
@@ -1701,3 +1701,3 @@ fn bindgen_test_layout_virtio_gpu_get_capset() {
         ::std::mem::align_of::<virtio_gpu_get_capset>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_get_capset))
@@ -1753,3 +1753,3 @@ fn bindgen_test_layout_virtio_gpu_resp_capset() {
         ::std::mem::align_of::<virtio_gpu_resp_capset>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_capset))
@@ -1796,3 +1796,3 @@ fn bindgen_test_layout_virtio_gpu_cmd_get_edid() {
         ::std::mem::align_of::<virtio_gpu_cmd_get_edid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_cmd_get_edid))
@@ -1849,3 +1849,3 @@ fn bindgen_test_layout_virtio_gpu_resp_edid() {
         ::std::mem::align_of::<virtio_gpu_resp_edid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_edid))
@@ -1993,3 +1993,3 @@ fn bindgen_test_layout_virtio_gpu_resource_assign_uuid() {
         ::std::mem::align_of::<virtio_gpu_resource_assign_uuid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_assign_uuid))
@@ -2045,3 +2045,3 @@ fn bindgen_test_layout_virtio_gpu_resp_resource_uuid() {
         ::std::mem::align_of::<virtio_gpu_resp_resource_uuid>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_resource_uuid))
@@ -2092,3 +2092,3 @@ fn bindgen_test_layout_virtio_gpu_resource_create_blob() {
         ::std::mem::align_of::<virtio_gpu_resource_create_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_create_blob))
@@ -2192,3 +2192,3 @@ fn bindgen_test_layout_virtio_gpu_set_scanout_blob() {
         ::std::mem::align_of::<virtio_gpu_set_scanout_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_set_scanout_blob))
@@ -2316,3 +2316,3 @@ fn bindgen_test_layout_virtio_gpu_resource_map_blob() {
         ::std::mem::align_of::<virtio_gpu_resource_map_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_map_blob))
@@ -2379,3 +2379,3 @@ fn bindgen_test_layout_virtio_gpu_resp_map_info() {
         ::std::mem::align_of::<virtio_gpu_resp_map_info>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resp_map_info))
@@ -2432,3 +2432,3 @@ fn bindgen_test_layout_virtio_gpu_resource_unmap_blob() {
         ::std::mem::align_of::<virtio_gpu_resource_unmap_blob>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(virtio_gpu_resource_unmap_blob))
diff --git a/src/virtio_ring.rs b/src/virtio_ring.rs
index 9512aaa..6682917 100644
--- a/src/virtio_ring.rs
+++ b/src/virtio_ring.rs
@@ -77,3 +77,3 @@ fn bindgen_test_layout_vring_desc() {
         ::std::mem::align_of::<vring_desc>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(vring_desc))
@@ -285,3 +285,3 @@ fn bindgen_test_layout_vring() {
         ::std::mem::size_of::<vring>(),
-        32usize,
+        ::std::mem::size_of::<usize>() * 4,
         concat!("Size of: ", stringify!(vring))
@@ -290,3 +290,3 @@ fn bindgen_test_layout_vring() {
         ::std::mem::align_of::<vring>(),
-        8usize,
+        ::std::mem::align_of::<usize>(),
         concat!("Alignment of ", stringify!(vring))
@@ -305,3 +305,3 @@ fn bindgen_test_layout_vring() {
         unsafe { ::std::ptr::addr_of!((*ptr).desc) as usize - ptr as usize },
-        8usize,
+        ::std::mem::size_of::<usize>(),
         concat!(
@@ -315,3 +315,3 @@ fn bindgen_test_layout_vring() {
         unsafe { ::std::ptr::addr_of!((*ptr).avail) as usize - ptr as usize },
-        16usize,
+        ::std::mem::size_of::<usize>() * 2,
         concat!(
@@ -325,3 +325,3 @@ fn bindgen_test_layout_vring() {
         unsafe { ::std::ptr::addr_of!((*ptr).used) as usize - ptr as usize },
-        24usize,
+        ::std::mem::size_of::<usize>() * 3,
         concat!(
@@ -404,3 +404,3 @@ fn bindgen_test_layout_vring_packed_desc() {
         ::std::mem::align_of::<vring_packed_desc>(),
-        8usize,
+        ::std::mem::align_of::<u64>(),
         concat!("Alignment of ", stringify!(vring_packed_desc))
