Remove Windows dependencies: we are building only for debian

Index: sequoia-policy-config/Cargo.toml
===================================================================
--- sequoia-policy-config.orig/Cargo.toml
+++ sequoia-policy-config/Cargo.toml
@@ -68,10 +68,5 @@ version = "1.11"
 features = ["crypto-nettle"]
 default-features = false

-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "1.11"
-features = ["crypto-cng"]
-default-features = false
-
 [badges.maintenance]
 status = "actively-developed"
 
