diff --git a/Cargo.toml b/Cargo.toml
index 8abc397..552c638 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,15 +40,15 @@ version = "0.2.87"
 [features]
 futures-core-03-stream = ["futures-core"]
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.futures-channel-preview]
-version = "0.3.0-alpha.18"
+#[target."cfg(target_arch = \"wasm32\")".dev-dependencies.futures-channel-preview]
+#version = "0.3.0-alpha.18"
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.futures-lite]
-version = "1.11.3"
-default-features = false
+#[target."cfg(target_arch = \"wasm32\")".dev-dependencies.futures-lite]
+#version = "1.11.3"
+#default-features = false
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3.37"
+#[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
+#version = "0.3.37"
 
 [target."cfg(target_feature = \"atomics\")".dependencies.web-sys]
 version = "0.3.24"
