Index: bzip2/Cargo.toml
===================================================================
--- bzip2.orig/Cargo.toml
+++ bzip2/Cargo.toml
@@ -25,16 +25,9 @@ repository = "https://github.com/alexcri
 [dependencies.bzip2-sys]
 version = "0.1"
 
-[dependencies.futures]
-version = "0.1"
-optional = true
-
 [dependencies.libc]
 version = "0.2"
 
-[dependencies.tokio-io]
-version = "0.1"
-optional = true
 [dev-dependencies.partial-io]
 version = "^0.2.1"
 features = ["quickcheck"]
@@ -45,8 +38,5 @@ version = "0.4"
 [dev-dependencies.rand]
 version = "0.3"
 
-[dev-dependencies.tokio-core]
-version = "0.1"
-
 [features]
-tokio = ["tokio-io", "futures"]
+tokio = []
