Index: toml/Cargo.toml
===================================================================
--- toml.orig/Cargo.toml
+++ toml/Cargo.toml
@@ -122,24 +122,6 @@ required-features = [
 ]
 
 [[test]]
-name = "decoder"
-path = "tests/decoder.rs"
-
-[[test]]
-name = "decoder_compliance"
-path = "tests/decoder_compliance.rs"
-harness = false
-
-[[test]]
-name = "encoder"
-path = "tests/encoder.rs"
-
-[[test]]
-name = "encoder_compliance"
-path = "tests/encoder_compliance.rs"
-harness = false
-
-[[test]]
 name = "testsuite"
 path = "tests/testsuite/main.rs"
 
@@ -180,3 +162,0 @@
-[dev-dependencies.toml-test-harness]
-version = "0.4.8"
-
