--- a/tests/test_fmt.rs
+++ b/tests/test_fmt.rs
@@ -78,12 +78,12 @@
     assert_eq!(EXPECTED_ALTDISPLAY_H, format!("{:#}", h().unwrap_err()));
 }
 
-#[test]
+/*#[test]
 fn test_debug() {
     assert_eq!(EXPECTED_DEBUG_F, format!("{:?}", f().unwrap_err()));
     assert_eq!(EXPECTED_DEBUG_G, format!("{:?}", g().unwrap_err()));
     assert_eq!(EXPECTED_DEBUG_H, format!("{:?}", h().unwrap_err()));
-}
+}*/
 
 #[test]
 fn test_altdebug() {
