--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,15 +37,6 @@
 name = "freedesktop_icons"
 path = "src/lib.rs"
 
-[[bench]]
-name = "simple_lookup"
-path = "benches/simple_lookup.rs"
-harness = false
-
-[[bench]]
-name = "tests"
-path = "benches/tests.rs"
-
 [dependencies.dirs]
 version = "5.0.1"
 
@@ -64,9 +55,6 @@
 [dependencies.xdg]
 version = "2.5.2"
 
-[dev-dependencies.criterion]
-version = "0.5"
-
 [dev-dependencies.gtk4]
 version = "0.9"
 
