diff --git a/Cargo.toml b/Cargo.toml
index 9f386fc..b2b2877 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,16 +47,16 @@ version = "0.2.0"
 version = "1.19.0"
 
 [dependencies.thiserror]
-version = "1.0.56"
+version = "2"
 
 [dependencies.tracing]
-version = "0.1.41"
+version = "0.1"
 
 [dependencies.xdg]
 version = "2.5.2"
 
 [dev-dependencies.gtk4]
-version = "0.9"
+version = "0.10"
 
 [dev-dependencies.speculoos]
 version = "0.11.0"
