Index: xattr/Cargo.toml
===================================================================
--- xattr.orig/Cargo.toml
+++ xattr/Cargo.toml
@@ -47,7 +47,7 @@ path = "tests/main.rs"
 version = "3"
 
 [target.'cfg(any(target_os = "android", target_os = "linux", target_os = "macos", target_os = "hurd"))'.dependencies.rustix]
-version = "1.0.0"
+version = ">= 0.38.37, < 2.0"
 features = [
     "fs",
     "std",
