Description: Change dependencies to available packaged versions
Author: Jeremy Bicha <jeremy.bicha@canonical.com>
Index: gstreamer-player-sys/Cargo.toml
===================================================================
--- gstreamer-player-sys.orig/Cargo.toml
+++ gstreamer-player-sys/Cargo.toml
@@ -66,7 +66,7 @@ version = "1.0.0"
 [dev-dependencies.tempfile]
 version = "3"
 [build-dependencies.system-deps]
-version = "3"
+version = "6"
 
 [features]
 dox = []
