Index: sysinfo/src/lib.rs
===================================================================
--- sysinfo.orig/src/common/system.rs
+++ sysinfo/src/common/system.rs
@@ -379,5 +379,5 @@
-            assert!(
-                !System::os_version()
-                    .expect("Failed to get os version")
-                    .is_empty()
-            );
+            //assert!(
+            //    !System::os_version()
+            //        .expect("Failed to get os version")
+            //        .is_empty()
+            //);
