Index: sysinfo/Cargo.toml
===================================================================
--- sysinfo.orig/Cargo.toml
+++ sysinfo/Cargo.toml
@@ -52,13 +52,6 @@ apple-app-store = ["apple-sandbox"]
 apple-sandbox = []
 c-interface = ["default"]
 component = [
-    "objc2-core-foundation/CFArray",
-    "objc2-core-foundation/CFBase",
-    "objc2-core-foundation/CFDictionary",
-    "objc2-core-foundation/CFNumber",
-    "objc2-core-foundation/CFString",
-    "objc2-io-kit",
-    "objc2-io-kit/hidsystem",
 ]
 debug = ["libc/extra_traits"]
 default = [
@@ -69,14 +62,6 @@ default = [
     "user",
 ]
 disk = [
-    "objc2-core-foundation/CFArray",
-    "objc2-core-foundation/CFBase",
-    "objc2-core-foundation/CFDictionary",
-    "objc2-core-foundation/CFError",
-    "objc2-core-foundation/CFNumber",
-    "objc2-core-foundation/CFString",
-    "objc2-core-foundation/CFURL",
-    "objc2-io-kit",
 ]
 linux-netdevs = []
 linux-tmpfs = []
@@ -85,11 +70,6 @@ network = [
 ]
 system = [
     "dep:memchr",
-    "objc2-core-foundation/CFBase",
-    "objc2-core-foundation/CFData",
-    "objc2-core-foundation/CFDictionary",
-    "objc2-core-foundation/CFString",
-    "objc2-io-kit",
 ]
 unknown-ci = []
 user = [
@@ -170,20 +150,6 @@ version = "1.0"
 [dev-dependencies.tempfile]
 version = "3.9"
 
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.objc2-core-foundation]
-version = "0.3.1"
-features = ["std"]
-optional = true
-default-features = false
-
-[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies.objc2-io-kit]
-version = "0.3.1"
-features = [
-    "std",
-    "libc",
-]
-optional = true
-default-features = false
 
 [target.'cfg(not(any(target_os = "unknown", target_arch = "wasm32")))'.dependencies.libc]
 version = "^0.2.173"
