Index: sysinfo/Cargo.toml
===================================================================
--- sysinfo.orig/Cargo.toml
+++ sysinfo/Cargo.toml
@@ -52,13 +52,6 @@ apple-app-store = ["apple-sandbox"]
 apple-sandbox = []
 c-interface = ["default"]
 component = [
-    "windows/Win32_Foundation",
-    "windows/Win32_Security",
-    "windows/Win32_System_Com",
-    "windows/Win32_System_Ole",
-    "windows/Win32_System_Rpc",
-    "windows/Win32_System_Variant",
-    "windows/Win32_System_Wmi",
     "objc2-core-foundation/CFArray",
     "objc2-core-foundation/CFBase",
     "objc2-core-foundation/CFDictionary",
@@ -76,13 +69,6 @@ default = [
     "user",
 ]
 disk = [
-    "windows/Win32_Foundation",
-    "windows/Win32_Storage_FileSystem",
-    "windows/Win32_Security",
-    "windows/Win32_System_IO",
-    "windows/Win32_System_Ioctl",
-    "windows/Win32_System_SystemServices",
-    "windows/Win32_System_WindowsProgramming",
     "objc2-core-foundation/CFArray",
     "objc2-core-foundation/CFBase",
     "objc2-core-foundation/CFDictionary",
@@ -96,31 +82,8 @@ linux-netdevs = []
 linux-tmpfs = []
 multithread = ["dep:rayon"]
 network = [
-    "windows/Win32_Foundation",
-    "windows/Win32_NetworkManagement_IpHelper",
-    "windows/Win32_NetworkManagement_Ndis",
-    "windows/Win32_Networking_WinSock",
 ]
 system = [
-    "windows/Win32_Foundation",
-    "windows/Win32_System_Diagnostics_ToolHelp",
-    "windows/Wdk_System_SystemInformation",
-    "windows/Wdk_System_SystemServices",
-    "windows/Wdk_System_Threading",
-    "windows/Win32_Security_Authorization",
-    "windows/Win32_System_Diagnostics_Debug",
-    "windows/Win32_System_Kernel",
-    "windows/Win32_System_Memory",
-    "windows/Win32_System_Performance",
-    "windows/Win32_System_Power",
-    "windows/Win32_System_ProcessStatus",
-    "windows/Win32_System_Registry",
-    "windows/Win32_System_RemoteDesktop",
-    "windows/Win32_System_SystemInformation",
-    "windows/Win32_System_SystemServices",
-    "windows/Win32_System_Threading",
-    "windows/Win32_UI_Shell",
-    "dep:ntapi",
     "dep:memchr",
     "objc2-core-foundation/CFBase",
     "objc2-core-foundation/CFData",
@@ -130,11 +93,6 @@ system = [
 ]
 unknown-ci = []
 user = [
-    "windows/Win32_Foundation",
-    "windows/Win32_NetworkManagement_NetManagement",
-    "windows/Win32_Security",
-    "windows/Win32_Security_Authentication_Identity",
-    "windows/Win32_Security_Authorization",
 ]
 
 [lib]
@@ -230,10 +188,4 @@ default-features = false
 [target.'cfg(not(any(target_os = "unknown", target_arch = "wasm32")))'.dependencies.libc]
 version = "^0.2.173"
 
-[target."cfg(windows)".dependencies.ntapi]
-version = "0.4"
-optional = true
 
-[target."cfg(windows)".dependencies.windows]
-version = ">=0.59, <0.62"
-optional = true
