Index: sysinfo/src/lib.rs
===================================================================
--- sysinfo.orig/src/common/system.rs
+++ sysinfo/src/common/system.rs
@@ -192,2 +192,3 @@
 mod test {
+    use std::time::Instant;
     use crate::*;
@@ -283,3 +284,5 @@ mod test {
             // Wait a bit to update CPU usage values
-            std::thread::sleep(MINIMUM_CPU_UPDATE_INTERVAL);
+            //std::thread::sleep(MINIMUM_CPU_UPDATE_INTERVAL);
+            let now = Instant::now();
+            while now.elapsed() < MINIMUM_CPU_UPDATE_INTERVAL {}
             if s.cpus().iter().any(|c| c.cpu_usage() > 0.0) {
