Index: sysinfo/tests/process.rs
===================================================================
--- sysinfo.orig/tests/process.rs
+++ sysinfo/tests/process.rs
@@ -791,7 +791,8 @@ fn test_process_run_time() {
     let current_pid = sysinfo::get_current_pid().expect("failed to get current pid");
     s.refresh_process(current_pid);
     let run_time = s.process(current_pid).expect("no process found").run_time();
-    std::thread::sleep(std::time::Duration::from_secs(2));
+    let now = std::time::Instant::now();
+    while now.elapsed() < std::time::Duration::from_secs(2) {};
     s.refresh_process(current_pid);
     let new_run_time = s.process(current_pid).expect("no process found").run_time();
     assert!(
