--- a/Cargo.toml
+++ b/Cargo.toml
@@ -115,10 +115,11 @@
 version = "3.4"
 
 [dependencies.directories]
-version = "4"
+version = "5"
 
 [dependencies.lscolors]
-version = "0.13"
+version = "0.20"
+features = ["ansi_term"]
 
 [dependencies.rayon]
 version = "1"
@@ -137,10 +138,10 @@
 version = "0.2"
 
 [dependencies.sysinfo]
-version = "0.27"
+version = "0.30"
 
 [dependencies.terminal_size]
-version = "0.2"
+version = "0.4"
 
 [dependencies.thousands]
 version = "0.2"
@@ -166,12 +167,6 @@
 [target.'cfg(not(target_has_atomic = "64"))'.dependencies.portable-atomic]
 version = "1.4"
 
-[target."cfg(windows)".dependencies.filesize]
-version = "0.2.0"
-
-[target."cfg(windows)".dependencies.winapi-util]
-version = "0.1"
-
 [profile.release]
 lto = true
 codegen-units = 1
--- a/src/main.rs
+++ b/src/main.rs
@@ -26,7 +26,7 @@
 use std::process;
 use std::sync::Arc;
 use std::sync::Mutex;
-use sysinfo::{System, SystemExt};
+use sysinfo::System;
 use utils::canonicalize_absolute_path;
 
 use self::display::draw_it;
