diff --git a/tests/redhat-v3.rs b/tests/redhat-v3.rs
index cf7e90f..28aafc0 100644
--- a/tests/redhat-v3.rs
+++ b/tests/redhat-v3.rs
@@ -27,13 +27,13 @@ fn run_tests_from_file(name: &str) {
     }
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v3_simple() {
     run_tests_from_file("vectors_simple3");
     run_tests_from_file("vectors_simple31");
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v3_random() {
     run_tests_from_file("vectors_random3");
     run_tests_from_file("vectors_random31");
diff --git a/tests/redhat-v4.rs b/tests/redhat-v4.rs
index 9508a8d..35b2c6e 100644
--- a/tests/redhat-v4.rs
+++ b/tests/redhat-v4.rs
@@ -22,37 +22,37 @@ fn run_tests_from_file(name: &str, test_serialization: bool) {
     }
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v4_base() {
     // All vector combinations with only mandatory fields, 104,976 vectors.
     run_tests_from_file("vectors_base4", true);
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v4_modified() {
     // All vector combinations of modified environmental fields, 373,248 vectors.
     run_tests_from_file("vectors_modified4", true);
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v4_supplemental() {
     // All vector combinations of supplemental fields, 576 vectors.
     run_tests_from_file("vectors_supplemental4", true);
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v4_security() {
     // All vector combinations of security fields, 54 vectors.
     run_tests_from_file("vectors_security4", true);
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v4_threat() {
     // All vector combinations of threat fields, 6 vectors.
     run_tests_from_file("vectors_threat4", true);
 }
 
-#[test]
+// #[test] disabled due to missing testdata
 fn cvss_v4_random() {
     // Random vector combinations across all fields, 10,000 vectors.
     run_tests_from_file("vectors_random4", false);
