diff --git a/tests/base.rs b/tests/base.rs
index ed25234..7af8d75 100644
--- a/tests/base.rs
+++ b/tests/base.rs
@@ -5,6 +5,7 @@
 use core::str::FromStr;
 
 /// CVE-2013-1937
+#[cfg(feature = "default")]
 #[test]
 fn cve_2013_1937() {
     let cvss_for_cve_2013_1937 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N";
@@ -14,6 +15,7 @@ fn cve_2013_1937() {
 }
 
 /// Missing CVSS:3.1 prefix
+#[cfg(feature = "default")]
 #[test]
 fn bad_prefix() {
     let cvss_for_cve_2013_1937e = "AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N";
@@ -21,6 +23,7 @@ fn bad_prefix() {
 }
 
 /// CVSS:3.0 prefix (parse these as for the purposes of this library they're identical)
+#[cfg(feature = "default")]
 #[test]
 fn cvss_v3_0_prefix() {
     let cvss_for_cve_2013_1937 = "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N";
@@ -30,6 +33,7 @@ fn cvss_v3_0_prefix() {
 }
 
 /// CVE-2013-0375
+#[cfg(feature = "default")]
 #[test]
 fn cve_2013_0375() {
     let cvss_for_cve_2013_0375 = "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:C/C:L/I:L/A:N";
@@ -39,6 +43,7 @@ fn cve_2013_0375() {
 }
 
 /// CVE-2014-3566
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_3566() {
     let cvss_for_cve_2014_3566 = "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:L/I:N/A:N";
@@ -48,6 +53,7 @@ fn cve_2014_3566() {
 }
 
 /// CVE-2012-1516
+#[cfg(feature = "default")]
 #[test]
 fn cve_2012_1516() {
     let cvss_for_cve_2012_1516 = "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:C/C:H/I:H/A:H";
@@ -57,6 +63,7 @@ fn cve_2012_1516() {
 }
 
 /// CVE-2009-0783
+#[cfg(feature = "default")]
 #[test]
 fn cve_2009_0783() {
     let cvss_for_cve_2009_0783 = "CVSS:3.1/AV:L/AC:L/PR:H/UI:N/S:U/C:L/I:L/A:L";
@@ -66,6 +73,7 @@ fn cve_2009_0783() {
 }
 
 /// CVE-2012-0384
+#[cfg(feature = "default")]
 #[test]
 fn cve_2012_0384() {
     let cvss_for_cve_2012_0384 = "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H";
@@ -75,6 +83,7 @@ fn cve_2012_0384() {
 }
 
 /// CVE-2015-1098
+#[cfg(feature = "default")]
 #[test]
 fn cve_2015_1098() {
     let cvss_for_cve_2015_1098 = "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H";
@@ -84,6 +93,7 @@ fn cve_2015_1098() {
 }
 
 /// CVE-2014-0160
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_0160() {
     let cvss_for_cve_2014_0160 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N";
@@ -93,6 +103,7 @@ fn cve_2014_0160() {
 }
 
 /// CVE-2014-6271
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_6271() {
     let cvss_for_cve_2014_6271 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H";
@@ -102,6 +113,7 @@ fn cve_2014_6271() {
 }
 
 /// CVE-2008-1447
+#[cfg(feature = "default")]
 #[test]
 fn cve_2008_1447() {
     let cvss_for_cve_2008_1447 = "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:C/C:N/I:H/A:N";
@@ -111,6 +123,7 @@ fn cve_2008_1447() {
 }
 
 /// CVE-2014-2005
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_2005() {
     let cvss_for_cve_2014_2005 = "CVSS:3.1/AV:P/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H";
@@ -120,6 +133,7 @@ fn cve_2014_2005() {
 }
 
 /// CVE-2010-0467
+#[cfg(feature = "default")]
 #[test]
 fn cve_2010_0467() {
     let cvss_for_cve_2010_0467 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:L/I:N/A:N";
@@ -129,6 +143,7 @@ fn cve_2010_0467() {
 }
 
 /// CVE-2012-1342
+#[cfg(feature = "default")]
 #[test]
 fn cve_2012_1342() {
     let cvss_for_cve_2012_1342 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:N/I:L/A:N";
@@ -138,6 +153,7 @@ fn cve_2012_1342() {
 }
 
 /// CVE-2013-6014
+#[cfg(feature = "default")]
 #[test]
 fn cve_2013_6014() {
     let cvss_for_cve_2013_6014 = "CVSS:3.1/AV:A/AC:L/PR:N/UI:N/S:C/C:H/I:N/A:H";
@@ -147,6 +163,7 @@ fn cve_2013_6014() {
 }
 
 /// CVE-2014-9253
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_9253() {
     let cvss_for_cve_2014_9253 = "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:C/C:L/I:L/A:N";
@@ -156,6 +173,7 @@ fn cve_2014_9253() {
 }
 
 /// CVE-2009-0658
+#[cfg(feature = "default")]
 #[test]
 fn cve_2009_0658() {
     let cvss_for_cve_2009_0658 = "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H";
@@ -165,6 +183,7 @@ fn cve_2009_0658() {
 }
 
 /// CVE-2011-1265
+#[cfg(feature = "default")]
 #[test]
 fn cve_2011_1265() {
     let cvss_for_cve_2011_1265 = "CVSS:3.1/AV:A/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H";
@@ -174,6 +193,7 @@ fn cve_2011_1265() {
 }
 
 /// CVE-2014-2019
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_2019() {
     let cvss_for_cve_2014_2019 = "CVSS:3.1/AV:P/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N";
@@ -183,6 +203,7 @@ fn cve_2014_2019() {
 }
 
 /// CVE-2015-0970
+#[cfg(feature = "default")]
 #[test]
 fn cve_2015_0970() {
     let cvss_for_cve_2015_0970 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H";
@@ -192,6 +213,7 @@ fn cve_2015_0970() {
 }
 
 /// CVE-2014-0224
+#[cfg(feature = "default")]
 #[test]
 fn cve_2014_0224() {
     let cvss_for_cve_2014_0224 = "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:N";
@@ -201,6 +223,7 @@ fn cve_2014_0224() {
 }
 
 /// CVE-2012-5376
+#[cfg(feature = "default")]
 #[test]
 fn cve_2012_5376() {
     let cvss_for_cve_2012_5376 = "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H";
@@ -210,6 +233,7 @@ fn cve_2012_5376() {
 }
 
 /// No impact scope changed
+#[cfg(feature = "default")]
 #[test]
 fn no_impact_scope_changed() {
     // https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:N/I:N/A:N => 0.0
@@ -220,6 +244,7 @@ fn no_impact_scope_changed() {
 }
 
 /// No impact scope unchanged
+#[cfg(feature = "default")]
 #[test]
 fn no_impact_scope_unchanged() {
     // https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N => 0.0
@@ -229,6 +254,7 @@ fn no_impact_scope_unchanged() {
     assert_eq!(base.score().value(), 0.0);
 }
 
+#[cfg(feature = "default")]
 #[test]
 fn low_scope_unchanged() {
     // https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:P/AC:H/PR:H/UI:R/S:U/C:N/I:N/A:L => 1.6
@@ -238,6 +264,7 @@ fn low_scope_unchanged() {
     assert_eq!(base.score().value(), 1.6);
 }
 
+#[cfg(feature = "default")]
 #[test]
 fn low_scope_changed() {
     // https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:P/AC:H/PR:H/UI:R/S:C/C:N/I:N/A:L => 1.8
