--- a/build.rs
+++ b/build.rs
@@ -32,8 +32,8 @@
 fn main() {
     println!("cargo:rerun-if-env-changed=OUCH_ARTIFACTS_FOLDER");
 
-    if let Some(dir) = env::var_os("OUCH_ARTIFACTS_FOLDER") {
-        let out = &Path::new(&dir);
+    //if let Some(dir) = env::var_os("OUCH_ARTIFACTS_FOLDER") {
+        let out = &Path::new("artifacts");
         create_dir_all(out).unwrap();
         let cmd = &mut Opts::command();
 
@@ -51,5 +51,5 @@
         for shell in Shell::value_variants() {
             generate_to(*shell, cmd, "ouch", out).unwrap();
         }
-    }
+    //}
 }
