Index: num-traits/src/identities.rs
===================================================================
--- num-traits.orig/src/identities.rs
+++ num-traits/src/identities.rs
@@ -5,7 +5,7 @@ use core::ops::{Add, Mul};
 ///
 /// # Laws
 ///
-/// ```{.text}
+/// ```text
 /// a + 0 = a       ∀ a ∈ Self
 /// 0 + a = a       ∀ a ∈ Self
 /// ```
@@ -83,7 +83,7 @@ where
 ///
 /// # Laws
 ///
-/// ```{.text}
+/// ```text
 /// a * 1 = a       ∀ a ∈ Self
 /// 1 * a = a       ∀ a ∈ Self
 /// ```
