diff --git a/Cargo.toml b/Cargo.toml
index 090d32f..a3960fc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -165,7 +165,6 @@ cmd = [
     "dep:os_pipe",
     "dep:wait-timeout",
     "dep:libc",
-    "dep:windows-sys",
 ]
 color = [
     "dep:anstream",
@@ -204,7 +203,3 @@ structured-data = ["dep:serde_json"]
 version = "0.2.137"
 optional = true
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = ["Win32_Foundation"]
-optional = true
