--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,4 +70,0 @@
-client-proxy-system = [
-    "dep:system-configuration",
-    "dep:windows-registry",
-]
@@ -79,1 +79,0 @@
-    "client-proxy-system",
@@ -243,1 +243,1 @@
-[target.'cfg(target_os = "macos")'.dependencies.system-configuration]
+[disabled.target.'cfg(target_os = "macos")'.dependencies.system-configuration]
@@ -247,1 +247,1 @@
-[target."cfg(windows)".dependencies.windows-registry]
+[disabled.target."cfg(windows)".dependencies.windows-registry]
