Index: b/Cargo.toml
===================================================================
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,16 +58,8 @@
 name = "pyo3_build_config"
 path = "src/lib.rs"
 
-[dependencies.python3-dll-a]
-version = "0.2.12"
-optional = true
-
 [dependencies.target-lexicon]
 version = "0.13"
 
-[build-dependencies.python3-dll-a]
-version = "0.2.12"
-optional = true
-
 [build-dependencies.target-lexicon]
 version = "0.13"
