--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67 +67 @@
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.js-sys]
+[disabled.target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.js-sys]
@@ -70 +70 @@
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.log]
+[disabled.target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.log]
@@ -73 +73 @@
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.wasm-bindgen]
+[disabled.target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.wasm-bindgen]
@@ -76 +76 @@
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dev-dependencies.getrandom]
+[disabled.target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dev-dependencies.getrandom]
@@ -80 +80 @@
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dev-dependencies.wasm-bindgen-test]
+[disabled.target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dev-dependencies.wasm-bindgen-test]
@@ -83 +83 @@
-[target.'cfg(target_os = "android")'.dependencies.android_system_properties]
+[disabled.target.'cfg(target_os = "android")'.dependencies.android_system_properties]
@@ -86 +86 @@
-[target.'cfg(target_os = "haiku")'.dependencies.iana-time-zone-haiku]
+[disabled.target.'cfg(target_os = "haiku")'.dependencies.iana-time-zone-haiku]
@@ -89 +89 @@
-[target.'cfg(target_os = "windows")'.dependencies.windows-core]
+[disabled.target.'cfg(target_os = "windows")'.dependencies.windows-core]
@@ -92 +92 @@
-[target.'cfg(target_vendor = "apple")'.dependencies.core-foundation-sys]
+[disabled.target.'cfg(target_vendor = "apple")'.dependencies.core-foundation-sys]
