diff --git a/Cargo.toml b/Cargo.toml
index c8dac2e..8e57059 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,8 +28,8 @@ gpg-error = "1.12"
 [build-dependencies.system-deps]
 version = "6.2.2"
 
-[target."cfg(windows)".build-dependencies.build-rs]
-version = "0.1.2"
+#[target."cfg(windows)".build-dependencies.build-rs]
+#version = "0.1.2"
 
-[target."cfg(windows)".build-dependencies.winreg]
-version = "0.52.0"
+#[target."cfg(windows)".build-dependencies.winreg]
+#version = "0.52.0"
