diff --git a/Cargo.toml b/Cargo.toml
index c8dac2e..8e57059 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,11 +28,11 @@ gpg-error = "1.12"
 [build-dependencies.system-deps]
 version = "6.0.2"
 
-[target."cfg(windows)".build-dependencies.build-rs]
-version = "0.1.2"
+#[target."cfg(windows)".build-dependencies.build-rs]
+#version = "0.1.2"
 
-[target."cfg(windows)".build-dependencies.winreg]
-version = "0.10.1"
+#[target."cfg(windows)".build-dependencies.winreg]
+#version = "0.10.1"
 
 [badges.maintenance]
 status = "experimental"
