--- a/Cargo.toml
+++ b/Cargo.toml
@@ -145,14 +145,14 @@
     "flate2?/rust_backend",
     "dep:thiserror",
 ]
-zlib-ng = [
-    "zlib",
-    "flate2?/zlib-ng",
-]
-zlib-ng-compat = [
-    "zlib",
-    "flate2?/zlib-ng-compat",
-]
+#zlib-ng = [
+#    "zlib",
+#    "flate2?/zlib-ng",
+#]
+#zlib-ng-compat = [
+#    "zlib",
+#    "flate2?/zlib-ng-compat",
+#]
 zlib-rust-backend = [
     "zlib",
     "flate2?/rust_backend",
