upstream: https://github.com/Byron/gitoxide/pull/1134

diff --git a/tests/hash.rs b/tests/hash.rs
index c8e48da..ecc321b 100644
--- a/tests/hash.rs
+++ b/tests/hash.rs
@@ -9,5 +9,5 @@ fn size_of_sha1() {
 #[cfg(feature = "fast-sha1")]
 #[test]
 fn size_of_sha1() {
-    assert_eq!(std::mem::size_of::<Sha1>(), 104)
+    assert_eq!(std::mem::size_of::<Sha1>(), if cfg!(target_arch = "x86") { 96 } else { 104 })
 }
