Index: parsec-interface/Cargo.toml
===================================================================
--- parsec-interface.orig/Cargo.toml
+++ parsec-interface/Cargo.toml
@@ -43,20 +43,20 @@ version = "0.4.11"
 version = "0.4.0"
 
 [dependencies.num-derive]
-version = "0.3.2"
+version = "0.3.0"
 
 [dependencies.num-traits]
 version = "0.2.12"
 
 [dependencies.prost]
-version = "0.8.0"
+version = "0.11.0"
 
 [dependencies.psa-crypto]
-version = "0.10.0"
+version = "0.9.0"
 default-features = false
 
 [dependencies.secrecy]
-version = "0.7.0"
+version = "0.8.0"
 features = ["serde"]
 
 [dependencies.serde]
@@ -64,7 +64,7 @@ version = "1.0.115"
 features = ["derive"]
 
 [dependencies.uuid]
-version = "0.8.1"
+version = "1.0.0"
 
 [dependencies.zeroize]
 version = "1.1.0"
