--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,7 +31,7 @@
 proc-macro = true
 
 [dependencies.proc-macro-crate]
-version = "3.4.0"
+version = ">=3.2.0, <4"
 default-features = false
 
 [dependencies.proc-macro-error2]
@@ -39,15 +39,15 @@
 default-features = false
 
 [dependencies.proc-macro2]
-version = "1.0.103"
+version = ">=1.0.101, <2"
 default-features = false
 
 [dependencies.quote]
-version = "1.0.41"
+version = ">=1.0.40, <2"
 default-features = false
 
 [dependencies.syn]
-version = "2.0.108"
+version = ">=2.0.105, <3"
 features = [
     "derive",
     "parsing",
