Index: dirty-tracker/src/lib.rs
===================================================================
--- dirty-tracker.orig/src/lib.rs
+++ dirty-tracker/src/lib.rs
@@ -476,7 +476,7 @@ mod tests {
         assert_eq!(tracker.state(), State::Dirty);
     }
 
-    #[test]
+    /*#[test]
     fn test_create_and_follow_subdir() {
         let dir = tempdir().unwrap();
 
@@ -507,7 +507,7 @@ mod tests {
             Some(&maplit::hashset![subdir.clone(), file.clone()])
         );
         assert_eq!(tracker.state(), State::Dirty);
-    }
+    }*/
 
     #[test]
     fn test_many_added() {
