--- rust-tracing-subscriber-0.3.18.orig/Cargo.toml
+++ rust-tracing-subscriber-0.3.18/Cargo.toml
@@ -76,7 +76,7 @@ optional = true
 default-features = false
 
 [dependencies.matchers]
-version = "0.1.0"
+version = "0.2.0"
 optional = true
 
 [dependencies.nu-ansi-term]
--- rust-tracing-subscriber-0.3.18.orig/src/filter/env/field.rs
+++ rust-tracing-subscriber-0.3.18/src/filter/env/field.rs
@@ -234,7 +234,7 @@ impl ValueMatch {
     /// This returns an error if the string didn't contain a valid `bool`,
     /// `u64`, `i64`, or `f64` literal, and couldn't be parsed as a regular
     /// expression.
-    fn parse_regex(s: &str) -> Result<Self, matchers::Error> {
+    fn parse_regex(s: &str) -> Result<Self, matchers::BuildError> {
         s.parse::<bool>()
             .map(ValueMatch::Bool)
             .or_else(|_| s.parse::<u64>().map(ValueMatch::U64))
@@ -279,7 +279,7 @@ impl fmt::Display for ValueMatch {
 // === impl MatchPattern ===
 
 impl FromStr for MatchPattern {
-    type Err = matchers::Error;
+    type Err = matchers::BuildError;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let matcher = s.parse::<Pattern>()?;
         Ok(Self {
