Index: tracing-subscriber/Cargo.toml
===================================================================
--- tracing-subscriber.orig/Cargo.toml
+++ tracing-subscriber/Cargo.toml
@@ -48,18 +48,22 @@ rustdoc-args = [
 [[bench]]
 name = "filter"
 harness = false
+required-features = ["env-filter", "std"]
 
 [[bench]]
 name = "filter_log"
 harness = false
+required-features = ["env-filter", "std"]
 
 [[bench]]
 name = "fmt"
 harness = false
+required-features = ["fmt", "std"]
 
 [[bench]]
 name = "enter"
 harness = false
+required-features = ["fmt", "std"]
 
 [dependencies.matchers]
 version = "0.1.0"
