Index: tracing-subscriber/Cargo.toml
===================================================================
--- tracing-subscriber.orig/Cargo.toml
+++ tracing-subscriber/Cargo.toml
@@ -65,10 +65,6 @@ name = "enter"
 harness = false
 required-features = ["fmt", "std"]
 
-[dependencies.matchers]
-version = "0.1.0"
-optional = true
-
 [dependencies.nu-ansi-term]
 version = "0.48.0"
 optional = true
@@ -191,14 +187,6 @@ default = [
     "tracing-log",
     "std",
 ]
-env-filter = [
-    "matchers",
-    "regex",
-    "once_cell",
-    "tracing",
-    "std",
-    "thread_local",
-]
 fmt = [
     "registry",
     "std",
