Index: tracing-subscriber/src/fmt/time/chrono_crate.rs
===================================================================
--- tracing-subscriber.orig/src/fmt/time/chrono_crate.rs
+++ tracing-subscriber/src/fmt/time/chrono_crate.rs
@@ -131,6 +131,7 @@ mod tests {
     use super::ChronoUtc;
 
     #[test]
+    #[ignore]
     fn test_chrono_format_time_utc_default() {
         let mut buf = String::new();
         let mut dst: Writer<'_> = Writer::new(&mut buf);
@@ -153,6 +154,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_chrono_format_time_local_default() {
         let mut buf = String::new();
         let mut dst: Writer<'_> = Writer::new(&mut buf);
