Index: opam-file-rs/Cargo.toml
===================================================================
--- opam-file-rs.orig/Cargo.toml
+++ opam-file-rs/Cargo.toml
@@ -24,9 +24,9 @@ categories = ["encoding", "parser-implem
 license = "MIT"
 repository = "https://github.com/puripuri2100/opam-file-rs"
 [dependencies.lalrpop-util]
-version = "0.19.4"
+version = "0.20"
 
 [dependencies.thiserror]
 version = "1.0.23"
 [build-dependencies.lalrpop]
-version = "0.19.4"
+version = "0.20"
