diff --git a/Cargo.toml b/Cargo.toml
index 031f45b..672122e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,7 +21,7 @@ include = [
     "LICENSE",
 ]
 description = "CLI and Rust library for caching subprocess invocations"
-homepage = "http://bkt.rs/"
+homepage = "https://bkt.rs/"
 readme = "README.md"
 keywords = [
     "cache",
@@ -35,7 +35,7 @@ categories = [
     "command-line-utilities",
 ]
 license = "MIT"
-repository = "http://github.com/dimo414/bkt"
+repository = "https://github.com/dimo414/bkt"
 
 [dependencies.anyhow]
 version = "1.0"
diff --git a/src/lib.rs b/src/lib.rs
index 10d9c09..c95370a 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -8,7 +8,7 @@
 //! # fn main() -> anyhow::Result<()> {
 //! # use std::time::Duration;
 //! let bkt = bkt::Bkt::in_tmp()?;
-//! let expensive_cmd = bkt::CommandDesc::new(["wget", "http://example.com"]);
+//! let expensive_cmd = bkt::CommandDesc::new(["wget", "https://example.com"]);
 //! let (result, age) = bkt.retrieve(&expensive_cmd, Duration::from_secs(3600))?;
 //! do_something(result.stdout_utf8());
 //! # Ok(()) }
@@ -974,7 +974,7 @@ impl CacheStatus {
 /// # fn main() -> anyhow::Result<()> {
 /// # use std::time::Duration;
 /// let bkt = bkt::Bkt::in_tmp()?;
-/// let cmd = bkt::CommandDesc::new(["curl", "http://expensive.api/foo"]);
+/// let cmd = bkt::CommandDesc::new(["curl", "https://expensive.api/foo"]);
 /// let (result, age) = bkt.retrieve(&cmd, Duration::from_secs(60*60))?;
 /// println!("Retrieved: {:?}\nAge: {:?}", result, age);
 /// # Ok(()) }
