--- a/Cargo.toml
+++ b/Cargo.toml
@@ -87,26 +87,3 @@
 version = "0.2.1"
 features = ["support-v0_6"]
 
-[target."cfg(windows)".dependencies.mio-anonymous-pipes]
-version = "0.2"
-
-[target."cfg(windows)".dependencies.miow]
-version = "0.3"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.7"
-features = [
-    "impl-default",
-    "basetsd",
-    "libloaderapi",
-    "minwindef",
-    "ntdef",
-    "processthreadsapi",
-    "winbase",
-    "wincon",
-    "wincontypes",
-    "winerror",
-    "winnt",
-    "winuser",
-    "consoleapi",
-]
