Index: alacritty-terminal/Cargo.toml
===================================================================
--- alacritty-terminal.orig/Cargo.toml
+++ alacritty-terminal/Cargo.toml
@@ -31,13 +31,13 @@ version = "0.1.1"
 version = "0.2.1"
 
 [dependencies.base64]
-version = "0.13.0"
+version = "0.21.0"
 
 [dependencies.bitflags]
 version = "1"
 
 [dependencies.dirs]
-version = "4.0.0"
+version = ">= 4.0, < 6.0"
 
 [dependencies.libc]
 version = "0.2"
@@ -71,14 +71,14 @@ version = "0.8"
 version = "0.1"
 
 [dependencies.vte]
-version = "0.10.0"
+version = "0.11.0"
 default-features = false
 
 [dev-dependencies.serde_json]
 version = "1.0.0"
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.24.1"
+version = "0.26"
 
 [target."cfg(unix)".dependencies.signal-hook]
 version = "0.3.10"
