--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,16 +37,16 @@
 version = "0.4"
 
 [dependencies.mio]
-version = "0.6.20"
+version = "0.6"
 
 [dependencies.mio-extras]
 version = "2"
 
 [dependencies.parking_lot]
-version = "0.11.0"
+version = "0.10"
 
 [dependencies.regex-automata]
-version = "0.1.9"
+version = "0.1.8"
 
 [dependencies.serde]
 version = "1"
@@ -64,7 +64,7 @@
 [dev-dependencies.serde_json]
 version = "1.0.0"
 [target."cfg(unix)".dependencies.nix]
-version = "0.22.0"
+version = "0.19.0"
 
 [target."cfg(unix)".dependencies.signal-hook]
 version = "0.1"
