diff --git a/Cargo.toml b/Cargo.toml
index 93aec92..98050bc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -109,7 +109,6 @@ js = [
     "dep:wasm-bindgen",
     "dep:js-sys",
 ]
-macro-diagnostics = ["dep:uuid-macro-internal"]
 md5 = ["dep:md-5"]
 rng = ["dep:getrandom"]
 rng-getrandom = [
@@ -184,10 +183,6 @@ default-features = false
 version = "2"
 optional = true
 
-[dependencies.uuid-macro-internal]
-version = "1.18.1"
-optional = true
-
 [dependencies.zerocopy]
 version = "0.8"
 features = ["derive"]
