--- a/Cargo.toml
+++ b/Cargo.toml
@@ -63,7 +63,6 @@
     "serde",
     "arbitrary",
     "slog",
-    "borsh",
     "v1",
     "v3",
     "v4",
@@ -96,10 +95,6 @@
 
 [features]
 atomic = ["dep:atomic"]
-borsh = [
-    "dep:borsh",
-    "dep:borsh-derive",
-]
 default = ["std"]
 fast-rng = [
     "rng",
@@ -142,15 +137,6 @@
 optional = true
 default-features = false
 
-[dependencies.borsh]
-version = "1"
-optional = true
-default-features = false
-
-[dependencies.borsh-derive]
-version = "1"
-optional = true
-default-features = false
 
 [dependencies.bytemuck]
 version = "1.18.1"
