diff --git a/Cargo.toml b/Cargo.toml
index 1c4b767..bce1f44 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -63,7 +63,6 @@ features = [
     "serde",
     "arbitrary",
     "slog",
-    "borsh",
     "v1",
     "v3",
     "v4",
@@ -96,10 +95,6 @@ status = "actively-developed"
 
 [features]
 atomic = ["dep:atomic"]
-borsh = [
-    "dep:borsh",
-    "dep:borsh-derive",
-]
 default = ["std"]
 fast-rng = [
     "rng",
@@ -145,16 +140,6 @@ version = "0.6"
 optional = true
 default-features = false
 
-[dependencies.borsh]
-version = "1"
-optional = true
-default-features = false
-
-[dependencies.borsh-derive]
-version = "1"
-optional = true
-default-features = false
-
 [dependencies.bytemuck]
 version = "1.18.1"
 features = ["derive"]
