Index: uuid/Cargo.toml
===================================================================
--- uuid.orig/Cargo.toml
+++ uuid/Cargo.toml
@@ -105,14 +105,10 @@ rng = ["dep:getrandom"]
 rng-getrandom = [
     "rng",
     "dep:getrandom",
-    "uuid-rng-internal-lib",
-    "uuid-rng-internal-lib/getrandom",
 ]
 rng-rand = [
     "rng",
     "dep:rand",
-    "uuid-rng-internal-lib",
-    "uuid-rng-internal-lib/rand",
 ]
 sha1 = ["dep:sha1"]
 std = []
@@ -194,11 +190,6 @@ version = "1.0.56"
 [dev-dependencies.trybuild]
 version = "1.0.52"
 
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.uuid-rng-internal-lib]
-version = "1.18.0"
-optional = true
-package = "uuid-rng-internal"
-
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.getrandom]
 version = "0.3"
 optional = true
