--- a/Cargo.toml
+++ b/Cargo.toml
@@ -63,7 +63,6 @@
     "serde",
     "arbitrary",
     "slog",
-    "borsh",
     "v1",
     "v3",
     "v4",
@@ -96,4 +95,0 @@
-borsh = [
-    "dep:borsh",
-    "dep:borsh-derive",
-]
@@ -141,15 +136,6 @@
 optional = true
 default-features = false
 
-[dependencies.borsh]
-version = "1"
-optional = true
-default-features = false
-
-[dependencies.borsh-derive]
-version = "1"
-optional = true
-default-features = false
 
 [dependencies.bytemuck]
 version = "1.18.0"
