--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,6 @@
     "serde",
     "arbitrary",
     "slog",
-    "borsh",
     "v1",
     "v3",
     "v4",
@@ -97,16 +96,6 @@
 optional = true
 default-features = false
 
-[dependencies.borsh]
-version = "1"
-optional = true
-default-features = false
-
-[dependencies.borsh-derive]
-version = "1"
-optional = true
-default-features = false
-
 [dependencies.bytemuck]
 version = "1.14.0"
 features = ["derive"]
@@ -164,10 +153,6 @@
 
 [features]
 atomic = ["dep:atomic"]
-borsh = [
-    "dep:borsh",
-    "dep:borsh-derive",
-]
 default = ["std"]
 fast-rng = [
     "rng",
