--- a/Cargo.toml
+++ b/Cargo.toml
@@ -100,15 +100,11 @@
     "dep:borsh",
     "dep:borsh-derive",
 ]
-default = ["std"]
+default = []
 fast-rng = [
     "rng",
     "dep:rand",
 ]
-js = [
-    "dep:wasm-bindgen",
-    "dep:js-sys",
-]
 macro-diagnostics = ["dep:uuid-macro-internal"]
 md5 = ["dep:md-5"]
 rng = ["dep:getrandom"]
@@ -125,10 +121,6 @@
     "uuid-rng-internal-lib/rand",
 ]
 sha1 = ["dep:sha1_smol"]
-std = [
-    "wasm-bindgen?/std",
-    "js-sys?/std",
-]
 v1 = ["atomic"]
 v3 = ["md5"]
 v4 = ["rng"]
@@ -216,23 +216,6 @@
 optional = true
 package = "uuid-rng-internal"
 
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.wasm-bindgen]
-version = "0.2"
-features = ["msrv"]
-optional = true
-default-features = false
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dev-dependencies.wasm-bindgen]
-version = "0.2"
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"), target_feature = "atomics"))'.dependencies.js-sys]
-version = "0.3"
-optional = true
-default-features = false
-
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.getrandom]
 version = "0.3"
 optional = true

