--- a/Cargo.toml
+++ b/Cargo.toml
@@ -181,11 +181,11 @@
 version = "1.0.52"
 
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.getrandom]
-version = "0.3"
+version = "0.2"
 optional = true
 
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.rand]
-version = "0.9"
+version = ">= 0.8, < 0.10"
 optional = true
 
 [lints.rust.unexpected_cfgs]
--- a/src/rng.rs
+++ b/src/rng.rs
@@ -54,7 +54,7 @@
         fn u128() -> u128 {
             let mut bytes = [0u8; 16];
 
-            getrandom::fill(&mut bytes).unwrap_or_else(|err| {
+            getrandom::getrandom(&mut bytes).unwrap_or_else(|err| {
                 // NB: getrandom::Error has no source; this is adequate display
                 panic!("could not retrieve random bytes for uuid: {}", err)
             });
@@ -65,7 +65,7 @@
         fn u64() -> u64 {
             let mut bytes = [0u8; 8];
 
-            getrandom::fill(&mut bytes).unwrap_or_else(|err| {
+            getrandom::getrandom(&mut bytes).unwrap_or_else(|err| {
                 // NB: getrandom::Error has no source; this is adequate display
                 panic!("could not retrieve random bytes for uuid: {}", err)
             });
@@ -76,7 +76,7 @@
         fn u16() -> u16 {
             let mut bytes = [0u8; 2];
 
-            getrandom::fill(&mut bytes).unwrap_or_else(|err| {
+            getrandom::getrandom(&mut bytes).unwrap_or_else(|err| {
                 // NB: getrandom::Error has no source; this is adequate display
                 panic!("could not retrieve random bytes for uuid: {}", err)
             });
