Index: uuid/Cargo.toml
===================================================================
--- uuid.orig/Cargo.toml
+++ uuid/Cargo.toml
@@ -146,6 +146,6 @@ v3 = ["md5"]
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen]
-version = "0.2"
-package = "wasm-bindgen"
+# [target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen]
+# version = "0.2"
+# package = "wasm-bindgen"
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+# [target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
+# version = "0.3"
