Index: uuid/Cargo.toml
===================================================================
--- uuid.orig/Cargo.toml
+++ uuid/Cargo.toml
@@ -188,3 +188,3 @@ v8 = []
-[target."cfg(all(target_arch = \"wasm32\", target_vendor = \"unknown\", target_os = \"unknown\"))".dependencies.wasm-bindgen]
-version = "0.2"
-optional = true
+#[target."cfg(all(target_arch = \"wasm32\", target_vendor = \"unknown\", target_os = \"unknown\"))".dependencies.wasm-bindgen]
+#version = "0.2"
+#optional = true
@@ -192,2 +192,2 @@ optional = true
-[target."cfg(all(target_arch = \"wasm32\", target_vendor = \"unknown\", target_os = \"unknown\"))".dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+#[target."cfg(all(target_arch = \"wasm32\", target_vendor = \"unknown\", target_os = \"unknown\"))".dev-dependencies.wasm-bindgen-test]
+#version = "0.3"
@@ -195,2 +195,2 @@ version = "0.3"
-[target."cfg(target = \"wasm32-unknown-unknown\")".dev-dependencies.wasm-bindgen]
-version = "0.2"
+#[target."cfg(target = \"wasm32-unknown-unknown\")".dev-dependencies.wasm-bindgen]
+#version = "0.2"
