Index: uuid/Cargo.toml
===================================================================
--- uuid.orig/Cargo.toml
+++ uuid/Cargo.toml
@@ -146,12 +146,12 @@ v3 = ["md5"]
 v4 = ["rng"]
 v5 = ["sha1"]
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-lib]
-version = "0.2"
-package = "wasm-bindgen"
+# [target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-lib]
+# version = "0.2"
+# package = "wasm-bindgen"
 
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3"
+# [target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
+# version = "0.3"
 
 [target."cfg(windows)".dev-dependencies.winapi]
 version = "0.3"
