Index: uuid/Cargo.toml
===================================================================
--- uuid.orig/Cargo.toml
+++ uuid/Cargo.toml
@@ -111,7 +111,7 @@ optional = true
 default-features = false
 
 [dependencies.rand]
-version = "0.8"
+version = ">= 0.8, < 0.10"
 optional = true
 
 [dependencies.serde]
@@ -169,15 +169,6 @@ v6 = ["atomic"]
 v7 = ["rng"]
 v8 = []
 
-[target.'cfg(all(target_arch = "wasm32", target_vendor = "unknown", target_os = "unknown"))'.dependencies.wasm-bindgen]
-version = "0.2"
-optional = true
-
-[target.'cfg(all(target_arch = "wasm32", target_vendor = "unknown", target_os = "unknown"))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
-
-[target.'cfg(target = "wasm32-unknown-unknown")'.dev-dependencies.wasm-bindgen]
-version = "0.2"
 
 [badges.is-it-maintained-issue-resolution]
 repository = "uuid-rs/uuid"
