--- a/Cargo.toml
+++ b/Cargo.toml
@@ -194,15 +194,6 @@
 v7 = ["rng"]
 v8 = []
 
-[target.'cfg(all(target_arch = "wasm32", target_vendor = "unknown", target_os = "unknown"))'.dependencies.wasm-bindgen]
-version = "0.2"
-optional = true
-
-[target.'cfg(all(target_arch = "wasm32", target_vendor = "unknown", target_os = "unknown"))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
-
-[target.'cfg(target = "wasm32-unknown-unknown")'.dev-dependencies.wasm-bindgen]
-version = "0.2"
 
 [badges.is-it-maintained-issue-resolution]
 repository = "uuid-rs/uuid"
