--- a/Cargo.toml
+++ b/Cargo.toml
@@ -181,11 +181,11 @@
 version = "1.0.52"
 
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.getrandom]
-version = "0.3"
+version = "0.2"
 optional = true
 
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.rand]
-version = "0.9"
+version = ">= 0.8, < 0.10"
 optional = true
 
 [lints.rust.unexpected_cfgs]
