Index: clap/Cargo.toml
===================================================================
--- clap.orig/Cargo.toml
+++ clap/Cargo.toml
@@ -41,15 +41,11 @@ cargo-args = [
     "-Zunstable-options",
     "-Zrustdoc-scrape-examples",
 ]
-features = ["unstable-doc"]
 rustdoc-args = [
     "--cfg",
     "docsrs",
 ]
 
-[package.metadata.playground]
-features = ["unstable-doc"]
-
 [package.metadata.release]
 shared-version = true
 tag-name = "v{{version}}"
@@ -428,10 +424,6 @@ std = ["clap_builder/std"]
 string = ["clap_builder/string"]
 suggestions = ["clap_builder/suggestions"]
 unicode = ["clap_builder/unicode"]
-unstable-doc = [
-    "clap_builder/unstable-doc",
-    "derive",
-]
 unstable-styles = ["clap_builder/unstable-styles"]
 unstable-v5 = [
     "clap_builder/unstable-v5",
