Index: clap/Cargo.toml
===================================================================
--- clap.orig/Cargo.toml
+++ clap/Cargo.toml
@@ -319,22 +319,6 @@ name = "busybox"
 path = "examples/multicall-busybox.rs"
 
 [[example]]
-name = "cargo-example"
-path = "examples/cargo-example.rs"
-required-features = [
-    "cargo",
-    "color",
-]
-
-[[example]]
-name = "cargo-example-derive"
-path = "examples/cargo-example-derive.rs"
-required-features = [
-    "derive",
-    "color",
-]
-
-[[example]]
 name = "demo"
 path = "examples/demo.rs"
 required-features = ["derive"]
@@ -417,10 +401,6 @@ optional = true
 [dev-dependencies.automod]
 version = "1.0.14"
 
-[dev-dependencies.clap-cargo]
-version = "0.14.1"
-default-features = false
-
 [dev-dependencies.humantime]
 version = "2.1.0"
 
