Index: clap/Cargo.toml
===================================================================
--- clap.orig/Cargo.toml
+++ clap/Cargo.toml
@@ -319,8 +319,0 @@
-[[example]]
-name = "cargo-example"
-path = "examples/cargo-example.rs"
-doc-scrape-examples = true
-required-features = [
-    "cargo",
-    "color",
-]
@@ -330,8 +330,0 @@
-[[example]]
-name = "cargo-example-derive"
-path = "examples/cargo-example-derive.rs"
-doc-scrape-examples = true
-required-features = [
-    "derive",
-    "color",
-]
@@ -420,4 +404,0 @@
-[dev-dependencies.clap-cargo]
-version = "0.15.0"
-default-features = false
-
