--- a/Cargo.toml
+++ b/Cargo.toml
@@ -354,7 +354,6 @@
 
 [dependencies.once_cell]
 version = "1.12.0"
-optional = true
 
 [dependencies.regex]
 version = "1.0"
@@ -404,7 +403,7 @@
 version = "1.0.18"
 
 [features]
-cargo = ["once_cell"]
+cargo = []
 color = [
     "atty",
     "termcolor",
@@ -420,8 +419,7 @@
 ]
 deprecated = []
 derive = [
-    "clap_derive",
-    "once_cell",
+    "clap_derive"
 ]
 env = []
 std = ["indexmap/std"]
