Index: clap/Cargo.toml
===================================================================
--- clap.orig/Cargo.toml
+++ clap/Cargo.toml
@@ -494,7 +494,7 @@ version = "1.0.14"
 version = "0.2.3"
 
 [dev-dependencies.rustversion]
-version = "1.0.15"
+version = "1.0.14"
 
 [dev-dependencies.semver]
 version = "1.0.26"
@@ -503,14 +503,14 @@ version = "1.0.26"
 version = "1.3.0"
 
 [dev-dependencies.snapbox]
-version = "0.6.16"
+version = ">= 0.4.16, < 0.7"
 features = ["term-svg"]
 
 [dev-dependencies.trybuild]
 version = "1.0.91"
 
 [dev-dependencies.trycmd]
-version = "0.15.3"
+version = ">= 0.14.20, < 0.16"
 features = [
     "color-auto",
     "diff",
