Description: Disable benches because they make the tests fail.
Bug: https://github.com/JeWaVe/hamming_rs/issues/2

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,6 +33,7 @@
 ]
 license = "MIT"
 repository = "https://github.com/JeWaVe/hamming_rs"
+autobenches = false
 
 [profile.dev]
 opt-level = 0
@@ -44,10 +45,6 @@
 debug-assertions = false
 overflow-checks = false
 
-[[bench]]
-name = "distance"
-harness = false
-
 [dependencies]
 
 [dev-dependencies.criterion]
