Index: tracing-log/Cargo.toml
===================================================================
--- tracing-log.orig/Cargo.toml
+++ tracing-log/Cargo.toml
@@ -31,6 +31,7 @@ categories = [
 ]
 license = "MIT"
 repository = "https://github.com/tokio-rs/tracing"
+autobenches = false
 
 [package.metadata.docs.rs]
 all-features = true
@@ -39,10 +40,6 @@ rustdoc-args = [
     "docsrs",
 ]
 
-[[bench]]
-name = "logging"
-harness = false
-
 [dependencies.ahash]
 version = "0.7.6"
 optional = true
