diff --git a/Cargo.toml b/Cargo.toml
index 8f8af33..64e777f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,7 +72,7 @@ version = "0.5"
 version = "5"
 
 [dependencies.fd-lock]
-version = ">= 3, < 5"
+version = "<5"
 default-features = false
 
 [dependencies.filetime]
@@ -118,13 +118,13 @@ features = ["socks"]
 version = "0.4"
 
 [dependencies.sequoia-ipc]
-version = "0.33"
+version = "0.31"
 
 [dependencies.sequoia-net]
 version = "0.28"
 
 [dependencies.sequoia-openpgp]
-version = "1.19"
+version = "1.18"
 features = ["compression"]
 default-features = false
 
@@ -196,7 +196,7 @@ version = "3"
 [features]
 crypto-botan = ["sequoia-openpgp/crypto-botan"]
 crypto-botan2 = ["sequoia-openpgp/crypto-botan2"]
-crypto-cng = ["sequoia-openpgp/crypto-cng"]
+#crypto-cng = ["sequoia-openpgp/crypto-cng"]
 crypto-nettle = ["sequoia-openpgp/crypto-nettle"]
 crypto-openssl = ["sequoia-openpgp/crypto-openssl"]
 default = ["crypto-nettle"]
