diff --git a/Cargo.toml b/Cargo.toml
index 08e4ca7..ffc9c6b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -77,7 +77,7 @@ version = "0.5"
 version = "5"
 
 [dependencies.fd-lock]
-version = ">= 3, < 5"
+version = "< 5"
 default-features = false
 
 [dependencies.filetime]
@@ -226,7 +226,7 @@ version = "3"
 [features]
 crypto-botan = ["sequoia-openpgp/crypto-botan"]
 crypto-botan2 = ["sequoia-openpgp/crypto-botan2"]
-crypto-cng = ["sequoia-openpgp/crypto-cng"]
+#crypto-cng = ["sequoia-openpgp/crypto-cng"]
 crypto-nettle = ["sequoia-openpgp/crypto-nettle"]
 crypto-openssl = ["sequoia-openpgp/crypto-openssl"]
 default = ["crypto-nettle"]
