Index: sequoia-chameleon-gnupg/Cargo.toml
===================================================================
--- sequoia-chameleon-gnupg.orig/Cargo.toml
+++ sequoia-chameleon-gnupg/Cargo.toml
@@ -52,7 +52,7 @@ repository = "https://gitlab.com/sequoia
 [features]
 crypto-botan = ["sequoia-openpgp/crypto-botan"]
 crypto-botan2 = ["sequoia-openpgp/crypto-botan2"]
-crypto-cng = ["sequoia-openpgp/crypto-cng"]
+#crypto-cng = ["sequoia-openpgp/crypto-cng"]
 crypto-nettle = ["sequoia-openpgp/crypto-nettle"]
 crypto-openssl = ["sequoia-openpgp/crypto-openssl"]
 default = ["crypto-nettle"]
@@ -173,7 +173,7 @@ version = "0.8"
 default-features = false
 
 [dependencies.sequoia-wot]
-version = "0.14"
+version = "0.15"
 default-features = false
 
 [dependencies.serde]
