Index: python-pkginfo/Cargo.toml
===================================================================
--- python-pkginfo.orig/Cargo.toml
+++ python-pkginfo/Cargo.toml
@@ -49,7 +49,7 @@ version = "1.0.33"
 version = "3.0.0"
 
 [dependencies.mailparse]
-version = "0.15"
+version = "0.16"
 
 [dependencies.rfc2047-decoder]
 version = "1.0.6"
@@ -71,7 +71,7 @@ optional = true
 package = "xz2"
 
 [dependencies.zip]
-version = ">=0.6,<3"
+version = ">=0.6,<6"
 features = ["deflate"]
 default-features = false
 
