--- a/Cargo.toml
+++ b/Cargo.toml
@@ -147,3 +147,3 @@
 
-[dev-dependencies.tower-test]
+[disabled.dev-dependencies.tower-test]
 version = "0.4"
--- a/src/balance/p2c/test.rs
+++ b/src/balance/p2c/test.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 use crate::discover::ServiceList;
--- a/src/balance/pool/test.rs
+++ b/src/balance/pool/test.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 use crate::load;
--- a/tests/load_shed/main.rs
+++ b/tests/load_shed/main.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(feature = "load-shed")]
--- a/tests/ready_cache/main.rs
+++ b/tests/ready_cache/main.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(feature = "ready-cache")]
--- a/tests/spawn_ready/main.rs
+++ b/tests/spawn_ready/main.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(feature = "spawn-ready")]
--- a/tests/retry/main.rs
+++ b/tests/retry/main.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(feature = "retry")]
--- a/tests/builder.rs
+++ b/tests/builder.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(all(feature = "buffer", feature = "limit", feature = "retry"))]
--- a/tests/balance/main.rs
+++ b/tests/balance/main.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(feature = "balance")]
--- a/tests/buffer/main.rs
+++ b/tests/buffer/main.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #![cfg(feature = "buffer")]
--- a/tests/limit/concurrency.rs
+++ b/tests/limit/concurrency.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 #[path = "../support.rs"]
--- a/tests/util/call_all.rs
+++ b/tests/util/call_all.rs
@@ -1 +1,2 @@
+#![cfg(feature = "tower-test")]
 use super::support;
