Origin: vendor
Subject: patch out Windows dependencies
Forwarded: not-needed

this should be handled automatically by debcargo...

Index: sequoia-ipc/Cargo.toml
===================================================================
--- sequoia-ipc.orig/Cargo.toml
+++ sequoia-ipc/Cargo.toml
@@ -112,18 +112,3 @@ features = [
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.ctor]
-version = "0.2"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["winsock2"]
-default-features = false
-
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "2"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
