Index: png/Cargo.toml
===================================================================
--- png.orig/Cargo.toml
+++ png/Cargo.toml
@@ -56,10 +56,6 @@ path = "examples/png-generate.rs"
 name = "pngcheck"
 path = "examples/pngcheck.rs"
 
-[[example]]
-name = "show"
-path = "examples/show.rs"
-
 [dependencies.bitflags]
 version = "1.0"
 
@@ -90,11 +86,6 @@ features = ["derive"]
 [dev-dependencies.getopts]
 version = "0.2.14"
 
-[dev-dependencies.glium]
-version = "0.32"
-features = ["glutin"]
-default-features = false
-
 [dev-dependencies.glob]
 version = "0.3"
 
