Index: png/Cargo.toml
===================================================================
--- png.orig/Cargo.toml
+++ png/Cargo.toml
@@ -19,11 +19,8 @@ description = "PNG decoding and encoding
 categories = ["multimedia::images"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/image-rs/image-png.git"
+autobenches = false
 
-[[bench]]
-name = "decoder"
-path = "benches/decoder.rs"
-harness = false
 [dependencies.bitflags]
 version = "1.0"
 
@@ -35,8 +32,6 @@ version = "1.0"
 
 [dependencies.miniz_oxide]
 version = "0.5.1"
-[dev-dependencies.criterion]
-version = "0.3.1"
 
 [dev-dependencies.getopts]
 version = "0.2.14"
