--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,9 +56,9 @@
 name = "pngcheck"
 path = "examples/pngcheck.rs"
 
-[[example]]
-name = "show"
-path = "examples/show.rs"
+#[[example]]
+#name = "show"
+#path = "examples/show.rs"
 
 [dependencies.bitflags]
 version = "1.0"
@@ -89,11 +89,6 @@
 [dev-dependencies.getopts]
 version = "0.2.14"
 
-[dev-dependencies.glium]
-version = "0.32"
-features = ["glutin"]
-default-features = false
-
 [dev-dependencies.glob]
 version = "0.3"
 
