--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,23 +60,6 @@
 name = "show"
 path = "examples/show.rs"
 
-[[bench]]
-name = "decoder"
-path = "benches/decoder.rs"
-harness = false
-
-[[bench]]
-name = "expand_paletted"
-path = "benches/expand_paletted.rs"
-harness = false
-required-features = ["benchmarks"]
-
-[[bench]]
-name = "unfilter"
-path = "benches/unfilter.rs"
-harness = false
-required-features = ["benchmarks"]
-
 [dependencies.bitflags]
 version = "1.0"
 
@@ -103,9 +86,6 @@
 version = "3.0"
 features = ["derive"]
 
-[dev-dependencies.criterion]
-version = "0.4.0"
-
 [dev-dependencies.getopts]
 version = "0.2.14"
 
@@ -124,7 +104,6 @@
 version = "1.0.1"
 
 [features]
-benchmarks = []
 unstable = ["crc32fast/nightly"]
 
 [lints.rust.unexpected_cfgs]
