Index: png/Cargo.toml
===================================================================
--- png.orig/Cargo.toml
+++ png/Cargo.toml
@@ -29,6 +29,10 @@ categories = ["multimedia::images"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/image-rs/image-png.git"
 autobenches = false
+autoexamples = false
+
+[[example]]
+name = "pngcheck"
 
 
 [dependencies.bitflags]
@@ -47,11 +51,6 @@ version = "0.5.0"
 [dev-dependencies.getopts]
 version = "0.2.14"
 
-[dev-dependencies.glium]
-version = "0.31"
-features = ["glutin"]
-default-features = false
-
 [dev-dependencies.glob]
 version = "0.3"
 
