diff --git a/Cargo.toml b/Cargo.toml
index a84641c..a89a98f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,8 +25,3 @@ repository = "https://github.com/softprops/atty"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
 default-features = false
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = ["consoleapi", "processenv", "minwinbase", "minwindef", "winbase"]
-[badges.travis-ci]
-repository = "softprops/atty"
