Index: trust-dns-client/Cargo.toml
===================================================================
--- trust-dns-client.orig/Cargo.toml
+++ trust-dns-client/Cargo.toml
@@ -171,10 +171,6 @@ dns-over-openssl = [
     "dnssec-openssl",
     "openssl",
 ]
-dns-over-quic = [
-    "dns-over-rustls",
-    "trust-dns-proto/dns-over-quic",
-]
 dns-over-rustls = [
     "dns-over-tls",
     "dnssec-ring",
