--- a/tests/all.rs
+++ b/tests/all.rs
@@ -1330,7 +1330,8 @@
     let data = t!(ar.into_inner());
 
     // Without sparse support, the size of the tarball exceed 1MiB.
-    #[cfg(target_os = "linux")]
+    // ppc64el has 64k pages and thus 64k blocks on tmpfs
+    #[cfg(all(target_os = "linux", not(target_arch = "powerpc64")))]
     assert!(data.len() <= 37 * 1024); // ext4 (defaults to 4k block size)
     #[cfg(target_os = "freebsd")]
     assert!(data.len() <= 273 * 1024); // UFS (defaults to 32k block size, last block isn't a hole)
