--- a/Cargo.toml
+++ b/Cargo.toml
@@ -69,7 +69,7 @@
 version = "1.0"
 
 [dependencies.tokio]
-version = "1.37"
+version = "1"
 features = [
     "macros",
     "rt-multi-thread",
@@ -80,10 +80,10 @@
 version = "2.5.0"
 
 [dev-dependencies.env_logger]
-version = "0.11.3"
+version = "0.10"
 
 [dev-dependencies.tokio]
-version = "1.37"
+version = "1"
 features = [
     "rt-multi-thread",
     "macros",

