--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,7 +56,6 @@
 only_u8x4 = []
 rayon = [
     "dep:rayon",
-    "resize/rayon",
     "image/rayon",
 ]
 
@@ -88,81 +87,6 @@
 name = "testing"
 path = "tests/testing.rs"
 
-[[bench]]
-name = "bench_alpha"
-path = "benches/bench_alpha.rs"
-harness = false
-
-[[bench]]
-name = "bench_color_mapper"
-path = "benches/bench_color_mapper.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_l"
-path = "benches/bench_compare_l.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_l16"
-path = "benches/bench_compare_l16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_l32f"
-path = "benches/bench_compare_l32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_la"
-path = "benches/bench_compare_la.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_la16"
-path = "benches/bench_compare_la16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_la32f"
-path = "benches/bench_compare_la32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgb"
-path = "benches/bench_compare_rgb.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgb16"
-path = "benches/bench_compare_rgb16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgb32f"
-path = "benches/bench_compare_rgb32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgba"
-path = "benches/bench_compare_rgba.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgba16"
-path = "benches/bench_compare_rgba16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgba32f"
-path = "benches/bench_compare_rgba32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_resize"
-path = "benches/bench_resize.rs"
-harness = false
-
 [dependencies.bytemuck]
 version = "1.21"
 optional = true
@@ -199,11 +123,6 @@
 [dev-dependencies.png]
 version = "0.17"
 
-[dev-dependencies.resize]
-version = "0.8.8"
-features = ["std"]
-default-features = false
-
 [dev-dependencies.rgb]
 version = "0.8.50"
 
@@ -220,9 +139,6 @@
 [dev-dependencies.walkdir]
 version = "2.5"
 
-[target.'cfg(all(not(target_arch = "wasm32"), not(target_os = "windows")))'.dev-dependencies.libvips]
-version = "1.7"
-
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.nix]
 version = "0.29.0"
 features = ["sched"]
