Index: fast-image-resize/Cargo.toml
===================================================================
--- fast-image-resize.orig/Cargo.toml
+++ fast-image-resize/Cargo.toml
@@ -57,7 +57,6 @@ image = [
 only_u8x4 = []
 rayon = [
     "dep:rayon",
-    "resize/rayon",
     "image/rayon",
 ]
 
@@ -89,86 +88,6 @@ path = "tests/resize_tests.rs"
 name = "testing"
 path = "tests/testing.rs"
 
-[[bench]]
-name = "bench_alpha"
-path = "benches/bench_alpha.rs"
-harness = false
-
-[[bench]]
-name = "bench_color_mapper"
-path = "benches/bench_color_mapper.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_l"
-path = "benches/bench_compare_l.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_l16"
-path = "benches/bench_compare_l16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_l32f"
-path = "benches/bench_compare_l32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_la"
-path = "benches/bench_compare_la.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_la16"
-path = "benches/bench_compare_la16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_la32f"
-path = "benches/bench_compare_la32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgb"
-path = "benches/bench_compare_rgb.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgb16"
-path = "benches/bench_compare_rgb16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgb32f"
-path = "benches/bench_compare_rgb32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgba"
-path = "benches/bench_compare_rgba.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgba16"
-path = "benches/bench_compare_rgba16.rs"
-harness = false
-
-[[bench]]
-name = "bench_compare_rgba32f"
-path = "benches/bench_compare_rgba32f.rs"
-harness = false
-
-[[bench]]
-name = "bench_resize"
-path = "benches/bench_resize.rs"
-harness = false
-
-[[bench]]
-name = "bench_threads"
-path = "benches/bench_threads.rs"
-harness = false
-
 [dependencies.bytemuck]
 version = "1.23"
 optional = true
@@ -205,11 +124,6 @@ version = "0.14.0"
 [dev-dependencies.png]
 version = "0.18.0"
 
-[dev-dependencies.resize]
-version = "0.8.8"
-features = ["std"]
-default-features = false
-
 [dev-dependencies.rgb]
 version = "0.8.52"
 
@@ -226,9 +140,6 @@ version = "1.20"
 [dev-dependencies.walkdir]
 version = "2.5"
 
-[target.'cfg(all(not(target_arch = "wasm32"), not(target_os = "windows")))'.dev-dependencies.libvips]
-version = "1.7"
-
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.nix]
 version = "0.30.1"
 features = ["sched"]
