--- a/Cargo.toml
+++ b/Cargo.toml
@@ -171,7 +171,7 @@
 version = "1.0"
 
 [dependencies.document-features]
-version = "0.2.11"
+version = "0.2"
 
 [dependencies.image]
 version = "0.25.5"
@@ -194,10 +194,10 @@
 default-features = false
 
 [dev-dependencies.itertools]
-version = "0.14.0"
+version = "0.13"
 
 [dev-dependencies.png]
-version = "0.17.16"
+version = "0.17"
 
 [dev-dependencies.resize]
 version = "0.8.8"
