Index: fast-image-resize/Cargo.toml
===================================================================
--- fast-image-resize.orig/Cargo.toml
+++ fast-image-resize/Cargo.toml
@@ -113,15 +113,15 @@ optional = true
 version = "2.0"
 
 [dev-dependencies.criterion]
-version = "0.7.0"
+version = "0.5.1"
 features = ["cargo_bench_support"]
 default-features = false
 
 [dev-dependencies.itertools]
-version = "0.14.0"
+version = "0.13"
 
 [dev-dependencies.png]
-version = "0.18.0"
+version = "0.17.16"
 
 [dev-dependencies.rgb]
 version = "0.8.52"
