diff --git a/Cargo.toml b/Cargo.toml
index 6999d3c..2888a95 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -150,10 +150,10 @@ features = [
 version = "0.2.137"
 features = ["extra_traits"]
 
-[target."cfg(windows)".dependencies.recvmsg]
+[disabled.target."cfg(windows)".dependencies.recvmsg]
 version = "1.0.0"
 
-[target."cfg(windows)".dependencies.tokio]
+[disabled.target."cfg(windows)".dependencies.tokio]
 version = "1.36.0"
 features = [
     "sync",
@@ -165,10 +165,10 @@ features = [
 ]
 optional = true
 
-[target."cfg(windows)".dependencies.widestring]
+[disabled.target."cfg(windows)".dependencies.widestring]
 version = "1.0.2"
 
-[target."cfg(windows)".dependencies.windows-sys]
+[disabled.target."cfg(windows)".dependencies.windows-sys]
 version = "0.52.0"
 features = [
     "Win32_Foundation",
