diff --git a/Cargo.toml b/Cargo.toml
index 6999d3c..c005883 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,7 +58,7 @@ targets = [
 [features]
 async = ["futures-core"]
 default = []
-doc_cfg = []
+#doc_cfg = []
 tokio = [
     "dep:tokio",
     "async",
@@ -116,9 +116,6 @@ path = "examples/named_pipe/tokio/stream/bytes.rs"
 name = "named_pipe_tokio_client_msg"
 path = "examples/named_pipe/tokio/stream/msg.rs"
 
-[dependencies.doctest-file]
-version = "1.0.0"
-
 [dependencies.futures-core]
 version = "0.3.28"
 optional = true
diff --git a/src/local_socket/listener/enum.rs b/src/local_socket/listener/enum.rs
index bcac1fb..c9cd16f 100644
--- a/src/local_socket/listener/enum.rs
+++ b/src/local_socket/listener/enum.rs
@@ -47,7 +47,7 @@ mkenum!(
 ///
 /// ## Basic server
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/local_socket/sync/listener.rs")]
+//#[doc = doctest_file::include_doctest!("examples/local_socket/sync/listener.rs")]
 /// ```
 Listener);
 
diff --git a/src/local_socket/stream/enum.rs b/src/local_socket/stream/enum.rs
index ab85a21..435be63 100644
--- a/src/local_socket/stream/enum.rs
+++ b/src/local_socket/stream/enum.rs
@@ -65,7 +65,7 @@ mkenum!(
 ///
 /// ## Basic client
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/local_socket/sync/stream.rs")]
+//#[doc = doctest_file::include_doctest!("examples/local_socket/sync/stream.rs")]
 /// ```
 Stream);
 impl r#trait::Stream for Stream {
diff --git a/src/local_socket/tokio/listener/enum.rs b/src/local_socket/tokio/listener/enum.rs
index 3d4d498..86c07ff 100644
--- a/src/local_socket/tokio/listener/enum.rs
+++ b/src/local_socket/tokio/listener/enum.rs
@@ -22,7 +22,7 @@ mkenum!(
 ///
 /// ## Basic server
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/local_socket/tokio/listener.rs")]
+//#[doc = doctest_file::include_doctest!("examples/local_socket/tokio/listener.rs")]
 /// ```
 Listener);
 
diff --git a/src/local_socket/tokio/stream/enum.rs b/src/local_socket/tokio/stream/enum.rs
index eb46369..a714764 100644
--- a/src/local_socket/tokio/stream/enum.rs
+++ b/src/local_socket/tokio/stream/enum.rs
@@ -66,7 +66,7 @@ mkenum!(
 ///
 /// ## Basic client
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/local_socket/tokio/listener.rs")]
+//#[doc = doctest_file::include_doctest!("examples/local_socket/tokio/listener.rs")]
 /// ```
 Stream);
 
diff --git a/src/unnamed_pipe.rs b/src/unnamed_pipe.rs
index afe7933..c554397 100644
--- a/src/unnamed_pipe.rs
+++ b/src/unnamed_pipe.rs
@@ -40,11 +40,11 @@ use {crate::Sealed, std::io};
 /// ## Basic communication
 /// In a parent process:
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_a.rs")]
+//#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_a.rs")]
 /// ```
 /// In a child process:
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_b.rs")]
+//#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_b.rs")]
 /// ```
 #[inline]
 pub fn pipe() -> io::Result<(Sender, Recver)> { pipe_impl() }
diff --git a/src/unnamed_pipe/tokio.rs b/src/unnamed_pipe/tokio.rs
index c5e23b1..1f0e55f 100644
--- a/src/unnamed_pipe/tokio.rs
+++ b/src/unnamed_pipe/tokio.rs
@@ -22,11 +22,11 @@ use std::io;
 /// ## Basic communication
 /// In a parent process, within a Tokio runtime:
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_a.rs")]
+//#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_a.rs")]
 /// ```
 /// In a child process, within a Tokio runtime:
 /// ```no_run
-#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_b.rs")]
+//#[doc = doctest_file::include_doctest!("examples/unnamed_pipe/sync/side_b.rs")]
 /// ```
 #[inline]
 pub fn pipe() -> io::Result<(Sender, Recver)> { pipe_impl() }
