--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,20 +45,20 @@
 optional = true
 
 [dependencies.lazy-regex]
-version = ">=2"
+version = "3"
 
 [dependencies.log]
 version = ">=0.4"
 
 [dependencies.mailparse]
-version = "0.15"
+version = ">=0.14"
 optional = true
 
 [dependencies.maplit]
 version = "1"
 
 [dependencies.reqwest]
-version = ">=0.11"
+version = ">=0.12"
 
 [dependencies.tokio]
 version = "1"
@@ -66,7 +66,7 @@
 optional = true
 
 [dependencies.xmltree]
-version = "0.11"
+version = ">=0.10"
 
 [features]
 blocking = ["reqwest/blocking"]
