diff --git a/src/converter.rs b/src/converter.rs
index ab020bc..21b04a4 100644
--- a/src/converter.rs
+++ b/src/converter.rs
@@ -18,7 +18,7 @@ use quote::quote;
 ///
 /// This requires that the crate using the generated data needs a different struct.
 ///
-/// ```
+/// ```ignore
 /// use databake::{converter::AsStaticStr, Bake};
 ///
 /// #[derive(Bake)]
@@ -133,7 +133,7 @@ pub type StringAsStaticStr = AsStaticStr<String>;
 ///
 /// This requires that the crate using the generated data needs a different struct.
 ///
-/// ```
+/// ```ignore
 /// use databake::{converter::IteratorAsRefSlice, Bake};
 ///
 /// #[derive(Bake, Default)]
diff --git a/src/lib.rs b/src/lib.rs
index ce55b7c..6c23da1 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -24,7 +24,7 @@
 //!
 //! `Bake` can be automatically derived if the `derive` Cargo feature is enabled.
 //!
-//! ```
+//! ```ignore
 //! use databake::*;
 //!
 //! #[derive(Bake)]
@@ -43,7 +43,7 @@
 //! # Testing
 //! The [`test_bake`] macro can be used to assert that a particular expression is a `Bake` fixed point.
 //!
-//! ```
+//! ```ignore
 //! # use databake::*;
 //! # #[derive(Bake)]
 //! # #[databake(path = my_crate)]
