Index: uucore/Cargo.toml
===================================================================
--- uucore.orig/Cargo.toml
+++ uucore/Cargo.toml
@@ -203,12 +203,12 @@ format = [
 fs = [
     "dunce",
     "libc",
-    "winapi-util",
-    "windows-sys",
+#    "winapi-util",
+#    "windows-sys",
 ]
 fsext = [
     "libc",
-    "windows-sys",
+#    "windows-sys",
 ]
 fsxattr = ["xattr"]
 lines = []
@@ -245,7 +245,7 @@ tty = []
 update-control = []
 uptime = [
     "libc",
-    "windows-sys",
+#    "windows-sys",
     "utmpx",
     "utmp-classic",
     "thiserror",
@@ -264,21 +264,21 @@ wide = []
 version = "0.1.6"
 optional = true
 
-[target.'cfg(target_os = "windows")'.dependencies.winapi-util]
-version = "0.1.8"
-optional = true
+#[target.'cfg(target_os = "windows")'.dependencies.winapi-util]
+#version = "0.1.8"
+#optional = true
 
-[target.'cfg(target_os = "windows")'.dependencies.windows-sys]
-version = "0.59.0"
-features = [
-    "Wdk_System_SystemInformation",
-    "Win32_Storage_FileSystem",
-    "Win32_Foundation",
-    "Win32_System_RemoteDesktop",
-    "Win32_System_WindowsProgramming",
-]
-optional = true
-default-features = false
+#[target.'cfg(target_os = "windows")'.dependencies.windows-sys]
+#version = "0.59.0"
+#features = [
+#    "Wdk_System_SystemInformation",
+#    "Win32_Storage_FileSystem",
+#    "Win32_Foundation",
+#    "Win32_System_RemoteDesktop",
+#    "Win32_System_WindowsProgramming",
+#]
+#optional = true
+#default-features = false
 
 [target."cfg(unix)".dependencies.nix]
 version = "0.29"
