Index: uucore/Cargo.toml
===================================================================
--- uucore.orig/Cargo.toml
+++ uucore/Cargo.toml
@@ -247,7 +247,7 @@ uptime = [
     "libc",
 #    "windows-sys",
     "utmpx",
-    "utmp-classic",
+#    "utmp-classic",
     "thiserror",
 ]
 utf8 = []
@@ -260,9 +260,9 @@ utmpx = [
 version-cmp = []
 wide = []
 
-[target.'cfg(target_os = "openbsd")'.dependencies.utmp-classic]
-version = "0.1.6"
-optional = true
+#[target.'cfg(target_os = "openbsd")'.dependencies.utmp-classic]
+#version = "0.1.6"
+#optional = true
 
 #[target.'cfg(target_os = "windows")'.dependencies.winapi-util]
 #version = "0.1.8"
