--- a/src/utils.rs
+++ b/src/utils.rs
@@ -117,7 +117,7 @@ pub fn get_cargo() -> std::ffi::OsString
     cargo.to_os_string()
 }
 
-#[test]
+/*#[test]
 fn test_format_rust_expression() {
     use crate::assert_snapshot;
     assert_snapshot!(format_rust_expression("vec![1,2,3]"), @"vec![1, 2, 3]");
@@ -125,4 +125,4 @@ fn test_format_rust_expression() {
     assert_snapshot!(format_rust_expression(r#"    "aoeu""#), @r#""aoeu""#);
     assert_snapshot!(format_rust_expression(r#"  "aoe😄""#), @r#""aoe😄""#);
     assert_snapshot!(format_rust_expression("😄😄😄😄😄"), @"😄😄😄😄😄")
-}
+}*/
