From 29e109cd2ebd653683b48d343a885564e3eb3bf7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20H=C3=A4cker?= <haeckerfelix@gnome.org>
Date: Fri, 20 Sep 2024 21:20:03 +0200
Subject: [PATCH] session: Use float for speed limits

---
 src/session.rs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/session.rs b/src/session.rs
index 0a4d33a..fa108c5 100644
--- a/src/session.rs
+++ b/src/session.rs
@@ -74,9 +74,9 @@ pub struct Session {
     pub seed_ratio_limited: bool,
     #[serde(default)]
     pub session_id: String,
-    pub speed_limit_down: i32,
+    pub speed_limit_down: f32,
     pub speed_limit_down_enabled: bool,
-    pub speed_limit_up: i32,
+    pub speed_limit_up: f32,
     pub speed_limit_up_enabled: bool,
     pub start_added_torrents: bool,
     pub trash_original_torrent_files: bool,
@@ -165,11 +165,11 @@ pub struct SessionMutator {
     #[serde(rename = "seedRatioLimited")]
     pub seed_ratio_limited: Option<bool>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub speed_limit_down: Option<i32>,
+    pub speed_limit_down: Option<f32>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub speed_limit_down_enabled: Option<bool>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub speed_limit_up: Option<i32>,
+    pub speed_limit_up: Option<f32>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub speed_limit_up_enabled: Option<bool>,
     #[serde(skip_serializing_if = "Option::is_none")]
-- 
GitLab

