Index: rust-gstreamer-player-0.20.5/Cargo.toml
===================================================================
--- rust-gstreamer-player-0.20.5.orig/Cargo.toml
+++ rust-gstreamer-player-0.20.5/Cargo.toml
@@ -66,12 +66,6 @@ version = "0.1"
 
 [features]
 default = []
-dox = [
-    "ffi/dox",
-    "glib/dox",
-    "gst/dox",
-    "gst-video/dox",
-]
 v1_16 = [
     "gst/v1_16",
     "gst-video/v1_16",
