Index: annotate-snippets/Cargo.toml
===================================================================
--- annotate-snippets.orig/Cargo.toml
+++ annotate-snippets/Cargo.toml
@@ -95,11 +95,6 @@ name = "examples"
 path = "tests/examples.rs"
 
 [[test]]
-name = "fixtures"
-path = "tests/fixtures/main.rs"
-harness = false
-
-[[test]]
 name = "formatter"
 path = "tests/formatter.rs"
 
@@ -142,9 +137,6 @@ features = [
 [dev-dependencies.toml]
 version = "0.5.11"
 
-[dev-dependencies.tryfn]
-version = "0.2.1"
-
 [features]
 default = []
 testing-colors = []
