Index: indicatif/Cargo.toml
===================================================================
--- indicatif.orig/Cargo.toml
+++ indicatif/Cargo.toml
@@ -23,7 +23,7 @@ keywords = ["cli", "progress", "pb", "co
 license = "MIT"
 repository = "https://github.com/mitsuhiko/indicatif"
 [dependencies.console]
-version = ">=0.9.1, <1.0.0"
+version = ">=0.8.0, <1.0.0"
 
 [dependencies.lazy_static]
 version = "1.0"
