Index: indicatif/Cargo.toml
===================================================================
--- indicatif.orig/Cargo.toml
+++ indicatif/Cargo.toml
@@ -64,10 +64,6 @@ optional = true
 version = "0.1"
 optional = true
 
-[dependencies.vt100]
-version = "0.15.1"
-optional = true
-
 [dev-dependencies.once_cell]
 version = "1"
 
@@ -95,4 +91,3 @@ improved_unicode = [
     "unicode-width",
     "console/unicode-width",
 ]
-in_memory = ["vt100"]
