diff --git a/Cargo.toml b/Cargo.toml
index 211ed5b..c17c5e5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -113,5 +113,5 @@
 ]
 in_memory = ["vt100"]
 
-[target."cfg(target_arch = \"wasm32\")".dependencies.instant]
-version = "0.1"
+[target."cfg(target_arch = \"wasm32\")".dependencies.web-time]
+version = "1.1.0"
diff --git a/src/draw_target.rs b/src/draw_target.rs
index a2c055f..d7fb707 100644
--- a/src/draw_target.rs
+++ b/src/draw_target.rs
@@ -7,7 +7,7 @@ use std::time::Instant;
 
 use console::Term;
 #[cfg(target_arch = "wasm32")]
-use instant::Instant;
+use web_time::Instant;
 
 use crate::multi::{MultiProgressAlignment, MultiState};
 use crate::TermLike;
diff --git a/src/multi.rs b/src/multi.rs
index 4409309..9c99439 100644
--- a/src/multi.rs
+++ b/src/multi.rs
@@ -8,7 +8,7 @@ use std::time::Instant;
 use crate::draw_target::{DrawState, DrawStateWrapper, LineAdjust, ProgressDrawTarget};
 use crate::progress_bar::ProgressBar;
 #[cfg(target_arch = "wasm32")]
-use instant::Instant;
+use web_time::Instant;
 
 /// Manages multiple progress bars from different threads
 #[derive(Debug, Clone)]
diff --git a/src/progress_bar.rs b/src/progress_bar.rs
index 938668e..b2065d7 100644
--- a/src/progress_bar.rs
+++ b/src/progress_bar.rs
@@ -8,7 +8,7 @@ use std::time::Instant;
 use std::{fmt, io, thread};
 
 #[cfg(target_arch = "wasm32")]
-use instant::Instant;
+use web_time::Instant;
 #[cfg(test)]
 use once_cell::sync::Lazy;
 
diff --git a/src/state.rs b/src/state.rs
index 3bcdc44..45094d0 100644
--- a/src/state.rs
+++ b/src/state.rs
@@ -6,7 +6,7 @@ use std::time::Duration;
 use std::time::Instant;
 
 #[cfg(target_arch = "wasm32")]
-use instant::Instant;
+use web_time::Instant;
 use portable_atomic::{AtomicU64, AtomicU8, Ordering};
 
 use crate::draw_target::ProgressDrawTarget;
diff --git a/src/style.rs b/src/style.rs
index 01b220f..49bdd1d 100644
--- a/src/style.rs
+++ b/src/style.rs
@@ -6,7 +6,7 @@ use std::time::Instant;
 
 use console::{measure_text_width, Style};
 #[cfg(target_arch = "wasm32")]
-use instant::Instant;
+use web_time::Instant;
 #[cfg(feature = "unicode-segmentation")]
 use unicode_segmentation::UnicodeSegmentation;
 
