Description: Remove rspec dev-dependency

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,8 +72,6 @@
 [dev-dependencies.insta]
 version = "1"
 
-[dev-dependencies.rspec]
-version = "1"
 
 [features]
 no-color = []
--- a/src/control.rs
+++ b/src/control.rs
@@ -158,7 +158,8 @@
     }
 }
 
-#[cfg(test)]
+/*#[cfg(test)]
+
 mod specs {
     use super::*;
     use rspec;
@@ -386,4 +387,4 @@
             });
         }));
     }
-}
+}*/
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -29,8 +29,6 @@
 //! modify them.
 #![warn(missing_docs)]
 
-#[cfg(test)]
-extern crate rspec;
 
 mod color;
 pub mod control;
