--- a/Cargo.toml
+++ b/Cargo.toml
@@ -75,8 +75,6 @@
 [dev-dependencies.insta]
 version = "1"
 
-[dev-dependencies.rspec]
-version = "1"
 
 [features]
 no-color = []
--- a/src/control.rs
+++ b/src/control.rs
@@ -159,7 +159,8 @@
     }
 }
 
-#[cfg(test)]
+/*#[cfg(test)]
+
 mod specs {
     use super::*;
     use rspec;
@@ -387,4 +388,4 @@
             });
         }));
     }
-}
+}*/
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -32,8 +32,6 @@
 #[macro_use]
 extern crate lazy_static;
 
-#[cfg(test)]
-extern crate rspec;
 
 mod color;
 pub mod control;
