Index: xkbcommon/Cargo.toml
===================================================================
--- xkbcommon.orig/Cargo.toml
+++ xkbcommon/Cargo.toml
@@ -26,13 +26,11 @@ keywords = [
 license = "MIT"
 repository = "https://github.com/rust-x-bindings/xkbcommon-rs"
 resolver = "2"
+autoexamples = false
 
 [package.metadata.docs.rs]
 all-features = true
 
-[[example]]
-name = "quick-evdev"
-
 [dependencies.libc]
 version = "0.2.125"
 
@@ -45,9 +43,6 @@ version = "1.1.1"
 features = ["xkb"]
 optional = true
 
-[dev-dependencies.evdev]
-version = "0.11.4"
-
 [features]
 default = ["wayland"]
 wayland = ["memmap2"]
