Index: onig-sys/Cargo.toml
===================================================================
--- onig-sys.orig/Cargo.toml
+++ onig-sys/Cargo.toml
@@ -41,8 +41,8 @@ license = "MIT"
 repository = "https://github.com/iwillspeak/rust-onig"
 
 [features]
-default = ["generate"]
-generate = ["bindgen"]
+#default = ["generate"]
+#generate = ["bindgen"]
 posix-api = []
 print-debug = []
 
@@ -51,9 +51,7 @@ name = "onig_sys"
 path = "src/lib.rs"
 
 [build-dependencies.bindgen]
-version = "0.71"
-features = ["runtime"]
-optional = true
+version = "0"
 
 [build-dependencies.cc]
 version = "1.0"
