Index: onig-sys/Cargo.toml
===================================================================
--- onig-sys.orig/Cargo.toml
+++ onig-sys/Cargo.toml
@@ -34,9 +34,9 @@ license = "MIT"
 repository = "http://github.com/iwillspeak/rust-onig"
 
 [build-dependencies.bindgen]
-version = "0.59"
-features = ["runtime"]
-optional = true
+version = "0"
+#features = ["runtime"]
+#optional = true
 
 [build-dependencies.cc]
 version = "1.0"
@@ -45,7 +45,7 @@ version = "1.0"
 version = "^0.3.16"
 
 [features]
-default = ["generate"]
-generate = ["bindgen"]
+#default = ["generate"]
+#generate = ["bindgen"]
 posix-api = []
 print-debug = []
