--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,8 +35,6 @@
 license = "GPL-3.0-or-later"
 repository = "https://github.com/inflation/jpegxl-rs"
 
-[package.metadata.docs.rs]
-features = ["docs"]
 
 [lib]
 name = "jpegxl_sys"
@@ -48,19 +46,12 @@
 default-features = false
 
 [dev-dependencies.pretty_assertions]
-version = "1.4.1"
+version = "1.4.0"
 
-[build-dependencies.jpegxl-src]
-version = "0.11.2"
-optional = true
 
 [build-dependencies.pkg-config]
 version = "0.3.31"
 
-[features]
-default = []
-docs = []
-vendored = ["jpegxl-src"]
 
 [lints.clippy]
 module_name_repetitions = "allow"
