From 82ab7559203a434a159d54f42b59ce9c901f7055 Mon Sep 17 00:00:00 2001
From: Mads Marquart <mads@marquart.dk>
Date: Tue, 23 Nov 2021 22:28:57 +0100
Subject: [PATCH] Fix tests on targets where c_char = u8

---
 src/sentinel.rs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/sentinel.rs b/src/sentinel.rs
index d765d50..2265be5 100644
--- a/src/sentinel.rs
+++ b/src/sentinel.rs
@@ -328,9 +328,9 @@ fn test_string() {
     unsafe {
         let src = gen_malloc::<c_char>(5).as_ptr();
         *src.offset(0) = 0x61;
-        *src.offset(1) = -0x19;
-        *src.offset(2) = -0x6c;
-        *src.offset(3) = -0x4e;
+        *src.offset(1) = -0x19i8 as c_char;
+        *src.offset(2) = -0x6ci8 as c_char;
+        *src.offset(3) = -0x4ei8 as c_char;
         *src.offset(4) = 0;
 
         let string = MString::from_raw_unchecked(src);
@@ -342,7 +342,7 @@ fn test_string() {
 fn test_non_utf8_string() {
     unsafe {
         let src = gen_malloc::<c_char>(2).as_ptr();
-        *src.offset(0) = -1;
+        *src.offset(0) = -1i8 as c_char;
         *src.offset(1) = 0;
 
         let string = MString::from_raw(src);
