Index: bigdecimal/src/impl_serde.rs
===================================================================
--- bigdecimal.orig/src/impl_serde.rs
+++ bigdecimal/src/impl_serde.rs
@@ -247,6 +247,7 @@ mod test {
 
 
         #[test]
+	#[cfg(not(target_arch = "x86"))]
         fn test_struct_parsing() {
             let json_src = r#"
                 { "name": "foo", "value": 0.0008741329382918, "number": "12.34" }
@@ -397,8 +398,11 @@ mod test_jsonification {
             };
         }
 
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_1d0: "1.0" => "1.0");
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_0d001: "0.001" => "0.001");
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_41009d2207etc: "41009.22075436852032769878903135430037010");
     }
 
@@ -437,15 +441,20 @@ mod test_jsonification {
             };
         }
 
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_1d0: r#"{ "value": 1.0 }"# => "1.0" );
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_2d01: r#"{ "value": 2.01 }"# => "2.01" );
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_50: r#"{ "value": 50 }"# => "50" );
+	#[cfg(not(target_arch = "x86"))]
         impl_case!(case_high_prec: r#"{ "value": 64771126779.35857825871133263810255301911 }"# => "64771126779.35857825871133263810255301911" );
 
         impl_case!(case_nan: r#"{ "value": nan }"# => (error) );
 
 
         #[test]
+	#[cfg(not(target_arch = "x86"))]
         fn scale_out_of_bounds() {
             use serde_crate::de::Error;
 
