Index: phf-generator/Cargo.toml
===================================================================
--- phf-generator.orig/Cargo.toml
+++ phf-generator/Cargo.toml
@@ -31,7 +31,7 @@ name = "benches"
 harness = false
 
 [dependencies.criterion]
-version = "0.3.6"
+version = "0.5"
 optional = true
 
 [dependencies.phf_shared]
@@ -44,4 +44,7 @@ features = ["small_rng"]
 default-features = false
 
 [dev-dependencies.criterion]
-version = "0.3.6"
+version = "0.5"
+
+[features]
+criterion = ["dep:criterion", "phf_shared/std"]
Index: phf-generator/src/bin/gen_hash_test.rs
===================================================================
--- phf-generator.orig/src/bin/gen_hash_test.rs
+++ phf-generator/src/bin/gen_hash_test.rs
@@ -10,7 +10,7 @@ fn gen_vec(len: usize) -> Vec<String> {
     let mut rng = SmallRng::seed_from_u64(0xAAAAAAAAAAAAAAAA).sample_iter(Alphanumeric);
 
     (0..len)
-        .map(move |_| rng.by_ref().take(64).collect::<String>())
+        .map(move |_| rng.by_ref().take(64).map(char::from).collect::<String>())
         .collect()
 }
 
