--- a/Cargo.toml	2018-08-12 22:12:24.182112437 +0000
+++ b/Cargo.toml	2018-08-12 22:12:19.208779241 +0000
@@ -22,4 +22,4 @@
 version = "=0.7.22"
 
 [dependencies.rand]
-version = "0.4"
+version = "0.5"
--- a/src/lib.rs	2018-08-12 22:11:53.442113286 +0000
+++ b/src/lib.rs	2018-08-12 22:12:13.285446071 +0000
@@ -7,7 +7,10 @@
 
 const DEFAULT_LAMBDA: usize = 5;
 
-const FIXED_SEED: [u32; 4] = [3141592653, 589793238, 462643383, 2795028841];
+const FIXED_SEED: [u8; 16] = [187, 64, 230, 77,
+                              35, 39, 135, 214,
+                              27, 147, 96, 183,
+                              166, 152, 193, 105];
 
 pub struct HashState {
     pub key: u64,
