Index: phf-generator/Cargo.toml
===================================================================
--- phf-generator.orig/Cargo.toml
+++ phf-generator/Cargo.toml
@@ -19,16 +19,9 @@ description = "PHF generation logic"
 license = "MIT"
 repository = "https://github.com/sfackler/rust-phf"
 
-[[bin]]
-name = "gen_hash_test"
-required-features = ["criterion"]
-
 [[bench]]
 name = "benches"
 harness = false
-[dependencies.criterion]
-version = "0.3"
-optional = true
 
 [dependencies.phf_shared]
 version = "0.8.0"
@@ -36,5 +29,3 @@ version = "0.8.0"
 [dependencies.rand]
 version = "0.7"
 features = ["small_rng"]
-[dev-dependencies.criterion]
-version = "0.3"
Index: phf-generator/benches/benches.rs
===================================================================
--- phf-generator.orig/benches/benches.rs
+++ phf-generator/benches/benches.rs
@@ -1,4 +1,4 @@
-use criterion::*;
+/*use criterion::*;
 
 use rand::distributions::Standard;
 use rand::rngs::SmallRng;
@@ -77,4 +77,7 @@ mod rayon {
     }
 
     criterion_group!(benches, gen_hash_small, gen_hash_med, gen_hash_large, gen_hash_xlarge);
+}*/
+fn main() {
+    println!("bench disabled because criterion is not in Debian");
 }
\ No newline at end of file
