Index: numbat-cli/Cargo.toml
===================================================================
--- numbat-cli.orig/Cargo.toml
+++ numbat-cli/Cargo.toml
@@ -42,7 +42,7 @@ path = "src/main.rs"
 version = "1"
 
 [dependencies.chrono-tz]
-version = "0.8.5"
+version = "0.10"
 
 [dependencies.clap]
 version = "4"
@@ -60,13 +60,13 @@ version = "2"
 version = "5"
 
 [dependencies.itertools]
-version = "0.12"
+version = ">=0.10"
 
 [dependencies.numbat]
 version = "1.11.0"
 
 [dependencies.rustyline]
-version = "13"
+version = ">=13"
 features = ["derive"]
 
 [dependencies.serde]
@@ -74,7 +74,7 @@ version = "1.0.195"
 features = ["derive"]
 
 [dependencies.terminal_size]
-version = "0.3.0"
+version = "0.4.0"
 
 [dependencies.toml]
 version = "0.8.8"
