Index: rust-smallvec-1.11.1/src/tests.rs
===================================================================
--- rust-smallvec-1.11.1.orig/src/tests.rs
+++ rust-smallvec-1.11.1/src/tests.rs
@@ -987,7 +987,7 @@ fn test_clone_from() {
 #[test]
 fn test_size() {
     use core::mem::size_of;
-    assert_eq!(24, size_of::<SmallVec<[u8; 8]>>());
+    assert!(24 >= size_of::<SmallVec<[u8; 8]>>(), "size_of::<SmallVec<[u8; 8]>>() = {} which is greater than 24",size_of::<SmallVec<[u8; 8]>>());
 }
 
 #[cfg(feature = "drain_filter")]
