--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61 +61 @@
-[dependencies.bear-lib-terminal]
+[disabled.dependencies.bear-lib-terminal]
@@ -91 +91 @@
-[dependencies.ncurses]
+[disabled.dependencies.ncurses]
@@ -96 +96 @@
-[dependencies.pancurses]
+[disabled.dependencies.pancurses]
@@ -125 +125 @@
-blt-backend = ["dep:bear-lib-terminal"]
+#blt-backend = ["dep:bear-lib-terminal"]
@@ -131,8 +131,8 @@
-ncurses-backend = [
-    "dep:ncurses",
-    "dep:maplit",
-]
-pancurses-backend = [
-    "dep:pancurses",
-    "dep:maplit",
-]
+#ncurses-backend = [
+#    "dep:ncurses",
+#    "dep:maplit",
+#]
+#pancurses-backend = [
+#    "dep:pancurses",
+#    "dep:maplit",
+#]
