diff --git a/src/backends/puppet/observed.rs b/src/backends/puppet/observed.rs
index 62f7c02..aa203ef 100644
--- a/src/backends/puppet/observed.rs
+++ b/src/backends/puppet/observed.rs
@@ -1,5 +1,5 @@
 //! Structs representing output of puppet backend
-use crate::reexports::enumset::EnumSet;
+use crate::reexports::wasmer_enumset::EnumSet;
 use crate::theme::ColorPair;
 use crate::theme::Effect;
 use crate::Vec2;
diff --git a/src/backends/puppet/static_values.rs b/src/backends/puppet/static_values.rs
index 608f352..bd77c7c 100644
--- a/src/backends/puppet/static_values.rs
+++ b/src/backends/puppet/static_values.rs
@@ -1,7 +1,7 @@
 /// Some default values to Puppet backend.
 use lazy_static::lazy_static;
 
-use crate::reexports::enumset::EnumSet;
+use crate::reexports::wasmer_enumset::EnumSet;
 use crate::theme::ColorPair;
 use crate::theme::{Color, Effect};
 use crate::Vec2;
