--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,7 +55,7 @@
 ]
 
 [dependencies.clap-verbosity-flag]
-version = "3"
+version = "2"
 
 [dependencies.flate2]
 version = "1"
@@ -74,7 +74,7 @@
 version = "0.17"
 
 [dependencies.itertools]
-version = "0.14"
+version = "0.13"
 
 [dependencies.log]
 version = "0.4"
@@ -95,7 +95,7 @@
 version = "0.9"
 
 [dependencies.stderrlog]
-version = "0.6"
+version = "0.5"
 
 [dependencies.thiserror]
 version = "2"
@@ -129,7 +129,7 @@
 ]
 
 [build-dependencies.clap-verbosity-flag]
-version = "3"
+version = "2"
 
 [build-dependencies.clap_complete]
 version = "4.4"
