--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,7 +54,7 @@
 ]
 
 [dependencies.clap-verbosity-flag]
-version = "3"
+version = "2"
 
 [dependencies.flate2]
 version = "1"
@@ -70,10 +70,10 @@
 version = "1"
 
 [dependencies.indicatif]
-version = "0.18"
+version = "0.17"
 
 [dependencies.itertools]
-version = "0.14"
+version = "0.13"
 
 [dependencies.log]
 version = "0.4"
@@ -94,7 +94,7 @@
 version = "0.9"
 
 [dependencies.stderrlog]
-version = "0.6"
+version = "0.5"
 
 [dependencies.thiserror]
 version = "2"
@@ -111,7 +111,7 @@
 version = "1.14"
 
 [dependencies.xdg]
-version = "3.0"
+version = "2.0"
 
 [dependencies.xz2]
 version = "0.1"
@@ -128,7 +128,7 @@
 ]
 
 [build-dependencies.clap-verbosity-flag]
-version = "3"
+version = "2"
 
 [build-dependencies.clap_complete]
 version = "4.4"
--- a/src/config.rs
+++ b/src/config.rs
@@ -257,7 +257,7 @@
 impl Cache {
     pub async fn new(force_download: bool, archive_mirror: &str) -> Result<Self> {
         let mut cache = Self {
-            base_directory: BaseDirectories::with_prefix("Debian-RT-tools"),
+            base_directory: BaseDirectories::with_prefix("Debian-RT-tools")?,
             downloader: Downloader::new(force_download),
             archive_mirror: archive_mirror.into(),
             unstable: empty_release(),
