Index: tokio-native-tls/Cargo.toml
===================================================================
--- tokio-native-tls.orig/Cargo.toml
+++ tokio-native-tls/Cargo.toml
@@ -68,9 +68,6 @@ features = [
 version = "0.7"
 features = ["full"]
 
-[features]
-vendored = ["native-tls/vendored"]
-
 [target."cfg(all(not(target_os = \"macos\"), not(windows), not(target_os = \"ios\")))".dev-dependencies.openssl]
 version = "0.10"
 
