diff --git a/Cargo.toml b/Cargo.toml
index 7983fa7..b33f23d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -71,13 +71,13 @@ features = ["full"]
 [target."cfg(all(not(target_os = \"macos\"), not(windows), not(target_os = \"ios\")))".dev-dependencies.openssl]
 version = "0.10"
 
-[target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dev-dependencies.security-framework]
+[disabled.target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dev-dependencies.security-framework]
 version = "0.2"
 
-[target."cfg(windows)".dev-dependencies.schannel]
+[disabled.target."cfg(windows)".dev-dependencies.schannel]
 version = "0.1"
 
-[target."cfg(windows)".dev-dependencies.winapi]
+[disabled.target."cfg(windows)".dev-dependencies.winapi]
 version = "0.3"
 features = [
     "lmcons",
