Index: tokio-native-tls/tests/smoke.rs
===================================================================
--- tokio-native-tls.orig/tests/smoke.rs
+++ tokio-native-tls/tests/smoke.rs
@@ -28,6 +28,7 @@ lazy_static! {
 }
 
 #[tokio::test]
+#[ignore = "smoke tests don't work from sbuild"]
 async fn client_to_server() {
     let srv = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = srv.local_addr().unwrap();
@@ -67,6 +68,7 @@ async fn client_to_server() {
 }
 
 #[tokio::test]
+#[ignore = "smoke tests don't work from sbuild"]
 async fn server_to_client() {
     // Create a server listening on a port, then figure out what that port is
     let srv = TcpListener::bind("127.0.0.1:0").await.unwrap();
@@ -94,6 +96,7 @@ async fn server_to_client() {
 }
 
 #[tokio::test]
+#[ignore = "smoke tests don't work from sbuild"]
 async fn one_byte_at_a_time() {
     const AMT: usize = 1024;
 
