--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,17 +44,6 @@
 name = "rinja"
 path = "src/lib.rs"
 
-[[bench]]
-name = "escape"
-path = "benches/escape.rs"
-harness = false
-
-[[bench]]
-name = "to-json"
-path = "benches/to-json.rs"
-harness = false
-required-features = ["serde_json"]
-
 [dependencies.humansize]
 version = "2"
 optional = true
@@ -81,9 +70,6 @@
 version = "1.0"
 optional = true
 
-[dev-dependencies.criterion]
-version = "0.5"
-
 [features]
 code-in-doc = ["rinja_derive/code-in-doc"]
 config = ["rinja_derive/config"]
