Index: sequoia-gpg-agent/Cargo.toml
===================================================================
--- sequoia-gpg-agent.orig/Cargo.toml
+++ sequoia-gpg-agent/Cargo.toml
@@ -89,7 +89,7 @@ version = "0.4"
 version = "0.3"
 
 [dependencies.lalrpop-util]
-version = ">=0.17, <0.23"
+version = "0.20"
 
 [dependencies.libc]
 version = "0.2.152"
@@ -147,14 +147,6 @@ features = [
     "__implicit-crypto-backend-for-tests",
 ]
 default-features = false
-
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "2"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
 
 [profile.release]
 debug = 2
