Description: Fix version

Last-Update: 2024-12-07
---

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,19 +48,14 @@
 optional = true
 
 [dependencies.serde]
-version = "1.0.213"
+version = "1.0.210"
 features = ["derive"]
 optional = true
 
-[dependencies.termion]
-version = "4.0.3"
-optional = true
-
 [dependencies.unicode-width]
-version = "0.2.0"
+version = "0.1.14"
 
 [features]
 crossterm = ["dep:ratatui"]
 default = ["crossterm"]
 serde = ["dep:serde"]
-termion = ["dep:termion"]
