Index: is-terminal/Cargo.toml
===================================================================
--- is-terminal.orig/Cargo.toml
+++ is-terminal/Cargo.toml
@@ -45,13 +45,9 @@ version = "0.2.14"
 version = "0.2.110"
 
 [target."cfg(not(any(windows, target_os = \"hermit\", target_os = \"unknown\")))".dependencies.rustix]
-version = "0.38.0"
+version = ">= 0.37.0, < 1.0"
 features = ["termios"]
 
-[target."cfg(not(any(windows, target_os = \"hermit\", target_os = \"unknown\")))".dev-dependencies.rustix]
-version = "0.38.0"
-features = ["stdio"]
-
 # [target."cfg(target_os = \"hermit\")".dependencies.hermit-abi]
 # version = "0.3.0"
 
Index: is-terminal/src/lib.rs
===================================================================
--- is-terminal.orig/src/lib.rs
+++ is-terminal/src/lib.rs
@@ -310,7 +310,7 @@ mod tests {
         )
     }
 
-    #[test]
+    /*#[test]
     #[cfg(any(unix, target_os = "wasi"))]
     fn stdin() {
         assert_eq!(
@@ -368,7 +368,7 @@ mod tests {
                 rustix::stdio::stderr().is_terminal()
             )
         }
-    }
+    } */
 
     // Verify that the msys_tty_on function works with long path.
     #[test]
