--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,28 +37,28 @@
 version = "0.5.15"
 
 [dependencies.log]
-version = "0.4.27"
+version = "0.4"
 
 [dependencies.serde]
-version = "1.0.219"
+version = "1.0"
 
 [dependencies.serde_derive]
-version = "1.0.219"
+version = "1.0"
 
 [dependencies.serde_json]
-version = "1.0.140"
+version = "1.0"
 
 [dev-dependencies.anyhow]
-version = "1.0.98"
+version = "1.0.95"
 
 [dev-dependencies.ctrlc]
-version = "3.4.7"
+version = "3.4.5"
 
 [dev-dependencies.lsp-types]
 version = "=0.95"
 
 [dev-dependencies.rustc-hash]
-version = "2.1.1"
+version = "2.0"
 
 [lints.clippy]
 assigning_clones = "allow"
