diff --git a/Cargo.toml b/Cargo.toml
index 6c5011f..37eeb26 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,9 +20,6 @@ readme = "README.md"
 keywords = ["rand", "random", "random-rs", "rng"]
 categories = ["algorithms", "no-std", "development-tools::testing", "cryptography"]
 license = "MIT/Apache-2.0"
-[dependencies.doc-comment]
-version = "0.3"
-optional = true
 
 [features]
 u128 = []
diff --git a/src/lib.rs b/src/lib.rs
index 63c6941..03c41bd 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -16,11 +16,6 @@
 
 use core::{char, mem};
 
-#[cfg(feature = "doc-comment")]
-extern crate doc_comment;
-#[cfg(feature = "doc-comment")]
-doc_comment::doctest!("../README.md");
-
 /// This trait is used by `Random::gen()` as a generic function to create a random value for any type which implements it.
 /// I try to give by default implementations for all the types in libcore, including arrays and tuples, if anything is missing please raise the issue.
 /// You can implement this for any of your types.
