Index: once-cell/Cargo.toml
===================================================================
--- once-cell.orig/Cargo.toml
+++ once-cell/Cargo.toml
@@ -52,9 +52,9 @@ required-features = ["std"]
 [[example]]
 name = "test_synchronization"
 required-features = ["std"]
-[dependencies.atomic-polyfill]
-version = "0.1"
-optional = true
+#[dependencies.atomic-polyfill]
+#version = "0.1"
+#optional = true
 
 [dependencies.parking_lot]
 version = "0.11"
