Index: once-cell/Cargo.toml
===================================================================
--- once-cell.orig/Cargo.toml
+++ once-cell/Cargo.toml
@@ -66,10 +66,10 @@ required-features = ["std"]
 name = "test_synchronization"
 required-features = ["std"]
 
-[dependencies.atomic_polyfill]
-version = "1"
-optional = true
-package = "atomic-polyfill"
+#[dependencies.atomic_polyfill]
+#version = "1"
+#optional = true
+#package = "atomic-polyfill"
 
 [dependencies.critical_section]
 version = "1"
@@ -97,11 +97,6 @@ version = "1.2.0"
 
 [features]
 alloc = ["race"]
-atomic-polyfill = ["critical-section"]
-critical-section = [
-    "critical_section",
-    "atomic_polyfill",
-]
 default = ["std"]
 parking_lot = ["parking_lot_core"]
 race = []
