diff --git a/Cargo.toml b/Cargo.toml
index b755923..ed7f1ae 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,18 +49,18 @@ name = "test_workflow"
 path = "tests/test_workflow.rs"
 
 [dependencies.indexmap]
-version = "2.10.0"
+version = "2"
 features = ["serde"]
 
 [dependencies.serde]
-version = "1.0.219"
+version = "1"
 features = ["derive"]
 
 [dependencies.serde_yaml]
-version = "0.9.34"
+version = "0.9"
 
 [dependencies.tracing]
-version = "0.1.41"
+version = "0.1"
 
 [dev-dependencies]
 
