Index: gix-archive/Cargo.toml
===================================================================
--- gix-archive.orig/Cargo.toml
+++ gix-archive/Cargo.toml
@@ -77,7 +77,7 @@ optional = true
 version = "2.0.0"
 
 [dependencies.zip]
-version = "2.1.0"
+version = "0.6"
 features = ["deflate"]
 optional = true
 default-features = false
Index: gix-archive/src/write.rs
===================================================================
--- gix-archive.orig/src/write.rs
+++ gix-archive/src/write.rs
@@ -153,7 +153,7 @@ where
             zdt.minute().try_into().expect("non-negative"),
             zdt.second().try_into().expect("non-negative"),
         )
-        .map_err(|err| Error::InvalidModificationTime(Box::new(err)))?;
+        .map_err(|_err| Error::InvalidModificationTime("Failed to parse modification time".into()))?;
         while let Some(entry) = next_entry(stream)? {
             append_zip_entry(
                 &mut ar,
@@ -180,9 +180,9 @@ fn append_zip_entry<W: std::io::Write +
     compression_level: Option<i64>,
     tree_prefix: Option<&bstr::BString>,
 ) -> Result<(), Error> {
-    let file_opts = zip::write::FileOptions::<'_, ()>::default()
+    let file_opts = zip::write::FileOptions::default()
         .compression_method(zip::CompressionMethod::Deflated)
-        .compression_level(compression_level)
+        .compression_level(compression_level.map(|v| v as i32))
         .large_file(entry.bytes_remaining().map_or(true, |len| len > u32::MAX as usize))
         .last_modified_time(mtime)
         .unix_permissions(if entry.mode.is_executable() { 0o755 } else { 0o644 });
