diff --git a/Cargo.toml b/Cargo.toml
index 7d353fb..3fda8ee 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -109,37 +109,10 @@ optional = true
 [dependencies.dyn-clone]
 version = "1"
 
-[dependencies.eax]
-version = "0.5"
-optional = true
-
 [dependencies.ecb]
 version = "0.1"
 optional = true
 
-[dependencies.ecdsa]
-version = "0.16"
-features = [
-    "hazmat",
-    "arithmetic",
-]
-optional = true
-
-[dependencies.ed25519]
-version = "1"
-features = ["std"]
-optional = true
-default-features = false
-
-[dependencies.ed25519-dalek]
-version = "1"
-features = [
-    "rand",
-    "u64_backend",
-]
-optional = true
-default-features = false
-
 [dependencies.flate2]
 version = "1.0.1"
 optional = true
@@ -153,7 +126,7 @@ version = "0.5"
 optional = true
 
 [dependencies.idna]
-version = ">=0.2, <0.4"
+version = ">=0.2, <0.5"
 
 [dependencies.lalrpop-util]
 version = ">=0.17, <0.20"
@@ -193,24 +166,11 @@ optional = true
 version = "0.9.82"
 optional = true
 
-[dependencies.p256]
-version = "0.13"
-features = [
-    "ecdh",
-    "ecdsa",
-]
-optional = true
-
 [dependencies.rand]
 version = "0.8"
 optional = true
 package = "rand"
 
-[dependencies.rand07]
-version = "0.7.3"
-optional = true
-package = "rand"
-
 [dependencies.rand_core]
 version = "0.6"
 optional = true
@@ -230,11 +190,6 @@ optional = true
 version = "0.9.0"
 optional = true
 
-[dependencies.sha-1]
-version = "0.10"
-features = ["oid"]
-optional = true
-
 [dependencies.sha1collisiondetection]
 version = "0.2.3"
 features = ["std"]
@@ -256,10 +211,6 @@ optional = true
 version = "1.12.0"
 optional = true
 
-[dependencies.x25519-dalek-ng]
-version = "1"
-optional = true
-
 [dependencies.xxhash-rust]
 version = "0.8"
 features = ["xxh3"]
@@ -300,50 +251,11 @@ compression-deflate = [
 ]
 crypto-botan = ["botan/botan3"]
 crypto-botan2 = ["botan"]
-crypto-cng = [
-    "cipher",
-    "eax",
-    "winapi",
-    "win-crypto-ng",
-    "ed25519-dalek",
-    "num-bigint-dig",
-]
 crypto-nettle = ["nettle"]
 crypto-openssl = [
     "openssl",
     "openssl-sys",
 ]
-crypto-rust = [
-    "aes",
-    "block-padding",
-    "blowfish",
-    "cast5",
-    "cfb-mode",
-    "cipher",
-    "des",
-    "digest",
-    "eax",
-    "ecb",
-    "ed25519",
-    "ed25519-dalek",
-    "generic-array",
-    "idea",
-    "md-5",
-    "num-bigint-dig",
-    "rand",
-    "rand07",
-    "ripemd",
-    "rsa",
-    "sha-1",
-    "sha2",
-    "twofish",
-    "typenum",
-    "x25519-dalek-ng",
-    "p256",
-    "rand_core",
-    "rand_core/getrandom",
-    "ecdsa",
-]
 default = [
     "compression",
     "crypto-nettle",
@@ -361,19 +273,6 @@ default-features = false
 version = "0.2"
 features = ["js"]
 
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.rand07]
-version = "0.7"
-features = ["wasm-bindgen"]
-package = "rand"
-
-[target."cfg(windows)".dependencies.win-crypto-ng]
-version = ">=0.4, <0.5.1"
-features = [
-    "rand",
-    "block-cipher",
-]
-optional = true
-
 [target."cfg(windows)".dependencies.winapi]
 version = "0.3.8"
 features = ["bcrypt"]
