Index: sequoia-openpgp/Cargo.toml
===================================================================
--- sequoia-openpgp.orig/Cargo.toml
+++ sequoia-openpgp/Cargo.toml
@@ -332,8 +332,8 @@ default-features = false
 
 [features]
 __implicit-crypto-backend-for-tests = []
-allow-experimental-crypto = []
-allow-variable-time-crypto = []
+#allow-experimental-crypto = []
+#allow-variable-time-crypto = []
 compression = [
     "compression-deflate",
     "compression-bzip2",
