diff --git a/Cargo.toml b/Cargo.toml
index a2ffdc7..aa1779b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -130,7 +130,7 @@ optional = true
 version = ">=0.2, <0.5"
 
 [dependencies.lalrpop-util]
-version = ">=0.17, <0.20"
+version = ">=0.17, <0.21"
 
 [dependencies.lazy_static]
 version = "1.4.0"
@@ -231,7 +231,7 @@ version = "0.8"
 version = "6.0"
 
 [build-dependencies.lalrpop]
-version = ">=0.17, <0.20"
+version = ">=0.17, <0.21"
 default-features = false
 
 [features]
