Index: sequoia-openpgp/Cargo.toml
===================================================================
--- sequoia-openpgp.orig/Cargo.toml
+++ sequoia-openpgp/Cargo.toml
@@ -75,7 +75,7 @@ version = "1.0.18"
 version = "0.5"
 
 [dependencies.base64]
-version = ">= 0.21, < 0.23"
+version = "0.22"
 
 [dependencies.block-padding]
 version = "0.3"
@@ -87,7 +87,7 @@ features = ["zeroize"]
 optional = true
 
 [dependencies.botan]
-version = ">= 0.10.6, < 0.12"
+version = "0.10.6"
 optional = true
 
 [dependencies.buffered-reader]
@@ -95,7 +95,7 @@ version = "1.3.0"
 default-features = false
 
 [dependencies.bzip2]
-version = ">= 0.4, < 0.6"
+version = "0.4"
 optional = true
 
 [dependencies.camellia]
@@ -194,7 +194,7 @@ features = ["oid"]
 optional = true
 
 [dependencies.memsec]
-version = ">=0.5, <0.8"
+version = "0.7"
 default-features = false
 
 [dependencies.nettle]
