Index: sequoia-openpgp/Cargo.toml
===================================================================
--- sequoia-openpgp.orig/Cargo.toml
+++ sequoia-openpgp/Cargo.toml
@@ -87,7 +87,7 @@ features = ["zeroize"]
 optional = true
 
 [dependencies.botan]
-version = "0.10.6"
+version = "0.12"
 optional = true
 
 [dependencies.buffered-reader]
@@ -348,9 +348,7 @@ compression-deflate = [
 ]
 crypto-botan = [
     "dep:botan",
-    "botan?/botan3",
 ]
-crypto-botan2 = ["dep:botan"]
 crypto-fuzzing = []
 crypto-nettle = ["dep:nettle"]
 crypto-openssl = [
