Index: lalrpop/Cargo.toml
===================================================================
--- lalrpop.orig/Cargo.toml
+++ lalrpop/Cargo.toml
@@ -40,7 +40,7 @@ name = "lalrpop"
 required-features = ["pico-args"]
 
 [dependencies.ascii-canvas]
-version = "3.0"
+version = "2.0"
 default_features = false
 
 [dependencies.bit-set]
@@ -71,7 +71,7 @@ version = "0.6"
 default_features = false
 
 [dependencies.pico-args]
-version = "0.5"
+version = "0.4"
 optional = true
 default_features = false
 
@@ -89,7 +89,7 @@ version = "0.8"
 default_features = false
 
 [dependencies.term]
-version = "0.7"
+version = "0.5"
 default_features = false
 
 [dependencies.tiny-keccak]
