--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,7 +37,7 @@
 version = "1.0"
 
 [dependencies.cargo_metadata]
-version = "0.19"
+version = "0.23"
 
 [dependencies.clap]
 version = "4.5.16"
@@ -50,7 +50,7 @@
 version = "0.5"
 
 [dependencies.dirs]
-version = "6"
+version = "5"
 
 [dependencies.env_logger]
 version = "0.11"
--- a/src/debian.rs
+++ b/src/debian.rs
@@ -49,7 +49,7 @@
     pub fn new(pkg: Package) -> Pkg {
         Pkg {
             id: pkg.id,
-            name: pkg.name,
+            name: pkg.name.to_string(),
             version: pkg.version,
             source: pkg.source,
             manifest_path: pkg.manifest_path.into(),
