--- git-absorb.orig/Cargo.toml
+++ git-absorb/Cargo.toml
@@ -28,3 +28,3 @@
 [dependencies.anyhow]
-version = "~1.0"
+version = "1.0"
 
@@ -35,3 +35,3 @@
 [dependencies.clap]
-version = "~2.33"
+version = "2.33"
 
@@ -50,14 +50,14 @@
 [dependencies.memchr]
-version = "~2.3"
+version = "2.3"
 
 [dependencies.slog]
-version = "~2.5"
+version = "2.5"
 
 [dependencies.slog-async]
-version = "~2.5"
+version = "2.5"
 
 [dependencies.slog-term]
-version = "~2.6"
+version = "2.6"
 
 [dev-dependencies.tempfile]
-version = "~3.1"
+version = "3.1"
