--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,11 +44,8 @@ features = [
 [dependencies.clap_complete]
 version = "4"
 
-[dependencies.clap_complete_nushell]
-version = "4"
-
 [dependencies.git2]
-version = "0.20"
+version = "0.19"
 default-features = false
 
 [dependencies.memchr]
--- a/src/main.rs
+++ b/src/main.rs
@@ -3,7 +3,7 @@ extern crate slog;
 
 use clap::{CommandFactory, Parser as _};
 use clap_complete::{generate, Shell};
-use clap_complete_nushell::Nushell;
+//use clap_complete_nushell::Nushell;
 use slog::Drain;
 use std::io;
 
@@ -55,7 +55,7 @@ fn main() {
         match shell.as_str() {
             "bash" => generate(Shell::Bash, &mut cmd, app_name, &mut io::stdout()),
             "fish" => generate(Shell::Fish, &mut cmd, app_name, &mut io::stdout()),
-            "nushell" => generate(Nushell, &mut cmd, app_name, &mut io::stdout()),
+            //"nushell" => generate(Nushell, &mut cmd, app_name, &mut io::stdout()),
             "zsh" => generate(Shell::Zsh, &mut cmd, app_name, &mut io::stdout()),
             "powershell" => generate(Shell::PowerShell, &mut cmd, app_name, &mut io::stdout()),
             "elvish" => generate(Shell::Elvish, &mut cmd, app_name, &mut io::stdout()),
