From: Matthias Geiger <werdahias@riseup.net>
Date: Sun, 25 Feb 2024 23:50:42 +0100
Subject: Remove Windows dependencies

Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -83,10 +83,6 @@
     "gdk-wayland",
     "gst-gl-wayland",
 ]
-winegl = [
-    "gdk-win32/egl",
-    "gst-gl-egl",
-]
 x11egl = [
     "gdk-x11",
     "gst-gl-egl",
@@ -172,26 +168,3 @@
 
 [build-dependencies.gst-plugin-version-helper]
 version = "0.8"
-
-[target.'cfg(target_os = "macos")'.dependencies.gst-gl]
-version = "0.24"
-features = ["v1_16"]
-package = "gstreamer-gl"
-
-[target.'cfg(target_os = "windows")'.dependencies.gdk-win32]
-version = "0.10"
-features = ["v4_4"]
-package = "gdk4-win32"
-
-[target.'cfg(target_os = "windows")'.dependencies.gst-gl]
-version = "0.24"
-features = ["v1_20"]
-package = "gstreamer-gl"
-
-[target.'cfg(target_os = "windows")'.dependencies.windows-sys]
-version = ">=0.52, <=0.61"
-features = [
-    "Win32_Graphics_OpenGL",
-    "Win32_Foundation",
-    "Win32_Graphics_Gdi",
-]
