Index: gst-plugin-gtk4/Cargo.toml
===================================================================
--- gst-plugin-gtk4.orig/Cargo.toml
+++ gst-plugin-gtk4/Cargo.toml
@@ -113,10 +113,6 @@ wayland = [
     "gst_gl",
     "gst_gl_wayland",
 ]
-winegl = [
-    "gdk_win32/egl",
-    "gst_gl_egl",
-]
 x11egl = [
     "gtk/v4_6",
     "gdk_x11",
@@ -129,36 +125,3 @@ x11glx = [
     "gst_gl",
     "gst_gl_x11",
 ]
-
-[target."cfg(target_os = \"macos\")".dependencies.gst_gl]
-version = "0.21"
-features = ["v1_16"]
-package = "gstreamer-gl"
-
-[target."cfg(target_os = \"macos\")".dependencies.gtk]
-version = "0.7"
-features = ["v4_6"]
-package = "gtk4"
-
-[target."cfg(target_os = \"windows\")".dependencies.gdk_win32]
-version = "0.7"
-features = ["v4_4"]
-package = "gdk4-win32"
-
-[target."cfg(target_os = \"windows\")".dependencies.gst_gl]
-version = "0.21"
-features = ["v1_20"]
-package = "gstreamer-gl"
-
-[target."cfg(target_os = \"windows\")".dependencies.gtk]
-version = "0.7"
-features = ["v4_6"]
-package = "gtk4"
-
-[target."cfg(target_os = \"windows\")".dependencies.windows-sys]
-version = "0.48"
-features = [
-    "Win32_Graphics_OpenGL",
-    "Win32_Foundation",
-    "Win32_Graphics_Gdi",
-]
