diff --git a/Cargo.toml b/Cargo.toml
index 4bb0a7f..595484e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,27 +32,27 @@ doctest = false
 version = "2"
 
 [dependencies.bstr]
-version = "1.3.0"
+version = "1"
 features = ["std"]
 default-features = false
 
 [dependencies.gix-attributes]
-version = "^0.22.0"
+version = "0.20"
 
 [dependencies.gix-config-value]
-version = "^0.14.4"
+version = "0.14"
 
 [dependencies.gix-glob]
-version = "^0.16.0"
+version = "0.14"
 
 [dependencies.gix-path]
-version = "^0.10.4"
+version = "0.10"
 
 [dependencies.thiserror]
-version = "1.0.26"
+version = "1"
 
 [dev-dependencies.once_cell]
-version = "1.12.0"
+version = "1"
 
 [dev-dependencies.serial_test]
-version = "2.0.0"
+version = "2"
