only in patch2:
--- futures-lite.orig/Cargo.toml
+++ futures-lite/Cargo.toml
@@ -61,5 +61,2 @@
 
-[dev-dependencies.spin_on]
-version = "0.1.0"
-
 [dev-dependencies.waker-fn]
only in patch2:
--- futures-lite.orig/src/future.rs
+++ futures-lite/src/future.rs
@@ -4,3 +4,3 @@
 //!
-//! ```
+//! ``` ignore
 //! use futures_lite::future;
@@ -110,3 +110,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future;
@@ -158,3 +158,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future;
@@ -211,3 +211,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future;
@@ -245,3 +245,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future;
@@ -328,3 +328,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future;
@@ -337,3 +337,3 @@
 /// # })
-/// ```
+/// ``` ignore
 pub fn try_zip<T1, T2, E, F1, F2>(future1: F1, future2: F2) -> TryZip<F1, T1, F2, T2>
@@ -404,3 +404,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future::{self, pending, ready};
@@ -414,3 +414,3 @@
 /// # })
-/// ```
+/// ``` ignore
 pub fn or<T, F1, F2>(future1: F1, future2: F2) -> Or<F1, F2>
@@ -465,3 +465,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future::{self, pending, ready};
@@ -499,3 +499,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future::{self, pending, ready};
@@ -595,3 +595,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::future::{self, FutureExt};
@@ -636,3 +636,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::future::{pending, ready, FutureExt};
@@ -668,3 +668,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::future::{pending, ready, FutureExt};
@@ -696,3 +696,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::future::FutureExt;
@@ -719,3 +719,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::future::{self, FutureExt};
@@ -743,3 +743,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::future::{self, FutureExt};
@@ -754,3 +754,3 @@
     /// # })
-    /// ```
+    /// ``` ignore
     #[cfg(feature = "alloc")]
only in patch2:
--- futures-lite.orig/src/io.rs
+++ futures-lite/src/io.rs
@@ -4,3 +4,3 @@
 //!
-//! ```
+//! ``` ignore
 //! use futures_lite::io::{self, AsyncReadExt};
@@ -49,3 +49,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{self, BufReader, BufWriter};
@@ -118,3 +118,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{AssertAsync, AsyncReadExt};
@@ -649,3 +649,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, BufReader};
@@ -960,3 +960,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncWriteExt, BufWriter};
@@ -1062,3 +1062,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncWriteExt, BufWriter};
@@ -1187,3 +1187,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt, Cursor, SeekFrom};
@@ -1281,3 +1281,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncSeekExt, Cursor, SeekFrom};
@@ -1427,3 +1427,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{self, AsyncReadExt};
@@ -1474,3 +1474,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{self, AsyncReadExt};
@@ -1509,3 +1509,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{self, AsyncWriteExt};
@@ -1552,3 +1552,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, BufReader};
@@ -1584,3 +1584,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, BufReader};
@@ -1613,3 +1613,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, BufReader};
@@ -1646,3 +1646,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, BufReader};
@@ -1678,3 +1678,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, BufReader};
@@ -1712,3 +1712,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncBufReadExt, Cursor};
@@ -1977,3 +1977,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, BufReader};
@@ -2016,3 +2016,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, Cursor};
@@ -2046,3 +2046,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, Cursor};
@@ -2076,3 +2076,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, Cursor};
@@ -2101,3 +2101,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, Cursor};
@@ -2124,3 +2124,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, Cursor};
@@ -2151,3 +2151,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncReadExt, Cursor};
@@ -2729,3 +2729,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncSeekExt, Cursor, SeekFrom};
@@ -2784,3 +2784,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncWriteExt, BufWriter};
@@ -2821,3 +2821,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncWriteExt, BufWriter};
@@ -2842,3 +2842,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncWriteExt, BufWriter};
@@ -2864,3 +2864,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::io::{AsyncWriteExt, BufWriter};
@@ -3031,3 +3031,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::io::{self, Cursor};
only in patch2:
--- futures-lite.orig/src/lib.rs
+++ futures-lite/src/lib.rs
@@ -106,3 +106,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::{future, pin};
only in patch2:
--- futures-lite.orig/src/stream.rs
+++ futures-lite/src/stream.rs
@@ -4,3 +4,3 @@
 //!
-//! ```
+//! ``` ignore
 //! use futures_lite::stream::{self, StreamExt};
@@ -141,3 +141,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -180,3 +180,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -222,3 +222,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -265,3 +265,3 @@
 ///
-/// ```no_run
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -305,3 +305,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -354,3 +354,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -393,3 +393,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -439,3 +439,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -528,3 +528,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, StreamExt};
@@ -623,3 +623,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::{stream, prelude::*};
@@ -682,3 +682,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -710,3 +710,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -733,3 +733,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -758,3 +758,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -783,3 +783,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -813,3 +813,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -840,3 +840,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::pin;
@@ -872,3 +872,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -899,3 +899,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -924,3 +924,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -946,3 +946,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -973,3 +973,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -995,3 +995,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1026,3 +1026,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1055,3 +1055,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1085,3 +1085,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1109,3 +1109,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1133,3 +1133,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1156,3 +1156,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1185,3 +1185,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1216,3 +1216,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1246,3 +1246,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1283,3 +1283,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1314,3 +1314,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1339,3 +1339,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1365,3 +1365,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1389,3 +1389,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1417,3 +1417,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1439,3 +1439,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1464,3 +1464,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1489,3 +1489,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1511,3 +1511,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1540,3 +1540,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1571,3 +1571,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1600,3 +1600,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1620,3 +1620,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1656,3 +1656,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1686,3 +1686,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1713,3 +1713,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1740,3 +1740,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1820,3 +1820,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -1844,3 +1844,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::stream::{self, StreamExt};
@@ -2374,3 +2374,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, once, pending, StreamExt};
@@ -2426,3 +2426,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, once, pending, StreamExt};
@@ -2454,3 +2454,3 @@
 ///
-/// ```
+/// ``` ignore
 /// use futures_lite::stream::{self, once, pending, StreamExt};
@@ -3243,3 +3243,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::{prelude::*, stream};
@@ -3262,3 +3262,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::{prelude::*, stream};
@@ -3281,3 +3281,3 @@
     ///
-    /// ```
+    /// ``` ignore
     /// use futures_lite::{prelude::*, stream};
