Subject: These fail in autopkgtests
diff --git a/build.rs b/build.rs
index 534f7cd..9b4722c 100644
--- a/build.rs
+++ b/build.rs
@@ -173,6 +173,8 @@ fn main() {
     let dict_out = paths.place_item(dict_folder);
 
-    let _ext_langs = vec![Catalan, Hungarian];
+    let mut _ext_langs = vec![Catalan, Hungarian];
+    _ext_langs.remove(1);
+    _ext_langs.remove(0);
     let _std_langs = vec![Afrikaans,
                           Albanian,
                           Armenian,
