--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,10 +51,6 @@
 path = "examples/custom_files.rs"
 
 [[example]]
-name = "peg_calculator"
-path = "examples/peg_calculator.rs"
-
-[[example]]
 name = "readme_preview"
 path = "examples/readme_preview.rs"
 
@@ -95,15 +91,9 @@
 [dev-dependencies.lazy_static]
 version = "1.4"
 
-[dev-dependencies.peg]
-version = "0.7"
-
 [dev-dependencies.pico-args]
 version = "0.5.0"
 
-[dev-dependencies.rustyline]
-version = "6"
-
 [dev-dependencies.unindent]
 version = "0.1"
 
