diff --git a/Cargo.toml b/Cargo.toml
index 5a3d588..926890a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,20 +36,11 @@ version = "0.1"
 version = "1"
 
 [dev-dependencies.insta]
-version = "1.6.3"
+version = "1.21.0"
 
 [dev-dependencies.lazy_static]
 version = "1.4"
 
-[dev-dependencies.peg]
-version = "0.6"
-
-[dev-dependencies.rustyline]
-version = "6"
-
-[dev-dependencies.structopt]
-version = "0.3"
-
 [dev-dependencies.unindent]
 version = "0.1"
 
