Index: actix-http/Cargo.toml
===================================================================
--- actix-http.orig/Cargo.toml
+++ actix-http/Cargo.toml
@@ -136,10 +136,6 @@ name = "actix_http"
 path = "src/lib.rs"
 
 [[example]]
-name = "actix-web"
-path = "examples/actix-web.rs"
-
-[[example]]
 name = "bench"
 path = "examples/bench.rs"
 
@@ -203,10 +199,6 @@ path = "tests/test_rustls.rs"
 name = "test_server"
 path = "tests/test_server.rs"
 
-[[test]]
-name = "test_ws"
-path = "tests/test_ws.rs"
-
 [[bench]]
 name = "date-formatting"
 path = "benches/date-formatting.rs"
