Index: actix-http/Cargo.toml
===================================================================
--- actix-http.orig/Cargo.toml
+++ actix-http/Cargo.toml
@@ -135,10 +135,12 @@ path = "examples/echo2.rs"
 [[example]]
 name = "h2c-detect"
 path = "examples/h2c-detect.rs"
+required-features = ["http2"]
 
 [[example]]
 name = "h2spec"
 path = "examples/h2spec.rs"
+required-features = ["http2"]
 
 [[example]]
 name = "hello-world"
@@ -168,6 +170,7 @@ required-features = [
 [[test]]
 name = "test_h2_timer"
 path = "tests/test_h2_timer.rs"
+required-features = ["http2"]
 
 
 
