--- a/Cargo.toml
+++ b/Cargo.toml
@@ -184,7 +184,2 @@
 
-[dependencies.actix-tls]
-version = "3.4"
-optional = true
-default-features = false
-
 [dependencies.actix-utils]
@@ -228,3 +223,3 @@
 [dependencies.h2]
-version = "0.3.26"
+version = "0.4"
 optional = true
@@ -232,3 +227,3 @@
 [dependencies.http]
-version = "0.2.7"
+version = "1"
 
@@ -297,9 +292,2 @@
 
-[dev-dependencies.actix-tls]
-version = "3.4"
-features = [
-    "openssl",
-    "rustls-0_23-webpki-roots",
-]
-
 [dev-dependencies.actix-web]
@@ -387,29 +375,12 @@
 openssl = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/openssl",
 ]
 rustls = [
-    "__tls",
-    "rustls-0_20",
 ]
 rustls-0_20 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_20",
 ]
 rustls-0_21 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_21",
 ]
 rustls-0_22 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_22",
 ]
 rustls-0_23 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_23",
 ]
