Index: actix-http/Cargo.toml
===================================================================
--- actix-http.orig/Cargo.toml
+++ actix-http/Cargo.toml
@@ -208,7 +208,7 @@ optional = true
 version = "2"
 
 [dependencies.brotli]
-version = "8"
+version = ">= 6, < 9"
 optional = true
 
 [dependencies.bytes]
@@ -236,7 +236,7 @@ version = "1.0.13"
 optional = true
 
 [dependencies.foldhash]
-version = "0.1"
+version = "0.2"
 
 [dependencies.futures-core]
 version = "0.3.17"
@@ -244,11 +244,11 @@ features = ["alloc"]
 default-features = false
 
 [dependencies.h2]
-version = "0.3.27"
+version = "0.4"
 optional = true
 
 [dependencies.http]
-version = "0.2.7"
+version = "1"
 
 [dependencies.httparse]
 version = "1.5.1"
@@ -276,7 +276,7 @@ version = "2.1"
 version = "0.2"
 
 [dependencies.rand]
-version = "0.9"
+version = ">= 0.8, < 0.10"
 optional = true
 
 [dependencies.sha1]
