Index: actix-http/Cargo.toml
===================================================================
--- actix-http.orig/Cargo.toml
+++ actix-http/Cargo.toml
@@ -188,11 +188,6 @@ path = "tests/test_h2_timer.rs"
 
 
 [[bench]]
-name = "date-formatting"
-path = "benches/date-formatting.rs"
-harness = false
-
-[[bench]]
 name = "response-body-compression"
 path = "benches/response-body-compression.rs"
 harness = false
@@ -351,9 +346,6 @@ version = "0.3"
 version = "0.5"
 features = ["html_reports"]
 
-[dev-dependencies.divan]
-version = "0.1.8"
-
 [dev-dependencies.env_logger]
 version = "0.11"
 
