Index: actix-http/Cargo.toml
===================================================================
--- actix-http.orig/Cargo.toml
+++ actix-http/Cargo.toml
@@ -102,22 +102,7 @@ openssl = [
 ]
 rustls = [
     "__tls",
-    "rustls-0_20",
-]
-rustls-0_20 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_20",
-]
-rustls-0_21 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_21",
-]
-rustls-0_22 = [
-    "__tls",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_22",
+    "rustls-0_23",
 ]
 rustls-0_23 = [
     "__tls",
@@ -389,6 +374,8 @@ package = "openssl"
 [dev-dependencies.tls-rustls_023]
 version = "0.23"
 package = "rustls"
+default-features = false
+features = ["ring"]
 
 [dev-dependencies.tokio]
 version = "1.38.2"
