--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,9 +54,6 @@
 features = ["require-cas"]
 default-features = false
 
-[dev-dependencies.build-context]
-version = "0.1"
-
 [features]
 alloc = []
 default = []
--- a/tests/arc.rs
+++ b/tests/arc.rs
@@ -93,7 +93,7 @@
 
 // For -C panic=abort -Z panic_abort_tests: https://github.com/rust-lang/rust/issues/67650
 fn is_panic_abort() -> bool {
-    build_context::PANIC.contains("abort")
+    false
 }
 
 // https://github.com/rust-lang/rust/blob/1.80.0/library/alloc/src/sync/tests.rs
