--- rust-uutils-term-grid-0.6.0.orig/tests/test.rs
+++ rust-uutils-term-grid-0.6.0/tests/test.rs
@@ -193,7 +193,7 @@ fn emoji() {
 // checking that we do not get a panic.
 #[test]
 fn possible_underflow() {
-    let cells: Vec<_> = (0..48).map(|i| 2_isize.pow(i).to_string()).collect();
+    let cells: Vec<_> = (0..48).map(|i| 2_i64.pow(i).to_string()).collect();
 
     let grid = Grid::new(
         cells,
