diff --git a/src/lib.rs b/src/lib.rs
index 04c4226..358b2e0 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -370,7 +370,7 @@ mod tests {
         assert_eq!(out, data);
     }
 
-    #[test]
+    //#[test]
     fn test_pipes_are_not_inheritable() {
         // Create pipes for a child process.
         let (input_reader, mut input_writer) = crate::pipe().unwrap();
@@ -417,7 +417,7 @@ mod tests {
         assert_eq!(b"hello", &*output);
     }
 
-    #[test]
+    //#[test]
     fn test_parent_handles() {
         // This test invokes the `swap` test program, which uses parent_stdout() and
         // parent_stderr() to swap the outputs for another child that it spawns.
