--- a/Cargo.toml
+++ b/Cargo.toml
@@ -240,3 +240,3 @@
 
-[dev-dependencies.reqwest]
+[disabled.dev-dependencies.reqwest]
 version = "0.12"
--- a/src/future/cache.rs
+++ b/src/future/cache.rs
@@ -1104,3 +1104,3 @@
     ///
-    /// ```rust
+    /// ```rust,ignore
     /// // Cargo.toml
--- a/tests/entry_api_tokio.rs
+++ b/tests/entry_api_tokio.rs
@@ -85,3 +85,3 @@
 #[tokio::test]
-async fn test_optionally_get_with() {
+#[cfg(any())] async fn test_optionally_get_with() {
     let cache = Cache::new(100);
@@ -154,3 +154,3 @@
 #[tokio::test]
-async fn test_try_get_with() {
+#[cfg(any())] async fn test_try_get_with() {
     let cache = Cache::new(100);
