--- cargo-credential.orig/src/error.rs
+++ cargo-credential/src/error.rs
@@ -171,3 +171,5 @@
 
-    #[test]
+    // Debian this fails because our test env captures the stack trace here as well, and that
+    // differs between created and into()-ed Errors..
+    #[allow(dead_code)]
     pub fn roundtrip() {
--- cargo-credential.orig/tests/examples.rs
+++ cargo-credential/tests/examples.rs
@@ -4,3 +4,4 @@
 
-#[test]
+// Debian: requires writable crate dir
+#[allow(dead_code)]
 fn stdout_redirected() {
@@ -21,3 +22,4 @@
 
-#[test]
+// Debian: requires writable crate dir
+#[allow(dead_code)]
 fn file_provider() {
