From edfa94a428f72207933997c8b08d44cec0fbcce6 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 5 Mar 2025 16:41:43 -0500
Subject: [PATCH] python: bump version to 0.4.3

Forwarded: https://github.com/integritychain/fips203/pull/25

---
 python/fips203.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ffi/python/fips203.py b/ffi/python/fips203.py
index 3c764b3..a01b2a1 100644
--- a/python/fips203.py
+++ b/python/fips203.py
@@ -100,7 +100,7 @@ Please report issues at https://github.com/integritychain/fips203/issues
 from __future__ import annotations
 
 '''__version__ should track package.version from  ../Cargo.toml'''
-__version__ = '0.4.1'
+__version__ = '0.4.3'
 __author__ = 'Daniel Kahn Gillmor <dkg@fifthhorseman.net>'
 __all__ = [
     'ML_KEM_512',
-- 
2.47.2

