Index: der-parser/Cargo.toml
===================================================================
--- der-parser.orig/Cargo.toml
+++ der-parser/Cargo.toml
@@ -25,9 +25,6 @@ repository = "https://github.com/rustica
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
-[dependencies.bitvec]
-version = "0.22"
-optional = true
 
 [dependencies.cookie-factory]
 version = "0.3.0"
