--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,9 +52,9 @@
 
-[[example]]
-name = "akamai"
-path = "examples/akamai.rs"
-
-[[example]]
-name = "client"
-path = "examples/client.rs"
+#[[example]]
+#name = "akamai"
+#path = "examples/akamai.rs"
+
+#[[example]]
+#name = "client"
+#path = "examples/client.rs"
 
@@ -141,4 +141,4 @@
 
-[dev-dependencies.tokio-rustls]
-version = "0.26"
+#[dev-dependencies.tokio-rustls]
+#version = "0.26"
 
@@ -147,4 +147,4 @@
 
-[dev-dependencies.webpki-roots]
-version = "0.26"
+#[dev-dependencies.webpki-roots]
+#version = "0.26"
 
