--- a/Cargo.toml
+++ b/Cargo.toml
@@ -73,10 +73,10 @@
 path = "tests/value_delegate_derive.rs"
 
 [dependencies.heck]
-version = "0.5"
+version = "0.4"
 
 [dependencies.proc-macro-crate]
-version = "3.3"
+version = "3.2"
 
 [dependencies.proc-macro2]
 version = "1.0"
@@ -85,7 +85,7 @@
 version = "1.0"
 
 [dependencies.syn]
-version = "2.0.104"
+version = "2.0"
 features = ["full"]
 
 [dev-dependencies.glib]
