--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,2 +42,2 @@
-[dev-dependencies.criterion]
-version = "0.5.1"
+#[dev-dependencies.criterion]
+#version = "0.5.1"
@@ -45,2 +45,2 @@
-[dev-dependencies.criterion-plot]
-version = "0.5"
+#[dev-dependencies.criterion-plot]
+#version = "0.5"
--- a/benches/bench.rs
+++ b/benches/bench.rs
@@ -1 +1,4 @@
-#[macro_use]
+fn main() {
+	eprintln!("Debian: benches disabled");
+}
+/*#[macro_use]
@@ -411 +414 @@
-criterion_main!(mono_thread, multi_thread, forward_multi_thread);
+criterion_main!(mono_thread, multi_thread, forward_multi_thread);*/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,2 +48,2 @@
-[dev-dependencies.object-pool]
-version = "0.5"
+#[dev-dependencies.object-pool]
+#version = "0.5"
