--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,10 +41,10 @@
 
-[dev-dependencies.criterion]
-version = "0.5.1"
+#[dev-dependencies.criterion]
+#version = "0.5.1"
 
-[dev-dependencies.criterion-plot]
-version = "0.5"
+#[dev-dependencies.criterion-plot]
+#version = "0.5"
 
-[dev-dependencies.object-pool]
-version = "0.5"
+#[dev-dependencies.object-pool]
+#version = "0.5"
 
--- a/benches/bench.rs
+++ b/benches/bench.rs
@@ -1,2 +1,5 @@
-#[macro_use]
+fn main() {
+	eprintln!("Debian: benches disabled");
+}
+/*#[macro_use]
 extern crate criterion;
@@ -410,2 +413,2 @@
 criterion_group!(mono_thread, bench_alloc, bench_free, bench_reuse);
-criterion_main!(mono_thread, multi_thread, forward_multi_thread);
+criterion_main!(mono_thread, multi_thread, forward_multi_thread);*/
