Index: maxminddb/Cargo.toml
===================================================================
--- maxminddb.orig/Cargo.toml
+++ maxminddb/Cargo.toml
@@ -49,7 +49,9 @@ path = "src/maxminddb/lib.rs"
 #harness = false
 
 [dependencies.ipnetwork]
-version = "0.20.0"
+version = ">= 0.17, < 0.22"
+default_features = false
+features = ["serde"]
 
 [dependencies.log]
 version = "0.4"
@@ -69,7 +71,7 @@ features = ["derive"]
 #version = "0.5"
 
 [dev-dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 
 #[dev-dependencies.fake]
 #version = "2.4"
