Last-Update: 2025-09-02
Author: Blair Noctis <ncts@debian.org>
Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,6 +41,6 @@
 [features]
-web_spin_lock = [
-    "dep:wasm_sync",
-    "rayon-core/web_spin_lock",
-]
+#web_spin_lock = [
+#    "dep:wasm_sync",
+#    "rayon-core/web_spin_lock",
+#]
 
@@ -121,3 +121,3 @@
 
-[dependencies.wasm_sync]
+[disabled.dependencies.wasm_sync]
 version = "0.1.0"
