--- rcgen.orig/tests/webpki.rs
+++ rcgen/tests/webpki.rs
@@ -107,2 +107,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki() {
@@ -119,2 +120,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_256() {
@@ -133,2 +135,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_384() {
@@ -147,2 +150,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_25519() {
@@ -161,2 +165,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_25519_v1_given() {
@@ -178,2 +183,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_25519_v2_given() {
@@ -195,2 +201,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_rsa_given() {
@@ -217,2 +224,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_rsa_combinations_given() {
@@ -315,2 +323,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn from_remote() {
@@ -517,2 +526,3 @@
 #[test]
+#[cfg(feature = "pem")]
 fn test_webpki_serial_number() {
--- rcgen.orig/examples/rsa-irc-openssl.rs
+++ rcgen/examples/rsa-irc-openssl.rs
@@ -0,0 +1 @@
+#![cfg(feature = "pem")]
--- rcgen.orig/examples/rsa-irc.rs
+++ rcgen/examples/rsa-irc.rs
@@ -0,0 +1 @@
+#![cfg(feature = "pem")]
