Index: rcgen/Cargo.toml
===================================================================
--- rcgen.orig/Cargo.toml
+++ rcgen/Cargo.toml
@@ -75,10 +75,6 @@ path = "tests/openssl.rs"
 name = "util"
 path = "tests/util.rs"
 
-[[test]]
-name = "webpki"
-path = "tests/webpki.rs"
-
 [dependencies.aws-lc-rs]
 version = "1.6.0"
 optional = true
@@ -124,10 +120,6 @@ version = "0.8"
 [dev-dependencies.ring]
 version = "0.17"
 
-[dev-dependencies.rustls-webpki]
-version = "0.102"
-features = ["std"]
-
 [features]
 aws_lc_rs = [
     "crypto",
