Index: rcgen/Cargo.toml
===================================================================
--- rcgen.orig/Cargo.toml
+++ rcgen/Cargo.toml
@@ -71,10 +71,6 @@ required-features = [
 ]
 
 [[test]]
-name = "botan"
-path = "tests/botan.rs"
-
-[[test]]
 name = "generic"
 path = "tests/generic.rs"
 
@@ -127,10 +123,6 @@ features = [
 version = "1.2"
 optional = true
 
-[dev-dependencies.botan]
-version = "0.10"
-features = ["vendored"]
-
 [dev-dependencies.openssl]
 version = "0.10"
 
