Index: rcgen/Cargo.toml
===================================================================
--- rcgen.orig/Cargo.toml
+++ rcgen/Cargo.toml
@@ -42,26 +42,12 @@ allowed_external_types = [
 features = ["x509-parser"]
 
 [features]
-aws_lc_rs = [
-    "crypto",
-    "dep:aws-lc-rs",
-    "aws-lc-rs/aws-lc-sys",
-]
-aws_lc_rs_unstable = [
-    "aws_lc_rs",
-    "aws-lc-rs/unstable",
-]
 crypto = []
 default = [
     "crypto",
     "pem",
     "ring",
 ]
-fips = [
-    "crypto",
-    "dep:aws-lc-rs",
-    "aws-lc-rs/fips",
-]
 ring = [
     "crypto",
     "dep:ring",
@@ -92,11 +78,6 @@ required-features = [
     "pem",
 ]
 
-[dependencies.aws-lc-rs]
-version = "1.13.3"
-optional = true
-default-features = false
-
 [dependencies.pem]
 version = "3.0.2"
 optional = true
