Index: watchman-client/Cargo.toml
===================================================================
--- watchman-client.orig/Cargo.toml
+++ watchman-client/Cargo.toml
@@ -28,10 +28,9 @@ version = "1.0"
 features = ["serde"]
 
 [dependencies.futures]
-version = "0.3.13"
+version = ">= 0.3.0"
 features = [
     "async-await",
-    "compat",
 ]
 
 [dependencies.maplit]
@@ -51,14 +50,14 @@ version = "0.4"
 version = "1.0"
 
 [dependencies.tokio]
-version = "1.7.1"
+version = ">= 1.7.1"
 features = [
     "full",
     "test-util",
 ]
 
 [dependencies.tokio-util]
-version = "0.6"
+version = "0.7"
 features = ["full"]
 
 [dev-dependencies.clap]
