Index: tinyvec/Cargo.toml
===================================================================
--- tinyvec.orig/Cargo.toml
+++ tinyvec/Cargo.toml
@@ -75,7 +75,6 @@ name = "smallvec"
 harness = false
 required-features = [
     "alloc",
-    "real_blackbox",
 ]
 
 [dependencies.arbitrary]
@@ -106,7 +105,6 @@ default = []
 experimental_write_impl = []
 grab_spare_slice = []
 nightly_slice_partition_dedup = []
-real_blackbox = ["criterion/real_blackbox"]
 rustc_1_40 = []
 rustc_1_55 = ["rustc_1_40"]
 rustc_1_57 = ["rustc_1_55"]
