Index: async-signal/Cargo.toml
===================================================================
--- async-signal.orig/Cargo.toml
+++ async-signal/Cargo.toml
@@ -61,7 +61,7 @@ version = "1.0.0"
 version = "0.3.26"
 
 [dev-dependencies.async-io]
-version = "2.0.0"
+version = "2.5.0"
 
 [dev-dependencies.fastrand]
 version = "2.0.1"
@@ -73,7 +73,7 @@ version = "2.3.0"
 version = "0.3.14"
 
 [target."cfg(unix)".dependencies.async-io]
-version = "2.0.0"
+version = "2.5.0"
 
 [target."cfg(unix)".dependencies.futures-io]
 version = "0.3.26"
