Index: pest-derive/Cargo.toml
===================================================================
--- pest-derive.orig/Cargo.toml
+++ pest-derive/Cargo.toml
@@ -44,7 +44,6 @@ default-features = false
 [features]
 default = ["std"]
 grammar-extras = ["pest_generator/grammar-extras"]
-not-bootstrap-in-src = ["pest_generator/not-bootstrap-in-src"]
 std = [
     "pest/std",
     "pest_generator/std",
