From: Matthias Geiger <werdahias@riseup.net>
Date: Tue, 5 Sep 2023 23:24:49 +0200
Subject: Fix the include path so the test can run

Forwarded: not-needed
---
 tests/abi.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/abi.rs b/tests/abi.rs
index 3ab1fd3..f9a7601 100644
--- a/tests/abi.rs
+++ b/tests/abi.rs
@@ -32,6 +32,7 @@ impl Compiler {
         args.extend(get_var("CFLAGS", "")?);
         args.extend(get_var("CPPFLAGS", "")?);
         args.extend(pkg_config_cflags(PACKAGES)?);
+        args.push("-I/usr/include/gtk-4.0/unix-print/".to_owned());
         Ok(Self { args })
     }
 
