Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,11 +72,11 @@
 version = "1.3.2"
 
 [dependencies.clap_complete]
-version = "4.5.44"
+version = "4.5"
 features = ["unstable-dynamic"]
 
 [dependencies.cli-log]
-version = "2.1"
+version = "2"
 
 [dependencies.crokey]
 version = "1.1"
@@ -88,7 +88,7 @@
 version = "2.0.0"
 
 [dependencies.gix]
-version = "0.67"
+version = "0.69"
 features = [
     "index",
     "excludes",
@@ -107,7 +107,7 @@
 version = "3.4.1"
 
 [dependencies.notify]
-version = "7.0"
+version = "6"
 
 [dependencies.rodio]
 version = "0.20"
@@ -126,7 +126,7 @@
 version = "1.0"
 
 [dependencies.termimad]
-version = "0.31.1"
+version = "0.33"
 
 [dependencies.toml]
 version = "0.8"
