Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,3 +61,3 @@
 [dependencies.cargo_metadata]
-version = "0.19"
+version = "0.15"
 
@@ -74,3 +74,3 @@
 [dependencies.clap_complete]
-version = "4.5.44"
+version = "4.5"
 features = ["unstable-dynamic"]
@@ -78,3 +78,3 @@
 [dependencies.cli-log]
-version = "2.1"
+version = "2"
 
@@ -90,3 +90,3 @@
 [dependencies.gix]
-version = "0.67"
+version = "0.69"
 features = [
@@ -109,3 +109,3 @@
 [dependencies.notify]
-version = "7.0"
+version = "6"
 
