Index: cargo/Cargo.toml
===================================================================
--- cargo.orig/Cargo.toml
+++ cargo/Cargo.toml
@@ -168,7 +168,7 @@ version = "0.1"
 version = "0.8.7"
 
 [features]
-vendored-openssl = ["openssl/vendored"]
+vendored-openssl = ["openssl"]
 [target."cfg(target_os = \"macos\")".dependencies.core-foundation]
 version = "0.6.0"
 features = ["mac_os_10_7_support"]
