Index: cargo/Cargo.toml
===================================================================
--- cargo.orig/Cargo.toml
+++ cargo/Cargo.toml
@@ -283,21 +283,3 @@ optional = true
 
 [target."cfg(target_os = \"linux\")".dependencies.cargo-credential-libsecret]
 version = "0.4.1"
-
-[target."cfg(target_os = \"macos\")".dependencies.cargo-credential-macos-keychain]
-version = "0.4.1"
-
-[target."cfg(windows)".dependencies.cargo-credential-wincred]
-version = "0.4.1"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.48"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_Console",
-    "Win32_System_JobObjects",
-    "Win32_System_Threading",
-]
