Index: cargo/Cargo.toml
===================================================================
--- cargo.orig/Cargo.toml
+++ cargo/Cargo.toml
@@ -75,10 +75,10 @@ features = ["zlib"]
 default-features = false
 
 [dependencies.git2]
-version = "0.16.0"
+version = "0.18.0"
 
 [dependencies.git2-curl]
-version = "0.17.0"
+version = "0.19.0"
 
 [dependencies.glob]
 version = "0.3.0"
@@ -120,7 +120,7 @@ version = "1.2.0"
 version = "0.2"
 
 [dependencies.libgit2-sys]
-version = "0.14.1"
+version = "0.16"
 
 [dependencies.log]
 version = "0.4.6"
