Index: cargo/Cargo.toml
===================================================================
--- cargo.orig/Cargo.toml
+++ cargo/Cargo.toml
@@ -194,7 +194,7 @@ deny-warnings = []
 pretty-env-logger = []
 vendored-openssl = ["openssl"]
 [target."cfg(target_os = \"macos\")".dependencies.core-foundation]
-version = "0.7.0"
+version = ">= 0.7, < 0.10"
 features = ["mac_os_10_7_support"]
 [target."cfg(windows)".dependencies.fwdansi]
 version = "1"
