Index: cargo/Cargo.toml
===================================================================
--- cargo.orig/Cargo.toml
+++ cargo/Cargo.toml
@@ -88,7 +88,7 @@ version = "0.18.1"
 version = "0.19.0"
 
 [dependencies.gix]
-version = "0.55.2"
+version = "0.62"
 features = [
     "blocking-http-transport-curl",
     "progress-tree",
@@ -97,7 +97,7 @@ features = [
 default-features = false
 
 [dependencies.gix-features-for-configuration-only]
-version = "0.35.0"
+version = "0.38.1"
 features = ["parallel"]
 package = "gix-features"
 
