Index: cargo/Cargo.toml
===================================================================
--- cargo.orig/Cargo.toml
+++ cargo/Cargo.toml
@@ -72,10 +72,10 @@ features = ["zlib"]
 default-features = false
 
 [dependencies.git2]
-version = "0.13.16"
+version = "0.14"
 
 [dependencies.git2-curl]
-version = "0.14.1"
+version = "0.15"
 
 [dependencies.glob]
 version = "0.3.0"
@@ -111,7 +111,7 @@ version = "1.2.0"
 version = "0.2"
 
 [dependencies.libgit2-sys]
-version = "0.12.18"
+version = "0.13"
 
 [dependencies.log]
 version = "0.4.6"
