Description: Make botan3 a default feature
Author: Bastian Germann
Bug-Debian: https://bugs.debian.org/1109768
---
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,7 +16,7 @@ name = "botan-sys"
 version = "0.10.5"
 authors = ["Jack Lloyd <jack@randombit.net>"]
 build = "build.rs"
-links = "botan-2"
+links = "botan-3"
 description = "FFI wrapper for Botan cryptography library"
 homepage = "https://botan.randombit.net/"
 documentation = "https://docs.rs/botan-sys"
@@ -36,7 +36,7 @@ resolver = "1"
 
 [features]
 botan3 = []
-default = []
+default = ["botan3"]
 no-std = []
 #vendored = [
 #    "botan-src",
