--- a/Cargo.toml
+++ b/Cargo.toml
@@ -66,22 +66,13 @@ version = "0.30.0"
 optional = true
 
 [dependencies.xkb]
-version = "0.3"
+version = "0.2"
 optional = true
 
 [dev-dependencies.gir-format-check]
-version = "^0.1"
+version = "0.1"
 
 [features]
-dox = [
-    "ffi/dox",
-    "gdk/dox",
-    "gio/dox",
-    "glib/dox",
-    "xkb_crate",
-    "wayland_crate",
-    "egl",
-]
 egl = ["khronos-egl"]
 v4_10 = [
     "ffi/v4_10",
