--- a/Cargo.toml
+++ b/Cargo.toml
@@ -117 +117 @@
-[dependencies.hashbrown]
+[disabled.dependencies.hashbrown]
@@ -214 +214 @@
-    "dep:hashbrown",
+    "hashbrown-0_14",
@@ -228 +228 @@
-hashbrown-0_15 = ["dep:hashbrown"]
+#hashbrown-0_15 = ["dep:hashbrown"]
--- a/src/impls/ext/mod.rs
+++ b/src/impls/ext/mod.rs
@@ -15,2 +15,2 @@
-#[cfg(feature = "hashbrown-0_15")]
-mod hashbrown_0_15;
+//#[cfg(feature = "hashbrown-0_15")]
+//mod hashbrown_0_15;
--- a/src/de/pooling/alloc.rs
+++ b/src/de/pooling/alloc.rs
@@ -3 +3 @@
-use hashbrown::hash_map::{Entry, HashMap};
+use hashbrown_0_14::hash_map::{Entry, HashMap};
--- a/src/ser/sharing/alloc.rs
+++ b/src/ser/sharing/alloc.rs
@@ -3 +3 @@
-use hashbrown::hash_map::{Entry, HashMap};
+use hashbrown_0_14::hash_map::{Entry, HashMap};
