--- a/Cargo.toml
+++ b/Cargo.toml
@@ -116,3 +116,3 @@
 
-[dependencies.hashbrown]
+[disabled.dependencies.hashbrown]
 version = "0.15"
@@ -213,3 +213,3 @@
 alloc = [
-    "dep:hashbrown",
+    "hashbrown-0_14",
     "tinyvec-1?/alloc",
@@ -227,3 +227,3 @@
 ]
-hashbrown-0_15 = ["dep:hashbrown"]
+#hashbrown-0_15 = ["dep:hashbrown"]
 indexmap-2 = [
--- a/src/impls/ext/mod.rs
+++ b/src/impls/ext/mod.rs
@@ -14,4 +14,4 @@
 mod hashbrown_0_14;
-#[cfg(feature = "hashbrown-0_15")]
-mod hashbrown_0_15;
+//#[cfg(feature = "hashbrown-0_15")]
+//mod hashbrown_0_15;
 #[cfg(feature = "indexmap-2")]
--- a/src/de/pooling/alloc.rs
+++ b/src/de/pooling/alloc.rs
@@ -2,3 +2,3 @@
 
-use hashbrown::hash_map::{Entry, HashMap};
+use hashbrown_0_14::hash_map::{Entry, HashMap};
 use rancor::{fail, Source};
--- a/src/ser/sharing/alloc.rs
+++ b/src/ser/sharing/alloc.rs
@@ -2,3 +2,3 @@
 
-use hashbrown::hash_map::{Entry, HashMap};
+use hashbrown_0_14::hash_map::{Entry, HashMap};
 use rancor::{fail, Source};
