--- rkyv.orig/Cargo.toml
+++ rkyv/Cargo.toml
@@ -94,2 +94,3 @@
 path = "examples/backwards_compat.rs"
+required-features = ["alloc", "bytecheck"]
 
@@ -98,2 +99,3 @@
 path = "examples/complex_wrapper_types.rs"
+required-features = ["alloc"]
 
@@ -102,2 +104,3 @@
 path = "examples/derive_partial_ord.rs"
+required-features = ["alloc"]
 
@@ -110,2 +113,3 @@
 path = "examples/json_like_schema.rs"
+required-features = ["alloc", "bytecheck", "std"]
 
@@ -114,2 +118,3 @@
 path = "examples/niching.rs"
+required-features = ["bytecheck"]
 
@@ -118,2 +123,3 @@
 path = "examples/readme.rs"
+required-features = ["alloc","bytecheck"]
 
@@ -122,2 +128,3 @@
 path = "examples/remote_types.rs"
+required-features = ["alloc", "bytecheck"]
 
