Index: rkyv/Cargo.toml
===================================================================
--- rkyv.orig/Cargo.toml
+++ rkyv/Cargo.toml
@@ -59,7 +59,6 @@ default = [
     "std",
     "bytecheck",
 ]
-hashbrown-0_15 = ["dep:hashbrown"]
 indexmap-2 = [
     "dep:indexmap-2",
     "alloc",
@@ -168,7 +167,7 @@ default-features = false
 package = "bytes"
 
 [dependencies.hashbrown]
-version = "0.15"
+version = ">= 0.15, < 0.17"
 optional = true
 default-features = false
 
@@ -241,7 +240,7 @@ default-features = false
 package = "uuid"
 
 [dev-dependencies.foldhash]
-version = "0.1.4"
+version = ">= 0.1.4, < 3"
 
 [disabled.dev-dependencies.divan]
 version = "0.1"
