--- a/Cargo.toml
+++ b/Cargo.toml
@@ -74,3 +74,3 @@
 [dependencies.ptr_meta]
-version = "~0.1.3"
+version = "0.3"
 default-features = false
--- a/src/collections/btree_map/mod.rs
+++ b/src/collections/btree_map/mod.rs
@@ -91,3 +91,3 @@
 
-impl<T> Pointee for Node<[T]> {
+unsafe impl<T> Pointee for Node<[T]> {
     type Metadata = usize;
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -495,3 +495,3 @@
 ///
-/// impl<H, T> Pointee for Block<H, [T]> {
+/// unsafe impl<H, T> Pointee for Block<H, [T]> {
 ///     type Metadata = usize;
