Index: rkyv/Cargo.toml
===================================================================
--- rkyv.orig/Cargo.toml
+++ rkyv/Cargo.toml
@@ -193,8 +193,8 @@ optional = true
 default-features = false
 package = "uuid"
 
-[dev-dependencies.ahash]
-version = "0.8"
+[dev-dependencies.foldhash]
+version = "=0.1.4"
 
 [disabled.dev-dependencies.divan]
 version = "0.1"
Index: rkyv/src/impls/std/collections/hash_map.rs
===================================================================
--- rkyv.orig/src/impls/std/collections/hash_map.rs
+++ rkyv/src/impls/std/collections/hash_map.rs
@@ -107,7 +107,7 @@ mod tests {
     use core::{fmt::Debug, hash::BuildHasher};
     use std::collections::HashMap;
 
-    use ahash::RandomState;
+    use foldhash::quality::RandomState;
 
     use crate::{
         api::test::{roundtrip, roundtrip_with, to_archived},
