--- a/Cargo.toml
+++ b/Cargo.toml
@@ -104,7 +104,7 @@
 default-features = false
 
 [dependencies.image_hasher]
-version = "2.0"
+version = "1"
 
 [dependencies.imagepipe]
 version = "0.5"
--- a/src/similar_images.rs
+++ b/src/similar_images.rs
@@ -862,7 +862,7 @@
         HashAlg::Blockhash => "Blockhash",
         HashAlg::VertGradient => "VertGradient",
         HashAlg::DoubleGradient => "DoubleGradient",
-        HashAlg::Median => "Median",
+        //HashAlg::Median => "Median",
     }
     .to_string()
 }
@@ -878,7 +878,7 @@
                 HashAlg::DoubleGradient,
                 HashAlg::VertGradient,
                 HashAlg::Mean,
-                HashAlg::Median,
+                //HashAlg::Median,
             ] {
                 for filter in [
                     FilterType::Lanczos3,
