Origin: vendor
Subject: disable crypto-cng
Forwarded: not-needed


Index: sequoia-octopus-librnp/Cargo.toml
===================================================================
--- sequoia-octopus-librnp.orig/Cargo.toml
+++ sequoia-octopus-librnp/Cargo.toml
@@ -59,10 +59,10 @@ crypto-botan2 = [
     "sequoia-openpgp/compression",
     "sequoia-openpgp/crypto-botan2",
 ]
-crypto-cng = [
-    "sequoia-openpgp/compression",
-    "sequoia-openpgp/crypto-cng",
-]
+#crypto-cng = [
+#    "sequoia-openpgp/compression",
+#    "sequoia-openpgp/crypto-cng",
+#]
 crypto-nettle = [
     "sequoia-openpgp/compression",
     "sequoia-openpgp/crypto-nettle",
