Origin: vendor
Subject: patch out Windows dependencies
Forwarded: not-needed

this should be handled automatically by debcargo...

Index: sequoia-octopus-librnp/Cargo.toml
===================================================================
--- sequoia-octopus-librnp.orig/Cargo.toml
+++ sequoia-octopus-librnp/Cargo.toml
@@ -172,7 +172,3 @@ features = [
     "git2",
 ]
 default-features = false
-
-[target."cfg(windows)".dependencies.rusqlite]
-version = ">=0.24, <0.32"
-features = ["bundled"]
