Origin: vendor
Subject: patch out Windows dependencies
Forwarded: not-needed

this should be handled automatically by debcargo...

Index: sequoia-octopus-librnp/Cargo.toml
===================================================================
--- sequoia-octopus-librnp.orig/Cargo.toml
+++ sequoia-octopus-librnp/Cargo.toml
@@ -174,15 +174,3 @@ features = [
     "git2",
 ]
 default-features = false
-
-[target."cfg(windows)".dependencies.rusqlite]
-version = ">=0.24, <0.32"
-features = ["bundled"]
-
-[target."cfg(windows)".build-dependencies.vergen]
-version = "8"
-features = [
-    "git",
-    "gitcl",
-]
-default-features = false
