Origin: vendor
Subject: Relax dependencies to work well with other rust crates in Debian
Forwarded: not-needed

Index: sequoia-octopus-librnp/Cargo.toml
===================================================================
--- sequoia-octopus-librnp.orig/Cargo.toml
+++ sequoia-octopus-librnp/Cargo.toml
@@ -85,7 +85,7 @@ version = "1"
 version = "0.4"
 
 [dependencies.configparser]
-version = ">=2, <4"
+version = "<4"
 
 [dependencies.csv]
 version = "1.1"
@@ -146,7 +146,7 @@ version = "0.8"
 default-features = false
 
 [dependencies.sequoia-wot]
-version = "0.14"
+version = "0.15"
 default-features = false
 
 [dependencies.serde]
