version ranges don't work very well

diff --git a/Cargo.toml b/Cargo.toml
index c384dfb..7489bad 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,7 +52,7 @@ version = "1"
 version = "0.4"
 
 [dependencies.configparser]
-version = ">=2, <4"
+version = "<4"
 
 [dependencies.csv]
 version = "1.1"
@@ -82,14 +82,14 @@ version = "0.4"
 default-features = false
 
 [dependencies.rusqlite]
-version = ">=0.24, <0.31"
+version = "<0.31"
 
 [dependencies.sequoia-autocrypt]
-version = ">=0.24, <0.26"
+version = "<0.26"
 default-features = false
 
 [dependencies.sequoia-ipc]
-version = ">=0.28, <0.32"
+version = "<0.32"
 default-features = false
 
 [dependencies.sequoia-net]
@@ -117,7 +117,7 @@ features = ["derive"]
 version = "1"
 
 [dependencies.tempfile]
-version = "3.0"
+version = "3"
 
 [dependencies.thiserror]
 version = "1"
