Index: html2pango/Cargo.toml
===================================================================
--- html2pango.orig/Cargo.toml
+++ html2pango/Cargo.toml
@@ -38,7 +38,7 @@ version = "4"
 version = "1.0.65"
 
 [dependencies.html5ever]
-version = "0.26.0"
+version = "0.27.0"
 
 [dependencies.linkify]
 version = "0.9.0"
@@ -47,7 +47,7 @@ version = "0.9.0"
 version = "1.0.2"
 
 [dependencies.markup5ever_rcdom]
-version = "0.2.0"
+version = "0.3.0"
 
 [dependencies.once_cell]
 version = "1.15.0"
