Description: patch for svg >=0.16 compatibility
Author: Ananthu C V <weepingclown@disroot.org>
Bug: https://github.com/Canop/csv2svg/issues/5
Last-Update: 2024-04-25
--- a/src/graph.rs
+++ b/src/graph.rs
@@ -57,7 +57,7 @@
                 .set("height", 8)
                 .set("fill", COLORS[seq_idx]);
             group.append(square);
-            let label = element::Text::new()
+            let label = element::Text::new("")
                 .set("x", x + 14)
                 .set("y", y + 7)
                 .set("fill", LEGEND_COLOR)
@@ -83,7 +83,7 @@
                 .set("stroke-dasharray", "10 7")
                 .set("d", data);
             group.append(path);
-            let tick_label = element::Text::new()
+            let tick_label = element::Text::new("")
                 .set("x", self.gr.left + self.gr.width + 2)
                 .set("y", self.projector.project_y(*tick) + 2)
                 .set("fill", TICK_LABEL_COLOR)
@@ -165,7 +165,7 @@
                 .set("opacity", 0.5)
                 .set("d", data);
             tick_opt_group.append(path);
-            let tick_label = element::Text::new()
+            let tick_label = element::Text::new("")
                 .set("x", tick.tx + 1)
                 .set("y", y + 9)
                 .set("fill", TICK_LABEL_COLOR)
@@ -215,7 +215,7 @@
                             .set("r", 8);
                         point_group.append(circle);
                         let mut point_opt_group = node::element::Group::new().set("class", "opt");
-                        let point_label_shadow = element::Text::new()
+                        let point_label_shadow = element::Text::new("")
                             .set("x", x - 5)
                             .set("y", y - 10)
                             .set("stroke", "#222")
@@ -230,7 +230,7 @@
                             .set("cy", y)
                             .set("r", 4);
                         point_opt_group.append(circle);
-                        let point_label = element::Text::new()
+                        let point_label = element::Text::new("")
                             .set("x", x - 5)
                             .set("y", y - 10)
                             .set("fill", TICK_LABEL_COLOR)
