--- a/src/auto/functions.rs
+++ b/src/auto/functions.rs
@@ -146,7 +146,7 @@
 pub fn shape(text: &str, analysis: &Analysis) -> GlyphString {
     let length = text.len() as _;
     unsafe {
-        let mut glyphs = GlyphString::uninitialized();
+        let mut glyphs = GlyphString::new();
         ffi::pango_shape(text.to_glib_none().0, length, analysis.to_glib_none().0, glyphs.to_glib_none_mut().0);
         glyphs
     }
