--- rust-gvdb-0.9.0.orig/Cargo.toml
+++ rust-gvdb-0.9.0/Cargo.toml
@@ -111,7 +111,7 @@ version = "0.1"
 version = "1.2"
 
 [dev-dependencies.rand]
-version = "0.9"
+version = ">= 0.8, < 0.10"
 
 [dev-dependencies.serde_json]
 version = "1.0"
--- rust-gvdb-0.9.0.orig/src/write/file.rs
+++ rust-gvdb-0.9.0/src/write/file.rs
@@ -740,7 +740,7 @@ mod test {
             let file_builder = FileWriter::new();
             let mut table_builder = HashTableBuilder::new();
             let mut numbers: Vec<usize> = (0..200).collect();
-            numbers.shuffle(&mut rand::rng());
+            numbers.shuffle(&mut rand::thread_rng());
 
             for num in numbers {
                 let str = format!("{num}");
--- rust-gvdb-0.9.0.orig/tests/gresource_bundle.rs
+++ rust-gvdb-0.9.0/tests/gresource_bundle.rs
@@ -66,7 +66,7 @@ fn test_from_dir_reproducible_build() {
 
     use rand::prelude::*;
     fn copy_random_order(from: &Path, to: &Path) {
-        let mut rng = rand::rng();
+        let mut rng = rand::thread_rng();
         let mut files: Vec<std::fs::DirEntry> = std::fs::read_dir(from)
             .unwrap()
             .map(|d| d.unwrap())
