Index: gvdb/Cargo.toml
===================================================================
--- gvdb.orig/Cargo.toml
+++ gvdb/Cargo.toml
@@ -50,7 +50,7 @@ version = "0.9"
 optional = true
 
 [dependencies.quick-xml]
-version = "0.31"
+version = "0.27"
 features = ["serialize"]
 optional = true
 
@@ -70,7 +70,7 @@ version = "2.3"
 optional = true
 
 [dependencies.zvariant]
-version = "3.2"
+version = "3.15"
 features = ["gvariant"]
 default-features = false
 
