From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Thu, 22 Aug 2024 09:59:51 -0400
Subject: relax deps

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 82962be..c667871 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,7 @@ version = "0.9"
 optional = true
 
 [dependencies.quick-xml]
-version = "0.36"
+version = "0.31"
 features = ["serialize"]
 optional = true
 
