From 92580b9aba4e4c5a5992989d6cae27e3e1e15142 Mon Sep 17 00:00:00 2001
From: Wolfgang Silbermayr <wolfgang@silbermayr.at>
Date: Tue, 22 Jan 2019 09:32:19 +0100
Subject: [PATCH] Remove futures preview feature

---
 Cargo.toml | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 5277a14..4aa263f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,10 +40,6 @@ version = "0.7.0"
 version = "0.3"
 optional = true
 
-[dependencies.futures-core-preview]
-version = "0.2"
-optional = true
-
 [dependencies.gdk]
 version = "0.9.0"
 
@@ -91,7 +87,6 @@ optional = true
 [features]
 dox = ["gdk/dox", "gtk-sys/dox"]
 embed-lgpl-docs = ["gtk-rs-lgpl-docs", "gdk/embed-lgpl-docs"]
-futures = ["futures-core-preview", "fragile", "gio/futures"]
 purge-lgpl-docs = ["gtk-rs-lgpl-docs", "gdk/purge-lgpl-docs"]
 v3_10 = ["v3_8", "gtk-sys/v3_10", "gdk/v3_10"]
 v3_12 = ["v3_10", "gtk-sys/v3_12", "gdk/v3_12"]
-- 
2.20.1

