Index: sloppy-rfc4880/Cargo.toml
===================================================================
--- sloppy-rfc4880.orig/Cargo.toml
+++ sloppy-rfc4880/Cargo.toml
@@ -22,7 +22,7 @@ categories = ["parser-implementations"]
 license = "Apache-2.0"
 repository = "https://github.com/kpcyrd/sloppy-rfc4880"
 [dependencies.base64]
-version = "0.10"
+version = ">= 0.10, < 0.12"
 
 [dependencies.byteorder]
 version = "1.2"
@@ -31,7 +31,7 @@ version = "1.2"
 version = "0.1"
 
 [dependencies.hex]
-version = "0.3.2"
+version = ">= 0.3.2, < 0.5"
 
 [dependencies.log]
 version = "0.4"
