--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,7 +48,7 @@
 version = "1.6"
 
 [dependencies.ttf-parser]
-version = "0.24.1"
+version = "0.25"
 features = [
     "opentype-layout",
     "apple-layout",
@@ -59,10 +59,10 @@
 default-features = false
 
 [dependencies.unicode-bidi-mirroring]
-version = "0.3.0"
+version = "0.4"
 
 [dependencies.unicode-ccc]
-version = "0.3.0"
+version = "0.4"
 
 [dependencies.unicode-properties]
 version = "0.1.0"
