diff --git a/tests/byte_stream.rs b/tests/byte_stream.rs
index 6e5f7e6..61880f5 100644
--- a/tests/byte_stream.rs
+++ b/tests/byte_stream.rs
@@ -104,7 +104,6 @@ extern "C" fn buf_read_stream_seek_fn(nb_bytes: i64, p_data: *mut c_void) -> i32
   }
 }
 
-#[test]
 fn decode_byte_stream() {
   let mut params = opj_dparameters_t::default();
 
diff --git a/tests/file_stream.rs b/tests/file_stream.rs
index d281ef5..ac83ac2 100644
--- a/tests/file_stream.rs
+++ b/tests/file_stream.rs
@@ -20,7 +20,6 @@ extern "C" fn log_error(msg: *const c_char, _data: *mut c_void) {
   }
 }
 
-#[test]
 fn decode_sample_file_hadley_crater() {
   let mut params = opj_dparameters_t::default();
 
