--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,2 +39,3 @@
 repository = "https://github.com/winnow-rs/winnow"
+autobenches = false
 
@@ -153,6 +154,2 @@
 [[bench]]
-name = "contains_token"
-harness = false
-
-[[bench]]
 name = "find_slice"
@@ -161,6 +158,2 @@
 [[bench]]
-name = "iter"
-harness = false
-
-[[bench]]
 name = "next_slice"
@@ -184,7 +177,2 @@
 
-[[bench]]
-name = "json"
-path = "examples/json/bench.rs"
-harness = false
-required-features = ["std"]
 
