Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -178,25 +178,25 @@
 path = "examples/string/main.rs"
 required-features = ["alloc"]
 
-[[bench]]
+[[disabled.bench]]
 name = "arithmetic"
 path = "examples/arithmetic/bench.rs"
 harness = false
 required-features = ["alloc"]
 
-[[bench]]
+[[disabled.bench]]
 name = "http"
 path = "examples/http/bench.rs"
 harness = false
 required-features = ["alloc"]
 
-[[bench]]
+[[disabled.bench]]
 name = "ini"
 path = "examples/ini/bench.rs"
 harness = false
 required-features = ["std"]
 
-[[bench]]
+[[disabled.bench]]
 name = "json"
 path = "examples/json/bench.rs"
 harness = false
