Index: winnow/Cargo.toml
===================================================================
--- winnow.orig/Cargo.toml
+++ winnow/Cargo.toml
@@ -168,27 +168,11 @@ path = "examples/json/bench.rs"
 harness = false
 required-features = ["std"]
 
-[dependencies.anstyle]
-version = "0.3.5"
-optional = true
-
-[dependencies.anstyle-stream]
-version = "0.2.2"
-optional = true
-
-[dependencies.is-terminal]
-version = "0.4.7"
-optional = true
-
 [dependencies.memchr]
 version = "2.5"
 optional = true
 default-features = false
 
-[dependencies.terminal_size]
-version = "0.2.6"
-optional = true
-
 [dev-dependencies.circular]
 version = "0.3.0"
 
@@ -216,12 +200,6 @@ version = "0.2.0"
 
 [features]
 alloc = []
-debug = [
-    "dep:anstyle",
-    "dep:is-terminal",
-    "dep:terminal_size",
-    "dep:anstyle-stream",
-]
 default = ["std"]
 simd = ["dep:memchr"]
 std = [
