Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -96,9 +96,9 @@
 alloc = []
-debug = [
-    "std",
-    "dep:anstream",
-    "dep:anstyle",
-    "dep:is-terminal",
-    "dep:terminal_size",
-]
+#debug = [
+#    "std",
+#    "dep:anstream",
+#    "dep:anstyle",
+#    "dep:is-terminal",
+#    "dep:terminal_size",
+#]
 default = ["std"]
@@ -230,3 +230,3 @@
 
-[dependencies.anstream]
+[disabled.dependencies.anstream]
 version = "0.3.2"
@@ -234,3 +234,3 @@
 
-[dependencies.anstyle]
+[disabled.dependencies.anstyle]
 version = "1.0.1"
@@ -238,3 +238,3 @@
 
-[dependencies.is-terminal]
+[disabled.dependencies.is-terminal]
 version = "0.4.9"
@@ -247,3 +247,3 @@
 
-[dependencies.terminal_size]
+[disabled.dependencies.terminal_size]
 version = "0.4.0"
@@ -282,3 +282,3 @@
 
-[dev-dependencies.term-transcript]
+[disabled.dependencies.term-transcript]
 version = "0.2.0"
