Forwarded: not-needed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -93,13 +93,13 @@
 
 [features]
 alloc = []
-debug = [
-    "std",
-    "dep:anstream",
-    "dep:anstyle",
-    "dep:is_terminal_polyfill",
-    "dep:terminal_size",
-]
+#debug = [
+#    "std",
+#    "dep:anstream",
+#    "dep:anstyle",
+#    "dep:is_terminal_polyfill",
+#    "dep:terminal_size",
+#]
 default = ["std"]
 simd = ["dep:memchr"]
 std = [
@@ -202,15 +202,15 @@
 harness = false
 required-features = ["std"]
 
-[dependencies.anstream]
+[disabled.dependencies.anstream]
 version = "0.3.2"
 optional = true
 
-[dependencies.anstyle]
+[disabled.dependencies.anstyle]
 version = "1.0.1"
 optional = true
 
-[dependencies.is_terminal_polyfill]
+[disabled.dependencies.is_terminal_polyfill]
 version = "1.48.0"
 optional = true
 
@@ -219,7 +219,7 @@
 optional = true
 default-features = false
 
-[dependencies.terminal_size]
+[disabled.dependencies.terminal_size]
 version = "0.4.0"
 optional = true
 
@@ -251,7 +251,7 @@
 version = "0.6.21"
 features = ["examples"]
 
-[dev-dependencies.term-transcript]
+[disabled.dependencies.term-transcript]
 version = "0.2.0"
 
 [lints.clippy]
