--- a/speakersafetyd.sysusers.conf
+++ b/speakersafetyd.sysusers.conf
@@ -1,2 +1,2 @@
-u    speakersafetyd    -        "Speakersafetyd daemon"    /var/lib/speakersafetyd    -
-m    speakersafetyd    audio
+u!   _speakersafetyd    -        "Speakersafetyd daemon"    /var/lib/speakersafetyd    -
+m    _speakersafetyd    audio
--- a/Makefile
+++ b/Makefile
@@ -8,8 +8,8 @@
 TMPFILESDIR ?= /usr/lib/tmpfiles.d
 SHAREDIR ?= /usr/share/
 VARDIR ?= /var/
-SPEAKERSAFETYD_GROUP ?= speakersafetyd
-SPEAKERSAFETYD_USER ?= speakersafetyd
+SPEAKERSAFETYD_GROUP ?= _speakersafetyd
+SPEAKERSAFETYD_USER ?= _speakersafetyd
 
 all:
 	cargo build --release
@@ -27,10 +27,8 @@
 	install -pm0644 95-speakersafetyd.rules $(DESTDIR)/$(UDEVDIR)/95-speakersafetyd.rules
 	install -dDm0755 $(DESTDIR)/$(SHAREDIR)/speakersafetyd/apple
 	install -pm0644 -t $(DESTDIR)/$(SHAREDIR)/speakersafetyd/apple $(wildcard conf/apple/*)
-	install -dDm0755 -o $(SPEAKERSAFETYD_USER) -g $(SPEAKERSAFETYD_GROUP) $(DESTDIR)/$(VARDIR)/lib/speakersafetyd/blackbox
 	install -dDm0755 $(DESTDIR)/$(TMPFILESDIR)
 	install -pm0644 speakersafetyd.tmpfiles $(DESTDIR)/$(TMPFILESDIR)/speakersafetyd.conf
-	install -dDm0755 -o $(SPEAKERSAFETYD_USER) -g $(SPEAKERSAFETYD_GROUP) /run/speakersafetyd
 
 uninstall:
 	rm -f $(DESTDIR)/$(BINDIR)/speakersafetyd $(DESTDIR)/$(UNITDIR)/speakersafetyd.service $(DESTDIR)/$(UDEVDIR)/95-speakersafetyd.rules $(DESTDIR)/$(TMPFILESDIR)/speakersafetyd.conf
--- a/speakersafetyd.service
+++ b/speakersafetyd.service
@@ -4,7 +4,7 @@
 [Service]
 Type=simple
 ExecStart=/usr/bin/speakersafetyd -c /usr/share/speakersafetyd/ -b /var/lib/speakersafetyd/blackbox -m 7
-User=speakersafetyd
+User=_speakersafetyd
 AmbientCapabilities=CAP_SYS_NICE
 CapabilityBoundingSet=CAP_SYS_NICE
 UMask=0066
--- a/speakersafetyd.tmpfiles
+++ b/speakersafetyd.tmpfiles
@@ -1,2 +1,2 @@
-d /var/lib/speakersafetyd/blackbox 0755 speakersafetyd speakersafetyd -
-d /run/speakersafetyd 0755 speakersafetyd speakersafetyd -
+d /var/lib/speakersafetyd/blackbox 0755 _speakersafetyd _speakersafetyd -
+d /run/speakersafetyd 0755 _speakersafetyd _speakersafetyd -
