Description: downgrading darling dependency version
Author: Eddie Billoir <lechatp@outlook.fr>
Index: bon-macros/Cargo.toml
===================================================================
--- bon-macros.orig/Cargo.toml
+++ bon-macros/Cargo.toml
@@ -47,7 +47,7 @@ path = "src/lib.rs"
 proc-macro = true
 
 [dependencies.darling]
-version = "0.21.0"
+version = "0.20"
 
 [dependencies.ident_case]
 version = "1"
