Index: netlink-sys/Cargo.toml
===================================================================
--- netlink-sys.orig/Cargo.toml
+++ netlink-sys/Cargo.toml
@@ -35,25 +35,6 @@ repository = "https://github.com/rust-ne
 name = "netlink_sys"
 path = "src/lib.rs"
 
-[[example]]
-name = "audit_events"
-path = "examples/audit_events.rs"
-
-[[example]]
-name = "audit_events_async_std"
-path = "examples/audit_events_async_std.rs"
-required-features = ["smol_socket"]
-
-[[example]]
-name = "audit_events_tokio"
-path = "examples/audit_events_tokio.rs"
-required-features = ["tokio_socket"]
-
-[[example]]
-name = "audit_events_tokio_manual_thread_builder"
-path = "examples/audit_events_tokio_manual_thread_builder.rs"
-required-features = ["tokio_socket"]
-
 [dependencies.async-io]
 version = "2"
 optional = true
@@ -89,9 +70,6 @@ default-features = false
 version = "1.13"
 features = ["attributes"]
 
-[dev-dependencies.netlink-packet-audit]
-version = "0.4.1"
-
 [dev-dependencies.tokio]
 version = "1.0.1"
 features = [
