Index: rustc-demangle/Cargo.toml
===================================================================
--- rustc-demangle.orig/Cargo.toml
+++ rustc-demangle/Cargo.toml
@@ -41,18 +41,5 @@ lto = true
 name = "rustc_demangle"
 path = "src/lib.rs"
 
-[dependencies.compiler_builtins]
-version = "0.1.2"
-optional = true
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [features]
-rustc-dep-of-std = [
-    "core",
-    "compiler_builtins",
-]
 std = []
