Index: fastrand/Cargo.toml
===================================================================
--- fastrand.orig/Cargo.toml
+++ fastrand/Cargo.toml
@@ -37,17 +37,3 @@ version = "0.8"
 
 [dev-dependencies.wyhash]
 version = "0.5"
-
-[target."cfg(target_arch = \"wasm32\")".dependencies.instant]
-version = "0.1"
-
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.instant]
-version = "0.1"
-features = ["wasm-bindgen"]
-
-[target."cfg(target_arch = \"wasm32\")".dev-dependencies.wasm-bindgen-test]
-version = "0.3"
