Index: fd-find/Cargo.toml
===================================================================
--- fd-find.orig/Cargo.toml
+++ fd-find/Cargo.toml
@@ -57,7 +57,7 @@ features = [
 default-features = false
 
 [dependencies.clap]
-version = "4.1.1"
+version = "4"
 features = [
     "suggestions",
     "color",
@@ -68,7 +68,7 @@ features = [
 ]
 
 [dependencies.clap_complete]
-version = "4.0.6"
+version = "4"
 optional = true
 
 [dependencies.crossbeam-channel]
