Index: fd-find/Cargo.toml
===================================================================
--- fd-find.orig/Cargo.toml
+++ fd-find/Cargo.toml
@@ -70,7 +70,7 @@ version = "1.0"
 version = "0.4.0"
 
 [dependencies.clap]
-version = "4.5.42"
+version = "4"
 features = [
     "suggestions",
     "color",
@@ -80,7 +80,7 @@ features = [
 ]
 
 [dependencies.clap_complete]
-version = "4.5.55"
+version = "4"
 optional = true
 
 [dependencies.crossbeam-channel]
