Index: fd-find/Cargo.toml
===================================================================
--- fd-find.orig/Cargo.toml
+++ fd-find/Cargo.toml
@@ -144,10 +144,10 @@ version = "0.9"
 #optional = true
 
 [target.'cfg(all(unix, not(target_os = "redox")))'.dependencies.libc]
-version = "0.2"
+version = "0"
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.30.1"
+version = "0"
 features = [
     "signal",
     "user",
