Index: fd-find/Cargo.toml
===================================================================
--- fd-find.orig/Cargo.toml
+++ fd-find/Cargo.toml
@@ -90,7 +90,7 @@ version = "0"
 version = "3.4"
 
 [dependencies.etcetera]
-version = "0.10"
+version = "0"
 
 [dependencies.faccess]
 version = "0.2.4"
@@ -105,7 +105,7 @@ version = "=0.5.9"
 version = "0.4.23"
 
 [dependencies.jiff]
-version = "0.2.14"
+version = "0"
 
 [dependencies.lscolors]
 version = "0"
