--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,11 +25,7 @@
 repository = "https://github.com/Farmadupe/vid_dup_finder_lib"
 
 [features]
-default = ["gstreamer_backend"]
-gstreamer_backend = [
-    "vid_frame_iter",
-    "glib",
-]
+default = []
 
 [lib]
 name = "ffmpeg_gst_wrapper"
@@ -59,7 +55,3 @@
 
 [dependencies.url]
 version = "2.2"
-
-[dependencies.vid_frame_iter]
-version = "0.2.1"
-optional = true
