Index: alacritty/Cargo.toml
===================================================================
--- alacritty.orig/Cargo.toml
+++ alacritty/Cargo.toml
@@ -39,3 +40,3 @@
 [dependencies.dirs]
-version = "4.0.0"
+version = ">= 4.0, < 6.0"
 
@@ -90,7 +90,7 @@ features = ["derive"]
 version = "1"
 
 [dependencies.serde_yaml]
-version = "0.8"
+version = ">= 0.8, < 1.0"
 
 [dependencies.unicode-width]
 version = "0.1"
