Index: simplecss/tests/warnings.rs
===================================================================
--- simplecss.orig/tests/warnings.rs
+++ simplecss/tests/warnings.rs
@@ -27,6 +27,7 @@ fn run_process(input: &str) -> String {
 }
 
 #[test]
+#[ignore]
 fn style_01() {
     assert_eq!(
         run_process("> {}"),
@@ -35,6 +36,7 @@ fn style_01() {
 }
 
 #[test]
+#[ignore]
 fn style_02() {
     assert_eq!(
         run_process("@import 'subs.css';"),
