Index: vt100/Cargo.toml
===================================================================
--- vt100.orig/Cargo.toml
+++ vt100/Cargo.toml
@@ -44,7 +44,7 @@ version = "0.4.17"
 version = "0.1.10"
 
 [dependencies.vte]
-version = "0.11.0"
+version = "0.13.0"
 
 [dev-dependencies.nix]
 version = "0.26.2"
@@ -66,4 +66,4 @@ version = "1.0.92"
 version = "< 0.3"
 
 [dev-dependencies.vte]
-version = "0.11.0"
+version = "0.13.0"
