--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,3 +46,3 @@
 [dependencies.vte]
-version = "0.11.0"
+version = "0.15"
 
@@ -68,2 +68,2 @@
 [dev-dependencies.vte]
-version = "0.11.0"
+version = "0.15"
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -24,5 +24,5 @@
     pub fn process(&mut self, bytes: &[u8]) {
-        for byte in bytes {
-            self.parser.advance(&mut self.screen, *byte);
-        }
+        //for byte in bytes {
+            self.parser.advance(&mut self.screen, bytes);
+        //}
     }
