Index: chrono/src/offset/local/tz_data.rs
===================================================================
--- chrono.orig/src/offset/local/tz_data.rs
+++ chrono/src/offset/local/tz_data.rs
@@ -170,7 +170,7 @@ const TZDATA_VERSION_LEN: usize = 12;
 mod tests {
     use super::*;
 
-    #[test]
+    /*#[test]
     fn test_ohos_tzdata_header_and_index() {
         let file = File::open("./tests/ohos/tzdata").unwrap();
         let header = TzDataHeader::new(&file).unwrap();
@@ -193,14 +193,14 @@ mod tests {
         let timezone = iter.find_timezone(b"Asia/Shanghai").unwrap();
         let tzdata = iter.find_tzdata(&file, &header, timezone).unwrap();
         assert_eq!(tzdata.len(), 393);
-    }
+    }*/
 
     #[test]
     fn test_invalid_tzdata_header() {
         TzDataHeader::new(&b"tzdaaa2024aaaaaaaaaaaaaaa\0"[..]).unwrap_err();
     }
 
-    #[test]
+    /*#[test]
     fn test_android_tzdata_header_and_index() {
         let file = File::open("./tests/android/tzdata").unwrap();
         let header = TzDataHeader::new(&file).unwrap();
@@ -249,7 +249,7 @@ mod tests {
     fn test_android_tzdata_find_missing() {
         let file = File::open("./tests/android/tzdata").unwrap();
         assert!(find_tz_data::<ANDROID_ENTRY_LEN>(file, b"Asia/S000000i").unwrap().is_none());
-    }
+    }*/
 
     #[cfg(target_env = "ohos")]
     #[test]
