Subject: Drop Windows-specific features

These are not going to be built on Debian

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -107,7 +107,6 @@
 __internal_bench = []
 alloc = []
 clock = [
-    "winapi",
     "iana-time-zone",
     "android-tzdata",
     "now",
@@ -144,7 +143,6 @@
     "wasm-bindgen",
     "js-sys",
 ]
-winapi = ["windows-targets"]
 
 [target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dependencies.js-sys]
 version = "0.3"
@@ -162,10 +160,3 @@
 version = "0.1.45"
 features = ["fallback"]
 optional = true
-
-[target."cfg(windows)".dependencies.windows-targets]
-version = "0.52"
-optional = true
-
-[target."cfg(windows)".dev-dependencies.windows-bindgen]
-version = "0.58"
