Subject: Drop Android-related Features

These are not going to be built on Debian.

Index: chrono/Cargo.toml
===================================================================
--- chrono.orig/Cargo.toml
+++ chrono/Cargo.toml
@@ -55,11 +55,9 @@ rustdoc-args = [
 features = ["serde"]
 
 [features]
-__internal_bench = []
 alloc = []
 clock = [
     "iana-time-zone",
-    "android-tzdata",
     "now",
 ]
 default = [
@@ -153,10 +151,6 @@ optional = true
 version = "0.2"
 optional = true
 
-[target.'cfg(target_os = "android")'.dependencies.android-tzdata]
-version = "0.1.1"
-optional = true
-
 [target."cfg(unix)".dependencies.iana-time-zone]
 version = "0.1.45"
 features = ["fallback"]
