Subject: Drop Windows-specific features

These are not going to be built on Debian

Index: chrono/Cargo.toml
===================================================================
--- chrono.orig/Cargo.toml
+++ chrono/Cargo.toml
@@ -60,3 +60,2 @@
 clock = [
-    "winapi",
     "iana-time-zone",
@@ -93,7 +92,6 @@ wasmbind = [
     "wasm-bindgen",
     "js-sys",
 ]
-winapi = ["windows-link"]
 
 [lib]
 name = "chrono"
@@ -164,9 +162,3 @@ version = "0.1.45"
 features = ["fallback"]
 optional = true
 
-[target."cfg(windows)".dependencies.windows-link]
-version = "0.2"
-optional = true
-
-[target."cfg(windows)".dev-dependencies.windows-bindgen]
-version = "0.63"
