--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,10 +53,6 @@
 version = "0.2"
 default-features = false
 
-[dependencies.pure-rust-locales]
-version = "0.5.2"
-optional = true
-
 [dependencies.rustc-serialize]
 version = "0.3.20"
 optional = true
@@ -93,16 +89,15 @@
 __doctest = []
 __internal_bench = []
 alloc = []
-clock = ["libc", "std", "winapi"]
+clock = ["libc", "std"]
 default = ["clock", "std", "oldtime"]
 oldtime = ["time"]
 std = []
-unstable-locales = ["pure-rust-locales", "alloc"]
+unstable-locales = ["alloc"]
 
 [target."cfg(windows)".dependencies.winapi]
 version = "0.3.0"
 features = ["std", "minwinbase", "minwindef", "timezoneapi"]
-optional = true
 [badges.appveyor]
 repository = "chronotope/chrono"
 
