--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,9 +21,6 @@
 categories = ["network-programming", "os"]
 license = "Apache-2.0"
 repository = "https://github.com/achanda/ipnetwork"
-[dependencies.clippy]
-version = "0.0.302"
-optional = true
 
 [dependencies.serde]
 version = ">=0.8.0, <2.0"
@@ -35,6 +32,5 @@
 
 [features]
 default = []
-dev = ["clippy"]
 [badges.travis-ci]
 repository = "achanda/ipnetwork"
