Index: libseccomp/Cargo.toml
===================================================================
--- libseccomp.orig/Cargo.toml
+++ libseccomp/Cargo.toml
@@ -30,6 +30,9 @@ categories = [
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/libseccomp-rs/libseccomp-rs"
 
+[lib]
+doctest = false
+
 [package.metadata.docs.rs]
 rustc-args = [
     "--cfg",
Index: libseccomp/tests/notify.rs
===================================================================
--- libseccomp.orig/tests/notify.rs
+++ libseccomp/tests/notify.rs
@@ -126,6 +126,7 @@ fn test_resp_new() {
 }
 
 #[test]
+#[ignore = "known to be flaky"]
 fn test_error() {
     skip_if_not_supported!();
 
Index: libseccomp/tests/tests.rs
===================================================================
--- libseccomp.orig/tests/tests.rs
+++ libseccomp/tests/tests.rs
@@ -21,6 +21,8 @@ fn test_check_version() {
 }
 
 #[test]
+#[ignore = "known to be flaky"
+]
 fn test_check_api() {
     assert!(check_api(3, ScmpVersion::from((2, 4, 0))).unwrap());
     assert!(!check_api(100, ScmpVersion::from((2, 4, 0))).unwrap());
@@ -229,6 +231,7 @@ fn test_get_syscall_from_name() {
 
 #[test]
 #[cfg(feature = "const-syscall")]
+#[cfg(not(target_arch = "s390x"))]
 fn test_syscall_new() {
     for name in known_syscall_names::KNOWN_SYSCALL_NAMES {
         if let Ok(nr) = ScmpSyscall::from_name(name) {
@@ -248,6 +251,7 @@ fn test_display_syscall() {
 }
 
 #[test]
+#[cfg(not(target_endian = "big"))]
 fn test_arch_functions() {
     let mut ctx = ScmpFilterContext::new_filter(ScmpAction::Allow).unwrap();
     ctx.add_arch(ScmpArch::X86).unwrap();
@@ -260,6 +264,7 @@ fn test_arch_functions() {
 }
 
 #[test]
+#[cfg(not(target_endian = "big"))]
 fn test_merge_filters() {
     let mut ctx1 = ScmpFilterContext::new_filter(ScmpAction::Allow).unwrap();
     let mut ctx2 = ScmpFilterContext::new_filter(ScmpAction::Allow).unwrap();
Index: libseccomp/src/api.rs
===================================================================
--- libseccomp.orig/src/api.rs
+++ libseccomp/src/api.rs
@@ -129,6 +129,7 @@ mod tests {
     use super::*;
 
     #[test]
+    #[cfg(not(target_arch = "loong64"))]
     fn test_ensure_supported_api() {
         assert!(ensure_supported_api("test", 3, ScmpVersion::from((2, 4, 0))).is_ok());
         assert!(ensure_supported_api("test", 100, ScmpVersion::from((2, 4, 0))).is_err());
