--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,20 +43,10 @@
 all-features = true
 
 [[example]]
-name = "hyphenation"
-path = "examples/hyphenation.rs"
-required-features = ["hyphenation"]
-
-[[example]]
 name = "termwidth"
 path = "examples/termwidth.rs"
 required-features = ["terminal_size"]
 
-[dependencies.hyphenation]
-version = "0.8.4"
-features = ["embed_en-us"]
-optional = true
-
 [dependencies.smawk]
 version = "0.3.1"
 optional = true
