--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,9 +58,6 @@
 version = "0.1.10"
 optional = true
 
-[dev-dependencies.version-sync]
-version = "0.9.4"
-
 [features]
 default = [
     "unicode-linebreak",
--- a/tests/version-numbers.rs
+++ b/tests/version-numbers.rs
@@ -1,4 +1,4 @@
-#[test]
+/*#[test]
 fn test_readme_deps() {
     version_sync::assert_markdown_deps_updated!("README.md");
 }
@@ -19,4 +19,4 @@
 #[test]
 fn test_dependency_graph() {
     version_sync::assert_contains_regex!("src/lib.rs", "master/images/textwrap-{version}.svg");
-}
+}*/
