--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,9 +84,6 @@
 version = "0.2.0"
 optional = true
 
-[dev-dependencies.unic-emoji-char]
-version = "0.9.0"
-
 [dev-dependencies.version-sync]
 version = "0.9.5"
 
--- a/src/core.rs
+++ b/src/core.rs
@@ -382,7 +382,7 @@
         assert_eq!(chars.next(), Some('H'));
     }
 
-    #[test]
+    /*#[test]
     fn emojis_have_correct_width() {
         use unic_emoji_char::is_emoji;
 
@@ -421,7 +421,7 @@
 
         // The remaining planes contain almost no assigned code points
         // and thus also no emojis.
-    }
+    }*/
 
     #[test]
     fn display_width_works() {
