--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,9 +84,6 @@
 version = "0.2.0"
 optional = true
 
-[dev-dependencies.version-sync]
-version = "0.9.5"
-
 [target."cfg(unix)".dev-dependencies.termion]
 version = "4.0.2"
 
--- a/tests/version-numbers.rs
+++ b/tests/version-numbers.rs
@@ -1,4 +1,4 @@
-#[test]
+/*#[test]
 fn test_readme_deps() {
     version_sync::assert_markdown_deps_updated!("README.md");
 }
@@ -19,4 +19,4 @@
 #[test]
 fn test_dependency_graph() {
     version_sync::assert_contains_regex!("src/lib.rs", "master/images/textwrap-{version}.svg");
-}
+}*/
