--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,19 +55,19 @@
 default-features = false
 
 [dependencies.itertools]
-version = "0.12"
+version = "0.13"
 
 [dependencies.katexit]
 version = "0.1.4"
 optional = true
 
 [dependencies.nalgebra]
-version = "0.32"
+version = "0.33"
 features = ["std"]
 default-features = false
 
 [dependencies.num]
-version = "0.4.1"
+version = "0.4"
 
 [dependencies.quickcheck]
 version = "1.0.3"
@@ -99,7 +99,7 @@
 version = "1.0.3"
 
 [dev-dependencies.wasm-bindgen-test]
-version = "0.3.38"
+version = "0.3"
 
 [features]
 default = [
