--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,12 +26,6 @@
 license = "MIT"
 repository = "https://github.com/image-rs/imageproc.git"
 
-[package.metadata.docs.rs]
-features = [
-    "property-testing",
-    "katexit",
-]
-
 [profile.bench]
 opt-level = 3
 lto = false
@@ -57,10 +51,6 @@
 [dependencies.itertools]
 version = "0.13"
 
-[dependencies.katexit]
-version = "0.1.4"
-optional = true
-
 [dependencies.nalgebra]
 version = "0.33"
 features = ["std"]
