Index: fake-instant/src/lib.rs
===================================================================
--- fake-instant.orig/src/lib.rs
+++ fake-instant/src/lib.rs
@@ -25,7 +25,6 @@
     mutable_transmutes,
     no_mangle_const_items,
     unknown_crate_types,
-    warnings
 )]
 #![deny(
     deprecated,
@@ -54,7 +53,6 @@
     unused_results
 )]
 #![allow(
-    box_pointers,
     missing_copy_implementations,
     missing_debug_implementations,
     variant_size_differences
