--- a/Cargo.toml
+++ b/Cargo.toml
@@ -76,6 +76,6 @@
 default = []
-ffi = [
-    "dep:http-body-util",
-    "dep:futures-util",
-]
+#ffi = [
+#    "dep:http-body-util",
+#    "dep:futures-util",
+#]
 full = [
@@ -97,3 +97,3 @@
 ]
-nightly = []
+#nightly = []
 server = [
@@ -103,3 +103,3 @@
 ]
-tracing = ["dep:tracing"]
+#tracing = ["dep:tracing"]
 
@@ -163,3 +163,3 @@
 
-[dependencies.tracing]
+[disabled.dependencies.tracing]
 version = "0.1"
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,3 +5,3 @@
 #![cfg_attr(all(test, feature = "full"), deny(warnings))]
-#![cfg_attr(all(test, feature = "nightly"), feature(test))]
+//#![cfg_attr(all(test, feature = "nightly"), feature(test))]
 #![cfg_attr(docsrs, feature(doc_cfg))]
