diff --git a/Cargo.toml b/Cargo.toml
index 517a1dd..51cffb5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -93,9 +93,9 @@ features = [
 ]
 default-features = false
 
-[dev-dependencies.futures01_crate]
-version = "0.1"
-package = "futures"
+#[dev-dependencies.futures01_crate]
+#version = "0.1"
+#package = "futures"
 
 [dev-dependencies.futures03_crate]
 version = "0.3"
@@ -123,27 +123,30 @@ version = "1"
 version = "1"
 package = "serde"
 
-[dev-dependencies.tokio01_crate]
-version = "0.1"
-features = ["io"]
-default-features = false
-package = "tokio"
+#[dev-dependencies.tokio01_crate]
+#version = "0.1"
+#features = ["io"]
+#default-features = false
+#package = "tokio"
 
-[dev-dependencies.tokio02_crate]
-version = "0.2"
-default-features = false
-package = "tokio"
+#[dev-dependencies.tokio02_crate]
+#version = "0.2"
+#default-features = false
+#package = "tokio"
 
-[dev-dependencies.tokio03_crate]
-version = "0.3"
-default-features = false
-package = "tokio"
+#[dev-dependencies.tokio03_crate]
+#version = "0.3"
+#default-features = false
+#package = "tokio"
 
 [dev-dependencies.tokio1_crate]
 version = "1"
 default-features = false
 package = "tokio"
 
+[dev-dependencies.macrotest]
+version = "1"
+
 [features]
 convert = []
 coroutine_trait = []
