--- rust-systemd-journal-logger-2.2.2.orig/Cargo.toml
+++ rust-systemd-journal-logger-2.2.2/Cargo.toml
@@ -77,30 +77,30 @@ path = "examples/custom_fields.rs"
 name = "systemd_service"
 path = "examples/systemd_service.rs"
 
-[[test]]
-name = "init"
-path = "tests/init.rs"
-
-[[test]]
-name = "init_with_extra_fields"
-path = "tests/init_with_extra_fields.rs"
+#[[test]]
+#name = "init"
+#path = "tests/init.rs"
+
+#[[test]]
+#name = "init_with_extra_fields"
+#path = "tests/init_with_extra_fields.rs"
 
 [[test]]
 name = "journal"
 path = "tests/journal.rs"
 
-[[test]]
-name = "journal_stream"
-path = "tests/journal_stream.rs"
-harness = false
-
-[[test]]
-name = "log_to_journal"
-path = "tests/log_to_journal.rs"
-
-[[test]]
-name = "log_with_extra_fields"
-path = "tests/log_with_extra_fields.rs"
+#[[test]]
+#name = "journal_stream"
+#path = "tests/journal_stream.rs"
+#harness = false
+
+#[[test]]
+#name = "log_to_journal"
+#path = "tests/log_to_journal.rs"
+
+#[[test]]
+#name = "log_with_extra_fields"
+#path = "tests/log_with_extra_fields.rs"
 
 [dependencies.log]
 version = "^0.4"
