--- a/Cargo.toml
+++ b/Cargo.toml
@@ -243,7 +243,6 @@
 console_writer = [
     "ansi_writer",
     "libc",
-    "winapi",
 ]
 default = [
     "all_components",
@@ -291,17 +290,6 @@
 version = "0.2"
 optional = true
 
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "handleapi",
-    "minwindef",
-    "processenv",
-    "winbase",
-    "wincon",
-]
-optional = true
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
