--- a/Cargo.toml
+++ b/Cargo.toml
@@ -125,3 +125,3 @@
 [dependencies.derive_more]
-version = "2.0.1"
+version = "1"
 features = ["debug"]
@@ -155,3 +155,3 @@
 [dependencies.rand]
-version = "0.9"
+version = "0.8"
 optional = true
@@ -188,3 +188,3 @@
 [dependencies.toml]
-version = "<0.8.10"
+version = ">= 0.8, <1"
 optional = true
--- a/src/append/rolling_file/policy/compound/trigger/time.rs
+++ b/src/append/rolling_file/policy/compound/trigger/time.rs
@@ -283,7 +283,7 @@
         let current = get_current_time();
         let next_time = self.get_next_time(current);
         let next_roll_time = if self.config.max_random_delay > 0 {
-            let random_delay = rand::rng().random_range(0..self.config.max_random_delay);
+            let random_delay = rand::thread_rng().gen_range(0..self.config.max_random_delay);
             next_time + Duration::seconds(random_delay as i64)
         } else {
             next_time
