Index: hyper-timeout/tests/client_upload.rs
===================================================================
--- hyper-timeout.orig/tests/client_upload.rs
+++ hyper-timeout/tests/client_upload.rs
@@ -77,7 +77,7 @@ async fn test_upload_timeout() {
 
     let h = hyper_util::client::legacy::connect::HttpConnector::new();
     let mut connector = TimeoutConnector::new(h);
-    connector.set_read_timeout(Some(Duration::from_millis(5)));
+    connector.set_read_timeout(Some(Duration::from_millis(500)));
 
     // comment this out and the test will fail
     connector.set_reset_reader_on_write(true);
