Index: rustix-linux-procfs/Cargo.toml
===================================================================
--- rustix-linux-procfs.orig/Cargo.toml
+++ rustix-linux-procfs/Cargo.toml
@@ -51,14 +51,15 @@ path = "src/lib.rs"
 version = "1.5.2"
 
 [dependencies.rustix]
-version = "1.0.0"
+version = ">= 0.38.37"
 features = [
     "alloc",
     "fs",
+    "pty",
 ]
 default-features = false
 
 [dev-dependencies.rustix]
-version = "1.0.0"
+version = ">= 0.38.37"
 features = ["std"]
 default-features = false
