--- a/src/metadata.rs
+++ b/src/metadata.rs
@@ -228,7 +228,7 @@
 #[cfg(test)]
 mod tests {
     use super::*;
-    use assert_matches::assert_matches;
+    // use assert_matches::assert_matches;
     use cfg_expr::Expression;
     use std::{env, path::PathBuf};
 
@@ -274,13 +274,13 @@
         )
     }
 
-    #[test]
+    /*#[test]
     fn parse_feature_not_string() {
         assert_matches!(
             parse_file("toml-feature-not-string"),
             Err(crate::Error::InvalidMetadata(_))
         );
-    }
+    }*/
 
     #[test]
     fn parse_override_name() {
--- a/src/test.rs
+++ b/src/test.rs
@@ -6,7 +6,7 @@
 use std::rc::Rc;
 use std::sync::Mutex;
 
-use assert_matches::assert_matches;
+//use assert_matches::assert_matches;
 
 use crate::Dependencies;
 
@@ -112,7 +112,7 @@
     toml(path, vec![]).unwrap_err()
 }
 
-fn toml_err_invalid(path: &str, err_ends_with: &str) {
+/*fn toml_err_invalid(path: &str, err_ends_with: &str) {
     let err = toml_err(path);
     assert_matches!(err, Error::InvalidMetadata(_));
 
@@ -122,7 +122,7 @@
             err_ends_with, err
         );
     }
-}
+}*/
 
 // Assert a PkgConfig error because requested lib version cannot be found
 fn toml_pkg_config_err_version(
@@ -146,12 +146,12 @@
     }
 }
 
-#[test]
+/*#[test]
 fn missing_file() {
     assert_matches!(toml_err("toml-missing-file"), Error::FailToRead(_, _));
-}
+}*/
 
-#[test]
+/*#[test]
 fn missing_key() {
     toml_err_invalid("toml-missing-key", "no package.metadata.system-deps");
 }
@@ -196,7 +196,7 @@
         "toml-unexpected-key",
         "metadata.system-deps.testlib: unexpected key color type string",
     );
-}
+}*/
 
 #[test]
 fn override_name() {
@@ -874,6 +874,7 @@
     assert!(libraries.get_by_name("testanotherlib").is_none());
 }
 
+/*
 #[test]
 fn invalid_cfg() {
     let err = toml(
@@ -884,6 +885,7 @@
 
     assert_matches!(err, Error::UnsupportedCfg(_));
 }
+*/
 
 #[test]
 fn static_one_lib() {
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,8 +51,6 @@
 
 [dependencies.version-compare]
 version = "0.0.10"
-[dev-dependencies.assert_matches]
-version = "1.5"
 
 [dev-dependencies.itertools]
 version = "0.10"
