--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,7 +46,7 @@ name = "system_deps"
 features = ["targets"]
 
 [dependencies.heck]
-version = "0.5"
+version = "0.4"
 
 [dependencies.pkg-config]
 version = "0.3.25"
@@ -57,13 +57,13 @@ features = ["parse"]
 default-features = false
 
 [dependencies.version-compare]
-version = "0.2"
+version = "0.1"
 
 [dev-dependencies.assert_matches]
 version = "1.5"
 
 [dev-dependencies.itertools]
-version = "0.14"
+version = "0.13"
 
 [dev-dependencies.lazy_static]
 version = "1"
