diff --git a/Cargo.toml b/Cargo.toml
index 880ebf7..e3b5097 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,20 +39,21 @@ features = ["sequoia-openpgp/default"]
 version = "1"
 
 [dependencies.env_logger]
-version = "0.11.0"
+version = "0.10"
 
 [dependencies.hex]
 version = "0.4.3"
 
 [dependencies.sequoia-openpgp]
-version = "1.21.2"
+version = "1"
 default-features = false
+features = ["crypto-nettle"]
 
 [dependencies.serde]
 version = "1"
 
 [dependencies.serde_yaml]
-version = "0.8.13"
+version = "0.9"
 
 [dependencies.structopt]
 version = "0.3.22"
@@ -71,13 +72,13 @@ features = [
 ]
 default-features = false
 
-[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
-version = "1"
-features = [
-    "crypto-cng",
-    "__implicit-crypto-backend-for-tests",
-]
-default-features = false
+#[target."cfg(windows)".dev-dependencies.sequoia-openpgp]
+#version = "1"
+#features = [
+#    "crypto-cng",
+#    "__implicit-crypto-backend-for-tests",
+#]
+#default-features = false
 
 [badges.gitlab]
 repository = "sequoia-pgp/sequoia-tpm"
