Index: hickory-server/tests/authority_battery/dynamic_update.rs
===================================================================
--- hickory-server.orig/tests/authority_battery/dynamic_update.rs
+++ hickory-server/tests/authority_battery/dynamic_update.rs
@@ -877,6 +877,7 @@ pub fn add_auth<A: DnssecAuthority>(auth
 macro_rules! define_update_test {
     ($new:ident; $( $f:ident, )*) => {
         $(
+            #[ignore]
             #[test]
             fn $f () {
                 let mut authority = crate::$new("../../tests/test-data/test_configs/example.com.zone", module_path!(), stringify!($f));
