--- chrono-tz.orig/src/lib.rs
+++ chrono-tz/src/lib.rs
@@ -147,2 +147,6 @@
 #[cfg(test)]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(test)]
 mod tests {
@@ -162,2 +166,3 @@
     use super::UTC;
+    use alloc::string::ToString;
     use chrono::{Duration, NaiveDate, TimeZone};
--- chrono-tz.orig/src/serde.rs
+++ chrono-tz/src/serde.rs
@@ -46,2 +46,3 @@
     #[test]
+    #[cfg(feature = "std")]
     fn serde_de_error() {
