Index: rust-argon2/Cargo.toml
===================================================================
--- rust-argon2.orig/Cargo.toml
+++ rust-argon2/Cargo.toml
@@ -32,13 +32,13 @@ repository = "https://github.com/sru-sys
 name = "argon2"
 
 [dependencies.base64]
-version = "0.21"
+version = "0.22"
 
 [dependencies.blake2b_simd]
 version = "1.0"
 
 [dependencies.constant_time_eq]
-version = "0.3.0"
+version = ">= 0.1.5"
 
 [dependencies.crossbeam-utils]
 version = "0.8"
