Index: adler2/Cargo.toml
===================================================================
--- adler2.orig/Cargo.toml
+++ adler2/Cargo.toml
@@ -71,7 +71,6 @@ search = 'https://docs.rs/adler/[a-z0-9\
 
 [features]
 default = ["std"]
-rustc-dep-of-std = ["core"]
 std = []
 
 [lib]
@@ -83,9 +82,4 @@ path = "src/lib.rs"
 #path = "benches/bench.rs"
 #harness = false
 
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [dev-dependencies]
