Index: postcard/Cargo.toml
===================================================================
--- postcard.orig/Cargo.toml
+++ postcard/Cargo.toml
@@ -89,7 +89,7 @@ name = "max_size"
 path = "tests/max_size.rs"
 
 [dependencies.cobs]
-version = "0.3.0"
+version = ">= 0.2.3, < 0.4"
 default-features = false
 
 [dependencies.crc]
@@ -97,7 +97,7 @@ version = "3.0.1"
 optional = true
 
 [dependencies.defmt]
-version = "1"
+version = ">= 0.3.10, < 2"
 optional = true
 
 [dependencies.embedded-io-06]
