--- a/Cargo.toml
+++ b/Cargo.toml
@@ -93,10 +93,6 @@
 version = "1.0.12"
 optional = true
 
-[dependencies.postcard-derive]
-version = "0.1.2"
-optional = true
-
 [dependencies.serde]
 version = "1.0.100"
 features = ["derive"]
@@ -109,7 +105,6 @@
 ]
 default = ["heapless-cas"]
 embedded-io = ["dep:embedded-io"]
-experimental-derive = ["postcard-derive"]
 heapless-cas = [
     "heapless",
     "heapless/portable-atomic",
