Index: trust-dns-recursor/Cargo.toml
===================================================================
--- trust-dns-recursor.orig/Cargo.toml
+++ trust-dns-recursor/Cargo.toml
@@ -144,11 +144,6 @@ dns-over-openssl = [
     "trust-dns-proto/dns-over-openssl",
     "trust-dns-resolver/dns-over-openssl",
 ]
-dns-over-quic = [
-    "dns-over-rustls",
-    "trust-dns-proto/dns-over-quic",
-    "trust-dns-resolver/dns-over-quic",
-]
 dns-over-rustls = [
     "dns-over-tls",
     "dnssec-ring",
