Index: rusqlite/Cargo.toml
===================================================================
--- rusqlite.orig/Cargo.toml
+++ rusqlite/Cargo.toml
@@ -83,13 +83,13 @@ version = "1.1"
 optional = true
 
 [dependencies.fallible-iterator]
-version = "0.2"
+version = ">= 0.2, < 1.0"
 
 [dependencies.fallible-streaming-iterator]
 version = "0.1"
 
 [dependencies.hashlink]
-version = "0.8"
+version = ">= 0.8, < 0.11"
 
 [dependencies.libsqlite3-sys]
 version = "0.26.0"
