--- a/src/build.rs
+++ b/src/build.rs
@@ -68,3 +68,3 @@
 /// Directory and enclosed contents will be removed when return object is dropped.
-#[cfg(feature = "default")]
+#[cfg(feature = "vendored-libbpf")]
 fn extract_libbpf_headers_to_disk(target_dir: &Path) -> Result<Option<PathBuf>> {
@@ -90,3 +90,3 @@
 
-#[cfg(not(feature = "default"))]
+#[cfg(not(feature = "vendored-libbpf"))]
 fn extract_libbpf_headers_to_disk(_target_dir: &Path) -> Result<Option<PathBuf>> {
