diff --git a/Cargo.toml b/Cargo.toml
index a09d196..137552f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,7 +54,7 @@ path = "src/main.rs"
 version = "1.0.40"
 
 [dependencies.cargo_metadata]
-version = "0.19.1"
+version = "0.23"
 
 [dependencies.clap]
 version = "4.0.32"
diff --git a/src/metadata.rs b/src/metadata.rs
index 73915fe..bf306a0 100644
--- a/src/metadata.rs
+++ b/src/metadata.rs
@@ -118,7 +118,7 @@ fn get_package(package: &Package, workspace_target_dir: &Path) -> Result<Vec<Unp
                         .to_string();
 
                     return Some(UnprocessedObj {
-                        package: package.name.clone(),
+                        package: package.name.to_string(),
                         out: target_dir.clone(),
                         path,
                         name,
