--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,8 +47,6 @@
 features = [
     "radix",
     "format",
-    "write-integers",
-    "write-floats",
     "parse-integers",
     "parse-floats",
 ]
@@ -74,69 +72,49 @@
 path = "tests/issue_97_tests.rs"
 
 [dependencies.lexical-parse-float]
-version = "1.0.5"
+version = ">= 0.8.5"
 optional = true
 default-features = false
 
 [dependencies.lexical-parse-integer]
-version = "1.0.5"
+version = ">= 0.8.5"
 optional = true
 default-features = false
 
 [dependencies.lexical-util]
-version = "1.0.5"
-default-features = false
-
-[dependencies.lexical-write-float]
-version = "1.0.5"
-optional = true
-default-features = false
-
-[dependencies.lexical-write-integer]
-version = "1.0.5"
-optional = true
+version = ">= 0.8.5"
 default-features = false
 
 [dev-dependencies.approx]
-version = "0.5.0"
+version = ">= 0.5.0"
 
 [features]
 compact = [
-    "lexical-write-integer?/compact",
-    "lexical-write-float?/compact",
     "lexical-parse-integer?/compact",
     "lexical-parse-float?/compact",
 ]
 default = [
     "std",
-    "write-integers",
-    "write-floats",
     "parse-integers",
     "parse-floats",
 ]
 f128 = [
     "lexical-util/f128",
     "lexical-parse-float?/f128",
-    "lexical-write-float?/f128",
 ]
 f16 = [
     "lexical-util/f16",
     "lexical-parse-float?/f16",
-    "lexical-write-float?/f16",
 ]
 floats = ["lexical-util/floats"]
 format = [
     "lexical-util/format",
     "lexical-parse-integer?/format",
     "lexical-parse-float?/format",
-    "lexical-write-integer?/format",
-    "lexical-write-float?/format",
 ]
 integers = ["lexical-util/integers"]
 lint = [
     "lexical-util/lint",
-    "lexical-write-integer?/lint",
-    "lexical-write-float?/lint",
     "lexical-parse-integer?/lint",
     "lexical-parse-float?/lint",
 ]
@@ -153,33 +131,17 @@
 ]
 power-of-two = [
     "lexical-util/power-of-two",
-    "lexical-write-integer?/power-of-two",
-    "lexical-write-float?/power-of-two",
     "lexical-parse-integer?/power-of-two",
     "lexical-parse-float?/power-of-two",
 ]
 radix = [
     "lexical-util/radix",
-    "lexical-write-integer?/radix",
-    "lexical-write-float?/radix",
     "lexical-parse-integer?/radix",
     "lexical-parse-float?/radix",
 ]
 std = [
     "lexical-util/std",
-    "lexical-write-integer/std",
-    "lexical-write-float/std",
     "lexical-parse-integer/std",
     "lexical-parse-float/std",
 ]
 write = ["lexical-util/write"]
-write-floats = [
-    "lexical-write-float",
-    "write",
-    "floats",
-]
-write-integers = [
-    "lexical-write-integer",
-    "write",
-    "integers",
-]
