--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,7 +39,7 @@ panic = "abort"
 [lib]
 crate-type = ["lib", "staticlib", "cdylib"]
 [dependencies.arrayvec]
-version = "0.4"
+version = ">= 0.4,< 0.6"
 features = ["array-sizes-33-128"]
 optional = true
 
@@ -67,7 +67,7 @@ version = "1.1.0"
 [dev-dependencies.approx]
 version = "0.3.0"
 [build-dependencies.rustc_version]
-version = "0.2"
+version = "0.4"
 
 [features]
 correct = ["arrayvec", "table"]
