Index: lexical-core/Cargo.toml
===================================================================
--- lexical-core.orig/Cargo.toml
+++ lexical-core/Cargo.toml
@@ -65,9 +65,9 @@ optional = true
 [dependencies.static_assertions]
 version = "1.1.0"
 [dev-dependencies.approx]
-version = "0.3.0"
+version = ">= 0.3, < 0.6"
 [build-dependencies.rustc_version]
-version = "0.2"
+version = "0.4"
 
 [features]
 correct = ["arrayvec", "table"]
