Index: thiserror-core/Cargo.toml
===================================================================
--- thiserror-core.orig/Cargo.toml
+++ thiserror-core/Cargo.toml
@@ -29,6 +29,7 @@ keywords = [
 categories = ["rust-patterns"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/FlorianUekermann/thiserror/tree/core"
+autotests = false
 
 [package.metadata.docs.rs]
 targets = ["x86_64-unknown-linux-gnu"]
@@ -53,3 +54,9 @@ features = ["diff"]
 [features]
 default = ["std"]
 std = ["thiserror-impl/std"]
+[[test]]
+name = "compiletest"
+
+[[test]]
+name = "test_option"
+
