diff --git a/src/models/link/metadata.rs b/src/models/link/metadata.rs
index 9625032..0825c50 100644
--- a/src/models/link/metadata.rs
+++ b/src/models/link/metadata.rs
@@ -221,2 +221,3 @@ mod test {
     #[test]
+    #[ignore = "https://github.com/in-toto/in-toto-rs/issues/99#issuecomment-2536495328"]
     fn serialize_linkmetadata() {
@@ -261,2 +262,3 @@ mod test {
     #[test]
+    #[ignore = "https://github.com/in-toto/in-toto-rs/issues/99#issuecomment-2536495328"]
     fn deserialize_linkmetadata() {
diff --git a/src/models/metadata.rs b/src/models/metadata.rs
index c0811a5..cb9ce75 100644
--- a/src/models/metadata.rs
+++ b/src/models/metadata.rs
@@ -510,2 +510,3 @@ mod tests {
     #[test]
+    #[ignore = "https://github.com/in-toto/in-toto-rs/issues/99#issuecomment-2536495328"]
     fn serialize_link_metablock() {
