--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,19 +33,19 @@
 repository = "https://github.com/watchexec/clearscreen"
 
 [dependencies.terminfo]
-version = "0.8.0"
+version = "0.9"
 
 [dependencies.thiserror]
 version = "1.0.38"
 
 [dependencies.which]
-version = "6.0.1"
+version = "8.0"
 
 [features]
 windows-console = []
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.28.0"
+version = "0.29"
 features = [
     "fs",
     "term",
