--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,3 +41,2 @@
 [features]
-windows-console = []
 
@@ -75,12 +74,1 @@ features = [
 default-features = false
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Console",
-    "Win32_System_SystemInformation",
-    "Win32_System_SystemServices",
-    "Win32_NetworkManagement_NetManagement",
-]
-default-features = false
