--- a/src/net.rs
+++ b/src/net.rs
@@ -15,7 +15,7 @@
 use futures_io::{AsyncRead, AsyncWrite};
 use futures_util::future::FutureExt;
 use hickory_resolver::proto::tcp::{Connect, DnsTcpStream};
-use hickory_resolver::proto::udp::{DnsUdpSocket, QuicLocalAddr, UdpSocket};
+use hickory_resolver::proto::udp::{DnsUdpSocket, UdpSocket};
 use pin_utils::pin_mut;
 use socket2::{Domain, Protocol, Socket, Type};
 
@@ -58,11 +58,6 @@
     }
 }
 
-impl QuicLocalAddr for AsyncStdUdpSocket {
-    fn local_addr(&self) -> io::Result<SocketAddr> {
-        self.0.local_addr()
-    }
-}
 
 #[async_trait]
 impl UdpSocket for AsyncStdUdpSocket {
--- a/src/runtime.rs
+++ b/src/runtime.rs
@@ -9,7 +9,7 @@
 use std::future::Future;
 use std::net::SocketAddr;
 use std::pin::Pin;
-
+use std::io;
 use hickory_resolver::proto::error::ProtoError;
 use hickory_resolver::proto::Executor;
 
@@ -126,7 +126,7 @@
         &self,
         config: &NameServerConfig,
         options: &ResolverOpts,
-    ) -> Self::FutureConn {
+    ) -> Result<Self::FutureConn, io::Error> {
         self.connection_provider.new_connection(config, options)
     }
 }
