--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,3 +60,3 @@
 [dev-dependencies.rand]
-version = "0.9"
+version = "0.8"
 
--- a/src/date.rs
+++ b/src/date.rs
@@ -567,7 +567,7 @@ mod test {
             .unwrap()
             .as_secs();
         for _ in 0..10000 {
-            let sec = rand::rng().random_range(0..upper);
+            let sec = rand::thread_rng().gen_range(0..upper);
             let (s, time) = from_sec(sec);
             assert_eq!(parse_rfc3339(&s).unwrap(), time);
             assert_eq!(format_rfc3339(time).to_string(), s);
@@ -577,7 +577,7 @@ mod test {
     #[test]
     fn random_wide_range() {
         for _ in 0..100_000 {
-            let sec = rand::rng().random_range(0..max::SECONDS);
+            let sec = rand::thread_rng().gen_range(0..max::SECONDS);
             let (s, time) = from_sec(sec);
             assert_eq!(parse_rfc3339(&s).unwrap(), time);
             assert_eq!(format_rfc3339(time).to_string(), s);
--- a/src/duration.rs
+++ b/src/duration.rs
@@ -724,7 +724,7 @@ mod test {
     #[test]
     fn random_second() {
         for _ in 0..10000 {
-            let sec = rand::rng().random_range(0..253_370_764_800);
+            let sec = rand::thread_rng().gen_range(0..253_370_764_800);
             let d = Duration::new(sec, 0);
             assert_eq!(d, parse_duration(&format_duration(d).to_string()).unwrap());
         }
@@ -733,8 +733,8 @@ mod test {
     #[test]
     fn random_any() {
         for _ in 0..10000 {
-            let sec = rand::rng().random_range(0..253_370_764_800);
-            let nanos = rand::rng().random_range(0..1_000_000_000);
+            let sec = rand::thread_rng().gen_range(0..253_370_764_800);
+            let nanos = rand::thread_rng().gen_range(0..1_000_000_000);
             let d = Duration::new(sec, nanos);
             assert_eq!(d, parse_duration(&format_duration(d).to_string()).unwrap());
         }
