Index: cargo-outdated/Cargo.toml
===================================================================
--- cargo-outdated.orig/Cargo.toml
+++ cargo-outdated/Cargo.toml
@@ -46,7 +46,7 @@ name = "cargo-outdated"
 version = "1.0"
 
 [dependencies.cargo]
-version = "0.77.0"
+version = "0.79.0"
 
 [dependencies.clap]
 version = "4.1.4"
Index: cargo-outdated/src/cargo_ops/elaborate_workspace.rs
===================================================================
--- cargo-outdated.orig/src/cargo_ops/elaborate_workspace.rs
+++ cargo-outdated/src/cargo_ops/elaborate_workspace.rs
@@ -18,7 +18,7 @@ use cargo::{
         Dependency, FeatureValue, Package, PackageId, Workspace,
     },
     ops::{self, Packages},
-    util::{interning::InternedString, CargoResult, Config},
+    util::{interning::InternedString, CargoResult, GlobalContext},
 };
 use serde::{Deserialize, Serialize};
 use tabwriter::TabWriter;
@@ -87,7 +87,7 @@ impl<'ela> ElaborateWorkspace<'ela> {
         // The CompileKind, this has no target since it's the temp workspace
         // targets are blank since we don't need to fully build for the targets to get
         // the dependencies
-        let compile_kind = CompileKind::from_requested_targets(workspace.config(), &[])?;
+        let compile_kind = CompileKind::from_requested_targets(workspace.gctx(), &[])?;
         let mut target_data = RustcTargetData::new(workspace, &compile_kind)?;
         let ws_resolve = ops::resolve_ws_with_opts(
             workspace,
@@ -97,8 +97,6 @@ impl<'ela> ElaborateWorkspace<'ela> {
             &specs,
             HasDevUnits::Yes,
             ForceAllTargets::Yes,
-            // TODO: respect MSRV here
-            None,
         )?;
         let packages = ws_resolve.pkg_set;
         let resolve = ws_resolve
@@ -215,7 +213,7 @@ impl<'ela> ElaborateWorkspace<'ela> {
         compat: &ElaborateWorkspace<'_>,
         latest: &ElaborateWorkspace<'_>,
         options: &Options,
-        _config: &Config,
+        _config: &GlobalContext,
         root: PackageId,
         skip: &HashSet<String>,
     ) -> CargoResult<()> {
Index: cargo-outdated/src/cargo_ops/temp_project.rs
===================================================================
--- cargo-outdated.orig/src/cargo_ops/temp_project.rs
+++ cargo-outdated/src/cargo_ops/temp_project.rs
@@ -16,7 +16,7 @@ use cargo::{
         config::SourceConfigMap,
         source::{QueryKind, Source},
     },
-    util::{cache_lock::CacheLockMode, network::PollExt, CargoResult, Config},
+    util::{cache_lock::CacheLockMode, network::PollExt, CargoResult, GlobalContext},
 };
 use semver::{Version, VersionReq};
 use tempfile::{Builder, TempDir};
@@ -30,7 +30,7 @@ pub struct TempProject<'tmp> {
     pub workspace: Rc<RefCell<Option<Workspace<'tmp>>>>,
     pub temp_dir: TempDir,
     manifest_paths: Vec<PathBuf>,
-    config: Config,
+    config: GlobalContext,
     relative_manifest: String,
     options: &'tmp Options,
     is_workspace_project: bool,
@@ -173,7 +173,7 @@ impl<'tmp> TempProject<'tmp> {
         root: &Path,
         relative_manifest: &str,
         options: &Options,
-    ) -> CargoResult<Config> {
+    ) -> CargoResult<GlobalContext> {
         let shell = ::cargo::core::Shell::new();
         let cwd = env::current_dir()
             .with_context(|| "Cargo couldn't get the current directory of the process")?;
@@ -191,7 +191,7 @@ impl<'tmp> TempProject<'tmp> {
         // if it is, set it in the configure options
         let cargo_home_path = std::env::var_os("CARGO_HOME").map(std::path::PathBuf::from);
 
-        let mut config = Config::new(shell, cwd, homedir);
+        let mut config = GlobalContext::new(shell, cwd, homedir);
         config.configure(
             0,
             options.verbose == 0,
@@ -212,7 +212,7 @@ impl<'tmp> TempProject<'tmp> {
             recursive: false,
             precise: None,
             to_update: Vec::new(),
-            config: &self.config,
+            gctx: &self.config,
             dry_run: false,
             workspace: self.is_workspace_project,
         };
@@ -390,7 +390,7 @@ impl<'tmp> TempProject<'tmp> {
         let version = package_id.version();
         let source_id = package_id.source_id().with_locked_precise();
         let query_result = {
-            let ws_config = workspace.workspace.config();
+            let ws_config = workspace.workspace.gctx();
             let _lock = ws_config.acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
             let source_config = SourceConfigMap::new(ws_config)?;
             let mut source = source_config.load(source_id, &HashSet::new())?;
Index: cargo-outdated/src/main.rs
===================================================================
--- cargo-outdated.orig/src/main.rs
+++ cargo-outdated/src/main.rs
@@ -16,7 +16,7 @@ use cargo::{
     util::{
         important_paths::find_root_manifest_for_wd,
         network::http::{http_handle, needs_custom_http_transport},
-        CargoResult, CliError, Config,
+        CargoResult, CliError, GlobalContext,
     },
 };
 
@@ -30,7 +30,7 @@ fn main() {
     env_logger::init();
     let options = cli::parse();
 
-    let mut config = match Config::default() {
+    let mut config = match GlobalContext::default() {
         Ok(cfg) => cfg,
         Err(e) => {
             let mut shell = cargo::core::Shell::new();
@@ -71,7 +71,7 @@ fn main() {
 }
 
 /// executes the cargo-outdate command with the cargo configuration and options
-pub fn execute(options: Options, config: &mut Config) -> CargoResult<i32> {
+pub fn execute(options: Options, config: &mut GlobalContext) -> CargoResult<i32> {
     // Check if $CARGO_HOME is set before capturing the config environment
     // if it is, set it in the configure options
     let cargo_home_path = std::env::var_os("CARGO_HOME").map(std::path::PathBuf::from);
