--- a/src/program/mod.rs
+++ b/src/program/mod.rs
@@ -80,7 +80,7 @@
                 args.insert_str(0, "credential-");
                 args.insert_str(0, " ");
                 args.insert_str(0, git_program.to_string_lossy().as_ref());
-                gix_command::prepare(gix_path::from_bstr(args.as_ref()).into_owned())
+                gix_command::prepare(gix_path::from_bstr(AsRef::<bstr::BStr>::as_ref(&args)).into_owned())
                     .arg(action.as_arg(true))
                     .with_shell_allow_argument_splitting()
                     .into()
