Index: eza/src/fs/file.rs
===================================================================
--- eza.orig/src/fs/file.rs
+++ eza/src/fs/file.rs
@@ -598,6 +598,7 @@ impl<'dir> File<'dir> {
             // allow it to compile cleanly on Linux.
             #[allow(trivial_numeric_casts)]
             #[allow(clippy::unnecessary_cast, clippy::useless_conversion)]
+            unsafe {
             f::Size::DeviceIDs(f::DeviceIDs {
                 major: libc::major(
                     device_id
@@ -610,6 +611,7 @@ impl<'dir> File<'dir> {
                         .expect("Malformed device major ID when getting filesize"),
                 ) as u32,
             })
+            }
         } else if self.is_file() {
             f::Size::Some(self.metadata().map_or(0, std::fs::Metadata::len))
         } else {
