Index: eza/src/fs/file.rs
===================================================================
--- eza.orig/src/fs/file.rs
+++ eza/src/fs/file.rs
@@ -542,6 +542,7 @@ impl<'dir> File<'dir> {
             self.recursive_size.map_or(f::Blocksize::None, |_, blocks| {
                 f::Blocksize::Some(blocks * 512)
             })
+            }
         } else if self.is_file() {
             // Note that metadata.blocks returns the number of blocks
             // for 512 byte blocks according to the POSIX standard
@@ -616,6 +617,7 @@ impl<'dir> File<'dir> {
                 let device_id = device_id
                     .try_into()
                     .expect("Malformed device major ID when getting filesize");
+                unsafe {
                 f::Size::DeviceIDs(f::DeviceIDs {
                     major: unsafe { libc::major(device_id) as u32 },
                     minor: unsafe { libc::minor(device_id) as u32 },
