Index: eza/Cargo.toml
===================================================================
--- eza.orig/Cargo.toml
+++ eza/Cargo.toml
@@ -226,7 +226,7 @@ nix-generated = []
 nix-local = []
 powertest = []
 vendored-libgit2 = ["git2/vendored-libgit2"]
-vendored-openssl = ["git2/vendored-openssl"]
+#vendored-openssl = ["git2/vendored-openssl"]
 
 [target.'cfg(target_os = "linux")'.dependencies.proc-mounts]
 version = "0.3"
