Index: eza/Cargo.toml
===================================================================
--- eza.orig/Cargo.toml
+++ eza/Cargo.toml
@@ -219,7 +219,7 @@ nix = []
 nix-generated = []
 nix-local = []
 powertest = []
-vendored-libgit2 = ["git2/vendored-libgit2"]
+#vendored-libgit2 = ["git2/vendored-libgit2"]
 #vendored-openssl = ["git2/vendored-openssl"]
 
 [target.'cfg(target_os = "linux")'.dependencies.proc-mounts]
