Index: eza/Cargo.toml
===================================================================
--- eza.orig/Cargo.toml
+++ eza/Cargo.toml
@@ -177,7 +177,7 @@ default = ["git"]
 git = ["git2"]
 nix = []
 nix-local = []
-vendored-libgit2 = ["git2/vendored-libgit2"]
+#vendored-libgit2 = ["git2/vendored-libgit2"]
 #vendored-openssl = ["git2/vendored-openssl"]
 
 [target."cfg(target_os = \"linux\")".dependencies.proc-mounts]
