Index: eza/Cargo.toml
===================================================================
--- eza.orig/Cargo.toml
+++ eza/Cargo.toml
@@ -121,7 +121,7 @@ version = "0.1.0"
 version = "0.3"
 
 [dependencies.chrono]
-version = "0.4.40"
+version = "0.4"
 features = ["clock"]
 default-features = false
 
@@ -208,7 +208,7 @@ features = ["html_reports"]
 version = "0.14"
 
 [build-dependencies.chrono]
-version = "0.4.40"
+version = "0.4"
 features = ["clock"]
 default-features = false
 
