Index: eza/Cargo.toml
===================================================================
--- eza.orig/Cargo.toml
+++ eza/Cargo.toml
@@ -97,7 +97,7 @@ version = "0.12.2"
 features = ["ansi_colours"]
 
 [dependencies.chrono]
-version = "0.4.33"
+version = "0.4"
 features = ["clock"]
 default-features = false
 
@@ -170,7 +170,7 @@ features = ["html_reports"]
 version = "0.14"
 
 [build-dependencies.chrono]
-version = "0.4.33"
+version = "0.4"
 features = ["clock"]
 default-features = false
 
