Index: eza/Cargo.toml
===================================================================
--- eza.orig/Cargo.toml
+++ eza/Cargo.toml
@@ -191,7 +191,7 @@ version = "0.9"
 version = "0.4.2"
 
 [dependencies.timeago]
-version = "0.4.2"
+version = "0.4"
 default-features = false
 
 [dependencies.unicode-width]
@@ -205,7 +205,7 @@ version = "0.5.1"
 features = ["html_reports"]
 
 [dev-dependencies.trycmd]
-version = "0.15"
+version = "0.14"
 
 [build-dependencies.chrono]
 version = "0.4.40"
