--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,7 @@
 repository = "https://github.com/Smithay/drm-rs"
 
 [build-dependencies.bindgen]
-version = "0.69.1"
+version = "0.71"
 optional = true
 
 [build-dependencies.pkg-config]
@@ -38,7 +38,7 @@
 ]
 
 [target."cfg(any(target_os = \"android\", target_os = \"linux\"))".dependencies.linux-raw-sys]
-version = "0.6"
+version = "0.4"
 features = [
     "general",
     "no_std",
