Index: mio-0.6/src/lib.rs
===================================================================
--- mio-0.6.orig/src/lib.rs
+++ mio-0.6/src/lib.rs
@@ -3,7 +3,8 @@
 // deprecated APIs.
 #![allow(bare_trait_objects, deprecated, unknown_lints)]
 #![deny(missing_docs, missing_debug_implementations)]
-#![cfg_attr(test, deny(warnings))]
+// #![cfg_attr(test, deny(warnings))]
+#![warn(useless_deprecated)]
 
 // Many of mio's public methods violate this lint, but they can't be fixed
 // without a breaking change.
