diff --git a/Cargo.toml b/Cargo.toml
index aa851dc..6df5481 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,16 +32,16 @@ repository = "https://github.com/la10736/rstest"
 proc-macro = true
 
 [dependencies.cfg-if]
-version = "1.0.0"
+version = "1"
 
 [dependencies.proc-macro2]
-version = "1.0.39"
+version = "1"
 
 [dependencies.quote]
-version = "1.0.19"
+version = "1"
 
 [dependencies.syn]
-version = "1.0.98"
+version = "1"
 features = [
     "full",
     "parsing",
@@ -51,30 +51,30 @@ features = [
 ]
 
 [dependencies.unicode-ident]
-version = "1.0.5"
+version = "1"
 
 [dev-dependencies.actix-rt]
-version = "2.7.0"
+version = "2"
 
 [dev-dependencies.async-std]
-version = "1.12.0"
+version = "1"
 features = ["attributes"]
 
 [dev-dependencies.pretty_assertions]
-version = "1.2.1"
+version = "1"
 
 [dev-dependencies.rstest]
-version = "0.15.0"
+version = "0.15"
 default-features = false
 
 [dev-dependencies.rstest_reuse]
-version = "0.4.0"
+version = "0.4"
 
 [dev-dependencies.rstest_test]
-version = "0.9.0"
+version = "0.9"
 
 [build-dependencies.rustc_version]
-version = "0.4.0"
+version = "0.4"
 
 [features]
 async-timeout = []
