--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,7 +46,7 @@
 version = "2.3.1"
 
 [dependencies.image]
-version = "0.24.7"
+version = "0.25"
 optional = true
 default-features = false
 
@@ -57,7 +57,7 @@
 version = "1.0.51"
 
 [dev-dependencies.image]
-version = "0.24.7"
+version = "0.25"
 features = [
     "jpeg",
     "png",
@@ -67,9 +67,6 @@
 [dev-dependencies.lcms2]
 version = "6.0.3"
 
-[dev-dependencies.testresult]
-version = "0.3.0"
-
 [features]
 bench = []
 default = [
@@ -79,7 +76,6 @@
 docs = ["jpegxl-sys/docs"]
 image = ["dep:image"]
 threads = ["jpegxl-sys/threads"]
-vendored = ["jpegxl-sys/vendored"]
 
 [target."cfg(not(target_family = \"wasm\"))".dev-dependencies.criterion]
 version = "0.5.1"
