Index: combine/Cargo.toml
===================================================================
--- combine.orig/Cargo.toml
+++ combine/Cargo.toml
@@ -167,17 +167,17 @@ package = "futures"
 version = "1.0"
 
 [dev-dependencies.partial-io]
-version = "0.3"
+version = "0.5.4"
 features = [
     "tokio",
     "quickcheck",
 ]
 
 [dev-dependencies.quick-error]
-version = "1.0"
+version = "2.0"
 
 [dev-dependencies.quickcheck]
-version = "0.6"
+version = "1.0.3"
 
 [dev-dependencies.tokio-02-dep]
 version = "0.2"
