Index: combine/Cargo.toml
===================================================================
--- combine.orig/Cargo.toml
+++ combine/Cargo.toml
@@ -66,16 +66,6 @@ required-features = [
     "futures-io-03",
 ]
 
-[[bench]]
-name = "json"
-harness = false
-required-features = ["std"]
-
-[[bench]]
-name = "http"
-harness = false
-required-features = ["std"]
-
 [dependencies.bytes]
 version = "1"
 optional = true
@@ -119,10 +109,6 @@ default-features = false
 [dev-dependencies.bytes]
 version = "1"
 
-[dev-dependencies.criterion]
-version = "0.3"
-default-features = false
-
 [dev-dependencies.futures-03-dep]
 version = "0.3.1"
 package = "futures"
