diff --git a/src/tests/mod.rs b/src/tests/mod.rs
index 42c5486..4ae4fa4 100644
--- a/src/tests/mod.rs
+++ b/src/tests/mod.rs
@@ -38,6 +38,7 @@ test_atomic_int_pub!(u64);
     ),
     cfg(not(qemu))
 )]
+#[cfg(feature = "fallback")]
 test_atomic_int_pub!(i128);
 #[cfg_attr(
     all(
@@ -49,6 +50,7 @@ test_atomic_int_pub!(i128);
     ),
     cfg(not(qemu))
 )]
+#[cfg(feature = "fallback")]
 test_atomic_int_pub!(u128);
 #[cfg(qemu)]
 #[cfg(target_arch = "powerpc64")]
