Relax the dependency-versions to what's in Debian

Index: gix-transport/Cargo.toml
===================================================================
--- gix-transport.orig/Cargo.toml
+++ gix-transport/Cargo.toml
@@ -50,7 +50,7 @@ version = "0.1.51"
 optional = true
 
 [dependencies.base64]
-version = "0.22.1"
+version = "0.21"
 optional = true
 
 [dependencies.bstr]
@@ -74,7 +74,7 @@ version = "0.3.16"
 optional = true
 
 [dependencies.futures-lite]
-version = "2.1.0"
+version = ">= 1.12.0"
 features = ["std"]
 optional = true
 default-features = false
@@ -106,11 +106,8 @@ version = "0.2.6"
 optional = true
 
 [dependencies.reqwest]
-version = "0.12.0"
+version = ">= 0.11.24"
 features = [
-    "charset",
-    "http2",
-    "macos-system-configuration",
     "blocking",
 ]
 optional = true
