Index: reqwest-0.11/Cargo.toml
===================================================================
--- reqwest-0.11.orig/Cargo.toml
+++ reqwest-0.11/Cargo.toml
@@ -191,13 +191,6 @@ gzip = [
     "tokio-util",
 ]
 hickory-dns = ["hickory-resolver"]
-http3 = [
-    "rustls-tls-manual-roots",
-    "h3",
-    "h3-quinn",
-    "quinn",
-    "futures-channel",
-]
 json = ["serde_json"]
 multipart = ["mime_guess"]
 native-tls = ["default-tls"]
@@ -243,14 +236,6 @@ optional = true
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.h2]
 version = ">= 0.3.14, < 0.5"
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.h3]
-version = "0.0.3"
-optional = true
-
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.h3-quinn]
-version = "0.0.4"
-optional = true
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.hickory-resolver]
 version = "0.24"
 features = ["tokio-runtime"]
@@ -302,16 +287,6 @@ version = "2.1"
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.pin-project-lite]
 version = "0.2.0"
 
-[target."cfg(not(target_arch = \"wasm32\"))".dependencies.quinn]
-version = "0.10"
-features = [
-    "tls-rustls",
-    "ring",
-    "runtime-tokio",
-]
-optional = true
-default-features = false
-
 [target."cfg(not(target_arch = \"wasm32\"))".dependencies.rustls]
 version = "0.21.6"
 features = ["dangerous_configuration"]
