--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,24 +64,9 @@
 [dependencies.walkdir]
 version = "2.5.0"
 
-[target.'cfg(target_os = "macos")'.dependencies.core-foundation]
-version = "0.10.0"
-features = ["mac_os_10_7_support"]
-
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.169"
 
-[target."cfg(windows)".dependencies.miow]
-version = "0.6.0"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.59"
-features = [
-    "Win32_Storage_FileSystem",
-    "Win32_Foundation",
-    "Win32_System_Console",
-]
-
 [lints.clippy]
 dbg_macro = "warn"
 disallowed_methods = "warn"
