diff --git a/Cargo.toml b/Cargo.toml
index 5875ce8..dbc37b3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -74,7 +74,7 @@ name = "stdout_stderr_printing"
 path = "examples/stdout_stderr_printing.rs"
 
 [dependencies.indicatif]
-version = "0.17.9"
+version = "0.17"
 features = ["in_memory"]
 
 [dependencies.tracing]
@@ -93,12 +93,12 @@ version = "0.15.8"
 version = "0.11.0"
 
 [dev-dependencies.futures]
-version = "0.3.31"
+version = "0.3"
 
 [dev-dependencies.rand]
 version = "0.8.5"
 features = ["std_rng"]
 
 [dev-dependencies.tokio]
-version = "1.41.1"
+version = "1"
 features = ["full"]
