--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,21 +67,2 @@
 [features]
-bindgen-source = ["bindgen"]
-default = ["vendored-libbpf"]
-novendor = []
-static = [
-    "static-libbpf",
-    "static-libelf",
-    "static-zlib",
-]
-static-libbpf = []
-static-libelf = ["static-libbpf"]
-static-zlib = ["static-libbpf"]
-vendored = [
-    "vendored-libbpf",
-    "vendored-libelf",
-    "vendored-zlib",
-]
-vendored-libbpf = ["static-libbpf"]
-vendored-libelf = ["static-libelf"]
-vendored-zlib = ["static-zlib"]
 
--- a/bindings.h
+++ b/bindings.h
@@ -1 +0,0 @@
-#ifdef __LIBBPF_SYS_NOVENDOR
@@ -7,7 +5,0 @@
-#else
-#include "libbpf/include/uapi/linux/if_link.h"
-#include "libbpf/include/uapi/linux/perf_event.h"
-#include "libbpf/src/bpf.h"
-#include "libbpf/src/btf.h"
-#include "libbpf/src/libbpf.h"
-#endif /* __LIBBPF_SYS_NOVENDOR */
