--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,33 +35,18 @@
 name = "html2text"
 path = "src/lib.rs"
 
-[[example]]
-name = "html2term"
-path = "examples/html2term.rs"
-
-[[example]]
-name = "html2text"
-path = "examples/html2text.rs"
-
-[[bench]]
-name = "tables"
-path = "benches/tables.rs"
 
 [dependencies.backtrace]
 version = "0.3"
 optional = true
 
 [dependencies.dashmap]
-version = "~5.4"
+version = "6.1"
 optional = true
 
 [dependencies.html5ever]
 version = "0.27.0"
 
-[dependencies.lightningcss]
-version = "1.0.0-alpha.54"
-optional = true
-
 [dependencies.log]
 version = "0.4.20"
 optional = true
@@ -76,19 +61,18 @@
 version = "1.0.50"
 
 [dependencies.unicode-width]
-version = "=0.1.13"
+version = "0.2"
 
 [dev-dependencies.argparse]
 version = "0.2.2"
 
 [dev-dependencies.env_logger]
-version = "0.10.1"
+version = "0.11"
 
 [dev-dependencies.log]
 version = "0.4.20"
 
 [features]
-css = ["dep:lightningcss"]
 default = []
 html_trace = ["dep:log"]
 html_trace_bt = [
@@ -97,4 +81,4 @@
 ]
 
 [target."cfg(unix)".dev-dependencies.termion]
-version = "3.0"
+version = "1.0"
