--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,10 +58,6 @@ debug = []
 default = []
 deprecated = []
 raw-deprecated = ["deprecated"]
-unstable-markdown = [
-    "dep:pulldown-cmark",
-    "dep:anstyle",
-]
 unstable-v5 = ["deprecated"]
 
 [lib]
@@ -61,3 +61,3 @@
 [dependencies.heck]
-version = "0.5.0"
+version = ">= 0.4"
 
@@ -80,7 +76,7 @@ version = "0.5.0"
 [dependencies.proc-macro2]
 version = "1.0.69"
 
-[dependencies.pulldown-cmark]
+[disabled.dependencies.pulldown-cmark]
 version = "0.13.0"
 optional = true
 default-features = false
