Origin: vendor
Subject: Relax dependencies to work well with other rust crates in Debian
Forwarded: not-needed

diff --git a/Cargo.toml b/Cargo.toml
index 9a37877..1821f35 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@
 version = "5"
 
 [dependencies.gethostname]
-version = ">=0.4, <0.6"
+version = "0.4"
 
 [dependencies.num_cpus]
 version = "1"
@@ -65,7 +65,7 @@
 version = "1"
 
 [dependencies.rusqlite]
-version = ">=0.29, <0.32"
+version = "0.29"
 features = [
     "collation",
     "blob",
