--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,13 +28,6 @@
 license = "MIT/Apache-2.0"
 repository = "https://github.com/async-email/async-native-tls"
 
-[[test]]
-name = "google"
-required-features = ["runtime-async-std"]
-
-[[test]]
-name = "google-tokio"
-required-features = ["runtime-tokio"]
 
 [[test]]
 name = "bad"
--- a/tests/google-tokio.rs
+++ b/tests/google-tokio.rs
@@ -45,6 +45,7 @@
 }
 
 #[tokio::test]
+#[ignore = "no phoning home to google"]
 async fn fetch_google() {
     drop(env_logger::try_init());
 
--- a/tests/google.rs
+++ b/tests/google.rs
@@ -17,6 +17,7 @@
 }
 
 #[async_std::test]
+#[ignore = "no phoning home to google"]
 async fn fetch_google() {
     drop(env_logger::try_init());
 
