--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,10 +67,10 @@
 version = "3.2.1"
 
 [dependencies.dialoguer]
-version = "0.11"
+version = "0.10"
 
 [dependencies.env_logger]
-version = "0.11"
+version = "0.10"
 
 [dependencies.futures]
 version = "0.3.12"
@@ -109,7 +109,7 @@
 features = ["serde"]
 
 [dev-dependencies.trycmd]
-version = "0.15"
+version = "0.14"
 
 [features]
 all = [
@@ -122,5 +122,4 @@
 ]
 experimental = ["experimental-transfer-v2"]
 experimental-transfer-v2 = ["magic-wormhole/experimental-transfer-v2"]
-native-tls = ["magic-wormhole/native-tls"]
 tls = ["magic-wormhole/tls"]
