--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,7 +70,7 @@
 version = "0.11"
 
 [dependencies.env_logger]
-version = "0.11"
+version = "0.10"
 
 [dependencies.futures]
 version = "0.3.12"
@@ -109,12 +109,11 @@
 features = ["serde"]
 
 [dev-dependencies.trycmd]
-version = "0.15"
+version = "0.14"
 
 [features]
 all = [
     "default",
-    "magic-wormhole/native-tls",
 ]
 default = [
     "magic-wormhole/default",
@@ -122,5 +121,4 @@
 ]
 experimental = ["experimental-transfer-v2"]
 experimental-transfer-v2 = ["magic-wormhole/experimental-transfer-v2"]
-native-tls = ["magic-wormhole/native-tls"]
 tls = ["magic-wormhole/tls"]
