Index: serialport/tests/test_file_locking.rs
===================================================================
--- serialport.orig/tests/test_file_locking.rs
+++ serialport/tests/test_file_locking.rs
@@ -66,7 +66,7 @@ fn second_open_fails_flock(hw_config: Ha
     assert_eq!(second.unwrap_err().kind(), ErrorKind::NoDevice);
 }
 
-#[rstest]
+/*#[rstest]
 #[cfg(unix)]
 #[cfg_attr(not(feature = "hardware-tests"), ignore)]
 fn second_open_fails_lock(hw_config: HardwareConfig) {
@@ -88,7 +88,7 @@ fn second_open_fails_lock(hw_config: Har
     let second = serialport::new(&hw_config.port_1, 115200).open();
     assert!(second.is_err());
     assert_eq!(second.unwrap_err().kind(), ErrorKind::NoDevice);
-}
+}*/
 
 #[rstest]
 #[cfg(unix)]
