Index: serialport/Cargo.toml
===================================================================
--- serialport.orig/Cargo.toml
+++ serialport/Cargo.toml
@@ -113,7 +113,7 @@ path = "tests/test_tty.rs"
 version = "1.0.0"
 
 [dependencies.quote]
-version = ">=1, <=1.0.40"
+version = "1"
 
 [dependencies.scopeguard]
 version = "1.1"
@@ -128,10 +128,10 @@ version = "3.1.6"
 features = ["derive"]
 
 [dev-dependencies.libc]
-version = ">=0.2.0, <=0.2.163"
+version = "0.2.0"
 
 [dev-dependencies.os_str_bytes]
-version = ">=6.0, <6.6.0"
+version = "7"
 
 [dev-dependencies.quickcheck]
 version = "1.0.3"
@@ -140,7 +140,7 @@ version = "1.0.3"
 version = "1.0.0"
 
 [dev-dependencies.rstest]
-version = "0.12.0"
+version = "0.17"
 default-features = false
 
 [dev-dependencies.rustversion]
