--- a/Cargo.toml
+++ b/Cargo.toml
@@ -159,12 +158,0 @@
-[target.'cfg(any(target_os = "ios", target_os = "macos"))'.dependencies.core-foundation]
-version = "=0.10.0"
-
-[target.'cfg(any(target_os = "ios", target_os = "macos"))'.dependencies.core-foundation-sys]
-version = "0.8.4"
-
-[target.'cfg(any(target_os = "ios", target_os = "macos"))'.dependencies.io-kit-sys]
-version = "0.4.0"
-
-[target.'cfg(any(target_os = "ios", target_os = "macos"))'.dependencies.mach2]
-version = "0.4.1"
-
@@ -184,14 +171,0 @@
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = [
-    "Win32_Devices_Communication",
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_Registry",
-    "Win32_System_Diagnostics_Debug",
-    "Win32_Devices_DeviceAndDriverInstallation",
-    "Win32_System_Threading",
-    "Win32_Security",
-    "Win32_System_SystemServices",
-]
