--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,13 +64,13 @@
 
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "fileapi",
-    "sysinfoapi",
-    "minwindef",
-    "winbase",
-    "winerror",
-    "ws2def",
-    "ws2ipdef",
-    "pdh",
-]
+#[target."cfg(windows)".dependencies.winapi]
+#version = "0.3"
+#features = [
+#    "fileapi",
+#    "sysinfoapi",
+#    "minwindef",
+#    "winbase",
+#    "winerror",
+#    "ws2def",
+#    "ws2ipdef",
+#    "pdh",
+#]
