--- a/src/platform/mod.rs
+++ b/src/platform/mod.rs
@@ -105,2 +105,3 @@
     #[test]
+	#[ignore = "/sys/class/power_supply does not exist in some environments"]
     fn test_on_ac_power() {
@@ -110,2 +111,3 @@
     #[test]
+	#[ignore = "mounting requires special handling in some environments"]
     fn test_mounts() {
@@ -117,2 +119,3 @@
     #[test]
+	#[ignore = "mounting requires special handling in some environments"]
     fn test_mount_at() {
