Index: cfg-if/Cargo.toml
===================================================================
--- cfg-if.orig/Cargo.toml
+++ cfg-if/Cargo.toml
@@ -30,7 +30,6 @@ license = "MIT OR Apache-2.0"
 repository = "https://github.com/rust-lang/cfg-if"
 
 [features]
-rustc-dep-of-std = ["core"]
 
 [lib]
 name = "cfg_if"
@@ -39,8 +38,3 @@ path = "src/lib.rs"
 [[test]]
 name = "xcrate"
 path = "tests/xcrate.rs"
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
