--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,6 +44,11 @@
 [dev-dependencies.criterion]
 version = "0.5.1"
 
+[dev-dependencies.plotters]
+default_features = false
+features = ["ttf", "line_series", "bitmap_backend"]
+version = "0.3.5"
+
 [dev-dependencies.rayon]
 version = "1.10"
 
@@ -67,3 +72,7 @@
 ]
 optional = true
 default-features = false
+
+#needed to avoid problems with two copies of plotters-bitmap in the build.
+[patch.crates-io]
+plotters-bitmap = { path = "." }
