Index: numpy/Cargo.toml.orig
===================================================================
--- numpy.orig/Cargo.toml.orig
+++ numpy/Cargo.toml.orig
@@ -22,7 +22,7 @@ num-complex = ">= 0.2, < 0.5"
 num-integer = "0.1"
 num-traits = "0.2"
 ndarray = ">= 0.15, < 0.17"
-pyo3 = { version = "0.22.0", default-features = false, features = ["macros"] }
+pyo3 = { version = "0.25.1", default-features = false, features = ["macros"] }
 rustc-hash = "1.1"
 
 [dev-dependencies]
Index: numpy/Cargo.toml
===================================================================
--- numpy.orig/Cargo.toml
+++ numpy/Cargo.toml
@@ -101,7 +101,7 @@ version = "0.1"
 version = "0.2"
 
 [dependencies.pyo3]
-version = "0.22.0"
+version = "0.25.1"
 features = ["macros"]
 default-features = false
 
@@ -114,7 +114,7 @@ features = ["std"]
 default-features = false
 
 [dev-dependencies.pyo3]
-version = "0.22.0"
+version = "0.25.1"
 features = ["auto-initialize"]
 default-features = false
