--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,10 +39,3 @@
 [dev-dependencies.rand]
 version = "0.8"
 
-[target."cfg(target_os = \"windows\")".dependencies.winapi]
-version = "0.3"
-features = [
-    "wincon",
-    "processenv",
-    "winbase",
-]
--- a/src/tty/mod.rs
+++ b/src/tty/mod.rs
@@ -22,7 +22,7 @@
 #[cfg(target_os = "wasi")]
 pub use self::wasi::*;
 
-#[cfg(windows)]
-mod windows;
-#[cfg(windows)]
-pub use self::windows::*;
+//#[cfg(windows)]
+//mod windows;
+//#[cfg(windows)]
+//pub use self::windows::*;
