Index: hashbrown/Cargo.toml
===================================================================
--- hashbrown.orig/Cargo.toml
+++ hashbrown/Cargo.toml
@@ -42,7 +42,6 @@ repository = "https://github.com/rust-la
 
 [package.metadata.docs.rs]
 features = [
-    "nightly",
     "rayon",
     "serde",
     "raw-entry",
@@ -59,4 +58,0 @@
-nightly = [
-    "foldhash?/nightly",
-    "bumpalo/allocator_api",
-]
@@ -70,5 +70,0 @@
-rustc-dep-of-std = [
-    "nightly",
-    "core",
-    "rustc-internal-api",
-]
