Index: hashbrown/Cargo.toml
===================================================================
--- hashbrown.orig/Cargo.toml
+++ hashbrown/Cargo.toml
@@ -42,7 +42,6 @@ repository = "https://github.com/rust-la
 
 [package.metadata.docs.rs]
 features = [
-    "nightly",
     "rayon",
     "serde",
     "raw-entry",
@@ -59,13 +58,7 @@ default = [
 ]
 default-hasher = ["dep:foldhash"]
 inline-more = []
-nightly = ["bumpalo/allocator_api"]
 raw-entry = []
-rustc-dep-of-std = [
-    "nightly",
-    "core",
-    "rustc-internal-api",
-]
 rustc-internal-api = []
 
 [lib]
