Index: hashbrown/Cargo.toml
===================================================================
--- hashbrown.orig/Cargo.toml
+++ hashbrown/Cargo.toml
@@ -64,7 +64,6 @@ raw-entry = []
 rustc-dep-of-std = [
     "nightly",
     "core",
-    "alloc",
     "rustc-internal-api",
 ]
 rustc-internal-api = []
@@ -105,11 +104,6 @@ path = "benches/insert_unique_unchecked.
 name = "set_ops"
 path = "benches/set_ops.rs"
 
-[dependencies.alloc]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-alloc"
-
 [dependencies.allocator-api2]
 version = "0.2.9"
 features = ["alloc"]
