--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,9 +20,9 @@
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/sharkdp/numbat"
 
-[dependencies.attohttpc]
-version = "0.26.0"
-features = ["tls-rustls-webpki-roots"]
+[dependencies.reqwest]
+version = ">= 0.11.22"
+features = ["rustls-tls", "blocking"]
 default-features = false
 
 [dependencies.quick-xml]
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -36,7 +36,7 @@
 const ECB_XML_URL: &str = "https://www.ecb.europa.eu/stats/eurofxref/eurofxref-daily.xml";
 
 fn fetch_ecb_xml() -> Option<String> {
-    attohttpc::get(ECB_XML_URL).send().ok()?.text().ok()
+    reqwest::blocking::get(ECB_XML_URL).ok()?.text().ok()
 }
 
 pub fn fetch_exchange_rates() -> Option<ExchangeRates> {
