diff --git a/Cargo.toml b/Cargo.toml
index c738a76..7db5b6f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,7 +45,7 @@ name = "lz4-compress"
 path = "src/main.rs"
 
 [dependencies.byteorder]
-version = "0.5"
+version = "1"
 
 [dependencies.quick-error]
-version = "1"
+version = "2"
diff --git a/src/decompress.rs b/src/decompress.rs
index b37b265..4510650 100644
--- a/src/decompress.rs
+++ b/src/decompress.rs
@@ -8,11 +8,11 @@ quick_error! {
     pub enum Error {
         /// Expected another byte, but none found.
         ExpectedAnotherByte {
-            description("Expected another byte, found none.")
+            display("Expected another byte, found none.")
         }
         /// Deduplication offset out of bounds (not in buffer).
         OffsetOutOfBounds {
-            description("The offset to copy is not contained in the decompressed buffer.")
+            display("The offset to copy is not contained in the decompressed buffer.")
         }
     }
 }
