--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@
 version = "1.15.0"
 
 [dependencies.itertools]
-version = "0.12"
+version = "0.13"
 
 [dependencies.kanata-keyberon]
 version = "0.190.0"
@@ -63,7 +63,7 @@
 default-features = false
 
 [dependencies.miette]
-version = "5.7.0"
+version = "7.4.0"
 features = ["fancy"]
 
 [dependencies.once_cell]
@@ -73,7 +73,7 @@
 version = "0.12"
 
 [dependencies.patricia_tree]
-version = "0.8"
+version = "0.9"
 
 [dependencies.rustc-hash]
 version = "1.1.0"
