Index: sequoia-sq/Cargo.toml
===================================================================
--- sequoia-sq.orig/Cargo.toml
+++ sequoia-sq/Cargo.toml
@@ -171,7 +171,7 @@
 version = "0.3"
 
 [dependencies.gethostname]
-version = ">=0.4, <0.6"
+version = "0.4"
 
 [dependencies.humantime]
 version = "2"
@@ -186,7 +186,7 @@
 version = "1"
 
 [dependencies.reqwest]
-version = ">=0.11.27, <0.13"
+version = "0.11"
 features = [
     "hickory-dns",
     "stream",
@@ -213,7 +213,7 @@
 default-features = false
 
 [dependencies.sequoia-openpgp]
-version = "1.18"
+version = "1.21"
 features = ["compression"]
 default-features = false
 
@@ -235,7 +235,7 @@
 version = "1.2.0"
 
 [dependencies.terminal_size]
-version = ">=0.2.6, <0.5"
+version = "0.3"
 
 [dependencies.textwrap]
 version = ">=0.15, <0.17"
@@ -246,7 +246,7 @@
 default-features = false
 
 [dependencies.thiserror]
-version = ">=1, <3"
+version = "1.0"
 
 [dependencies.tokio]
 version = "1.13.1"
