Index: sequoia-sq/Cargo.toml
===================================================================
--- sequoia-sq.orig/Cargo.toml
+++ sequoia-sq/Cargo.toml
@@ -81,7 +81,7 @@ version = "1"
 version = "1.0.18"
 
 [dependencies.buffered-reader]
-version = "1.4"
+version = "1.3.1"
 features = ["compression"]
 default-features = false
 
@@ -118,7 +118,7 @@ version = "1"
 version = "0.3"
 
 [dependencies.gethostname]
-version = ">=0.4, <0.6"
+version = "0.4"
 
 [dependencies.humantime]
 version = "2"
@@ -133,7 +133,7 @@ version = "1.17"
 version = "1"
 
 [dependencies.reqwest]
-version = ">=0.12, <0.13"
+version = "0.12"
 features = [
     "hickory-dns",
     "stream",
@@ -143,7 +143,7 @@ features = [
 version = "7.0"
 
 [dependencies.rusqlite]
-version = ">=0.31, <0.33"
+version = "0.29"
 
 [dependencies.sequoia-autocrypt]
 version = "0.26"
@@ -190,10 +190,10 @@ version = "3.1"
 version = "1.2.0"
 
 [dependencies.terminal_size]
-version = ">=0.2.6, <0.5"
+version = "0.4"
 
 [dependencies.textwrap]
-version = ">=0.15, <0.17"
+version = "0.16"
 features = [
     "smawk",
     "unicode-width",
@@ -201,7 +201,7 @@ features = [
 default-features = false
 
 [dependencies.thiserror]
-version = ">=1, <3"
+version = "2"
 
 [dependencies.tokio]
 version = "1.13.1"
@@ -224,7 +224,7 @@ version = "2"
 version = "0.2"
 
 [dev-dependencies.predicates]
-version = ">=2, <4"
+version = "3"
 
 [dev-dependencies.regex]
 version = "1"
@@ -233,7 +233,7 @@ version = "1"
 version = "1.0.18"
 
 [build-dependencies.buffered-reader]
-version = "1.4"
+version = "1.3.1"
 features = ["compression"]
 default-features = false
 
@@ -285,10 +285,10 @@ version = "1.0.137"
 features = ["derive"]
 
 [build-dependencies.terminal_size]
-version = ">=0.2.6, <0.5"
+version = "0.4"
 
 [build-dependencies.textwrap]
-version = ">=0.15, <0.17"
+version = "0.16"
 
 [build-dependencies.typenum]
 version = "1"
