Index: sequoia-sq/Cargo.toml
===================================================================
--- sequoia-sq.orig/Cargo.toml
+++ sequoia-sq/Cargo.toml
@@ -195,12 +195,12 @@
 default-features = false
 
 [dependencies.sequoia-openpgp]
-version = "1.18"
+version = "1.21"
 features = ["compression"]
 default-features = false
 
 [dependencies.sequoia-policy-config]
-version = "0.6"
+version = ">=0.6, <=0.7"
 
 [dependencies.sequoia-wot]
 version = "0.12"
@@ -217,7 +217,7 @@
 version = "1.2.0"
 
 [dependencies.terminal_size]
-version = ">=0.2.6, <0.4"
+version = ">=0.2.6, <1"
 
 [dependencies.textwrap]
 version = ">=0.15, <0.17"
