Index: sequoia-sq/Cargo.toml
===================================================================
--- sequoia-sq.orig/Cargo.toml
+++ sequoia-sq/Cargo.toml
@@ -87,7 +87,7 @@ version = "2"
 version = "0.17"
 
 [dependencies.itertools]
-version = ">=0.10, <0.13"
+version = ">=0.10, <1"
 
 [dependencies.once_cell]
 version = "1.17"
@@ -113,7 +113,7 @@ version = "0.28"
 default-features = false
 
 [dependencies.sequoia-openpgp]
-version = "1.18"
+version = "1.21"
 features = ["compression"]
 default-features = false
 
@@ -138,7 +138,7 @@ version = "3.1"
 version = "1.2.0"
 
 [dependencies.terminal_size]
-version = ">=0.2.6, <0.4"
+version = ">=0.2.6, <1"
 
 [dependencies.textwrap]
 version = ">=0.15, <0.17"
