Index: rust-decimal/Cargo.toml
===================================================================
--- rust-decimal.orig/Cargo.toml
+++ rust-decimal/Cargo.toml
@@ -200,16 +200,8 @@ rkyv = ["dep:rkyv"]
 rkyv-safe = ["rkyv/validation"]
 rust-fuzz = ["dep:arbitrary"]
 serde = ["dep:serde"]
-serde-arbitrary-precision = ["serde-with-arbitrary-precision"]
 serde-bincode = ["serde-str"]
-serde-float = ["serde-with-float"]
 serde-str = ["serde-with-str"]
-serde-with-arbitrary-precision = [
-    "serde",
-    "serde_json/arbitrary_precision",
-    "serde_json/std",
-]
-serde-with-float = ["serde"]
 serde-with-str = ["serde"]
 std = [
     "arrayvec/std",
