Index: rust-decimal/Cargo.toml
===================================================================
--- rust-decimal.orig/Cargo.toml
+++ rust-decimal/Cargo.toml
@@ -193,12 +193,6 @@ version = "0.8"
 optional = true
 default-features = false
 
-[dependencies.rand-0_9]
-version = "0.9"
-optional = true
-default-features = false
-package = "rand"
-
 [disabled.dependencies.rkyv]
 version = "0.7.42"
 features = [
@@ -269,12 +263,6 @@ version = "0.8"
 features = ["getrandom"]
 default-features = false
 
-[dev-dependencies.rand-0_9]
-version = "0.9"
-features = ["thread_rng"]
-default-features = false
-package = "rand"
-
 [dev-dependencies.rkyv-0_8]
 version = "0.8"
 package = "rkyv"
