Index: rust-decimal/src/decimal.rs
===================================================================
--- rust-decimal.orig/src/decimal.rs
+++ rust-decimal/src/decimal.rs
@@ -179,7 +179,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::MIN, dec!(-79_228_162_514_264_337_593_543_950_335));
@@ -190,7 +190,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::MAX, dec!(79_228_162_514_264_337_593_543_950_335));
@@ -201,7 +201,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::ZERO, dec!(0));
@@ -212,7 +212,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::ONE, dec!(1));
@@ -223,7 +223,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::NEGATIVE_ONE, dec!(-1));
@@ -234,7 +234,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::TWO, dec!(2));
@@ -245,7 +245,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::TEN, dec!(10));
@@ -256,7 +256,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::ONE_HUNDRED, dec!(100));
@@ -267,7 +267,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::ONE_THOUSAND, dec!(1000));
@@ -287,7 +287,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::PI, dec!(3.1415926535897932384626433833));
@@ -304,7 +304,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::HALF_PI, dec!(1.5707963267948966192313216916));
@@ -321,7 +321,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::QUARTER_PI, dec!(0.7853981633974483096156608458));
@@ -338,7 +338,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::TWO_PI, dec!(6.2831853071795864769252867666));
@@ -355,7 +355,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::E, dec!(2.7182818284590452353602874714));
@@ -372,7 +372,7 @@ impl Decimal {
     /// # Examples
     ///
     /// Basic usage:
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// assert_eq!(Decimal::E_INVERSE, dec!(0.3678794411714423215955237702));
@@ -747,7 +747,7 @@ impl Decimal {
     ///
     /// Basic usage:
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::prelude::*;
     /// #
     /// # fn main() -> Result<(), rust_decimal::Error> {
@@ -983,7 +983,7 @@ impl Decimal {
     ///
     /// # Example
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::prelude::*;
     /// # use rust_decimal_macros::dec;
     ///
@@ -1428,7 +1428,7 @@ impl Decimal {
     ///
     /// # Example
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::{Decimal, RoundingStrategy};
     /// # use rust_decimal_macros::dec;
     /// #
@@ -1581,7 +1581,7 @@ impl Decimal {
     ///
     /// # Example
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     /// #
@@ -1611,7 +1611,7 @@ impl Decimal {
     ///
     /// # Example
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     ///
@@ -1653,7 +1653,7 @@ impl Decimal {
     ///
     /// # Example
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::{Decimal, RoundingStrategy};
     /// # use rust_decimal_macros::dec;
     ///
@@ -1765,7 +1765,7 @@ impl Decimal {
     ///
     /// # Example
     ///
-    /// ```
+    /// ```ignore
     /// # use rust_decimal::Decimal;
     /// # use rust_decimal_macros::dec;
     ///
