Index: rust-decimal/Cargo.toml
===================================================================
--- rust-decimal.orig/Cargo.toml
+++ rust-decimal/Cargo.toml
@@ -86,12 +86,6 @@ version = "1.0"
 optional = true
 default-features = false
 
-
-[dependencies.ndarray]
-version = "0.15.6"
-optional = true
-default-features = false
-
 [dependencies.num-traits]
 version = "0.2"
 features = ["i128"]
@@ -193,7 +187,6 @@ default = [
 legacy-ops = []
 maths = []
 maths-nopanic = ["maths"]
-ndarray = ["dep:ndarray"]
 proptest = ["dep:proptest"]
 rand = ["dep:rand"]
 rkyv = ["dep:rkyv"]
