--- a/Cargo.toml
+++ b/Cargo.toml
@@ -219,4 +219,4 @@
-borsh = [
-    "dep:borsh",
-    "std",
-]
+#borsh = [
+#    "dep:borsh",
+#    "std",
+#]
@@ -224,20 +224,20 @@
-db-diesel-mysql = [
-    "diesel/mysql_backend",
-    "std",
-]
-db-diesel-postgres = [
-    "diesel/postgres_backend",
-    "std",
-]
-db-diesel2-mysql = ["db-diesel-mysql"]
-db-diesel2-postgres = ["db-diesel-postgres"]
-db-postgres = [
-    "dep:bytes",
-    "dep:postgres-types",
-    "std",
-]
-db-tokio-postgres = [
-    "dep:bytes",
-    "dep:postgres-types",
-    "std",
-]
+#db-diesel-mysql = [
+#    "diesel/mysql_backend",
+#    "std",
+#]
+#db-diesel-postgres = [
+#    "diesel/postgres_backend",
+#    "std",
+#]
+#db-diesel2-mysql = ["db-diesel-mysql"]
+#db-diesel2-postgres = ["db-diesel-postgres"]
+#db-postgres = [
+#    "dep:bytes",
+#    "dep:postgres-types",
+#    "std",
+#]
+#db-tokio-postgres = [
+#    "dep:bytes",
+#    "dep:postgres-types",
+#    "std",
+#]
@@ -256,4 +256,4 @@
-rocket-traits = [
-    "dep:rocket",
-    "std",
-]
+#rocket-traits = [
+#    "dep:rocket",
+#    "std",
+#]
@@ -272 +272 @@
-    "borsh?/std",
+    #"borsh?/std",
@@ -279 +279 @@
-tokio-pg = ["db-tokio-postgres"]
+#tokio-pg = ["db-tokio-postgres"]
@@ -290 +290 @@
-[dependencies.borsh]
+[disabled.dependencies.borsh]
@@ -300 +300 @@
-[dependencies.diesel]
+[disabled.dependencies.diesel]
@@ -310 +310 @@
-[dependencies.rocket]
+[disabled.dependencies.rocket]
@@ -320 +320 @@
-[dependencies.tokio-postgres]
+[disabled.dependencies.tokio-postgres]
@@ -330 +330 @@
-[dev-dependencies.futures]
+[disabled.dev-dependencies.futures]
@@ -340 +340 @@
-[dev-dependencies.tokio]
+[disabled.dev-dependencies.tokio]
