Index: rust-decimal/Cargo.toml
===================================================================
--- rust-decimal.orig/Cargo.toml
+++ rust-decimal/Cargo.toml
@@ -79,7 +79,6 @@ default = [
     "std",
 ]
 legacy-ops = []
-macros = ["dep:rust_decimal_macros"]
 maths = []
 maths-nopanic = ["maths"]
 #ndarray = ["dep:ndarray"]
@@ -119,11 +118,6 @@ std = [
 name = "rust_decimal"
 path = "src/lib.rs"
 
-[[example]]
-name = "rkyv-remote"
-path = "examples/rkyv-remote.rs"
-required-features = ["macros"]
-
 [[test]]
 name = "decimal_tests"
 path = "tests/decimal_tests.rs"
@@ -219,7 +213,7 @@ version = "0.5.0-rc.3"
 optional = true
 default-features = false
 
-[dependencies.rust_decimal_macros]
+[disabled.dependencies.rust_decimal_macros]
 version = "1"
 optional = true
 default-features = false
