Index: rust-decimal/Cargo.toml
===================================================================
--- rust-decimal.orig/Cargo.toml
+++ rust-decimal/Cargo.toml
@@ -153,10 +153,6 @@ version = "0.8"
 features = ["getrandom"]
 default-features = false
 
-[dev-dependencies.rust_decimal_macros]
-version = "1.33"
-default-features = false
-
 [dev-dependencies.serde]
 version = "1.0"
 features = ["derive"]
