--- a/Cargo.toml
+++ b/Cargo.toml
@@ -134,3 +134,2 @@ default = [
     "bracketed-paste",
-    "windows",
     "events",
@@ -154,6 +153,2 @@ use-dev-tty = [
 ]
-windows = [
-    "dep:winapi",
-    "dep:crossterm_winapi",
-]
 
@@ -191,12 +186 @@ optional = true
 
-[target."cfg(windows)".dependencies.crossterm_winapi]
-version = "0.9.1"
-optional = true
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.9"
-features = [
-    "winuser",
-    "winerror",
-]
-optional = true
