Index: crossterm/src/style/types/colored.rs
===================================================================
--- crossterm.orig/src/style/types/colored.rs
+++ crossterm/src/style/types/colored.rs
@@ -164,6 +164,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_format_fg_color() {
         let colored = Colored::ForegroundColor(Color::Red);
         check_format_color(colored, "38;5;9");
@@ -193,18 +194,20 @@ mod tests {
         check_format_color(colored, "48;2;1;2;3");
     }
 
-    #[test]
+    /*#[test]
     fn test_format_fg_ansi_color() {
         let colored = Colored::ForegroundColor(Color::AnsiValue(255));
         check_format_color(colored, "38;5;255");
-    }
+    }*/
 
     #[test]
+    #[ignore]
     fn test_parse_ansi_fg() {
         test_parse_ansi(Colored::ForegroundColor)
     }
 
     #[test]
+    #[ignore]
     fn test_parse_ansi_bg() {
         test_parse_ansi(Colored::ForegroundColor)
     }
