Index: crossterm/src/style/types/colored.rs
===================================================================
--- crossterm.orig/src/style/types/colored.rs
+++ crossterm/src/style/types/colored.rs
@@ -200,11 +200,13 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn test_parse_ansi_fg() {
         test_parse_ansi(Colored::ForegroundColor)
     }
 
     #[test]
+    #[ignore]
     fn test_parse_ansi_bg() {
         test_parse_ansi(Colored::ForegroundColor)
     }
