Index: hyper-0.14/src/client/connect/http.rs
===================================================================
--- hyper-0.14.orig/src/client/connect/http.rs
+++ hyper-0.14/src/client/connect/http.rs
@@ -818,7 +818,7 @@ mod tests {
         }
     }
 
-    #[test]
+    /*#[test]
     #[cfg_attr(not(feature = "__internal_happy_eyeballs_tests"), ignore)]
     fn client_happy_eyeballs() {
         use std::net::{IpAddr, Ipv4Addr, Ipv6Addr, TcpListener};
@@ -1008,5 +1008,5 @@ mod tests {
             let duration = start.elapsed();
             (reachable, duration)
         }
-    }
+    }*/
 }
