--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,3 +56,3 @@
 
-[dependencies.wasm-gen]
+[disabled.dependencies.wasm-gen]
 version = "0.1.4"
@@ -61,3 +61,3 @@
 version = "0.9.0"
-features = ["wasm"]
+#features = ["wasm"]
 
--- a/src/binary_file.rs
+++ b/src/binary_file.rs
@@ -36,5 +36,5 @@
     // https://github.com/WebAssembly/tool-conventions/blob/master/Linking.md
-    wasm_gen::write_custom_section(&mut result, "linking", &[2]);
+    //wasm_gen::write_custom_section(&mut result, "linking", &[2]);
 
-    wasm_gen::write_custom_section(&mut result, ".dep-v0", contents);
+    //wasm_gen::write_custom_section(&mut result, ".dep-v0", contents);
     result
