diff --git a/src/object_file.rs b/src/object_file.rs
index 5fed86c..16375e0 100644
--- a/src/object_file.rs
+++ b/src/object_file.rs
@@ -62,7 +62,7 @@ fn create_object_file(
         "arm" => Architecture::Arm,
         "aarch64" => {
             if info["target_pointer_width"].as_str() == "32" {
-                Architecture::Aarch64_Ilp32
+                return None
             } else {
                 Architecture::Aarch64
             }
