Index: prost-reflect/src/dynamic/mod.rs
===================================================================
--- prost-reflect.orig/src/dynamic/mod.rs
+++ prost-reflect/src/dynamic/mod.rs
@@ -1196,8 +1196,9 @@ impl fmt::Display for Value {
 
 #[test]
 fn type_sizes() {
+    let ptr_width = usize::BITS as usize;
-    assert_eq!(std::mem::size_of::<DynamicMessage>(), 40);
-    assert_eq!(std::mem::size_of::<Value>(), 56);
+    assert_eq!(std::mem::size_of::<DynamicMessage>() * 8, 5 * ptr_width);
+    assert_eq!(std::mem::size_of::<Value>() * 8, 5 * ptr_width + 128);
 }
 
 pub(crate) enum Either<L, R> {
