From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Wed, 21 Aug 2024 17:53:43 -0400
Subject: relax deps

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 6d08b44..d4d6fb5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,7 +30,7 @@ version = "1.1.0"
 version = "0.2"
 
 [build-dependencies.system-deps]
-version = "6.0"
+version = "7"
 
 [features]
 v1_1 = []
