--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,12 +34,2 @@
 
-[[example]]
-name = "simple_proxy"
-path = "examples/simple_proxy.rs"
-required-features = ["proxy"]
-
-[[example]]
-name = "simple_plaintext"
-path = "examples/simple_plaintext.rs"
-required-features = ["tls-native"]
-
 [dependencies.chrono]
@@ -131,3 +124,3 @@
 [dependencies.toml]
-version = "0.7.3"
+version = "0.8"
 optional = true
@@ -141,8 +134,2 @@
 
-[dev-dependencies.args]
-version = "2.2.0"
-
-[dev-dependencies.env_logger]
-version = "0.11.0"
-
 [dev-dependencies.futures]
@@ -150,5 +137,2 @@
 
-[dev-dependencies.getopts]
-version = "0.2.21"
-
 [dev-dependencies.tokio]
