Index: irc/Cargo.toml
===================================================================
--- irc.orig/Cargo.toml
+++ irc/Cargo.toml
@@ -69,7 +69,7 @@ version = "0.12.1"
 version = "1.0.12"
 
 [dependencies.rustls-native-certs]
-version = "0.8"
+version = "0.6"
 optional = true
 
 [dependencies.rustls-pemfile]
Index: irc/src/client/conn.rs
===================================================================
--- irc.orig/src/client/conn.rs
+++ irc/src/client/conn.rs
@@ -346,9 +346,10 @@ impl Connection {
             #[cfg(feature = "webpki-roots")]
             root_store.extend(webpki_roots::TLS_SERVER_ROOTS.iter().cloned());
 
-            let native_certs = rustls_native_certs::load_native_certs();
-            for cert in native_certs.certs {
-                root_store.add(cert.into())?;
+            if let Ok(native_certs) = rustls_native_certs::load_native_certs() {
+              for cert in native_certs {
+                root_store.add(cert.0.into())?;
+              }
             }
 
             if let Some(cert_path) = config.cert_path() {
