diff --git a/Cargo.toml b/Cargo.toml
index 8605b90..dbfac22 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -63,2 +63,0 @@ serde = [
-[target."cfg(windows)".dependencies.gix-path]
-version = "^0.10.7"
@@ -70,9 +70,0 @@ serde = [
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.52.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Security_Authorization",
-    "Win32_Storage_FileSystem",
-    "Win32_System_Memory",
-    "Win32_System_Threading",
-]
