--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,7 +26,7 @@ resolver = "2"
 [lib]
 proc-macro = true
 [dependencies.proc-macro-crate]
-version = "1.0.0"
+version = "0.1.4"
 
 [dependencies.proc-macro2]
 version = "1.0"
--- a/src/utils.rs
+++ b/src/utils.rs
@@ -1,10 +1,10 @@
 use proc_macro2::TokenStream;
-use proc_macro_crate::{crate_name, FoundCrate};
+use proc_macro_crate::crate_name;
 use quote::{format_ident, quote};
 use syn::{Attribute, Lit, Meta, Meta::List, MetaList, NestedMeta, Result};
 
 pub fn zvariant_path() -> TokenStream {
-    if let Ok(FoundCrate::Name(name)) = crate_name("zvariant") {
+    if let Ok(name) = crate_name("zvariant") {
         let ident = format_ident!("{}", name);
         quote! { ::#ident }
     } else {
